---
title: "Good_Illumina"
output: html_document
---



# Set up
## Knitr
```{r Knitrsetup, cache=TRUE,warning=FALSE,message=FALSE,out.width="100%"}
knitr::opts_chunk$set(#eval = TRUE, 
                      #echo = TRUE, 
                      cache = FALSE,
                      #include = FALSE,
                      collapse = FALSE,
                      dependson = NULL,
                      warning = FALSE,
                      engine = "R", # Chunks will always have R code, unless noted
                      error = TRUE,
                      fig.align = "center",
                      cache.lazy = FALSE,
                      tidy.opts=list(width.cutoff=60),
                      tidy=TRUE
                      )
```


## Load the packages
```{r load packages, message=FALSE, warning=FALSE, results='hide'}
setwd("/Projects2/Yorick/Rstudio_projects/Task1.2TransitSHIME/TransitSHIME")
# Function to check whether package is installed
  is.installed <- function(package){
    is.element(package, installed.packages()[,1])
  } 
  
# loading in data
# library(rJava)
library(remotes, lib.loc = "/usr/local/lib/R/site-library")
# library(devtools)
library(readxl)  #to read excel files
  
# plotting data
library(ggplot2)
# library(ggExtra) not installed # Add marginal histograms to ggplot2, and more ggplot2 enhancements
library(ggfortify) # ordination plots
library(ggrepel)
library(cowplot)
library(gganimate)
library(grid) 
library(gridExtra) # Extra pretty grids
library(circlize) # Pretty colours
library(pheatmap) # Pretty heatmaps
library(ComplexHeatmap) # Even prettier heatmaps
  
# Statistics
# install_github("pmartinezarbizu/pairwiseAdonis/pairwiseAdonis")
library(pairwiseAdonis) # Post-hoc of PerMANOVA
library(ggpubr) # Add statistics to ggplots
library(rstatix)
  
# formatting data
library(reshape2)
library(plyr)
library(gtable)
library(grid)
library(plyr) 
library(dplyr)
library(tidyr)
library(tidyverse)
library(data.table) # Useful to change ggplots!
library(cluster)
library(gtools)
library(gdata) # Various R Programming Tools for Data Manipulation
library(multcompView)
library(lemon) 
  
# formatting sequencing data
library(vegan) # Community Ecology Package
library(phyloseq)
library(microbiome) # To further manipulate phyloseq data
library(ape) # Analyses of Phylogenetics and Evolution
# library(MicEco)# Various functions for microbial community data, incl. copy number corrections
# library(DESeq2)
library(rBLAST) # Interface for BLAST search
library(Biostrings)
  
# Other scripts
library(scales)
library(stringr)
library(RColorBrewer)
library(openxlsx)
# library(plotly)
# library(leaflet)
library(gganimate)
library(gtable)
library(stringi)
# remotes::install_github("thomasp85/transformr")
# library(transformr)
library(officer) # to create ppt
  
# Do stuff in parallel with multiple cores
library(foreach)
library(doParallel)
corenumber = 10

  
  
  
# Own scripts
source('YM_KDP_papertheme.R')
source('YM_changetaxa.R') # Change taxa that don't correspond with RDP database
source("KDP_tax_summarize.R")
source("KDP_volcano.R")
source("KDP_splittaxfunction.R")
source("YM_aggregate_top_OTU.R")
source("YM_fusetoplayer.R")
source("YM_extrafunctions.R")
# devtools::install_github("teunbrand/ggh4x")


# citation(package = "base") # To know what to cite
version #Current R version
citation(package = "igraph")
sessionInfo() #To know all package versions

# Make supplementary figures in grey values for cheaper printing. For paper make = false
BW = TRUE
```

## Colours and labels
```{r labels and colours, message=FALSE, warning=FALSE}
# Universal colours for the project
source("YM_colours.R")
# Labels
donor.labs <- c("Donor 1", "Donor 2", "Donor 3", "Donor 4", "Donor 5", "Donor 6")
names(donor.labs) <- c("1", "2", "3", "4","5","6")
PD_Colon.labs <- c("Faecal sample","Proximal colon", "Distal colon")
names(PD_Colon.labs) <- c("FS","P", "D")
PD_Colon.labsPC <- c("Faecal sample","Proximal colon", "Distal colon")
names(PD_Colon.labsPC) <- c("FS","P", "D")
# Labels
VFA.labs <- c("Acetate", "Propionate", "Butyrate", "Total")
names(VFA.labs) <- c("Acetate", "Propionate", "Butyrate", "total_SCFA")
donor.labs <- c("Donor 1 - Short", "Donor 2 - Short", "Donor 3 - Medium", "Donor 4 - Long", "Donor 5 - Long", "Donor 6 - Medium")
names(donor.labs) <- c("1", "2", "3", "4","5","6")
Vessel.labs <- c("Faecal sample","Proximal colon", "Distal colon")
names(Vessel.labs) <- c("FS","P", "D")
Transit.labs <- c("Short transit", "Medium transit", "Long transit")
names(Transit.labs) <- c("ST", "MT", "LT")
donor.labs <- c("Donor 1", "Donor 2", "Donor 3", "Donor 4", "Donor 5", "Donor 6")
names(donor.labs) <- c("1", "2", "3", "4","5","6")
saveplots <- function () setwd("/Projects2/Yorick/Rstudio_projects/Task1.2TransitSHIME/Data/Results/Plots") #save plots
RDP <- T
otu <- FALSE
```


```{r colour quest, fig.height=10, fig.width=16, include=FALSE}
library(ggsci)
library(scico)
# ggsci
getPalette = colorRampPalette(pal_npg("nrc", alpha = 0.7)(10)); show_col(getPalette(20));getPalette(20)
getPalette = colorRampPalette(pal_startrek("uniform", alpha = 0.7)(7)); show_col(getPalette(26));getPalette(26)
getPalette = colorRampPalette(pal_rickandmorty("schwifty", alpha = 0.7)(12)); show_col(getPalette(26));getPalette(26)
getPalette = colorRampPalette(pal_tron("legacy", alpha = 0.7)(7)); show_col(getPalette(7));getPalette(7)
colors1 <- c("#FAFD7C", "#C5AD52", "#905E29", "#634132", "#3A374F", "#41557D",
              "#82A3C1", "#BCD9ED", "#DAA1AD", "#F8696C", "#B7684F", "#6D6C36", 
              "#7B6A5C","#BD65A7", "#E769BF", "#E77E7D", "#E89544", "#F0B965", 
              "#F8DD85", "#DBE7AB", "#B6ECD3", "#A0D2C5",  "#8C856E", "#7AA6AD",
               "#69C8EC","#97A088"); show_col(colors1)
getPalette = colorRampPalette(brewer.pal(11, "Paired")); show_col(getPalette(26));getPalette(26)
getPalette = colorRampPalette(brewer.pal(9, "Pastel1")); show_col(getPalette(26));getPalette(26)
getPalette = colorRampPalette(brewer.pal(9, "Blues")); show_col(getPalette(26));getPalette(26)
x <- getPalette(26) 
  colors <- c("#7FC97F", "#66A61E","#1B9E77", "#6D6D36", "#B2DF8A", "#FDC086", 
              "#1F78B4", "#A6CEE3","#7570B3",  "#3383BE","#BF5B17",  "#A6761D",
              "#E7298A", "#E31A1C",  "#B12934",  "#F0027F",  "#CC0C00","#BEAED4",
               "#FB9A99",  "#FDBF6F", "#FF7F00", "#CAB2D6", "#FFFF99","#E6AB02",
              "#6A3D9A","#666666"); show_col(colors)
  
  colors1 <- c("#DBE7AB","#7FC97F", "#66A61E","#1B9E77", "#BD65A7", "#E769BF", 
               "#B6ECD3","#BCD9ED", "#82A3C1","#69C8EC", "#41557D","#3A374F",  
               "#DAA1AD","#E77E7D", "#F8696C", "#B7684F", "#CC0C00", "#7D54A6",
               "#C5AD52", "#A383BE","#E89544", "#F0B965", "#634132","#FAFD7C", 
               "#97A088", "#7B6A5C"); show_col(colors1)
  
   desired = c("Acidaminococcus"="#DA5724",  "Akkermansia"="#94cfd5", "Alistipes"="#DAA1AD",
  "Bacteroides"="#3F4921", "Bifidobacterium"="#CBD588", "Bilophila"="#5F7FC7", "Blautia"="#673770", 
  "Cloacibacillus" = "#D7C1B1", "Dialister"= "#FAFD7C","Enterobacteriaceae"="#38333E", 
  "Clostridium_XlVa" = "#689030", "Eisenbergiella" = "#508578", "Lachnospiraceae"="#CD9BCD",
  "Megamonas" ="#D14285","Megasphaera"= "#6DDE88", "Mitsuokella" ="#652926", "Parabacteroides"="#7FDCC0",
  "Phascolarctobacterium"="#5E738F", "Prevotella"= "#C84248", "Pseudomonas"="#D1A33D", "Roseburia"="#8A7C64", 
  "Ruminococcus" = "#AD6F3B", "Ruminococcaceae" = "#CC0C00", "Sutterella" = "#046582", "Veillonella" = "#F8698C", 
  "Other"="#bfbfbf")
   
   colors = 21
  par(mfrow=c(2,2));show_col(desired); show_col(desired2); show_col(colors1); 
  getPalette = colorRampPalette(pal_rickandmorty("schwifty", alpha = 0.7)(12)); show_col(getPalette(colors));getPalette(colors)
  getPalette = colorRampPalette(brewer.pal(9, "Pastel2")); show_col(getPalette(colors));getPalette(colors)
getPalette = colorRampPalette(brewer.pal(11, "Paired")); show_col(getPalette(colors));getPalette(colors)
# getPalette = colorRampPalette(pal_npg("nrc", alpha = 0.7)(10)); show_col(getPalette(26));getPalette(26)
# scico_palette_show()
show_col(scico(26, palette = 'lapaz'))
qual_col_pals <- brewer.pal.info[brewer.pal.info$category == "qual", ]
col_vector <- unlist(mapply(brewer.pal, qual_col_pals$maxcolors, rownames(qual_col_pals)))
```


# DNA database matches
## BLAST FASTA file
```{r}
library(rBLAST)
library(Biostrings)
setwd("/media/projects2/yorick/Rstudio_projects/Task1.2_Transittime/Data/BLAST")
new_db = F
## download the 16S Microbial data base from NCBI
if(new_db == T){ 
download.file("https://ftp.ncbi.nlm.nih.gov/blast/db/16S_ribosomal_RNA.tar.gz", "16S_ribosomal_RNA.tar.gz")
untar("16S_ribosomal_RNA.tar.gz", exdir="/media/projects2/yorick/Rstudio_projects/Task1.2_Transittime/Data/BLAST/16SMicrobialDB")
bl <- blast(db="/media/projects2/yorick/Rstudio_projects/Task1.2_Transittime/Data/BLAST/16SMicrobialDB/16S_ribosomal_RNA", type = "blastn")
}else{
# Old data base
setwd("/media/projects2/yorick/Rstudio_projects/Task1.2_Transittime/Data/BLAST")
download.file("https://ftp.ncbi.nlm.nih.gov/blast/db/v4/16SMicrobial_v4.tar.gz", "16SMicrobial_v4.tar.gz", mode='w')
untar("16SMicrobial_v4.tar.gz", exdir="/media/projects2/yorick/Rstudio_projects/Task1.2_Transittime/Data/BLAST/16SMicrobialDB")
bl <- blast(db="/media/projects2/yorick/Rstudio_projects/Task1.2_Transittime/Data/BLAST/16SMicrobialDB/16SMicrobial", type = "blastn")
}
## load test data 
setwd("/media/projects2/yorick/Rstudio_projects/Task1.2_Transittime/Data/BLAST")
seq <- readRNAStringSet(system.file("OTU_fasta_seqmatch.fasta",	package="rBLAST"))
seq
names(seq)
# ## load a BLAST database (replace db with the location + name of the BLAST DB)
print(bl, info=TRUE)
## query a sequence using BLAST
cl <- predict(bl, seq[3,])
cl[1:5,]
```

## Interprete RDP seqmatch results
```{r RDP seqmatch}
library(stringr)
RDPseqmatch <- read.csv2("RDPseqmatch_results.csv", header = TRUE, sep = ";")
# Split sequence name and remove (T) from the string
RDPseqmatch[c('Species', 'code1', "code2")] <- str_split_fixed(RDPseqmatch$sequence.name, ';', 3)
RDPseqmatch$Species <- str_remove(RDPseqmatch$Species, "(T)")
RDPseqmatch$Species <- gsub( "(", "",RDPseqmatch$Species, fixed = TRUE)
RDPseqmatch$Species <- gsub( ")", "",RDPseqmatch$Species, fixed = TRUE)
RDPseqmatch$OTU <- str_remove(RDPseqmatch$query.name, "OTU"); RDPseqmatch$OTU <- as.numeric(RDPseqmatch$OTU)
# Rename columns
colnames(RDPseqmatch)[colnames(RDPseqmatch) == "X_1"] <- "Domain" 
colnames(RDPseqmatch)[colnames(RDPseqmatch) == "X_3"] <- "Phylum" 
colnames(RDPseqmatch)[colnames(RDPseqmatch) == "X_5"] <- "Class"
colnames(RDPseqmatch)[colnames(RDPseqmatch) == "X_7"] <- "Order" 
colnames(RDPseqmatch)[colnames(RDPseqmatch) == "X_9"] <- "Family" 
colnames(RDPseqmatch)[colnames(RDPseqmatch) == "X_11"] <- "Genus" 
# Select cols you want
RDPseqmatch <- subset(RDPseqmatch, select = c("OTU","Species", "S_ab.score", "unique.common.oligomers","Domain",
                                    "Phylum", "Class", "Order", "Family", "Genus",  "code1", "code2", "query.name"))
# Remove NA
RDPseqmatch <- na.omit(RDPseqmatch) 
# Take top x Species level sequences
uniqueOTU <- sort(unique(RDPseqmatch$OTU)); top = 2
RDPcondensed <- data.frame()
for(i in uniqueOTU){
  df <- subset(RDPseqmatch, OTU == i); df3 <- data.frame()
  for(x in c(1:top)){
    max = max(df$S_ab.score)
    df2 <- subset(df, S_ab.score == max) 
    df3 <- rbind(df3, df2)
    df <- subset(df, S_ab.score != max)
  }
  df3 <- df3[!duplicated(df3[c('Species')]), ]
  RDPcondensed <- rbind(RDPcondensed,df3)
}
RDPcondensed$S_ab.score <- RDPcondensed$S_ab.score/1000
RDPcondensed$combo <- paste0(RDPcondensed$OTU,RDPcondensed$Species)
# write.csv2(RDPcondensed, "RDPseqmatch_condensed_all.csv")
write.csv2(RDPcondensed, "RDPseqmatch_condensed.csv")
```



# RUN ALL FIRST data manipulation 
## Load data and create phyloseq object
```{r Create and manipulate phyloseq objects, fig.height=20, fig.width=40, message=F}
# Load source data 
if(RDP == TRUE) {
  otutaxonomy <- read.table("fasta_for_RDPclassifier.wang.taxonomy")
  # herschikken van RDP data <-- gekopieerde splittax2 functie
    x <- otutaxonomy; y <- stri_split_regex(x$V2,"\\(|\\;", simplify = TRUE)
    # add the first column with the id's of the sample to the dateframe with the taxonomy
    x <- cbind(x[,1],as.data.frame(y))
    # remove the empty columns
    x <- x[,colSums(x != "") != 0] 
    # silva only goes till genus level so we need to add empty columns in that case, since it will try to give names in 
    # next step and name vector will be to long if not adapted
    if (ncol(x)==13){x <- mutate(x,x14="NA",x15="NA")}
    # add column names
    colnames(x) <-c("id", "Kingdom", "prob_k", "Phylum", "prob_p", "Class", "prob_c", "Order",
                    "prob_o","Family","prob_f","Genus","prob_g","Species","prob_s")
    taxtable <- x; taxtable$ID2 <- taxtable$id; taxtable$ID2<- gsub("OTU", "", taxtable$ID2) 
    taxtable2 <- subset(taxtable, ID2 == ""); taxtable2$ID2 <- c(32,33,34,35,36,37,38)
    taxtable <- subset(taxtable, ID2 != ""); taxtable <- rbind(taxtable2, taxtable)
    taxtable$ID2 <- as.numeric(taxtable$ID2); test <- taxtable[duplicated(taxtable$ID2),]
    taxtable <- arrange(taxtable,ID2)
    taxtable$ID2 <- str_pad(taxtable$ID2, 5, pad = "0")
    taxtable$ID2 <- paste0("Otu", taxtable$ID2, sep = "")  # Add zeros in front Otu number
    row.names(taxtable) <- taxtable$ID2
    taxtable <- subset(taxtable, select = c(Kingdom,Phylum,Class,Order,Family,Genus)); rm(x, y)
} else {
  taxfile = "stabilityfile.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.0.03.cons.taxonomy"
  test <- as.data.frame(tax_table(import_mothur(mothur_constaxonomy_file = taxfile)))
  }
sharedfile = "stabilityfile.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.shared"

# Load sampledata and alter it
sampledata <- read.csv2("Sampledata_Illu.csv", header = TRUE, sep = ",")
sampledata <- na.omit(sampledata)
sampledata$Donor[sampledata$Donor == 4] <- 7; sampledata$Donor[sampledata$Donor == 6] <- 4
sampledata$Donor[sampledata$Donor == 7] <- 6; 
sampledata$Transit <- as.ordered(factor(sampledata$Transit,
                                          levels=c("FS","ST","MT", "LT")))
sampledata$PD_Colon <- as.ordered(factor(sampledata$PD_Colon, levels=c("P", "D", "FS")))
sampledata$Donor <- as.ordered(factor(sampledata$Donor, levels=c("1","2","3","4","5","6")))
sampledata <- cbind(sampledata, `Interaction`= interaction(sampledata$Donor, sampledata$Transit, sep=":"))
sampledata$Day <- as.numeric(sampledata$Day)

 ## Merge fcm counts with sampledata
if(exists("by_nr") == F){by_nr <- read.csv("cellcounts.csv")}
 Countdata <- by_nr
 sampledata <-  merge(sampledata, by_nr[, c("Nr", "SDtot", "meantot")], by.x = "Samplenumber",by.y="Nr", all.x=TRUE)
 
 ## Assign rownames to be Sample ID's
 row.names(sampledata) <- sampledata$Samplenumber
 
# Import mothur data and
if(RDP == TRUE){
  mothur_data <- import_mothur(mothur_shared_file = sharedfile
                             # mothur_constaxonomy_file = taxfile
                             )
  taxtable <- tax_table(as.matrix(taxtable))
  mothur_data <- merge_phyloseq(mothur_data, taxtable)
  
} else{
  mothur_data <- import_mothur(mothur_shared_file = sharedfile,
                             mothur_constaxonomy_file = taxfile)
  xxx <- as.data.frame(tax_table(mothur_data))
  # Rename the columns to reflect tax
  colnames(tax_table(mothur_data)) <- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus")
}
# Check if the samples are correct 
sample_names(mothur_data); x <- as.data.frame(sample_names(mothur_data))
# Rename samplenames of long transit in donors 3 and 6: 279-288 -> 281 - 290 and 379-388 -> 381 - 390
 txx <- as.data.frame(phyloseq::otu_table(mothur_data), stringsAsFactors = F)
 colnames(txx)[colnames(txx) %in% c("279", "280", "281", "282", "283", "284", "285", "286", "287", "288",
                                    "379", "380", "381", "382", "383", "384", "385", "386", "387", 
                                    "388")]  <- c("281", "282", "283", "284", "285", "286", "287", "288", 
                       "289", "290", "381", "382",	"383", "384", "385", "386", "387", "388", "389", "390")
 otu_table(mothur_data) <- otu_table(as.matrix(txx), taxa_are_rows = TRUE)
 sample_names(mothur_data)

# Merge mothurdata object with sample metadata
moth_merge <- merge_phyloseq(mothur_data,sample_data(sampledata))
phylobj.mothur <- moth_merge

# Singleton and waste not, want not corrections
moth_merge_NS <- prune_taxa(taxa_sums(moth_merge) > 1, moth_merge) 

 ## Sample removal (sample 121 remove, 1 read - see QC)
 moth_merge_NSx <- subset_samples(moth_merge_NS, Samplenumber !="121")
 df.OTU_NS <- as.data.frame(otu_table(moth_merge_NSx)); df.Tax_NS <- as.data.frame(tax_table(moth_merge_NSx))
 
 ## If no removal run code line below:
 # df.OTU_NS <- as.data.frame(otu_table(moth_merge_NS)); df.Tax_NS <- as.data.frame(tax_table(moth_merge_NS))
 
## Apply 'arbitrary cut-offs' from the waste not want not paper
 minobs = 1
 prevalence <- apply(as.matrix(df.OTU_NS), 1, 
                    function (x, minobs){sum(x>=minobs)}, minobs)/ncol(df.OTU_NS)
 prevalencefilter <- prevalence>0.05
 df.OTUwnwn <- df.OTU_NS[prevalencefilter,]
 df.OTU_NSwnwn <- df.OTUwnwn[rowSums(df.OTUwnwn)>0.5*ncol(df.OTUwnwn),]
 df.OTU_NS_wnwn <- as.data.frame(t(df.OTU_NSwnwn))# Transpose dataframe
 df.OTU_NS_wnwn[] <- lapply(df.OTU_NS_wnwn, as.numeric)
 colnames(df.OTU_NS_wnwn) <- rownames(df.OTU_NSwnwn)
 rownames(df.OTU_NS_wnwn) <- colnames(df.OTU_NSwnwn)
 moth_merge_NS_wnwn <- moth_merge_NS
 otu_table(moth_merge_NS_wnwn) <- otu_table(as.matrix(df.OTU_NSwnwn), taxa_are_rows = TRUE)

# Create phylobjects
# ON OTU LEVEL
phylobj <- moth_merge_NS_wnwn
phylobj.otu <- phylobj
OTU1 = data.frame(otu_table(phylobj.otu)); tax.tab = data.frame(tax_table(phylobj.otu))
OTU1$Genus = tax.tab$Genus; otu.tab <- OTU1 %>% select(Genus, everything())
otu.tab$order <- 1:nrow(otu.tab); otu.tab <- otu.tab %>% select(order, everything())

# Copy number correction 
## Read in the RDP files containing information for copy number correction and format files  
RDPclass_QMP <- read.csv2("fasta_for_RDPclassifier.fastahier.csv")
RDPclass_QMP_copynrcorrected <- read.csv2("fasta_for_RDPclassifier.fastahier_COPYNRCORRECTED.csv", as.is = TRUE)
RDPclass_QMP$Seqnr <- as.numeric(RDPclass_QMP$Seqnr)
RDPclass_QMP_copynrcorrected$Seqnr_corrected <- as.numeric(RDPclass_QMP_copynrcorrected$Seqnr_corrected)
RDPclass_QMP_merge <- cbind(RDPclass_QMP,RDPclass_QMP_copynrcorrected)
RDPclass_QMP_merge <- mutate(RDPclass_QMP_merge,copynr=Seqnr/Seqnr_corrected)
RDPclass_QMP_merge$copynr[is.na(RDPclass_QMP_merge$copynr)] <- 1 # if the copy number is unknown take copy number 1 
RDPclass_QMP_merge <- RDPclass_QMP_merge[,-c(3:6)]

## Run own script to change non corresponding taxa from RDP database
otu.tab <- Change_taxaOTU(OTU.df = otu.tab, RDP = TRUE)
RDPclass_QMP_merge <- Change_taxaRDP(RDP.df = RDPclass_QMP_merge, RDP = TRUE)

## Combine the copy number correction factor with the genus table 
otu.tab$Genus%in%RDPclass_QMP_merge$Taxa; test <- cbind(data.frame(x = otu.tab$Genus%in%RDPclass_QMP_merge$Taxa),otu.tab)
if("FALSE" %in% test$x){stop("Genera in OTU table are not matching RDP set!")}; rm(test)
copynr_genus <- RDPclass_QMP_merge[match(otu.tab$Genus,RDPclass_QMP_merge$Taxa),]$copynr
otu.tab2 <- subset(otu.tab, select = -c(Genus, order))
genusotutable_copynr_corrected <- sweep(otu.tab2,1,copynr_genus,'/')
colnames(genusotutable_copynr_corrected) <- sub("X", "", colnames(genusotutable_copynr_corrected))
genusotutable_copynr_corrected2 <- otu_table(genusotutable_copynr_corrected, taxa_are_rows = TRUE)
otu_table(phylobj.otu) <- genusotutable_copynr_corrected2
sample_names(genusotutable_copynr_corrected2)

## Rename certain genera in taxonomy of phyloseq 
tax.tab <- data.frame(tax_table(phylobj.otu))
tax.tab$Genus <- as.character(tax.tab$Genus)
tax.tab$Genus[tax.tab$Genus == "Lachnospiraceae_ge"] <- "Lachnospiraceae"
tax.tab$Genus[tax.tab$Genus == "Ruminococcus2"] <- "Ruminococcus"
tax.tab$Genus[tax.tab$Genus == "Enterobacteriaceae_unclassified"] <- "Enterobacteriaceae"
tax.tab$Genus[tax.tab$Genus == "Lachnospiraceae_unclassified"] <- "Lachnospiraceae"
tax.tab$Genus[tax.tab$Genus == "Ruminococcaceae_unclassified"] <- "Ruminococcaceae"
tax.tab$Genus[tax.tab$Genus == "Erysipelotrichaceae_unclassified"] <- "Erysipelotrichaceae"
tax.tab$Genus[tax.tab$Genus == "Bacteria_unclassified"] <- "Bacteria"
tax.tab$Genus[tax.tab$Genus == "Clostridiales_unclassified"] <- "Clostridiales"
tax.tab$Genus[tax.tab$Genus == "Actinobacteria_unclassified"] <- "Actinobacteria"
tax.tab$Genus[tax.tab$Genus == "Alphaproteobacteria_unclassified"] <- "Alphaproteobacteria"
tax.tab$Genus[tax.tab$Genus == "Bacillales_unclassified"] <- "Bacillales"
tax.tab$Genus[tax.tab$Genus == "Bacteroidales_unclassified"] <- "Bacteroidales"
tax.tab$Genus[tax.tab$Genus == "Eubacteriaceae_unclassified"] <- "Eubacteriaceae"
tax.tab$Genus[tax.tab$Genus == "Gammaproteobacteria_unclassified"] <- "Gammaproteobacteria"
tax.tab$Genus[tax.tab$Genus == "Peptostreptococcaceae_unclassified"] <- "Peptostreptococcaceae"
tax.tab$Genus[tax.tab$Genus == "Pseudomonadaceae_unclassified"] <- "Pseudomonadaceae"
tax.tab$Genus[tax.tab$Genus == "Veillonellaceae_unclassified"] <- "Veillonellaceae"
tax.tab$Genus[tax.tab$Genus == "Firmicutes_unclassified"] <- "Firmicutes"
tax.tab$Genus <- as.factor(tax.tab$Genus)
tax.tab <- tax_table(as.matrix(tax.tab))
tax_table(phylobj.otu) <- tax.tab; rm(tax.tab)

# Add faecal samples to colon PD_Colons and remove from phyloseq 
phylobj.FS.PD_Colon.otu <- phylobj.otu
sample_data(phylobj.FS.PD_Colon.otu)$Shape <- "Nutrient_Load"
for (i in c("0", "200", "300", "400", "500", "83")) {
  phylobj1.0 <- subset_samples(phylobj.otu, sample_names(phylobj.otu)==i)
  sample_data(phylobj1.0)$PD_Colon <- "1"
  sample_data(phylobj1.0)$Shape <- "Faecal sample"
  sample_data(phylobj1.0)$sample <- paste(i, "P", sep="")
  sample_names(phylobj1.0) <- paste(i, "P", sep="")
  phylobj1.1 <- subset_samples(phylobj.otu, sample_names(phylobj)==i)
  sample_data(phylobj1.1)$PD_Colon <- "2"
  sample_data(phylobj1.1)$Shape <- "Faecal sample"
  sample_data(phylobj1.1)$sample <- paste(i, "D", sep="")
  sample_names(phylobj1.1) <- paste(i, "C", sep="")
  phylobj.FS.PD_Colon.otu <- merge_phyloseq(phylobj1.0, phylobj1.1,phylobj.FS.PD_Colon.otu)
  phylobj.FS.PD_Colon.otu <- subset_samples(phylobj.FS.PD_Colon.otu, sample_names(phylobj.FS.PD_Colon.otu)!=i)
  }
sample_data(phylobj.FS.PD_Colon.otu)$PD_Colon[sample_data(phylobj.FS.PD_Colon.otu)$PD_Colon == "1"] <- "P"
sample_data(phylobj.FS.PD_Colon.otu)$PD_Colon[sample_data(phylobj.FS.PD_Colon.otu)$PD_Colon == "2"] <- "D"

# Create proportional phyloseq on 100%
phylobj.prop.otu = transform_sample_counts(phylobj.FS.PD_Colon.otu, function(x) 100 * x/sum(x))

# QMP, incorporate cell counts in sampledata
phylobj.QMP.otu <- transform_sample_counts(phylobj.FS.PD_Colon.otu, function(x) x/sum(x))
for(n in 1:nsamples(phylobj.QMP.otu)){
# otu_table(phylobj.QMP.otu)[,n] <- as.numeric(otu_table(phylobj.QMP.otu)[,n])*as.numeric(sample_data(phylobj.QMP.otu)$Total_cells[n])
otu_table(phylobj.QMP.otu)[,n] <- as.numeric(otu_table(phylobj.QMP.otu)[,n])*as.numeric(sample_data(phylobj.QMP.otu)$meantot[n])
}

# Remove faecal samples from phylobj
phylobj.QMPFS.otu <- subset_samples(phylobj.QMP.otu, Transit == "FS")
phylobj.QMP.otu = subset_samples(phylobj.QMP.otu, Transit != "FS")


# ON GENUS LEVEL
phylobj <- moth_merge_NS_wnwn

# Aggregate according to genus
phylobj.agg <- tax_glom(phylobj, "Genus")
OTU1 = data.frame(otu_table(phylobj.agg)); tax.tab = data.frame(tax_table(phylobj.agg))
OTU1$Genus = tax.tab$Genus; otu.tab <- OTU1 %>% select(Genus, everything())
otu.tab$order <- 1:nrow(otu.tab); otu.tab <- otu.tab %>% select(order, everything())

# Copy number correction 
 ## Read in the RDP files containing information for copy number correction and format files
 RDPclass_QMP <- read.csv2("fasta_for_RDPclassifier.fastahier.csv")
 RDPclass_QMP_copynrcorrected <- read.csv2("fasta_for_RDPclassifier.fastahier_COPYNRCORRECTED.csv", as.is = TRUE)
 RDPclass_QMP$Seqnr <- as.numeric(RDPclass_QMP$Seqnr)
 RDPclass_QMP_copynrcorrected$Seqnr_corrected <- as.numeric(RDPclass_QMP_copynrcorrected$Seqnr_corrected)
 RDPclass_QMP_merge <- cbind(RDPclass_QMP,RDPclass_QMP_copynrcorrected)
 RDPclass_QMP_merge <- mutate(RDPclass_QMP_merge,copynr=Seqnr/Seqnr_corrected)
 RDPclass_QMP_merge$copynr[is.na(RDPclass_QMP_merge$copynr)] <- 1 # if the copy number is unknown take copy number 1 
 RDPclass_QMP_merge <- RDPclass_QMP_merge[,-c(3:6)]
 ## Run own script to change non corresponding taxa from RDP database
 otu.tab <- Change_taxaOTU(OTU.df = otu.tab, RDP = TRUE) 
 RDPclass_QMP_merge$Taxa <- gsub("unclassified_(.+)","\\1_unclassified",RDPclass_QMP_merge$Taxa)# old, new
 RDPclass_QMP_merge$Taxa <- gsub("Clostridium XlVa","Clostridium_XlVa",RDPclass_QMP_merge$Taxa)
 RDPclass_QMP_merge$Taxa <- gsub("\"Bacteroidetes\"_unclassified","Bacteroidetes_unclassified",RDPclass_QMP_merge$Taxa)
 RDPclass_QMP_merge$Taxa <- gsub("Clostridium sensu stricto", "Clostridium_sensu_stricto", RDPclass_QMP_merge$Taxa)
 RDPclass_QMP_merge$Taxa <- gsub("Clostridium IV","Clostridium_IV",RDPclass_QMP_merge$Taxa)
 RDPclass_QMP_merge$Taxa <- gsub("Clostridium XlVb","Clostridium_XlVb",RDPclass_QMP_merge$Taxa)
 RDPclass_QMP_merge$Taxa <- gsub("\"Bacteroidales\"_unclassified","Bacteroidales_unclassified",RDPclass_QMP_merge$Taxa)
 RDPclass_QMP_merge <- Change_taxaRDP(RDP.df = RDPclass_QMP_merge, RDP = TRUE)
 ## Combine the copy number correction factor with the genus table 
 otu.tab$Genus%in%RDPclass_QMP_merge$Taxa; test <- cbind(data.frame(x = otu.tab$Genus%in%RDPclass_QMP_merge$Taxa),otu.tab)
 if("FALSE" %in% test$x){stop("Taxa in OTU table are not matching RDP set!")}; rm(test)
 copynr_genus <- RDPclass_QMP_merge[match(otu.tab$Genus,RDPclass_QMP_merge$Taxa),]$copynr
 otu.tab2 <- subset(otu.tab, select = -c(Genus, order))
 genusotutable_copynr_corrected <- sweep(otu.tab2,1,copynr_genus,'/')
 colnames(genusotutable_copynr_corrected) <- sub("X", "", colnames(genusotutable_copynr_corrected))
 genusotutable_copynr_corrected2 <- otu_table(genusotutable_copynr_corrected, taxa_are_rows = TRUE)
 otu_table(phylobj.agg) <- genusotutable_copynr_corrected2
 sample_names(genusotutable_copynr_corrected2)
 
 ## Rename certain genera in taxonomy of phyloseq 
 tax.tab <- data.frame(tax_table(phylobj.agg))
 tax.tab$Genus <- as.character(tax.tab$Genus)
 tax.tab$Genus[tax.tab$Genus == "Lachnospiraceae_ge"] <- "Ruminococcus"
 tax.tab$Genus[tax.tab$Genus == "Ruminococcus2"] <- "Ruminococcus"
 tax.tab$Genus[tax.tab$Genus == "Enterobacteriaceae_unclassified"] <- "Enterobacteriaceae"
 tax.tab$Genus[tax.tab$Genus == "Lachnospiraceae_unclassified"] <- "Lachnospiraceae"
 tax.tab$Genus[tax.tab$Genus == "Ruminococcaceae_unclassified"] <- "Ruminococcaceae"
 tax.tab$Genus[tax.tab$Genus == "Erysipelotrichaceae_unclassified"] <- "Erysipelotrichaceae"
 tax.tab$Genus[tax.tab$Genus == "Bacteria_unclassified"] <- "Bacteria"
 tax.tab$Genus[tax.tab$Genus == "Clostridiales_unclassified"] <- "Clostridiales"
 tax.tab$Genus[tax.tab$Genus == "Actinobacteria_unclassified"] <- "Actinobacteria"
 tax.tab$Genus[tax.tab$Genus == "Alphaproteobacteria_unclassified"] <- "Alphaproteobacteria"
 tax.tab$Genus[tax.tab$Genus == "Bacillales_unclassified"] <- "Bacillales"
 tax.tab$Genus[tax.tab$Genus == "Bacteroidales_unclassified"] <- "Bacteroidales"
 tax.tab$Genus[tax.tab$Genus == "Eubacteriaceae_unclassified"] <- "Eubacteriaceae"
 tax.tab$Genus[tax.tab$Genus == "Gammaproteobacteria_unclassified"] <- "Gammaproteobacteria"
 tax.tab$Genus[tax.tab$Genus == "Peptostreptococcaceae_unclassified"] <- "Peptostreptococcaceae"
 tax.tab$Genus[tax.tab$Genus == "Pseudomonadaceae_unclassified"] <- "Pseudomonadaceae"
 tax.tab$Genus[tax.tab$Genus == "Veillonellaceae_unclassified"] <- "Veillonellaceae"
 tax.tab$Genus[tax.tab$Genus == "Firmicutes_unclassified"] <- "Firmicutes"
 tax.tab$Genus <- as.factor(tax.tab$Genus)
 tax.tab <- tax_table(as.matrix(tax.tab))
 tax_table(phylobj.agg) <- tax.tab
 
 
# Add faecal samples to colon PD_Colons and remove from phyloseq 
 phylobj.FS.PD_Colon <- phylobj.agg
 sample_data(phylobj.FS.PD_Colon)$Shape <- "Nutrient_Load"
 for (i in c("0", "200", "300", "400", "500", "83")) {
   phylobj1.0 <- subset_samples(phylobj.agg, sample_names(phylobj.agg)==i)
   sample_data(phylobj1.0)$PD_Colon <- "1"
   sample_data(phylobj1.0)$Shape <- "Faecal sample"
   sample_data(phylobj1.0)$sample <- paste(i, "P", sep="")
   sample_names(phylobj1.0) <- paste(i, "P", sep="")
   phylobj1.1 <- subset_samples(phylobj.agg, sample_names(phylobj)==i)
   sample_data(phylobj1.1)$PD_Colon <- "2"
   sample_data(phylobj1.1)$Shape <- "Faecal sample"
   sample_data(phylobj1.1)$sample <- paste(i, "D", sep="")
   sample_names(phylobj1.1) <- paste(i, "C", sep="")
   phylobj.FS.PD_Colon <- merge_phyloseq(phylobj1.0, phylobj1.1,phylobj.FS.PD_Colon)
   phylobj.FS.PD_Colon <- subset_samples(phylobj.FS.PD_Colon, sample_names(phylobj.FS.PD_Colon)!=i)
   }
 
 sample_data(phylobj.FS.PD_Colon)$PD_Colon[sample_data(phylobj.FS.PD_Colon)$PD_Colon == "1"] <- "P"
 sample_data(phylobj.FS.PD_Colon)$PD_Colon[sample_data(phylobj.FS.PD_Colon)$PD_Colon == "2"] <- "D"
 
# Create proportional phyloseq on 100%
phylobj.prop = transform_sample_counts(phylobj.FS.PD_Colon, function(x) 100 * x/sum(x))

# QMP, incorporate cell counts in sampledata
phylobj.QMP <- transform_sample_counts(phylobj.FS.PD_Colon, function(x) x/sum(x))
for(n in 1:nsamples(phylobj.QMP)){
# otu_table(phylobj.QMP)[,n] <- as.numeric(otu_table(phylobj.QMP)[,n])*as.numeric(sample_data(phylobj.QMP)$Total_cells[n])
otu_table(phylobj.QMP)[,n] <- as.numeric(otu_table(phylobj.QMP)[,n])*as.numeric(sample_data(phylobj.QMP)$meantot[n])
}

# Make phylobj of all samples
phylobj.QMPFS <- subset_samples(phylobj.QMP, Transit == "FS")

# Remove faecal samples from phylobj
phylobj.QMPFS <- subset_samples(phylobj.QMP, Transit == "FS")
phylobj.QMP = subset_samples(phylobj.QMP, Transit != "FS")

# Remove NA
# phylobj.QMP <- subset_samples(phylobj.QMP, !is.na(meantot))
# Remove unneeded objects
# rm(sharedfile,taxfile, faecal, phylobj1.0, phylobj1.1, OTU1, tax.tab, otu.tab, copynr_genus, otu.tab2, genusotutable_copynr_corrected, genusotutable_copynr_corrected2, RDPclass_QMP, RDPclass_QMP_copynrcorrected, RDPclass_QMP_merge, mothur_data, minobs)




# Import SCFA data for MIXOMICS
functdata <- read.csv2("Dataframe_alt.csv", sep=",")
# order levels and transform from continuous scale to factor
functdata$Donor[functdata$Donor == 4] <- 7; functdata$Donor[functdata$Donor == 6] <- 4
functdata$Donor[functdata$Donor == 7] <- 6; 
functdata$Donor <- as.ordered(factor(functdata$Donor,levels=c("1","2","3","4","5","6")))
functdata$Vessel <- as.ordered(factor(functdata$Vessel,levels=c("P","D","FS")))
functdata$Ninetransit <- as.ordered(factor(functdata$Ninetransit,levels=c("0","1","2","3","4")))
functdata$Transit <- as.ordered(factor(functdata$Transit,levels=c("ST","MT","LT","FS")))
# Make columns numeric
i <- c(9:34) # Specify columns you want to change
functdata[ , i] <- apply(functdata[ , i], 2,  # Specify own function within apply
                    function(x) as.numeric(as.character(x)))
class(functdata[,9:34]); head(functdata[,9:34])
# x2 compensation to faecal sample of donor 5 (number 400, was diluted extra in preparation process)
functdatatrt <- subset(functdata, Samplenumber == 400)
functdatatrt[,9:18] <- functdatatrt[,9:18]*2
functdata <- rbind(subset(functdata, Samplenumber != 400), functdatatrt)

# Add total_SCFA ratio of SCFA and branched to dataframe
functdatanet <- functdata # To use down below
functdata <- mutate(functdata,total_SCFA=Acetate+Propionate+Butyrate+Isobutyrate+Isovalerate+Valerate+Isocaproate+Caproate+Heptanoate)
functdata <- mutate(functdata,Acetateratio=Acetate/total_SCFA,
                    Propionateratio=Propionate/total_SCFA,
                    Butyrateratio=Butyrate/total_SCFA,
                    Branched = Isovalerate+Isobutyrate, 
                    Branchedratio = Branched/total_SCFA,
                    But_prop=Butyrate/Propionate)

# Omit NA from dataframe 
functdata <- functdata %>% filter(!is.na(Acetate))
functdatabrut <- functdata

# Create dataframe without stabilisation
functdatatrt <- functdata %>% filter(!is.na(Ninetransit))

## Create dataframe without pumps and halfway transit
SCFAdata <- subset(functdatatrt,Ninetransit=='0'|Ninetransit=='3')


# Remove the intermediates
rm(functdata, functdatatrt, functdatanet)
```

## Percentage OTU per genus
```{r OTU per genus, fig.height=10, fig.width=16}
`%nin%` = Negate(`%in%`)
library(genefilter)
phylo <- subset_samples(phylobj.prop.otu, Transit != "FS")
taxtable <- data.frame(tax_table(phylo))
otutable <- data.frame(otu_table(phylo))
combinedtable <- cbind(taxtable, otutable)
combinedtable2 <- data.frame(); x = 1
for (i in unique(combinedtable$Genus)){
  test2 <- subset(combinedtable, combinedtable$Genus == i)
  test <- test2[,c(7:ncol(test2))]
  for(col in names(test)) {
  test[paste0(col, "_pct")] = test[col] / sum(test[col])
  }
  test3 <- test[,c(269:ncol(test))]
  test3 <- test3 *100
  # test3[is.na(test3)] <- 1
  test3$SD <- rowSds(test3, na.rm=TRUE)
  test3$mean=rowMeans(test3, na.rm=TRUE)
  test3$genus <- rep(i,nrow(test3))
  cols <- ncol(test3)-2;  test3 <- test3[,c(cols:ncol(test3))]
  # if(i != "Clostridium_XlVa"){
    combinedtable2 <- rbind(combinedtable2, test3)
    # }
    print(paste0(i, ", ",x, " of ", length(unique(combinedtable$Genus)), " genera done")); x <- x + 1
  }
combinedtable2$mean <- format(round(combinedtable2$mean, 2), nsmall = 2)
combinedtable2$SD <- format(round(combinedtable2$SD, 2), nsmall = 2)
combinedtable2$text <- paste0(combinedtable2$mean, " Â± ", combinedtable2$SD) 
combinedtable2 <- rownames_to_column(combinedtable2)
view(combinedtable2)
df.combined <- combinedtable2
df.combined$SD <- as.numeric(df.combined$SD); df.combined$mean <- as.numeric(df.combined$mean)
df.combined$sum <- df.combined$mean + df.combined$SD
df.combined <-subset(df.combined, sum > 5)
write.csv2(df.combined, "Genera_OTU.csv")
```

## TAKES A WHILE - Retrieve list of Top 25 genera and genera with significant changes to transit time alterations
```{r Significant list, fig.height=30, fig.width=25, warning=F}
my_comparisons <- list(c("ST", "MT"), c("MT", "LT"), c("ST", "LT"))
bars = T; barstransit = F

# As it takes long, check first if it already exists
check = F; if(check == T){
if(exists("df.sig")){stop("The significantlist already exists, no need to wait for a new list")}}

redo = F #Make T if you want to redo the calculations
if(redo == T){
# Absolute
phylobj.QMP_stable <- subset_samples(phylobj.QMP, Ninetransit == 3)
Top25 <- aggregate_top_taxa(phylobj.QMP_stable, top = 25, "Genus")
Top25 <- psmelt(Top25); Top25list <- unique(Top25$OTU); Top25list <- Top25list[Top25list != "Other"]
Top25list <- sort(Top25list)
# Add extra column days for graphs
Top <- psmelt(subset_samples(phylobj.QMP, Ninetransit == 3))
Top$Transit <- as.ordered(factor(Top$Transit, levels = c("ST", "MT", "LT")))
Top$PD_Colon <- as.ordered(factor(Top$PD_Colon, levels = c("P", "D", "FS")))
Top$Donor <- as.ordered(factor(Top$Donor, levels = c("1", "2", "3", "4", "5", "6")))
highlist <- data.frame()
for (i in unique(Top$Genus)) {
  if(i!= "Other"){Top2 <- subset(Top, Genus == i)
  Top3 <- data.frame(Genus = i, Abundance = max(Top2$Abundance))
  highlist <- rbind(highlist, Top3)} 
}
# Order with highest abundances
sequence <- highlist[order(-highlist$Abundance),]; sequence <- sequence$Genus
# Proportional
phylobj.prop_2 = transform_sample_counts(phylobj.QMP, function(x) 100 * x/sum(x))
phylobj.prop_stable <- subset_samples(phylobj.prop_2, Ninetransit == 3)
Top15phylobj.prop <- phylobj.prop_stable
Top15.prop <- psmelt(Top15phylobj.prop); unique(Top15.prop$Genus)
# Add extra column days for graphs
Top.prop <- Top15.prop
Top.prop$Transit <- as.ordered(factor(Top.prop$Transit, levels = c("ST", "MT", "LT")))
Top.prop$PD_Colon <- as.ordered(factor(Top.prop$PD_Colon, levels = c("P", "D", "FS")))
Top.prop$Donor <- as.ordered(factor(Top.prop$Donor, levels = c("1", "2", "3", "4", "5", "6")))
unique(Top$Genus)%in%unique(Top.prop$Genus) 
# Statistics
# Non parametric to 1-way ANOVA, alternative - Kruskal-Wallis
correction = "holm"
# Between transits
gglist <- list(); gglist2 <- list(); y = 1; df.grpPMP <- data.frame(); df.grpQMP <- data.frame(); statlist <- list()
message("Kruskal-Wallis tests between transit times")
df.pwc <- data.frame(); df.pwc2 <- data.frame(); df.sig <- data.frame()
# PLOTS
comparisonlistbar <- list(); x =1; y =1
for (i in sequence) {
  Top1 <- subset(Top, Genus == i); Top2 <- subset(Top.prop, Genus == i)
  
  T1 <- Top1 %>% group_by(Transit, PD_Colon) %>% summarise(SD = sd(Abundance), 
    Abundance = mean(Abundance),Transit = Transit, PD_Colon = PD_Colon) 
  T1<- T1[!duplicated(T1), ]; T1$min = T1$Abundance - T1$SD; T1$max = T1$Abundance + T1$SD
    max = max(T1$max); min = min(T1$min)
    
  # Create data frame with means and SD
    T1 <- subset(Top1, Abundance > 0); T1$Abundance <- as.numeric(T1$Abundance)
  T1 <- T1 %>% group_by(Transit, PD_Colon) %>% summarise(SD = sd(Abundance), 
    Abundance = mean(Abundance),Transit = Transit, PD_Colon = PD_Colon); T1<- T1[!duplicated(T1),]
  T1$Genus <- i; T1 <- subset(T1, select = c(Genus, Transit, PD_Colon, Abundance, SD))
  df.grpQMP <- rbind(df.grpQMP, T1)
  
  
  T1 <- subset(Top2, Abundance > 0); T1$Abundance <- as.numeric(T1$Abundance)
  T1 <- T1 %>% group_by(Transit, PD_Colon) %>% summarise(SD = sd(Abundance), 
    Abundance = mean(Abundance),Transit = Transit, PD_Colon = PD_Colon); T1<- T1[!duplicated(T1),]
  T1$Genus <- i; T1 <- subset(T1, select = c(Genus, Transit, PD_Colon, Abundance, SD)); df.grpPMP <- rbind(df.grpPMP, T1)  
    
# Absolute abundance
p <- papertheme(ggbarplot(Top1, x="Transit", y="Abundance" , fill = "Donor", 
                  add = "mean_sd", facet.by = c("PD_Colon"), position = position_dodge())) 
p<-p + facet_grid(PD_Colon~., labeller = labeller(PD_Colon = PD_Colon.labs)) +
  scale_fill_manual(values = sixdonors) + scale_x_discrete(labels = c("Short", "Medium", "Long"))+
  theme(strip.background = element_rect(fill = "#D6E0F4"), strip.text.x = element_text(size = 20),
        strip.text.y = element_text(size = 20, angle = 270), legend.text = element_text(size = 18), 
        legend.title = element_text(size = 18, face = "bold"),
        legend.title.align = 0.5) + xlab("Transit time") +
  ylab(expression(paste("Absolute abundance (cells m",L^{"-1"}, ")"))) +
  guides(fill = guide_legend("Donor", ncol = 6)) + theme(strip.text.y = element_blank())
  if(x%%4 != 0){p <- p + theme(legend.position = "none")}
p3 <- papertheme(ggbarplot(Top1, x="Transit", y="Abundance" , fill = "Transit", 
                           add = "mean_sd", facet.by = c("PD_Colon")))
if(barstransit == F){
  p3 <- papertheme(ggboxplot(Top1, x="Transit", y="Abundance" , fill = "Transit", facet.by = c("PD_Colon")))
  } 
p3 <- p3 + facet_grid(PD_Colon~., labeller = labeller(PD_Colon = PD_Colon.labs)) +
    scale_fill_manual(values = transitcolors) + scale_x_discrete(labels = c("Short", "Medium", "Long"))+
    theme(strip.background = element_rect(fill = "#D6E0F4"), strip.text.x = element_text(size = 20),
    strip.text.y = element_text(size = 20, angle = 270), legend.text = element_text(size = 18), 
    legend.title = element_text(size = 18, face = "bold"),
    legend.title.align = 0.5) + xlab("Transit time") +
  ylab(expression(paste("Absolute abundance (cells m",L^{"-1"}, ")"))) + 
  guides(fill = guide_legend("Transit", ncol = 5)) +
  theme(axis.title.y = element_blank(), axis.text.y=element_blank(), axis.ticks.y = element_blank()) 
if(x%%4 != 0){p3 <- p3 + theme(legend.position = "none")}
gen = i
ymax = max(ggplot_build(p)$layout$panel_scales_y[[1]]$range$range)
message(paste0("Quantitative - ", round(100*x/length(sequence)), "%"))
plotstat <- statplot(dataframe = Top1, AboveZero = T, ymax = ymax) + theme(legend.position = "none")
ymax = max(ggplot_build(plotstat)$layout$panel_scales_y[[1]]$range$range); ymin = min(ggplot_build(p)$layout$panel_scales_y[[1]]$range$range)
plotstat <- plotstat + scale_y_continuous(limits = c(ymin, ymax))
if(is.infinite(ymax)){ymax = 1}; if(is.infinite(ymin)){ymin = 0}
plot1 <- ggarrange(p + theme(legend.position = "none") +scale_y_continuous(limits = c(ymin, ymax)) , plotstat, widths = c(1,0.3))
stats <- statplot(dataframe = Top1, AboveZero = T, plot = "ymax", showstat = F, yaxis = T)
### Calculate p values and put them in a data frame
  Top5 <- subset(Top1, Abundance > 0); Top5$Abundance <- as.numeric(Top5$Abundance)
  
 #### Proximal colon
  my_data5 <- subset(Top5, PD_Colon == "P"); unique(my_data5$Transit); my_data5$Transit <- droplevels(my_data5$Transit)
  
  skip_to_next <- FALSE
  tryCatch(res.kruskal <- my_data5 %>% rstatix::kruskal_test(Abundance ~ Transit), error = function(e) {skip_to_next <<- TRUE
  message(crayon::red("Only 1 transit time has", gen,  "in the proximal colon!
                      Skipped to next")) })
  
  if(skip_to_next == FALSE){
  res.kruskal <- my_data5 %>% rstatix::kruskal_test(Abundance ~ Transit)
  pvaluep <- res.kruskal$p; message(pvaluep); effectsize2 <- Effectsize(my_data5)
  }else{
    pvaluep <- 1
    effectsize2 <- "Effectsize does not matter"
  }
  df.prox.qmp <- data.frame(genus = gen, vessel = "Proximal colon", pvalue = pvaluep)
 
  #### Distal colon
  my_data5 <- subset(Top5, PD_Colon == "D"); unique(my_data5$Transit); my_data5$Transit <- droplevels(my_data5$Transit)
  
  skip_to_next <- FALSE
  tryCatch(res.kruskal <- my_data5 %>% rstatix::kruskal_test(Abundance ~ Transit), error = function(e) {skip_to_next <<- TRUE
  message(crayon::red("Only 1 transit time has", gen,  "in the distal colon!
                      Skipped to next")) })
  
  if(skip_to_next == FALSE){
  res.kruskal <- my_data5 %>% rstatix::kruskal_test(Abundance ~ Transit)
  pvaluep <- res.kruskal$p; message(pvaluep); effectsize2 <- Effectsize(my_data5)
  }else{
    pvaluep <- 1
    effectsize2 <- "Effectsize does not matter"
  }
  df.dist.qmp <- data.frame(genus = gen, vessel = "Distal colon", pvalue = pvaluep)
  
  
# Proportional abundance
p2 <- papertheme(ggbarplot(Top2, x="Transit", y="Abundance" , fill = "Donor", 
                  add = "mean_sd", facet.by = c("PD_Colon"), position = position_dodge())) 
  if(bars == F){ 
    p2 <- papertheme(ggboxplot(Top2, x="Transit", y="Abundance" , fill = "Donor", 
                              facet.by = c("PD_Colon")))
  }
p2<-p2 + facet_grid(PD_Colon~., labeller = labeller(PD_Colon = PD_Colon.labs)) +
  scale_fill_manual(values = sixdonors) + scale_x_discrete(labels = c("Short", "Medium", "Long"))+
  theme(strip.background = element_rect(fill = "#D6E0F4"), strip.text.x = element_text(size = 20),
        strip.text.y = element_text(size = 20, angle = 270), legend.text = element_text(size = 18), 
        legend.title = element_text(size = 18, face = "bold"),
        legend.title.align = 0.5) + xlab("Transit time") + ylab("Proportional abundance (%)") +
  guides(fill = guide_legend("Donor", ncol = 6)) + theme(strip.text.y = element_blank())
  if(x%%4 != 0){p2 <- p2 + theme(legend.position = "none")}
  # + stat_compare_means(comparisons = my_comparisons, label = "p.signif", hide.ns = T, hide.NS = T) # Add significance levels
  # stat_compare_means(label.y = 50) 
p4 <- papertheme(ggbarplot(Top2, x="Transit", y="Abundance" , fill = "Transit", 
                           add = "mean_sd", facet.by = c("PD_Colon")))
if(barstransit == F){
  p4 <- papertheme(ggboxplot(Top2, x="Transit", y="Abundance" , fill = "Transit", facet.by = c("PD_Colon")))
  } 
p4 <- p4 + facet_grid(PD_Colon~., labeller = labeller(PD_Colon = PD_Colon.labs)) +
    scale_fill_manual(values = transitcolors) + scale_x_discrete(labels = c("Short", "Medium", "Long"))+
    theme(strip.background = element_rect(fill = "#D6E0F4"), strip.text.x = element_text(size = 20),
    strip.text.y = element_text(size = 20, angle = 270), legend.text = element_text(size = 18), 
    legend.title = element_text(size = 18, face = "bold"),
    legend.title.align = 0.5) + xlab("Transit time") + ylab("Proportional abundance (%)") + 
  guides(fill = guide_legend("Transit", ncol = 5)) +
  theme(axis.title.y = element_blank(), axis.text.y=element_blank(), axis.ticks.y = element_blank()) 
if(x%%4 != 0){p4 <- p4 + theme(legend.position = "none")}
# plot2 <- ggarrange(p2, p4, widths = c(1,0.3))
ymax = max(ggplot_build(p2)$layout$panel_scales_y[[1]]$range$range)
message("Proportional")
plotstat <- statplot(dataframe = Top2, AboveZero = T, ymax = ymax); boxlegend <-cowplot::get_legend(plotstat)
legends <- ggdraw(plot_grid(NULL,cowplot::get_legend(p2), boxlegend,NULL, ncol=4, align='v'))
ymax = max(xx <- ggplot_build(plotstat)$layout$panel_scales_y[[1]]$range$range)
ymin = min(ggplot_build(p2)$layout$panel_scales_y[[1]]$range$range)
if(is.infinite(ymax)){ymax = 1}; if(is.infinite(ymin)){ymin = 0}
plotstat <- plotstat + scale_y_continuous(limits = c(ymin, ymax)) + theme(legend.position = "none")
p2 <- p2 + theme(legend.position = "none") + scale_y_continuous(limits = c(ymin, ymax)); box <- box + theme(legend.position = "none")
gen = i; plot2 <- ggarrange(p2, plotstat, widths = c(1,0.3))
stats2 <- statplot(dataframe = Top2, AboveZero = T, plot = "ymax", showstat = F, yaxis = T)
### Calculate p values and put them in a data frame
  Top5 <- subset(Top2, Abundance > 0); Top5$Abundance <- as.numeric(Top5$Abundance)
  
 #### Proximal colon
  my_data5 <- subset(Top5, PD_Colon == "P"); unique(my_data5$Transit); my_data5$Transit <- droplevels(my_data5$Transit)
  
  skip_to_next <- FALSE
  tryCatch(res.kruskal <- my_data5 %>% rstatix::kruskal_test(Abundance ~ Transit), error = function(e) {skip_to_next <<- TRUE
  message(crayon::red("Only 1 transit time has", gen,  "in the proximal colon!
                      Skipped to next")) })
  
  if(skip_to_next == FALSE){
  res.kruskal <- my_data5 %>% rstatix::kruskal_test(Abundance ~ Transit)
  pvaluep <- res.kruskal$p; message(pvaluep); effectsize2 <- Effectsize(my_data5)
  }else{
    pvaluep <- 1
    effectsize2 <- "Effectsize does not matter"
  }
  df.prox.prop <- data.frame(genus = gen, vessel = "Proximal colon", pvalue = pvaluep)
 
  #### Distal colon
  my_data5 <- subset(Top5, PD_Colon == "D"); unique(my_data5$Transit); my_data5$Transit <- droplevels(my_data5$Transit)
  
  skip_to_next <- FALSE
  tryCatch(res.kruskal <- my_data5 %>% rstatix::kruskal_test(Abundance ~ Transit), error = function(e) {skip_to_next <<- TRUE
  message(crayon::red("Only 1 transit time has", gen,  "in the distal colon!
                      Skipped to next")) })
  
  if(skip_to_next == FALSE){
  res.kruskal <- my_data5 %>% rstatix::kruskal_test(Abundance ~ Transit)
  pvaluep <- res.kruskal$p; message(pvaluep); effectsize2 <- Effectsize(my_data5)
  }else{
    pvaluep <- 1
    effectsize2 <- "Effectsize does not matter"
  }
  df.dist.prop <- data.frame(genus = gen, vessel = "Distal colon", pvalue = pvaluep)
  
  df.test <- cbind(df.prox.qmp, df.prox.prop)
  df.test2 <- cbind(df.dist.qmp, df.dist.prop)
  df.sig <- rbind(df.sig, df.test, df.test2)
statlist[[x]] <- ggarrange(stats, stats2, ncol = 2); rm(stats, stats2)
  if (x%%4 != 0){p2 <- p2 + theme(legend.position = "none")
  p <- ggarrange(plot1, plot2, ncol = 2)}
  p<- ggarrange(plot1, plot2, ncol = 2)
  if(x%%4 == 0){p <- ggarrange(plot1, plot2, ncol = 2, common.legend = T, legend = "bottom")
  p <- ggarrange(p, legends, nrow = 2, heights = c(1,0.1))} 
  p <- annotate_figure(p, top = text_grob(i, color = "black", face = "italic", size = 20))
  # print(p)
  comparisonlistbar[[x]] <- p; x = x + 1
}
for (i in seq(from = 1, to = (length(comparisonlistbar)-2), by = 4)) {
 comparisons <- ggarrange(comparisonlistbar[[i]], comparisonlistbar[[i+1]], comparisonlistbar[[i+2]],comparisonlistbar[[i+3]], nrow = 4)
   print(comparisons)
}
write.csv2(df.sig, "df.sig.csv")
  }else{df.sig <- read.csv2("df.sig.csv")
  if(length(colnames(df.sig)) ==7){
    df.sig <- subset(df.sig, select = -c(X))
    
    phylobj.QMP_stable <- subset_samples(phylobj.QMP, Ninetransit == 3)
    Top25 <- aggregate_top_taxa(phylobj.QMP_stable, top = 25, "Genus")
    Top25 <- psmelt(Top25); Top25list <- unique(Top25$OTU); Top25list <- Top25list[Top25list != "Other"]
    Top25list <- sort(Top25list)
  }
  }
# View(df.grpQMP); View(df.grpPMP)
View(df.sig)

df.sigQMP <- df.sig[,1:3]; df.sigprop <- df.sig[,4:6]; colnames(df.sigprop)  <- c("genus", "vessel", "pvalue")
df.sigQMP <- subset(df.sigQMP, pvalue <=0.05); df.sigprop <- subset(df.sigprop, pvalue <=0.05)
significantlist <- rbind(data.frame(Genus = unique(df.sigQMP$genus)), 
                         data.frame(Genus = unique(df.sigprop$genus)), 
                         data.frame(Genus = Top25list))
significantlist <- unique(significantlist$Genus)
significantOTU <- data.frame()
for(i in sort(significantlist)){
  df <- subset(df.combined, genus ==i)
  significantOTU <- rbind(significantOTU, df)
}
View(significantOTU)



df.sig2P <- subset(df.sig, vessel == "Proximal colon"); rownames(df.sig2P) <- df.sig2P$genus
df.sig2P$p_P_QMP <- df.sig2P$pvalue; df.sig2P$p_P_PMP <- df.sig2P$pvalue.1
df.sig2D <- subset(df.sig, vessel == "Distal colon"); rownames(df.sig2D) <- df.sig2D$Genus
df.sig2D$p_D_QMP <- df.sig2D$pvalue; df.sig2D$p_D_PMP <- df.sig2D$pvalue.1

df.sig3 <- data.frame(genus = df.sig2P$genus, p_P_QMP = df.sig2P$p_P_QMP, p_D_QMP= df.sig2D$p_D_QMP, 
                      p_P_PMP = df.sig2P$p_P_PMP, p_D_PMP= df.sig2D$p_D_PMP)

df.sig3 <- subset(df.sig3, p_P_QMP <= 0.05 | p_D_QMP <= 0.05 | p_P_PMP <= 0.05 | p_D_PMP <= 0.05)
```

## Genera and OTUs
```{r Look into genera and the OTUs underneath, fig.height=30, fig.width=15, warning=F}
if(!exists("combinedtable2")){stop("table is not present, run chunk above first!")}
my_comparisons <- list(c("ST", "MT"), c("MT", "LT"), c("ST", "LT"))
bars = T; barstransit = F; pptlist <- list()

# Absolute
phylobj.QMP_stable <- subset_samples(phylobj.QMP.otu, Ninetransit == 3)
phylobj.QMP_stable_genus <- subset_samples(phylobj.QMP, Ninetransit == 3)

# Proportional
phylobj.prop_2 = transform_sample_counts(phylobj.QMP.otu, function(x) 100 * x/sum(x))
phylobj.prop_stable <- subset_samples(phylobj.prop_2, Ninetransit == 3)
phylobj.prop_2 = transform_sample_counts(phylobj.QMP, function(x) 100 * x/sum(x))
phylobj.prop_stable_genus <- subset_samples(phylobj.prop_2, Ninetransit == 3)
onlysig = T
if(onlysig ==T){
  total_genera <- significantlist
}else{total_genera <- psmelt(phylobj.QMP_stable_genus); total_genera <- unique(total_genera$Genus)}
total_genera <- sort(total_genera); m =1
for(i in total_genera){
# if(i == "Clostridiales"){i = "Clostridiales_unclassified"}
phylobj.QMP_otu <- subset_taxa(phylobj.QMP_stable, Genus == i)
phylobj.prop_otu <- subset_taxa(phylobj.prop_stable, Genus == i)
# if(i == "Clostridiales_unclassified"){i = "Clostridiales"}
phylobj.QMP_genus <- subset_taxa(phylobj.QMP_stable_genus, Genus == i)
phylobj.prop_genus <- subset_taxa(phylobj.prop_stable_genus, Genus == i)
df.QMP_otu <- psmelt(phylobj.QMP_otu); df.QMP_genus <- psmelt(phylobj.QMP_genus) 
df.prop_otu <- psmelt(phylobj.prop_otu); df.prop_genus <- psmelt(phylobj.prop_genus) 
# Add extra column days for graphs - Absolute abundance
df.QMP_otu$Transit <- as.ordered(factor(df.QMP_otu$Transit, levels = c("ST", "MT", "LT")))
df.QMP_otu$PD_Colon <- as.ordered(factor(df.QMP_otu$PD_Colon, levels = c("P", "D", "FS")))
df.QMP_otu$Donor <- as.ordered(factor(df.QMP_otu$Donor, levels = c("1", "2", "3", "4", "5", "6")))
df.QMP_genus$Transit <- as.ordered(factor(df.QMP_genus$Transit, levels = c("ST", "MT", "LT")))
df.QMP_genus$PD_Colon <- as.ordered(factor(df.QMP_genus$PD_Colon, levels = c("P", "D", "FS")))
df.QMP_genus$Donor <- as.ordered(factor(df.QMP_genus$Donor, levels = c("1", "2", "3", "4", "5", "6")))
# Add extra column days for graphs - Proportional abundance
df.prop_otu$Transit <- as.ordered(factor(df.prop_otu$Transit, levels = c("ST", "MT", "LT")))
df.prop_otu$PD_Colon <- as.ordered(factor(df.prop_otu$PD_Colon, levels = c("P", "D", "FS")))
df.prop_otu$Donor <- as.ordered(factor(df.prop_otu$Donor, levels = c("1", "2", "3", "4", "5", "6")))
df.prop_genus$Transit <- as.ordered(factor(df.prop_genus$Transit, levels = c("ST", "MT", "LT")))
df.prop_genus$PD_Colon <- as.ordered(factor(df.prop_genus$PD_Colon, levels = c("P", "D", "FS")))
df.prop_genus$Donor <- as.ordered(factor(df.prop_genus$Donor, levels = c("1", "2", "3", "4", "5", "6")))
plotlist <- list(); y = 1    
plot <- papertheme(ggbarplot(df.QMP_genus, x="Transit", y="Abundance" , fill = "Donor", 
                  add = "mean_sd", facet.by = c("PD_Colon"), position = position_dodge())) +
  facet_grid(PD_Colon~., labeller = labeller(PD_Colon = PD_Colon.labs)) +
  ggtitle(i)+
  scale_fill_manual(values = sixdonors) + scale_x_discrete(labels = c("Short", "Medium", "Long"))+
  theme(strip.background = element_rect(fill = "#D6E0F4"), strip.text.x = element_text(size = 20),
        strip.text.y = element_text(size = 20, angle = 270), legend.text = element_text(size = 18), 
        legend.title = element_text(size = 18, face = "bold"),
        legend.title.align = 0.5) + xlab("SHIME transit time") +
  ylab(expression(paste("Absolute abundance (cells m",L^{"-1"}, ")"))) +
  guides(fill = guide_legend("Donor", ncol = 6)) + theme(strip.text.y = element_blank())

plot2 <- papertheme(ggbarplot(df.prop_genus, x="Transit", y="Abundance" , fill = "Donor", 
                  add = "mean_sd", facet.by = c("PD_Colon"), position = position_dodge())) +
  facet_grid(PD_Colon~., labeller = labeller(PD_Colon = PD_Colon.labs)) +
  ggtitle(i)+
  scale_fill_manual(values = sixdonors) + scale_x_discrete(labels = c("Short", "Medium", "Long"))+
  theme(strip.background = element_rect(fill = "#D6E0F4"), strip.text.x = element_text(size = 20),
        strip.text.y = element_text(size = 20, angle = 270), legend.text = element_text(size = 18), 
        legend.title = element_text(size = 18, face = "bold"),
        legend.title.align = 0.5) + xlab("SHIME transit time") +
  ylab("Proportional abundance (%)") +
  guides(fill = guide_legend("Donor", ncol = 6)) + theme(strip.text.y = element_blank())
plot <- ggarrange(plot, plot2, common.legend = T, legend = "none")
plotlist[[1]] <- plot; 
y = 2

for(x in unique(df.QMP_otu$OTU)){
  test <- subset(combinedtable2, rowname == x)
  test$SD <- as.numeric(test$SD); test$mean <- as.numeric(test$mean); combo =test$mean +test$SD  
  if(combo > 5) {
plot <- papertheme(ggbarplot(subset(df.QMP_otu, OTU ==x), x="Transit", y="Abundance" , fill = "Donor", 
                  add = "mean_sd", facet.by = c("PD_Colon"), position = position_dodge())) +
  facet_grid(PD_Colon~., labeller = labeller(PD_Colon = PD_Colon.labs)) +
  ggtitle(paste0(x, "(",test$text,"%)", sep = ""))+
  scale_fill_manual(values = sixdonors) + scale_x_discrete(labels = c("Short", "Medium", "Long"))+
  theme(strip.background = element_rect(fill = "#D6E0F4"), strip.text.x = element_text(size = 20),
        strip.text.y = element_text(size = 20, angle = 270), legend.text = element_text(size = 18), 
        legend.title = element_text(size = 18, face = "bold"),
        legend.title.align = 0.5) + xlab("SHIME Transit time") +
  ylab(expression(paste("Absolute abundance (cells m",L^{"-1"}, ")"))) +
  guides(fill = guide_legend("Donor", ncol = 6)) + theme(strip.text.y = element_blank())

plot2 <- papertheme(ggbarplot(subset(df.prop_otu, OTU ==x), x="Transit", y="Abundance" , fill = "Donor", 
                  add = "mean_sd", facet.by = c("PD_Colon"), position = position_dodge())) +
  facet_grid(PD_Colon~., labeller = labeller(PD_Colon = PD_Colon.labs)) +
  ggtitle(paste0(x, "(",test$text,"%)", sep = ""))+
  scale_fill_manual(values = sixdonors) + scale_x_discrete(labels = c("Short", "Medium", "Long"))+
  theme(strip.background = element_rect(fill = "#D6E0F4"), strip.text.x = element_text(size = 20),
        strip.text.y = element_text(size = 20, angle = 270), legend.text = element_text(size = 18), 
        legend.title = element_text(size = 18, face = "bold"),
        legend.title.align = 0.5) + xlab("SHIME transit time") +
  ylab(expression(paste("Absolute abundance (cells m",L^{"-1"}, ")"))) +
  guides(fill = guide_legend("Donor", ncol = 6)) + theme(strip.text.y = element_blank())

# only show legend when last figure
if(x == unique(df.QMP_otu$OTU)[length(unique(df.QMP_otu$OTU))]){
  plot <- ggarrange(plot, plot2, common.legend = T, legend = "bottom")
}else{plot <- ggarrange(plot, plot2, common.legend = T, legend = "none")}
plotlist[[y]] <- plot; y = y +1}
}

arrangedplot <- ggarrange(plotlist = plotlist, ncol = 1, common.legend = T, legend = "bottom")
pptlist[[m]] <- arrangedplot
print(arrangedplot)

if(m%%5 == 0){message(paste0( m,"/", length(total_genera)," - "   , round(m/length(total_genera)*100), "%"))}; m <- m +1
}
ppt = T
if(ppt == T){
    ppt_presentation <- read_pptx()
    for(i in 1: length(pptlist)) {
      print(pptlist[[i]])
      plot <- ggarrange(list)
      ppt_presentation <- add_slide(ppt_presentation, layout="Title and Content", master="Office Theme") %>%
        ph_with(res = 75,  value = pptlist[[i]], 
                # location = ph_location_type(type = "body")
                location = ph_location_fullsize()
                )
      }
    ppt_presentation <- ppt_presentation %>% print(target = "Genus_with_OTU.pptx")
} 

  cat(crayon::yellow(whale3))
```


# Dirichlet Multinomial Mixtures
```{r Dirichlet Multinomial Mixtures}
library(microbiome)
library(DirichletMultinomial)
library(reshape2)
library(magrittr)
library(dplyr)

pseq1 <- prune_taxa(taxa_names(phylobj.QMP.otu) != "Otu00002", phylobj.QMP.otu )
pseq1 <- prune_taxa(taxa_names(pseq1) != "Otu00009", pseq1 )

pseq <- transform_sample_counts(aggregate_taxa(subset_samples(pseq1, Ninetransit ==3), "Genus"), function(x) 1 * x/sum(x))
pseqOTU <- as.data.frame(otu_table(pseq))
pseqST <- as.data.frame(sample_data(subset_samples(pseq, Transit == "LT")))

pseqOTUST <- subset(pseqOTU, select = c(row.names(pseqST)))



full <- FALSE
.qualitative <- DirichletMultinomial:::.qualitative
dev.off <- function(...) invisible(grDevices::dev.off(...))

# Data
fl <- system.file(package="DirichletMultinomial", "extdata", "Twins.csv")
count <- t(as.matrix(read.csv(fl, row.names=1)))
 count[1:5, 1:3]

cnts <- log10(colSums(count))
densityplot(cnts, xlim=range(cnts),xlab="Taxon representation (log 10 count)")



if (full) {
fit <- mclapply(1:7, dmn, count=count, verbose=TRUE)
save(fit, file=file.path(tempdir(), "fit.rda"))
} else data(fit)
fit[[4]]



lplc <- sapply(fit, laplace)
plot(lplc, type="b", xlab="Number of Dirichlet Components",ylab="Model Fit")

(best <- fit[[which.min(lplc)]])



mixturewt(best)
head(mixture(best), 3)
splom(log(fitted(best)))


p0 <- fitted(fit[[1]], scale=TRUE) # scale by theta
p4 <- fitted(best, scale=TRUE)
colnames(p4) <- paste("m", 1:4, sep="")
(meandiff <- colSums(abs(p4 - as.vector(p0))))
sum(meandiff)



diff <- rowSums(abs(p4 - as.vector(p0)))
o <- order(diff, decreasing=TRUE)
cdiff <- cumsum(diff[o]) / sum(diff)
df <- head(cbind(Mean=p0[o], p4[o,], diff=diff[o], cdiff), 10)

heatmapdmn(count, fit[[1]], best, 30)




# Generative classifier
fl <- system.file(package="DirichletMultinomial", "extdata", "TwinStudy.t")
pheno0 <- scan(fl)
lvls <- c("Lean", "Obese", "Overwt")
pheno <- factor(lvls[pheno0 + 1], levels=lvls)
names(pheno) <- rownames(count)
table(pheno)


counts <- lapply(levels(pheno), csubset, count, pheno)
sapply(counts, dim)

keep <- c("Lean", "Obese")
count <- count[pheno %in% keep,]
pheno <- factor(pheno[pheno %in% keep], levels=keep)






```

# Mixomics - based on http://mixomics.org/mixdiablo/diablo-tcga-case-study/
## To begin
```{r Starting with Mixomics, fig.height=10, fig.width=10}
subsetcolon = T; subsetvessel = "P"
otus <- F

library(mixOmics) # import the mixOmics library
library(igraph)
# Check the dimensions. The distribution of class labels is also examined. Make sure that the rows are the same. Note that the dataframes in the data list should be named. 


# Extract genus table from phylobj
phylobj.QMP_stable <- subset_samples(phylobj.QMP, Ninetransit == 3)
phylobj.QMP_stable <- subset_samples(phylobj.prop, Ninetransit == 3)
phylobj.QMP_stable <- aggregate_taxa(phylobj.QMP_stable, "Genus")
# phylobj.QMP_stable <- subset_samples(phylobj.QMP.otu, Ninetransit == 3)
# phylobj.QMP_stable <- subset_samples(phylobj.prop.otu, Ninetransit == 3)

if(subsetcolon ==T){phylobj.QMP_stable <- subset_samples(phylobj.QMP_stable, PD_Colon == subsetvessel)}
# 
genus_table <- as.data.frame(t(otu_table(phylobj.QMP_stable)))




if(otus ==T){
  genusbu <- subset(genus_table, select = c("Otu00001"))
  for(i in unique(colnames(genus_table))){
  df <- subset(genus_table, select = c(i))

  m =i
  if(stringr::str_detect(m, "Otu0000")){
      y <- str_replace(m, "Otu0000", "OTU")
    }else{if(stringr::str_detect(m, "Otu000")){
      y <- str_replace(m, "Otu000", "OTU")
    }else{if(stringr::str_detect(m, "Otu00")){
      y <- str_replace(m, "Otu00", "OTU")
    }else{if(stringr::str_detect(m, "Otu0")){
      y <- str_replace(m, "Otu0", "OTU")
    }}}}

 colnames(df) <- c(y)
 genusbu <- cbind(genusbu, df)
  }
  genusbu <- subset(genusbu, select = -c(Otu00001))
genus_table <- genusbu
}



# Extract sampleinfo from phyloseq object
sampleinfo <- data.frame(sample_data(phylobj.QMP_stable))
if(subsetcolon ==T){sampleinfo <- subset(sampleinfo, PD_Colon == subsetvessel)}
# 



# Extract VFA data and melt with sampleinfo
SCFAdata <- read.csv2("netSCFAproduction.csv", sep = ",")
if(subsetcolon ==T){SCFAdata <- subset(SCFAdata, Vessel == subsetvessel)}

SCFAdata_table <- merge(sampleinfo, SCFAdata, by="Samplenumber")
row.names(SCFAdata_table) <- SCFAdata_table$Samplenumber
SCFAdata_table <- subset(SCFAdata_table, select = c("Acetate", "Propionate", "Isobutyrate", "Butyrate",
                                                    "Isovalerate", "Valerate", "Isocaproate", "Caproate", "Heptanoate",
                                                    "Octanoic.acid"))
test <- data.frame(sampleinfo, x = sampleinfo$Samplenumber %in% SCFAdata$Samplenumber)
test <- subset(test, x == F); warning(paste(test$Samplenumber, "is conflicting!"))


# Extract cell count tables and melt with sampleinfo
Countdata <- read.csv2("netcellrate.csv", sep = ",")
sampleinfo2 <- subset(sampleinfo, select = c(Samplenumber, Donor, Transit, PD_Colon, Date, Ninetransit, Day))
count_table <- merge(sampleinfo2, Countdata, by.x="Samplenumber", by.y = "Nr")
row.names(count_table) <- count_table$Samplenumber
count_table <- subset(count_table, select = c("meantot", "meanperc", "meanint"))
test <- data.frame(sampleinfo, x = sampleinfo$Samplenumber %in% Countdata$Nr)
test <- subset(test, x == F); warning(paste(test$Samplenumber, "is conflicting!"))

# Remove sample 114, because its missing in the SCFA data
genus_table <- rownames_to_column(genus_table)
genus_table <- subset(genus_table, rowname!=114)
row.names(genus_table) <- genus_table$rowname
genus_table <- subset(genus_table, select = -c(rowname))
sampleinfo <- subset(sampleinfo, Samplenumber !=114)



count_table <- rownames_to_column(count_table)
count_table <- subset(count_table, rowname!=114)
row.names(count_table) <- count_table$rowname
count_table <- subset(count_table, select = -c(rowname))


SCFAdata_table <- merge(sampleinfo, SCFAdata, by="Samplenumber")
row.names(SCFAdata_table) <- SCFAdata_table$Samplenumber
SCFAdata_table <- subset(SCFAdata_table, select = c("Acetate", "Propionate", "Isobutyrate", "Butyrate",
                                                    "Isovalerate", "Valerate", "Isocaproate", "Caproate", "Heptanoate",
                                                    "Octanoic.acid"))
# Re-check row names
row.names(sampleinfo) %in% row.names(genus_table)
row.names(SCFAdata_table) %in% row.names(genus_table)
row.names(SCFAdata_table) %in% row.names(sampleinfo)
row.names(count_table) %in% row.names(sampleinfo)
sampleinfo <- sampleinfo[order(as.numeric(row.names(sampleinfo))), ]
SCFAdata_table <- SCFAdata_table[order(as.numeric(row.names(SCFAdata_table))), ]
SCFAdata_table[] <- lapply(SCFAdata_table, as.numeric)
genus_table <- genus_table[order(as.numeric(row.names(genus_table))), ]
genus_table[] <- lapply(genus_table, as.numeric)
count_table <- count_table[order(as.numeric(row.names(count_table))), ]
count_table[] <- lapply(count_table, as.numeric)
## set a list of all the X dataframes
data = list(genera = genus_table, 
            cells = count_table,
            SCFA = SCFAdata_table
            )
test <- data.frame(genera = row.names(genus_table),SCFA = row.names(SCFAdata_table),cells = row.names(count_table))
## Check their dimension
lapply(data, dim) 

## Set the response variable as the Y df
Y = data.frame(Transit=sampleinfo$Transit, Donor = sampleinfo$Donor, PD_Colon = sampleinfo$PD_Colon, row = sampleinfo$Samplenumber)
row.names(Y) = Y$row; Y = subset(Y, select = -c(row))
Y = as.matrix(Y)
Y = sampleinfo$Transit
summary(Y)

result.diablo.tcga <- block.plsda(data, Y) # run the method
plotIndiv(result.diablo.tcga) # plot the samples
plotVar(result.diablo.tcga) # plot the variables
```


## Initial Analysis - Pairwise PLS Comparisons
As mentioned in the methods pages, it is strongly advised that prior to using the DIABLO framework that the data be examined in a non-integrative context. Here, the correlation between the top 25 features of each dataframe (in a pairwise fashion) are shown
```{r Initial Analysis - Pairwise PLS Comparisons, fig.height=10, fig.width=10}
list.keepSCFA = c(10, 10) # select arbitrary values of features to keep
if(otus == T){list.keepGenera = c(400, 400)}else{
list.keepGenera = c(25, 25)}
list.keepCell = c(3, 3)

# generate three pairwise PLS models
pls1 <- spls(data[["SCFA"]], data[["genera"]], keepX = list.keepSCFA, keepY = list.keepGenera) 
pls2 <- spls(data[["cells"]], data[["genera"]], keepX = list.keepCell, keepY = list.keepGenera)
pls3 <- spls(data[["SCFA"]], data[["cells"]], keepX = list.keepSCFA, keepY = list.keepCell)


# plot features of first PLS
plot1<- plotVar(pls1, cutoff = 0.5, title = "(a) SCFA vs genera", 
        legend = c("SCFA", "genera"), 
        var.names = FALSE, style = 'graphics', 
        pch = c(16, 17), cex = c(2,2), 
        col = c('darkorchid', 'lightgreen'))

plot2 <- plotVar(pls2, cutoff = 0.5, title = "(b) cells vs genera",
        legend = c("cells", "genera"),
        var.names = FALSE, style = 'graphics',
        pch = c(16, 17), cex = c(2,2),
        col = c('darkorchid', 'lightgreen'))

plot3 <- plotVar(pls3, cutoff = 0.5, title = "(c) SCFA vs cells",
        legend = c("SCFA", "cells"),
        var.names = FALSE, style = 'graphics',
        pch = c(16, 17), cex = c(2,2),
        col = c('darkorchid', 'lightgreen'))


# Correlations between the first component of each dataset. It can be seen the values are around ~0.8 â 0.9. Which is good
cor(pls1$variates$X, pls1$variates$Y) 
cor(pls2$variates$X, pls2$variates$Y)
cor(pls3$variates$X, pls3$variates$Y)
```

## Initial DIABLO Model
### Design
The moderate to high correlation between these features means that an appropriate value for the design matrix would be about ~0.8 â 0.9. However, as discussed in the N-Integration Methods page, values above 0.5 will cause a reduction in predictive ability of the model â and prediction is what is desired in this context. Hence, a value of 0.1 will be used to prioritise the discriminative ability of the model.
```{r Design}
# for square matrix filled with 0.1s
design = matrix(0.1, ncol = length(data), nrow = length(data), 
                dimnames = list(names(data), names(data)))
diag(design) = 0 # set diagonal to 0s

design

# With a design in place, the initial DIABLO model can be generated. An arbitrarily high number of components (ncomp = 5) will be used.
# form basic DIABLO model
basic.diablo.model = block.splsda(X = data, Y = Y, ncomp = 2, design = design)
```


### Tuning the number of components 
```{r Tuning the number of components}
# To choose the number of components for the final DIABLO model, the function perf() is run with 10-fold cross-validation repeated 10 times. 
# run component number tuning with repeated CV
perf.diablo = perf(basic.diablo.model, validation = 'Mfold', 
                   folds = 10, nrepeat = 10) 

# If you get this error, then it means that you : Error in solve.default(t(Pmat[, 1:x]) %*% Wmat[, 1:x]) : system is computationally singular: reciprocal condition number = 3.08032e-30


plot(perf.diablo) # plot output of tuning

# we observe that both overall and balanced error rate (BER) decrease from 1 to 2 components. The standard deviation indicates a potential slight gain in adding more components. The centroids.dist distance seems to give the best accuracy (see Supplemental Material in [6]). Considering this distance and the BER, the output $choice.ncomp indicates the optimal number of components for the final DIABLO model.

# set the optimal ncomp value
ncomp = perf.diablo$choice.ncomp$WeightedVote["Overall.BER", "centroids.dist"] 
ncomp =2
# show the optimal choice for ncomp for each dist metric
perf.diablo$choice.ncomp$WeightedVote
```


### Tuning the numbers feature - takes a while
This tuning function should be used to tune the keepX parameters in the block.splsda() function.

We choose the optimal number of variables to select in each data set using the tune.block.splsda() function, for a grid of keepX values for each type of omics. Note that the function has been set to favour a relatively small signature while allowing us to obtain a sufficient number of variables for downstream validation and/or interpretation. See ?tune.block.splsda.

The function tune is run with 10-fold cross validation, but repeated only once. Note that for a more thorough tuning process, provided sufficient computational time, we could increase the nrepeat argument. Here we have saved the results into an RData object that is available for download as the tuning can take a very long time, especially on lower end machines.
```{r Tuning the numbers feature - THIS TAKES LONG}
# set grid of values for each component to test
if(otus == T){test.keepX = list (genera = c(5:9, seq(112, 118, 2), seq(289,294,5)),
                   cells = c(1:3),
                   SCFA = c(1,3,6))}else{
                   test.keepX = list (genera = c(5:9, seq(12, 18, 2), seq(69,74,5)),
                   cells = c(1:3),
                   SCFA = c(1,3,6))
                   }
# run the feature selection tuning
tune.TCGA = tune.block.splsda(X = data, Y = Y, ncomp = ncomp, 
                              test.keepX = test.keepX, design = design,
                              validation = 'Mfold', folds = 10, nrepeat = 1,
                              dist = "centroids.dist" 
                              )

list.keepX = tune.TCGA$choice.keepX # set the optimal values of features to retain
list.keepX
```

### Final model
```{r}
# set the optimised DIABLO model
final.diablo.model = block.splsda(X = data, Y = Y, ncomp = ncomp, 
                          keepX = list.keepX, design = design)

final.diablo.model$design # design matrix for the final model


# The selected variables can be extracted with the function selectVar(), for example in the genera block, as seen below. Note that the stability of selected variables can be extracted from the output of the perf() function.
# the features selected to form the first component
selectVar(final.diablo.model, block = 'genera', comp = 1)$genera$name
selectVar(final.diablo.model, block = 'cells', comp = 1)$cells$name
selectVar(final.diablo.model, block = 'SCFA', comp = 1)$SCFA$name
```


### Plots
#### Sample plots
```{r fig.height=12, fig.width=12}
# Diagnostic plot to check whether the correlation between components from each data set has been maximised as specified in the design matrix. Specify which dimension to be assessed with the ncomp argument.
plotDiablo(final.diablo.model, ncomp = 2)

# Project each sample into the space spanned by the components of each block. Clustering of the samples can be better assessed with this plot.
plotIndiv(final.diablo.model, ind.names = FALSE, legend = TRUE, 
          title = 'DIABLO Sample Plots')

# In the arrow plot, the start of the arrow indicates the centroid between all data sets for a given sample and the tips of the arrows indicate the location of that sample in each block. Such graphics highlight the agreement between all data sets at the sample level. While somewhat difficult to interpret, even qualitatively.
plotArrow(final.diablo.model, ind.names = FALSE, legend = TRUE, 
          title = 'DIABLO')
```


#### Variable plots 
```{r Variable plots, fig.height=10, fig.width=10}
# The best starting point to evaluate the correlation structure between variables is with the correlation circle plot. A majority of the miRNA variables are positively correlated with the first component while the mRNA variables seem to separate along this dimension. These first two components correlate highly with the selected variables from the proteomics dataset. From this, the correlation of each selected feature from all three datasets can be evaluated based on their proximity.

plotVar(final.diablo.model, var.names = FALSE, 
        style = 'graphics', legend = TRUE,
        pch = c(16, 17, 15), cex = c(2,2,2), 
        col = c('darkorchid', 'brown1', 'lightgreen'))



# The circos plot is exclusive to integrative frameworks and represents the correlations between variables of different types, represented on the side quadrants. From Figure 7, it seems that the miRNA variables are almost entirely negatively correlated with the other two dataframes. The proteomics features are the opposite, such that they display primarily positive correlations while the mRNA variables are more mixed. Note that these correlations are above a value of 0.7 (cutoff = 0.7). All the interpretations made above are only relevant for features with very strong correlations. 
circosPlot(final.diablo.model, cutoff = 0.7, line = TRUE,
           color.blocks= c('darkorchid', 'brown1', 'lightgreen'),
           color.cor = c("darkgreen", "darkred"), size.labels = 1.5, size.variables = 0.9, 
           showIntraLinks = F, size.legend = 1.5)

library(circlize)


# Another visualisation of the correlations between the different types of variables is the relevance network, which is also built on the similarity matrix (as is the circos plot). Each colour represents a type of variable. Figure 8 shows this network which has a lower cutoff an Figure 7 (cutoff = 0.4). Two distinct clusters can be observed, though due to the density of the plot the relationships within the cluster are hard to determine. The interactive version of this plot would be useful here.
network(final.diablo.model, blocks = c(1,2,3),
        color.node = c('darkorchid', 'brown1', 'lightgreen'), cutoff = 0.4)



# The network can be saved in a .gml format to be input into the software Cytoscape, using the R package igraph. An example is shown directly below.
my.network = network(final.diablo.model, blocks = c(1,2,3),
        color.node = c('lightgreen', 'brown1', 'darkorchid'), cutoff = 0.4)
write.graph(my.network$gR, file = "myNetwork.gml", format = "gml")


# The function plotLoadings() visualises the loading weights of each selected variable on each component and each data set (Figure 9). The colour indicates the class in which the variable has the maximum level of expression (contrib = 'max') using the median (method = 'median'). Figure 9 depicts the loading values for the second dimension.
plotLoadings(final.diablo.model, comp = 1, contrib = 'max', method = 'median')




# The cimDIABLO() function is a clustered image map specifically implemented to represent the multi-omics molecular signature expression for each sample. From Figure 10, the areas of homogeneous expression levels for a set of samples across a set of features can be determined. For instance, the Her2 samples were the only group to show extremely high levels of expression for a specific set of proteins (seen by the small, red block in the middle bottom of Figure 10). This indicates these features are fairly discriminating for this subtype
cimDiablo(final.diablo.model)
```

```{r fig.height=15, fig.width=15}
my.network = network(final.diablo.model, blocks = c(1,2,3),
        color.node = c('lightgreen', 'brown1', 'darkorchid'), cutoff = 0.7)
```



### Performance of the model
We assess the performance of the model using 10-fold cross-validation repeated 10 times, using the function perf(). The method runs a block.splsda() model on the pre-specified arguments input from the final.diablo.model object, but on cross-validated samples. We then assess the accuracy of the prediction on the left-out samples.

In addition to the usual (balanced) classification error rates, predicted dummy variables and variates, as well as the stability of the selected features, the perf() function for DIABLO outputs the performance based on Majority Vote (each data set votes for a class for a particular test sample) or a weighted vote, where the weight is defined according to the correlation between the latent component associated to a particular data set and the outcome.

Since the tune() function was used with the centroid.dist argument, the outputs of the perf() function for that same distance are examined. The following code may take a few minutes to run.
```{r}
perf.diablo = perf(final.diablo.model, validation = 'Mfold', 
                   M = 10, nrepeat = 10, 
                   dist = 'centroids.dist') 

perf.diablo$MajorityVote.error.rate

perf.diablo$WeightedVote.error.rate


# An AUC plot per block can also be obtained using the function auroc(). The interpretation of this output may not be particularly insightful in relation to the performance evaluation of our methods, but can complement the statistical analysis.
auc.splsda = auroc(final.diablo.model, roc.block = "genera", roc.comp = 2, print = FALSE)
auc.splsda = auroc(final.diablo.model, roc.block = "SCFA", roc.comp = 2, print = FALSE)
auc.splsda = auroc(final.diablo.model, roc.block = "cells", roc.comp = 2, print = FALSE)
```



# Mixomics - NGS and SCFA -sPLS
## To begin
```{r Starting with Mixomics, fig.height=10, fig.width=10}
subsetcolon = T; subsetvessel = "P"
otus <- F

library(mixOmics) # import the mixOmics library
library(igraph)
# Check the dimensions. The distribution of class labels is also examined. Make sure that the rows are the same. Note that the dataframes in the data list should be named. 


# Extract genus table from phylobj
phylobj.QMP_stable <- subset_samples(phylobj.QMP, Ninetransit == 3)
phylobj.QMP_stable <- subset_samples(phylobj.prop, Ninetransit == 3)

topX_genera <- T
if(topX_genera == T){
phylobj.QMP_stable <- aggregate_top_taxa(phylobj.QMP_stable, top = 50, "Genus")}else{
  phylobj.QMP_stable <- aggregate_taxa(phylobj.QMP_stable, "Genus")
}
# 
# phylobj.QMP_stable <- subset_samples(phylobj.QMP.otu, Ninetransit == 3)
# phylobj.QMP_stable <- subset_samples(phylobj.prop.otu, Ninetransit == 3)

if(subsetcolon ==T){phylobj.QMP_stable <- subset_samples(phylobj.QMP_stable, PD_Colon == subsetvessel)}
# 
genus_table <- as.data.frame(t(otu_table(phylobj.QMP_stable)))




if(otus ==T){
  genusbu <- subset(genus_table, select = c("Otu00001"))
  for(i in unique(colnames(genus_table))){
  df <- subset(genus_table, select = c(i))

  m =i
  if(stringr::str_detect(m, "Otu0000")){
      y <- str_replace(m, "Otu0000", "OTU")
    }else{if(stringr::str_detect(m, "Otu000")){
      y <- str_replace(m, "Otu000", "OTU")
    }else{if(stringr::str_detect(m, "Otu00")){
      y <- str_replace(m, "Otu00", "OTU")
    }else{if(stringr::str_detect(m, "Otu0")){
      y <- str_replace(m, "Otu0", "OTU")
    }}}}

 colnames(df) <- c(y)
 genusbu <- cbind(genusbu, df)
  }
  genusbu <- subset(genusbu, select = -c(Otu00001))
genus_table <- genusbu
}



# Extract sampleinfo from phyloseq object
sampleinfo <- data.frame(sample_data(phylobj.QMP_stable))
if(subsetcolon ==T){sampleinfo <- subset(sampleinfo, PD_Colon == subsetvessel)}
# 



# Extract VFA data and melt with sampleinfo
SCFAdata <- read.csv2("netSCFAproduction.csv", sep = ",")
if(subsetcolon ==T){SCFAdata <- subset(SCFAdata, Vessel == subsetvessel)}

SCFAdata_table <- merge(sampleinfo, SCFAdata, by="Samplenumber")
row.names(SCFAdata_table) <- SCFAdata_table$Samplenumber
SCFAdata_table <- subset(SCFAdata_table, select = c("Acetate", "Propionate", "Isobutyrate", "Butyrate",
                                                    "Isovalerate", "Valerate", "Isocaproate", "Caproate", "Heptanoate",
                                                    "Octanoic.acid"))
test <- data.frame(sampleinfo, x = sampleinfo$Samplenumber %in% SCFAdata$Samplenumber)
test <- subset(test, x == F); warning(paste(test$Samplenumber, "is conflicting!"))


# Extract SCFA relative to nutrient load
SCFA_loaddata <- read.csv2("SCFA_load.csv", sep = ",")

SCFAloaddata_table <- merge(sampleinfo, SCFA_loaddata, by="Samplenumber")
row.names(SCFAloaddata_table) <- SCFAloaddata_table$Samplenumber
SCFAloaddata_table <- subset(SCFAloaddata_table, select = c("Acet_load",   "Prop_load",    
"Buty_load",     "Val_load",      "Cap_load"   ,   "Hep_load"  ,    "Oct_load" ,     "Isoval_load" ,  "Isobut_load"))
test <- data.frame(sampleinfo, x = sampleinfo$Samplenumber %in% SCFA_loaddata$Samplenumber)
test <- subset(test, x == F); warning(paste(test$Samplenumber, "is conflicting!"))

# Extract cell count tables and melt with sampleinfo
Countdata <- read.csv2("netcellrate.csv", sep = ",")
sampleinfo2 <- subset(sampleinfo, select = c(Samplenumber, Donor, Transit, PD_Colon, Date, Ninetransit, Day))
count_table <- merge(sampleinfo2, Countdata, by.x="Samplenumber", by.y = "Nr")
row.names(count_table) <- count_table$Samplenumber
count_table <- subset(count_table, select = c("meantot", "meanperc", "meanint"))
test <- data.frame(sampleinfo, x = sampleinfo$Samplenumber %in% Countdata$Nr)
test <- subset(test, x == F); warning(paste(test$Samplenumber, "is conflicting!"))

# Remove sample 114, because its missing in the SCFA data
genus_table <- rownames_to_column(genus_table)
genus_table <- subset(genus_table, rowname!=114)
row.names(genus_table) <- genus_table$rowname
genus_table <- subset(genus_table, select = -c(rowname))
sampleinfo <- subset(sampleinfo, Samplenumber !=114)

SCFAloaddata_table <- rownames_to_column(SCFAloaddata_table)
SCFAloaddata_table <- subset(SCFAloaddata_table,rowname!=114)

count_table <- rownames_to_column(count_table)
count_table <- subset(count_table, rowname!=114)
row.names(count_table) <- count_table$rowname
count_table <- subset(count_table, select = -c(rowname))


SCFAdata_table <- merge(sampleinfo, SCFAdata, by="Samplenumber")
row.names(SCFAdata_table) <- SCFAdata_table$Samplenumber
SCFAdata_table <- subset(SCFAdata_table, select = c("Acetate", "Propionate", "Isobutyrate", "Butyrate",
                                                    "Isovalerate", "Valerate", "Isocaproate", "Caproate", "Heptanoate",
                                                    "Octanoic.acid"))


SCFAloaddata_table <- merge(sampleinfo, SCFA_loaddata, by="Samplenumber")
row.names(SCFAloaddata_table) <- SCFAloaddata_table$Samplenumber
SCFAloaddata_table <- subset(SCFAloaddata_table, select = c("Acet_load",   "Prop_load",    
"Buty_load",     "Val_load",      "Cap_load"   ,   "Hep_load"  ,    "Oct_load" ,     "Isoval_load" ,  "Isobut_load"))
# Re-check row names
row.names(sampleinfo) %in% row.names(genus_table)
row.names(SCFAdata_table) %in% row.names(genus_table)
row.names(SCFAdata_table) %in% row.names(sampleinfo)
row.names(count_table) %in% row.names(sampleinfo)
sampleinfo <- sampleinfo[order(as.numeric(row.names(sampleinfo))), ]
SCFAdata_table <- SCFAdata_table[order(as.numeric(row.names(SCFAdata_table))), ]
SCFAdata_table[] <- lapply(SCFAdata_table, as.numeric)

SCFAloaddata_table <- SCFAloaddata_table[order(as.numeric(row.names(SCFAloaddata_table))), ]
SCFAloaddata_table[] <- lapply(SCFAloaddata_table, as.numeric)

genus_table <- genus_table[order(as.numeric(row.names(genus_table))), ]
genus_table[] <- lapply(genus_table, as.numeric)
count_table <- count_table[order(as.numeric(row.names(count_table))), ]
count_table[] <- lapply(count_table, as.numeric)
## set a list of all the X dataframes
data = list(genera = genus_table, 
            SCFA_load = SCFAloaddata_table,
            SCFA = SCFAdata_table)
test <- data.frame(genera = row.names(genus_table),
                   SCFAload = row.names(SCFAloaddata_table),
                   SCFA = row.names(SCFAdata_table))
## Check their dimension
lapply(data, dim) 

## Set the response variable as the Y df
Y = data.frame(Transit=sampleinfo$Transit, Donor = sampleinfo$Donor, PD_Colon = sampleinfo$PD_Colon, row = sampleinfo$Samplenumber)
row.names(Y) = Y$row; Y = subset(Y, select = -c(row))
Y = as.matrix(Y)
Y = sampleinfo$Transit
summary(Y)
sampledata <- sampleinfo

result.diablo.tcga <- block.plsda(data, Y) # run the method
plotIndiv(result.diablo.tcga) # plot the samples
plotVar(result.diablo.tcga) # plot the variables

# 
X <- data$genera
Y <- data$SCFA
# Y <- data$SCFA_load
```


## Initial Analysis - Preliminary Analysis with PCA
Both datasets should be explored prior to any form of analysis. This will aid in making decisions in constructing the best model possible. PCA is an unsupervised, exploratory method that is useful here. When using PCA on these datasets, both should be centered and scaled due to the varying scales of each variables (especially the Y dataframe). It is preferable to first run a PCA with a large number of components (i.e. ncomp = 10). Then, use the âelbowâ (a sudden drop) on the barplot to choose the final number of PCs. Further information can be found in the PCA Methods Page.
```{r Initial Analysis - Pairwise PLS Comparisons, fig.height=10, fig.width=10}
set.seed(0604)
# Trim dataframes because:
# Error: columns with zero variance in 'X': 1,3,5,7,25,35,46,47,48,51,65,80. Remove these columns before scaling.
if(topX_genera == F){
X <- subset(data$genera, select = -c(1,3,5,7,25,35,46,47,48,51,65,80))
# X <- subset(X, select = -c(Other))
}else{
  X <- subset(data$genera, select = -c(5))
  X <- subset(X, select = -c(Other))
}
# 
# Error: columns with zero variance in 'X': 7,10. Remove these columns before scaling.
Y <- subset(data$SCFA, select = -c(7,10))


pca.genera <- pca(X, ncomp = 10, center = TRUE, scale = TRUE)
pca.SCFA <- pca(Y, ncomp = 8, center = TRUE, scale = TRUE)

plot(pca.genera)
plot(pca.SCFA)


# The explained variance of each Principal Components for both datasets can be seen in Figure 1. In both cases, use of the 'elbow method' indicates two components would be sufficient. Beyond this, the added components provide little novel information. The next step is to assess the clustering of samples in the Principal Component subspace. 
plotIndiv(pca.genera, comp = c(1, 2), 
          group = sampledata$Transit, 
          ind.names = row.names(sampledata), 
          legend = TRUE, title = 'Genera, PCA comp 1 - 2')

plotIndiv(pca.SCFA, comp = c(1, 2), 
          group = sampledata$Transit, 
          ind.names = row.names(sampledata), 
          legend = TRUE, title = 'net SCFA production, PCA comp 1 - 2')


# Initial sPLS model
# A basic sPLS model needs to be created such that it can be optimised and tuned.Note that as no keepX or keepY parameters are passed into the function, it is equivalent to the pls() function for now. The regression mode of sPLS is used as the gene expression data is being attempting to be used to explain the clinical data. 
# spls <- spls(X = X, Y = Y, ncomp = 5, mode = 'regression')  


# TUNING SPLS
# Selecting number of components
  ## repeated CV tuning of component count
  # perf.spls <- perf(spls, validation = 'Mfold',
  #                          folds = 10, nrepeat = 5)
  
 t <- plot(perf.spls, criterion = 'Q2.total')
  
# Selecting the number of variables 
  ## set range of test values for number of variables to use from X dataframe
  list.keepX <- c(seq(15, 35, 5))
  ## set range of test values for number of variables to use from Y dataframe
  list.keepY <- c(3:8) 
  
  
  if(topX_genera == T){
    df.select <- c(1:26, 28,30, 31,32, 33, 34,36,39,40)
  }
  
  
spls <- spls(X = subset(X, select = c(1:20, 22)), 
             Y = Y, ncomp = 5, mode = 'regression'); perf.spls <- perf(spls, validation = 'Mfold',
                            folds = 10, nrepeat = 5)


  tune.spls <- tune.spls(subset(X, select = df.select), 
                         Y, ncomp = 3, test.keepX = list.keepX,
                                test.keepY = list.keepY,
                                nrepeat = 1, folds = 10, # use 10 folds
                                mode = 'regression', measure = 'cor') 
# 1:26, 28,30, 31,32, 33, 34,36,39,40, 41, 42, 43,44,45

  plot(tune.spls)         # use the correlation measure for tuning  
  # The output of this tuning can be seen in Figure 4. For every grid value of keepX and keepY, the averaged correlation coefficients between the t and u components (latent variables) are shown across repeated cross validation folds. The value is depicted by circle size. Optimal values (here corresponding to the highest mean correlation) for each dimension and data set are indicated by the green square.

  # The optimal number of features to use for both datasets can be extracted through the below calls.
  tune.spls$choice.keepX
  tune.spls$choice.keepY
  
  # extract optimal number of variables for X dataframe
  optimal.keepX <- tune.spls$choice.keepX 
  
  # extract optimal number of variables for Y datafram
  optimal.keepY <- tune.spls$choice.keepY
  
  optimal.ncomp <-  length(optimal.keepX) # extract optimal number of components

# FINAL MODEL
  # use all tuned values from above
  final.spls.liver <- spls(data$genera, Y, ncomp = optimal.ncomp, 
                      keepX = optimal.keepX,
                      keepY = optimal.keepY,
                      mode = "regression") # explanitory approach being used, 
                                           # hence use regression mode
  
 
```


## Plot
```{r fig.height=13, fig.width=13}
plotIndiv(final.spls.liver, ind.names = FALSE, 
         rep.space = "X-variate", # plot in X-variate subspace
         group = sampledata$Transit, # colour by time group
         pch = sampledata$Donor, 
         col.per.group = color.mixo(1:3), 
         legend = TRUE, legend.title = 'Transit', legend.title.pch = 'Donor')

plotIndiv(final.spls.liver, ind.names = FALSE, 
         rep.space = "Y-variate", # plot in X-variate subspace
         group = sampledata$Transit, # colour by time group
         pch = sampledata$Donor, 
         col.per.group = color.mixo(1:3), 
         legend = TRUE, legend.title = 'Transit', legend.title.pch = 'Donor')


# colours
ramp = colorRampPalette(c("green", "darkgreen", "black", "darkred", "red"))
    ramp = ramp(101)
    green = ramp[1:43]
    red = ramp[59:101]
    ramp = colorRamp(c(red, "black",green), space = "Lab")
    
    rgb(ramp(seq(0, 1, length = 50)), 
        # alpha = 0, 
        maxColorValue = 255)
    

color.edge <- color.GreenRed(50)  # set the colours of the connecting lines
# color.edge <- color.jet(50)  # set the colours of the connecting lines
 

network <- network(final.spls.liver, comp = 1:2,
        cutoff = 0.7, # only show connections with a correlation above 0.7
        shape.node = c("rectangle", "circle"),
        # color.node = c("cyan", "pink"),
        color.node = c("orange","lightblue"),
        show.edge.labels = T,
        lwd.edge = c(3, 3),
        cex.edge.label = 4,
              # interactive = T,
        color.edge = rgb(ramp(seq(0, 1, length = 50)), maxColorValue = 255),
      layout.fun = layout_as_tree
        )


saveplots()
network <- network(final.spls.liver, comp = 1:2,
        cutoff = 0.55, # only show connections with a correlation above 0.7
        shape.node = c("rectangle", "circle"),
        # color.node = c("cyan", "pink"),
        color.node = c("orange","lightblue"),
        show.edge.labels = T,
        lwd.edge = c(3, 3),
        cex.edge.label = 4,
              # interactive = T,
        color.edge = rgb(ramp(seq(0, 1, length = 50)), maxColorValue = 255),
      layout.fun = layout_as_tree,
      save = 'tiff', # save as a png to the current working directory
        name.save = 'sPLS_Network_Plot_DC'
        )


```



# Mixomics - OTU NGS and SCFA -sPLS
## To begin
```{r Starting with Mixomics, fig.height=10, fig.width=10}
subsetcolon = T; subsetvessel = "P"
otus <- T

library(mixOmics) # import the mixOmics library
library(igraph)
# Check the dimensions. The distribution of class labels is also examined. Make sure that the rows are the same. Note that the dataframes in the data list should be named. 


# Extract genus table from phylobj
phylobj.QMP_stable <- subset_samples(phylobj.QMP.otu, Ninetransit == 3)
# phylobj.QMP_stable <- subset_samples(phylobj.prop, Ninetransit == 3)

topX_genera <- F
if(otus ==F){
if(topX_genera == T){
phylobj.QMP_stable <- aggregate_top_taxa(phylobj.QMP_stable, top = 50, "Genus")}else{
  phylobj.QMP_stable <- aggregate_taxa(phylobj.QMP_stable, "Genus")
}}
# 
# phylobj.QMP_stable <- subset_samples(phylobj.QMP.otu, Ninetransit == 3)
# phylobj.QMP_stable <- subset_samples(phylobj.prop.otu, Ninetransit == 3)

if(subsetcolon ==T){phylobj.QMP_stable <- subset_samples(phylobj.QMP_stable, PD_Colon == subsetvessel)}
# 
genus_table <- as.data.frame(t(otu_table(phylobj.QMP_stable)))




if(otus ==T){
  genusbu <- subset(genus_table, select = c("Otu00001"))
  for(i in unique(colnames(genus_table))){
  df <- subset(genus_table, select = c(i))

  m =i
  if(stringr::str_detect(m, "Otu0000")){
      y <- str_replace(m, "Otu0000", "OTU")
    }else{if(stringr::str_detect(m, "Otu000")){
      y <- str_replace(m, "Otu000", "OTU")
    }else{if(stringr::str_detect(m, "Otu00")){
      y <- str_replace(m, "Otu00", "OTU")
    }else{if(stringr::str_detect(m, "Otu0")){
      y <- str_replace(m, "Otu0", "OTU")
    }}}}

 colnames(df) <- c(y)
 genusbu <- cbind(genusbu, df)
  }
  genusbu <- subset(genusbu, select = -c(Otu00001))
genus_table <- genusbu
}



# Extract sampleinfo from phyloseq object
sampleinfo <- data.frame(sample_data(phylobj.QMP_stable))
if(subsetcolon ==T){sampleinfo <- subset(sampleinfo, PD_Colon == subsetvessel)}
# 



# Extract VFA data and melt with sampleinfo
SCFAdata <- read.csv2("netSCFAproduction.csv", sep = ",")
if(subsetcolon ==T){SCFAdata <- subset(SCFAdata, Vessel == subsetvessel)}

SCFAdata_table <- merge(sampleinfo, SCFAdata, by="Samplenumber")
row.names(SCFAdata_table) <- SCFAdata_table$Samplenumber
SCFAdata_table <- subset(SCFAdata_table, select = c("Acetate", "Propionate", "Isobutyrate", "Butyrate",
                                                    "Isovalerate", "Valerate", "Isocaproate", "Caproate", "Heptanoate",
                                                    "Octanoic.acid"))
test <- data.frame(sampleinfo, x = sampleinfo$Samplenumber %in% SCFAdata$Samplenumber)
test <- subset(test, x == F); warning(paste(test$Samplenumber, "is conflicting!"))


# Extract SCFA relative to nutrient load
SCFA_loaddata <- read.csv2("SCFA_load.csv", sep = ",")

SCFAloaddata_table <- merge(sampleinfo, SCFA_loaddata, by="Samplenumber")
row.names(SCFAloaddata_table) <- SCFAloaddata_table$Samplenumber
SCFAloaddata_table <- subset(SCFAloaddata_table, select = c("Acet_load",   "Prop_load",    
"Buty_load",     "Val_load",      "Cap_load"   ,   "Hep_load"  ,    "Oct_load" ,     "Isoval_load" ,  "Isobut_load"))
test <- data.frame(sampleinfo, x = sampleinfo$Samplenumber %in% SCFA_loaddata$Samplenumber)
test <- subset(test, x == F); warning(paste(test$Samplenumber, "is conflicting!"))

# Extract cell count tables and melt with sampleinfo
Countdata <- read.csv2("netcellrate.csv", sep = ",")
sampleinfo2 <- subset(sampleinfo, select = c(Samplenumber, Donor, Transit, PD_Colon, Date, Ninetransit, Day))
count_table <- merge(sampleinfo2, Countdata, by.x="Samplenumber", by.y = "Nr")
row.names(count_table) <- count_table$Samplenumber
count_table <- subset(count_table, select = c("meantot", "meanperc", "meanint"))
test <- data.frame(sampleinfo, x = sampleinfo$Samplenumber %in% Countdata$Nr)
test <- subset(test, x == F); warning(paste(test$Samplenumber, "is conflicting!"))

# Remove sample 114, because its missing in the SCFA data
genus_table <- rownames_to_column(genus_table)
genus_table <- subset(genus_table, rowname!=114)
row.names(genus_table) <- genus_table$rowname
genus_table <- subset(genus_table, select = -c(rowname))
sampleinfo <- subset(sampleinfo, Samplenumber !=114)

SCFAloaddata_table <- rownames_to_column(SCFAloaddata_table)
SCFAloaddata_table <- subset(SCFAloaddata_table,rowname!=114)

count_table <- rownames_to_column(count_table)
count_table <- subset(count_table, rowname!=114)
row.names(count_table) <- count_table$rowname
count_table <- subset(count_table, select = -c(rowname))


SCFAdata_table <- merge(sampleinfo, SCFAdata, by="Samplenumber")
row.names(SCFAdata_table) <- SCFAdata_table$Samplenumber
SCFAdata_table <- subset(SCFAdata_table, select = c("Acetate", "Propionate", "Isobutyrate", "Butyrate",
                                                    "Isovalerate", "Valerate", "Isocaproate", "Caproate", "Heptanoate",
                                                    "Octanoic.acid"))


SCFAloaddata_table <- merge(sampleinfo, SCFA_loaddata, by="Samplenumber")
row.names(SCFAloaddata_table) <- SCFAloaddata_table$Samplenumber
SCFAloaddata_table <- subset(SCFAloaddata_table, select = c("Acet_load",   "Prop_load",    
"Buty_load",     "Val_load",      "Cap_load"   ,   "Hep_load"  ,    "Oct_load" ,     "Isoval_load" ,  "Isobut_load"))
# Re-check row names
row.names(sampleinfo) %in% row.names(genus_table)
row.names(SCFAdata_table) %in% row.names(genus_table)
row.names(SCFAdata_table) %in% row.names(sampleinfo)
row.names(count_table) %in% row.names(sampleinfo)
sampleinfo <- sampleinfo[order(as.numeric(row.names(sampleinfo))), ]
SCFAdata_table <- SCFAdata_table[order(as.numeric(row.names(SCFAdata_table))), ]
SCFAdata_table[] <- lapply(SCFAdata_table, as.numeric)

SCFAloaddata_table <- SCFAloaddata_table[order(as.numeric(row.names(SCFAloaddata_table))), ]
SCFAloaddata_table[] <- lapply(SCFAloaddata_table, as.numeric)

genus_table <- genus_table[order(as.numeric(row.names(genus_table))), ]
genus_table[] <- lapply(genus_table, as.numeric)
count_table <- count_table[order(as.numeric(row.names(count_table))), ]
count_table[] <- lapply(count_table, as.numeric)
## set a list of all the X dataframes
data = list(genera = genus_table, 
            SCFA_load = SCFAloaddata_table,
            SCFA = SCFAdata_table)
test <- data.frame(genera = row.names(genus_table),
                   SCFAload = row.names(SCFAloaddata_table),
                   SCFA = row.names(SCFAdata_table))
## Check their dimension
lapply(data, dim) 

## Set the response variable as the Y df
Y = data.frame(Transit=sampleinfo$Transit, Donor = sampleinfo$Donor, PD_Colon = sampleinfo$PD_Colon, row = sampleinfo$Samplenumber)
row.names(Y) = Y$row; Y = subset(Y, select = -c(row))
Y = as.matrix(Y)
Y = sampleinfo$Transit
summary(Y)
sampledata <- sampleinfo

result.diablo.tcga <- block.plsda(data, Y) # run the method
plotIndiv(result.diablo.tcga) # plot the samples
plotVar(result.diablo.tcga) # plot the variables

# 
X <- data$genera
Y <- data$SCFA
# Y <- data$SCFA_load
```


## Initial Analysis - Preliminary Analysis with PCA
Both datasets should be explored prior to any form of analysis. This will aid in making decisions in constructing the best model possible. PCA is an unsupervised, exploratory method that is useful here. When using PCA on these datasets, both should be centered and scaled due to the varying scales of each variables (especially the Y dataframe). It is preferable to first run a PCA with a large number of components (i.e. ncomp = 10). Then, use the âelbowâ (a sudden drop) on the barplot to choose the final number of PCs. Further information can be found in the PCA Methods Page.
```{r Initial Analysis - Pairwise PLS Comparisons, fig.height=10, fig.width=10}
set.seed(0604)
# Trim dataframes because:
# Error: columns with zero variance in 'X': 1,3,5,7,25,35,46,47,48,51,65,80. Remove these columns before scaling.


    X <- subset(data$genera, select =
    -c(26,39,85,96,101,106,133,157,158,191,193,197,199,203,207,211,213,215,216,236,255,270,278,279,286,289,291,294,295,304,305,309,314,322,332,339,341,349,351,355,357,362,363,381,388,390,393,396,400))

    X <- subset(X, select =
    -c(26,39,85,95,105,132,156,157,192,195,197,201,205,209,211,213,214,253,267,276,284,287,292,293,302,304,307,312,315,319,329,336,338,348,351,352,357,358,376,383,385,388,391,395))
# 
# Error: columns with zero variance in 'X': 7,10. Remove these columns before scaling.
Y <- subset(data$SCFA, select = -c(7,10))


pca.genera <- pca(X, ncomp = 10, center = TRUE, scale = TRUE)
pca.SCFA <- pca(Y, ncomp = 8, center = TRUE, scale = TRUE)

plot(pca.genera)
plot(pca.SCFA)


# The explained variance of each Principal Components for both datasets can be seen in Figure 1. In both cases, use of the 'elbow method' indicates two components would be sufficient. Beyond this, the added components provide little novel information. The next step is to assess the clustering of samples in the Principal Component subspace. 
plotIndiv(pca.genera, comp = c(1, 2), 
          group = sampledata$Transit, 
          ind.names = row.names(sampledata), 
          legend = TRUE, title = 'Genera, PCA comp 1 - 2')

plotIndiv(pca.SCFA, comp = c(1, 2), 
          group = sampledata$Transit, 
          ind.names = row.names(sampledata), 
          legend = TRUE, title = 'net SCFA production, PCA comp 1 - 2')


# Initial sPLS model
# A basic sPLS model needs to be created such that it can be optimised and tuned.Note that as no keepX or keepY parameters are passed into the function, it is equivalent to the pls() function for now. The regression mode of sPLS is used as the gene expression data is being attempting to be used to explain the clinical data. 
# spls <- spls(X = X, Y = Y, ncomp = 5, mode = 'regression')  


# TUNING SPLS
# Selecting number of components
  ## repeated CV tuning of component count
 #  perf.spls <- perf(spls, validation = 'Mfold',
 #                           folds = 10, nrepeat = 5)
 #  
 # t <- plot(perf.spls, criterion = 'Q2.total')
  
# Selecting the number of variables 
  ## set range of test values for number of variables to use from X dataframe
  list.keepX <- c(seq(15, 35, 5))
  ## set range of test values for number of variables to use from Y dataframe
  list.keepY <- c(3:8) 
  
  
  
    df.select <- c(1:26, 28,30, 31,32, 33, 34,36,39,40)

  
  
spls <- spls(X = subset(X, select = c(1:20, 22)), 
             Y = Y, ncomp = 5, mode = 'regression'); perf.spls <- perf(spls, validation = 'Mfold',
                            folds = 10, nrepeat = 5)


  tune.spls <- tune.spls(subset(X, select = df.select), 
                         Y, ncomp = 3, test.keepX = list.keepX,
                                test.keepY = list.keepY,
                                nrepeat = 1, folds = 10, # use 10 folds
                                mode = 'regression', measure = 'cor') 
# 1:26, 28,30, 31,32, 33, 34,36,39,40, 41, 42, 43,44,45

  plot(tune.spls)         # use the correlation measure for tuning  
  # The output of this tuning can be seen in Figure 4. For every grid value of keepX and keepY, the averaged correlation coefficients between the t and u components (latent variables) are shown across repeated cross validation folds. The value is depicted by circle size. Optimal values (here corresponding to the highest mean correlation) for each dimension and data set are indicated by the green square.

  # The optimal number of features to use for both datasets can be extracted through the below calls.
  tune.spls$choice.keepX
  tune.spls$choice.keepY
  
  # extract optimal number of variables for X dataframe
  optimal.keepX <- tune.spls$choice.keepX 
  
  # extract optimal number of variables for Y datafram
  optimal.keepY <- tune.spls$choice.keepY
  
  optimal.ncomp <-  length(optimal.keepX) # extract optimal number of components

# FINAL MODEL
  # use all tuned values from above
  final.spls.liver <- spls(data$genera, Y, ncomp = optimal.ncomp, 
                      keepX = optimal.keepX,
                      keepY = optimal.keepY,
                      mode = "regression") # explanitory approach being used, 
                                           # hence use regression mode
  
 
```


## Plot
```{r fig.height=13, fig.width=13}
plotIndiv(final.spls.liver, ind.names = FALSE, 
         rep.space = "X-variate", # plot in X-variate subspace
         group = sampledata$Transit, # colour by time group
         pch = sampledata$Donor, 
         col.per.group = color.mixo(1:3), 
         legend = TRUE, legend.title = 'Transit', legend.title.pch = 'Donor')

plotIndiv(final.spls.liver, ind.names = FALSE, 
         rep.space = "Y-variate", # plot in X-variate subspace
         group = sampledata$Transit, # colour by time group
         pch = sampledata$Donor, 
         col.per.group = color.mixo(1:3), 
         legend = TRUE, legend.title = 'Transit', legend.title.pch = 'Donor')


# colours
ramp = colorRampPalette(c("green", "darkgreen", "black", "darkred", "red"))
    ramp = ramp(101)
    green = ramp[1:43]
    red = ramp[59:101]
    ramp = colorRamp(c(red, "black",green), space = "Lab")
    
    rgb(ramp(seq(0, 1, length = 50)), 
        # alpha = 0, 
        maxColorValue = 255)
    

color.edge <- color.GreenRed(50)  # set the colours of the connecting lines
# color.edge <- color.jet(50)  # set the colours of the connecting lines
 

network <- network(final.spls.liver, comp = 1:2,
        cutoff = 0.55, # only show connections with a correlation above 0.7
        shape.node = c("rectangle", "circle"),
        # color.node = c("cyan", "pink"),
        color.node = c("orange","lightblue"),
        show.edge.labels = T,
        lwd.edge = c(3, 3),
        cex.edge.label = 4,
              # interactive = T,
        color.edge = rgb(ramp(seq(0, 1, length = 50)), maxColorValue = 255),
      layout.fun = layout_as_tree
        )

saveplots()
network <- network(final.spls.liver, comp = 1:2,
        cutoff = 0.55, # only show connections with a correlation above 0.7
        shape.node = c("rectangle", "circle"),
        # color.node = c("cyan", "pink"),
        color.node = c("orange","lightblue"),
        show.edge.labels = T,
        lwd.edge = c(3, 3),
        cex.edge.label = 4,
              # interactive = T,
        color.edge = rgb(ramp(seq(0, 1, length = 50)), maxColorValue = 255),
      layout.fun = layout_as_tree,
      save = 'tiff', # save as a png to the current working directory
        name.save = 'sPLS_Network_Plot_DC'
        )


```

play with igraph
```{r fig.height=15, fig.width=15}

network.res <- network(final.spls.liver)
write.graph(network.res$gR, file = "network.gml", format = "gml")
```








play with igraph
```{r fig.height=15, fig.width=15}

network.res <- network(final.spls.liver)
write.graph(network.res$gR, file = "network.gml", format = "gml")
```




# Mixomics - NGS and SCFA -rCCA
## To begin
```{r Starting with Mixomics, fig.height=10, fig.width=10}
subsetcolon = T; subsetvessel = "P"
otus <- F

library(mixOmics) # import the mixOmics library
library(igraph)
# Check the dimensions. The distribution of class labels is also examined. Make sure that the rows are the same. Note that the dataframes in the data list should be named. 


# Extract genus table from phylobj
phylobj.QMP_stable <- subset_samples(phylobj.QMP, Ninetransit == 3)
phylobj.QMP_stable <- subset_samples(phylobj.prop, Ninetransit == 3)

topX_genera <- T
if(topX_genera == T){
phylobj.QMP_stable <- aggregate_top_taxa(phylobj.QMP_stable, top = 50, "Genus")}else{
  phylobj.QMP_stable <- aggregate_taxa(phylobj.QMP_stable, "Genus")
}
# 
# phylobj.QMP_stable <- subset_samples(phylobj.QMP.otu, Ninetransit == 3)
# phylobj.QMP_stable <- subset_samples(phylobj.prop.otu, Ninetransit == 3)

if(subsetcolon ==T){phylobj.QMP_stable <- subset_samples(phylobj.QMP_stable, PD_Colon == subsetvessel)}
# 
genus_table <- as.data.frame(t(otu_table(phylobj.QMP_stable)))




if(otus ==T){
  genusbu <- subset(genus_table, select = c("Otu00001"))
  for(i in unique(colnames(genus_table))){
  df <- subset(genus_table, select = c(i))

  m =i
  if(stringr::str_detect(m, "Otu0000")){
      y <- str_replace(m, "Otu0000", "OTU")
    }else{if(stringr::str_detect(m, "Otu000")){
      y <- str_replace(m, "Otu000", "OTU")
    }else{if(stringr::str_detect(m, "Otu00")){
      y <- str_replace(m, "Otu00", "OTU")
    }else{if(stringr::str_detect(m, "Otu0")){
      y <- str_replace(m, "Otu0", "OTU")
    }}}}

 colnames(df) <- c(y)
 genusbu <- cbind(genusbu, df)
  }
  genusbu <- subset(genusbu, select = -c(Otu00001))
genus_table <- genusbu
}



# Extract sampleinfo from phyloseq object
sampleinfo <- data.frame(sample_data(phylobj.QMP_stable))
if(subsetcolon ==T){sampleinfo <- subset(sampleinfo, PD_Colon == subsetvessel)}
# 



# Extract VFA data and melt with sampleinfo
SCFAdata <- read.csv2("netSCFAproduction.csv", sep = ",")
if(subsetcolon ==T){SCFAdata <- subset(SCFAdata, Vessel == subsetvessel)}

SCFAdata_table <- merge(sampleinfo, SCFAdata, by="Samplenumber")
row.names(SCFAdata_table) <- SCFAdata_table$Samplenumber
SCFAdata_table <- subset(SCFAdata_table, select = c("Acetate", "Propionate", "Isobutyrate", "Butyrate",
                                                    "Isovalerate", "Valerate", "Isocaproate", "Caproate", "Heptanoate",
                                                    "Octanoic.acid"))
test <- data.frame(sampleinfo, x = sampleinfo$Samplenumber %in% SCFAdata$Samplenumber)
test <- subset(test, x == F); warning(paste(test$Samplenumber, "is conflicting!"))


# Extract SCFA relative to nutrient load
SCFA_loaddata <- read.csv2("SCFA_load.csv", sep = ",")

SCFAloaddata_table <- merge(sampleinfo, SCFA_loaddata, by="Samplenumber")
row.names(SCFAloaddata_table) <- SCFAloaddata_table$Samplenumber
SCFAloaddata_table <- subset(SCFAloaddata_table, select = c("Acet_load",   "Prop_load",    
"Buty_load",     "Val_load",      "Cap_load"   ,   "Hep_load"  ,    "Oct_load" ,     "Isoval_load" ,  "Isobut_load"))
test <- data.frame(sampleinfo, x = sampleinfo$Samplenumber %in% SCFA_loaddata$Samplenumber)
test <- subset(test, x == F); warning(paste(test$Samplenumber, "is conflicting!"))

# Extract cell count tables and melt with sampleinfo
Countdata <- read.csv2("netcellrate.csv", sep = ",")
sampleinfo2 <- subset(sampleinfo, select = c(Samplenumber, Donor, Transit, PD_Colon, Date, Ninetransit, Day))
count_table <- merge(sampleinfo2, Countdata, by.x="Samplenumber", by.y = "Nr")
row.names(count_table) <- count_table$Samplenumber
count_table <- subset(count_table, select = c("meantot", "meanperc", "meanint"))
test <- data.frame(sampleinfo, x = sampleinfo$Samplenumber %in% Countdata$Nr)
test <- subset(test, x == F); warning(paste(test$Samplenumber, "is conflicting!"))

# Remove sample 114, because its missing in the SCFA data
genus_table <- rownames_to_column(genus_table)
genus_table <- subset(genus_table, rowname!=114)
row.names(genus_table) <- genus_table$rowname
genus_table <- subset(genus_table, select = -c(rowname))
sampleinfo <- subset(sampleinfo, Samplenumber !=114)

SCFAloaddata_table <- rownames_to_column(SCFAloaddata_table)
SCFAloaddata_table <- subset(SCFAloaddata_table,rowname!=114)

count_table <- rownames_to_column(count_table)
count_table <- subset(count_table, rowname!=114)
row.names(count_table) <- count_table$rowname
count_table <- subset(count_table, select = -c(rowname))


SCFAdata_table <- merge(sampleinfo, SCFAdata, by="Samplenumber")
row.names(SCFAdata_table) <- SCFAdata_table$Samplenumber
SCFAdata_table <- subset(SCFAdata_table, select = c("Acetate", "Propionate", "Isobutyrate", "Butyrate",
                                                    "Isovalerate", "Valerate", "Isocaproate", "Caproate", "Heptanoate",
                                                    "Octanoic.acid"))


SCFAloaddata_table <- merge(sampleinfo, SCFA_loaddata, by="Samplenumber")
row.names(SCFAloaddata_table) <- SCFAloaddata_table$Samplenumber
SCFAloaddata_table <- subset(SCFAloaddata_table, select = c("Acet_load",   "Prop_load",    
"Buty_load",     "Val_load",      "Cap_load"   ,   "Hep_load"  ,    "Oct_load" ,     "Isoval_load" ,  "Isobut_load"))
# Re-check row names
row.names(sampleinfo) %in% row.names(genus_table)
row.names(SCFAdata_table) %in% row.names(genus_table)
row.names(SCFAdata_table) %in% row.names(sampleinfo)
row.names(count_table) %in% row.names(sampleinfo)
sampleinfo <- sampleinfo[order(as.numeric(row.names(sampleinfo))), ]
SCFAdata_table <- SCFAdata_table[order(as.numeric(row.names(SCFAdata_table))), ]
SCFAdata_table[] <- lapply(SCFAdata_table, as.numeric)

SCFAloaddata_table <- SCFAloaddata_table[order(as.numeric(row.names(SCFAloaddata_table))), ]
SCFAloaddata_table[] <- lapply(SCFAloaddata_table, as.numeric)

genus_table <- genus_table[order(as.numeric(row.names(genus_table))), ]
genus_table[] <- lapply(genus_table, as.numeric)
count_table <- count_table[order(as.numeric(row.names(count_table))), ]
count_table[] <- lapply(count_table, as.numeric)
## set a list of all the X dataframes
data = list(genera = genus_table, 
            SCFA_load = SCFAloaddata_table,
            SCFA = SCFAdata_table)
test <- data.frame(genera = row.names(genus_table),
                   SCFAload = row.names(SCFAloaddata_table),
                   SCFA = row.names(SCFAdata_table))
## Check their dimension
lapply(data, dim) 

## Set the response variable as the Y df
Y = data.frame(Transit=sampleinfo$Transit, Donor = sampleinfo$Donor, PD_Colon = sampleinfo$PD_Colon, row = sampleinfo$Samplenumber)
row.names(Y) = Y$row; Y = subset(Y, select = -c(row))
Y = as.matrix(Y)
Y = sampleinfo$Transit
summary(Y)
sampledata <- sampleinfo

result.diablo.tcga <- block.plsda(data, Y) # run the method
plotIndiv(result.diablo.tcga) # plot the samples
plotVar(result.diablo.tcga) # plot the variables

# 
X <- data$genera
Y <- data$SCFA
# Y <- data$SCFA_load


imgCor(X, Y, sideColors = c("purple", "green"))
```


## Tuning rCCA
Both datasets should be explored prior to any form of analysis. This will aid in making decisions in constructing the best model possible. PCA is an unsupervised, exploratory method that is useful here. When using PCA on these datasets, both should be centered and scaled due to the varying scales of each variables (especially the Y dataframe). It is preferable to first run a PCA with a large number of components (i.e. ncomp = 10). Then, use the âelbowâ (a sudden drop) on the barplot to choose the final number of PCs. Further information can be found in the PCA Methods Page.
```{r Initial Analysis - Pairwise PLS Comparisons, fig.height=10, fig.width=10}
set.seed(0604)
# Trim dataframes because:
# Error: columns with zero variance in 'X': 1,3,5,7,25,35,46,47,48,51,65,80. Remove these columns before scaling.
if(topX_genera == F){
X <- subset(data$genera, select = -c(1,3,5,7,25,35,46,47,48,51,65,80))
# X <- subset(X, select = -c(Other))
}else{
  X <- subset(data$genera, select = -c(5))
  X <- subset(X, select = -c(Other))
}
# 
# Error: columns with zero variance in 'X': 7,10. Remove these columns before scaling.
Y <- subset(data$SCFA, select = -c(7,10))


# set grid search values for each regularisation parameter
grid1 <- seq(0.001, 0.2, length = 10) 
grid2 <- seq(0.001, 0.2, length = 10)

# optimise the regularisation parameter values
cv.tune.rcc.nutrimouse <- tune.rcc(X, Y, grid1 = grid1, grid2 = grid2, 
                                   validation = "loo") 
  
cv.tune.rcc.nutrimouse  


opt.l1 <- cv.tune.rcc.nutrimouse$opt.lambda1 # extract the optimal lambda values
opt.l2 <- cv.tune.rcc.nutrimouse$opt.lambda2

# formed optimised CV rCCA
CV.rcc.nutrimouse <- rcc(X, Y, method = "ridge", 
                         lambda1 = opt.l1, lambda2 = opt.l2) 


# run the rCCA method using shrinkage
shrink.rcc.nutrimouse <- rcc(X,Y, method = 'shrinkage') 
# examine the optimal lambda values after shrinkage 
shrink.rcc.nutrimouse$lambda 


# barplot of cross validation method rCCA canonical correlations
plot(CV.rcc.nutrimouse, type = "barplot", main = "Cross Validation") 

# barplot of shrinkage method rCCA canonical correlations
plot(shrink.rcc.nutrimouse, type = "barplot", main = "Shrinkage") 
```


## Plot
```{r fig.height=13, fig.width=13}
network(CV.rcc.nutrimouse, comp = 1:2, interactive = FALSE,
        lwd.edge = 3,
        shape.node = c("rectangle", "circle"),
        # color.node = c("cyan", "pink"),
        color.node = c("orange","lightblue"),
        show.edge.labels = T,
        # lwd.edge = c(3, 3),
        cex.edge.label = 4,
              # interactive = T,
        color.edge = rgb(ramp(seq(0, 1, length = 50)), maxColorValue = 255),
      layout.fun = layout_as_tree,
        cutoff = 0.55)

network <- network(final.spls.liver, comp = 1:2,
        cutoff = 0.55, # only show connections with a correlation above 0.7
        shape.node = c("rectangle", "circle"),
        # color.node = c("cyan", "pink"),
        color.node = c("orange","lightblue"),
        show.edge.labels = T,
        lwd.edge = c(3, 3),
        cex.edge.label = 4,
              # interactive = T,
        color.edge = rgb(ramp(seq(0, 1, length = 50)), maxColorValue = 255),
      layout.fun = layout_as_tree,
      save = 'tiff', # save as a png to the current working directory
        name.save = 'sPLS_Network_Plot_DC'
        )


```

play with igraph
```{r fig.height=15, fig.width=15}

network.res <- network(final.spls.liver)
write.graph(network.res$gR, file = "network.gml", format = "gml")
```




# Operate from phyloseq objects
## Heatmaps
```{r Plot heatmap, fig.width=20, include=FALSE}
# Manipulate dataframe for heatmaps
df.heatmap <- aggregate_top_taxa(phylobj.prop, top= 15, "Genus")
df.heatmap <- psmelt(df.heatmap)
df.heatmap <- subset(df.heatmap, Ninetransit == 3)
# df.heatmap$Nutrient_Load.Day <- factor(df.heatmap$Nutrient_Load.Day, ordered = TRUE)
df.heatmap$Abundance <- as.numeric(df.heatmap$Abundance)
# Create heatmaps between donors with as columns PD_Colons
for (i in c("1", "2", "3", "4", "5", "6")){
df.heatmap2 <- subset(df.heatmap, Donor == i)
# df.heatmap3 <- subset(df.heatmap2, PD_Colon == "1")
plot <- ggplot(df.heatmap2, aes(Transit, OTU)) +
  geom_tile(aes(fill = Abundance),colour = "white") + 
  scale_fill_gradient(low = "white",high = "steelblue") +
  facet_grid(~PD_Colon, labeller = labeller(PD_Colon = PD_Colon.labsPC, Donor= donor.labs)) +
  ggtitle(paste("Donor:", i)) 
plot <- papertheme(plot, 20)
plot <- plot + theme(strip.background =element_rect(fill="#D6E0F4"),
        strip.text.x = element_text(size = 20),
        strip.text.y = element_text(size = 20),
        axis.text=element_text(size=15),
        legend.text=element_text(size=15),
        legend.title = element_text(size = 15, face= "bold"),
        legend.title.align = 0.5,
        legend.box.background = element_rect())
print(plot)
}
f2 = colorRamp2(seq(min(0), max(100), length = 3), c("blue", "#EEEEEE", "red"), 
    space = "RGB")
# Create complex heatmaps
for (i in c("1", "2", "3", "4", "5", "6")){
df.heatmap2 <- subset(df.heatmap, Donor == i)
for (j in c("P", "D")){
df.heatmap3 <- subset(df.heatmap2, PD_Colon == j)
df.heatmap3<-  acast(df.heatmap3, OTU~Nutrient_Load.Day, value.var="Abundance")
plot <- Heatmap(df.heatmap3, cluster_columns = FALSE, col = f2, row_names_rot = 0, row_names_side = "left", name = "%", row_title = "OTU", column_title = "Days", row_dend_side = "right", column_title_side = "bottom") 
print (plot)
}}
rm(df.heatmap, df.heatmap2, plot)
library(vegan)
data(varespec)
pc <- rda(varespec)
biplot(pc) # species scores as biplot arrows
plot(envfit(pc ~ Pleuschr + Cladarbu + Cladrang + Cladstel, varespec))
ordisurf(pc ~ Cladstel, varespec, knots = 1, add = TRUE)
```

## Faecal samples
```{r FS, fig.width=20, fig.height=10, include = FALSE}
Top25 <- aggregate_top_taxa(phylobj.QMPFS, top= 15, "Genus")
# Top25 <- aggregate_top_taxa(phylobj.QMPFS, top= 15, "Family")
Top25 <- subset_samples(Top25, Donor == 3 & PD_Colon == "P")
taxa_names(Top25)
plot <- plot_bar(Top25, fill="Genus", x= "Ninetransit" 
                 # title="Absolute abundance top 15 genera"
                 )  
# plot <- plot_bar(Top25, fill="Family", x= "Ninetransit" 
#                  # title="Absolute abundance top 15 genera"
#                  )  
plot <- papertheme(plot)
newtab = data.table(plot$data)
newtab$Donor <- factor(newtab$Donor, levels = c(1,2,3,4,5,6))
plot$data <- newtab
plot1 <- plot + scale_shape_manual(values=c(15, 16)) +
  geom_bar(aes(color=Genus, fill=Genus), stat="identity", position="stack") +
  facet_grid(.~Donor, labeller = labeller(PD_Colon = PD_Colon.labsPC, Donor= donor.labs)) +
  scale_fill_manual(values = colorRampPalette(brewer.pal(16,"Set3"))(16)) +
  scale_color_manual(values = colorRampPalette(brewer.pal(16,"Set3"))(16)) +
  # scale_fill_manual(values = desired) +
  # scale_color_manual(values = desired) +
  labs(color = "Nutrient_Load", shape = "Phase", x = "", y =  expression(paste("Abundance (cells m", L^{"-1"},")"))) +
  guides(color = FALSE) +
  # scale_x_discrete(labels = c("100","33","20","10")) +
  # scale_y_continuous(labels=scaleFUN)+
  # annotate("rect", xmin="0", xmax="5", ymin=-Inf, ymax=Inf, alpha=0.2, fill="#d18fa9") +
  theme(strip.background =element_rect(fill="#D6E0F4"),
        strip.text.x = element_text(size = 20),
        strip.text.y = element_text(size = 20),
        axis.text=element_text(size=15),
        axis.text.x = element_blank(),
        legend.text=element_text(size=15),
        legend.title = element_text(size = 15, face= "bold"),
        legend.title.align = 0.5, 
        legend.box.background = element_rect())
print(plot1)
plotA <- plot1

phylobj.prop.3 = transform_sample_counts(phylobj.QMPFS, function(x) 100 * x/sum(x))
Top25 <- aggregate_top_taxa(phylobj.prop.3, top= 15, "Genus")
Top25 <- subset_samples(Top25, Donor == 3 & PD_Colon == "P")
plot <- plot_bar(Top25, fill="Genus", x= "Ninetransit" 
                 # title="Absolute abundance top 15 genera"
                 )  
plot <- papertheme(plot)
newtab = data.table(plot$data)
newtab$Donor <- factor(newtab$Donor, levels = c(1,2,3,4,5,6))
plot$data <- newtab
plot1 <- plot + scale_shape_manual(values=c(15, 16)) +
  geom_bar(aes(color=Genus, fill=Genus), stat="identity", position="stack") +
  facet_grid(.~Donor, labeller = labeller(PD_Colon = PD_Colon.labsPC, Donor= donor.labs)) +
  scale_fill_manual(values = colorRampPalette(brewer.pal(16,"Set3"))(16)) +
  scale_color_manual(values = colorRampPalette(brewer.pal(16,"Set3"))(16)) +
  # scale_fill_manual(values = desired) +
  # scale_color_manual(values = desired) +
  labs(color = "Nutrient_Load", shape = "Phase", x = "", y =  "Proportional abundance (%)") +
  guides(color = FALSE) +
  # scale_x_discrete(labels = c("100","33","20","10")) +
  # scale_y_continuous(labels=scaleFUN)+
  # annotate("rect", xmin="0", xmax="5", ymin=-Inf, ymax=Inf, alpha=0.2, fill="#d18fa9") +
  theme(strip.background =element_rect(fill="#D6E0F4"),
        strip.text.x = element_text(size = 20),
        strip.text.y = element_text(size = 20),
        axis.text=element_text(size=15),
        axis.text.x = element_blank(),
        legend.text=element_text(size=15),
        legend.title = element_text(size = 15, face= "bold"),
        legend.title.align = 0.5, 
        legend.box.background = element_rect())
print(plot1)
# ggsave(plot = plot1, width = 14,4, height = 9, dpi = 300, filename = "QMP.genus.png")



ggarrange(plotA, plot1, ncol = 2, common.legend = T, legend = "bottom")


```

## Percentage OTU per genus
```{r OTU per genus, fig.height=10, fig.width=16}
`%nin%` = Negate(`%in%`)
library(genefilter)
phylo <- subset_samples(phylobj.prop.otu, Nutrient_Load != "FS")
taxtable <- data.frame(tax_table(phylo))
otutable <- data.frame(otu_table(phylo))
combinedtable <- cbind(taxtable, otutable)
test2 <- subset(combinedtable, combinedtable$Genus == "Clostridium_XlVa")
  test <- test2[,c(7:ncol(test2))]
  for(col in names(test)) {
  test[paste0(col, "_pct")] = test[col] / sum(test[col])
  }
  test3 <- test[,c(154:ncol(test))]
  test3 <- test3 *100
  # test3[is.na(test3)] <- 1
  test3$SD <- rowSds(test3, na.rm=TRUE)
  test3$mean=rowMeans(test3, na.rm=TRUE)
  test3$genus <- rep("Clostridium_XlVa",nrow(test3))
  cols <- ncol(test3)-2;  test3 <- test3[,c(cols:ncol(test3))]
  combinedtable2 <- test3
  
test2 <- subset(combinedtable, combinedtable$Phylum == "Actinobacteria")
  test <- test2[,c(7:ncol(test2))]
  for(col in names(test)) {
  test[paste0(col, "_pct")] = test[col] / sum(test[col])
  }
  test3 <- test[,c(154:ncol(test))]
  test3 <- test3 *100
  # test3[is.na(test3)] <- 1
  test3$SD <- rowSds(test3, na.rm=TRUE)
  test3$mean=rowMeans(test3, na.rm=TRUE)
  test3$phylum <- rep("Actinobacteria",nrow(test3))
  cols <- ncol(test3)-2;  test3 <- test3[,c(cols:ncol(test3))]
  combinedtable2 <- test3  
  
for (i in unique(combinedtable$Genus)){
  test2 <- subset(combinedtable, combinedtable$Genus == i)
  test <- test2[,c(7:ncol(test2))]
  for(col in names(test)) {
  test[paste0(col, "_pct")] = test[col] / sum(test[col])
  }
  test3 <- test[,c(154:ncol(test))]
  test3 <- test3 *100
  # test3[is.na(test3)] <- 1
  test3$SD <- rowSds(test3, na.rm=TRUE)
  test3$mean=rowMeans(test3, na.rm=TRUE)
  test3$genus <- rep(i,nrow(test3))
  cols <- ncol(test3)-2;  test3 <- test3[,c(cols:ncol(test3))]
  if(i != "Clostridium_XlVa"){
    combinedtable2 <- rbind(combinedtable2, test3)
    }
  }
combinedtable2$mean <- format(round(combinedtable2$mean, 2), nsmall = 2)
combinedtable2$SD <- format(round(combinedtable2$SD, 2), nsmall = 2)
combinedtable2$text <- paste0(combinedtable2$mean, " Â± ", combinedtable2$SD) 
view(combinedtable2)
combinedtable2 <- rownames_to_column(combinedtable2)
```

## Species richness
```{r Species richness}
my_comparisons <- list(c("ST", "MT"), c("MT", "LT"), c("ST", "LT"))
physeqobj.meta.s <- subset_samples(phylobj.agg, Ninetransit == 3)
physeqobj.meta.s@sam_data$Transit <- factor(physeqobj.meta.s@sam_data$Transit,ordered=T,levels=c("ST","MT","LT"))
physeqobj.meta.s@sam_data$PD_Colon <- factor(physeqobj.meta.s@sam_data$PD_Colon,ordered=T,levels=c("P","D"))
test <- estimate_richness(physeqobj.meta.s, measures=c("Shannon", NULL));
test <- rownames_to_column(test)
test <- merge(test, sampledata, by.x="rowname", by.y = "Samplenumber",all.x=TRUE)
plot <- papertheme(ggplot(test, aes(x = Transit, y = Shannon, color = Transit))+ geom_boxplot()+   facet_grid(PD_Colon ~ Donor, 
      labeller = labeller(PD_Colon = PD_Colon.labsPC, Donor = donor.labs), scale = "free_x")) + scale_color_manual(values = transitcolors) + 
  theme(strip.background = element_rect(fill = "#D6E0F4"), strip.text.x = element_text(size = 20),
    strip.text.y = element_text(size = 20, angle = 270), axis.text.x = element_blank(),
    axis.ticks.x = element_blank(), axis.title.x = element_blank(), legend.text = element_text(size = 15,
        face = "italic"), legend.title = element_text(size = 15, face = "bold"),
    legend.title.align = 0.5) + xlab("Transit time") +
  stat_compare_means(comparisons = my_comparisons, label = "p.signif", hide.ns = T, hide.NS = T) # ; plot
plot
```


# Operate from dataframe
## Graphs
### mean per transit
```{r Plot graphs from dataframe means - DONT RUN, fig.height=10, fig.width=16, include=TRUE}
saveplots()
# formatR::tidy_app()
# Generate labels
diverging.genera <- c("1_P_Bilophila", "1_D_Akkermansia", "1_D_Bilophila", 
    "2_P_Enterobacteriaceae", "3_P_Enterobacteriaceae", "3_D_Enterobacteriaceae", 
    "3_D_Mitsuokella", "5_P_Bilophila", "5_P_Enterobacteriaceae", 
    "5_D_Akkermansia", "5_D_Bilophila", "5_D_Enterobacteriaceae", 
    "6_P_Bilophila", "6_D_Bilophila")
Genus.labs <- c("Donor 1; DC", "Donor 5; DC", "Donor 1; PC", "Donor 5; PC", 
    "Donor 6; PC", "Donor 1; DC", "Donor 5; DC", "Donor 6; DC", "Donor 2; PC", 
    "Donor 3; PC", "Donor 5; PC", "Donor 3; DC", "Donor 5; DC", "Donor 3; DC")
names(Genus.labs) <- c("1_D_Akkermansia", "5_D_Akkermansia", "1_P_Bilophila", 
    "5_P_Bilophila", "6_P_Bilophila", "1_D_Bilophila", "5_D_Bilophila", 
    "6_D_Bilophila", "2_P_Enterobacteriaceae", "3_P_Enterobacteriaceae", 
    "5_P_Enterobacteriaceae", "3_D_Enterobacteriaceae", "5_D_Enterobacteriaceae", 
    "3_D_Mitsuokella")
Genus2.labs <- c("Donor 1; PC", "Donor 1; DC", "Donor 2; PC", "Donor 3; PC", 
    "Donor 3; DC", "Donor 5; PC", "Donor 5; DC", "Donor 6; PC", "Donor 6; DC")
names(Genus2.labs) <- c("1_P", "1_D", "2_P", "3_P", "3_D", "5_P", "5_D", "6_P", "6_D")
qual_col_pals = brewer.pal.info[brewer.pal.info$category == 'qual',]
col_vector = unlist(mapply(brewer.pal, qual_col_pals$maxcolors, rownames(qual_col_pals)))
# Make a data frame with a column for the read counts of each sample
phylobj.QMP_stable <- subset_samples(phylobj.QMP, Ninetransit == 3)
Top15phylobj <- aggregate_top_taxa(phylobj.QMP_stable, top= 25, "Genus")
Top15 <- psmelt(Top15phylobj); dftestQMP <- Top15
unique(Top15$OTU)
unique(Top15$OTU)%in%Top15levels; x <- data.frame(x = unique(Top15$OTU)%in%Top15levels, genus = unique(Top15$OTU))  #check if labelled
Top15$OTU <- as.ordered(factor(Top15$OTU,levels=Top15levels))
# Make means
Top15.mean <- Top15
Top15.mean <- group_by_at(Top15.mean, vars(Donor, Transit, PD_Colon, Genus))
Top15.mean <- summarize(Top15.mean,SD = sd(Abundance), Abundance = mean(Abundance, na.rm = T), median = median (Abundance))
Top15.mean$Genus <- as.ordered(factor(Top15.mean$Genus,levels=Top15levels))
Top15.mean$Transit <- as.ordered(factor(Top15.mean$Transit,levels=c("ST", "MT", "LT")))
Top15.mean$PD_Colon <- as.ordered(factor(Top15.mean$PD_Colon,levels=c("P", "D")))
Top15.mean$Donor <- as.ordered(factor(Top15.mean$Donor,levels=c(1,2,3,4,5,6)))
# Make Top15.means dataframe in printable and readable format
Top15.mean.print <- subset(Top15.mean, Top15.mean$Donor == "1")
Top15.mean.print$median <- NULL
Top15.mean.print <- Top15.mean.print[c(1,2,3,4,6,5)]
for (i in 2:6){
Top15.meanx <- subset(Top15.mean, Top15.mean$Donor == i)
Top15.meanx$Nutrient_Load <- Top15.meanx$PD_Colon <- Top15.meanx$Genus <- Top15.meanx$median <- NULL
names(Top15.meanx)[names(Top15.meanx) == 'Abundance'] <- paste("Abundance,",i)
names(Top15.meanx)[names(Top15.meanx) == 'SD'] <- paste("SD,",i)
Top15.mean.print <- cbind(Top15.mean.print, Top15.meanx)}
unique(Top15.mean$Genus)
# Plot average abundance
bplot <- ggplot(data = Top15.mean, aes(x = Top15.mean$Transit, y = as.numeric(Abundance), fill = factor(Genus))) +
  geom_bar(position="stack", stat="identity") 
bplot <- bplot + scale_x_discrete(labels = c("S","M","L"))
bplot <- papertheme(bplot,20)
bplot <- bplot + guides(color = guide_legend("Donor")) +
  xlab("Transit time") + ylab(expression(paste("Absolute abundance (cells m", L^{"-1"},")"))) +
  theme(strip.background =element_rect(fill="#D6E0F4")) +
  guides(color = guide_legend("SCFA"), fill = guide_legend("")) +
  ggsci::scale_fill_npg() +
  # scale_fill_manual(values=desired) +
  scale_fill_manual(values=col_vector) +
  # scale_color_manual(values=desired,labels = Top15levels) +
  theme(strip.background =element_rect(fill="#D6E0F4")) +
  guides(fill=guide_legend(nrow=5)) +
  theme(
    legend.text = element_text(face = "italic"),
    strip.text.x = element_text(size = 20),
    strip.text.y = element_text(size = 20, angle = 270)) +
  facet_grid(PD_Colon~Donor,labeller = labeller(PD_Colon = PD_Colon.labsPC, Donor= donor.labs))
bplot
# ggsave(plot = bplot, width = 16, height = 10, dpi = 300, filename = "Paperplot_QMP.png")
# Plot average abundance in geom_area plot
plot <- ggplot(Top15, aes(x=as.numeric(Top15$Day), y=Top15$Abundance, fill=Top15$OTU)) +
  geom_area(aes(color=OTU, fill=OTU), stat="identity", position="stack") 
plot <- papertheme(plot)
plot <- plot + scale_shape_manual(values=c(15, 16)) +
  facet_grid(PD_Colon~Donor, labeller = labeller(PD_Colon = PD_Colon.labsPC, Donor= donor.labs)) +
  scale_fill_manual(values = desired) +
  scale_color_manual(values = desired) +
  labs(color = "Nutrient_Load", fill= "", shape = "Phase", x = "Nutrient load (%)", y =  expression(paste("Absolute abundance (cells m", L^{"-1"},")"))) +
  guides(color = FALSE) +
  # scale_x_discrete(labels = c("100","33","20","10")) +
  theme(strip.background =element_rect(fill="#D6E0F4"),
        strip.text.x = element_text(size = 20),
        strip.text.y = element_text(size = 20),
        axis.text=element_text(size=15),
        legend.text=element_text(size=15),
        legend.title = element_text(size = 15, face= "bold"),
        legend.title.align = 0.5, 
        legend.box.background = element_rect())
print(plot)
# Proportional
Top15phylobj <- aggregate_top_taxa(phylobj.QMP, top= 15, "Genus")
# Top15.prop = transform_sample_counts(Top15phylobj, function(x) 100 * x/sum(x))
Top15 <- psmelt(Top15phylobj) 
unique(Top15$OTU)
Top15$OTU <- as.ordered(factor(Top15$OTU,levels=Top15levels))
plot <- ggplot(Top15, aes(x=Transit, y=Abundance, fill=OTU)) +
  geom_bar(aes(color=OTU, fill=OTU), stat="identity", position="fill") 
plot <- papertheme(plot)
plot <- plot + scale_shape_manual(values=c(15, 16)) +
  facet_grid(PD_Colon~Donor, labeller = labeller(PD_Colon = PD_Colon.labsPC, Donor= donor.labs)) +
  scale_fill_manual(values = desired) +
  scale_color_manual(values = desired) +
  scale_y_continuous(labels = percent_format()) +
  labs(color = "Nutrient_Load", fill= "", shape = "Phase", x = "Nutrient load (%)", y = "Relative abundance (%)") +
  guides(color = FALSE) +
  scale_x_discrete(labels = c("100","33","20","10")) +
  theme(strip.background =element_rect(fill="#D6E0F4"),
        strip.text.x = element_text(size = 20),
        strip.text.y = element_text(size = 20, angle = 270),
        axis.text=element_text(size=15),
        legend.text=element_text(size=15),
        legend.title = element_text(size = 15, face= "bold"),
        legend.title.align = 0.5 
        # legend.box.background = element_rect()
        )
print(plot)
ggsave(plot = plot, width = 16, height = 10, dpi = 300, filename = "Paperplot_Relative.png")
# 
for (i in Top15levels) {
  df <- subset(Top15.mean, Top15.mean$Genus == i)
plot <- ggplot(df, aes(x=Transit, y=Abundance, fill=Genus)) +
geom_bar(position="stack", stat="identity") + 
  geom_errorbar(aes(ymin = Abundance - SD, ymax = Abundance + SD, width=0.1), alpha = 0.8)
plot <- papertheme(plot)
plot <- plot + scale_shape_manual(values=c(15, 16)) +
  facet_grid(PD_Colon~Donor, labeller = labeller(PD_Colon = PD_Colon.labsPC, Donor= donor.labs)) +
  scale_fill_manual(values = desired) +
  scale_color_manual(values = desired) +
  labs(color = "Nutrient_Load", fill= "", shape = "Phase", x = "Nutrient load (%)", y =  expression(paste("Absolute abundance (cells m", L^{"-1"},")"))) +
  guides(color = FALSE) +
  scale_x_discrete(labels = c("100","33","20","10")) +
  theme(strip.background =element_rect(fill="#D6E0F4"),
        strip.text.x = element_text(size = 20),
        strip.text.y = element_text(size = 20),
        axis.text=element_text(size=15),
        legend.text=element_text(size=15),
        legend.title = element_text(size = 15, face= "bold"),
        legend.title.align = 0.5,
        legend.box.background = element_rect())
# print(plot)
# ggsave(plot = plot, width = 16, height = 10, dpi = 300, filename = paste("Change_", i,".jpg"))
}
# Plot diverging genera in one plot
Top15.mean$combo <- paste(Top15.mean$Donor, Top15.mean$PD_Colon, Top15.mean$Genus, sep="_")
df <- subset(Top15.mean, Top15.mean$combo == "1_D_Akkermansia" |Top15.mean$combo =="5_D_Akkermansia"  |Top15.mean$combo =="1_P_Bilophila" |Top15.mean$combo == "5_P_Bilophila" |Top15.mean$combo =="6_P_Bilophila" |Top15.mean$combo =="1_D_Bilophila" |Top15.mean$combo =="5_D_Bilophila" |Top15.mean$combo =="6_D_Bilophila" |Top15.mean$combo =="2_P_Enterobacteriaceae" |Top15.mean$combo =="3_P_Enterobacteriaceae"  |Top15.mean$combo == "5_P_Enterobacteriaceae"  |Top15.mean$combo =="3_D_Enterobacteriaceae"  |Top15.mean$combo =="5_D_Enterobacteriaceae"  |Top15.mean$combo =="3_D_Mitsuokella")
df$combo <- as.ordered(factor(df$combo,levels=diverging.genera))
df$combo2 <- paste(df$Donor, df$PD_Colon, sep="_")
plot <- ggplot(df, aes(x=df$Nutrient_Load, y=df$Abundance, fill=df$Genus)) +
# geom_bar(position="stack", stat="identity") + 
  geom_bar(position=position_dodge(), stat="identity") + 
  geom_errorbar(aes(ymin = Abundance - SD, ymax = Abundance + SD, width=0.1), alpha = 0.8, position = position_dodge(width = 0.9))
plot <- papertheme(plot)
plot <- plot + scale_y_continuous(breaks= seq(0,7.5e8, by = 0.25e9),limits = c(0, 7.5e8)) +
  scale_fill_manual(values = desired) +
  scale_color_manual(values = desired) +
  labs(color = "Nutrient_Load", fill= "", shape = "Phase", x = "Nutrient load (%)", y =  expression(paste("Absolute abundance (cells m", L^{"-1"},")"))) +
  guides(color = FALSE) +
  scale_x_discrete(labels = c("100","33","20","10")) +
  theme(strip.background =element_rect(fill="#D6E0F4"),
        strip.text.x = element_text(size = 20), strip.text.y = element_text(size = 20),axis.text=element_text(size=15),
        legend.text=element_text(size=15), legend.title = element_text(size = 15, face= "bold"), legend.title.align = 0.5) +
  facet_wrap(vars(combo2), ncol = 5,
             # labeller = labeller(combo = Genus.labs)
             labeller = labeller(combo2 = Genus2.labs)
             )
print(plot)
ggsave(plot = plot, width = 16, height = 10, dpi = 300, filename = "Paperplot_Diverging.Genera.jpg")
rm(Top15, Top15f, Top15.mean.print, Top15.meanx, Top15.mean, Genus.labs, Genus2.labs)


```

### Figure3 - Paperplot mean per day
#

```{r Paperplot proportional abundance, fig.height=11, fig.width=17, include=TRUE}
saveplots()
# formatR::tidy_app()
# Make a data frame with a column for the read counts of each sample
phylobj.QMP_stable <- subset_samples(phylobj.QMP, Ninetransit == 3)
Top15phylobj <- aggregate_top_taxa(phylobj.QMP_stable, top = 20, "Genus")
Top15 <- psmelt(Top15phylobj)
unique(Top15$OTU)
unique(Top15$OTU) %in% Top15levels
x <- data.frame(x = unique(Top15$OTU) %in% Top15levels, genus = unique(Top15$OTU)) #check if labelled
x$genus <- as.ordered(factor(x$genus, levels = Top15levels))
Top15$OTU <- as.ordered(factor(Top15$OTU, levels = Top15levels))
sort(unique(x$genus))
desired = c("Acidaminococcus"="#DA5724",  "Akkermansia"="#94cfd5", "Alistipes"="#3F4921",
            "Bacteroides"="#DAA1AD", "Bifidobacterium"="#CBD588", "Bilophila"="#5F7FC7", 
            # "Blautia"="#673770", "Eisenbergiella" = "#508578","Megasphaera"= "#6DDE88",
            # "Roseburia"="#8A7C64", "Sutterella" = "#046582",
            "Cloacibacillus" = "#D7C1B1", "Dialister"= "#FAFD7C","Enterobacteriaceae"="#38333E", 
            "Clostridium_XlVa" = "#689030",  "Lachnospiraceae"="#CD9BCD",
            "Megamonas" ="#D14285", "Mitsuokella" ="#652926", "Parabacteroides"="#fff8dc",
            "Phascolarctobacterium"="#5E738F", "Prevotella"= "#C84248", "Pseudomonas"="#D1A33D", 
            "Ruminococcus" = "#AD6F3B", "Ruminococcaceae" = "#CC0C00",  "Veillonella" = "#F8698C", 
            "Other"="#bfbfbf")
desired = c("Acidaminococcus"="#DA5724",  "Akkermansia"="#94cfd5", 
            "Bacteroides"="#DAA1AD", "Bifidobacterium"="#CBD588", "Bilophila"="#5F7FC7", 
            # "Blautia"="#673770", "Eisenbergiella" = "#508578","Megasphaera"= "#6DDE88",
            # "Roseburia"="#8A7C64", "Sutterella" = "#046582",
            "Cloacibacillus" = "#D7C1B1", "Dialister"= "#FAFD7C","Enterobacteriaceae"="#38333E", 
            "Clostridium_XlVa" = "#689030",  "Lachnospiraceae"="#CD9BCD",
            "Megamonas" ="#D14285", "Mitsuokella" ="#652926", "Parabacteroides"="#fff8dc",
            "Prevotella"= "#C84248", "Pseudomonas"="#D1A33D", 
            "Ruminococcus" = "#AD6F3B", "Veillonella" = "#F8698C", 
            "Other"="#bfbfbf")
# Add extra column days for graphs
Top <- rename_dates(Top15)
Top$Transit <- as.ordered(factor(Top$Transit, levels = c("ST", "MT", "LT")))
Top$PD_Colon <- as.ordered(factor(Top$PD_Colon, levels = c("P", "D", "FS")))
Top$Donor <- as.ordered(factor(Top$Donor, levels = c("1", "2", "3", "4", "5", "6")))
Top$Genus <- as.ordered(factor(Top$Genus, levels = Top15levels))
plot <- ggplot(Top, aes(x = Day, y = Abundance, fill = Genus)) + geom_bar(position = "stack",
    stat = "identity") + facet_grid(PD_Colon ~ Donor + Transit, labeller = labeller(PD_Colon = PD_Colon.labsPC,
    Donor = donor.labs), scale = "free_x")
plot <- papertheme(plot)
plot <- plot + theme(strip.background = element_rect(fill = "#D6E0F4"), strip.text.x = element_text(size = 20),
    strip.text.y = element_text(size = 20, angle = 270), axis.text.x = element_blank(),
    axis.ticks.x = element_blank(), axis.title.x = element_blank(), legend.text = element_text(size = 15, hjust = 0 #,face = "italic"
        ), legend.title = element_text(size = 15, face = "bold"), 
    legend.title.align = 0.5) + xlab("Transit time") + 
  ylab(expression(paste("Absolute abundance (cells m",L^{"-1"}, ")"))) + 
  guides(fill = guide_legend("Genus", nrow = 3))
# scale_fill_manual(values = desired) +
plot <- plot + scale_fill_manual(values = desired,
              labels = c(expression(italic("Acidaminococcus")),
                         expression(italic("Akkermansia")), 
                         expression(italic("Bacteroides")), 
                         expression(italic("Bifidobacterium")), 
                         expression(italic("Bilophila")),
                         expression(italic("Cloacibacillus")), 
                         expression(italic("Dialister")),
                         expression(italic("Enterobacteriaceae")), 
                         expression(italic("Clostridium_XlVa")), 
                         expression(italic("Lachnospiraceae")),
                         expression(italic("Megamonas")) , 
                         expression(italic("Mitsuokella")), 
                         expression(italic("Parabacteroides")),
                         expression(italic("Prevotella")), 
                         expression(italic("Pseudomonas")), 
                         expression(italic("Ruminococcus")), 
                         expression(italic("Veillonella")),
                         "Other")) 
  
plot <- Fusetoplayer(plot, Type = "DNA"); plot


# Proportional
plot <- ggplot(Top, aes(x = Day, y = Abundance, fill = Genus)) + geom_bar(position = "fill",
    stat = "identity") + facet_grid(PD_Colon ~ Donor + Transit, labeller = labeller(PD_Colon = PD_Colon.labsPC,
    Donor = donor.labs), scale = "free_x")
plot <- papertheme(plot)
plot <- plot + theme(strip.background = element_rect(fill = "#D6E0F4"), strip.text.x = element_text(size = 20),
    strip.text.y = element_text(size = 20, angle = 270), axis.text.x = element_blank(),
    axis.ticks.x = element_blank(), axis.title.x = element_blank(), legend.text = element_text(size = 15, hjust = 0 #, face = "italic"
        ), legend.title = element_text(size = 15, face = "bold"),
    legend.title.align = 0.5) + xlab("Transit time") + 
  ylab("Proportional abundance (%)") + 
  scale_y_continuous(labels = scales::percent_format(accuracy = 1))+
  guides(fill = guide_legend("Genus", nrow = 3))
# scale_fill_manual(values = desired) +
plot <- plot + scale_fill_manual(values = desired, 
          labels = c(expression(italic("Acidaminococcus")),
                         expression(italic("Akkermansia")),
                         # expression(italic("Alistipes")),
                         expression(italic("Bacteroides")), 
                         expression(italic("Bifidobacterium")), 
                         expression(italic("Bilophila")),
                         expression(italic("Cloacibacillus")), 
                         # expression(italic("Cloacibacillus")),
                         expression(italic("Dialister")),
                         expression(italic("Enterobacteriaceae")), 
                         expression(italic("Clostridium_XlVa")), 
                         expression(italic("Lachnospiraceae")),
                         expression(italic("Megamonas")) , 
                         expression(italic("Mitsuokella")), 
                         expression(italic("Parabacteroides")),
                         expression(italic("Prevotella")), 
                         expression(italic("Pseudomonas")), 
                         expression(italic("Ruminococcus")), 
                         expression(italic("Veillonella")),
                         "Other")) 
plot <- Fusetoplayer(plot, "DNA"); plot

# Add extra toplayer with original transit times to df and plot
df1 <- subset(Top, Donor == 1 |Donor == 2); df1$intTransit <- "Short transit donor"
df2 <- subset(Top, Donor == 3 |Donor == 4); df2$intTransit <- "Medium transit donor"
df3 <- subset(Top, Donor == 5 |Donor == 6); df3$intTransit <- "Long transit donor"
Top <- rbind(df1, df2,df3); rm(df1, df2, df3)
Top$Transit <- as.ordered(factor(Top$Transit, levels = c("ST", "MT", "LT")))
Top$PD_Colon <- as.ordered(factor(Top$PD_Colon, levels = c("P", "D", "FS")))
Top$Donor <- as.ordered(factor(Top$Donor, levels = c("1", "2", "3", "4", "5", "6")))
Top$intTransit <- as.ordered(factor(Top$intTransit, levels = c("Short transit donor","Medium transit donor","Long transit donor")))
Top$Genus <- as.ordered(factor(Top$Genus, levels = Top15levels))

xlabs <- c( "1.ST" = "", "2.ST" = "", "3.ST" = "Short", "4.ST" = "", "5.ST" = "", 
            "1.MT" = "", "2.MT" = "", "3.MT" = "Med", "4.MT" = "", "5.MT" = "",
            "1.LT" = "", "2.LT" = "", "3.LT" = "Long", "4.LT" = "", "5.LT" = "")

## Plot it
plot <- ggplot(Top, aes(x = interaction(Day, Transit), y = Abundance, fill = Genus)) + geom_bar(position = "fill",
    stat = "identity") + facet_grid(PD_Colon ~ intTransit + Donor + Transit, labeller = labeller(PD_Colon = PD_Colon.labsPC,
    Donor = donor.labs), scale = "free_x") +  scale_x_discrete(labels=xlabs) 
plot <- papertheme(plot)
plot <- plot + theme(strip.background = element_rect(fill = "#D6E0F4"), strip.text.x = element_text(size = 20),
    strip.text.y = element_text(size = 20, angle = 270), 
    # axis.text.x = element_blank(),
    axis.ticks.x = element_blank(), 
    # axis.title.x = element_blank(), 
    legend.text = element_text(size = 15, hjust = 0 #, face = "italic"
        ), legend.title = element_text(size = 15, face = "bold"),
    legend.title.align = 0.5) + xlab("SHIME Transit time") + 
  
  ylab("Proportional abundance (%)") + 
  scale_y_continuous(labels = scales::percent_format(accuracy = 1))+
  guides(fill = guide_legend("Genus", nrow = 3))
# scale_fill_manual(values = desired) +
plot <- plot + scale_fill_manual(values = desired, 
          labels = c(expression(italic("Acidaminococcus")),
                         expression(italic("Akkermansia")), 
                         expression(italic("Bacteroides")), 
                         expression(italic("Bifidobacterium")), 
                         expression(italic("Bilophila")),
                         expression(italic("Cloacibacillus")), 
                         expression(italic("Dialister")),
                         expression(italic("Enterobacteriaceae")), 
                         expression(italic("Clostridium_XlVa")), 
                         expression(italic("Lachnospiraceae")),
                         expression(italic("Megamonas")) , 
                         expression(italic("Mitsuokella")), 
                         expression(italic("Parabacteroides")),
                         expression(italic("Prevotella")), 
                         expression(italic("Pseudomonas")), 
                         expression(italic("Ruminococcus")), 
                         expression(italic("Veillonella")),
                         "Other"))

## Combine top label rows
    pg <- ggplotGrob(plot)
    ### Get a list of strips from the original plot
    strip <- lapply(grep("strip-t", pg$layout$name), function(x) {pg$grobs[[x]]})
    
    ### Construct gtable to contain the new strip
    newStrip.inttransit <- gtable(widths = unit(rep(1, 18), "null"), heights = strip[[1]]$heights)
    newStrip.donors <- gtable(widths = unit(rep(1, 18), "null"), heights = strip[[2]]$heights)
    
    ## Populate the gtable first row
    cols <- seq(1, by = 6, length.out = 3)  # seq(start, by = panels under this row, length.out = panel amount)
    newStrip.inttransit <- gtable_add_grob(newStrip.inttransit, lapply(strip[cols], `[`, 1), 
                                t = 1, l = cols, r = cols + 5)
    
    # t = row, l = amount of cols, r = # of panels under this row Put the strip
    # into the plot
    pg <- gtable_add_grob(pg, newStrip.inttransit, t = 7, l = 5, r = 39)
    
    cols <- seq(1, by = 3, length.out = 6)  # seq(start, by = panels under this row, length.out = panel amount)
    newStrip.donors <- gtable_add_grob(newStrip.donors, lapply(strip[cols], `[`, 2), 
                                t = 2, l = cols, r = cols + 2)
    
    pgNew <- gtable_add_grob(pg, newStrip.donors, t = 7, l = 5, r = 39)
pgNew <- ggdraw() + draw_grob(pgNew)
pgNew

## Combine top label rows
    pg <- ggplotGrob(plot)
    ### Get a list of strips from the original plot
    strip <- lapply(grep("strip-t", pg$layout$name), function(x) {pg$grobs[[x]]})
    
    ### Construct gtable to contain the new strip
    newStrip.inttransit <- gtable(widths = unit(rep(1, 18), "null"), heights = strip[[1]]$heights)
    newStrip.donors <- gtable(widths = unit(rep(1, 18), "null"), heights = strip[[2]]$heights)
    
    ## Populate the gtable first row
    cols <- seq(1, by = 6, length.out = 3)  # seq(start, by = panels under this row, length.out = panel amount)
    newStrip.inttransit <- gtable_add_grob(newStrip.inttransit, lapply(strip[cols], `[`, 1), 
                                t = 1, l = cols, r = cols + 5)
    
    cols <- seq(1, by = 3, length.out = 6)  # seq(start, by = panels under this row, length.out = panel amount)
    newStrip.donors <- gtable_add_grob(newStrip.donors, lapply(strip[cols], `[`, 2), 
                                t = 2, l = cols, r = cols + 2)
    # t = row, l = amount of cols, r = # of panels under this row Put the strip
    # into the plot
    pg <- gtable_add_grob(pg, newStrip.inttransit, t = 7.0, l = 5, r = 39)
    pgNew <- gtable_add_grob(pg, newStrip.donors, t = 7.1, l = 5, r = 39)
pgNew <- ggdraw() + draw_grob(pgNew)
pgNew

plotFigure3 <- pgNew
saveplots()
ggsave(plot = plotFigure3 , width = 17, height = 11, dpi = 200, filename = "Figure3_lo_res.png")
ggsave(plot = plotFigure3 , width = 17, height = 11, dpi = 400, filename = "Figure3_hi_res.tiff")


```


```{r Paperplot proportional abundance, fig.height=11, fig.width=17, include=TRUE}
saveplots()
# formatR::tidy_app()
# Make a data frame with a column for the read counts of each sample
phylobj.QMP_stable <- subset_samples(phylobj.QMP, Ninetransit == 3)
Top15phylobj <- aggregate_top_taxa(phylobj.QMP_stable, top = 20, "Genus")
Top15 <- psmelt(Top15phylobj)
unique(Top15$OTU)
unique(Top15$OTU) %in% Top15levels
x <- data.frame(x = unique(Top15$OTU) %in% Top15levels, genus = unique(Top15$OTU)) #check if labelled
x$genus <- as.ordered(factor(x$genus, levels = Top15levels))
Top15$OTU <- as.ordered(factor(Top15$OTU, levels = Top15levels))
sort(unique(x$genus))
desired = c("Acidaminococcus"="#DA5724",  "Akkermansia"="#94cfd5", "Alistipes"="#3F4921",
            "Bacteroides"="#DAA1AD", "Bifidobacterium"="#CBD588", "Bilophila"="#5F7FC7", 
            # "Blautia"="#673770", "Eisenbergiella" = "#508578","Megasphaera"= "#6DDE88",
            # "Roseburia"="#8A7C64", "Sutterella" = "#046582",
            "Cloacibacillus" = "#D7C1B1", "Dialister"= "#FAFD7C","Enterobacteriaceae"="#38333E", 
            "Clostridium_XlVa" = "#689030",  "Lachnospiraceae"="#CD9BCD",
            "Megamonas" ="#D14285", "Mitsuokella" ="#652926", "Parabacteroides"="#fff8dc",
            "Phascolarctobacterium"="#5E738F", "Prevotella"= "#C84248", "Pseudomonas"="#D1A33D", 
            "Ruminococcus" = "#AD6F3B", "Ruminococcaceae" = "#CC0C00",  "Veillonella" = "#F8698C", 
            "Other"="#bfbfbf")
desired = c("Acidaminococcus"="#DA5724",  "Akkermansia"="#94cfd5", 
            "Bacteroides"="#DAA1AD", "Bifidobacterium"="#CBD588", "Bilophila"="#5F7FC7", 
            # "Blautia"="#673770", "Eisenbergiella" = "#508578","Megasphaera"= "#6DDE88",
            # "Roseburia"="#8A7C64", "Sutterella" = "#046582",
            "Cloacibacillus" = "#D7C1B1", "Dialister"= "#FAFD7C","Enterobacteriaceae"="#38333E", 
            "Clostridium_XlVa" = "#689030",  "Lachnospiraceae"="#CD9BCD",
            "Megamonas" ="#D14285", "Mitsuokella" ="#652926", "Parabacteroides"="#fff8dc",
            "Prevotella"= "#C84248", "Pseudomonas"="#D1A33D", 
            "Ruminococcus" = "#AD6F3B", "Veillonella" = "#F8698C", 
            "Other"="#bfbfbf")
# Add extra column days for graphs
Top <- rename_dates(Top15)
Top$Transit <- as.ordered(factor(Top$Transit, levels = c("ST", "MT", "LT")))
Top$PD_Colon <- as.ordered(factor(Top$PD_Colon, levels = c("P", "D", "FS")))
Top$Donor <- as.ordered(factor(Top$Donor, levels = c("1", "2", "3", "4", "5", "6")))
Top$Genus <- as.ordered(factor(Top$Genus, levels = Top15levels))
plot <- ggplot(Top, aes(x = Day, y = Abundance, fill = Genus)) + geom_bar(position = "stack",
    stat = "identity") + facet_grid(PD_Colon ~ Donor + Transit, labeller = labeller(PD_Colon = PD_Colon.labsPC,
    Donor = donor.labs), scale = "free_x")
plot <- papertheme(plot)
plot <- plot + theme(strip.background = element_rect(fill = "#D6E0F4"), strip.text.x = element_text(size = 20),
    strip.text.y = element_text(size = 20, angle = 270), axis.text.x = element_blank(),
    axis.ticks.x = element_blank(), axis.title.x = element_blank(), legend.text = element_text(size = 15, hjust = 0 #,face = "italic"
        ), legend.title = element_text(size = 15, face = "bold"), 
    legend.title.align = 0.5) + xlab("Transit time") + 
  ylab(expression(paste("Absolute abundance (cells m",L^{"-1"}, ")"))) + 
  guides(fill = guide_legend("Genus", nrow = 3))
# scale_fill_manual(values = desired) +
plot <- plot + scale_fill_manual(values = desired,
              labels = c(expression(italic("Acidaminococcus")),
                         expression(italic("Akkermansia")), 
                         expression(italic("Bacteroides")), 
                         expression(italic("Bifidobacterium")), 
                         expression(italic("Bilophila")),
                         expression(italic("Cloacibacillus")), 
                         expression(italic("Dialister")),
                         expression(italic("Enterobacteriaceae")), 
                         expression(italic("Clostridium_XlVa")), 
                         expression(italic("Lachnospiraceae")),
                         expression(italic("Megamonas")) , 
                         expression(italic("Mitsuokella")), 
                         expression(italic("Parabacteroides")),
                         expression(italic("Prevotella")), 
                         expression(italic("Pseudomonas")), 
                         expression(italic("Ruminococcus")), 
                         expression(italic("Veillonella")),
                         "Other")) 
  
plot <- Fusetoplayer(plot, Type = "DNA"); plot


# Proportional
plot <- ggplot(Top, aes(x = Day, y = Abundance, fill = Genus)) + geom_bar(position = "fill",
    stat = "identity") + facet_grid(PD_Colon ~ Donor + Transit, labeller = labeller(PD_Colon = PD_Colon.labsPC,
    Donor = donor.labs), scale = "free_x")
plot <- papertheme(plot)
plot <- plot + theme(strip.background = element_rect(fill = "#D6E0F4"), strip.text.x = element_text(size = 20),
    strip.text.y = element_text(size = 20, angle = 270), axis.text.x = element_blank(),
    axis.ticks.x = element_blank(), axis.title.x = element_blank(), legend.text = element_text(size = 15, hjust = 0 #, face = "italic"
        ), legend.title = element_text(size = 15, face = "bold"),
    legend.title.align = 0.5) + xlab("Transit time") + 
  ylab("Proportional abundance (%)") + 
  scale_y_continuous(labels = scales::percent_format(accuracy = 1))+
  guides(fill = guide_legend("Genus", nrow = 3))
# scale_fill_manual(values = desired) +
plot <- plot + scale_fill_manual(values = desired, 
          labels = c(expression(italic("Acidaminococcus")),
                         expression(italic("Akkermansia")),
                         # expression(italic("Alistipes")),
                         expression(italic("Bacteroides")), 
                         expression(italic("Bifidobacterium")), 
                         expression(italic("Bilophila")),
                         expression(italic("Cloacibacillus")), 
                         # expression(italic("Cloacibacillus")),
                         expression(italic("Dialister")),
                         expression(italic("Enterobacteriaceae")), 
                         expression(italic("Clostridium_XlVa")), 
                         expression(italic("Lachnospiraceae")),
                         expression(italic("Megamonas")) , 
                         expression(italic("Mitsuokella")), 
                         expression(italic("Parabacteroides")),
                         expression(italic("Prevotella")), 
                         expression(italic("Pseudomonas")), 
                         expression(italic("Ruminococcus")), 
                         expression(italic("Veillonella")),
                         "Other")) 
plot <- Fusetoplayer(plot, "DNA"); plot
# Add extra toplayer with original transit times to df and plot
df1 <- subset(Top, Donor == 1 |Donor == 2); df1$intTransit <- "Short transit donor"
df2 <- subset(Top, Donor == 3 |Donor == 4); df2$intTransit <- "Medium transit donor"
df3 <- subset(Top, Donor == 5 |Donor == 6); df3$intTransit <- "Long transit donor"
Top <- rbind(df1, df2,df3); rm(df1, df2, df3)
Top$Transit <- as.ordered(factor(Top$Transit, levels = c("ST", "MT", "LT")))
Top$PD_Colon <- as.ordered(factor(Top$PD_Colon, levels = c("P", "D", "FS")))
Top$Donor <- as.ordered(factor(Top$Donor, levels = c("1", "2", "3", "4", "5", "6")))
Top$intTransit <- as.ordered(factor(Top$intTransit, levels = c("Short transit donor","Medium transit donor","Long transit donor")))
Top$Genus <- as.ordered(factor(Top$Genus, levels = Top15levels))
## Plot it
plot <- ggplot(Top, aes(x = Day, y = Abundance, fill = Genus)) + geom_bar(position = "fill",
    stat = "identity") + facet_grid(PD_Colon ~ intTransit + Donor + Transit, labeller = labeller(PD_Colon = PD_Colon.labsPC,
    Donor = donor.labs), scale = "free_x")
plot <- papertheme(plot)
plot <- plot + theme(strip.background = element_rect(fill = "#D6E0F4"), strip.text.x = element_text(size = 20),
    strip.text.y = element_text(size = 20, angle = 270), axis.text.x = element_blank(),
    axis.ticks.x = element_blank(), axis.title.x = element_blank(), legend.text = element_text(size = 15, hjust = 0 #, face = "italic"
        ), legend.title = element_text(size = 15, face = "bold"),
    legend.title.align = 0.5) + xlab("Transit time") + 
  ylab("Proportional abundance (%)") + 
  scale_y_continuous(labels = scales::percent_format(accuracy = 1))+
  guides(fill = guide_legend("Genus", nrow = 3))
# scale_fill_manual(values = desired) +
plot <- plot + scale_fill_manual(values = desired, 
          labels = c(expression(italic("Acidaminococcus")),
                         expression(italic("Akkermansia")), 
                         expression(italic("Bacteroides")), 
                         expression(italic("Bifidobacterium")), 
                         expression(italic("Bilophila")),
                         expression(italic("Cloacibacillus")), 
                         expression(italic("Dialister")),
                         expression(italic("Enterobacteriaceae")), 
                         expression(italic("Clostridium_XlVa")), 
                         expression(italic("Lachnospiraceae")),
                         expression(italic("Megamonas")) , 
                         expression(italic("Mitsuokella")), 
                         expression(italic("Parabacteroides")),
                         expression(italic("Prevotella")), 
                         expression(italic("Pseudomonas")), 
                         expression(italic("Ruminococcus")), 
                         expression(italic("Veillonella")),
                         "Other")) 
## Combine top label rows
    pg <- ggplotGrob(plot)
    ### Get a list of strips from the original plot
    strip <- lapply(grep("strip-t", pg$layout$name), function(x) {pg$grobs[[x]]})
    
    ### Construct gtable to contain the new strip
    newStrip.inttransit <- gtable(widths = unit(rep(1, 18), "null"), heights = strip[[1]]$heights)
    newStrip.donors <- gtable(widths = unit(rep(1, 18), "null"), heights = strip[[2]]$heights)
    
    ## Populate the gtable first row
    cols <- seq(1, by = 6, length.out = 3)  # seq(start, by = panels under this row, length.out = panel amount)
    newStrip.inttransit <- gtable_add_grob(newStrip.inttransit, lapply(strip[cols], `[`, 1), 
                                t = 1, l = cols, r = cols + 5)
    
    # t = row, l = amount of cols, r = # of panels under this row Put the strip
    # into the plot
    pg <- gtable_add_grob(pg, newStrip.inttransit, t = 7, l = 5, r = 39)
    
    cols <- seq(1, by = 3, length.out = 6)  # seq(start, by = panels under this row, length.out = panel amount)
    newStrip.donors <- gtable_add_grob(newStrip.donors, lapply(strip[cols], `[`, 2), 
                                t = 2, l = cols, r = cols + 2)
    
    pgNew <- gtable_add_grob(pg, newStrip.donors, t = 7, l = 5, r = 39)
pgNew <- ggdraw() + draw_grob(pgNew)
pgNew
## Combine top label rows
    pg <- ggplotGrob(plot)
    ### Get a list of strips from the original plot
    strip <- lapply(grep("strip-t", pg$layout$name), function(x) {pg$grobs[[x]]})
    
    ### Construct gtable to contain the new strip
    newStrip.inttransit <- gtable(widths = unit(rep(1, 18), "null"), heights = strip[[1]]$heights)
    newStrip.donors <- gtable(widths = unit(rep(1, 18), "null"), heights = strip[[2]]$heights)
    
    ## Populate the gtable first row
    cols <- seq(1, by = 6, length.out = 3)  # seq(start, by = panels under this row, length.out = panel amount)
    newStrip.inttransit <- gtable_add_grob(newStrip.inttransit, lapply(strip[cols], `[`, 1), 
                                t = 1, l = cols, r = cols + 5)
    cols <- seq(1, by = 3, length.out = 6)  # seq(start, by = panels under this row, length.out = panel amount)
    newStrip.donors <- gtable_add_grob(newStrip.donors, lapply(strip[cols], `[`, 2), 
                                t = 2, l = cols, r = cols + 2)
    # t = row, l = amount of cols, r = # of panels under this row Put the strip
    # into the plot
    pg <- gtable_add_grob(pg, newStrip.inttransit, t = 7, l = 5, r = 39)
    pgNew <- gtable_add_grob(pg, newStrip.donors, t = 7.1, l = 5, r = 39)
pgNew <- ggdraw() + draw_grob(pgNew)
pgNew

plotFigure3 <- pgNew
saveplots()
ggsave(plot = plotFigure3 , width = 17, height = 11, dpi = 200, filename = "Figure3_lo_res.png")
ggsave(plot = plotFigure3 , width = 17, height = 11, dpi = 400, filename = "Figure3_hi_res.tiff")

# Stabilised + Faecal sample
# Make a data frame with a column for the read counts of each sample
phylobj.QMP_stable_FS <- merge_phyloseq(subset_samples(phylobj.QMP, Ninetransit == 3), phylobj.QMPFS)
# phylobj.QMP_stable_FS <- subset_samples(phylobj.QMP, Ninetransit == 3)
ntop = 21
Top15phylobj <- aggregate_top_taxa(phylobj.QMP_stable_FS, top = ntop, "Genus")
# Top15phylobj <- aggregate_top_taxa(phylobj.QMP_stable_FS, top = 15, "Family")
Top15 <- psmelt(Top15phylobj)
unique(Top15$OTU)
unique(Top15$OTU) %in% Top15levels
x <- data.frame(x = unique(Top15$OTU) %in% Top15levels, genus = unique(Top15$OTU))  #check if labelled
Top15$OTU <- as.ordered(factor(Top15$OTU, levels = Top15levels))
# Add extra column days for graphs
Top <- rename_dates(Top15, FS = T)
Top$Transit <- as.ordered(factor(Top$Transit, levels = c("ST", "MT", "LT", "FS")))
Top$PD_Colon <- as.ordered(factor(Top$PD_Colon, levels = c("P", "D")))
Top$Donor <- as.ordered(factor(Top$Donor, levels = c("1", "2", "3", "4", "5", "6")))
Top$Genus <- as.ordered(factor(Top$Genus, levels = Top15levels))
getPalette = colorRampPalette(brewer.pal(11, "Paired"))
extracolours <- c(desired, getPalette(ntop))
# Proportional
plot <- ggplot(Top, aes(x = Day, y = Abundance, fill = Genus)) + geom_bar(position = "fill",
    stat = "identity") + facet_grid(PD_Colon ~ Donor + Transit, 
                  labeller = labeller(PD_Colon = PD_Colon.labsPC, Donor = donor.labs
                                      ), scale = "free_x")
plot <- papertheme(plot)
plot <- plot + theme(strip.background = element_rect(fill = "#D6E0F4"), strip.text.x = element_text(size = 20),
    strip.text.y = element_text(size = 20, angle = 270), axis.text.x = element_blank(),
    axis.ticks.x = element_blank(), axis.title.x = element_blank(), legend.text = element_text(size = 15,
        face = "italic"), legend.title = element_text(size = 15, face = "bold"),
    legend.title.align = 0.5) + xlab("Transit time")  +
  ylab("Proportional abundance (%)") + 
  scale_y_continuous(labels = scales::percent_format(accuracy = 1))+
  guides(fill = guide_legend("Genus", nrow = 3))
# plot <- plot + scale_fill_manual(values = desired)
plot <- plot + scale_fill_manual(values = extracolours)
plot <- Fusetoplayer(plot, "DNA_FS"); plot
setwd("/Projects2/Yorick/Rstudio_projects/Task1.2TransitSHIME/TransitSHIME")
source("YM_colours.R")




phylobj.QMP_stable_FS <- merge_phyloseq(subset_samples(phylobj.QMP, Ninetransit == 3), phylobj.QMPFS)
# phylobj.QMP_stable_FS <- subset_samples(phylobj.QMP, Ninetransit == 3)
ntop = 10
Top15phylobj <- aggregate_top_taxa(phylobj.QMP_stable_FS, top = ntop, "Class")
# Top15phylobj <- aggregate_top_taxa(phylobj.QMP_stable_FS, top = 15, "Family")
Top15 <- psmelt(Top15phylobj)

# Add extra column days for graphs
Top <- rename_dates(Top15, FS = T)
Top$Transit <- as.ordered(factor(Top$Transit, levels = c("ST", "MT", "LT", "FS")))
Top$PD_Colon <- as.ordered(factor(Top$PD_Colon, levels = c("P", "D")))
Top$Donor <- as.ordered(factor(Top$Donor, levels = c("1", "2", "3", "4", "5", "6")))
# Top$Genus <- as.ordered(factor(Top$Genus, levels = Top15levels))
getPalette = colorRampPalette(brewer.pal(11, "Paired"))
extracolours <- c(desired, getPalette(ntop))
# Proportional
plot <- ggplot(Top, aes(x = Day, y = Abundance, fill = Class)) + geom_bar(position = "fill",
    stat = "identity") + facet_grid(PD_Colon ~ Donor + Transit, 
                  labeller = labeller(PD_Colon = PD_Colon.labsPC, Donor = donor.labs
                                      ), scale = "free_x")
plot <- papertheme(plot)
plot <- plot + theme(strip.background = element_rect(fill = "#D6E0F4"), strip.text.x = element_text(size = 20),
    strip.text.y = element_text(size = 20, angle = 270), axis.text.x = element_blank(),
    axis.ticks.x = element_blank(), axis.title.x = element_blank(), legend.text = element_text(size = 15,
        face = "italic"), legend.title = element_text(size = 15, face = "bold"),
    legend.title.align = 0.5) + xlab("Transit time")  +
  ylab("Proportional abundance (%)") + 
  scale_y_continuous(labels = scales::percent_format(accuracy = 1))+
  guides(fill = guide_legend("Class", nrow = 3))
# plot <- plot + scale_fill_manual(values = desired)
plot <- plot + scale_fill_manual(values = getPalette(11))
plot <- Fusetoplayer(plot, "DNA_FS"); plot




phylobj.QMP_stable_FS <- merge_phyloseq(subset_samples(phylobj.QMP, Ninetransit == 3), phylobj.QMPFS)
# phylobj.QMP_stable_FS <- subset_samples(phylobj.QMP, Ninetransit == 3)
ntop = 10
Top15phylobj <- aggregate_top_taxa(phylobj.QMP_stable_FS, top = ntop, "Phylum")
# Top15phylobj <- aggregate_top_taxa(phylobj.QMP_stable_FS, top = 15, "Family")
Top15 <- psmelt(Top15phylobj)

# Add extra column days for graphs
Top <- rename_dates(Top15, FS = T)
Top$Transit <- as.ordered(factor(Top$Transit, levels = c("ST", "MT", "LT", "FS")))
Top$PD_Colon <- as.ordered(factor(Top$PD_Colon, levels = c("P", "D")))
Top$Donor <- as.ordered(factor(Top$Donor, levels = c("1", "2", "3", "4", "5", "6")))
# Top$Genus <- as.ordered(factor(Top$Genus, levels = Top15levels))
getPalette = colorRampPalette(brewer.pal(11, "Paired"))
extracolours <- c(desired, getPalette(ntop))
# Proportional
plot <- ggplot(Top, aes(x = Day, y = Abundance, fill = Phylum)) + geom_bar(position = "fill",
    stat = "identity") + facet_grid(PD_Colon ~ Donor + Transit, 
                  labeller = labeller(PD_Colon = PD_Colon.labsPC, Donor = donor.labs
                                      ), scale = "free_x")
plot <- papertheme(plot)
plot <- plot + theme(strip.background = element_rect(fill = "#D6E0F4"), strip.text.x = element_text(size = 20),
    strip.text.y = element_text(size = 20, angle = 270), axis.text.x = element_blank(),
    axis.ticks.x = element_blank(), axis.title.x = element_blank(), legend.text = element_text(size = 15,
        face = "italic"), legend.title = element_text(size = 15, face = "bold"),
    legend.title.align = 0.5) + xlab("Transit time")  +
  ylab("Proportional abundance (%)") + 
  scale_y_continuous(labels = scales::percent_format(accuracy = 1))+
  guides(fill = guide_legend("Phylum", nrow = 3))
# plot <- plot + scale_fill_manual(values = desired)
plot <- plot + scale_fill_manual(values = getPalette(11))
plot <- Fusetoplayer(plot, "DNA_FS"); plot



```




### Washout faeces
```{r}
phylobj.QMP_stable <- subset_samples(phylobj.prop.otu, Ninetransit == 3 | Ninetransit == 0)
# Top15phylobj <- aggregate_top_taxa(phylobj.QMP_stable, top = 17, "Genus")
Top15 <- psmelt(phylobj.QMP_stable)
Top15$comb <- paste0(Top15$OTU, Top15$Transit, Top15$PD_Colon)
dfOTU <- data.frame()
for(donor in c(1:6)){
  print(message(paste("Donor ", donor)))
df.1 <- subset(Top15, Donor ==donor)
df.1 <- df.1 %>% group_by(comb) %>% summarise(Abundance = mean(Abundance), OTU = OTU, Transit = Transit, PD_Colon = PD_Colon, Donor = Donor)
  df.1<- df.1[!duplicated(df.1),]
  
  number = 1
  for(i in unique(df.1$OTU)){
    df1 <- subset(df.1, OTU == i)
    for(x in c(1:nrow(df1))){
      df2 <- df1[x,]
      if(df2$Abundance ==0){df2$Abundance <- 0}else{df2$Abundance <- 1}
      dfOTU <- rbind(dfOTU, df2)
      if(number%%500 == 0){message(paste(number, "of 3560 combinations done for donor ", donor))}
      number = number +1
    }
  }
}  

```


#### Plot as pie
```{r fig.height=20, fig.width=20}
desiredcolor <-  c("FS" = "#CDC774", "ST" = "#7ACD74", "MT"= "#757BCD","LT"= "#CD747A")
plotsP <- list(); lop = 1; lod = 1; plotsD <- list()
for(donors in c(1:6)){
df <- subset(dfOTU, Donor == donors)
dfFS <- subset(df, Transit == "FS")
dfTransit <- subset(df, Transit != "FS")
for(i in c("ST", "MT", "LT")){
  df2 <- subset(dfTransit, Transit == i)
  df3 <- rbind(dfFS, df2)
  for(x in c("P", "D")){
    df4 <- subset(df3, PD_Colon == x)
    df4 <- df4 %>% group_by(Transit) %>% summarise(Abundance = sum(Abundance),
                                                   # OTU = OTU, 
                                                   Transit = Transit, PD_Colon = PD_Colon, Donor = Donor)
  df4<- df4[!duplicated(df4),]
  df6 <- data.frame()
 for(looper in c(1:nrow(df4))){
   df5 <- df4[looper,]
   num = df5$Abundance
   df5$Abundance <- round(100*(num/sum(df4$Abundance)))
   df6 <- rbind(df6, df5)
 }
   plot <- ggplot(df6, aes(x="", y=Abundance, fill=Transit)) +
      geom_bar(stat="identity", width=1) +
     geom_text(aes(label = paste(Abundance, "%")),
            position = position_stack(vjust = 0.5)) +
      coord_polar("y", start=0) + ggtitle(paste(donors, i, x)) + scale_fill_manual(values = desiredcolor)
   
   
   if(x == "P"){plotsP[[lop]] <- plot
   lop = lop +1}else{plotsD[[lod]] <- plot
   lod = lod +1}
  }
}
}



plots <- list(); lop = 1; lod = 1; plotsD <- list()
for(donors in c(1:6)){
df <- subset(dfOTU, Donor == donors)
dfFS <- subset(df, Transit == "FS")
dfTransit <- subset(df, Transit != "FS")

  dfSF <- subset(dfTransit, Transit == "ST")
  dfSF <- rbind(dfFS, dfSF)
  dfMF <- subset(dfTransit, Transit == "MT")
  dfMF <- rbind(dfFS, dfMF)
  dfLF <- subset(dfTransit, Transit == "LT")
  dfLF <- rbind(dfFS, dfLF)
  
  
  for(x in c("P", "D")){
  dfSF2 <- subset(dfSF, PD_Colon == x)
  dfSF2 <- dfSF2 %>% group_by(Transit) %>% summarise(Abundance = sum(Abundance),
                                                   # OTU = OTU, 
                                                   Transit = Transit, PD_Colon = PD_Colon, Donor = Donor)
  dfSF2<- dfSF2[!duplicated(dfSF2),]; dfSF2$int <- "SF"
  
  dfSF1 <- data.frame()
  for(looper in c(1:nrow(dfSF2))){
   df5 <- dfSF2[looper,]
   num = df5$Abundance
   df5$Abundance <- round(100*(num/sum(dfSF2$Abundance)))
   dfSF1 <- rbind(dfSF1, df5)
  }
  dfSF2 <- dfSF1
  
  
  dfMF2 <- subset(dfMF, PD_Colon == x)
  dfMF2 <- dfMF2 %>% group_by(Transit) %>% summarise(Abundance = sum(Abundance),
                                                   # OTU = OTU, 
                                                   Transit = Transit, PD_Colon = PD_Colon, Donor = Donor)
  dfMF2<- dfMF2[!duplicated(dfMF2),]; dfMF2$int <- "MF"
  dfMF1 <- data.frame()
  for(looper in c(1:nrow(dfMF2))){
   df5 <- dfMF2[looper,]
   num = df5$Abundance
   df5$Abundance <- round(100*(num/sum(dfMF2$Abundance)))
   dfMF1 <- rbind(dfMF1, df5)
  }
  dfMF2 <- dfMF1
  
  dfLF2 <- subset(dfLF, PD_Colon == x)
  dfLF2 <- dfLF2 %>% group_by(Transit) %>% summarise(Abundance = sum(Abundance),
                                                   # OTU = OTU, 
                                                   Transit = Transit, PD_Colon = PD_Colon, Donor = Donor)
  dfLF2<- dfLF2[!duplicated(dfLF2),]; dfLF2$int <- "LF"
  dfLF1 <- data.frame()
  for(looper in c(1:nrow(dfLF2))){
   df5 <- dfLF2[looper,]
   num = df5$Abundance
   df5$Abundance <- round(100*(num/sum(dfLF2$Abundance)))
   dfLF1 <- rbind(dfLF1, df5)
  }
  dfLF2 <- dfLF1
  
  if(x == "P"){dftotP <- rbind(dfSF2, dfMF2, dfLF2)}else{dftotD <- rbind(dfSF2, dfMF2, dfLF2)}
  }
  dftot <- rbind(dftotP, dftotD)
  dftot$PD_Colon <- as.ordered(factor(dftot$PD_Colon, levels = c("P", "D")))
  
  ggplot(dftot, aes(x = int, y = Abundance, fill = Transit)) +
  geom_col() +    coord_polar("y")

   plot <- papertheme(ggplot(dftot, aes(x=int, y=Abundance, fill=Transit))) + facet_grid(PD_Colon~., labeller = labeller(PD_Colon = PD_Colon.labs))+ 
     geom_col()+
      # geom_bar(stat="identity", width=1) +
     scale_x_discrete(limits = c(" ", "SF","MF","LF")) +
     geom_text(aes(label = paste(Abundance, "%")),
            position = position_stack(vjust = 0.5)) +
      coord_polar("y") + 
          ggtitle(paste("Donor", donors)) + scale_fill_manual(values = desiredcolor, labels = c("Faecal sample", "Short SHIME transit", "Medium SHIME transit", "Long SHIME transit"))+
     theme(axis.text.x = element_blank(),
            axis.title.y  = element_blank(),
            axis.title.x = element_blank(),
        axis.ticks.x = element_blank(),
        axis.text.y = element_blank(),
        axis.ticks.y = element_blank(),plot.title = element_text(hjust = 0.5),
        panel.grid  = element_blank(), strip.background = element_rect(fill = "#D6E0F4"), strip.text.x = element_text(size = 20),
    strip.text.y = element_text(size = 20, angle = 270))
   
   
   plots[[donors]] <- plot
   }


ggarrange(plotlist = plots, common.legend = T)


```


#### Plot as Venn
```{r fig.height=20, fig.width=20}
library(ggvenn)

plotsP <- list(); lop = 1; lod = 1; plotsD <- list()
for(donors in c(1:6)){
df <- subset(dfOTU, Donor == donors)
df <- subset(df, Abundance != 0)

for(x in c("P", "D")){
  df1 <- subset(df, PD_Colon == x)
dfFS <- subset(df1, Transit == "FS")$OTU
dfST <- subset(df1, Transit == "ST")$OTU
dfMT <- subset(df1, Transit == "MT")$OTU
dfLT <- subset(df1, Transit == "LT")$OTU

vennlist <- list(FS = dfFS, ST = dfST, MT = dfMT, LT = dfLT)
names(vennlist) <- c("Faecal sample","Short transit","Medium transit", "Long transit")
plot <- ggvenn(vennlist, fill_color = c("#CDC774", "#7ACD74", "#757BCD", "#CD747A"),
  stroke_size = 0.5, set_name_size = 4) + ggtitle(donors)
   if(x == "P"){plotsP[[lop]] <- plot
   lop = lop +1}else{plotsD[[lod]] <- plot
   lod = lod +1}
}}

ggarrange(plotlist = plotsP)
ggarrange(plotlist = plotsD)
ggarrange(plotlist = plots)

```



### mean per day + faeces
```{r Paperplot proportional abundance, fig.height=11, fig.width=17, include=TRUE}
saveplots()
# formatR::tidy_app()
# Make a data frame with a column for the read counts of each sample
phylobj.QMP_stable <- subset_samples(phylobj.QMP, Ninetransit == 3 | Ninetransit == 0)

Top15phylobj <- aggregate_top_taxa(phylobj.QMP_stable, top = 17, "Genus")
Top15 <- psmelt(Top15phylobj)
unique(Top15$OTU)
unique(Top15$OTU) %in% Top15levels
x <- data.frame(x = unique(Top15$OTU) %in% Top15levels, genus = unique(Top15$OTU)) #check if labelled
x$genus <- as.ordered(factor(x$genus, levels = Top15levels))
Top15$OTU <- as.ordered(factor(Top15$OTU, levels = Top15levels))
sort(unique(x$genus))
desired = c("Acidaminococcus"="#DA5724",  "Akkermansia"="#94cfd5", "Alistipes"="#3F4921",
            "Bacteroides"="#DAA1AD", "Bifidobacterium"="#CBD588", "Bilophila"="#5F7FC7", 
            # "Blautia"="#673770", "Eisenbergiella" = "#508578","Megasphaera"= "#6DDE88",
            # "Roseburia"="#8A7C64", "Sutterella" = "#046582",
            "Cloacibacillus" = "#D7C1B1", "Dialister"= "#FAFD7C","Enterobacteriaceae"="#38333E", 
            "Clostridium_XlVa" = "#689030",  "Lachnospiraceae"="#CD9BCD",
            "Megamonas" ="#D14285", "Mitsuokella" ="#652926", "Parabacteroides"="#fff8dc",
            "Phascolarctobacterium"="#5E738F", "Prevotella"= "#C84248", "Pseudomonas"="#D1A33D", 
            "Ruminococcus" = "#AD6F3B", "Ruminococcaceae" = "#CC0C00",  "Veillonella" = "#F8698C", 
            "Other"="#bfbfbf")
desired = c("Acidaminococcus"="#DA5724",  "Akkermansia"="#94cfd5", 
            "Bacteroides"="#DAA1AD", "Bifidobacterium"="#CBD588", "Bilophila"="#5F7FC7", 
            # "Blautia"="#673770", "Eisenbergiella" = "#508578","Megasphaera"= "#6DDE88",
            # "Roseburia"="#8A7C64", "Sutterella" = "#046582",
            "Cloacibacillus" = "#D7C1B1", "Dialister"= "#FAFD7C","Enterobacteriaceae"="#38333E", 
            "Clostridium_XlVa" = "#689030",  "Lachnospiraceae"="#CD9BCD",
            "Megamonas" ="#D14285", "Mitsuokella" ="#652926", "Parabacteroides"="#fff8dc",
            "Prevotella"= "#C84248", "Pseudomonas"="#D1A33D", 
            "Ruminococcus" = "#AD6F3B", "Veillonella" = "#F8698C", 
            "Other"="#bfbfbf")
desired2 = c("Acidaminococcus"="#DA5724",  "Akkermansia"="#94cfd5", "Alistipes"="#72BF5B", "Anaerostipes"="#D14285",
            "Bacteroides"="#DAA1AD", "Bifidobacterium"="#CBD588", "Bilophila"="#5F7FC7", "Blautia"=  "#33A02C" ,
            # "Blautia"="#673770", "Eisenbergiella" = "#508578","Megasphaera"= "#6DDE88",
            # "Roseburia"="#8A7C64", "Sutterella" = "#046582",
            "Cloacibacillus" = "#D7C1B1","Clostridium_XlVa" = "#689030","Coprococcus"= "#68AB9F" ,  
            "Dialister"= "#FAFD7C","Enterobacteriaceae"="#38333E", "Erysipelotrichaceae" ="#B2DF8A", "Faecalibacterium"="#A6CEE3",
            "Fusicatenibacter"="#1F78B4"  , 
              "Lachnospiraceae"="#CD9BCD",
            "Megamonas" ="#62A3CB", "Mitsuokella" ="#652926", "Parabacteroides"="#fff8dc",
            "Prevotella"= "#C84248", "Pseudomonas"="#D1A33D",  "Roseburia"= "#969D62",
            "Ruminococcus" = "#AD6F3B", "Ruminococcaceae"="#EF595A","Veillonella" = "#F8698C", 
            "Other"="#bfbfbf")


              
                                                    






# Add extra column days for graphs
Top <- rename_dates(Top15)
Top$Transit <- as.ordered(factor(Top$Transit, levels = c("ST", "MT", "LT", "FS")))
Top$PD_Colon <- as.ordered(factor(Top$PD_Colon, levels = c("P", "D", "FS")))
Top$Donor <- as.ordered(factor(Top$Donor, levels = c("1", "2", "3", "4", "5", "6")))
Top$Genus <- as.ordered(factor(Top$Genus, levels = Top15levels))
plot <- ggplot(Top, aes(x = Day, y = Abundance, fill = Genus)) + geom_bar(position = "stack",
    stat = "identity") + facet_grid(PD_Colon ~ Donor + Transit,  labeller = labeller(PD_Colon = PD_Colon.labsPC,
    Donor = donor.labs), scale = "free_x")
plot <- papertheme(plot)
plot <- plot + theme(strip.background = element_rect(fill = "#D6E0F4"), strip.text.x = element_text(size = 20),
    strip.text.y = element_text(size = 20, angle = 270), axis.text.x = element_blank(),
    axis.ticks.x = element_blank(), axis.title.x = element_blank(), legend.text = element_text(size = 15, hjust = 0 #,face = "italic"
        ), legend.title = element_text(size = 15, face = "bold"), 
    legend.title.align = 0.5) + xlab("Transit time") + 
  ylab(expression(paste("Absolute abundance (cells m",L^{"-1"}, ")"))) + 
  guides(fill = guide_legend("Genus", nrow = 3))
# scale_fill_manual(values = desired) +
plot <- plot + scale_fill_manual(values = desired,
              labels = c(expression(italic("Acidaminococcus")),
                         expression(italic("Akkermansia")), 
                         expression(italic("Bacteroides")), 
                         expression(italic("Bifidobacterium")), 
                         expression(italic("Bilophila")),
                         expression(italic("Cloacibacillus")), 
                         expression(italic("Dialister")),
                         expression(italic("Enterobacteriaceae")), 
                         expression(italic("Clostridium_XlVa")), 
                         expression(italic("Lachnospiraceae")),
                         expression(italic("Megamonas")) , 
                         expression(italic("Mitsuokella")), 
                         expression(italic("Parabacteroides")),
                         expression(italic("Prevotella")), 
                         expression(italic("Pseudomonas")), 
                         expression(italic("Ruminococcus")), 
                         expression(italic("Veillonella")),
                         "Other")) 
  
plot <- Fusetoplayer(plot, Type = "DNA"); plot

# Proportional
plot <- ggplot(Top, aes(x = Day, y = Abundance, fill = Genus)) + geom_bar(position = "fill",
    stat = "identity") + facet_grid(PD_Colon ~ Donor + Transit, labeller = labeller(PD_Colon = PD_Colon.labsPC,
    Donor = donor.labs), scale = "free_x")
plot <- papertheme(plot)
plot <- plot + theme(strip.background = element_rect(fill = "#D6E0F4"), strip.text.x = element_text(size = 20),
    strip.text.y = element_text(size = 20, angle = 270), axis.text.x = element_blank(),
    axis.ticks.x = element_blank(), axis.title.x = element_blank(), legend.text = element_text(size = 15, hjust = 0 #, face = "italic"
        ), legend.title = element_text(size = 15, face = "bold"),
    legend.title.align = 0.5) + xlab("Transit time") + 
  ylab("Proportional abundance (%)") + 
  scale_y_continuous(labels = scales::percent_format(accuracy = 1))+
  guides(fill = guide_legend("Genus", nrow = 3))
# scale_fill_manual(values = desired) +
plot <- plot + scale_fill_manual(values = desired, 
          labels = c(expression(italic("Acidaminococcus")),
                         expression(italic("Akkermansia")), 
                         expression(italic("Bacteroides")), 
                         expression(italic("Bifidobacterium")), 
                         expression(italic("Bilophila")),
                         expression(italic("Cloacibacillus")), 
                         expression(italic("Dialister")),
                         expression(italic("Enterobacteriaceae")), 
                         expression(italic("Clostridium_XlVa")), 
                         expression(italic("Lachnospiraceae")),
                         expression(italic("Megamonas")) , 
                         expression(italic("Mitsuokella")), 
                         expression(italic("Parabacteroides")),
                         expression(italic("Prevotella")), 
                         expression(italic("Pseudomonas")), 
                         expression(italic("Ruminococcus")), 
                         expression(italic("Veillonella")),
                         "Other")) 
plot <- Fusetoplayer(plot, "DNA"); plot
# Add extra toplayer with original transit times to df and plot
df1 <- subset(Top, Donor == 1 |Donor == 2); df1$intTransit <- "Short transit donor"
df2 <- subset(Top, Donor == 3 |Donor == 4); df2$intTransit <- "Medium transit donor"
df3 <- subset(Top, Donor == 5 |Donor == 6); df3$intTransit <- "Long transit donor"
Top <- rbind(df1, df2,df3); rm(df1, df2, df3)
Top$Transit <- as.ordered(factor(Top$Transit, levels = c("ST", "MT", "LT")))
Top$PD_Colon <- as.ordered(factor(Top$PD_Colon, levels = c("P", "D", "FS")))
Top$Donor <- as.ordered(factor(Top$Donor, levels = c("1", "2", "3", "4", "5", "6")))
Top$intTransit <- as.ordered(factor(Top$intTransit, levels = c("Short transit donor","Medium transit donor","Long transit donor")))
Top$Genus <- as.ordered(factor(Top$Genus, levels = Top15levels))
## Plot it
plot <- ggplot(Top, aes(x = Day, y = Abundance, fill = Genus)) + geom_bar(position = "fill",
    stat = "identity") + facet_grid(PD_Colon ~ intTransit + Donor + Transit, labeller = labeller(PD_Colon = PD_Colon.labsPC,
    Donor = donor.labs), scale = "free_x")
plot <- papertheme(plot)
plot <- plot + theme(strip.background = element_rect(fill = "#D6E0F4"), strip.text.x = element_text(size = 20),
    strip.text.y = element_text(size = 20, angle = 270), axis.text.x = element_blank(),
    axis.ticks.x = element_blank(), axis.title.x = element_blank(), legend.text = element_text(size = 15, hjust = 0 #, face = "italic"
        ), legend.title = element_text(size = 15, face = "bold"),
    legend.title.align = 0.5) + xlab("Transit time") + 
  ylab("Proportional abundance (%)") + 
  scale_y_continuous(labels = scales::percent_format(accuracy = 1))+
  guides(fill = guide_legend("Genus", nrow = 3))
# scale_fill_manual(values = desired) +
plot <- plot + scale_fill_manual(values = desired, 
          labels = c(expression(italic("Acidaminococcus")),
                         expression(italic("Akkermansia")), 
                         expression(italic("Bacteroides")), 
                         expression(italic("Bifidobacterium")), 
                         expression(italic("Bilophila")),
                         expression(italic("Cloacibacillus")), 
                         expression(italic("Dialister")),
                         expression(italic("Enterobacteriaceae")), 
                         expression(italic("Clostridium_XlVa")), 
                         expression(italic("Lachnospiraceae")),
                         expression(italic("Megamonas")) , 
                         expression(italic("Mitsuokella")), 
                         expression(italic("Parabacteroides")),
                         expression(italic("Prevotella")), 
                         expression(italic("Pseudomonas")), 
                         expression(italic("Ruminococcus")), 
                         expression(italic("Veillonella")),
                         "Other")) 
## Combine top label rows
    pg <- ggplotGrob(plot)
    ### Get a list of strips from the original plot
    strip <- lapply(grep("strip-t", pg$layout$name), function(x) {pg$grobs[[x]]})
    
    ### Construct gtable to contain the new strip
    newStrip.inttransit <- gtable(widths = unit(rep(1, 18), "null"), heights = strip[[1]]$heights)
    newStrip.donors <- gtable(widths = unit(rep(1, 18), "null"), heights = strip[[2]]$heights)
    
    ## Populate the gtable first row
    cols <- seq(1, by = 6, length.out = 3)  # seq(start, by = panels under this row, length.out = panel amount)
    newStrip.inttransit <- gtable_add_grob(newStrip.inttransit, lapply(strip[cols], `[`, 1), 
                                t = 1, l = cols, r = cols + 5)
    
    # t = row, l = amount of cols, r = # of panels under this row Put the strip
    # into the plot
    pg <- gtable_add_grob(pg, newStrip.inttransit, t = 7, l = 5, r = 39)
    
    cols <- seq(1, by = 3, length.out = 6)  # seq(start, by = panels under this row, length.out = panel amount)
    newStrip.donors <- gtable_add_grob(newStrip.donors, lapply(strip[cols], `[`, 2), 
                                t = 2, l = cols, r = cols + 2)
    
    pgNew <- gtable_add_grob(pg, newStrip.donors, t = 7, l = 5, r = 39)
pgNew <- ggdraw() + draw_grob(pgNew)
pgNew
## Combine top label rows
    pg <- ggplotGrob(plot)
    ### Get a list of strips from the original plot
    strip <- lapply(grep("strip-t", pg$layout$name), function(x) {pg$grobs[[x]]})
    
    ### Construct gtable to contain the new strip
    newStrip.inttransit <- gtable(widths = unit(rep(1, 18), "null"), heights = strip[[1]]$heights)
    newStrip.donors <- gtable(widths = unit(rep(1, 18), "null"), heights = strip[[2]]$heights)
    
    ## Populate the gtable first row
    cols <- seq(1, by = 6, length.out = 3)  # seq(start, by = panels under this row, length.out = panel amount)
    newStrip.inttransit <- gtable_add_grob(newStrip.inttransit, lapply(strip[cols], `[`, 1), 
                                t = 1, l = cols, r = cols + 5)
    cols <- seq(1, by = 3, length.out = 6)  # seq(start, by = panels under this row, length.out = panel amount)
    newStrip.donors <- gtable_add_grob(newStrip.donors, lapply(strip[cols], `[`, 2), 
                                t = 2, l = cols, r = cols + 2)
    # t = row, l = amount of cols, r = # of panels under this row Put the strip
    # into the plot
    pg <- gtable_add_grob(pg, newStrip.inttransit, t = 7, l = 5, r = 39)
    pgNew <- gtable_add_grob(pg, newStrip.donors, t = 7.1, l = 5, r = 39)
pgNew <- ggdraw() + draw_grob(pgNew)
pgNew
ggnewplotobject
plot <- Fusetoplayer(plot, "DNA"); plot


# Stabilised + Faecal sample
# Make a data frame with a column for the read counts of each sample
phylobj.QMP_stable_FS <- merge_phyloseq(subset_samples(phylobj.QMP, Ninetransit == 3), phylobj.QMPFS)
# phylobj.QMP_stable_FS <- subset_samples(phylobj.QMP, Ninetransit == 3)
ntop = 21
Top15phylobj <- aggregate_top_taxa(phylobj.QMP_stable_FS, top = ntop, "Genus")
# Top15phylobj <- aggregate_top_taxa(phylobj.QMP_stable_FS, top = 15, "Family")
Top15 <- psmelt(Top15phylobj)
unique(Top15$OTU)
unique(Top15$OTU) %in% Top15levels
x <- data.frame(x = unique(Top15$OTU) %in% Top15levels, genus = unique(Top15$OTU))  #check if labelled
Top15$OTU <- as.ordered(factor(Top15$OTU, levels = Top15levels))
# Add extra column days for graphs
Top <- rename_dates(Top15, FS = T)
Top$Transit <- as.ordered(factor(Top$Transit, levels = c("ST", "MT", "LT", "FS")))
Top$PD_Colon <- as.ordered(factor(Top$PD_Colon, levels = c("P", "D")))
Top$Donor <- as.ordered(factor(Top$Donor, levels = c("1", "2", "3", "4", "5", "6")))
Top$Genus <- as.ordered(factor(Top$Genus, levels = c("Akkermansia", "Alistipes", "Anaerostipes" ,    "Bacteroides", "Bifidobacterium", "Bilophila","Blautia" ,   "Clostridium_XlVa" ,"Coprococcus", "Enterobacteriaceae" , "Erysipelotrichaceae" ,  "Faecalibacterium",      "Fusicatenibacter"  ,    "Lachnospiraceae" ,   
                                                     "Megamonas",     "Mitsuokella",   "Prevotella",  "Roseburia", "Ruminococcus" ,  "Ruminococcaceae",                                 
                      "Other"  )))
getPalette = colorRampPalette(brewer.pal(11, "Paired"))
extracolours <- desired2                            
              

# Proportional
plot <- ggplot(Top, aes(x = Day, y = Abundance, fill = Genus)) + geom_bar(position = "fill",
    stat = "identity") + facet_grid(PD_Colon ~ Donor + Transit, 
                  labeller = labeller(PD_Colon = PD_Colon.labsPC, Donor = donor.labs
                                      ), scale = "free_x")
plot <- papertheme(plot)
plot <- plot + theme(strip.background = element_rect(fill = "#D6E0F4"), strip.text.x = element_text(size = 20),
    strip.text.y = element_text(size = 20, angle = 270), axis.text.x = element_blank(),
    axis.ticks.x = element_blank(), axis.title.x = element_blank(), legend.text = element_text(size = 15,
        face = "italic"), legend.title = element_text(size = 15, face = "bold"),
    legend.title.align = 0.5) + xlab("Transit time")  +
  ylab("Proportional abundance (%)") + 
  scale_y_continuous(labels = scales::percent_format(accuracy = 1))+
  guides(fill = guide_legend("Genus", nrow = 4))
# plot <- plot + scale_fill_manual(values = desired)
plot <- plot + scale_fill_manual(values = extracolours)
plot <- Fusetoplayer(plot, "DNA_FS"); plot

```

### Genera differences between transits with statistics - QMP and RMP
```{r Per genus and per donor, fig.height=20, fig.width=15}
my_comparisons <- list(c("ST", "MT"), c("MT", "LT"), c("ST", "LT"))
# Absolute
phylobj.QMP_stable <- subset_samples(phylobj.QMP, Ninetransit == 3)
Top15phylobj <- aggregate_top_taxa(phylobj.QMP_stable, top = 20, "Genus")
Top15 <- psmelt(Top15phylobj); unique(Top15$OTU); unique(Top15$OTU) %in% Top15levels
x <- data.frame(x = unique(Top15$OTU) %in% Top15levels, genus = unique(Top15$OTU))  #check if labelled
Top15$OTU <- as.ordered(factor(Top15$OTU, levels = Top15levels))
# Add extra column days for graphs
Top <- Top15
Top$Transit <- as.ordered(factor(Top$Transit, levels = c("ST", "MT", "LT")))
Top$PD_Colon <- as.ordered(factor(Top$PD_Colon, levels = c("P", "D", "FS")))
Top$Donor <- as.ordered(factor(Top$Donor, levels = c("1", "2", "3", "4", "5", "6")))
Top$Genus <- as.ordered(factor(Top$Genus, levels = Top15levels))
dftopQMP <- Top
# Proportional
phylobj.prop_2 = transform_sample_counts(phylobj.QMP, function(x) 100 * x/sum(x))
phylobj.prop_stable <- subset_samples(phylobj.prop_2, Ninetransit == 3)
Top15phylobj.prop <- aggregate_top_taxa(phylobj.prop_stable, top = 25, "Genus")
Top15.prop <- psmelt(Top15phylobj.prop); unique(Top15.prop$OTU); unique(Top15.prop$OTU) %in% Top15levels
x <- data.frame(x = unique(Top15.prop$OTU) %in% Top15levels, genus = unique(Top15.prop$OTU))  #check if labelled
Top15.prop$OTU <- as.ordered(factor(Top15.prop$OTU, levels = Top15levels))
# Add extra column days for graphs
Top.prop <- Top15.prop; dftopPMP <- Top15.prop
Top.prop$Transit <- as.ordered(factor(Top.prop$Transit, levels = c("ST", "MT", "LT")))
Top.prop$PD_Colon <- as.ordered(factor(Top.prop$PD_Colon, levels = c("P", "D", "FS")))
Top.prop$Donor <- as.ordered(factor(Top.prop$Donor, levels = c("1", "2", "3", "4", "5", "6")))
Top.prop$Genus <- as.ordered(factor(Top.prop$Genus, levels = Top15levels))
unique(Top$Genus)%in%unique(Top.prop$Genus) 
x <- data.frame(Genus = unique(Top$Genus), x =unique(Top$Genus)%in%unique(Top.prop$Genus)); x <- subset(x, x ==TRUE)
y <- data.frame(Genus = unique(Top.prop$Genus), x =unique(Top.prop$Genus)%in%unique(Top$Genus)); y <- subset(y, x ==TRUE)
z <- rbind(x,y); z <- distinct(z)
# In one graph
# BOXPLOTS
comparisonlistbox <- list(); x =1
for (i in z$Genus) {
  Top1 <- subset(Top, Genus == i); Top2 <- subset(Top.prop, Genus == i)
  # Top1$comb <- paste0(Top1$Donor, Top1$PD_Colon, Top1$Transit)
  # stat.test <- Top1 %>% group_by(comb) %>% t_test(Abundance ~ Transit) add_significance() %>% filter(p.adj.signif != "ns")
  
  p <- papertheme(ggplot(Top1, aes(x=Transit, y=Abundance, fill = Transit)) + 
    geom_boxplot() +
    geom_jitter(shape=16, position=position_jitter(0.2)) + 
    facet_grid(PD_Colon~Donor, labeller = labeller(PD_Colon = PD_Colon.labs)) +
    scale_fill_manual(values = transitcolors, labels=c("Short", "Medium", "Long")) + 
    scale_color_manual(values = transitcolors)) + 
    theme(strip.background = element_rect(fill = "#D6E0F4"), strip.text.x = element_text(size = 20),
    strip.text.y = element_text(size = 20, angle = 270), axis.text.x = element_blank(),
    axis.ticks.x = element_blank(), axis.title.x = element_blank(), legend.text = element_text(size = 18), 
    legend.title = element_text(size = 18, face = "bold"),
    legend.title.align = 0.5) + xlab("Transit time") + 
    # ggtitle(i) +
  ylab(expression(paste("Absolute abundance (cells m",L^{"-1"}, ")"))) + 
  guides(fill = guide_legend("Transit", ncol = 5))
  if(x%%4 != 0){p <- p + theme(legend.position = "none")}
  # + stat_compare_means(comparisons = my_comparisons, label = "p.signif", hide.ns = T, hide.NS = T) # Add significance levels
  # stat_compare_means(label.y = 50) 
  
  
  p2 <- papertheme(ggplot(Top2, aes(x=Transit, y=Abundance, fill = Transit)) + 
    geom_boxplot() +
    geom_jitter(shape=16, position=position_jitter(0.2)) + 
    facet_grid(PD_Colon~Donor, labeller = labeller(PD_Colon = PD_Colon.labs)) +
    scale_fill_manual(values = transitcolors, labels=c("Short", "Medium", "Long")) + 
    scale_color_manual(values = transitcolors)) + 
    theme(strip.background = element_rect(fill = "#D6E0F4"), strip.text.x = element_text(size = 20),
    strip.text.y = element_text(size = 20, angle = 270), axis.text.x = element_blank(),
    axis.ticks.x = element_blank(), axis.title.x = element_blank(), legend.text = element_text(size = 18), 
    legend.title = element_text(size = 18, face = "bold"),
    legend.title.align = 0.5) + xlab("Transit time") + 
    # ggtitle(i) +
  ylab("Proportional abundance (%)") + 
  guides(fill = guide_legend("Transit", ncol = 5)) 
  # + stat_compare_means(comparisons = my_comparisons, label = "p.signif", hide.ns = T, hide.NS = T) # Add significance levels
  # stat_compare_means(label.y = 50) 
  if (x%%4 != 0){p2 <- p2 + theme(legend.position = "none")
  p <- ggarrange(p, p2, ncol = 2)}
  if(x%%4 == 0){p <- ggarrange(p, p2, ncol = 2, common.legend = T, legend = "bottom")} 
  p <- annotate_figure(p, top = text_grob(i, color = "black", face = "italic", size = 20))
  # print(p)
  comparisonlistbox[[x]] <- p; x = x + 1
}
for (i in seq(from = 1, to = (length(comparisonlistbox)-1), by = 4)) {
  print(ggarrange(comparisonlistbox[[i]], comparisonlistbox[[i+1]], comparisonlistbox[[i+2]],comparisonlistbox[[i+3]], nrow = 4))
  }
# BARPLOTS
comparisonlistbar <- list(); x =1
for (i in z$Genus) {
  Top1 <- subset(Top, Genus == i); Top2 <- subset(Top.prop, Genus == i)
  # Top1$comb <- paste0(Top1$Donor, Top1$PD_Colon, Top1$Transit)
  # stat.test <- Top1 %>% group_by(comb) %>% t_test(Abundance ~ Transit) add_significance() %>% filter(p.adj.signif != "ns")
  
p <- papertheme(ggbarplot(Top1, x="Transit", y="Abundance" , fill = "Transit", add = "mean_sd", facet.by = c("PD_Colon", "Donor"))+
   facet_grid(PD_Colon~Donor, labeller = labeller(PD_Colon = PD_Colon.labs)) +
    scale_fill_manual(values = transitcolors, labels=c("Short", "Medium", "Long"))) +
    theme(strip.background = element_rect(fill = "#D6E0F4"), strip.text.x = element_text(size = 20),
    strip.text.y = element_text(size = 20, angle = 270), axis.text.x = element_blank(),
    axis.ticks.x = element_blank(), axis.title.x = element_blank(), legend.text = element_text(size = 18), 
    legend.title = element_text(size = 18, face = "bold"),
    legend.title.align = 0.5) + xlab("Transit time") +
  ylab(expression(paste("Absolute abundance (cells m",L^{"-1"}, ")"))) + 
  guides(fill = guide_legend("Transit", ncol = 5))
  if(x%%4 != 0){p <- p + theme(legend.position = "none")}
  # + stat_compare_means(comparisons = my_comparisons, label = "p.signif", hide.ns = T, hide.NS = T) # Add significance levels
  # stat_compare_means(label.y = 50) 
  
  
p2 <- papertheme(ggbarplot(Top2, x="Transit", y="Abundance" , fill = "Transit", add = "mean_sd", facet.by = c("PD_Colon", "Donor"))+
   facet_grid(PD_Colon~Donor, labeller = labeller(PD_Colon = PD_Colon.labs)) +
    scale_fill_manual(values = transitcolors, labels=c("Short", "Medium", "Long"))) +
    theme(strip.background = element_rect(fill = "#D6E0F4"), strip.text.x = element_text(size = 20),
    strip.text.y = element_text(size = 20, angle = 270), axis.text.x = element_blank(),
    axis.ticks.x = element_blank(), axis.title.x = element_blank(), legend.text = element_text(size = 18), 
    legend.title = element_text(size = 18, face = "bold"),
    legend.title.align = 0.5) + xlab("Transit time") +
  ylab("Proportional abundance (%)") +  
  guides(fill = guide_legend("Transit", ncol = 5))
  # + stat_compare_means(comparisons = my_comparisons, label = "p.signif", hide.ns = T, hide.NS = T) # Add significance levels
  # stat_compare_means(label.y = 50) 
  if (x%%4 != 0){p2 <- p2 + theme(legend.position = "none")
  p <- ggarrange(p, p2, ncol = 2)}
  if(x%%4 == 0){p <- ggarrange(p, p2, ncol = 2, common.legend = T, legend = "bottom")} 
  p <- annotate_figure(p, top = text_grob(i, color = "black", face = "italic", size = 20))
  # print(p)
  comparisonlistbar[[x]] <- p; x = x + 1
}
for (i in seq(from = 1, to = (length(comparisonlistbar)-1), by = 4)) {
  print(ggarrange(comparisonlistbar[[i]], comparisonlistbar[[i+1]], comparisonlistbar[[i+2]],comparisonlistbar[[i+3]], nrow = 4))
}
```


```{r Per genus and per donor, fig.height=20, fig.width=15}
# Statistics
dftopQMP; dftopPMP
dftopQMP$comb <- paste0(dftopQMP$Genus, dftopQMP$PD_Colon, dftopQMP$Transit)
dftopPMP$comb <- paste0(dftopPMP$Genus, dftopPMP$PD_Colon, dftopPMP$Transit)

T1 <- subset(dftopPMP, Abundance > 0); T1$Abundance <- as.numeric(T1$Abundance)
  T1 <- T1 %>% group_by(comb) %>% summarise(SD = sd(Abundance), 
    Abundance = mean(Abundance),Transit = Transit, PD_Colon = PD_Colon)
  T1<- T1[!duplicated(T1),]
  T1$Genus <- i; T1 <- subset(T1, select = c(Genus, Transit, PD_Colon, Abundance, SD))


```

```{r Per genus and per transit, fig.height=30, fig.width=25, warning=F}
my_comparisons <- list(c("ST", "MT"), c("MT", "LT"), c("ST", "LT"))
# Absolute
phylobj.QMP_stable <- subset_samples(phylobj.QMP, Ninetransit == 3)
Top15phylobj <- aggregate_top_taxa(phylobj.QMP_stable, top = 25, "Genus")
Top15 <- psmelt(Top15phylobj); unique(Top15$OTU); unique(Top15$OTU) %in% Top15levels
x <- data.frame(x = unique(Top15$OTU) %in% Top15levels, genus = unique(Top15$OTU))  #check if labelled
Top15$OTU <- as.ordered(factor(Top15$OTU, levels = Top15levels))
# Add extra column days for graphs
Top <- Top15
Top$Transit <- as.ordered(factor(Top$Transit, levels = c("ST", "MT", "LT")))
Top$PD_Colon <- as.ordered(factor(Top$PD_Colon, levels = c("P", "D", "FS")))
Top$Donor <- as.ordered(factor(Top$Donor, levels = c("1", "2", "3", "4", "5", "6")))
Top$Genus <- as.ordered(factor(Top$Genus, levels = Top15levels))
highlist <- data.frame()
for (i in unique(Top$Genus)) {
  if(i!= "Other"){Top2 <- subset(Top, Genus == i)
  Top3 <- data.frame(Genus = i, Abundance = max(Top2$Abundance))
  highlist <- rbind(highlist, Top3)} 
  }
sequence <- highlist[order(-highlist$Abundance),]; sequence <- sequence$Genus
# Proportional
phylobj.prop_2 = transform_sample_counts(phylobj.QMP, function(x) 100 * x/sum(x))
phylobj.prop_stable <- subset_samples(phylobj.prop_2, Ninetransit == 3)
Top15phylobj.prop <- aggregate_top_taxa(phylobj.prop_stable, top = 30, "Genus")
Top15.prop <- psmelt(Top15phylobj.prop); unique(Top15.prop$OTU); unique(Top15.prop$OTU) %in% Top15levels
x <- data.frame(x = unique(Top15.prop$OTU) %in% Top15levels, genus = unique(Top15.prop$OTU))  #check if labelled
Top15.prop$OTU <- as.ordered(factor(Top15.prop$OTU, levels = Top15levels))
# Add extra column days for graphs
Top.prop <- Top15.prop
Top.prop$Transit <- as.ordered(factor(Top.prop$Transit, levels = c("ST", "MT", "LT")))
Top.prop$PD_Colon <- as.ordered(factor(Top.prop$PD_Colon, levels = c("P", "D", "FS")))
Top.prop$Donor <- as.ordered(factor(Top.prop$Donor, levels = c("1", "2", "3", "4", "5", "6")))
Top.prop$Genus <- as.ordered(factor(Top.prop$Genus, levels = Top15levels))
unique(Top$Genus)%in%unique(Top.prop$Genus) 
x <- data.frame(Genus = unique(Top$Genus), x =unique(Top$Genus)%in%unique(Top.prop$Genus)); x <- subset(x, x ==TRUE)
y <- data.frame(Genus = unique(Top.prop$Genus), x =unique(Top.prop$Genus)%in%unique(Top$Genus)); y <- subset(y, x ==TRUE)
z <- rbind(x,y); z <- distinct(z)
# In one graph
# BOXPLOTS
comparisonlistbox <- list(); x =1; rows = 4
for (i in c(1:(length(sequence)/rows))) {
  j = sequence[[i*rows-3]]; k = sequence[[i*rows-2]]; l = sequence[[i*rows-1]]; m = sequence[[i*rows]]
  Topj <- subset(Top, Genus == j); Topk <- subset(Top, Genus == k); Topl <- subset(Top, Genus == l); Topm <- subset(Top, Genus == m); 
  Top1 <- rbind(Topj, Topk, Topl, Topm)
  p <- papertheme(ggplot(Top1, aes(x=Transit, y=Abundance, fill = Transit)) + 
    geom_boxplot(alpha = 0.8) +
    geom_jitter(shape=16, position=position_jitter(0.2), aes(color = Donor)) + 
    facet_grid(PD_Colon~Genus, labeller = labeller(PD_Colon = PD_Colon.labs)) +
    scale_fill_manual(values = transitcolors, labels=c("Short", "Medium", "Long")) + 
    scale_color_manual(values = sixdonors)
    ) + 
    theme(strip.background = element_rect(fill = "#D6E0F4"), strip.text.x = element_text(size = 20, face = "italic"),
    strip.text.y = element_text(size = 20, angle = 270), axis.text.x = element_blank(),
    axis.ticks.x = element_blank(), axis.title.x = element_blank(), legend.text = element_text(size = 18), 
    legend.title = element_text(size = 18, face = "bold"),
    legend.title.align = 0.5) + xlab("Transit time") + 
    # ggtitle(i) +
  ylab(expression(paste("Absolute abundance (cells m",L^{"-1"}, ")"))) + 
  guides(fill = guide_legend("Transit", ncol = 5)) +
  stat_compare_means(comparisons = my_comparisons, label = "p.signif", hide.ns = T, hide.NS = T) # Add significance levels
  if(x%%6 != 0){p <- p + theme(legend.position = "none")}
    # stat_compare_means(label.y = 50) 
  Topj <- subset(Top.prop, Genus == j); Topk <- subset(Top.prop, Genus == k)
  Topl <- subset(Top.prop, Genus == l); Topm <- subset(Top.prop, Genus == m);
  Top2 <- rbind(Topj, Topk, Topl, Topm)
  p2 <- papertheme(ggplot(Top2, aes(x=Transit, y=Abundance, fill = Transit)) + 
    geom_boxplot(alpha = 0.8) +
    geom_jitter(shape=16, position=position_jitter(0.2), aes(color = Donor)) + 
    facet_grid(PD_Colon~Genus, labeller = labeller(PD_Colon = PD_Colon.labs)) +
    scale_fill_manual(values = transitcolors, labels=c("Short", "Medium", "Long")) + 
    scale_color_manual(values = sixdonors)
    ) + 
    theme(strip.background = element_rect(fill = "#D6E0F4"), strip.text.x = element_text(size = 20, face = "italic"),
    strip.text.y = element_text(size = 20, angle = 270), axis.text.x = element_blank(),
    axis.ticks.x = element_blank(), axis.title.x = element_blank(), legend.text = element_text(size = 18), 
    legend.title = element_text(size = 18, face = "bold"),
    legend.title.align = 0.5) + xlab("Transit time") + 
    # ggtitle(i) +
  ylab("Proportional abundance (%)") + 
  guides(fill = guide_legend("Transit", ncol = 5)) +
  stat_compare_means(comparisons = my_comparisons, label = "p.signif", hide.ns = T, hide.NS = T) # Add significance levels
  # 
  if (x%%6 != 0){p2 <- p2 + theme(legend.position = "none")
  p <- ggarrange(p, p2, ncol = 2)}
  if(x%%6 == 0){p <- ggarrange(p, p2, ncol = 2, common.legend = T, legend = "bottom")}
  # print(p); print(p2)
  comparisonlistbox[[x]] <- p;
  y <- x + 1; x <- y
}
for (i in seq(from = 1, to = length(comparisonlistbox), by = 4)) {
  if((i+3) < (length(comparisonlistbox))+1){ print(ggarrange(comparisonlistbox[[i]], comparisonlistbox[[i+1]],
                                                         comparisonlistbox[[i+2]],comparisonlistbox[[i+3]],
                                                         nrow = 4))
  }else{
      if((i+2) < (length(comparisonlistbox))+1){ print(ggarrange(comparisonlistbox[[i]], comparisonlistbox[[i+1]], 
                                                         comparisonlistbox[[i+2]],nrow = 3))
      }else{
          if((i+1) < (length(comparisonlistbox))+1){ print(ggarrange(comparisonlistbox[[i]], comparisonlistbox[[i+1]],
                                                         nrow = 2))
          }else{
               if((i) < (length(comparisonlistbox))+1){ print(ggarrange(comparisonlistbox[[i]],nrow = 1))}
            }
        }
    }
  } 
print(ggarrange(comparisonlistbox[[1]], comparisonlistbox[[2]], comparisonlistbox[[3]], comparisonlistbox[[4]],
                comparisonlistbox[[5]], comparisonlistbox[[6]],  nrow = 6))
# BARPLOTS
comparisonlistbar <- list(); x =1
for (i in c(1:(length(sequence)/rows))) {
  j = sequence[[i*rows-3]]; k = sequence[[i*rows-2]]; l = sequence[[i*rows-1]]; m = sequence[[i*rows]]
  Topj <- subset(Top, Genus == j); Topk <- subset(Top, Genus == k); Topl <- subset(Top, Genus == l); Topm <- subset(Top, Genus == m); 
  Top1 <- rbind(Topj, Topk, Topl, Topm)
  # Top1$comb <- paste0(Top1$Donor, Top1$PD_Colon, Top1$Transit)
  # stat.test <- Top1 %>% group_by(comb) %>% t_test(Abundance ~ Transit) add_significance() %>% filter(p.adj.signif != "ns")
  
  p <- papertheme(ggbarplot(Top1, x="Transit", y="Abundance" , fill = "Transit", add = "mean_sd", facet.by = c("PD_Colon", "Genus"))+
   facet_grid(PD_Colon~Genus, labeller = labeller(PD_Colon = PD_Colon.labs)) +
    scale_fill_manual(values = transitcolors, labels=c("Short", "Medium", "Long"))) +
    theme(strip.background = element_rect(fill = "#D6E0F4"), strip.text.x = element_text(size = 20, face = "italic"),
    strip.text.y = element_text(size = 20, angle = 270), axis.text.x = element_blank(),
    axis.ticks.x = element_blank(), axis.title.x = element_blank(), legend.text = element_text(size = 18), 
    legend.title = element_text(size = 18, face = "bold"),
    legend.title.align = 0.5) + xlab("Transit time") +
  ylab(expression(paste("Absolute abundance (cells m",L^{"-1"}, ")"))) + 
  guides(fill = guide_legend("Transit", ncol = 5)) +
 stat_compare_means(comparisons = my_comparisons, label = "p.signif", hide.ns = T, hide.NS = T) # Add significance levels
  if(x%%6 != 0){p <- p + theme(legend.position = "none")}
  # + stat_compare_means(comparisons = my_comparisons, label = "p.signif", hide.ns = T, hide.NS = T) # Add significance levels
  # stat_compare_means(label.y = 50) 
  
  Topj <- subset(Top.prop, Genus == j); Topk <- subset(Top.prop, Genus == k)
  Topl <- subset(Top.prop, Genus == l); Topm <- subset(Top.prop, Genus == m);
  Top2 <- rbind(Topj, Topk, Topl, Topm)
  
 p2 <- papertheme(ggbarplot(Top2, x="Transit", y="Abundance" , fill = "Transit", add = "mean_sd", facet.by = c("PD_Colon", "Genus"))+
   facet_grid(PD_Colon~Genus, labeller = labeller(PD_Colon = PD_Colon.labs)) +
    scale_fill_manual(values = transitcolors, labels=c("Short", "Medium", "Long"))) +
    theme(strip.background = element_rect(fill = "#D6E0F4"), strip.text.x = element_text(size = 20, face = "italic"),
    strip.text.y = element_text(size = 20, angle = 270), axis.text.x = element_blank(),
    axis.ticks.x = element_blank(), axis.title.x = element_blank(), legend.text = element_text(size = 18), 
    legend.title = element_text(size = 18, face = "bold"),
    legend.title.align = 0.5) + xlab("Transit time") +
  ylab("Proportional abundance (%)") +
  stat_compare_means(comparisons = my_comparisons, label = "p.signif", hide.ns = T, hide.NS = T)+ # Add significance levels
  guides(fill = guide_legend("Transit", ncol = 5))
  # + stat_compare_means(comparisons = my_comparisons, label = "p.signif", hide.ns = T, hide.NS = T) # Add significance levels
  # stat_compare_means(label.y = 50) 
  if (x%%6 != 0){p2 <- p2 + theme(legend.position = "none")
  p <- ggarrange(p, p2, ncol = 2)}
  if(x%%6 == 0){p <- ggarrange(p, p2, ncol = 2, common.legend = T, legend = "bottom")} 
  # p <- annotate_figure(p, top = text_grob(i, color = "black", face = "italic", size = 20))
  # print(p)
  comparisonlistbar[[x]] <- p; x = x + 1
}
for (i in seq(from = 1, to = length(comparisonlistbar), by = 4)) {
  if((i+3) < (length(comparisonlistbar))+1){print(ggarrange(comparisonlistbar[[i]], comparisonlistbar[[i+1]],
                                                            comparisonlistbar[[i+2]],comparisonlistbar[[i+3]],
                                                         nrow = 4))
  }else{
      if((i+2) < (length(comparisonlistbar))+1){print(ggarrange(comparisonlistbar[[i]], comparisonlistbar[[i+1]], 
                                                                comparisonlistbar[[i+2]],nrow = 3))
      }else{
          if((i+1) < (length(comparisonlistbar))+1){print(ggarrange(comparisonlistbar[[i]], comparisonlistbar[[i+1]],
                                                                    nrow = 2))
          }else{
               if((i) < (length(comparisonlistbar))+1){print(ggarrange(comparisonlistbar[[i]],nrow = 1))}
            }
        }
    }
}
print(ggarrange(comparisonlistbar[[1]], comparisonlistbar[[2]], comparisonlistbar[[3]], comparisonlistbar[[4]],
                comparisonlistbar[[5]], comparisonlistbar[[6]],  nrow = 6))
```

```{r Paperplot genus, transit and donor in one, fig.height=30, fig.width=25, warning=F}
my_comparisons <- list(c("ST", "MT"), c("MT", "LT"), c("ST", "LT"))
bars = T; barstransit = F
# Absolute
phylobj.QMP_stable <- subset_samples(phylobj.QMP, Ninetransit == 3)
Top15phylobj <- aggregate_top_taxa(phylobj.QMP_stable, top = 20, "Genus")
Top15 <- psmelt(Top15phylobj); unique(Top15$OTU); unique(Top15$OTU) %in% Top15levels
x <- data.frame(x = unique(Top15$OTU) %in% Top15levels, genus = unique(Top15$OTU))  #check if labelled
Top15$OTU <- as.ordered(factor(Top15$OTU, levels = Top15levels))
# Add extra column days for graphs
Top <- Top15
Top$Transit <- as.ordered(factor(Top$Transit, levels = c("ST", "MT", "LT")))
Top$PD_Colon <- as.ordered(factor(Top$PD_Colon, levels = c("P", "D", "FS")))
Top$Donor <- as.ordered(factor(Top$Donor, levels = c("1", "2", "3", "4", "5", "6")))
Top$Genus <- as.ordered(factor(Top$Genus, levels = Top15levels))
# Proportional
phylobj.prop_2 = transform_sample_counts(phylobj.QMP, function(x) 100 * x/sum(x))
phylobj.prop_stable <- subset_samples(phylobj.prop_2, Ninetransit == 3)
Top15phylobj.prop <- aggregate_top_taxa(phylobj.prop_stable, top = 25, "Genus")
Top15.prop <- psmelt(Top15phylobj.prop); unique(Top15.prop$OTU); unique(Top15.prop$OTU) %in% Top15levels
x <- data.frame(x = unique(Top15.prop$OTU) %in% Top15levels, genus = unique(Top15.prop$OTU))  #check if labelled
Top15.prop$OTU <- as.ordered(factor(Top15.prop$OTU, levels = Top15levels))
# Add extra column days for graphs
Top.prop <- Top15.prop
Top.prop$Transit <- as.ordered(factor(Top.prop$Transit, levels = c("ST", "MT", "LT")))
Top.prop$PD_Colon <- as.ordered(factor(Top.prop$PD_Colon, levels = c("P", "D", "FS")))
Top.prop$Donor <- as.ordered(factor(Top.prop$Donor, levels = c("1", "2", "3", "4", "5", "6")))
Top.prop$Genus <- as.ordered(factor(Top.prop$Genus, levels = Top15levels))
unique(Top$Genus)%in%unique(Top.prop$Genus) 
x <- data.frame(Genus = unique(Top$Genus), x =unique(Top$Genus)%in%unique(Top.prop$Genus)); x <- subset(x, x ==TRUE)
y <- data.frame(Genus = unique(Top.prop$Genus), x =unique(Top.prop$Genus)%in%unique(Top$Genus)); y <- subset(y, x ==TRUE)
z <- rbind(x,y); z <- distinct(z)
# Statistics
# Non parametric to 1-way ANOVA, alternative - Kruskal-Wallis
correction = "holm"
# Between transits
gglist <- list(); gglist2 <- list(); y = 1; df.grpPMP <- data.frame(); df.grpQMP <- data.frame(); statlist <- list()
message("Kruskal-Wallis tests between transit times")
df.pwc <- data.frame(); df.pwc2 <- data.frame()
# PLOTS
comparisonlistbar <- list(); x =1; y =1
for (i in sort(z$Genus)) {
  Top1 <- subset(Top, Genus == i); Top2 <- subset(Top.prop, Genus == i)
  
  T1 <- Top1 %>% group_by(Transit, PD_Colon) %>% summarise(SD = sd(Abundance), 
    Abundance = mean(Abundance),Transit = Transit, PD_Colon = PD_Colon) 
  T1<- T1[!duplicated(T1), ]; T1$min = T1$Abundance - T1$SD; T1$max = T1$Abundance + T1$SD
    max = max(T1$max); min = min(T1$min)
    
  # Create data frame with means and SD
    T1 <- subset(Top1, Abundance > 0); T1$Abundance <- as.numeric(T1$Abundance)
  T1 <- T1 %>% group_by(Transit, PD_Colon) %>% summarise(SD = sd(Abundance), 
    Abundance = mean(Abundance),Transit = Transit, PD_Colon = PD_Colon); T1<- T1[!duplicated(T1),]
  T1$Genus <- i; T1 <- subset(T1, select = c(Genus, Transit, PD_Colon, Abundance, SD))
  df.grpQMP <- rbind(df.grpQMP, T1)
  
  
  T1 <- subset(Top2, Abundance > 0); T1$Abundance <- as.numeric(T1$Abundance)
  T1 <- T1 %>% group_by(Transit, PD_Colon) %>% summarise(SD = sd(Abundance), 
    Abundance = mean(Abundance),Transit = Transit, PD_Colon = PD_Colon); T1<- T1[!duplicated(T1),]
  T1$Genus <- i; T1 <- subset(T1, select = c(Genus, Transit, PD_Colon, Abundance, SD)); df.grpPMP <- rbind(df.grpPMP, T1)  
    
    
    
  # Top1$comb <- paste0(Top1$Donor, Top1$PD_Colon, Top1$Transit)
  # stat.test <- Top1 %>% group_by(comb) %>% t_test(Abundance ~ Transit) add_significance() %>% filter(p.adj.signif != "ns")
# Absolute abundance
p <- papertheme(ggbarplot(Top1, x="Transit", y="Abundance" , fill = "Donor", 
                  add = "mean_sd", facet.by = c("PD_Colon"), position = position_dodge())) 
  if(bars == F){ 
    p <- papertheme(ggboxplot(Top1, x="Transit", y="Abundance" , fill = "Donor", 
                              facet.by = c("PD_Colon")))
  }
p<-p + facet_grid(PD_Colon~., labeller = labeller(PD_Colon = PD_Colon.labs)) +
  scale_fill_manual(values = sixdonors) + scale_x_discrete(labels = c("Short", "Medium", "Long"))+
  theme(strip.background = element_rect(fill = "#D6E0F4"), strip.text.x = element_text(size = 20),
        strip.text.y = element_text(size = 20, angle = 270), legend.text = element_text(size = 18), 
        legend.title = element_text(size = 18, face = "bold"),
        legend.title.align = 0.5) + xlab("Transit time") +
  ylab(expression(paste("Absolute abundance (cells m",L^{"-1"}, ")"))) +
  guides(fill = guide_legend("Donor", ncol = 6)) + theme(strip.text.y = element_blank())
  if(x%%4 != 0){p <- p + theme(legend.position = "none")}
  # + stat_compare_means(comparisons = my_comparisons, label = "p.signif", hide.ns = T, hide.NS = T) # Add significance levels
  # stat_compare_means(label.y = 50) 
p3 <- papertheme(ggbarplot(Top1, x="Transit", y="Abundance" , fill = "Transit", 
                           add = "mean_sd", facet.by = c("PD_Colon")))
if(barstransit == F){
  p3 <- papertheme(ggboxplot(Top1, x="Transit", y="Abundance" , fill = "Transit", facet.by = c("PD_Colon")))
  } 
p3 <- p3 + facet_grid(PD_Colon~., labeller = labeller(PD_Colon = PD_Colon.labs)) +
    scale_fill_manual(values = transitcolors) + scale_x_discrete(labels = c("Short", "Medium", "Long"))+
    theme(strip.background = element_rect(fill = "#D6E0F4"), strip.text.x = element_text(size = 20),
    strip.text.y = element_text(size = 20, angle = 270), legend.text = element_text(size = 18), 
    legend.title = element_text(size = 18, face = "bold"),
    legend.title.align = 0.5) + xlab("Transit time") +
  ylab(expression(paste("Absolute abundance (cells m",L^{"-1"}, ")"))) + 
  guides(fill = guide_legend("Transit", ncol = 5)) +
  theme(axis.title.y = element_blank(), axis.text.y=element_blank(), axis.ticks.y = element_blank()) 
if(x%%4 != 0){p3 <- p3 + theme(legend.position = "none")}
gen = i
# plot1 <- ggarrange(p, p3, widths = c(1,0.3))
ymax = max(ggplot_build(p)$layout$panel_scales_y[[1]]$range$range)
message("Quantitative")
plotstat <- statplot(dataframe = Top1, AboveZero = T, ymax = ymax) + theme(legend.position = "none")
ymax = max(ggplot_build(plotstat)$layout$panel_scales_y[[1]]$range$range); ymin = min(ggplot_build(p)$layout$panel_scales_y[[1]]$range$range)
plotstat <- plotstat + scale_y_continuous(limits = c(ymin, ymax))
plot1 <- ggarrange(p + theme(legend.position = "none") +scale_y_continuous(limits = c(ymin, ymax)) , plotstat, widths = c(1,0.3))
# if(x ==1|x==5|x==9|x==13|x==17){
#   title <- grobTree(rectGrob(gp=gpar(fill="#D6E0F4", color = "#D6E0F4")), textGrob("Absolute abundance", gp=gpar(fontsize=15, col="Black")))
#   plot1 <- ggarrange(
#     ggarrange(NULL, title, NULL, ncol = 3, widths = c(0.1,1,0.1)), 
#     plot1, nrow = 2, heights = c(0.05,1))
#   }
# print(plot1)
stats <- statplot(dataframe = Top1, AboveZero = T, plot = "ymax", showstat = F, yaxis = T)
# Proportional abundance
p2 <- papertheme(ggbarplot(Top2, x="Transit", y="Abundance" , fill = "Donor", 
                  add = "mean_sd", facet.by = c("PD_Colon"), position = position_dodge())) 
  if(bars == F){ 
    p2 <- papertheme(ggboxplot(Top2, x="Transit", y="Abundance" , fill = "Donor", 
                              facet.by = c("PD_Colon")))
  }
p2<-p2 + facet_grid(PD_Colon~., labeller = labeller(PD_Colon = PD_Colon.labs)) +
  scale_fill_manual(values = sixdonors) + scale_x_discrete(labels = c("Short", "Medium", "Long"))+
  theme(strip.background = element_rect(fill = "#D6E0F4"), strip.text.x = element_text(size = 20),
        strip.text.y = element_text(size = 20, angle = 270), legend.text = element_text(size = 18), 
        legend.title = element_text(size = 18, face = "bold"),
        legend.title.align = 0.5) + xlab("Transit time") + ylab("Proportional abundance (%)") +
  guides(fill = guide_legend("Donor", ncol = 6)) + theme(strip.text.y = element_blank())
  if(x%%4 != 0){p2 <- p2 + theme(legend.position = "none")}
  # + stat_compare_means(comparisons = my_comparisons, label = "p.signif", hide.ns = T, hide.NS = T) # Add significance levels
  # stat_compare_means(label.y = 50) 
p4 <- papertheme(ggbarplot(Top2, x="Transit", y="Abundance" , fill = "Transit", 
                           add = "mean_sd", facet.by = c("PD_Colon")))
if(barstransit == F){
  p4 <- papertheme(ggboxplot(Top2, x="Transit", y="Abundance" , fill = "Transit", facet.by = c("PD_Colon")))
  } 
p4 <- p4 + facet_grid(PD_Colon~., labeller = labeller(PD_Colon = PD_Colon.labs)) +
    scale_fill_manual(values = transitcolors) + scale_x_discrete(labels = c("Short", "Medium", "Long"))+
    theme(strip.background = element_rect(fill = "#D6E0F4"), strip.text.x = element_text(size = 20),
    strip.text.y = element_text(size = 20, angle = 270), legend.text = element_text(size = 18), 
    legend.title = element_text(size = 18, face = "bold"),
    legend.title.align = 0.5) + xlab("Transit time") + ylab("Proportional abundance (%)") + 
  guides(fill = guide_legend("Transit", ncol = 5)) +
  theme(axis.title.y = element_blank(), axis.text.y=element_blank(), axis.ticks.y = element_blank()) 
if(x%%4 != 0){p4 <- p4 + theme(legend.position = "none")}
# plot2 <- ggarrange(p2, p4, widths = c(1,0.3))
ymax = max(ggplot_build(p2)$layout$panel_scales_y[[1]]$range$range)
message("Proportional")
plotstat <- statplot(dataframe = Top2, AboveZero = T, ymax = ymax); boxlegend <-cowplot::get_legend(plotstat)
legends <- ggdraw(plot_grid(NULL,cowplot::get_legend(p2), boxlegend,NULL, ncol=4, align='v'))
ymax = max(xx <- ggplot_build(plotstat)$layout$panel_scales_y[[1]]$range$range)
ymin = min(ggplot_build(p2)$layout$panel_scales_y[[1]]$range$range)
plotstat <- plotstat + scale_y_continuous(limits = c(ymin, ymax)) + theme(legend.position = "none")
p2 <- p2 + theme(legend.position = "none") + scale_y_continuous(limits = c(ymin, ymax)); box <- box + theme(legend.position = "none")
gen = i; plot2 <- ggarrange(p2, plotstat, widths = c(1,0.3))
# if(x ==1|x==5|x==9|x==13|x==17){
#   title <- grobTree(rectGrob(gp=gpar(fill="#D6E0F4", color = "#D6E0F4")), textGrob("Proportional abundance", gp=gpar(fontsize=15, col="Black")))
#   plot2 <- ggarrange(
#     ggarrange(NULL, title, NULL, ncol = 3, widths = c(0.1,1,0.1)), 
#     plot2, nrow = 2, heights = c(0.05,1))
#   }
# print(plot2)
stats2 <- statplot(dataframe = Top2, AboveZero = T, plot = "ymax", showstat = F, yaxis = T)
statlist[[x]] <- ggarrange(stats, stats2, ncol = 2); rm(stats, stats2)
  if (x%%4 != 0){p2 <- p2 + theme(legend.position = "none")
  p <- ggarrange(plot1, plot2, ncol = 2)}
  p<- ggarrange(plot1, plot2, ncol = 2)
  if(x%%4 == 0){p <- ggarrange(plot1, plot2, ncol = 2, common.legend = T, legend = "bottom")
  p <- ggarrange(p, legends, nrow = 2, heights = c(1,0.1))} 
  p <- annotate_figure(p, top = text_grob(i, color = "black", face = "italic", size = 20))
  # print(p)
  comparisonlistbar[[x]] <- p; x = x + 1
}
for (i in seq(from = 1, to = (length(comparisonlistbar)-1), by = 4)) {
 comparisons <- ggarrange(comparisonlistbar[[i]], comparisonlistbar[[i+1]], comparisonlistbar[[i+2]],comparisonlistbar[[i+3]], nrow = 4)
   print(comparisons)
}
# ggarrange(statlist[[1]] + border(), statlist[[2]] + border(), statlist[[3]] + border(), statlist[[4]] + border(), statlist[[5]] + border(), 
#           statlist[[6]] + border(), statlist[[8]] + border(), statlist[[9]] + border(), statlist[[10]] + border(), statlist[[11]] + border(),
#           statlist[[12]] + border(), statlist[[13]] + border(), nrow = 4, ncol = 3)
# 
# ggarrange(statlist[[14]] + border(), statlist[[15]] + border(), statlist[[16]] + border(), statlist[[17]] + border(), 
#           statlist[[18]] + border(), statlist[[19]] + border(), statlist[[20]] + border(), statlist[[21]] + border(), nrow = 4, ncol = 2)
# View(df.grpQMP); View(df.grpPMP)
```



### Paperplot differences faeces and eventual
```{r fig.height=37, fig.width=30}

# Top15 in faeces
Top25 <- aggregate_top_taxa(phylobj.QMPFS, top= 16, "Genus")
taxa <- taxa_names(Top25)
taxa <- c("Alistipes","Anaerostipes", "Bacteroides","Bifidobacterium"  ,   "Blautia",            
"Clostridiales" ,      "Coprococcus"  ,       "Erysipelotrichaceae", "Faecalibacterium" ,   "Fusicatenibacter"   ,
 "Lachnospiraceae",  "Prevotella"    ,      "Roseburia"   ,        "Ruminococcaceae",   "Ruminococcus")

# Stabilised + Faecal sample
# Make a data frame with a column for the read counts of each sample
phylobj.QMP_stable_FS <- merge_phyloseq(subset_samples(phylobj.QMP, Ninetransit == 3), phylobj.QMPFS)
# phylobj.QMP_stable_FS <- subset_samples(phylobj.QMP, Ninetransit == 3)
ntop = 75
phylobj.QMP_stable_FS = transform_sample_counts(phylobj.QMP_stable_FS, function(x) 100 * x/sum(x))
Top15phylobj <- aggregate_top_taxa(phylobj.QMP_stable_FS, top = ntop, "Genus") 
Top15 <- psmelt(Top15phylobj)

samp1 <- subset(Top15, Donor ==1 | Donor ==2); samp1$intTransit <- "Short transit donor"
samp2 <- subset(Top15, Donor ==3 | Donor ==4); samp2$intTransit <- "Medium transit donor"
samp3 <- subset(Top15, Donor ==5 | Donor ==6); samp3$intTransit <- "Long transit donor"
Top15 <- rbind(samp1, samp2, samp3) 

# for loop
plotlist <- list(); x =1
for (i in taxa) {
df <- subset(Top15, Genus == i)
df$comb <- paste0(df$PD_Colon, df$Donor, df$Transit)
df <- df %>%  group_by(comb) %>%
  summarise(SD = sd(Abundance), mean = mean(Abundance), Transit = Transit, intTransit = intTransit, PD_Colon = PD_Colon, Donor = Donor)
df <- df[!duplicated(df), ]
df$Transit <- as.ordered(factor(df$Transit, levels = c("FS", "ST", "MT", "LT")))
df$intTransit <- as.ordered(factor(df$intTransit, levels = c("Short transit donor", "Medium transit donor", "Long transit donor")))
df$PD_Colon <- as.ordered(factor(df$PD_Colon, levels = c("P", "D")))
df$Donor <- as.ordered(factor(df$Donor, levels = c("1", "2", "3", "4", "5", "6")))

plot <- papertheme(ggplot(df, aes(x = Transit, y = mean, fill = Transit))) + geom_bar(stat="identity") + 
  facet_grid(PD_Colon~intTransit+Donor , labeller = labeller(Donor = donor.labs, PD_Colon = PD_Colon.labs))+
  scale_fill_manual(values = c(transitcolors, "FS" = "#cdc774"), labels = c("F", "S", "M", "L"))+
  scale_x_discrete(labels = c("F", "S", "M", "L"))+
  geom_errorbar(aes(ymin = mean - SD, ymax = mean + SD, width=0.1), alpha = 0.8)+
  theme(strip.background = element_rect(fill = "#D6E0F4"), strip.text.x = element_text(size = 20),
    strip.text.y = element_text(size = 20, angle = 270), 
    legend.position = "none",
    # axis.text.x = element_blank(),
    # axis.ticks.x = element_blank(), 
    # axis.title.x = element_blank(), 
    legend.text = element_text(size = 15,
        face = "italic"), legend.title = element_text(size = 15, face = "bold"),
    legend.title.align = 0.5) + xlab("Transit time")  +
  ggtitle(i)+
  # scale_y_continuous(labels = scales::percent_format(accuracy = 1))+
  ylab("Proportional abundance (%)")
plot <- Fusetoplayer(plot, "most")  
print(plot) 
plotlist[[x]] <- plot
x = x +1
}


plot <- ggarrange(plotlist[[1]],plotlist[[2]],plotlist[[3]],plotlist[[4]],plotlist[[5]],
                  plotlist[[6]],plotlist[[7]],plotlist[[8]],plotlist[[9]],plotlist[[10]],
                  plotlist[[11]],plotlist[[12]],plotlist[[13]],plotlist[[14]],plotlist[[15]], nrow = 5, ncol = 3)
annotate_figure(plot, bottom = text_grob("F = faecal sample, S, M and L are resp. short, medium and long SHIME transit times", 
               color = "black", face = "bold", size = 20))

# plot <- ggarrange(plotlist[[6]],plotlist[[7]],plotlist[[8]],plotlist[[9]],plotlist[[10]], nrow = 5) 
# annotate_figure(plot, bottom = text_grob("F = faecal sample, S, M and L are resp. short, medium and long SHIME transit times", 
#                color = "black", face = "bold", size = 14))
# 
# plot <- ggarrange(plotlist[[11]],plotlist[[12]],plotlist[[13]],plotlist[[14]],plotlist[[15]], nrow = 5) 
# annotate_figure(plot, bottom = text_grob("F = faecal sample, S, M and L are resp. short, medium and long SHIME transit times", 
#                color = "black", face = "bold", size = 14))
```


### Spearman correlation
```{r Correlation colons split, fig.height=15, fig.width=15}
order <- c("Acidaminococcus", "Akkermansia", "Alistipes", "Anaeroglobus", "Anaerostipes",	"Bacteria", 	"Bacteroides", 	"Bifidobacterium", 	"Bilophila", 	"Blautia", 	"Cloacibacillus", 	"Clostridiales", 	"Clostridium_XlVa", 	"Coprococcus", 	"Desulfovibrio", 	"Dialister", 	"Dorea", 	"Eisenbergiella", 	"Enterobacteriaceae", 	"Erysipelotrichaceae", 	"Faecalibacterium", 	"Fusicatenibacter", 	"Fusobacterium", 	"Lachnospiraceae", 	"Megamonas", 	"Megasphaera", 	"Mitsuokella", 	"Parabacteroides", 	"Phascolarctobacterium", 	"Prevotella", 	"Pseudomonas", 	"Roseburia", 	"Ruminococcaceae", 	"Ruminococcus", 	"Sutterella", 	"Veillonella")
order <- c("Veillonella", 	"Sutterella", 	"Ruminococcus", 	"Ruminococcaceae", 	"Roseburia", 	"Pseudomonas", 	"Prevotella", 	"Phascolarctobacterium", 	"Parabacteroides", 	"Mitsuokella", 	"Megasphaera", 	"Megamonas", 	"Lachnospiraceae", 	"Fusobacterium", 	"Fusicatenibacter", 	"Faecalibacterium", 	"Erysipelotrichaceae", 	"Enterobacteriaceae", 	"Eisenbergiella", 	"Dorea", 	"Dialister", 	"Desulfovibrio", 	"Coprococcus", 	"Clostridium_XlVa", 	"Clostridiales", 	"Cloacibacillus", 	"Blautia", 	"Bilophila", 	"Bifidobacterium", 	"Bacteroides", 	"Bacteria", 	"Anaerostipes", 	"Anaeroglobus", 	"Alistipes", 	"Akkermansia", 	"Acidaminococcus")
# Absolute
phylobj.QMP_stable <- subset_samples(phylobj.QMP, Ninetransit == 3)
Top15phylobj <- aggregate_taxa(phylobj.QMP_stable, level = "Genus")
# Top15phylobj <- aggregate_top_taxa(phylobj.QMP_stable, top = 20, "Genus")
Top15 <- psmelt(Top15phylobj); unique(Top15$OTU); unique(Top15$OTU) %in% Top15levels
x <- data.frame(x = unique(Top15$OTU) %in% Top15levels, genus = unique(Top15$OTU))  #check if labelled
Top15$OTU <- as.ordered(factor(Top15$OTU, levels = Top15levels))
# Add extra column days for graphs
Top <- Top15
Top$Transit <- as.ordered(factor(Top$Transit, levels = c("ST", "MT", "LT")))
Top$PD_Colon <- as.ordered(factor(Top$PD_Colon, levels = c("P", "D", "FS")))
Top$Donor <- as.ordered(factor(Top$Donor, levels = c("1", "2", "3", "4", "5", "6")))
Top$Genus <- as.ordered(factor(Top$Genus, levels = Top15levels))
# Add numbers
df.1 <- subset(Top, PD_Colon == "P" & Transit == "ST"); df.1$TT <- 8
df.2 <- subset(Top, PD_Colon == "P" & Transit == "MT"); df.2$TT <- 16
df.3 <- subset(Top, PD_Colon == "P" & Transit == "LT"); df.3$TT <- 24
df.4 <- subset(Top, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 13
df.5 <- subset(Top, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 26
df.6 <- subset(Top, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 39
Top <- rbind(df.1, df.2, df.3, df.4, df.5, df.6)
rm(df.1, df.2, df.3, df.4, df.5, df.6)
dfcor <- data.frame()
for(i in na.omit(unique(Top$Genus))){
  df1 <- subset(Top, Genus == i);  df1 <- subset(df1, Abundance > 0)
  for(x in c("P", "D")){
  df<- subset(df1, PD_Colon == x)
  t <- cor(df$Abundance, df$TT, method = "spearman")
  corr <- cor.test(df$Abundance, df$TT, method = 'spearman'); 
  
  if(is.na(corr[["p.value"]])){
    sig <- "-"
    }else{
      if(corr[["p.value"]]<0.001){
        sig <- "***"
        }else{
          if(corr[["p.value"]]<0.01){
            sig <- "**"
          }else{
              if(corr[["p.value"]]<0.05){
                sig<-"*"
              }else{
                  sig<-""}}}}
  
  dfcor <- rbind(dfcor, data.frame(genus = i, cor = corr[["estimate"]][["rho"]], 
                 PD_Colon = x, pvalue = corr[["p.value"]], sig = sig, comm = "Quantitative microbial community"))
  }}
# Proportional
phylobj.prop_2 = transform_sample_counts(phylobj.QMP, function(x) 100 * x/sum(x))
phylobj.prop_stable <- subset_samples(phylobj.prop_2, Ninetransit == 3)
Top15phylobj.prop <- aggregate_taxa(phylobj.prop_stable, level = "Genus")
Top15.prop <- psmelt(Top15phylobj.prop); unique(Top15.prop$OTU); unique(Top15.prop$OTU) %in% Top15levels
x <- data.frame(x = unique(Top15.prop$OTU) %in% Top15levels, genus = unique(Top15.prop$OTU))  #check if labelled
Top15.prop$OTU <- as.ordered(factor(Top15.prop$OTU, levels = Top15levels))
# Add extra column days for graphs
Top.prop <- Top15.prop
Top.prop$Transit <- as.ordered(factor(Top.prop$Transit, levels = c("ST", "MT", "LT")))
Top.prop$PD_Colon <- as.ordered(factor(Top.prop$PD_Colon, levels = c("P", "D", "FS")))
Top.prop$Donor <- as.ordered(factor(Top.prop$Donor, levels = c("1", "2", "3", "4", "5", "6")))
Top.prop$Genus <- as.ordered(factor(Top.prop$Genus, levels = Top15levels))
# Add numbers
df.1 <- subset(Top.prop, PD_Colon == "P" & Transit == "ST"); df.1$TT <- 8
df.2 <- subset(Top.prop, PD_Colon == "P" & Transit == "MT"); df.2$TT <- 16
df.3 <- subset(Top.prop, PD_Colon == "P" & Transit == "LT"); df.3$TT <- 24
df.4 <- subset(Top.prop, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 13
df.5 <- subset(Top.prop, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 26
df.6 <- subset(Top.prop, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 39
Top.prop <- rbind(df.1, df.2, df.3, df.4, df.5, df.6)
rm(df.1, df.2, df.3, df.4, df.5, df.6)
dfcor2 <- data.frame()
for(i in na.omit(unique(Top.prop$Genus))){
  df1 <- subset(Top.prop, Genus == i);  df1 <- subset(df1, Abundance > 0)
  for(x in c("P", "D")){
  df<- subset(df1, PD_Colon == x)
  t <- cor(df$Abundance, df$TT, method = "spearman")
  corr <- cor.test(df$Abundance, df$TT, method = 'spearman'); 
  
  if(is.na(corr[["p.value"]])){
    sig <- "-"
    }else{
      if(corr[["p.value"]]<0.001){
        sig <- "***"
        }else{
          if(corr[["p.value"]]<0.01){
            sig <- "**"
          }else{
              if(corr[["p.value"]]<0.05){
                sig<-"*"
              }else{
                  sig<-""}}}}
  
  dfcor2 <- rbind(dfcor2, data.frame(genus = i, cor = corr[["estimate"]][["rho"]], 
                 PD_Colon = x, pvalue = corr[["p.value"]], sig = sig, comm = "Proportional microbial community"))
  }}
#Combine abs and prop abundances in one
dfcor <- rbind(dfcor, dfcor2)
dfcor <- na.omit(dfcor); dfcor$comm <- as.ordered(factor(dfcor$comm, 
                              levels = c("Quantitative microbial community", "Proportional microbial community")))
dfcor$PD_Colon <- as.ordered(factor(dfcor$PD_Colon, levels = c("P", "D")))
dfcor$genus <- as.ordered(factor(dfcor$genus, levels = order))
plot <- papertheme(ggplot(dfcor, aes(x = PD_Colon, y = genus, fill =cor))) + geom_tile() +
  scale_fill_gradient2(na.value = "grey50",low="blue", high="red") + 
  geom_text(aes(label = sig)) +
  theme(axis.text.y = element_text(face="italic")) + ylab("Genus") + xlab("Colon vessel") + 
  scale_x_discrete(breaks=c("P","D"), labels=c("Proximal colon", "Distal colon")) +
  labs(fill = expression(rho)) +
  facet_grid(cols = vars(comm)) +theme(strip.background = element_rect(fill = "#D6E0F4"))
plot
```

```{r Correlation colons together, fig.height=15, fig.width=12}
order <- c("Acidaminococcus", "Akkermansia", "Alistipes", "Anaeroglobus", "Anaerostipes",	"Bacteria", 	"Bacteroides", 	"Bifidobacterium", 	"Bilophila", 	"Blautia", 	"Cloacibacillus", 	"Clostridiales", 	"Clostridium_XlVa", 	"Coprococcus", 	"Desulfovibrio", 	"Dialister", 	"Dorea", 	"Eisenbergiella", 	"Enterobacteriaceae", 	"Erysipelotrichaceae", 	"Faecalibacterium", 	"Fusicatenibacter", 	"Fusobacterium", 	"Lachnospiraceae", 	"Megamonas", 	"Megasphaera", 	"Mitsuokella", 	"Parabacteroides", 	"Phascolarctobacterium", 	"Prevotella", 	"Pseudomonas", 	"Roseburia", 	"Ruminococcaceae", 	"Ruminococcus", 	"Sutterella", 	"Veillonella")
order <- c("Veillonella", 	"Sutterella", 	"Ruminococcus", 	"Ruminococcaceae", 	"Roseburia", 	"Pseudomonas", 	"Prevotella", 	"Phascolarctobacterium", 	"Parabacteroides", 	"Mitsuokella", 	"Megasphaera", 	"Megamonas", 	"Lachnospiraceae", 	"Fusobacterium", 	"Fusicatenibacter", 	"Faecalibacterium", 	"Erysipelotrichaceae", 	"Enterobacteriaceae", 	"Eisenbergiella", 	"Dorea", 	"Dialister", 	"Desulfovibrio", 	"Coprococcus", 	"Clostridium_XlVa", 	"Clostridiales", 	"Cloacibacillus", 	"Blautia", 	"Bilophila", 	"Bifidobacterium", 	"Bacteroides", 	"Bacteria", 	"Anaerostipes", 	"Anaeroglobus", 	"Alistipes", 	"Akkermansia", 	"Acidaminococcus")
# Absolute
phylobj.QMP_stable <- subset_samples(phylobj.QMP, Ninetransit == 3)
Top15phylobj <- aggregate_taxa(phylobj.QMP_stable, level = "Genus")
# Top15phylobj <- aggregate_top_taxa(phylobj.QMP_stable, top = 20, "Genus")
Top15 <- psmelt(Top15phylobj); unique(Top15$OTU); unique(Top15$OTU) %in% Top15levels
x <- data.frame(x = unique(Top15$OTU) %in% Top15levels, genus = unique(Top15$OTU))  #check if labelled
Top15$OTU <- as.ordered(factor(Top15$OTU, levels = Top15levels))
# Add extra column days for graphs
Top <- Top15
Top$Transit <- as.ordered(factor(Top$Transit, levels = c("ST", "MT", "LT")))
Top$PD_Colon <- as.ordered(factor(Top$PD_Colon, levels = c("P", "D", "FS")))
Top$Donor <- as.ordered(factor(Top$Donor, levels = c("1", "2", "3", "4", "5", "6")))
Top$Genus <- as.ordered(factor(Top$Genus, levels = Top15levels))
# Add numbers
df.1 <- subset(Top, PD_Colon == "P" & Transit == "ST"); df.1$TT <- 8
df.2 <- subset(Top, PD_Colon == "P" & Transit == "MT"); df.2$TT <- 16
df.3 <- subset(Top, PD_Colon == "P" & Transit == "LT"); df.3$TT <- 24
df.4 <- subset(Top, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 13
df.5 <- subset(Top, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 26
df.6 <- subset(Top, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 39
Top <- rbind(df.1, df.2, df.3, df.4, df.5, df.6)
rm(df.1, df.2, df.3, df.4, df.5, df.6)
dfcor <- data.frame()
for(i in na.omit(unique(Top$Genus))){
  df1 <- subset(Top, Genus == i);  df1 <- subset(df1, Abundance > 0)
  
  df<- df1
  t <- cor(df$Abundance, df$TT, method = "spearman")
  corr <- cor.test(df$Abundance, df$TT, method = 'spearman'); 
  
  if(is.na(corr[["p.value"]])){
    sig <- "-"
    }else{
      if(corr[["p.value"]]<0.001){
        sig <- "***"
        }else{
          if(corr[["p.value"]]<0.01){
            sig <- "**"
          }else{
              if(corr[["p.value"]]<0.05){
                sig<-"*"
              }else{
                  sig<-""}}}}
  
  dfcor <- rbind(dfcor, data.frame(genus = i, cor = corr[["estimate"]][["rho"]], 
                 PD_Colon = "Combined", pvalue = corr[["p.value"]], sig = sig, comm = "Quantitative microbial community"))
  }
# Proportional
phylobj.prop_2 = transform_sample_counts(phylobj.QMP, function(x) 100 * x/sum(x))
phylobj.prop_stable <- subset_samples(phylobj.prop_2, Ninetransit == 3)
Top15phylobj.prop <- aggregate_taxa(phylobj.prop_stable, level = "Genus")
Top15.prop <- psmelt(Top15phylobj.prop); unique(Top15.prop$OTU); unique(Top15.prop$OTU) %in% Top15levels
x <- data.frame(x = unique(Top15.prop$OTU) %in% Top15levels, genus = unique(Top15.prop$OTU))  #check if labelled
Top15.prop$OTU <- as.ordered(factor(Top15.prop$OTU, levels = Top15levels))
# Add extra column days for graphs
Top.prop <- Top15.prop
Top.prop$Transit <- as.ordered(factor(Top.prop$Transit, levels = c("ST", "MT", "LT")))
Top.prop$PD_Colon <- as.ordered(factor(Top.prop$PD_Colon, levels = c("P", "D", "FS")))
Top.prop$Donor <- as.ordered(factor(Top.prop$Donor, levels = c("1", "2", "3", "4", "5", "6")))
Top.prop$Genus <- as.ordered(factor(Top.prop$Genus, levels = Top15levels))
# Add numbers
df.1 <- subset(Top.prop, PD_Colon == "P" & Transit == "ST"); df.1$TT <- 8
df.2 <- subset(Top.prop, PD_Colon == "P" & Transit == "MT"); df.2$TT <- 16
df.3 <- subset(Top.prop, PD_Colon == "P" & Transit == "LT"); df.3$TT <- 24
df.4 <- subset(Top.prop, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 13
df.5 <- subset(Top.prop, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 26
df.6 <- subset(Top.prop, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 39
Top.prop <- rbind(df.1, df.2, df.3, df.4, df.5, df.6)
rm(df.1, df.2, df.3, df.4, df.5, df.6)
dfcor2 <- data.frame()
for(i in na.omit(unique(Top.prop$Genus))){
  df1 <- subset(Top.prop, Genus == i);  df1 <- subset(df1, Abundance > 0)
  
  df<- df1
  t <- cor(df$Abundance, df$TT, method = "spearman")
  corr <- cor.test(df$Abundance, df$TT, method = 'spearman'); 
  
  if(is.na(corr[["p.value"]])){
    sig <- "-"
    }else{
      if(corr[["p.value"]]<0.001){
        sig <- "***"
        }else{
          if(corr[["p.value"]]<0.01){
            sig <- "**"
          }else{
              if(corr[["p.value"]]<0.05){
                sig<-"*"
              }else{
                  sig<-""}}}}
  
  dfcor2 <- rbind(dfcor2, data.frame(genus = i, cor = corr[["estimate"]][["rho"]], 
                 PD_Colon = "Combined", pvalue = corr[["p.value"]], sig = sig, comm = "Proportional microbial community"))
  }
#Combine abs and prop abundances in one
dfcor <- rbind(dfcor, dfcor2)
dfcor <- na.omit(dfcor); dfcor$comm <- as.ordered(factor(dfcor$comm, 
                              levels = c("Quantitative microbial community", "Proportional microbial community")))
# dfcor$PD_Colon <- as.ordered(factor(dfcor$PD_Colon, levels = c("P", "D")))
dfcor$genus <- as.ordered(factor(dfcor$genus, levels = order))
plot <- papertheme(ggplot(dfcor, aes(x = "1", y = genus, fill =cor))) + geom_tile() +
  scale_fill_gradient2(na.value = "grey50",low="blue", high="red", 
                       breaks = c(-0.8,-0.4,0,0.4,0.8), labels = c(-0.8,-0.4,0,0.4,0.8)) + 
  geom_text(aes(label = sig)) +
  theme(axis.text.y = element_text(face="italic")) + ylab("Genus") + xlab("") + 
  scale_x_discrete(breaks=c("P","D"), labels=c("Proximal colon", "Distal colon")) +
  labs(fill = expression(rho)) +
  facet_grid(cols = vars(comm)) +theme(strip.background = element_rect(fill = "#D6E0F4")) + 
  theme(legend.key.width= unit(2, 'cm'))
  # scale_colour_gradient(limits = c(257, 8888), 
  #                       breaks = c(257, 2000, 4000, 6000, 8000, 8888),
  #                       labels = c(257, 2000, 4000, 6000, 8000, 8888))
plot
```

```{r Correlation colons split AND combined, fig.height=15, fig.width=15}
order <- c("Veillonella", 	"Sutterella", 	"Ruminococcus", 	"Ruminococcaceae", 	"Roseburia", 	"Pseudomonas", 	"Prevotella", 	"Phascolarctobacterium", 	"Parabacteroides", 	"Mitsuokella", 	"Megasphaera", 	"Megamonas", 	"Lachnospiraceae", 	"Fusobacterium", 	"Fusicatenibacter", 	"Faecalibacterium", 	"Erysipelotrichaceae", 	"Enterobacteriaceae", 	"Eisenbergiella", 	"Dorea", 	"Dialister", 	"Desulfovibrio", 	"Coprococcus", 	"Clostridium_XlVa", 	"Clostridiales", 	"Cloacibacillus", 	"Blautia", 	"Bilophila", 	"Bifidobacterium", 	"Bacteroides", 	"Bacteria", 	"Anaerostipes", 	"Anaeroglobus", 	"Alistipes", 	"Akkermansia", 	"Acidaminococcus")
# Absolute
phylobj.QMP_stable <- subset_samples(phylobj.QMP, Ninetransit == 3)
Top15phylobj <- aggregate_taxa(phylobj.QMP_stable, level = "Genus")
# Top15phylobj <- aggregate_top_taxa(phylobj.QMP_stable, top = 20, "Genus")
Top15 <- psmelt(Top15phylobj); unique(Top15$OTU); unique(Top15$OTU) %in% Top15levels
x <- data.frame(x = unique(Top15$OTU) %in% Top15levels, genus = unique(Top15$OTU))  #check if labelled
Top15$OTU <- as.ordered(factor(Top15$OTU, levels = Top15levels))
# Add extra column days for graphs
Top <- Top15
Top$Transit <- as.ordered(factor(Top$Transit, levels = c("ST", "MT", "LT")))
Top$PD_Colon <- as.ordered(factor(Top$PD_Colon, levels = c("P", "D", "FS")))
Top$Donor <- as.ordered(factor(Top$Donor, levels = c("1", "2", "3", "4", "5", "6")))
Top$Genus <- as.ordered(factor(Top$Genus, levels = Top15levels))
# Add numbers
df.1 <- subset(Top, PD_Colon == "P" & Transit == "ST"); df.1$TT <- 8
df.2 <- subset(Top, PD_Colon == "P" & Transit == "MT"); df.2$TT <- 16
df.3 <- subset(Top, PD_Colon == "P" & Transit == "LT"); df.3$TT <- 24
df.4 <- subset(Top, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 13
df.5 <- subset(Top, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 26
df.6 <- subset(Top, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 39
Top <- rbind(df.1, df.2, df.3, df.4, df.5, df.6)
rm(df.1, df.2, df.3, df.4, df.5, df.6)
dfcor <- data.frame()
for(i in na.omit(unique(Top$Genus))){
  df1 <- subset(Top, Genus == i);  df1 <- subset(df1, Abundance > 0)
  for(x in c("P", "D")){
  df<- subset(df1, PD_Colon == x)
  t <- cor(df$Abundance, df$TT, method = "spearman")
  corr <- cor.test(df$Abundance, df$TT, method = 'spearman'); 
  
  if(is.na(corr[["p.value"]])){
    sig <- "-"
    }else{
      if(corr[["p.value"]]<0.0001){
        sig <- "****"
        }else{
          if(corr[["p.value"]]<0.001){
            sig <- "***"
            }else{
              if(corr[["p.value"]]<0.01){
                sig <- "**"
                }else{
                  if(corr[["p.value"]]<0.05){
                    sig<-"*"
                  }else{
                      sig<-""}}}}}
  
  dfcor <- rbind(dfcor, data.frame(genus = i, cor = corr[["estimate"]][["rho"]], 
                 PD_Colon = x, pvalue = corr[["p.value"]], sig = sig, comm = "Quantitative microbial community"))
  }}
# Proportional
phylobj.prop_2 = transform_sample_counts(phylobj.QMP, function(x) 100 * x/sum(x))
phylobj.prop_stable <- subset_samples(phylobj.prop_2, Ninetransit == 3)
Top15phylobj.prop <- aggregate_taxa(phylobj.prop_stable, level = "Genus")
Top15.prop <- psmelt(Top15phylobj.prop); unique(Top15.prop$OTU); unique(Top15.prop$OTU) %in% Top15levels
x <- data.frame(x = unique(Top15.prop$OTU) %in% Top15levels, genus = unique(Top15.prop$OTU))  #check if labelled
Top15.prop$OTU <- as.ordered(factor(Top15.prop$OTU, levels = Top15levels))
# Add extra column days for graphs
Top.prop <- Top15.prop
Top.prop$Transit <- as.ordered(factor(Top.prop$Transit, levels = c("ST", "MT", "LT")))
Top.prop$PD_Colon <- as.ordered(factor(Top.prop$PD_Colon, levels = c("P", "D", "FS")))
Top.prop$Donor <- as.ordered(factor(Top.prop$Donor, levels = c("1", "2", "3", "4", "5", "6")))
Top.prop$Genus <- as.ordered(factor(Top.prop$Genus, levels = Top15levels))
# Add numbers
df.1 <- subset(Top.prop, PD_Colon == "P" & Transit == "ST"); df.1$TT <- 8
df.2 <- subset(Top.prop, PD_Colon == "P" & Transit == "MT"); df.2$TT <- 16
df.3 <- subset(Top.prop, PD_Colon == "P" & Transit == "LT"); df.3$TT <- 24
df.4 <- subset(Top.prop, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 13
df.5 <- subset(Top.prop, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 26
df.6 <- subset(Top.prop, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 39
Top.prop <- rbind(df.1, df.2, df.3, df.4, df.5, df.6)
rm(df.1, df.2, df.3, df.4, df.5, df.6)
dfcor2 <- data.frame()
for(i in na.omit(unique(Top.prop$Genus))){
  df1 <- subset(Top.prop, Genus == i);  df1 <- subset(df1, Abundance > 0)
  for(x in c("P", "D")){
  df<- subset(df1, PD_Colon == x)
  t <- cor(df$Abundance, df$TT, method = "spearman")
  corr <- cor.test(df$Abundance, df$TT, method = 'spearman'); 
  
  if(is.na(corr[["p.value"]])){
    sig <- "-"
    }else{
      if(corr[["p.value"]]<0.0001){
        sig <- "****"
        }else{
          if(corr[["p.value"]]<0.001){
            sig <- "***"
            }else{
              if(corr[["p.value"]]<0.01){
                sig <- "**"
                }else{
                  if(corr[["p.value"]]<0.05){
                    sig<-"*"
                  }else{
                      sig<-""}}}}}
  
  dfcor2 <- rbind(dfcor2, data.frame(genus = i, cor = corr[["estimate"]][["rho"]], 
                 PD_Colon = x, pvalue = corr[["p.value"]], sig = sig, comm = "Proportional microbial community"))
  }}
#Combine abs and prop abundances in one
dfcor <- rbind(dfcor, dfcor2)
dfcor <- na.omit(dfcor); dfcor$comm <- as.ordered(factor(dfcor$comm, 
                              levels = c("Quantitative microbial community", "Proportional microbial community")))
dfcor$PD_Colon <- as.ordered(factor(dfcor$PD_Colon, levels = c("P", "D")))
dfcor$genus <- as.ordered(factor(dfcor$genus, levels = order))
plot <- papertheme(ggplot(dfcor, aes(x = PD_Colon, y = genus, fill =cor))) + geom_tile() +
  scale_fill_gradient2(na.value = "grey50",low="blue", high="red") + 
  geom_text(aes(label = sig)) +
  theme(axis.text.y = element_text(face="italic")) + ylab("Genus") + xlab("Colon vessel") + 
  scale_x_discrete(breaks=c("P","D"), labels=c("Proximal colon", "Distal colon")) +
  labs(fill = expression(rho)) +
  facet_grid(cols = vars(comm)) +theme(strip.background = element_rect(fill = "#D6E0F4"))
plot
dfcorall <- dfcor
dfcor <- data.frame()
for(i in na.omit(unique(Top$Genus))){
  df1 <- subset(Top, Genus == i);  df1 <- subset(df1, Abundance > 0)
  
  df<- df1
  t <- cor(df$Abundance, df$TT, method = "spearman")
  corr <- cor.test(df$Abundance, df$TT, method = 'spearman'); 
  
  if(is.na(corr[["p.value"]])){
    sig <- "-"
    }else{
      if(corr[["p.value"]]<0.0001){
        sig <- "****"
        }else{
          if(corr[["p.value"]]<0.001){
            sig <- "***"
            }else{
              if(corr[["p.value"]]<0.01){
                sig <- "**"
                }else{
                  if(corr[["p.value"]]<0.05){
                    sig<-"*"
                  }else{
                      sig<-""}}}}}
  
  dfcor <- rbind(dfcor, data.frame(genus = i, cor = corr[["estimate"]][["rho"]], 
                 PD_Colon = "Combined", pvalue = corr[["p.value"]], sig = sig, comm = "Quantitative microbial community"))
  }
dfcor2 <- data.frame()
for(i in na.omit(unique(Top.prop$Genus))){
  df1 <- subset(Top.prop, Genus == i);  df1 <- subset(df1, Abundance > 0)
  
  df<- df1
  t <- cor(df$Abundance, df$TT, method = "spearman")
  corr <- cor.test(df$Abundance, df$TT, method = 'spearman'); 
  
  if(is.na(corr[["p.value"]])){
    sig <- "-"
    }else{
      if(corr[["p.value"]]<0.0001){
        sig <- "****"
        }else{
          if(corr[["p.value"]]<0.001){
            sig <- "***"
            }else{
              if(corr[["p.value"]]<0.01){
                sig <- "**"
                }else{
                  if(corr[["p.value"]]<0.05){
                    sig<-"*"
                  }else{
                      sig<-""}}}}}
  
  dfcor2 <- rbind(dfcor2, data.frame(genus = i, cor = corr[["estimate"]][["rho"]], 
                 PD_Colon = "Combined", pvalue = corr[["p.value"]], sig = sig, comm = "Proportional microbial community"))
  }
#Combine abs and prop abundances in one
dfcor <- rbind(dfcor, dfcor2)
dfcor <- na.omit(dfcor); dfcor$comm <- as.ordered(factor(dfcor$comm, 
                              levels = c("Quantitative microbial community", "Proportional microbial community")))
# dfcor$PD_Colon <- as.ordered(factor(dfcor$PD_Colon, levels = c("P", "D")))
dfcor$genus <- as.ordered(factor(dfcor$genus, levels = order))
plot <- papertheme(ggplot(dfcor, aes(x = "1", y = genus, fill =cor))) + geom_tile() +
  scale_fill_gradient2(na.value = "grey50",low="blue", high="red", 
                       breaks = c(-0.8,-0.4,0,0.4,0.8), labels = c(-0.8,-0.4,0,0.4,0.8)) + 
  geom_text(aes(label = sig)) +
  theme(axis.text.y = element_text(face="italic")) + ylab("Genus") + xlab("") + 
  scale_x_discrete(breaks=c("P","D"), labels=c("Proximal colon", "Distal colon")) +
  labs(fill = expression(rho)) +
  facet_grid(cols = vars(comm)) +theme(strip.background = element_rect(fill = "#D6E0F4")) + 
  theme(legend.key.width= unit(2, 'cm'))
  # scale_colour_gradient(limits = c(257, 8888), 
  #                       breaks = c(257, 2000, 4000, 6000, 8000, 8888),
  #                       labels = c(257, 2000, 4000, 6000, 8000, 8888))
plot
dfcorall <- rbind(dfcorall, dfcor)
dfcorall$PD_Colon <- as.ordered(factor(dfcorall$PD_Colon, levels = c("P", "D", "Combined")))
dfcorall$genus <- as.ordered(factor(dfcorall$genus, levels = order))
plot <- papertheme(ggplot(dfcorall, aes(x = PD_Colon, y = genus, fill =cor))) + geom_tile() +
  # scale_fill_gradient2(na.value = "grey50",low="green", high="red") + 
  scale_fill_gradient2(na.value = "grey50",low="#5e135c", high="#125c13", breaks = c(-0.8,-0.4,0,0.4,0.8)) +
  geom_text(aes(label = sig, fontface = "bold")) +
  theme(axis.text.y = element_text(face="italic")) + ylab("Genus") + xlab("Colon vessel") + 
  scale_x_discrete(breaks=c("P","D", "Combined"), labels=c("Proximal colon", "Distal colon", "Combined")) +
  labs(fill = expression(rho)) +
  facet_grid(cols = vars(comm)) +theme(strip.background = element_rect(fill = "#D6E0F4")) + 
  theme(legend.key.width= unit(2, 'cm'))
plot
```






### Paperplot main genera correlations of Top most abundant genera
```{r Paperplot correlation colons split AND combined, fig.height=30, fig.width=30}
# order <- c("Veillonella", 	"Sutterella", 	"Ruminococcus", 	"Ruminococcaceae", 	"Roseburia", 	"Pseudomonas", 	"Prevotella", 	"Phascolarctobacterium", 	"Parabacteroides", 	"Mitsuokella", 	"Megasphaera", 	"Megamonas", 	"Lachnospiraceae", 	"Fusobacterium", 	"Fusicatenibacter", 	"Faecalibacterium", 	"Erysipelotrichaceae", 	"Enterobacteriaceae", 	"Eisenbergiella", 	"Dorea", 	"Dialister", 	"Desulfovibrio", 	"Coprococcus", 	"Clostridium_XlVa", 	"Clostridiales", 	"Cloacibacillus", 	"Blautia", 	"Bilophila", 	"Bifidobacterium", 	"Bacteroides", 	"Bacteria", 	"Anaerostipes", 	"Anaeroglobus", 	"Alistipes", 	"Akkermansia", 	"Acidaminococcus")
# df.OTU <-subset(df.combined, sum > 10)
df.OTU <-subset(df.combined, mean > 5); x_lab = 12
topnum <- 14
# Retrieve top most abundant genera
## Absolute
phylobj.QMP_stable <- subset_samples(phylobj.QMP, Ninetransit == 3)
Top15phylobj <- aggregate_taxa(phylobj.QMP_stable, level = "Genus")
Top15phylobj <- aggregate_top_taxa(phylobj.QMP_stable, top = topnum, "Genus")
## Proportional
phylobj.prop_2 = transform_sample_counts(phylobj.QMP, function(x) 100 * x/sum(x))
phylobj.prop_stable <- subset_samples(phylobj.prop_2, Ninetransit == 3)
phylobj.prop_stable <- aggregate_top_taxa(phylobj.prop_stable, top = topnum, "Genus")
## Take consensus top genera
Top.genera <- rbind(subset(as.data.frame(phylobj.prop_stable@tax_table), Genus != "Other"), 
                    subset(as.data.frame(Top15phylobj@tax_table), Genus != "Other"))
## Remove unsignificant genera
df <- data.frame()
for(i in significantlist){
  df2 <- subset(Top.genera, Genus == i)
  df <- rbind(df, df2)
}
Top.genera <- unique(df$Genus)
# Remove those that are not in the significant changes list (significantlist)
Top.genera <- sort(Top.genera, decreasing = T)


# GENUS LEVEL
# Absolute
phylobj.QMP_stable <- subset_samples(phylobj.QMP, Ninetransit == 3)
Top15phylobj <- aggregate_taxa(phylobj.QMP_stable, level = "Genus")
Top15 <- psmelt(Top15phylobj); unique(Top15$OTU); unique(Top15$OTU) %in% Top15levels
## Subset according to top most abundant genera
df.test <- data.frame();
for(i in Top.genera){
  df.test2 <- subset(Top15, Genus == i)
  df.test <- rbind(df.test, df.test2)
}
Top15 <- df.test
x <- data.frame(x = unique(Top15$OTU) %in% Top15levels, genus = unique(Top15$OTU))  #check if labelled
Top15$OTU <- as.ordered(factor(Top15$OTU, levels = Top15levels))
## Add extra column days for graphs
Top <- Top15
Top$Transit <- as.ordered(factor(Top$Transit, levels = c("ST", "MT", "LT")))
Top$PD_Colon <- as.ordered(factor(Top$PD_Colon, levels = c("P", "D", "FS")))
Top$Donor <- as.ordered(factor(Top$Donor, levels = c("1", "2", "3", "4", "5", "6")))
Top$Genus <- as.ordered(factor(Top$Genus, levels = Top15levels))
## Add numbers
df.1 <- subset(Top, PD_Colon == "P" & Transit == "ST"); df.1$TT <- 8; df.1$TT1 <- 8
df.2 <- subset(Top, PD_Colon == "P" & Transit == "MT"); df.2$TT <- 16; df.2$TT1 <- 16
df.3 <- subset(Top, PD_Colon == "P" & Transit == "LT"); df.3$TT <- 24; df.3$TT1 <- 24
df.4 <- subset(Top, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 13; df.4$TT1 <- 8
df.5 <- subset(Top, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 26; df.5$TT1 <- 16
df.6 <- subset(Top, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 39; df.6$TT1 <- 24
# df.4 <- subset(Top, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 8
# df.5 <- subset(Top, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 16
# df.6 <- subset(Top, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 24

Top <- rbind(df.1, df.2, df.3, df.4, df.5, df.6)
rm(df.1, df.2, df.3, df.4, df.5, df.6)
## Perform test
dfcor <- data.frame(); df.otutest <- data.frame(); QMPlist <- list(); y =1
for(i in na.omit(unique(Top$Genus))){
  df1 <- subset(Top, Genus == i);  df1 <- subset(df1, Abundance > 0)
  for(x in c("P", "D")){
  df<- subset(df1, PD_Colon == x)
  t <- cor(df$Abundance, df$TT, method = "spearman")
  corr <- cor.test(df$Abundance, df$TT, method = 'spearman'); 
  
  if(is.na(corr[["p.value"]])){
    sig <- "-"
    }else{
      if(corr[["p.value"]]<0.0001){
        sig <- "****"
        }else{
          if(corr[["p.value"]]<0.001){
            sig <- "***"
            }else{
              if(corr[["p.value"]]<0.01){
                sig <- "**"
                }else{
                  if(corr[["p.value"]]<0.05){
                    sig<-"*"
                  }else{
                      sig<-""}}}}}
  
  dfcor <- rbind(dfcor, data.frame(genus = i, cor = corr[["estimate"]][["rho"]], 
             PD_Colon = x, pvalue = corr[["p.value"]], sig = sig, comm = "Quantitative microbial community"))
  
  if(x == "P"){corP <- paste0("Ï = ",round(t, digits = 2)," ", sig, sep = ""); corrP <- corr[["p.value"]]
  }else{corD <- paste0("Ï = ",round(t, digits = 2)," ", sig, sep = ""); corrD <- corr[["p.value"]]}
  }
  
  # dat_text <- data.frame(label = c(corP, corD),
  # PD_Colon   = c("P","D"),
  # value   = c(max(df1$Abundance), max(df1$Abundance)), 
  # TT = c(8,8))

  
plot <- papertheme(ggplot(df1, aes(y = Abundance, x = TT1))) + geom_boxplot(aes(fill = Transit)) + geom_jitter(aes(color = Donor), size = 4) + 
    facet_grid(rows = vars(PD_Colon), labeller = labeller(PD_Colon = PD_Colon.labs), scales = "free_x") 

  if (corrP > 0.05 & corrD > 0.05) {
    plot <- plot
    dat_text <- data.frame(label = c("", ""), PD_Colon = c("P",
        "D"), value = c(max(df1$Abundance), max(df1$Abundance)),
        TT1 = c(x_lab, x_lab))
} else {
    if (corrP > 0.05 & corrD <= 0.05 ) {
        plot <- plot + geom_smooth(data = subset(df1, PD_Colon =="D"), aes(y = Abundance, x = TT1), method = lm,
                                   color = "black")
        dat_text <- data.frame(label = c("", corD), PD_Colon = c("P",
            "D"), value = c(max(df1$Abundance), max(df1$Abundance)),
            TT1 = c(x_lab, x_lab))
    } else {
        if (corrP <= 0.05 & corrD > 0.05) {
            plot <- plot + geom_smooth(data = subset(df1,  PD_Colon == "P"), aes(y = Abundance, x = TT1),
                method = lm, color = "black")
            dat_text <- data.frame(label = c(corP, ""),
                PD_Colon = c("P", "D"), value = c(max(df1$Abundance),
                  max(df1$Abundance)), TT1 = c(x_lab, x_lab))
        } else {
            plot <- plot + geom_smooth(aes(y = Abundance, x = TT1), method = lm, color = "black")
            dat_text <- data.frame(label = c(corP, corD),
                PD_Colon = c("P", "D"), value = c(max(df1$Abundance),
                  max(df1$Abundance)), TT1 = c(x_lab, x_lab))
        }
    }
}
  dat_text$PD_Colon <- as.ordered(factor(dat_text$PD_Colon, levels = c("P", "D")))
  
 plot <- plot + scale_color_manual(values = sixdonors) + scale_fill_manual("SHIME transit", labels = c("Short", "Medium", "Long"), values = transitcolors) +
  scale_x_discrete(labels = c("Short", "Medium", "Long"))+
  ylab(expression(paste("Absolute abundance (cells ",g^{"-1"},")")))+ xlab("Transit time")  
   
   if(i == "Akkermansia"){plot <- plot + theme(strip.background =element_rect(fill="#D6E0F4"),
          legend.position = c(0.5, 0.8),
          legend.direction = "vertical",
          legend.box = "horizontal",
          strip.text.x = element_text(size = 20),
          strip.text.y = element_blank(),
          legend.title = element_text(size=20), 
          legend.spacing.x = unit(0.01, 'cm'),
          legend.background = element_blank()) + 
     guides(shape = guide_legend(ncol = 2, byrow = TRUE))}else{
    plot <- plot + theme(strip.background =element_rect(fill="#D6E0F4"),
          legend.position = "none",
          strip.text.x = element_text(size = 20),
          strip.text.y = element_blank(),
          legend.title = element_text(size=20), legend.background = element_blank())} 
  
if(i == "X"){plot <- plot + geom_text(correlations)}else{
  plot <- plot + geom_text(data= dat_text,  mapping = aes(x = as.numeric(TT1), y = as.numeric(value), label = label, family = 'sans'), size = 7)
}
QMPlist[[i]] <- plot; y = y+1
  }




# Proportional
phylobj.prop_2 = transform_sample_counts(phylobj.QMP, function(x) 100 * x/sum(x))
phylobj.prop_stable <- subset_samples(phylobj.prop_2, Ninetransit == 3)
Top15phylobj.prop <- aggregate_taxa(phylobj.prop_stable, level = "Genus")
Top15.prop <- psmelt(Top15phylobj.prop); unique(Top15.prop$OTU); unique(Top15.prop$OTU) %in% Top15levels
## Subset according to top most abundant genera
df.test <- data.frame();
for(i in Top.genera){
  df.test2 <- subset(Top15.prop, Genus == i)
  df.test <- rbind(df.test, df.test2)
}
Top15.prop <- df.test
x <- data.frame(x = unique(Top15$OTU) %in% Top15levels, genus = unique(Top15$OTU))  #check if labelled
Top15$OTU <- as.ordered(factor(Top15$OTU, levels = Top15levels))
x <- data.frame(x = unique(Top15.prop$OTU) %in% Top15levels, genus = unique(Top15.prop$OTU))  #check if labelled
Top15.prop$OTU <- as.ordered(factor(Top15.prop$OTU, levels = Top15levels))
## Add extra column days for graphs
Top.prop <- Top15.prop
Top.prop$Transit <- as.ordered(factor(Top.prop$Transit, levels = c("ST", "MT", "LT")))
Top.prop$PD_Colon <- as.ordered(factor(Top.prop$PD_Colon, levels = c("P", "D", "FS")))
Top.prop$Donor <- as.ordered(factor(Top.prop$Donor, levels = c("1", "2", "3", "4", "5", "6")))
Top.prop$Genus <- as.ordered(factor(Top.prop$Genus, levels = Top15levels))
## Add numbers
df.1 <- subset(Top.prop, PD_Colon == "P" & Transit == "ST"); df.1$TT <- 8; df.1$TT1 <- 8
df.2 <- subset(Top.prop, PD_Colon == "P" & Transit == "MT"); df.2$TT <- 16; df.2$TT1 <- 16
df.3 <- subset(Top.prop, PD_Colon == "P" & Transit == "LT"); df.3$TT <- 24; df.3$TT1 <- 24
df.4 <- subset(Top.prop, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 13; df.4$TT1 <- 8
df.5 <- subset(Top.prop, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 26; df.5$TT1 <- 16
df.6 <- subset(Top.prop, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 39; df.6$TT1 <- 24
# df.4 <- subset(Top.prop, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 8
# df.5 <- subset(Top.prop, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 16
# df.6 <- subset(Top.prop, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 24
Top.prop <- rbind(df.1, df.2, df.3, df.4, df.5, df.6)
rm(df.1, df.2, df.3, df.4, df.5, df.6)
dfcor2 <- data.frame(); PMPlist <- list(); y = 1

for(i in na.omit(unique(Top.prop$Genus))){
  df1 <- subset(Top.prop, Genus == i);  df1 <- subset(df1, Abundance > 0)
  for(x in c("P", "D")){
  df<- subset(df1, PD_Colon == x)
  t <- cor(df$Abundance, df$TT, method = "spearman")
  corr <- cor.test(df$Abundance, df$TT, method = 'spearman'); 
  
  if(is.na(corr[["p.value"]])){
    sig <- "-"
    }else{
      if(corr[["p.value"]]<0.0001){
        sig <- "****"
        }else{
          if(corr[["p.value"]]<0.001){
            sig <- "***"
            }else{
              if(corr[["p.value"]]<0.01){
                sig <- "**"
                }else{
                  if(corr[["p.value"]]<0.05){
                    sig<-"*"
                  }else{
                      sig<-""}}}}}
  
  dfcor2 <- rbind(dfcor2, data.frame(genus = i, cor = corr[["estimate"]][["rho"]], 
                 PD_Colon = x, pvalue = corr[["p.value"]], sig = sig, comm = "Proportional microbial community"))
  
  if(x == "P"){corP <- paste0("Ï = ",round(t, digits = 2)," ", sig, sep = ""); corrP <- corr[["p.value"]]}
  if(x == "D"){corD <- paste0("Ï = ",round(t, digits = 2)," ", sig, sep = ""); corrD <- corr[["p.value"]]}
  }
  
  #   dat_text <- data.frame(label = c(corP, corD),
  # PD_Colon   = c("P","D"),
  # value   = c(max(df1$Abundance), max(df1$Abundance)), 
  # TT = c(8,8))
  # dat_text$PD_Colon <- as.ordered(factor(dat_text$PD_Colon, levels = c("P", "D")))
  
plot <- papertheme(ggplot(df1, aes(y = Abundance, x = TT1))) + geom_boxplot(aes(fill = Transit)) + geom_jitter(aes(color = Donor), size =4) + 
      facet_grid(rows = vars(PD_Colon), labeller = labeller(PD_Colon = PD_Colon.labs), scales = "free_x") 

  if (corrP > 0.05 & corrD > 0.05) {
    plot <- plot
    dat_text <- data.frame(label = c("", ""), PD_Colon = c("P",
        "D"), value = c(max(df1$Abundance), max(df1$Abundance)),
        TT1 = c(x_lab, x_lab))
} else {
    if (corrP > 0.05 & corrD <= 0.05) {
        plot <- plot + geom_smooth(data = subset(df1, PD_Colon =="D"), 
                                   aes(y = Abundance, x = TT1), method = lm,
                                   color = "black")
        dat_text <- data.frame(label = c("", corD), PD_Colon = c("P",
            "D"), value = c(max(df1$Abundance), max(df1$Abundance)),
            TT1 = c(x_lab, x_lab))
    } else {
        if (corrP <= 0.05 & corrD > 0.05) {
            plot <- plot + geom_smooth(data = subset(df1,
                PD_Colon == "P"), aes(y = Abundance, x = TT1),
                method = lm, color = "black")
            dat_text <- data.frame(label = c(corP, ""),
                PD_Colon = c("P", "D"), value = c(max(df1$Abundance),
                  max(df1$Abundance)), TT1 = c(x_lab, x_lab))
        } else {
            plot <- plot + geom_smooth(aes(y = Abundance,
                x = TT1), method = lm, color = "black")
            dat_text <- data.frame(label = c(corP, corD),
                PD_Colon = c("P", "D"), value = c(max(df1$Abundance),
                  max(df1$Abundance)), TT1 = c(x_lab, x_lab))
        }
    }
}
  dat_text$PD_Colon <- as.ordered(factor(dat_text$PD_Colon, levels = c("P", "D")))

plot <- plot + scale_color_manual(values = sixdonors) + scale_fill_manual(values = transitcolors) +
  ylab("Proportional abundance (%)")+ xlab("Transit time") + 
  scale_x_discrete(labels = c("Short", "Medium", "Long"))+
    theme(strip.background =element_rect(fill="#D6E0F4"),
          legend.position = "none",
          legend.box = "horizontal",
          strip.text.x = element_text(size = 20),
          strip.text.y = element_text(size = 20, angle = 270),
          legend.title = element_text(size=20), legend.background = element_blank()) 
  
if(i == "X"){plot <- plot + geom_text(correlations)}else{
  plot <- plot + geom_text(data= dat_text,  mapping = aes(x = as.numeric(TT1), y = as.numeric(value), label = label, family = 'sans'), size = 7)
}
PMPlist[[i]] <- plot; y = y+1
    }

#Combine abs and prop abundances in one
dfcor <- rbind(dfcor, dfcor2)
dfcor <- na.omit(dfcor); dfcor$comm <- as.ordered(factor(dfcor$comm, 
                              levels = c("Quantitative microbial community", "Proportional microbial community")))
dfcor$PD_Colon <- as.ordered(factor(dfcor$PD_Colon, levels = c("P", "D")))
dfcor$genus <- as.ordered(factor(dfcor$genus, levels = Top.genera))
plot <- papertheme(ggplot(dfcor, aes(x = PD_Colon, y = genus, fill =cor))) + geom_tile() +
  scale_fill_gradient2(na.value = "grey50",low="blue", high="red") + 
  geom_text(aes(label = sig)) +
  theme(axis.text.y = element_text(face="italic")) + ylab("Genus") + xlab("Colon vessel") + 
  scale_x_discrete(breaks=c("P","D"), labels=c("Proximal colon", "Distal colon")) +
  labs(fill = expression(rho)) +
  facet_grid(cols = vars(comm)) +theme(strip.background = element_rect(fill = "#D6E0F4"))
plot
dfcorall <- dfcor




# Combine plots
plots <- list(); x = 1
genera <- na.omit(unique(Top.prop$Genus))
genera <- sort(genera, decreasing = F)
for(i in genera){ 
  plot1 <- QMPlist[[i]]; plot2 <- PMPlist[[i]]
  plot <- ggarrange(plot1, plot2)
  plot <- annotate_figure(plot, text_grob(i, color = "black", face = "italic", size = 25))
  # print(plot)
  plots[[x]] <- plot
  x = x +1
}

plots1 <- plots[1:8]; plots2 <- plots[9:16]
# ggarrange(plotlist = plots1, nrow = 2, ncol = 4)
# ggarrange(plotlist = plots2, nrow = 2, ncol = 4)
ggarrange(plotlist = plots, nrow = 5, ncol = 3)
```



# Paperplot main genera correlations and OTU of Top most abundant genera

## Top30 generas without OTUs
```{r Paperplot correlation colons split AND combined, fig.height=42, fig.width=40}
PD_Colon.labs <- c("Faecal sample","Prox colon", "Dist colon")
names(PD_Colon.labs) <- c("FS","P", "D")
nY = 0.75
if(BW == TRUE){fillcolor <- "lightgrey"
  labelsx <- c("S", "M", "L")
  
}else{fillcolor <- "#D6E0F4"
  labelsx <- c("ST", "MT", "LT")
  }


# fig.height = 6.5 per row and width = 8.75 per col
corenumber <- 10; scientificdigits <- 2
df.pwc <- data.frame(); gglist_stats <- list(); statscount <- 1
df.pwc.otu <- data.frame(); gglist_stats.otu <- list(); statscount.otu <- 1
df.res.kruskal.all <- data.frame()
registerDoParallel(corenumber)  

# df.OTU <-subset(df.combined, sum > 10)
df.OTU <-subset(df.combined, mean > 5); x_lab = 16; 
hjustv = 1.1; vjustv = 1.5; xInf <- Inf; yInf <- Inf # Top Left (h0,v1, -inf, +inf)  Top Right (h1,v1, +inf, +inf)
topnum <- 30; abovezero = TRUE; smoothline = F

# Retrieve top most abundant genera
## Absolute
phylobj.QMP_stable <- subset_samples(phylobj.QMP, Ninetransit == 3)
Top15phylobj <- aggregate_taxa(phylobj.QMP_stable, level = "Genus")
Top15phylobj <- aggregate_top_taxa(phylobj.QMP_stable, top = topnum, "Genus")
extraphylo <- subset_taxa(phylobj.QMP_stable, Genus == "Clostridiales"|Genus == "Megasphaera")
extraphylo <- subset_taxa(phylobj.QMP_stable, Genus == "Clostridiales")
Top15phylobj <- merge_phyloseq(Top15phylobj, extraphylo)

## Proportional
phylobj.prop_2 = transform_sample_counts(phylobj.QMP, function(x) 100 * x/sum(x))
phylobj.prop_stable <- subset_samples(phylobj.prop_2, Ninetransit == 3)
extraphylo <- subset_taxa(phylobj.prop_stable, Genus == "Clostridiales"|Genus == "Megasphaera")
extraphylo <- subset_taxa(phylobj.prop_stable, Genus == "Clostridiales")
phylobj.prop_stable <- aggregate_top_taxa(phylobj.prop_stable, top = topnum, "Genus")
phylobj.prop_stable <- merge_phyloseq(phylobj.prop_stable, extraphylo)

## Take consensus top genera
Top.genera <- rbind(subset(as.data.frame(phylobj.prop_stable@tax_table), Genus != "Other"), 
                    subset(as.data.frame(Top15phylobj@tax_table), Genus != "Other"))
## Remove insignificant genera
df <- data.frame()
for(i in significantlist){
  df2 <- subset(Top.genera, Genus == i)
  df <- rbind(df, df2)
}
Top.genera <- unique(df$Genus)

# Remove those that are not in the significant changes list (significantlist)
Top.genera <- sort(Top.genera, decreasing = T)
# Top.genera <- c("Veillonella","Ruminococcus",  "Mitsuokella",	"Enterobacteriaceae", "Bifidobacterium")

# GENUS LEVEL
# Absolute
phylobj.QMP_stable <- subset_samples(phylobj.QMP, Ninetransit == 3)
Top15phylobj <- aggregate_taxa(phylobj.QMP_stable, level = "Genus")
Top15 <- psmelt(Top15phylobj); unique(Top15$OTU); unique(Top15$OTU) %in% Top15levels

## Subset according to top most abundant genera
df.test <- data.frame();
for(i in Top.genera){
  df.test2 <- subset(Top15, Genus == i)
  df.test <- rbind(df.test, df.test2)
}
Top15 <- df.test
x <- data.frame(x = unique(Top15$OTU) %in% Top15levels, genus = unique(Top15$OTU))  #check if labelled
Top15$OTU <- as.ordered(factor(Top15$OTU, levels = Top15levels))

## Add extra column days for graphs
Top <- Top15
Top$Transit <- as.ordered(factor(Top$Transit, levels = c("ST", "MT", "LT")))
Top$PD_Colon <- as.ordered(factor(Top$PD_Colon, levels = c("P", "D", "FS")))
Top$Donor <- as.ordered(factor(Top$Donor, levels = c("1", "2", "3", "4", "5", "6")))
Top$Genus <- as.ordered(factor(Top$Genus, levels = Top15levels))

## Add numbers
df.1 <- subset(Top, PD_Colon == "P" & Transit == "ST"); df.1$TT <- 8; df.1$TT1 <- 8
df.2 <- subset(Top, PD_Colon == "P" & Transit == "MT"); df.2$TT <- 16; df.2$TT1 <- 16
df.3 <- subset(Top, PD_Colon == "P" & Transit == "LT"); df.3$TT <- 24; df.3$TT1 <- 24
df.4 <- subset(Top, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 13; df.4$TT1 <- 8
df.5 <- subset(Top, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 26; df.5$TT1 <- 16
df.6 <- subset(Top, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 39; df.6$TT1 <- 24


Top <- rbind(df.1, df.2, df.3, df.4, df.5, df.6)
rm(df.1, df.2, df.3, df.4, df.5, df.6, extraphylo)

## Perform test
dfcor <- data.frame(); df.otutest <- data.frame(); QMPlist <- list(); y =1
# foreach (i = na.omit(unique(Top$Genus))) %dopar% {
for(i in na.omit(unique(Top$Genus))){
  df1 <- subset(Top, Genus == i);  
  if(abovezero == T){
    df1.1 <- subset(df1, Donor == 1); df1.1 <- subset(df1.1, Abundance > 0)
    if (dim(df1.1)[1] == 0) {df1 <- subset(df1, Donor !=1)}
    df1.2 <- subset(df1, Donor == 2); df1.2 <- subset(df1.2, Abundance > 0) 
    if (dim(df1.2)[1] == 0) {df1 <- subset(df1, Donor !=2)}
    df1.3 <- subset(df1, Donor == 3); df1.3 <- subset(df1.3, Abundance > 0)
    if (dim(df1.3)[1] == 0) {df1 <- subset(df1, Donor !=3)}
    df1.4 <- subset(df1, Donor == 4); df1.4 <- subset(df1.4, Abundance > 0) 
    if (dim(df1.4)[1] == 0) {df1 <- subset(df1, Donor !=4)}
    df1.5 <- subset(df1, Donor == 5); df1.5 <- subset(df1.5, Abundance > 0) 
    if (dim(df1.5)[1] == 0) {df1 <- subset(df1, Donor !=5)}
    df1.6 <- subset(df1, Donor == 6); df1.6 <- subset(df1.6, Abundance > 0) 
    if (dim(df1.6)[1] == 0) {df1 <- subset(df1, Donor !=6)}
    }
  for(x in c("P", "D")){
  df<- subset(df1, PD_Colon == x)
  t <- cor(df$Abundance, df$TT, method = "spearman")
  corr <- cor.test(df$Abundance, df$TT, method = 'spearman'); 
  if(is.na(corr[["p.value"]])){
    sig <- "-"
    }else{
      if(corr[["p.value"]]<0.0001){
        sig <- "****"
        }else{
          if(corr[["p.value"]]<0.001){
            sig <- "***"
            }else{
              if(corr[["p.value"]]<0.01){
                sig <- "**"
                }else{
                  if(corr[["p.value"]]<0.05){
                    sig<-"*"
                  }else{
                      sig<-""}}}}}
  
  dfcor <- rbind(dfcor, data.frame(genus = i, cor = corr[["estimate"]][["rho"]], 
             PD_Colon = x, pvalue = corr[["p.value"]], sig = sig, comm = "Quantitative microbial community"))
  
  if(x == "P"){corP <- paste0("Ï = ",round(t, digits = 2)," ", sig, sep = ""); corrP <- corr[["p.value"]]
  }else{corD <- paste0("Ï = ",round(t, digits = 2)," ", sig, sep = ""); corrD <- corr[["p.value"]]}
  }
  
plot <- papertheme(ggplot(df1, aes(y = Abundance, x = TT1))) + geom_boxplot(aes(fill = Transit)) + 
  geom_jitter(aes(color = Donor, shape = Donor), size = 4, height = 0) + 
  facet_grid(rows = vars(PD_Colon), labeller = labeller(PD_Colon = PD_Colon.labs), scales = "free_x") 

  if(is.na(corrP)){corrP <- 1} 
  if(is.na(corrD)){corrD <- 1} 
  if (corrP > 0.05 & corrD > 0.05) {
    plot <- plot
    dat_text <- data.frame(label = c("", ""), 
                           PD_Colon = c("P", "D"), 
                           value = c(max(df1$Abundance), max(df1$Abundance)),
                           xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                           hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
                           TT1 = c(x_lab, x_lab))
} else {
    if (corrP > 0.05 & corrD <= 0.05 ) {
        if(smoothline ==T){plot <- plot + geom_smooth(data = subset(df1, PD_Colon =="D"), aes(y = Abundance, x = TT1), 
                                                      method = lm, color = "black")}
        dat_text <- data.frame(label = c("", corD), PD_Colon = c("P","D"), 
                               value = c(max(df1$Abundance), max(df1$Abundance)),
                               xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                               hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
                               TT1 = c(x_lab, x_lab))
    } else {
        if (corrP <= 0.05 & corrD > 0.05) {
            if(smoothline ==T){plot <- plot + geom_smooth(data = subset(df1,  PD_Colon == "P"), aes(y = Abundance, x = TT1),
                method = lm, color = "black")}
            dat_text <- data.frame(label = c(corP, ""),
                                   PD_Colon = c("P", "D"), 
                                   value = c(max(df1$Abundance),max(df1$Abundance)),
                                   xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                                   hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
                                   TT1 = c(x_lab, x_lab))
        } else {
            if(smoothline ==T){plot <- plot + geom_smooth(aes(y = Abundance, x = TT1), method = lm, color = "black")}
            dat_text <- data.frame(label = c(corP, corD),
                                   PD_Colon = c("P", "D"), 
                                   value = c(max(df1$Abundance),max(df1$Abundance)),
                                   xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                                   hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
                                   TT1 = c(x_lab, x_lab))
        }
    }
}
  dat_text$PD_Colon <- as.ordered(factor(dat_text$PD_Colon, levels = c("P", "D")))
  
 plot <- plot + scale_color_manual(values = sixdonors) + 
   scale_fill_manual("SHIME transit", values = transitcolors, labels= c("Short", "Medium", "Long")) +
   scale_x_continuous(breaks=c(8,16,24), labels=labelsx) +
   # scale_x_discrete(labels = c("Short", "Medium", "Long"))+
   ylab(expression(paste("Abs abundance (cells ",mL^{"-1"},")")))+ xlab("Transit time") + 
   scale_y_continuous(labels = label_scientific(digits = scientificdigits)) 
 
 # if(i == "Acidaminococcus"){plot <- plot + 
 #   guides(fill=guide_legend(ncol=1, title.position = "top", hjust = 0.5),
 #          color=guide_legend(nrow=3, title.position = "top", hjust = 0.5)) +
 #   theme(legend.title.align = 0.5, legend.title = element_text(size=20), 
 #         legend.text = element_text(size=20), legend.position = c(0.5,0.35), 
 #         legend.box = "horizontal")
 #  }else{
    plot <- plot + theme(legend.position = "none")
    # }
 
         plot <- plot + theme(strip.background =element_rect(fill=fillcolor),
         strip.text.x = element_text(size = 20),
         strip.text.y = element_blank(),
         legend.title = element_text(size=20), legend.background = element_blank()) 
  
 
 
         
    ### Statistics figures
    vessels = "P"
    test <- subset(df1, PD_Colon == vessels)
    res.kruskal25 <- subset(df1, PD_Colon == vessels) %>% rstatix::kruskal_test(Abundance ~ Transit)
     res.kruskal22 <- res.kruskal25; res.kruskal22$vessel <- vessels; res.kruskal22$Genus <- i; 
     res.kruskal22$Abundance <- "ABS"
     df.res.kruskal.all <- rbind(df.res.kruskal.all, res.kruskal22)
     pwc <- subset(df1, PD_Colon == vessels) %>%  
       wilcox_test(Abundance ~ Transit, 
                   p.adjust.method = "holm", paired = F)
    pwc <- subset(pwc, p.adj.signif != "ns")
    pwc$Genus <- i; pwc$Vessel <- vessels; pwc$abundance <- "ABS"
    pwc$maxST <- max(subset(test, Transit =="ST")$Abundance) 
    pwc$maxMT <- max(subset(test, Transit =="MT")$Abundance)
    pwc$maxLT <- max(subset(test, Transit =="LT")$Abundance)
    df.pwc <- rbind(df.pwc, pwc)
    # print(pwc) # print pairwise comparisons
    
    #### Visualization: box plots with p-values
    pwc <- pwc %>% add_xy_position(x = "Transit")
    plot64 <- ggboxplot(subset(df1, PD_Colon == vessels), x = "Transit", y = "Abundance") +
    stat_pvalue_manual(pwc, hide.ns = TRUE, label = "p") +
    labs(title = paste0("Genus ",i,", Vessel ",vessels),
    subtitle = get_test_label(res.kruskal25, detailed = TRUE),
    caption = get_pwc_label(pwc))
    
    
    vessels = "D"
    test <- subset(df1, PD_Colon == vessels)
    res.kruskal25 <- subset(df1, PD_Colon == vessels) %>% rstatix::kruskal_test(Abundance ~ Transit)
        res.kruskal22 <- res.kruskal25; res.kruskal22$vessel <- vessels; res.kruskal22$Genus <- i; 
        res.kruskal22$Abundance <- "ABS"
     df.res.kruskal.all <- rbind(df.res.kruskal.all, res.kruskal22)
    pwc <- subset(df1, PD_Colon == vessels) %>%  wilcox_test(Abundance ~ Transit, p.adjust.method = "holm", paired = F)
 
    pwc <- subset(pwc, p.adj.signif != "ns")
    pwc$Genus <- i; pwc$Vessel <- vessels; pwc$abundance <- "ABS"
    pwc$maxST <- max(subset(test, Transit =="ST")$Abundance)
    pwc$maxMT <- max(subset(test, Transit =="MT")$Abundance)
    pwc$maxLT <- max(subset(test, Transit =="LT")$Abundance)
    df.pwc <- rbind(df.pwc, pwc)
    # print(pwc) # print pairwise comparisons
    
    #### Visualization: box plots with p-values
    pwc <- pwc %>% add_xy_position(x = "Transit")
    plot65 <- ggboxplot(subset(df1, PD_Colon == vessels), x = "Transit", y = "Abundance") +
    stat_pvalue_manual(pwc, hide.ns = TRUE, label = "p") +
    labs(title = paste0("Genus ",i,", Vessel ",vessels),
    subtitle = get_test_label(res.kruskal25, detailed = TRUE),
    caption = get_pwc_label(pwc)); 
    
    gglist_stats[[statscount]] <- ggarrange(plot64, plot65, nrow=2)  
    statscount <- statscount +1
rm(plot64, plot65, vessels)
y = y+1        
         
    ## plot statistics
    # stats_dataframe is own script!
   correlations <- stats_dataframe(df1,i, profile = "quant") 
   

        print_rho = F
if(print_rho == F){plot <- plot + geom_text(data= correlations,  mapping = aes(x = TT1, y = Abundance, label = label, 
                                        family = 'sans'), size = 7)
}else{
  plot <- plot + geom_text(data= correlations,  mapping = aes(x = TT1, y = Abundance, label = label, 
                                        family = 'sans'), size = 7)
  # plot <- plot + geom_text(data= dat_text,  mapping = aes(x = as.numeric(TT1), y = as.numeric(value), label = label, family = 'sans'), size = 7)
  plot <- plot + geom_text(data= dat_text,  mapping = aes(x = xpos, y = ypos, label = label, 
                                        hjust=hjustvar,vjust=vjustvar, family = 'sans'), size = 7)
}
        
   if(BW == TRUE){plot <- plot + scale_fill_grey(name = "SHIME transit", start = 0.7, end = 1, labels= c("Short", "Medium", "Long")) + scale_color_grey(start = 0, end = 0.3) +scale_shape_manual(name = "Donor", values=c("1" = 15, "2" = 16, "3" = 17 , "4" = 18, "5" = 25, "6" = 9)) + guides(color = "none") + xlab("Transit")
   }else{plot <- plot + scale_shape_manual(name = "Donor", values=c("1" = 16, "2" = 16, "3" = 16 , "4" = 16, "5" = 16, "6" = 16)) + guides(shape = "none")}     
        
        
        
QMPlist[[i]] <- plot; 

  }



# Proportional
phylobj.prop_2 = transform_sample_counts(phylobj.QMP, function(x) 100 * x/sum(x))
phylobj.prop_stable <- subset_samples(phylobj.prop_2, Ninetransit == 3)
Top15phylobj.prop <- aggregate_taxa(phylobj.prop_stable, level = "Genus")
Top15.prop <- psmelt(Top15phylobj.prop); unique(Top15.prop$OTU); unique(Top15.prop$OTU) %in% Top15levels
## Subset according to top most abundant genera
df.test <- data.frame();
for(i in Top.genera){
  df.test2 <- subset(Top15.prop, Genus == i)
  df.test <- rbind(df.test, df.test2)
}
Top15.prop <- df.test
x <- data.frame(x = unique(Top15$OTU) %in% Top15levels, genus = unique(Top15$OTU))  #check if labelled
Top15$OTU <- as.ordered(factor(Top15$OTU, levels = Top15levels))
x <- data.frame(x = unique(Top15.prop$OTU) %in% Top15levels, genus = unique(Top15.prop$OTU))  #check if labelled
Top15.prop$OTU <- as.ordered(factor(Top15.prop$OTU, levels = Top15levels))
## Add extra column days for graphs
Top.prop <- Top15.prop
Top.prop$Transit <- as.ordered(factor(Top.prop$Transit, levels = c("ST", "MT", "LT")))
Top.prop$PD_Colon <- as.ordered(factor(Top.prop$PD_Colon, levels = c("P", "D", "FS")))
Top.prop$Donor <- as.ordered(factor(Top.prop$Donor, levels = c("1", "2", "3", "4", "5", "6")))
Top.prop$Genus <- as.ordered(factor(Top.prop$Genus, levels = Top15levels))
## Add numbers
df.1 <- subset(Top.prop, PD_Colon == "P" & Transit == "ST"); df.1$TT <- 8; df.1$TT1 <- 8
df.2 <- subset(Top.prop, PD_Colon == "P" & Transit == "MT"); df.2$TT <- 16; df.2$TT1 <- 16
df.3 <- subset(Top.prop, PD_Colon == "P" & Transit == "LT"); df.3$TT <- 24; df.3$TT1 <- 24
df.4 <- subset(Top.prop, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 13; df.4$TT1 <- 8
df.5 <- subset(Top.prop, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 26; df.5$TT1 <- 16
df.6 <- subset(Top.prop, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 39; df.6$TT1 <- 24
# df.4 <- subset(Top.prop, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 8
# df.5 <- subset(Top.prop, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 16
# df.6 <- subset(Top.prop, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 24
Top.prop <- rbind(df.1, df.2, df.3, df.4, df.5, df.6)
rm(df.1, df.2, df.3, df.4, df.5, df.6)
dfcor2 <- data.frame(); PMPlist <- list(); y = 1

# foreach (i = na.omit(unique(Top.prop$Genus)), .combine = 'c') %dopar% {
for(i in na.omit(unique(Top.prop$Genus))){
  df1 <- subset(Top.prop, Genus == i)
  if(abovezero == T){
    df1.1 <- subset(df1, Donor == 1); df1.1 <- subset(df1.1, Abundance > 0)
    if (dim(df1.1)[1] == 0) {df1 <- subset(df1, Donor !=1)}
    df1.2 <- subset(df1, Donor == 2); df1.2 <- subset(df1.2, Abundance > 0) 
    if (dim(df1.2)[1] == 0) {df1 <- subset(df1, Donor !=2)}
    df1.3 <- subset(df1, Donor == 3); df1.3 <- subset(df1.3, Abundance > 0)
    if (dim(df1.3)[1] == 0) {df1 <- subset(df1, Donor !=3)}
    df1.4 <- subset(df1, Donor == 4); df1.4 <- subset(df1.4, Abundance > 0) 
    if (dim(df1.4)[1] == 0) {df1 <- subset(df1, Donor !=4)}
    df1.5 <- subset(df1, Donor == 5); df1.5 <- subset(df1.5, Abundance > 0) 
    if (dim(df1.5)[1] == 0) {df1 <- subset(df1, Donor !=5)}
    df1.6 <- subset(df1, Donor == 6); df1.6 <- subset(df1.6, Abundance > 0) 
    if (dim(df1.6)[1] == 0) {df1 <- subset(df1, Donor !=6)}
    }
  for(x in c("P", "D")){
  df<- subset(df1, PD_Colon == x)
  t <- cor(df$Abundance, df$TT, method = "spearman")
  corr <- cor.test(df$Abundance, df$TT, method = 'spearman'); 
  
  if(is.na(corr[["p.value"]])){
    sig <- "-"
    }else{
      if(corr[["p.value"]]<0.0001){
        sig <- "****"
        }else{
          if(corr[["p.value"]]<0.001){
            sig <- "***"
            }else{
              if(corr[["p.value"]]<0.01){
                sig <- "**"
                }else{
                  if(corr[["p.value"]]<0.05){
                    sig<-"*"
                  }else{
                      sig<-""}}}}}
  
  dfcor2 <- rbind(dfcor2, data.frame(genus = i, cor = corr[["estimate"]][["rho"]], 
                 PD_Colon = x, pvalue = corr[["p.value"]], sig = sig, comm = "Proportional microbial community"))
  
  if(x == "P"){corP <- paste0("Ï = ",round(t, digits = 2)," ", sig, sep = ""); corrP <- corr[["p.value"]]}
  if(x == "D"){corD <- paste0("Ï = ",round(t, digits = 2)," ", sig, sep = ""); corrD <- corr[["p.value"]]}
  }
  
  #   dat_text <- data.frame(label = c(corP, corD),
  # PD_Colon   = c("P","D"),
  # value   = c(max(df1$Abundance), max(df1$Abundance)), 
  # TT = c(8,8))
  # dat_text$PD_Colon <- as.ordered(factor(dat_text$PD_Colon, levels = c("P", "D")))
  
plot <- papertheme(ggplot(df1, aes(y = Abundance, x = TT1))) + geom_boxplot(aes(fill = Transit)) + geom_jitter(aes(color = Donor, shape = Donor), size =4, height = 0) + 
      facet_grid(rows = vars(PD_Colon), labeller = labeller(PD_Colon = PD_Colon.labs), scales = "free_x") 

  if(is.na(corrP)){corrP <- 1} 
  if(is.na(corrD)){corrD <- 1} 
  if (corrP > 0.05 & corrD > 0.05) {
    plot <- plot
    dat_text <- data.frame(label = c("", ""), PD_Colon = c("P",
        "D"), value = c(max(df1$Abundance), max(df1$Abundance)),
        xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
        hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
        TT1 = c(x_lab, x_lab))
} else {
    if (corrP > 0.05 & corrD <= 0.05) {
        if(smoothline ==T){plot <- plot + geom_smooth(data = subset(df1, PD_Colon =="D"), 
                                   aes(y = Abundance, x = TT1), method = lm, color = "black")}
        dat_text <- data.frame(label = c("", corD), PD_Colon = c("P",
            "D"), value = c(max(df1$Abundance), max(df1$Abundance)),
            xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
            hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
            TT1 = c(x_lab, x_lab))
    } else {
        if (corrP <= 0.05 & corrD > 0.05) {
            if(smoothline ==T){plot <- plot + geom_smooth(data = subset(df1,
                PD_Colon == "P"), aes(y = Abundance, x = TT1),
                method = lm, color = "black")}
            dat_text <- data.frame(label = c(corP, ""),
                PD_Colon = c("P", "D"), value = c(max(df1$Abundance),
                  max(df1$Abundance)), 
                xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),TT1 = c(x_lab, x_lab))
        } else {
            if(smoothline ==T){plot <- plot + geom_smooth(aes(y = Abundance,
                x = TT1), method = lm, color = "black")}
            dat_text <- data.frame(label = c(corP, corD),
                PD_Colon = c("P", "D"), value = c(max(df1$Abundance),
                  max(df1$Abundance)), 
                xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
                TT1 = c(x_lab, x_lab))
        }
    }
}
  dat_text$PD_Colon <- as.ordered(factor(dat_text$PD_Colon, levels = c("P", "D")))

plot <- plot + scale_color_manual(values = sixdonors) + scale_fill_manual(values = transitcolors) +
  scale_x_continuous(breaks=c(8,16,24), labels=labelsx) +
  ylab("Prop abundance (%)")+ xlab("Transit time") + 
  # scale_x_discrete(labels = c("Short", "Medium", "Long"))+
    theme(strip.background =element_rect(fill=fillcolor),
          legend.position = "none",
          legend.box = "horizontal",
          strip.text.x = element_text(size = 20),
          strip.text.y = element_blank(), 
          # strip.text.y = element_text(size = 20, angle = 270),
          legend.title = element_text(size=20), legend.background = element_blank()) 
# if(i == "Veillonella"){
#     plot <- plot + theme(strip.text.y = element_text(size = 20, angle = 270))
#   }
  
correlations <- stats_dataframe(df1,i, profile = "prop")

      if(print_rho == F){plot <- plot + geom_text(data= correlations,  mapping = aes(x = TT1, y = Abundance, label = label, 
                                        family = 'sans'), size = 7)
}else{
  plot <- plot + geom_text(data= correlations,  mapping = aes(x = TT1, y = Abundance, label = label, 
                                        family = 'sans'), size = 7)
  # plot <- plot + geom_text(data= dat_text,  mapping = aes(x = as.numeric(TT1), y = as.numeric(value), label = label, family = 'sans'), size = 7)
  plot <- plot + geom_text(data= dat_text,  mapping = aes(x = xpos, y = ypos, label = label, 
                                        hjust=hjustvar,vjust=vjustvar, family = 'sans'), size = 7)
}

if(BW == TRUE){plot <- plot + scale_fill_grey(name = "SHIME transit", start = 0.7, end = 1, labels= c("Short", "Medium", "Long")) + scale_color_grey(start = 0, end = 0.3) +scale_shape_manual(name = "Donor", values=c("1" = 15, "2" = 16, "3" = 17 , "4" = 18, "5" = 25, "6" = 9)) + guides(color = "none")+ xlab("Transit")}else{plot <- plot + scale_shape_manual(name = "Donor", values=c("1" = 16, "2" = 16, "3" = 16 , "4" = 16, "5" = 16, "6" = 16)) + guides(shape = "none")}   


PMPlist[[i]] <- plot


   ### Statistics figures
    vessels = "P"
    test <- subset(df1, PD_Colon == vessels)
    res.kruskal25 <- subset(df1, PD_Colon == vessels) %>% rstatix::kruskal_test(Abundance ~ Transit)
         res.kruskal22 <- res.kruskal25; res.kruskal22$vessel <- vessels; res.kruskal22$Genus <- i; 
         res.kruskal22$Abundance <- "PROP"
     df.res.kruskal.all <- rbind(df.res.kruskal.all, res.kruskal22)
    pwc <- subset(df1, PD_Colon == vessels) %>%  wilcox_test(Abundance ~ Transit, p.adjust.method = "holm", paired = F)
 
    pwc <- subset(pwc, p.adj.signif != "ns")
    pwc$Genus <- i; pwc$Vessel <- vessels; pwc$abundance <- "PROP"
    pwc$maxST <- max(subset(test, Transit =="ST")$Abundance); pwc$maxMT <- max(subset(test, Transit =="MT")$Abundance)
    pwc$maxLT <- max(subset(test, Transit =="LT")$Abundance) ; df.pwc <- rbind(df.pwc, pwc)
    # print(pwc) # print pairwise comparisons
    
    # Visualization: box plots with p-values
    pwc <- pwc %>% add_xy_position(x = "Transit")
    plot64 <- ggboxplot(subset(df1, PD_Colon == vessels), x = "Transit", y = "Abundance") +
    stat_pvalue_manual(pwc, hide.ns = TRUE, label = "p") +
    labs(title = paste0("Genus ",i,", Vessel ",vessels),
    subtitle = get_test_label(res.kruskal25, detailed = TRUE),
    caption = get_pwc_label(pwc))
    
    
    vessels = "D"
    test <- subset(df1, PD_Colon == vessels)
    res.kruskal25 <- subset(df1, PD_Colon == vessels) %>% rstatix::kruskal_test(Abundance ~ Transit)
         res.kruskal22 <- res.kruskal25; res.kruskal22$vessel <- vessels; res.kruskal22$Genus <- i; 
         res.kruskal22$Abundance <- "PROP"
     df.res.kruskal.all <- rbind(df.res.kruskal.all, res.kruskal22)
    pwc <- subset(df1, PD_Colon == vessels) %>%  wilcox_test(Abundance ~ Transit, p.adjust.method = "holm", paired = F)
 
    pwc <- subset(pwc, p.adj.signif != "ns")
    pwc$Genus <- i; pwc$Vessel <- vessels; pwc$abundance <- "PROP"
    pwc$maxST <- max(subset(test, Transit =="ST")$Abundance); pwc$maxMT <- max(subset(test, Transit =="MT")$Abundance)
    pwc$maxLT <- max(subset(test, Transit =="LT")$Abundance)
    df.pwc <- rbind(df.pwc, pwc)
    # print(pwc) # print pairwise comparisons
    
    # Visualization: box plots with p-values
    pwc <- pwc %>% add_xy_position(x = "Transit")
    plot65 <- ggboxplot(subset(df1, PD_Colon == vessels), x = "Transit", y = "Abundance") +
    stat_pvalue_manual(pwc, hide.ns = TRUE, label = "p") +
    labs(title = paste0("Genus ",i,", Vessel ",vessels),
    subtitle = get_test_label(res.kruskal25, detailed = TRUE),
    caption = get_pwc_label(pwc)); 
    
    gglist_stats[[statscount]] <- ggarrange(plot64, plot65, nrow=2)  
    statscount <- statscount +1
    
    rm(plot64, plot65, vessels)

y = y+1
    }

# Combine abs and prop abundances in one
dfcor <- rbind(dfcor, dfcor2)
dfcor <- na.omit(dfcor); dfcor$comm <- as.ordered(factor(dfcor$comm,
                              levels = c("Quantitative microbial community", "Proportional microbial community")))
dfcor$PD_Colon <- as.ordered(factor(dfcor$PD_Colon, levels = c("P", "D")))
dfcor$genus <- as.ordered(factor(dfcor$genus, levels = Top.genera))
plot <- papertheme(ggplot(dfcor, aes(x = PD_Colon, y = genus, fill =cor))) + geom_tile() +
  scale_fill_gradient2(na.value = "grey50",low="blue", high="red") +
  geom_text(aes(label = sig)) +
  theme(axis.text.y = element_text(face="italic")) + ylab("Genus") + xlab("Colon vessel") +
  scale_x_discrete(breaks=c("P","D"), labels=c("Proximal colon", "Distal colon")) +
  labs(fill = expression(rho)) +
  facet_grid(cols = vars(comm)) +theme(strip.background = element_rect(fill = fillcolor))
plot
dfcorall <- dfcor


# Combine plots
plots <- list(); x = 1
genera <- na.omit(unique(Top.prop$Genus))
genera <- sort(genera, decreasing = F)
# foreach (i = genera) %dopar% {
for(i in genera){
  plot1 <- QMPlist[[i]]; plot2 <- PMPlist[[i]]
  
  plot1_Y <- ggplot_build(plot1)$layout$panel_params[[1]]$y$get_labels()
  plot1_Y <- plot1_Y[!is.na(plot1_Y)]
  plot1_Y <- nchar(plot1_Y[[1]])+nY
  
  plot2_Y <- ggplot_build(plot2)$layout$panel_params[[1]]$y$get_labels()
  plot2_Y <- plot2_Y[!is.na(plot2_Y)]
  plot2_Y <- as.numeric(plot2_Y)
  plot2_Y <- nchar(max(plot2_Y))+nY
  
 
  
  plot <- ggarrange(plot1, plot2, ncol = 2, widths = c(1,  nY))
  plot <- annotate_figure(plot, text_grob(i, color = "black", face = "bold.italic", size = 25))
  # print(plot)
  plots[[i]] <- plot
  x = x +1
}

plots1 <- plots[1:8]; plots2 <- plots[9:16]
ggarrange(plotlist = plots, nrow = 5, ncol = 6)
# ggarrange(plotlist = plots2, nrow = 2, ncol = 4)
# ggarrange(plotlist = plots, nrow = 4, ncol = 4)









########################################################################

# OTU LEVEL
# Create OTU list with the genera
Top.OTU <- data.frame(); df.OTU <-df.combined
for(i in Top.genera){
  df.test <- subset(df.OTU, genus == i)
  Top.OTU <- rbind(Top.OTU, df.test)
}
Top.OTU <- unique(Top.OTU$rowname)

# Abs abundanceE
Top15.otu <- psmelt(subset_samples(phylobj.QMP.otu, Ninetransit == 3))
# unique(Top15.otu$OTU)

## Subset according to top most abundant
df.test <- data.frame()
for(i in Top.OTU){
  df.test2 <- subset(Top15.otu, OTU == i)
  df.test <- rbind(df.test, df.test2)
}
Top15.otu <- df.test
unique(Top15.otu$OTU)

## Add extra column days for graphs
Top.otu <- Top15.otu
Top.otu$Transit <- as.ordered(factor(Top.otu$Transit, levels = c("ST", "MT", "LT")))
Top.otu$PD_Colon <- as.ordered(factor(Top.otu$PD_Colon, levels = c("P", "D", "FS")))
Top.otu$Donor <- as.ordered(factor(Top.otu$Donor, levels = c("1", "2", "3", "4", "5", "6")))

## Add transit numbers
df.1 <- subset(Top.otu, PD_Colon == "P" & Transit == "ST"); df.1$TT <- 8; df.1$TT1 <- 8
df.2 <- subset(Top.otu, PD_Colon == "P" & Transit == "MT"); df.2$TT <- 16; df.2$TT1 <- 16
df.3 <- subset(Top.otu, PD_Colon == "P" & Transit == "LT"); df.3$TT <- 24; df.3$TT1 <- 24
df.4 <- subset(Top.otu, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 13; df.4$TT1 <- 8
df.5 <- subset(Top.otu, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 26; df.5$TT1 <- 16
df.6 <- subset(Top.otu, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 39; df.6$TT1 <- 24
Top.otu <- rbind(df.1, df.2, df.3, df.4, df.5, df.6)
rm(df.1, df.2, df.3, df.4, df.5, df.6)


# Remove unwanted OTUs here 
Top.otu <- subset(Top.otu, OTU!= "Otu00142" & OTU != "Otu00429" & OTU != "Otu00477"& OTU != "Otu00219")

## Perform tests
dfcor.otu <- data.frame(); OTUQMPlist <- list()
for(i in na.omit(unique(Top.otu$OTU))){
  df1 <- subset(Top.otu, OTU == i)
  if(abovezero == T){
    df1.1 <- subset(df1, Donor == 1); df1.1 <- subset(df1.1, Abundance > 0)
    if (dim(df1.1)[1] == 0) {df1 <- subset(df1, Donor !=1)}
    df1.2 <- subset(df1, Donor == 2); df1.2 <- subset(df1.2, Abundance > 0) 
    if (dim(df1.2)[1] == 0) {df1 <- subset(df1, Donor !=2)}
    df1.3 <- subset(df1, Donor == 3); df1.3 <- subset(df1.3, Abundance > 0)
    if (dim(df1.3)[1] == 0) {df1 <- subset(df1, Donor !=3)}
    df1.4 <- subset(df1, Donor == 4); df1.4 <- subset(df1.4, Abundance > 0) 
    if (dim(df1.4)[1] == 0) {df1 <- subset(df1, Donor !=4)}
    df1.5 <- subset(df1, Donor == 5); df1.5 <- subset(df1.5, Abundance > 0) 
    if (dim(df1.5)[1] == 0) {df1 <- subset(df1, Donor !=5)}
    df1.6 <- subset(df1, Donor == 6); df1.6 <- subset(df1.6, Abundance > 0) 
    if (dim(df1.6)[1] == 0) {df1 <- subset(df1, Donor !=6)}
    }
  for(x in c("P", "D")){
  df<- subset(df1, PD_Colon == x)
  t <- cor(df$Abundance, df$TT, method = "spearman")
  
  if(is.na(t)){corr[["p.value"]] <- NA}else{
   corr <- cor.test(df$Abundance, df$TT, method = 'spearman')  
  }
  
  
  if(is.na(corr[["p.value"]])){
    sig <- ""
    }else{
      if(corr[["p.value"]]<0.0001){
        sig <- "****"
        }else{
          if(corr[["p.value"]]<0.001){
            sig <- "***"
            }else{
              if(corr[["p.value"]]<0.01){
                sig <- "**"
                }else{
                  if(corr[["p.value"]]<0.05){
                    sig<-"*"
                  }else{
                      sig<-""}}}}}
  
  if(is.na(t)){dfcor.otu <- rbind(dfcor.otu, data.frame(genus = i, cor = NA, 
                 PD_Colon = x, pvalue = corr[["p.value"]], sig = sig, comm = "Quantitative microbial community"))
  }else{
    dfcor.otu <- rbind(dfcor.otu, data.frame(genus = i, cor = corr[["estimate"]][["rho"]], 
                 PD_Colon = x, pvalue = corr[["p.value"]], sig = sig, comm = "Quantitative microbial community"))}
  
  if(x == "P"){corP <- paste0("Ï = ",round(t, digits = 2)," ", sig, sep = ""); corrP <- corr[["p.value"]]
  }else{corD <- paste0("Ï = ",round(t, digits = 2)," ", sig, sep = ""); corrD <- corr[["p.value"]]}
  }
  
  
  if(i == "Otu00017"){backup <- df1; backup$Abundance <- 0
  backup$Abundance <- as.numeric(backup$Abundance)}
  if(i == "Otu00142"|i =="Otu00219"){plot <- papertheme(ggplot(backup, 
                            aes(y = Abundance, x = TT1))) + 
    geom_boxplot(aes(fill = Transit)) + geom_jitter(aes(color = Donor), size =4, height = 0) + 
      facet_grid(rows = vars(PD_Colon), labeller = labeller(PD_Colon = PD_Colon.labs), scales = "free_x") 
  }else{
    plot <- papertheme(ggplot(df1, aes(y = Abundance, x = TT1))) + geom_boxplot(aes(fill = Transit)) +
      geom_jitter(aes(color = Donor, shape = Donor), size =4, height = 0) + 
      facet_grid(rows = vars(PD_Colon), labeller = labeller(PD_Colon = PD_Colon.labs), scales = "free_x") 
  }
  
  if(is.na(corrP)){corrP <- 1} 
  if(is.na(corrD)){corrD <- 1} 
  if (corrP > 0.05 & corrD > 0.05) {
    plot <- plot
    dat_text <- data.frame(label = c("", ""), PD_Colon = c("P",
        "D"), value = c(max(df1$Abundance), max(df1$Abundance)),
        xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
        hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
        TT1 = c(x_lab, x_lab))
} else {
    if (corrP > 0.05 & corrD <= 0.05) {
        if(smoothline ==T){plot <- plot + geom_smooth(data = subset(df1, PD_Colon =="D"), 
                                   aes(y = Abundance, x = TT1), method = lm,
                                   color = "black")}
        dat_text <- data.frame(label = c("", corD), PD_Colon = c("P",
            "D"), value = c(max(df1$Abundance), max(df1$Abundance)),
            xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
            hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
            TT1 = c(x_lab, x_lab))
    } else {
        if (corrP <= 0.05 & corrD > 0.05) {
            if(smoothline ==T){plot <- plot + geom_smooth(data = subset(df1,
                PD_Colon == "P"), aes(y = Abundance, x = TT1),
                method = lm, color = "black")}
            dat_text <- data.frame(label = c(corP, ""),
                PD_Colon = c("P", "D"), value = c(max(df1$Abundance),
                  max(df1$Abundance)),
                xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),TT1 = c(x_lab, x_lab))
        } else {
            if(smoothline ==T){plot <- plot + geom_smooth(aes(y = Abundance,
                x = TT1), method = lm, color = "black")}
            dat_text <- data.frame(label = c(corP, corD),
                PD_Colon = c("P", "D"), value = c(max(df1$Abundance),
                  max(df1$Abundance)), 
                xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
                TT1 = c(x_lab, x_lab))
        }
    }
}
  dat_text$PD_Colon <- as.ordered(factor(dat_text$PD_Colon, levels = c("P", "D")))

plot <- plot + scale_color_manual(values = sixdonors) + 
  scale_fill_manual("SHIME transit", values = transitcolors, labels = c("Short", "Medium", "Long")) +
  # scale_x_discrete(labels = c("Short", "Medium", "Long"))+
  scale_x_continuous(breaks=c(8,16,24), labels=labelsx) +
  ylab(expression(paste("Abs abundance (cells ",mL^{"-1"},")")))+ xlab("Transit time") + 
  scale_y_continuous(labels = label_scientific(digits = scientificdigits)) 
if(i == "Otu00119a" | i == "Otu00062a"){plot <- plot + 
   guides(fill=guide_legend(ncol=1, title.position = "top", hjust = 0.5),
          color=guide_legend(nrow=3, title.position = "top", hjust = 0.5)) +
   theme(legend.title.align = 0.5, legend.title = element_text(size=20), 
         legend.text = element_text(size=20), legend.position = c(0.5,0.85), legend.box = "horizontal")}else{
           if(i == "Otu00042a"){plot <- plot + 
   guides(fill=guide_legend(ncol=1, title.position = "top", hjust = 0.5),
          color=guide_legend(nrow=3, title.position = "top", hjust = 0.5)) +
   theme(legend.title.align = 0.5, legend.title = element_text(size=20), 
         legend.text = element_text(size=20), legend.position = c(0.5,0.30), legend.box = "horizontal")}else{
   plot <- plot + theme(legend.position = "none")}}
         plot <- plot + theme(strip.background =element_rect(fill=fillcolor),
         # legend.box = "horizontal",
         strip.text.x = element_text(size = 20),
         strip.text.y = element_blank(),
         # strip.text.y = element_text(size = 20, angle = 270),
         legend.title = element_text(size=20), legend.background = element_blank())
         
    
stats_otu = T
if(stats_otu == T){correlations <- stats_dataframe(df1,i, profile = "quant")} 

      if(print_rho == F){
        if(stats_otu == T){
        plot <- plot + geom_text(data= correlations,  
                                                  mapping = aes(x = TT1, y = Abundance, label = label, 
                                        family = 'sans'), size = 7)}else{plot <- plot}
}else{
  if(stats_otu == T){
        plot <- plot + geom_text(data= correlations,  
                                                  mapping = aes(x = TT1, y = Abundance, label = label, 
                                        family = 'sans'), size = 7)}
  # plot <- plot + geom_text(data= dat_text,  mapping = aes(x = as.numeric(TT1), y = as.numeric(value), label = label, family = 'sans'), size = 7)
  plot <- plot + geom_text(data= dat_text,  mapping = aes(x = xpos, y = ypos, label = label, 
                                        hjust=hjustvar,vjust=vjustvar, family = 'sans'), size = 7)
}

if(BW == TRUE){plot <- plot + scale_fill_grey(name = "SHIME transit", start = 0.7, end = 1, labels= c("Short", "Medium", "Long")) + scale_color_grey(start = 0, end = 0.3) +scale_shape_manual(name = "Donor", values=c("1" = 15, "2" = 16, "3" = 17 , "4" = 18, "5" = 25, "6" = 9)) + guides(color = "none") + xlab("Transit")
}else{plot <- plot + scale_shape_manual(name = "Donor", values=c("1" = 16, "2" = 16, "3" = 16 , "4" = 16, "5" = 16, "6" = 16)) + guides(shape = "none")}

 OTUQMPlist[[i]] <- plot; 
 
 
 ### Statistics
    vessels = "P"
    test <- subset(df1, PD_Colon == vessels)
    dfje <- subset(subset(df1, PD_Colon == vessels), Abundance >0) 
    
    if(length(unique(dfje$Transit))>1){
      res.kruskal25 <- subset(df1, PD_Colon == vessels) %>% rstatix::kruskal_test(Abundance ~ Transit)
           res.kruskal22 <- res.kruskal25; res.kruskal22$vessel <- vessels; res.kruskal22$Genus <- i; 
           res.kruskal22$Abundance <- "ABS"
     df.res.kruskal.all <- rbind(df.res.kruskal.all, res.kruskal22)
    pwc <- subset(df1, PD_Colon == vessels) %>%  wilcox_test(Abundance ~ Transit, p.adjust.method = "holm", paired = F)
    }else{
      pwc <- data.frame(.y.="Abundance",
      group1 ="ST",
      group2 = "MT",
      n1 = 30,
      n2 = 30,
      statistic = 700,
      p = 1,
      p.adj = 1,
      p.adj.signif = "ns")
    }
 
    pwc <- subset(pwc, p.adj.signif != "ns")
    if(nrow(pwc)>0){
    pwc$Genus <- i; pwc$Vessel <- vessels; pwc$abundance <- "ABS"; df.pwc.otu <- rbind(df.pwc.otu, pwc)
    # print(pwc) # print pairwise comparisons
    
    # Visualization: box plots with p-values
    pwc <- pwc %>% add_xy_position(x = "Transit")
    plot64 <- ggboxplot(subset(df1, PD_Colon == vessels), x = "Transit", y = "Abundance") +
    stat_pvalue_manual(pwc, hide.ns = TRUE, label = "p") +
    labs(title = paste0("OTU ",i,", Vessel ",vessels),
    subtitle = get_test_label(res.kruskal25, detailed = TRUE),
    caption = get_pwc_label(pwc))
    }else{
      plot64 <- ggboxplot(subset(df1, PD_Colon == vessels), x = "Transit", y = "Abundance") +
      labs(title = paste0("OTU ",i,", Vessel ",vessels))}
    
    
    vessels = "D"
    
    dfje <- subset(subset(df1, PD_Colon == vessels), Abundance >0) 
    
    if(length(unique(dfje$Transit))>1){
      res.kruskal25 <- subset(df1, PD_Colon == vessels) %>% rstatix::kruskal_test(Abundance ~ Transit)
           res.kruskal22 <- res.kruskal25; res.kruskal22$vessel <- vessels; res.kruskal22$Genus <- i; res.kruskal22$Abundance <- "ABS"
     df.res.kruskal.all <- rbind(df.res.kruskal.all, res.kruskal22)
    pwc <- subset(df1, PD_Colon == vessels) %>%  wilcox_test(Abundance ~ Transit, p.adjust.method = "holm", paired = F)
    }else{
      pwc <- data.frame(.y.="Abundance",
      group1 ="ST",
      group2 = "MT",
      n1 = 30,
      n2 = 30,
      statistic = 700,
      p = 1,
      p.adj = 1,
      p.adj.signif = "ns")
    }
 
    pwc <- subset(pwc, p.adj.signif != "ns")
    
    if(nrow(pwc)>0){
    pwc$Genus <- i; pwc$Vessel <- vessels; pwc$abundance <- "ABS"; df.pwc.otu <- rbind(df.pwc.otu, pwc)
    # print(pwc) # print pairwise comparisons
    
    # Visualization: box plots with p-values
    pwc <- pwc %>% add_xy_position(x = "Transit")
    plot65 <- ggboxplot(subset(df1, PD_Colon == vessels), x = "Transit", y = "Abundance") +
    stat_pvalue_manual(pwc, hide.ns = TRUE, label = "p") +
    labs(title = paste0("OTU ",i,", Vessel ",vessels),
    subtitle = get_test_label(res.kruskal25, detailed = TRUE),
    caption = get_pwc_label(pwc))
    }else{
      plot65 <- ggboxplot(subset(df1, PD_Colon == vessels), x = "Transit", y = "Abundance") +
      labs(title = paste0("OTU ",i,", Vessel ",vessels))}
    
    gglist_stats.otu[[statscount.otu]] <- ggarrange(plot64, plot65, nrow=2)  
    statscount.otu <- statscount.otu +1
 
 # y = y+1
  
  }


# Proportional
Top15.prop.otu <- psmelt(subset_samples(phylobj.prop.otu, Ninetransit == 3)); unique(Top15.prop.otu$OTU)
## Subset according to top most abundant
df.test <- data.frame()
for(i in Top.OTU){
  df.test2 <- subset(Top15.prop.otu, OTU == i)
  df.test <- rbind(df.test, df.test2)
}
Top15.prop.otu <- df.test
# unique(Top15.prop.otu$OTU)

## Add extra column days for graphs
Top.prop.otu <- Top15.prop.otu
Top.prop.otu$Transit <- as.ordered(factor(Top.prop.otu$Transit, levels = c("ST", "MT", "LT")))
Top.prop.otu$PD_Colon <- as.ordered(factor(Top.prop.otu$PD_Colon, levels = c("P", "D", "FS")))
Top.prop.otu$Donor <- as.ordered(factor(Top.prop.otu$Donor, levels = c("1", "2", "3", "4", "5", "6")))

## Add numbers
df.1 <- subset(Top.prop.otu, PD_Colon == "P" & Transit == "ST"); df.1$TT <- 8; df.1$TT1 <- 8
df.2 <- subset(Top.prop.otu, PD_Colon == "P" & Transit == "MT"); df.2$TT <- 16; df.2$TT1 <- 16
df.3 <- subset(Top.prop.otu, PD_Colon == "P" & Transit == "LT"); df.3$TT <- 24; df.3$TT1 <- 24
df.4 <- subset(Top.prop.otu, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 13; df.4$TT1 <- 8
df.5 <- subset(Top.prop.otu, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 26; df.5$TT1 <- 16
df.6 <- subset(Top.prop.otu, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 39; df.6$TT1 <- 24
Top.prop.otu <- rbind(df.1, df.2, df.3, df.4, df.5, df.6)
rm(df.1, df.2, df.3, df.4, df.5, df.6)

# Remove unwanted OTUs here 
Top.prop.otu <- subset(Top.prop.otu, OTU!= "Otu00142" & OTU != "Otu00429" & OTU != "Otu00477"& OTU != "Otu00219")


## Perform tests
dfcor.otu <- data.frame(); OTUPMPlist <- list()
for(i in na.omit(unique(Top.prop.otu$OTU))){
  df1 <- subset(Top.prop.otu, OTU == i); df.bu <- df1
  if(abovezero == T){
    df1.1 <- subset(df1, Donor == 1); df1.1 <- subset(df1.1, Abundance > 0)
    if (dim(df1.1)[1] == 0) {df1 <- subset(df1, Donor !=1)}
    df1.2 <- subset(df1, Donor == 2); df1.2 <- subset(df1.2, Abundance > 0) 
    if (dim(df1.2)[1] == 0) {df1 <- subset(df1, Donor !=2)}
    df1.3 <- subset(df1, Donor == 3); df1.3 <- subset(df1.3, Abundance > 0)
    if (dim(df1.3)[1] == 0) {df1 <- subset(df1, Donor !=3)}
    df1.4 <- subset(df1, Donor == 4); df1.4 <- subset(df1.4, Abundance > 0) 
    if (dim(df1.4)[1] == 0) {df1 <- subset(df1, Donor !=4)}
    df1.5 <- subset(df1, Donor == 5); df1.5 <- subset(df1.5, Abundance > 0) 
    if (dim(df1.5)[1] == 0) {df1 <- subset(df1, Donor !=5)}
    df1.6 <- subset(df1, Donor == 6); df1.6 <- subset(df1.6, Abundance > 0) 
    if (dim(df1.6)[1] == 0) {df1 <- subset(df1, Donor !=6)}
  }
  
  if(nrow(df1) == 0){df1 <- subset(df.bu, Donor == 1&Transit == "ST"); df1$Abundance <- 0.00001}
  for(x in c("P", "D")){
  df<- subset(df1, PD_Colon == x)
  t <- cor(df$Abundance, df$TT, method = "spearman")
  
  if(is.na(t)){corr[["p.value"]] <- NA}else{
   corr <- cor.test(df$Abundance, df$TT, method = 'spearman')  
  }
  
  
  if(is.na(corr[["p.value"]])){
    sig <- ""
    }else{
      if(corr[["p.value"]]<0.0001){
        sig <- "****"
        }else{
          if(corr[["p.value"]]<0.001){
            sig <- "***"
            }else{
              if(corr[["p.value"]]<0.01){
                sig <- "**"
                }else{
                  if(corr[["p.value"]]<0.05){
                    sig<-"*"
                  }else{
                      sig<-""}}}}}
  
  if(is.na(t)){dfcor.otu <- rbind(dfcor.otu, data.frame(genus = i, cor = NA, 
                 PD_Colon = x, pvalue = corr[["p.value"]], sig = sig, comm = "Quantitative microbial community"))
  }else{
    dfcor.otu <- rbind(dfcor.otu, data.frame(genus = i, cor = corr[["estimate"]][["rho"]], 
                 PD_Colon = x, pvalue = corr[["p.value"]], sig = sig, comm = "Quantitative microbial community"))}
  
  if(x == "P"){corP <- paste0("Ï = ",round(t, digits = 2)," ", sig, sep = ""); corrP <- corr[["p.value"]]
  }else{corD <- paste0("Ï = ",round(t, digits = 2)," ", sig, sep = ""); corrD <- corr[["p.value"]]}
  }
  
  plot <- papertheme(ggplot(df1, aes(y = Abundance, x = TT1))) + geom_boxplot(aes(fill = Transit)) + geom_jitter(aes(color = Donor, shape = Donor), size =4, height = 0) + 
      facet_grid(rows = vars(PD_Colon), labeller = labeller(PD_Colon = PD_Colon.labs), scales = "free_x") 

  
  
  if(is.na(corrP)){corrP <- 1} 
  if(is.na(corrD)){corrD <- 1} 
  if (corrP > 0.05 & corrD > 0.05) {
    plot <- plot
    dat_text <- data.frame(label = c("", ""), PD_Colon = c("P",
        "D"), value = c(max(df1$Abundance), max(df1$Abundance)),
        xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
        TT1 = c(x_lab, x_lab))
} else {
    if (corrP > 0.05 & corrD <= 0.05) {
        if(smoothline ==T){plot <- plot + geom_smooth(data = subset(df1, PD_Colon =="D"), 
                                   aes(y = Abundance, x = TT1), method = lm,
                                   color = "black")}
        dat_text <- data.frame(label = c("", corD), PD_Colon = c("P",
            "D"), value = c(max(df1$Abundance), max(df1$Abundance)),
            xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
            TT1 = c(x_lab, x_lab))
    } else {
        if (corrP <= 0.05 & corrD > 0.05) {
            if(smoothline ==T){plot <- plot + geom_smooth(data = subset(df1,
                PD_Colon == "P"), aes(y = Abundance, x = TT1),
                method = lm, color = "black")}
            dat_text <- data.frame(label = c(corP, ""),
                PD_Colon = c("P", "D"), value = c(max(df1$Abundance),
                  max(df1$Abundance)), 
                xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),TT1 = c(x_lab, x_lab))
        } else {
            if(smoothline ==T){plot <- plot + geom_smooth(aes(y = Abundance,
                x = TT1), method = lm, color = "black")}
            dat_text <- data.frame(label = c(corP, corD),
                PD_Colon = c("P", "D"), value = c(max(df1$Abundance),
                  max(df1$Abundance)), 
                xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),TT1 = c(x_lab, x_lab))
        }
    }
}
  dat_text$PD_Colon <- as.ordered(factor(dat_text$PD_Colon, levels = c("P", "D")))

plot <- plot + scale_color_manual(values = sixdonors) + 
  scale_fill_manual("SHIME transit", values = transitcolors, labels = c("Short", "Medium", "Long")) + 
  guides(fill=guide_legend(ncol=1, title.position = "top", hjust = 0.5), 
         color=guide_legend(nrow=3, title.position = "top", hjust = 0.5)) +
  # scale_x_discrete(labels = c("Short", "Medium", "Long"))+ 
  scale_x_continuous(breaks=c(8,16,24), labels=labelsx) +
  theme(legend.title.align = 0.5, legend.title = element_text(size=25), legend.text = element_text(size=25))+
  ylab("Prop abundance (%)")+ xlab("Transit time") 
  
if(BW == TRUE){plot <- plot + scale_fill_grey(name = "SHIME transit", start = 0.7, end = 1, labels= c("Short", "Medium", "Long")) + scale_color_grey(start = 0, end = 0.3) +scale_shape_manual(name = "Donor", values=c("1" = 15, "2" = 16, "3" = 17 , "4" = 18, "5" = 25, "6" = 9)) + guides(color = "none")+ xlab("Transit")}else{plot <- plot + scale_shape_manual(name = "Donor", values=c("1" = 16, "2" = 16, "3" = 16 , "4" = 16, "5" = 16, "6" = 16)) + guides(shape = "none")}


if(i == "Otu00002"){
plt_legend <- g_legend(plot + guides(shape=guide_legend(nrow=3, title.position = "top", hjust = 0.5, override.aes = list(size = 4))))
}

if(i == "Otu00144a"){plot <- plot + 
   guides(fill=guide_legend(ncol=1, title.position = "top", hjust = 0.5),
          color=guide_legend(nrow=3, title.position = "top", hjust = 0.5)) +
   theme(legend.title.align = 0.5, legend.title = element_text(size=20), 
         legend.text = element_text(size=20), legend.position = c(0.5,0.30), legend.box = "horizontal")}else{
   plot <- plot + theme(legend.position = "none")}
         plot <- plot + theme(strip.background =element_rect(fill=fillcolor),
         # legend.box = "horizontal",
         strip.text.x = element_text(size = 20),
         strip.text.y = element_blank(),
         # strip.text.y = element_text(size = 20, angle = 270),
         legend.title = element_text(size=20), legend.background = element_blank())
         
  # if(i == "Otu00005" | i == "Otu00236" | i == "Otu00073"| i == "Otu00188" |  i == "Otu00198"|
  #    # i == "Otu00025" 
  #    i == "Otu00018" | i == "Otu00155"| i == "Otu00377" |  i == "Otu00009"|
  #    i == "Otu00026" | i == "Otu00133" | i == "Otu00048"| i == "Otu00181" |
  #    i == "Otu00254" | i == "Otu00074" | i == "Otu00266"| i == "Otu00131" |  i == "Otu00058"|
  #    i == "Otu00274" | i == "Otu00384" | i == "Otu00087"| i == "Otu00144" |  i == "Otu00051"|
  #    i == "Otu00019" | i == "Otu00149" | i == "Otu00085"| i == "Otu00116" |  i == "Otu00154"|
  #    i == "Otu00440" | i == "Otu00189" | i == "Otu00161"| i == "Otu00033" |  i == "Otu00214"|
  #    i == "Otu00261" | i == "Otu00325" | i == "Otu00126"| i == "Otu00237" |  i == "Otu00455"|
  #    i == "Otu00043" | i == "Otu00050" | i == "Otu00061"){
  #   plot <- plot + theme(strip.text.y = element_text(size = 20, angle = 270))
  # }

if(i == "Otu00018" | i == "Otu00150" | i == "Otu00377"| i == "Otu00048" |  i == "Otu00009"|
   i == "Otu00026" | i == "Otu00005"| i == "Otu00254" |  i == "Otu00069"|
   i == "Otu00266" | i == "Otu00133" | i == "Otu00061"| i == "Otu00046" |  i == "Otu00274"| 
   i == "Otu00384" | i == "Otu00087" | i == "Otu00105"| i == "Otu00144" |  i == "Otu00051"| 
   i == "Otu00019" | i == "Otu00149" | i == "Otu00076"| i == "Otu00103" |  i == "Otu00129"| 
   i == "Otu00165" | i == "Otu00440" | i == "Otu00189"| i == "Otu00236" |  i == "Otu00181"|
   i == "Otu00033" | i == "Otu00042" | i == "Otu00214"| i == "Otu00261" |  i == "Otu00325"| 
   i == "Otu00122" | i == "Otu00186" | i == "Otu00307"| i == "Otu00455" |  i == "Otu00072"| 
   i == "Otu00188" | i == "Otu00043" | i == "Otu00050"| i == "Otu00198")
  {
    plot <- plot + theme(strip.text.y = element_text(size = 20, angle = 270))
   }


 if(stats_otu == T){ correlations <- stats_dataframe(df1,i, profile = "prop") }

      if(print_rho == F){
        if(stats_otu == T){
        plot <- plot + geom_text(data= correlations,  
                                                  mapping = aes(x = TT1, y = Abundance, label = label, 
                                        family = 'sans'), size = 7)}else{plot <- plot}
}else{
  if(stats_otu == T){
        plot <- plot + geom_text(data= correlations,  
                                                  mapping = aes(x = TT1, y = Abundance, label = label, 
                                        family = 'sans'), size = 7)}
  # plot <- plot + geom_text(data= dat_text,  mapping = aes(x = as.numeric(TT1), y = as.numeric(value), label = label, family = 'sans'), size = 7)
  plot <- plot + geom_text(data= dat_text,  mapping = aes(x = xpos, y = ypos, label = label, 
                                        hjust=hjustvar,vjust=vjustvar, family = 'sans'), size = 7)
}
         
         
 OTUPMPlist[[i]] <- plot; y = y+1
 
 
 
  ### Statistics
    vessels = "P"
    
    dfje <- subset(subset(df1, PD_Colon == vessels), Abundance >0) 
    
    if(length(unique(dfje$Transit))>1){
      res.kruskal25 <- subset(df1, PD_Colon == vessels) %>% rstatix::kruskal_test(Abundance ~ Transit)
           res.kruskal22 <- res.kruskal25; res.kruskal22$vessel <- vessels; res.kruskal22$Genus <- i; 
           res.kruskal22$Abundance <- "PROP"
     df.res.kruskal.all <- rbind(df.res.kruskal.all, res.kruskal22)
    pwc <- subset(df1, PD_Colon == vessels) %>%  wilcox_test(Abundance ~ Transit, p.adjust.method = "holm", paired = F)
    }else{
      pwc <- data.frame(.y.="Abundance",
      group1 ="ST",
      group2 = "MT",
      n1 = 30,
      n2 = 30,
      statistic = 700,
      p = 1,
      p.adj = 1,
      p.adj.signif = "ns")
    }
 
    pwc <- subset(pwc, p.adj.signif != "ns")
    if(nrow(pwc)>0){
    pwc$Genus <- i; pwc$Vessel <- vessels; pwc$abundance <- "PROP"; df.pwc.otu <- rbind(df.pwc.otu, pwc)
    # print(pwc) # print pairwise comparisons
    
    # Visualization: box plots with p-values
    pwc <- pwc %>% add_xy_position(x = "Transit")
    plot64 <- ggboxplot(subset(df1, PD_Colon == vessels), x = "Transit", y = "Abundance") +
    stat_pvalue_manual(pwc, hide.ns = TRUE, label = "p") +
    labs(title = paste0("OTU ",i,", Vessel ",vessels),
    subtitle = get_test_label(res.kruskal25, detailed = TRUE),
    caption = get_pwc_label(pwc))
    }else{
      plot64 <- ggboxplot(subset(df1, PD_Colon == vessels), x = "Transit", y = "Abundance") +
      labs(title = paste0("OTU ",i,", Vessel ",vessels))}
    
    
    vessels = "D"
    
        dfje <- subset(subset(df1, PD_Colon == vessels), Abundance >0) 
    
    if(length(unique(dfje$Transit))>1){
      res.kruskal25 <- subset(df1, PD_Colon == vessels) %>% rstatix::kruskal_test(Abundance ~ Transit)
           res.kruskal22 <- res.kruskal25; res.kruskal22$vessel <- vessels; res.kruskal22$Genus <- i; 
           res.kruskal22$Abundance <- "PROP"
     df.res.kruskal.all <- rbind(df.res.kruskal.all, res.kruskal22)
    pwc <- subset(df1, PD_Colon == vessels) %>%  wilcox_test(Abundance ~ Transit, p.adjust.method = "holm", paired = F)
    }else{
      pwc <- data.frame(.y.="Abundance",
      group1 ="ST",
      group2 = "MT",
      n1 = 30,
      n2 = 30,
      statistic = 700,
      p = 1,
      p.adj = 1,
      p.adj.signif = "ns")
    }
 
    pwc <- subset(pwc, p.adj.signif != "ns")
    if(nrow(pwc)>0){
    pwc$Genus <- i; pwc$Vessel <- vessels; pwc$abundance <- "PROP"; df.pwc.otu <- rbind(df.pwc.otu, pwc)
    # print(pwc) # print pairwise comparisons
    
    # Visualization: box plots with p-values
    pwc <- pwc %>% add_xy_position(x = "Transit")
    plot65 <- ggboxplot(subset(df1, PD_Colon == vessels), x = "Transit", y = "Abundance") +
    stat_pvalue_manual(pwc, hide.ns = TRUE, label = "p") +
    labs(title = paste0("OTU ",i,", Vessel ",vessels),
    subtitle = get_test_label(res.kruskal25, detailed = TRUE),
    caption = get_pwc_label(pwc))
    }else{
      plot65 <- ggboxplot(subset(df1, PD_Colon == vessels), x = "Transit", y = "Abundance") +
      labs(title = paste0("OTU ",i,", Vessel ",vessels))}
    
    gglist_stats.otu[[statscount.otu]] <- ggarrange(plot64, plot65, nrow=2)  
    statscount.otu <- statscount.otu +1
    

  }

rm(plot64, plot65, vessels )





# Combine OTUplots
otuplots <- list(); x = 1
genera <- na.omit(unique(Top.otu$OTU))
genera <- sort(genera, decreasing = F)

# foreach (i = genera, .combine = "c") %dopar% {
  for(i in genera){
  if(i != "Otu00219" & i!="Otu00429" & i!="Otu00477"){
  plot1 <- OTUQMPlist[[i]]; plot2 <- OTUPMPlist[[i]]
  plot <- ggarrange(plot1, plot2)
  
  if(i == "Otu00018" | i == "Otu00150" | i == "Otu00377"| i == "Otu00048" |  i == "Otu00009"|
   i == "Otu00026" | i == "Otu00005"| i == "Otu00254" |  i == "Otu00069"|
   i == "Otu00266" | i == "Otu00133" | i == "Otu00061"| i == "Otu00046" |  i == "Otu00274"| 
   i == "Otu00384" | i == "Otu00087" | i == "Otu00105"| i == "Otu00144" |  i == "Otu00051"| 
   i == "Otu00019" | i == "Otu00149" | i == "Otu00076"| i == "Otu00103" |  i == "Otu00129"| 
   i == "Otu00165" | i == "Otu00440" | i == "Otu00189"| i == "Otu00236" |  i == "Otu00181"|
   i == "Otu00033" | i == "Otu00042" | i == "Otu00214"| i == "Otu00261" |  i == "Otu00325"| 
   i == "Otu00122" | i == "Otu00186" | i == "Otu00307"| i == "Otu00455" |  i == "Otu00072"| 
   i == "Otu00188" | i == "Otu00043" | i == "Otu00050"| i == "Otu00198")
  {
    plot <- ggarrange(plot1, plot2, ncol = 2, widths = c(1,  nY+0.15))
    # plot <- plot + theme(strip.text.y = element_text(size = 20, angle = 270))
   }else{plot <- ggarrange(plot1, plot2, ncol = 2, widths = c(1,  nY))}
  
  
  
  # print(plot)
  
  m =i
  if(stringr::str_detect(m, "Otu0000")){
      y <- str_replace(m, "Otu0000", "OTU")
    }else{if(stringr::str_detect(m, "Otu000")){
      y <- str_replace(m, "Otu000", "OTU")
    }else{if(stringr::str_detect(m, "Otu00")){
      y <- str_replace(m, "Otu00", "OTU")
    }else{if(stringr::str_detect(m, "Otu0")){
      y <- str_replace(m, "Otu0", "OTU")
    }}}}
  
  plot <- annotate_figure(plot, text_grob(y, color = "black", face = "bold", size = 25))
  # print(plot)
  otuplots[[i]] <- plot
  # x = x +1
  }
}

# ggarrange(plotlist = otuplots)





# Combine genera and OTU's
allplotslist <- list(); extraplotjes <- list(); nogplotjes <- list()
for(i in sort(Top.genera, decreasing = F)){
  plot1 <- plots[[i]]
dfje <- subset(df.combined, genus == i)
dfje <- subset(dfje, rowname!="Otu00102")
dfje <- subset(dfje, rowname!="Otu00021")
dfje <- subset(dfje, rowname!="Otu00216")
dfje <- subset(dfje, rowname!="Otu00219")
dfje <- subset(dfje, rowname!="Otu00191")
dfje <- subset(dfje, rowname!="Otu00277")
dfje <- subset(dfje, rowname!="Otu00078")
dfje <- subset(dfje, rowname!="Otu00133")
dfje <- subset(dfje, rowname!="Otu00142")
dfje <- subset(dfje, rowname!="Otu00114")
dfje <- subset(dfje, rowname!="Otu00178")
dfje <- subset(dfje, rowname!="Otu00332")
dfje <- subset(dfje, rowname!="Otu00395")
dfje <- subset(dfje, rowname!="Otu00407")
dfje <- subset(dfje, rowname!="Otu00429")
dfje <- subset(dfje, rowname!="Otu00477")
# dfje <- subset(dfje, mean > 5)
plotjes <- list()
plotjes[[1]] <- plot1; y = 2
for(x in unique(dfje$rowname)){
  plot2 <- otuplots[[x]]
    plotjes[[y]] <- plot2
    y = y+1
    
}
if(i == "Dialister"){dialisterplots <- plotjes}
if(i == "Eisenbergiella"){Eisenplots <- plotjes}

if(i == "Stenotrophomonas"){stenoplots <- plotjes}
if(i == "Sutterella"){suttplots <- plotjes}



# if(i == "Bifidobacterium"){ plotjes[[y]] <- ggarrange(plt_legend, NULL ,nrow = 2)}
if(i == "Acid"){ plotjes[[y]] <- ggarrange(plt_legend, NULL ,nrow = 2)}

if(
  i == "Anaeroglobus" |
   # i == "Bacteroides"| 
  i == "Clostridiales"| 
  i == "Collinsella"| 
   i == "Eisenbergiella"|
  i == "Ruminococcus" |
  # i == "Phascolarctobacterium"|
  i == "Prevotella"| 
  # i == "Stenotrophomonas"| 
   i == "Ruminococcaceae"
  ){plotjes[[y]] <- ggarrange(ggarrange(plt_legend, ncol = 1), NULL ,nrow = 2)}


if(i == "Anaeroglobus"){anplots <- plotjes}
if(i == "Bacteroides"){bacplots <- plotjes}


allplotslist[[i]] <- ggarrange(plotlist = plotjes, 
                               nrow = ceiling((nrow(dfje)+1)/4), 
                               ncol = 4)

if(i == "Acidaminococcus"){extraplotjes[[i]] <- ggarrange(plotlist = plotjes, nrow = 1, ncol =2)}
if(i == "Akkermansia"){ extraplotjes[[i]] <- ggarrange(plotlist = plotjes, nrow = 1, ncol =2)}
if(i == "Cloacibacillus"){extraplotjes[[i]] <- ggarrange(plotlist = plotjes, nrow = 1, ncol =3)}
if(i == "Clostridiales"){ extraplotjes[[i]] <- ggarrange(plotlist = plotjes, nrow = 1, ncol =2)}
if(i == "Stenotrophomonas"){ extraplotjes[[i]] <- ggarrange(plotlist = plotjes, nrow = 1, ncol =2)}
if(i == "Sutterella"){ extraplotjes[[i]] <- ggarrange(plotlist = plotjes, nrow = 1, ncol =2)}



}

nogplotjes0 <- ggarrange(anplots[[1]], anplots[[2]], anplots[[3]], bacplots[[1]], bacplots[[2]], bacplots[[3]], 
                         bacplots[[4]], bacplots[[5]], bacplots[[6]], bacplots[[7]], bacplots[[8]], bacplots[[9]], 
                         nrow = 3, ncol =4)

nogplotjes <- ggarrange(dialisterplots[[1]],dialisterplots[[2]],dialisterplots[[3]],dialisterplots[[4]],
                        dialisterplots[[5]],NULL,Eisenplots[[1]], Eisenplots[[2]], nrow = 2, ncol = 4)

nogplotjes2 <- ggarrange(stenoplots[[1]], stenoplots[[2]], NULL, suttplots[[1]],suttplots[[2]],ncol = 5, widths = c(1,1,0.1,1,1))
# ggarrange(allplotslist[[1]] + ggpubr::border(color = "grey", size = 1),
#           allplotslist[[2]] + ggpubr::border(color = "grey", size = 1), 
#           allplotslist[[3]] + ggpubr::border(color = "grey", size = 1),
#           allplotslist[[4]] + ggpubr::border(color = "grey", size = 1), ncol = 1, heights = c(1,1,2,1))
# ggarrange(plotlist = allplotslist, nrow = 5)

PD_Colon.labs <- c("Faecal sample","Proximal colon", "Distal colon")
names(PD_Colon.labs) <- c("FS","P", "D")
```

#### Plot it
```{r Paperplot correlation colons split AND combined, fig.height=25, fig.width=25}
replacelegend <- function(ggplotobject){
  ggplotnewobject <- ggplotobject + guides(fill=guide_legend(ncol=3, title.position = "left", hjust = 0.5),
          color=guide_legend(nrow=6, title.position = "left", hjust = 0.5))
  
  return(ggplotnewobject)}



plot1 <- ggarrange(ggarrange(replacelegend(extraplotjes[[1]]), NULL, replacelegend(extraplotjes[[2]]), 
                            ncol = 3, widths = c(2,0.1,2)), 
                   replacelegend(allplotslist[[3]]),
                  replacelegend(allplotslist[[4]]), 
                  nrow =3, heights = c(1,2,1), common.legend = T, legend = "bottom")
plot1 

plot2 <- ggarrange(allplotslist[[5]],
                  allplotslist[[6]], allplotslist[[7]], nrow =3, heights = c(3,1,1))
plot2


plot3 <- ggarrange(allplotslist[[8]], allplotslist[[9]],allplotslist[[10]], nrow = 3, heights = c(2,1,1))
plot3

plot4 <- ggarrange(allplotslist[[11]], allplotslist[[12]], allplotslist[[13]], nrow = 3, heights = c(2,1,1))
plot4

plot5 <- ggarrange(allplotslist[[14]], allplotslist[[15]], allplotslist[[16]], allplotslist[[17]],
                   nrow = 4, heights = c(2,1,1,1))
plot5

plot6 <- ggarrange(allplotslist[[18]], 
                   nrow = 1, heights = c(1))
plot6

plot7 <- ggarrange(allplotslist[[19]], allplotslist[[20]], allplotslist[[21]], allplotslist[[22]],
                           nrow = 4, heights = c(1,1,1,1))
plot7


plot8 <- ggarrange(allplotslist[[23]], allplotslist[[24]], allplotslist[[25]],
                           nrow = 3, heights = c(2,1,1))
plot8

plot9 <- allplotslist[[26]]
plot9

plot10 <- ggarrange(allplotslist[[27]], allplotslist[[28]], allplotslist[[29]], allplotslist[[30]],
                           nrow = 4, heights = c(2,1,1,1))
plot10




saveplots()
ggsave(plot = plot1, width = 40, height = 40, dpi = 300, filename = "FigureS8_1.png")
ggsave(plot = plot2, width = 40, height = 40, dpi = 300, filename = "FigureS8_2.png")
ggsave(plot = plot3, width = 40, height = 40, dpi = 300, filename = "FigureS8_3.png")
ggsave(plot = plot4, width = 40, height = 40, dpi = 300, filename = "FigureS8_4.png")
ggsave(plot = plot5, width = 40, height = 40, dpi = 300, filename = "FigureS8_5.png")
ggsave(plot = plot6, width = 40, height = 40, dpi = 300, filename = "FigureS8_6.png") 
ggsave(plot = plot7, width = 40, height = 40, dpi = 300, filename = "FigureS8_7.png")
ggsave(plot = plot8, width = 40, height = 40, dpi = 300, filename = "FigureS8_8.png")
ggsave(plot = plot9, width = 40, height = 40, dpi = 300, filename = "FigureS8_9.png")
ggsave(plot = plot10, width = 40, height = 40, dpi = 300, filename = "FigureS8_10.png")           
      

```

# Figure S10
#### Plot it - condensed
```{r Paperplot correlation colons split AND combined, fig.height=21, fig.width=15.5}
# Height 6 = 15, else = 21
replacelegend <- function(ggplotobject){
  ggplotnewobject <- ggplotobject + guides(fill=guide_legend(ncol=3, title.position = "left", hjust = 0.5),
          color=guide_legend(nrow=6, title.position = "left", hjust = 0.5))
  
  return(ggplotnewobject)}



plot1 <- ggarrange(ggarrange(replacelegend(extraplotjes[[1]]), NULL, replacelegend(extraplotjes[[2]]), 
                            ncol = 3, widths = c(2,0.1,2)), 
                   replacelegend(allplotslist[[3]]),
                  
                  nogplotjes0,
                  nrow =3, heights = c(1,2,3), common.legend = T, legend = "bottom")
plot1 

plot2 <- ggarrange(
                  allplotslist[[6]], allplotslist[[7]], allplotslist[[8]], allplotslist[[9]],allplotslist[[10]],
                  
                  nrow =5, heights = c(1,1,2,1,1))
plot2


plot3 <- ggarrange( allplotslist[[11]], allplotslist[[12]], allplotslist[[13]],
                    nogplotjes,
                    # allplotslist[[14]], allplotslist[[15]],
                    nrow =4, heights = c(2,1,1, 2))
plot3


plot4 <- ggarrange( allplotslist[[16]], allplotslist[[17]],allplotslist[[18]], nrow =3, heights = c(1,1,5))
plot4


plot5 <- ggarrange(allplotslist[[19]], allplotslist[[20]], allplotslist[[21]], allplotslist[[22]],allplotslist[[23]],
                           nrow = 5, heights = c(1,1,1,1, 2))
plot5


plot6 <- ggarrange( allplotslist[[24]], allplotslist[[25]],allplotslist[[26]],
                           nrow = 3, heights = c(1,1,4))
plot6


plot7 <- ggarrange(allplotslist[[27]], allplotslist[[28]], nogplotjes2,
                   # allplotslist[[29]], allplotslist[[30]],
                           nrow = 3, heights = c(2,1,1))
plot7




saveplots()
# ggsave(plot = plot1, width = 40, height = 40, dpi = 300, filename = "FigureS8_1.png")
# ggsave(plot = plot2, width = 40, height = 40, dpi = 300, filename = "FigureS8_2.png")
# ggsave(plot = plot3, width = 40, height = 40, dpi = 300, filename = "FigureS8_3.png")
# ggsave(plot = plot4, width = 40, height = 40, dpi = 300, filename = "FigureS8_4.png")
# ggsave(plot = plot5, width = 40, height = 40, dpi = 300, filename = "FigureS8_5.png")
# ggsave(plot = plot6, width = 40, height = 40, dpi = 300, filename = "FigureS8_6.png") 
# ggsave(plot = plot7, width = 40, height = 40, dpi = 300, filename = "FigureS8_7.png")
# ggsave(plot = plot8, width = 40, height = 40, dpi = 300, filename = "FigureS8_8.png")
# ggsave(plot = plot9, width = 40, height = 40, dpi = 300, filename = "FigureS8_9.png")
# ggsave(plot = plot10, width = 40, height = 40, dpi = 300, filename = "FigureS8_10.png")           
      

```


```{r Paperplot correlation colons split AND combined, fig.height=42, fig.width=40}
n = 1
plot1 <- ggarrange(ggarrange(extraplotjes[[1]],  extraplotjes[[2]], 
                            ncol = 2, widths = c(2,2)), allplotslist[[3]],
                  allplotslist[[4]], allplotslist[[5]],nrow =4, heights = c(1,2,1,2))
plot1 

n = 11
plot2 <- ggarrange(allplotslist[[6]],allplotslist[[7]],allplotslist[[8]], 
          ggarrange(extraplotjes[[3]],extraplotjes[[4]], ncol = 2, widths = c(3,2)), allplotslist[[n]],
          nrow = 5, heights = c(1,1,2,1,2))
plot2 
        

n = 12
plot3 <- ggarrange(allplotslist[[n]], allplotslist[[n+1]], allplotslist[[n+2]],
          allplotslist[[n+3]], allplotslist[[n+4]],allplotslist[[n+5]], nrow = 6, heights = c(1,1,1,1,1,1))
plot3

n = 18

plot4 <- ggarrange(allplotslist[[n]], allplotslist[[n+1]], allplotslist[[n+2]],
          allplotslist[[n+3]],nrow = 4, heights = c(4,1,1,1))
plot4 


n = 22
plot5 <- ggarrange(allplotslist[[n]], allplotslist[[n+1]],allplotslist[[n+2]], allplotslist[[n+3]],allplotslist[[n+4]],
          nrow = 5, heights = c(1,1,1,1,3))
plot5 
      

n=27
plot6 <- ggarrange(allplotslist[[n]], 
          ggarrange(extraplotjes[[5]], NULL, extraplotjes[[6]], ncol = 3, widths = c(2,1,2)),
          allplotslist[[n+3]],
          nrow = 3, heights = c(2,1,1,1))
plot6


# ggsave(plot = plot1, width = 40, height = 42, dpi = 300, filename = "FigureS8_1.png")
# ggsave(plot = plot2, width = 40, height = 42, dpi = 300, filename = "FigureS8_2.png")
# ggsave(plot = plot3, width = 40, height = 42, dpi = 300, filename = "FigureS8_3.png")
# ggsave(plot = plot4, width = 40, height = 42, dpi = 300, filename = "FigureS8_4.png")
# ggsave(plot = plot5, width = 40, height = 42, dpi = 300, filename = "FigureS8_5.png")
# ggsave(plot = plot6, width = 40, height = 42, dpi = 300, filename = "FigureS8_6.png")           
          




```

#### print stat plots
```{r Paperplot correlation colons split AND combined, fig.height=25, fig.width=20}
# Statisticplots
ggarrange(plotlist = gglist_stats, nrow = 4, ncol = 4)
ggarrange(plotlist = gglist_stats.otu, nrow = 4, ncol = 4)

view(df.res.kruskal.all)

df.res.kruskal.sig <- subset(df.res.kruskal.all, p < 0.05)
```


## Decreasing generas
```{r Paperplot correlation colons split AND combined, fig.height=32.5, fig.width=35}

# fig.height = 6.5 per row and width = 8.75 per col

# order <- c("Veillonella", 	"Sutterella", 	"Ruminococcus", 	"Ruminococcaceae", 	"Roseburia", 	"Pseudomonas", 	"Prevotella", 	"Phascolarctobacterium", 	"Parabacteroides", 	"Mitsuokella", 	"Megasphaera", 	"Megamonas", 	"Lachnospiraceae", 	"Fusobacterium", 	"Fusicatenibacter", 	"Faecalibacterium", 	"Erysipelotrichaceae", 	"Enterobacteriaceae", 	"Eisenbergiella", 	"Dorea", 	"Dialister", 	"Desulfovibrio", 	"Coprococcus", 	"Clostridium_XlVa", 	"Clostridiales", 	"Cloacibacillus", 	"Blautia", 	"Bilophila", 	"Bifidobacterium", 	"Bacteroides", 	"Bacteria", 	"Anaerostipes", 	"Anaeroglobus", 	"Alistipes", 	"Akkermansia", 	"Acidaminococcus")



# df.OTU <-subset(df.combined, sum > 10)
df.OTU <-subset(df.combined, mean > 5); x_lab = 16; 
hjustv = 1.1; vjustv = 1.5; xInf <- Inf; yInf <- Inf # Top Left (h0,v1, -inf, +inf)  Top Right (h1,v1, +inf, +inf)
topnum <- 10; abovezero = T
# Retrieve top most abundant genera
## Absolute
phylobj.QMP_stable <- subset_samples(phylobj.QMP, Ninetransit == 3)
Top15phylobj <- aggregate_taxa(phylobj.QMP_stable, level = "Genus")
Top15phylobj <- aggregate_top_taxa(phylobj.QMP_stable, top = topnum, "Genus")
## Proportional
phylobj.prop_2 = transform_sample_counts(phylobj.QMP, function(x) 100 * x/sum(x))
phylobj.prop_stable <- subset_samples(phylobj.prop_2, Ninetransit == 3)
phylobj.prop_stable <- aggregate_top_taxa(phylobj.prop_stable, top = topnum, "Genus")
## Take consensus top genera
Top.genera <- rbind(subset(as.data.frame(phylobj.prop_stable@tax_table), Genus != "Other"), 
                    subset(as.data.frame(Top15phylobj@tax_table), Genus != "Other"))
## Remove insignificant genera
df <- data.frame()
for(i in significantlist){
  df2 <- subset(Top.genera, Genus == i)
  df <- rbind(df, df2)
}
Top.genera <- unique(df$Genus)
# Remove those that are not in the significant changes list (significantlist)
Top.genera <- sort(Top.genera, decreasing = T)
Top.genera <- c("Veillonella","Ruminococcus",  "Mitsuokella",	
                # "Enterobacteriaceae", 	
                "Bifidobacterium")

# GENUS LEVEL
# Absolute
phylobj.QMP_stable <- subset_samples(phylobj.QMP, Ninetransit == 3)
Top15phylobj <- aggregate_taxa(phylobj.QMP_stable, level = "Genus")
Top15 <- psmelt(Top15phylobj); unique(Top15$OTU); unique(Top15$OTU) %in% Top15levels
## Subset according to top most abundant genera
df.test <- data.frame();
for(i in Top.genera){
  df.test2 <- subset(Top15, Genus == i)
  df.test <- rbind(df.test, df.test2)
}
Top15 <- df.test
x <- data.frame(x = unique(Top15$OTU) %in% Top15levels, genus = unique(Top15$OTU))  #check if labelled
Top15$OTU <- as.ordered(factor(Top15$OTU, levels = Top15levels))
## Add extra column days for graphs
Top <- Top15
Top$Transit <- as.ordered(factor(Top$Transit, levels = c("ST", "MT", "LT")))
Top$PD_Colon <- as.ordered(factor(Top$PD_Colon, levels = c("P", "D", "FS")))
Top$Donor <- as.ordered(factor(Top$Donor, levels = c("1", "2", "3", "4", "5", "6")))
Top$Genus <- as.ordered(factor(Top$Genus, levels = Top15levels))
## Add numbers
df.1 <- subset(Top, PD_Colon == "P" & Transit == "ST"); df.1$TT <- 8; df.1$TT1 <- 8
df.2 <- subset(Top, PD_Colon == "P" & Transit == "MT"); df.2$TT <- 16; df.2$TT1 <- 16
df.3 <- subset(Top, PD_Colon == "P" & Transit == "LT"); df.3$TT <- 24; df.3$TT1 <- 24
df.4 <- subset(Top, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 13; df.4$TT1 <- 8
df.5 <- subset(Top, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 26; df.5$TT1 <- 16
df.6 <- subset(Top, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 39; df.6$TT1 <- 24
# df.4 <- subset(Top, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 8
# df.5 <- subset(Top, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 16
# df.6 <- subset(Top, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 24

Top <- rbind(df.1, df.2, df.3, df.4, df.5, df.6)
rm(df.1, df.2, df.3, df.4, df.5, df.6)
## Perform test
dfcor <- data.frame(); df.otutest <- data.frame(); QMPlist <- list(); y =1
for(i in na.omit(unique(Top$Genus))){
  df1 <- subset(Top, Genus == i);  
  if(abovezero == T){
    df1.1 <- subset(df1, Donor == 1); df1.1 <- subset(df1.1, Abundance > 0)
    if (dim(df1.1)[1] == 0) {df1 <- subset(df1, Donor !=1)}
    df1.2 <- subset(df1, Donor == 2); df1.2 <- subset(df1.2, Abundance > 0) 
    if (dim(df1.2)[1] == 0) {df1 <- subset(df1, Donor !=2)}
    df1.3 <- subset(df1, Donor == 3); df1.3 <- subset(df1.3, Abundance > 0)
    if (dim(df1.3)[1] == 0) {df1 <- subset(df1, Donor !=3)}
    df1.4 <- subset(df1, Donor == 4); df1.4 <- subset(df1.4, Abundance > 0) 
    if (dim(df1.4)[1] == 0) {df1 <- subset(df1, Donor !=4)}
    df1.5 <- subset(df1, Donor == 5); df1.5 <- subset(df1.5, Abundance > 0) 
    if (dim(df1.5)[1] == 0) {df1 <- subset(df1, Donor !=5)}
    df1.6 <- subset(df1, Donor == 6); df1.6 <- subset(df1.6, Abundance > 0) 
    if (dim(df1.6)[1] == 0) {df1 <- subset(df1, Donor !=6)}
    }
  for(x in c("P", "D")){
  df<- subset(df1, PD_Colon == x)
  t <- cor(df$Abundance, df$TT, method = "spearman")
  corr <- cor.test(df$Abundance, df$TT, method = 'spearman'); 
  if(is.na(corr[["p.value"]])){
    sig <- "-"
    }else{
      if(corr[["p.value"]]<0.0001){
        sig <- "****"
        }else{
          if(corr[["p.value"]]<0.001){
            sig <- "***"
            }else{
              if(corr[["p.value"]]<0.01){
                sig <- "**"
                }else{
                  if(corr[["p.value"]]<0.05){
                    sig<-"*"
                  }else{
                      sig<-""}}}}}
  
  dfcor <- rbind(dfcor, data.frame(genus = i, cor = corr[["estimate"]][["rho"]], 
             PD_Colon = x, pvalue = corr[["p.value"]], sig = sig, comm = "Quantitative microbial community"))
  
  if(x == "P"){corP <- paste0("Ï = ",round(t, digits = 2)," ", sig, sep = ""); corrP <- corr[["p.value"]]
  }else{corD <- paste0("Ï = ",round(t, digits = 2)," ", sig, sep = ""); corrD <- corr[["p.value"]]}
  }
  
plot <- papertheme(ggplot(df1, aes(y = Abundance, x = TT1))) + geom_boxplot(aes(fill = Transit)) + geom_jitter(aes(color = Donor), size = 4, height = 0) + 
    facet_grid(rows = vars(PD_Colon), labeller = labeller(PD_Colon = PD_Colon.labs), scales = "free_x") 

  if(is.na(corrP)){corrP <- 1} 
  if(is.na(corrD)){corrD <- 1} 
  if (corrP > 0.05 & corrD > 0.05) {
    plot <- plot
    dat_text <- data.frame(label = c("", ""), 
                           PD_Colon = c("P", "D"), 
                           value = c(max(df1$Abundance), max(df1$Abundance)),
                           xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                           hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
                           TT1 = c(x_lab, x_lab))
} else {
    if (corrP > 0.05 & corrD <= 0.05 ) {
        plot <- plot + geom_smooth(data = subset(df1, PD_Colon =="D"), aes(y = Abundance, x = TT1), method = lm,
                                   color = "black")
        dat_text <- data.frame(label = c("", corD), PD_Colon = c("P","D"), 
                               value = c(max(df1$Abundance), max(df1$Abundance)),
                               xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                               hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
                               TT1 = c(x_lab, x_lab))
    } else {
        if (corrP <= 0.05 & corrD > 0.05) {
            plot <- plot + geom_smooth(data = subset(df1,  PD_Colon == "P"), aes(y = Abundance, x = TT1),
                method = lm, color = "black")
            dat_text <- data.frame(label = c(corP, ""),
                                   PD_Colon = c("P", "D"), 
                                   value = c(max(df1$Abundance),max(df1$Abundance)),
                                   xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                                   hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
                                   TT1 = c(x_lab, x_lab))
        } else {
            plot <- plot + geom_smooth(aes(y = Abundance, x = TT1), method = lm, color = "black")
            dat_text <- data.frame(label = c(corP, corD),
                                   PD_Colon = c("P", "D"), 
                                   value = c(max(df1$Abundance),max(df1$Abundance)),
                                   xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                                   hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
                                   TT1 = c(x_lab, x_lab))
        }
    }
}
  dat_text$PD_Colon <- as.ordered(factor(dat_text$PD_Colon, levels = c("P", "D")))
  
 plot <- plot + scale_color_manual(values = sixdonors) + scale_fill_manual("SHIME transit", values = transitcolors, labels= c("Short", "Medium", "Long")) +
  scale_x_continuous(breaks=c(8,16,24), labels=c("ST", "MT", "LT")) +
   # scale_x_discrete(labels = c("Short", "Medium", "Long"))+
   ylab(expression(paste("Absolute abundance (cells ",mL^{"-1"},")")))+ xlab("Transit time") + 
   scale_y_continuous(labels = scales::scientific) 
 
 if(i == "Acidaminococcus"){plot <- plot + 
   guides(fill=guide_legend(ncol=1, title.position = "top", hjust = 0.5),
          color=guide_legend(nrow=3, title.position = "top", hjust = 0.5)) +
   theme(legend.title.align = 0.5, legend.title = element_text(size=20), 
         legend.text = element_text(size=20), legend.position = c(0.5,0.35), legend.box = "horizontal")}else{
   plot <- plot + theme(legend.position = "none")}
         plot <- plot + theme(strip.background =element_rect(fill="#D6E0F4"),
         # legend.box = "horizontal",
         strip.text.x = element_text(size = 20),
         strip.text.y = element_blank(),
         # strip.text.y = element_text(size = 20, angle = 270),
         legend.title = element_text(size=20), legend.background = element_blank()) 
  
 
 
if(i == "X"){plot <- plot + geom_text(correlations)}else{
  # plot <- plot + geom_text(data= dat_text,  mapping = aes(x = as.numeric(TT1), y = as.numeric(value), label = label, family = 'sans'), size = 7)
  plot <- plot + geom_text(data= dat_text,  mapping = aes(x = xpos, y = ypos, label = label, 
                                        hjust=hjustvar,vjust=vjustvar, family = 'sans'), size = 7)
  }
QMPlist[[i]] <- plot; y = y+1
  }



# Proportional
phylobj.prop_2 = transform_sample_counts(phylobj.QMP, function(x) 100 * x/sum(x))
phylobj.prop_stable <- subset_samples(phylobj.prop_2, Ninetransit == 3)
Top15phylobj.prop <- aggregate_taxa(phylobj.prop_stable, level = "Genus")
Top15.prop <- psmelt(Top15phylobj.prop); unique(Top15.prop$OTU); unique(Top15.prop$OTU) %in% Top15levels
## Subset according to top most abundant genera
df.test <- data.frame();
for(i in Top.genera){
  df.test2 <- subset(Top15.prop, Genus == i)
  df.test <- rbind(df.test, df.test2)
}
Top15.prop <- df.test
x <- data.frame(x = unique(Top15$OTU) %in% Top15levels, genus = unique(Top15$OTU))  #check if labelled
Top15$OTU <- as.ordered(factor(Top15$OTU, levels = Top15levels))
x <- data.frame(x = unique(Top15.prop$OTU) %in% Top15levels, genus = unique(Top15.prop$OTU))  #check if labelled
Top15.prop$OTU <- as.ordered(factor(Top15.prop$OTU, levels = Top15levels))
## Add extra column days for graphs
Top.prop <- Top15.prop
Top.prop$Transit <- as.ordered(factor(Top.prop$Transit, levels = c("ST", "MT", "LT")))
Top.prop$PD_Colon <- as.ordered(factor(Top.prop$PD_Colon, levels = c("P", "D", "FS")))
Top.prop$Donor <- as.ordered(factor(Top.prop$Donor, levels = c("1", "2", "3", "4", "5", "6")))
Top.prop$Genus <- as.ordered(factor(Top.prop$Genus, levels = Top15levels))
## Add numbers
df.1 <- subset(Top.prop, PD_Colon == "P" & Transit == "ST"); df.1$TT <- 8; df.1$TT1 <- 8
df.2 <- subset(Top.prop, PD_Colon == "P" & Transit == "MT"); df.2$TT <- 16; df.2$TT1 <- 16
df.3 <- subset(Top.prop, PD_Colon == "P" & Transit == "LT"); df.3$TT <- 24; df.3$TT1 <- 24
df.4 <- subset(Top.prop, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 13; df.4$TT1 <- 8
df.5 <- subset(Top.prop, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 26; df.5$TT1 <- 16
df.6 <- subset(Top.prop, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 39; df.6$TT1 <- 24
# df.4 <- subset(Top.prop, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 8
# df.5 <- subset(Top.prop, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 16
# df.6 <- subset(Top.prop, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 24
Top.prop <- rbind(df.1, df.2, df.3, df.4, df.5, df.6)
rm(df.1, df.2, df.3, df.4, df.5, df.6)
dfcor2 <- data.frame(); PMPlist <- list(); y = 1

for(i in na.omit(unique(Top.prop$Genus))){
  df1 <- subset(Top.prop, Genus == i)
  if(abovezero == T){
    df1.1 <- subset(df1, Donor == 1); df1.1 <- subset(df1.1, Abundance > 0)
    if (dim(df1.1)[1] == 0) {df1 <- subset(df1, Donor !=1)}
    df1.2 <- subset(df1, Donor == 2); df1.2 <- subset(df1.2, Abundance > 0) 
    if (dim(df1.2)[1] == 0) {df1 <- subset(df1, Donor !=2)}
    df1.3 <- subset(df1, Donor == 3); df1.3 <- subset(df1.3, Abundance > 0)
    if (dim(df1.3)[1] == 0) {df1 <- subset(df1, Donor !=3)}
    df1.4 <- subset(df1, Donor == 4); df1.4 <- subset(df1.4, Abundance > 0) 
    if (dim(df1.4)[1] == 0) {df1 <- subset(df1, Donor !=4)}
    df1.5 <- subset(df1, Donor == 5); df1.5 <- subset(df1.5, Abundance > 0) 
    if (dim(df1.5)[1] == 0) {df1 <- subset(df1, Donor !=5)}
    df1.6 <- subset(df1, Donor == 6); df1.6 <- subset(df1.6, Abundance > 0) 
    if (dim(df1.6)[1] == 0) {df1 <- subset(df1, Donor !=6)}
    }
  for(x in c("P", "D")){
  df<- subset(df1, PD_Colon == x)
  t <- cor(df$Abundance, df$TT, method = "spearman")
  corr <- cor.test(df$Abundance, df$TT, method = 'spearman'); 
  
  if(is.na(corr[["p.value"]])){
    sig <- "-"
    }else{
      if(corr[["p.value"]]<0.0001){
        sig <- "****"
        }else{
          if(corr[["p.value"]]<0.001){
            sig <- "***"
            }else{
              if(corr[["p.value"]]<0.01){
                sig <- "**"
                }else{
                  if(corr[["p.value"]]<0.05){
                    sig<-"*"
                  }else{
                      sig<-""}}}}}
  
  dfcor2 <- rbind(dfcor2, data.frame(genus = i, cor = corr[["estimate"]][["rho"]], 
                 PD_Colon = x, pvalue = corr[["p.value"]], sig = sig, comm = "Proportional microbial community"))
  
  if(x == "P"){corP <- paste0("Ï = ",round(t, digits = 2)," ", sig, sep = ""); corrP <- corr[["p.value"]]}
  if(x == "D"){corD <- paste0("Ï = ",round(t, digits = 2)," ", sig, sep = ""); corrD <- corr[["p.value"]]}
  }
  
  #   dat_text <- data.frame(label = c(corP, corD),
  # PD_Colon   = c("P","D"),
  # value   = c(max(df1$Abundance), max(df1$Abundance)), 
  # TT = c(8,8))
  # dat_text$PD_Colon <- as.ordered(factor(dat_text$PD_Colon, levels = c("P", "D")))
  
plot <- papertheme(ggplot(df1, aes(y = Abundance, x = TT1))) + geom_boxplot(aes(fill = Transit)) + geom_jitter(aes(color = Donor), size =4) + 
      facet_grid(rows = vars(PD_Colon), labeller = labeller(PD_Colon = PD_Colon.labs), scales = "free_x") 

  if(is.na(corrP)){corrP <- 1} 
  if(is.na(corrD)){corrD <- 1} 
  if (corrP > 0.05 & corrD > 0.05) {
    plot <- plot
    dat_text <- data.frame(label = c("", ""), PD_Colon = c("P",
        "D"), value = c(max(df1$Abundance), max(df1$Abundance)),
        xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
        hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
        TT1 = c(x_lab, x_lab))
} else {
    if (corrP > 0.05 & corrD <= 0.05) {
        plot <- plot + geom_smooth(data = subset(df1, PD_Colon =="D"), 
                                   aes(y = Abundance, x = TT1), method = lm,
                                   color = "black")
        dat_text <- data.frame(label = c("", corD), PD_Colon = c("P",
            "D"), value = c(max(df1$Abundance), max(df1$Abundance)),
            xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
            hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
            TT1 = c(x_lab, x_lab))
    } else {
        if (corrP <= 0.05 & corrD > 0.05) {
            plot <- plot + geom_smooth(data = subset(df1,
                PD_Colon == "P"), aes(y = Abundance, x = TT1),
                method = lm, color = "black")
            dat_text <- data.frame(label = c(corP, ""),
                PD_Colon = c("P", "D"), value = c(max(df1$Abundance),
                  max(df1$Abundance)), 
                xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),TT1 = c(x_lab, x_lab))
        } else {
            plot <- plot + geom_smooth(aes(y = Abundance,
                x = TT1), method = lm, color = "black")
            dat_text <- data.frame(label = c(corP, corD),
                PD_Colon = c("P", "D"), value = c(max(df1$Abundance),
                  max(df1$Abundance)), 
                xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
                TT1 = c(x_lab, x_lab))
        }
    }
}
  dat_text$PD_Colon <- as.ordered(factor(dat_text$PD_Colon, levels = c("P", "D")))

plot <- plot + scale_color_manual(values = sixdonors) + scale_fill_manual(values = transitcolors) +
  scale_x_continuous(breaks=c(8,16,24), labels=c("ST", "MT", "LT")) +
  ylab("Proportional abundance (%)")+ xlab("Transit time") + 
  # scale_x_discrete(labels = c("Short", "Medium", "Long"))+
    theme(strip.background =element_rect(fill="#D6E0F4"),
          legend.position = "none",
          legend.box = "horizontal",
          strip.text.x = element_text(size = 20),
          strip.text.y = element_blank(), 
          # strip.text.y = element_text(size = 20, angle = 270),
          legend.title = element_text(size=20), legend.background = element_blank()) 
# if(i == "Veillonella"){
#     plot <- plot + theme(strip.text.y = element_text(size = 20, angle = 270))
#   }
  
if(i == "X"){plot <- plot + geom_text(correlations)}else{
  # plot <- plot + geom_text(data= dat_text,  mapping = aes(x = as.numeric(TT1), y = as.numeric(value),
  #                                                         label = label, family = 'sans'), size = 7)
  plot <- plot + geom_text(data= dat_text,  mapping = aes(x = xpos, y = ypos, label = label, 
                                        hjust=hjustvar,vjust=vjustvar, family = 'sans'), size = 7)
}
PMPlist[[i]] <- plot; y = y+1
    }

#Combine abs and prop abundances in one
dfcor <- rbind(dfcor, dfcor2)
dfcor <- na.omit(dfcor); dfcor$comm <- as.ordered(factor(dfcor$comm, 
                              levels = c("Quantitative microbial community", "Proportional microbial community")))
dfcor$PD_Colon <- as.ordered(factor(dfcor$PD_Colon, levels = c("P", "D")))
dfcor$genus <- as.ordered(factor(dfcor$genus, levels = Top.genera))
plot <- papertheme(ggplot(dfcor, aes(x = PD_Colon, y = genus, fill =cor))) + geom_tile() +
  scale_fill_gradient2(na.value = "grey50",low="blue", high="red") + 
  geom_text(aes(label = sig)) +
  theme(axis.text.y = element_text(face="italic")) + ylab("Genus") + xlab("Colon vessel") + 
  scale_x_discrete(breaks=c("P","D"), labels=c("Proximal colon", "Distal colon")) +
  labs(fill = expression(rho)) +
  facet_grid(cols = vars(comm)) +theme(strip.background = element_rect(fill = "#D6E0F4"))
plot
dfcorall <- dfcor


# Combine plots
plots <- list(); x = 1
genera <- na.omit(unique(Top.prop$Genus))
genera <- sort(genera, decreasing = F)
for(i in genera){ 
  plot1 <- QMPlist[[i]]; plot2 <- PMPlist[[i]]
  plot <- ggarrange(plot1, plot2)
  plot <- annotate_figure(plot, text_grob(i, color = "black", face = "bold.italic", size = 25))
  # print(plot)
  plots[[i]] <- plot
  x = x +1
}

plots1 <- plots[1:8]; plots2 <- plots[9:16]
# ggarrange(plotlist = plots1, nrow = 2, ncol = 4)
# ggarrange(plotlist = plots2, nrow = 2, ncol = 4)
# ggarrange(plotlist = plots, nrow = 4, ncol = 4)








# OTU LEVEL
# Create OTU list with the genera
Top.OTU <- data.frame(); df.OTU <-df.combined
for(i in Top.genera){
  df.test <- subset(df.OTU, genus == i)
  Top.OTU <- rbind(Top.OTU, df.test)
}
Top.OTU <- unique(Top.OTU$rowname)

# ABSOLUTE ABUNDANCE
Top15.otu <- psmelt(subset_samples(phylobj.QMP.otu, Ninetransit == 3)); unique(Top15.otu$OTU)
## Subset according to top most abundant
df.test <- data.frame()
for(i in Top.OTU){
  df.test2 <- subset(Top15.otu, OTU == i)
  df.test <- rbind(df.test, df.test2)
}
Top15.otu <- df.test
unique(Top15.otu$OTU)
## Add extra column days for graphs
Top.otu <- Top15.otu
Top.otu$Transit <- as.ordered(factor(Top.otu$Transit, levels = c("ST", "MT", "LT")))
Top.otu$PD_Colon <- as.ordered(factor(Top.otu$PD_Colon, levels = c("P", "D", "FS")))
Top.otu$Donor <- as.ordered(factor(Top.otu$Donor, levels = c("1", "2", "3", "4", "5", "6")))

## Add numbers
df.1 <- subset(Top.otu, PD_Colon == "P" & Transit == "ST"); df.1$TT <- 8; df.1$TT1 <- 8
df.2 <- subset(Top.otu, PD_Colon == "P" & Transit == "MT"); df.2$TT <- 16; df.2$TT1 <- 16
df.3 <- subset(Top.otu, PD_Colon == "P" & Transit == "LT"); df.3$TT <- 24; df.3$TT1 <- 24
df.4 <- subset(Top.otu, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 13; df.4$TT1 <- 8
df.5 <- subset(Top.otu, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 26; df.5$TT1 <- 16
df.6 <- subset(Top.otu, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 39; df.6$TT1 <- 24
Top.otu <- rbind(df.1, df.2, df.3, df.4, df.5, df.6)
rm(df.1, df.2, df.3, df.4, df.5, df.6)


## Perform tests
dfcor.otu <- data.frame(); OTUQMPlist <- list()
for(i in na.omit(unique(Top.otu$OTU))){
  df1 <- subset(Top.otu, OTU == i)
  if(abovezero == T){
    df1.1 <- subset(df1, Donor == 1); df1.1 <- subset(df1.1, Abundance > 0)
    if (dim(df1.1)[1] == 0) {df1 <- subset(df1, Donor !=1)}
    df1.2 <- subset(df1, Donor == 2); df1.2 <- subset(df1.2, Abundance > 0) 
    if (dim(df1.2)[1] == 0) {df1 <- subset(df1, Donor !=2)}
    df1.3 <- subset(df1, Donor == 3); df1.3 <- subset(df1.3, Abundance > 0)
    if (dim(df1.3)[1] == 0) {df1 <- subset(df1, Donor !=3)}
    df1.4 <- subset(df1, Donor == 4); df1.4 <- subset(df1.4, Abundance > 0) 
    if (dim(df1.4)[1] == 0) {df1 <- subset(df1, Donor !=4)}
    df1.5 <- subset(df1, Donor == 5); df1.5 <- subset(df1.5, Abundance > 0) 
    if (dim(df1.5)[1] == 0) {df1 <- subset(df1, Donor !=5)}
    df1.6 <- subset(df1, Donor == 6); df1.6 <- subset(df1.6, Abundance > 0) 
    if (dim(df1.6)[1] == 0) {df1 <- subset(df1, Donor !=6)}
    }
  for(x in c("P", "D")){
  df<- subset(df1, PD_Colon == x)
  t <- cor(df$Abundance, df$TT, method = "spearman")
  
  if(is.na(t)){corr[["p.value"]] <- NA}else{
   corr <- cor.test(df$Abundance, df$TT, method = 'spearman')  
  }
  
  
  if(is.na(corr[["p.value"]])){
    sig <- ""
    }else{
      if(corr[["p.value"]]<0.0001){
        sig <- "****"
        }else{
          if(corr[["p.value"]]<0.001){
            sig <- "***"
            }else{
              if(corr[["p.value"]]<0.01){
                sig <- "**"
                }else{
                  if(corr[["p.value"]]<0.05){
                    sig<-"*"
                  }else{
                      sig<-""}}}}}
  
  if(is.na(t)){dfcor.otu <- rbind(dfcor.otu, data.frame(genus = i, cor = NA, 
                 PD_Colon = x, pvalue = corr[["p.value"]], sig = sig, comm = "Quantitative microbial community"))
  }else{
    dfcor.otu <- rbind(dfcor.otu, data.frame(genus = i, cor = corr[["estimate"]][["rho"]], 
                 PD_Colon = x, pvalue = corr[["p.value"]], sig = sig, comm = "Quantitative microbial community"))}
  
  if(x == "P"){corP <- paste0("Ï = ",round(t, digits = 2)," ", sig, sep = ""); corrP <- corr[["p.value"]]
  }else{corD <- paste0("Ï = ",round(t, digits = 2)," ", sig, sep = ""); corrD <- corr[["p.value"]]}
  }
  
  plot <- papertheme(ggplot(df1, aes(y = Abundance, x = TT1))) + geom_boxplot(aes(fill = Transit)) + geom_jitter(aes(color = Donor), size =4) + 
      facet_grid(rows = vars(PD_Colon), labeller = labeller(PD_Colon = PD_Colon.labs), scales = "free_x") 
  
  if(is.na(corrP)){corrP <- 1} 
  if(is.na(corrD)){corrD <- 1} 
  if (corrP > 0.05 & corrD > 0.05) {
    plot <- plot
    dat_text <- data.frame(label = c("", ""), PD_Colon = c("P",
        "D"), value = c(max(df1$Abundance), max(df1$Abundance)),
        xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
        hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
        TT1 = c(x_lab, x_lab))
} else {
    if (corrP > 0.05 & corrD <= 0.05) {
        plot <- plot + geom_smooth(data = subset(df1, PD_Colon =="D"), 
                                   aes(y = Abundance, x = TT1), method = lm,
                                   color = "black")
        dat_text <- data.frame(label = c("", corD), PD_Colon = c("P",
            "D"), value = c(max(df1$Abundance), max(df1$Abundance)),
            xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
            hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
            TT1 = c(x_lab, x_lab))
    } else {
        if (corrP <= 0.05 & corrD > 0.05) {
            plot <- plot + geom_smooth(data = subset(df1,
                PD_Colon == "P"), aes(y = Abundance, x = TT1),
                method = lm, color = "black")
            dat_text <- data.frame(label = c(corP, ""),
                PD_Colon = c("P", "D"), value = c(max(df1$Abundance),
                  max(df1$Abundance)),
                xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),TT1 = c(x_lab, x_lab))
        } else {
            plot <- plot + geom_smooth(aes(y = Abundance,
                x = TT1), method = lm, color = "black")
            dat_text <- data.frame(label = c(corP, corD),
                PD_Colon = c("P", "D"), value = c(max(df1$Abundance),
                  max(df1$Abundance)), 
                xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
                TT1 = c(x_lab, x_lab))
        }
    }
}
  dat_text$PD_Colon <- as.ordered(factor(dat_text$PD_Colon, levels = c("P", "D")))

plot <- plot + scale_color_manual(values = sixdonors) + 
  scale_fill_manual("SHIME transit", values = transitcolors, labels = c("Short", "Medium", "Long")) +
  # scale_x_discrete(labels = c("Short", "Medium", "Long"))+
  scale_x_continuous(breaks=c(8,16,24), labels=c("Short", "Medium", "Long")) +
  ylab(expression(paste("Absolute abundance (cells ",mL^{"-1"},")")))+ xlab("Transit time") + 
  scale_y_continuous(labels = scales::scientific) 
if(i == "Otu00119" | i == "Otu00062"){plot <- plot + 
   guides(fill=guide_legend(ncol=1, title.position = "top", hjust = 0.5),
          color=guide_legend(nrow=3, title.position = "top", hjust = 0.5)) +
   theme(legend.title.align = 0.5, legend.title = element_text(size=20), 
         legend.text = element_text(size=20), legend.position = c(0.5,0.85), legend.box = "horizontal")}else{
           if(i == "Otu00042"){plot <- plot + 
   guides(fill=guide_legend(ncol=1, title.position = "top", hjust = 0.5),
          color=guide_legend(nrow=3, title.position = "top", hjust = 0.5)) +
   theme(legend.title.align = 0.5, legend.title = element_text(size=20), 
         legend.text = element_text(size=20), legend.position = c(0.5,0.30), legend.box = "horizontal")}else{
   plot <- plot + theme(legend.position = "none")}}
         plot <- plot + theme(strip.background =element_rect(fill="#D6E0F4"),
         # legend.box = "horizontal",
         strip.text.x = element_text(size = 20),
         strip.text.y = element_blank(),
         # strip.text.y = element_text(size = 20, angle = 270),
         legend.title = element_text(size=20), legend.background = element_blank())
         
    


  if(i == "X"){plot <- plot + geom_text(correlations)}else{
  # plot <- plot + geom_text(data= dat_text,  mapping = aes(x = as.numeric(TT1), y = as.numeric(value), label = label, family = 'sans'), size = 7)
  plot <- plot + geom_text(data= dat_text,  mapping = aes(x = xpos, y = ypos, label = label, 
                                        hjust=hjustvar,vjust=vjustvar, family = 'sans'), size = 7)
}
 OTUQMPlist[[i]] <- plot; y = y+1
  
  }


# Proportional
Top15.prop.otu <- psmelt(subset_samples(phylobj.prop.otu, Ninetransit == 3)); unique(Top15.prop.otu$OTU)
## Subset according to top most abundant
df.test <- data.frame()
for(i in Top.OTU){
  df.test2 <- subset(Top15.prop.otu, OTU == i)
  df.test <- rbind(df.test, df.test2)
}
Top15.prop.otu <- df.test
unique(Top15.prop.otu$OTU)
## Add extra column days for graphs
Top.prop.otu <- Top15.prop.otu
Top.prop.otu$Transit <- as.ordered(factor(Top.prop.otu$Transit, levels = c("ST", "MT", "LT")))
Top.prop.otu$PD_Colon <- as.ordered(factor(Top.prop.otu$PD_Colon, levels = c("P", "D", "FS")))
Top.prop.otu$Donor <- as.ordered(factor(Top.prop.otu$Donor, levels = c("1", "2", "3", "4", "5", "6")))

## Add numbers
df.1 <- subset(Top.prop.otu, PD_Colon == "P" & Transit == "ST"); df.1$TT <- 8; df.1$TT1 <- 8
df.2 <- subset(Top.prop.otu, PD_Colon == "P" & Transit == "MT"); df.2$TT <- 16; df.2$TT1 <- 16
df.3 <- subset(Top.prop.otu, PD_Colon == "P" & Transit == "LT"); df.3$TT <- 24; df.3$TT1 <- 24
df.4 <- subset(Top.prop.otu, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 13; df.4$TT1 <- 8
df.5 <- subset(Top.prop.otu, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 26; df.5$TT1 <- 16
df.6 <- subset(Top.prop.otu, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 39; df.6$TT1 <- 24
Top.prop.otu <- rbind(df.1, df.2, df.3, df.4, df.5, df.6)
rm(df.1, df.2, df.3, df.4, df.5, df.6)


## Perform tests
dfcor.otu <- data.frame(); OTUPMPlist <- list()
for(i in na.omit(unique(Top.prop.otu$OTU))){
  df1 <- subset(Top.prop.otu, OTU == i)
  if(abovezero == T){
    df1.1 <- subset(df1, Donor == 1); df1.1 <- subset(df1.1, Abundance > 0)
    if (dim(df1.1)[1] == 0) {df1 <- subset(df1, Donor !=1)}
    df1.2 <- subset(df1, Donor == 2); df1.2 <- subset(df1.2, Abundance > 0) 
    if (dim(df1.2)[1] == 0) {df1 <- subset(df1, Donor !=2)}
    df1.3 <- subset(df1, Donor == 3); df1.3 <- subset(df1.3, Abundance > 0)
    if (dim(df1.3)[1] == 0) {df1 <- subset(df1, Donor !=3)}
    df1.4 <- subset(df1, Donor == 4); df1.4 <- subset(df1.4, Abundance > 0) 
    if (dim(df1.4)[1] == 0) {df1 <- subset(df1, Donor !=4)}
    df1.5 <- subset(df1, Donor == 5); df1.5 <- subset(df1.5, Abundance > 0) 
    if (dim(df1.5)[1] == 0) {df1 <- subset(df1, Donor !=5)}
    df1.6 <- subset(df1, Donor == 6); df1.6 <- subset(df1.6, Abundance > 0) 
    if (dim(df1.6)[1] == 0) {df1 <- subset(df1, Donor !=6)}
    }
  for(x in c("P", "D")){
  df<- subset(df1, PD_Colon == x)
  t <- cor(df$Abundance, df$TT, method = "spearman")
  
  if(is.na(t)){corr[["p.value"]] <- NA}else{
   corr <- cor.test(df$Abundance, df$TT, method = 'spearman')  
  }
  
  
  if(is.na(corr[["p.value"]])){
    sig <- ""
    }else{
      if(corr[["p.value"]]<0.0001){
        sig <- "****"
        }else{
          if(corr[["p.value"]]<0.001){
            sig <- "***"
            }else{
              if(corr[["p.value"]]<0.01){
                sig <- "**"
                }else{
                  if(corr[["p.value"]]<0.05){
                    sig<-"*"
                  }else{
                      sig<-""}}}}}
  
  if(is.na(t)){dfcor.otu <- rbind(dfcor.otu, data.frame(genus = i, cor = NA, 
                 PD_Colon = x, pvalue = corr[["p.value"]], sig = sig, comm = "Quantitative microbial community"))
  }else{
    dfcor.otu <- rbind(dfcor.otu, data.frame(genus = i, cor = corr[["estimate"]][["rho"]], 
                 PD_Colon = x, pvalue = corr[["p.value"]], sig = sig, comm = "Quantitative microbial community"))}
  
  if(x == "P"){corP <- paste0("Ï = ",round(t, digits = 2)," ", sig, sep = ""); corrP <- corr[["p.value"]]
  }else{corD <- paste0("Ï = ",round(t, digits = 2)," ", sig, sep = ""); corrD <- corr[["p.value"]]}
  }
  
  plot <- papertheme(ggplot(df1, aes(y = Abundance, x = TT1))) + geom_boxplot(aes(fill = Transit)) + geom_jitter(aes(color = Donor), size =4) + 
      facet_grid(rows = vars(PD_Colon), labeller = labeller(PD_Colon = PD_Colon.labs), scales = "free_x") 

  
  
  if(is.na(corrP)){corrP <- 1} 
  if(is.na(corrD)){corrD <- 1} 
  if (corrP > 0.05 & corrD > 0.05) {
    plot <- plot
    dat_text <- data.frame(label = c("", ""), PD_Colon = c("P",
        "D"), value = c(max(df1$Abundance), max(df1$Abundance)),
        xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
        TT1 = c(x_lab, x_lab))
} else {
    if (corrP > 0.05 & corrD <= 0.05) {
        plot <- plot + geom_smooth(data = subset(df1, PD_Colon =="D"), 
                                   aes(y = Abundance, x = TT1), method = lm,
                                   color = "black")
        dat_text <- data.frame(label = c("", corD), PD_Colon = c("P",
            "D"), value = c(max(df1$Abundance), max(df1$Abundance)),
            xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
            TT1 = c(x_lab, x_lab))
    } else {
        if (corrP <= 0.05 & corrD > 0.05) {
            plot <- plot + geom_smooth(data = subset(df1,
                PD_Colon == "P"), aes(y = Abundance, x = TT1),
                method = lm, color = "black")
            dat_text <- data.frame(label = c(corP, ""),
                PD_Colon = c("P", "D"), value = c(max(df1$Abundance),
                  max(df1$Abundance)), 
                xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),TT1 = c(x_lab, x_lab))
        } else {
            plot <- plot + geom_smooth(aes(y = Abundance,
                x = TT1), method = lm, color = "black")
            dat_text <- data.frame(label = c(corP, corD),
                PD_Colon = c("P", "D"), value = c(max(df1$Abundance),
                  max(df1$Abundance)), 
                xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),TT1 = c(x_lab, x_lab))
        }
    }
}
  dat_text$PD_Colon <- as.ordered(factor(dat_text$PD_Colon, levels = c("P", "D")))

plot <- plot + scale_color_manual(values = sixdonors) + 
  scale_fill_manual("SHIME transit", values = transitcolors, labels = c("Short", "Medium", "Long")) + 
  guides(fill=guide_legend(ncol=1, title.position = "top", hjust = 0.5), 
         color=guide_legend(nrow=3, title.position = "top", hjust = 0.5)) +
  # scale_x_discrete(labels = c("Short", "Medium", "Long"))+ 
  scale_x_continuous(breaks=c(8,16,24), labels=c("Short", "Medium", "Long")) +
  theme(legend.title.align = 0.5, legend.title = element_text(size=25), legend.text = element_text(size=25))+
  ylab("Proportional abundance (%)")+ xlab("Transit time") 
  
plt_legend <- g_legend(plot)

if(i == "Otu00144"){plot <- plot + 
   guides(fill=guide_legend(ncol=1, title.position = "top", hjust = 0.5),
          color=guide_legend(nrow=3, title.position = "top", hjust = 0.5)) +
   theme(legend.title.align = 0.5, legend.title = element_text(size=20), 
         legend.text = element_text(size=20), legend.position = c(0.5,0.30), legend.box = "horizontal")}else{
   plot <- plot + theme(legend.position = "none")}
         plot <- plot + theme(strip.background =element_rect(fill="#D6E0F4"),
         # legend.box = "horizontal",
         strip.text.x = element_text(size = 20),
         strip.text.y = element_blank(),
         # strip.text.y = element_text(size = 20, angle = 270),
         legend.title = element_text(size=20), legend.background = element_blank())
         
  if(i == "Otu00005" | i == "Otu00236" | i == "Otu00072"| i == "Otu00188" |  i == "Otu00198"){
    plot <- plot + theme(strip.text.y = element_text(size = 20, angle = 270))
  }




  if(i == "X"){plot <- plot + geom_text(correlations)}else{
  # plot <- plot + geom_text(data= dat_text,  mapping = aes(x = as.numeric(TT1), y = as.numeric(value), label = label, family = 'sans'), size = 7)
  plot <- plot + geom_text(data= dat_text,  mapping = aes(x = xpos, y = ypos, label = label, 
                                        hjust=hjustvar,vjust=vjustvar, family = 'sans'), size = 7)
}
 OTUPMPlist[[i]] <- plot; y = y+1
  
  }







# Combine OTUplots
otuplots <- list(); x = 1
genera <- na.omit(unique(Top.otu$OTU))
genera <- sort(genera, decreasing = F)
for(i in genera){ 
  plot1 <- OTUQMPlist[[i]]; plot2 <- OTUPMPlist[[i]]
  plot <- ggarrange(plot1, plot2)
  # print(plot)
  
  m =i
  if(stringr::str_detect(m, "Otu0000")){
      y <- str_replace(m, "Otu0000", "OTU")
    }else{if(stringr::str_detect(m, "Otu000")){
      y <- str_replace(m, "Otu000", "OTU")
    }else{if(stringr::str_detect(m, "Otu00")){
      y <- str_replace(m, "Otu00", "OTU")
    }else{if(stringr::str_detect(m, "Otu0")){
      y <- str_replace(m, "Otu0", "OTU")
    }}}}
  
  plot <- annotate_figure(plot, text_grob(y, color = "black", face = "bold", size = 25))
  # print(plot)
  otuplots[[i]] <- plot
  x = x +1
}

ggarrange(plotlist = otuplots)



# Combine genera and OTU's
allplotslist <- list()
for(i in sort(Top.genera, decreasing = F)){
  plot1 <- plots[[i]]
dfje <- subset(df.combined, genus == i)
dfje <- subset(dfje, rowname!="Otu00102")
dfje <- subset(dfje, rowname!="Otu00216")
# dfje <- subset(dfje, rowname!="Otu00261")
# dfje <- subset(dfje, rowname!="Otu00078")
# dfje <- subset(dfje, rowname!="Otu00133")
# dfje <- subset(dfje, rowname!="Otu00214")
# dfje <- subset(dfje, mean > 5)
plotjes <- list()
plotjes[[1]] <- plot1; y = 2
for(x in unique(dfje$rowname)){
  plot2 <- otuplots[[x]]
    plotjes[[y]] <- plot2
    y = y+1
}
if(i == "Bifidobacterium"){ plotjes[[y]] <- ggarrange(plt_legend, NULL ,nrow = 2)}


allplotslist[[i]] <- ggarrange(plotlist = plotjes, 
                               nrow = ceiling((nrow(dfje)+1)/4), 
                               ncol = 4)
}

ggarrange(allplotslist[[1]] + ggpubr::border(color = "grey", size = 1),
          allplotslist[[2]] + ggpubr::border(color = "grey", size = 1), 
          allplotslist[[3]] + ggpubr::border(color = "grey", size = 1),
          allplotslist[[4]] + ggpubr::border(color = "grey", size = 1), ncol = 1, heights = c(1,1,2,1))

```




## Increasing generas
```{r Paperplot correlation colons split AND combined, fig.height=39, fig.width=40}

# fig.height = 6.5 per row and width = 8.75 per col

# order <- c("Veillonella", 	"Sutterella", 	"Ruminococcus", 	"Ruminococcaceae", 	"Roseburia", 	"Pseudomonas", 	"Prevotella", 	"Phascolarctobacterium", 	"Parabacteroides", 	"Mitsuokella", 	"Megasphaera", 	"Megamonas", 	"Lachnospiraceae", 	"Fusobacterium", 	"Fusicatenibacter", 	"Faecalibacterium", 	"Erysipelotrichaceae", 	"Enterobacteriaceae", 	"Eisenbergiella", 	"Dorea", 	"Dialister", 	"Desulfovibrio", 	"Coprococcus", 	"Clostridium_XlVa", 	"Clostridiales", 	"Cloacibacillus", 	"Blautia", 	"Bilophila", 	"Bifidobacterium", 	"Bacteroides", 	"Bacteria", 	"Anaerostipes", 	"Anaeroglobus", 	"Alistipes", 	"Akkermansia", 	"Acidaminococcus")



# df.OTU <-subset(df.combined, sum > 10)
df.OTU <-subset(df.combined, mean > 5); x_lab = 16; 
hjustv = -0.1; vjustv = 1.5; xInf <- -Inf; yInf <- Inf # Top Left (h0,v1, -inf, +inf)  Top Right (h1,v1, +inf, +inf)
topnum <- 10; abovezero = T
# Retrieve top most abundant genera
## Absolute
phylobj.QMP_stable <- subset_samples(phylobj.QMP, Ninetransit == 3)
Top15phylobj <- aggregate_taxa(phylobj.QMP_stable, level = "Genus")
Top15phylobj <- aggregate_top_taxa(phylobj.QMP_stable, top = topnum, "Genus")
## Proportional
phylobj.prop_2 = transform_sample_counts(phylobj.QMP, function(x) 100 * x/sum(x))
phylobj.prop_stable <- subset_samples(phylobj.prop_2, Ninetransit == 3)
phylobj.prop_stable <- aggregate_top_taxa(phylobj.prop_stable, top = topnum, "Genus")
## Take consensus top genera
Top.genera <- rbind(subset(as.data.frame(phylobj.prop_stable@tax_table), Genus != "Other"), 
                    subset(as.data.frame(Top15phylobj@tax_table), Genus != "Other"))
## Remove insignificant genera
df <- data.frame()
for(i in significantlist){
  df2 <- subset(Top.genera, Genus == i)
  df <- rbind(df, df2)
}
Top.genera <- unique(df$Genus)
# Remove those that are not in the significant changes list (significantlist)
Top.genera <- sort(Top.genera, decreasing = T)
Top.genera <- c("Ruminococcus",  	"Pseudomonas", 	"Prevotella", 	"Megamonas", 	"Dialister", 	"Cloacibacillus", 	"Bilophila", 		"Bacteroides"
                # , "Acidaminococcus"
                )

# GENUS LEVEL
# Absolute
phylobj.QMP_stable <- subset_samples(phylobj.QMP, Ninetransit == 3)
Top15phylobj <- aggregate_taxa(phylobj.QMP_stable, level = "Genus")
Top15 <- psmelt(Top15phylobj); unique(Top15$OTU); unique(Top15$OTU) %in% Top15levels
## Subset according to top most abundant genera
df.test <- data.frame();
for(i in Top.genera){
  df.test2 <- subset(Top15, Genus == i)
  df.test <- rbind(df.test, df.test2)
}
Top15 <- df.test
x <- data.frame(x = unique(Top15$OTU) %in% Top15levels, genus = unique(Top15$OTU))  #check if labelled
Top15$OTU <- as.ordered(factor(Top15$OTU, levels = Top15levels))
## Add extra column days for graphs
Top <- Top15
Top$Transit <- as.ordered(factor(Top$Transit, levels = c("ST", "MT", "LT")))
Top$PD_Colon <- as.ordered(factor(Top$PD_Colon, levels = c("P", "D", "FS")))
Top$Donor <- as.ordered(factor(Top$Donor, levels = c("1", "2", "3", "4", "5", "6")))
Top$Genus <- as.ordered(factor(Top$Genus, levels = Top15levels))
## Add numbers
df.1 <- subset(Top, PD_Colon == "P" & Transit == "ST"); df.1$TT <- 8; df.1$TT1 <- 8
df.2 <- subset(Top, PD_Colon == "P" & Transit == "MT"); df.2$TT <- 16; df.2$TT1 <- 16
df.3 <- subset(Top, PD_Colon == "P" & Transit == "LT"); df.3$TT <- 24; df.3$TT1 <- 24
df.4 <- subset(Top, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 13; df.4$TT1 <- 8
df.5 <- subset(Top, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 26; df.5$TT1 <- 16
df.6 <- subset(Top, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 39; df.6$TT1 <- 24
# df.4 <- subset(Top, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 8
# df.5 <- subset(Top, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 16
# df.6 <- subset(Top, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 24

Top <- rbind(df.1, df.2, df.3, df.4, df.5, df.6)
rm(df.1, df.2, df.3, df.4, df.5, df.6)
## Perform test
dfcor <- data.frame(); df.otutest <- data.frame(); QMPlist <- list(); y =1
for(i in na.omit(unique(Top$Genus))){
  df1 <- subset(Top, Genus == i);  
  if(abovezero == T){
    df1.1 <- subset(df1, Donor == 1); df1.1 <- subset(df1.1, Abundance > 0)
    if (dim(df1.1)[1] == 0) {df1 <- subset(df1, Donor !=1)}
    df1.2 <- subset(df1, Donor == 2); df1.2 <- subset(df1.2, Abundance > 0) 
    if (dim(df1.2)[1] == 0) {df1 <- subset(df1, Donor !=2)}
    df1.3 <- subset(df1, Donor == 3); df1.3 <- subset(df1.3, Abundance > 0)
    if (dim(df1.3)[1] == 0) {df1 <- subset(df1, Donor !=3)}
    df1.4 <- subset(df1, Donor == 4); df1.4 <- subset(df1.4, Abundance > 0) 
    if (dim(df1.4)[1] == 0) {df1 <- subset(df1, Donor !=4)}
    df1.5 <- subset(df1, Donor == 5); df1.5 <- subset(df1.5, Abundance > 0) 
    if (dim(df1.5)[1] == 0) {df1 <- subset(df1, Donor !=5)}
    df1.6 <- subset(df1, Donor == 6); df1.6 <- subset(df1.6, Abundance > 0) 
    if (dim(df1.6)[1] == 0) {df1 <- subset(df1, Donor !=6)}
    }
  for(x in c("P", "D")){
  df<- subset(df1, PD_Colon == x)
  t <- cor(df$Abundance, df$TT, method = "spearman")
  corr <- cor.test(df$Abundance, df$TT, method = 'spearman'); 
  
  if(is.na(corr[["p.value"]])){
    sig <- "-"
    }else{
      if(corr[["p.value"]]<0.0001){
        sig <- "****"
        }else{
          if(corr[["p.value"]]<0.001){
            sig <- "***"
            }else{
              if(corr[["p.value"]]<0.01){
                sig <- "**"
                }else{
                  if(corr[["p.value"]]<0.05){
                    sig<-"*"
                  }else{
                      sig<-""}}}}}
  
  dfcor <- rbind(dfcor, data.frame(genus = i, cor = corr[["estimate"]][["rho"]], 
             PD_Colon = x, pvalue = corr[["p.value"]], sig = sig, comm = "Quantitative microbial community"))
  
  if(x == "P"){corP <- paste0("Ï = ",round(t, digits = 2)," ", sig, sep = ""); corrP <- corr[["p.value"]]
  }else{corD <- paste0("Ï = ",round(t, digits = 2)," ", sig, sep = ""); corrD <- corr[["p.value"]]}
  }
  
  # dat_text <- data.frame(label = c(corP, corD),
  # PD_Colon   = c("P","D"),
  # value   = c(max(df1$Abundance), max(df1$Abundance)), 
  # TT = c(8,8))

  
plot <- papertheme(ggplot(df1, aes(y = Abundance, x = TT1))) + geom_boxplot(aes(fill = Transit)) + geom_jitter(aes(color = Donor), size = 4) + 
    facet_grid(rows = vars(PD_Colon), labeller = labeller(PD_Colon = PD_Colon.labs), scales = "free_x") 

  if(is.na(corrP)){corrP <- 1} 
  if(is.na(corrD)){corrD <- 1} 
  if (corrP > 0.05 & corrD > 0.05) {
    plot <- plot
    dat_text <- data.frame(label = c("", ""), 
                           PD_Colon = c("P", "D"), 
                           value = c(max(df1$Abundance), max(df1$Abundance)),
                           xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                           hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
                           TT1 = c(x_lab, x_lab))
} else {
    if (corrP > 0.05 & corrD <= 0.05 ) {
        plot <- plot + geom_smooth(data = subset(df1, PD_Colon =="D"), aes(y = Abundance, x = TT1), method = lm,
                                   color = "black")
        dat_text <- data.frame(label = c("", corD), PD_Colon = c("P","D"), 
                               value = c(max(df1$Abundance), max(df1$Abundance)),
                               xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                               hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
                               TT1 = c(x_lab, x_lab))
    } else {
        if (corrP <= 0.05 & corrD > 0.05) {
            plot <- plot + geom_smooth(data = subset(df1,  PD_Colon == "P"), aes(y = Abundance, x = TT1),
                method = lm, color = "black")
            dat_text <- data.frame(label = c(corP, ""),
                                   PD_Colon = c("P", "D"), 
                                   value = c(max(df1$Abundance),max(df1$Abundance)),
                                   xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                                   hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
                                   TT1 = c(x_lab, x_lab))
        } else {
            plot <- plot + geom_smooth(aes(y = Abundance, x = TT1), method = lm, color = "black")
            dat_text <- data.frame(label = c(corP, corD),
                                   PD_Colon = c("P", "D"), 
                                   value = c(max(df1$Abundance),max(df1$Abundance)),
                                   xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                                   hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
                                   TT1 = c(x_lab, x_lab))
        }
    }
}
  dat_text$PD_Colon <- as.ordered(factor(dat_text$PD_Colon, levels = c("P", "D")))
  
 plot <- plot + scale_color_manual(values = sixdonors) + scale_fill_manual("SHIME transit", values = transitcolors, labels= c("Short", "Medium", "Long")) +
  scale_x_continuous(breaks=c(8,16,24), labels=c("Short", "Medium", "Long")) +
   # scale_x_discrete(labels = c("Short", "Medium", "Long"))+
   ylab(expression(paste("Absolute abundance (cells ",mL^{"-1"},")")))+ xlab("Transit time") + 
   scale_y_continuous(labels = scales::scientific) 
 
 if(i == "Acidaminococcus"){plot <- plot + 
   guides(fill=guide_legend(ncol=1, title.position = "top", hjust = 0.5),
          color=guide_legend(nrow=3, title.position = "top", hjust = 0.5)) +
   theme(legend.title.align = 0.5, legend.title = element_text(size=20), 
         legend.text = element_text(size=20), legend.position = c(0.5,0.35), legend.box = "horizontal")}else{
   plot <- plot + theme(legend.position = "none")}
         plot <- plot + theme(strip.background =element_rect(fill="#D6E0F4"),
         # legend.box = "horizontal",
         strip.text.x = element_text(size = 20),
         strip.text.y = element_blank(),
         # strip.text.y = element_text(size = 20, angle = 270),
         legend.title = element_text(size=20), legend.background = element_blank()) 
  
 
 
if(i == "X"){plot <- plot + geom_text(correlations)}else{
  # plot <- plot + geom_text(data= dat_text,  mapping = aes(x = as.numeric(TT1), y = as.numeric(value), label = label, family = 'sans'), size = 7)
  plot <- plot + geom_text(data= dat_text,  mapping = aes(x = xpos, y = ypos, label = label, 
                                        hjust=hjustvar,vjust=vjustvar, family = 'sans'), size = 7)
  }
QMPlist[[i]] <- plot; y = y+1
  }



# Proportional
phylobj.prop_2 = transform_sample_counts(phylobj.QMP, function(x) 100 * x/sum(x))
phylobj.prop_stable <- subset_samples(phylobj.prop_2, Ninetransit == 3)
Top15phylobj.prop <- aggregate_taxa(phylobj.prop_stable, level = "Genus")
Top15.prop <- psmelt(Top15phylobj.prop); unique(Top15.prop$OTU); unique(Top15.prop$OTU) %in% Top15levels
## Subset according to top most abundant genera
df.test <- data.frame();
for(i in Top.genera){
  df.test2 <- subset(Top15.prop, Genus == i)
  df.test <- rbind(df.test, df.test2)
}
Top15.prop <- df.test
x <- data.frame(x = unique(Top15$OTU) %in% Top15levels, genus = unique(Top15$OTU))  #check if labelled
Top15$OTU <- as.ordered(factor(Top15$OTU, levels = Top15levels))
x <- data.frame(x = unique(Top15.prop$OTU) %in% Top15levels, genus = unique(Top15.prop$OTU))  #check if labelled
Top15.prop$OTU <- as.ordered(factor(Top15.prop$OTU, levels = Top15levels))
## Add extra column days for graphs
Top.prop <- Top15.prop
Top.prop$Transit <- as.ordered(factor(Top.prop$Transit, levels = c("ST", "MT", "LT")))
Top.prop$PD_Colon <- as.ordered(factor(Top.prop$PD_Colon, levels = c("P", "D", "FS")))
Top.prop$Donor <- as.ordered(factor(Top.prop$Donor, levels = c("1", "2", "3", "4", "5", "6")))
Top.prop$Genus <- as.ordered(factor(Top.prop$Genus, levels = Top15levels))
## Add numbers
df.1 <- subset(Top.prop, PD_Colon == "P" & Transit == "ST"); df.1$TT <- 8; df.1$TT1 <- 8
df.2 <- subset(Top.prop, PD_Colon == "P" & Transit == "MT"); df.2$TT <- 16; df.2$TT1 <- 16
df.3 <- subset(Top.prop, PD_Colon == "P" & Transit == "LT"); df.3$TT <- 24; df.3$TT1 <- 24
df.4 <- subset(Top.prop, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 13; df.4$TT1 <- 8
df.5 <- subset(Top.prop, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 26; df.5$TT1 <- 16
df.6 <- subset(Top.prop, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 39; df.6$TT1 <- 24
# df.4 <- subset(Top.prop, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 8
# df.5 <- subset(Top.prop, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 16
# df.6 <- subset(Top.prop, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 24
Top.prop <- rbind(df.1, df.2, df.3, df.4, df.5, df.6)
rm(df.1, df.2, df.3, df.4, df.5, df.6)
dfcor2 <- data.frame(); PMPlist <- list(); y = 1

for(i in na.omit(unique(Top.prop$Genus))){
  df1 <- subset(Top.prop, Genus == i)
  if(abovezero == T){
    df1.1 <- subset(df1, Donor == 1); df1.1 <- subset(df1.1, Abundance > 0)
    if (dim(df1.1)[1] == 0) {df1 <- subset(df1, Donor !=1)}
    df1.2 <- subset(df1, Donor == 2); df1.2 <- subset(df1.2, Abundance > 0) 
    if (dim(df1.2)[1] == 0) {df1 <- subset(df1, Donor !=2)}
    df1.3 <- subset(df1, Donor == 3); df1.3 <- subset(df1.3, Abundance > 0)
    if (dim(df1.3)[1] == 0) {df1 <- subset(df1, Donor !=3)}
    df1.4 <- subset(df1, Donor == 4); df1.4 <- subset(df1.4, Abundance > 0) 
    if (dim(df1.4)[1] == 0) {df1 <- subset(df1, Donor !=4)}
    df1.5 <- subset(df1, Donor == 5); df1.5 <- subset(df1.5, Abundance > 0) 
    if (dim(df1.5)[1] == 0) {df1 <- subset(df1, Donor !=5)}
    df1.6 <- subset(df1, Donor == 6); df1.6 <- subset(df1.6, Abundance > 0) 
    if (dim(df1.6)[1] == 0) {df1 <- subset(df1, Donor !=6)}
    }
  for(x in c("P", "D")){
  df<- subset(df1, PD_Colon == x)
  t <- cor(df$Abundance, df$TT, method = "spearman")
  corr <- cor.test(df$Abundance, df$TT, method = 'spearman'); 
  
  if(is.na(corr[["p.value"]])){
    sig <- "-"
    }else{
      if(corr[["p.value"]]<0.0001){
        sig <- "****"
        }else{
          if(corr[["p.value"]]<0.001){
            sig <- "***"
            }else{
              if(corr[["p.value"]]<0.01){
                sig <- "**"
                }else{
                  if(corr[["p.value"]]<0.05){
                    sig<-"*"
                  }else{
                      sig<-""}}}}}
  
  dfcor2 <- rbind(dfcor2, data.frame(genus = i, cor = corr[["estimate"]][["rho"]], 
                 PD_Colon = x, pvalue = corr[["p.value"]], sig = sig, comm = "Proportional microbial community"))
  
  if(x == "P"){corP <- paste0("Ï = ",round(t, digits = 2)," ", sig, sep = ""); corrP <- corr[["p.value"]]}
  if(x == "D"){corD <- paste0("Ï = ",round(t, digits = 2)," ", sig, sep = ""); corrD <- corr[["p.value"]]}
  }
  
  #   dat_text <- data.frame(label = c(corP, corD),
  # PD_Colon   = c("P","D"),
  # value   = c(max(df1$Abundance), max(df1$Abundance)), 
  # TT = c(8,8))
  # dat_text$PD_Colon <- as.ordered(factor(dat_text$PD_Colon, levels = c("P", "D")))
  
plot <- papertheme(ggplot(df1, aes(y = Abundance, x = TT1))) + geom_boxplot(aes(fill = Transit)) + geom_jitter(aes(color = Donor), size =4) + 
      facet_grid(rows = vars(PD_Colon), labeller = labeller(PD_Colon = PD_Colon.labs), scales = "free_x") 

  if(is.na(corrP)){corrP <- 1} 
  if(is.na(corrD)){corrD <- 1} 
  if (corrP > 0.05 & corrD > 0.05) {
    plot <- plot
    dat_text <- data.frame(label = c("", ""), PD_Colon = c("P",
        "D"), value = c(max(df1$Abundance), max(df1$Abundance)),
        xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
        hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
        TT1 = c(x_lab, x_lab))
} else {
    if (corrP > 0.05 & corrD <= 0.05) {
        plot <- plot + geom_smooth(data = subset(df1, PD_Colon =="D"), 
                                   aes(y = Abundance, x = TT1), method = lm,
                                   color = "black")
        dat_text <- data.frame(label = c("", corD), PD_Colon = c("P",
            "D"), value = c(max(df1$Abundance), max(df1$Abundance)),
            xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
            hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
            TT1 = c(x_lab, x_lab))
    } else {
        if (corrP <= 0.05 & corrD > 0.05) {
            plot <- plot + geom_smooth(data = subset(df1,
                PD_Colon == "P"), aes(y = Abundance, x = TT1),
                method = lm, color = "black")
            dat_text <- data.frame(label = c(corP, ""),
                PD_Colon = c("P", "D"), value = c(max(df1$Abundance),
                  max(df1$Abundance)), 
                xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),TT1 = c(x_lab, x_lab))
        } else {
            plot <- plot + geom_smooth(aes(y = Abundance,
                x = TT1), method = lm, color = "black")
            dat_text <- data.frame(label = c(corP, corD),
                PD_Colon = c("P", "D"), value = c(max(df1$Abundance),
                  max(df1$Abundance)), 
                xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
                TT1 = c(x_lab, x_lab))
        }
    }
}
  dat_text$PD_Colon <- as.ordered(factor(dat_text$PD_Colon, levels = c("P", "D")))

plot <- plot + scale_color_manual(values = sixdonors) + scale_fill_manual(values = transitcolors) +
  scale_x_continuous(breaks=c(8,16,24), labels=c("Short", "Medium", "Long")) +
  ylab("Proportional abundance (%)")+ xlab("Transit time") + 
  # scale_x_discrete(labels = c("Short", "Medium", "Long"))+
    theme(strip.background =element_rect(fill="#D6E0F4"),
          legend.position = "none",
          legend.box = "horizontal",
          strip.text.x = element_text(size = 20),
          strip.text.y = element_blank(), 
          # strip.text.y = element_text(size = 20, angle = 270),
          legend.title = element_text(size=20), legend.background = element_blank()) 
if(i == "Ruminococcus"){
    plot <- plot + theme(strip.text.y = element_text(size = 20, angle = 270))
  }
  
if(i == "X"){plot <- plot + geom_text(correlations)}else{
  # plot <- plot + geom_text(data= dat_text,  mapping = aes(x = as.numeric(TT1), y = as.numeric(value),
  #                                                         label = label, family = 'sans'), size = 7)
  plot <- plot + geom_text(data= dat_text,  mapping = aes(x = xpos, y = ypos, label = label, 
                                        hjust=hjustvar,vjust=vjustvar, family = 'sans'), size = 7)
}
PMPlist[[i]] <- plot; y = y+1
    }

#Combine abs and prop abundances in one
dfcor <- rbind(dfcor, dfcor2)
dfcor <- na.omit(dfcor); dfcor$comm <- as.ordered(factor(dfcor$comm, 
                              levels = c("Quantitative microbial community", "Proportional microbial community")))
dfcor$PD_Colon <- as.ordered(factor(dfcor$PD_Colon, levels = c("P", "D")))
dfcor$genus <- as.ordered(factor(dfcor$genus, levels = Top.genera))
plot <- papertheme(ggplot(dfcor, aes(x = PD_Colon, y = genus, fill =cor))) + geom_tile() +
  scale_fill_gradient2(na.value = "grey50",low="blue", high="red") + 
  geom_text(aes(label = sig)) +
  theme(axis.text.y = element_text(face="italic")) + ylab("Genus") + xlab("Colon vessel") + 
  scale_x_discrete(breaks=c("P","D"), labels=c("Proximal colon", "Distal colon")) +
  labs(fill = expression(rho)) +
  facet_grid(cols = vars(comm)) +theme(strip.background = element_rect(fill = "#D6E0F4"))
plot
dfcorall <- dfcor


# Combine plots
plots <- list(); x = 1
genera <- na.omit(unique(Top.prop$Genus))
genera <- sort(genera, decreasing = F)
for(i in genera){ 
  plot1 <- QMPlist[[i]]; plot2 <- PMPlist[[i]]
  plot <- ggarrange(plot1, plot2)
  plot <- annotate_figure(plot, text_grob(i, color = "black", face = "bold.italic", size = 25))
  # print(plot)
  plots[[i]] <- plot
  x = x +1
}

plots1 <- plots[1:8]; plots2 <- plots[9:16]
# ggarrange(plotlist = plots1, nrow = 2, ncol = 4)
# ggarrange(plotlist = plots2, nrow = 2, ncol = 4)
# ggarrange(plotlist = plots, nrow = 4, ncol = 4)








# OTU LEVEL
# Create OTU list with the genera
Top.OTU <- data.frame(); df.OTU <-df.combined
for(i in Top.genera){
  df.test <- subset(df.OTU, genus == i)
  Top.OTU <- rbind(Top.OTU, df.test)
}
Top.OTU <- unique(Top.OTU$rowname)

# ABSOLUTE ABUNDANCE
Top15.otu <- psmelt(subset_samples(phylobj.QMP.otu, Ninetransit == 3)); unique(Top15.otu$OTU)
## Subset according to top most abundant
df.test <- data.frame()
for(i in Top.OTU){
  df.test2 <- subset(Top15.otu, OTU == i)
  df.test <- rbind(df.test, df.test2)
}
Top15.otu <- df.test
unique(Top15.otu$OTU)
## Add extra column days for graphs
Top.otu <- Top15.otu
Top.otu$Transit <- as.ordered(factor(Top.otu$Transit, levels = c("ST", "MT", "LT")))
Top.otu$PD_Colon <- as.ordered(factor(Top.otu$PD_Colon, levels = c("P", "D", "FS")))
Top.otu$Donor <- as.ordered(factor(Top.otu$Donor, levels = c("1", "2", "3", "4", "5", "6")))

## Add numbers
df.1 <- subset(Top.otu, PD_Colon == "P" & Transit == "ST"); df.1$TT <- 8; df.1$TT1 <- 8
df.2 <- subset(Top.otu, PD_Colon == "P" & Transit == "MT"); df.2$TT <- 16; df.2$TT1 <- 16
df.3 <- subset(Top.otu, PD_Colon == "P" & Transit == "LT"); df.3$TT <- 24; df.3$TT1 <- 24
df.4 <- subset(Top.otu, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 13; df.4$TT1 <- 8
df.5 <- subset(Top.otu, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 26; df.5$TT1 <- 16
df.6 <- subset(Top.otu, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 39; df.6$TT1 <- 24
Top.otu <- rbind(df.1, df.2, df.3, df.4, df.5, df.6)
rm(df.1, df.2, df.3, df.4, df.5, df.6)


## Perform tests
dfcor.otu <- data.frame(); OTUQMPlist <- list()
for(i in na.omit(unique(Top.otu$OTU))){
  df1 <- subset(Top.otu, OTU == i)
  if(abovezero == T){
    df1.1 <- subset(df1, Donor == 1); df1.1 <- subset(df1.1, Abundance > 0)
    if (dim(df1.1)[1] == 0) {df1 <- subset(df1, Donor !=1)}
    df1.2 <- subset(df1, Donor == 2); df1.2 <- subset(df1.2, Abundance > 0) 
    if (dim(df1.2)[1] == 0) {df1 <- subset(df1, Donor !=2)}
    df1.3 <- subset(df1, Donor == 3); df1.3 <- subset(df1.3, Abundance > 0)
    if (dim(df1.3)[1] == 0) {df1 <- subset(df1, Donor !=3)}
    df1.4 <- subset(df1, Donor == 4); df1.4 <- subset(df1.4, Abundance > 0) 
    if (dim(df1.4)[1] == 0) {df1 <- subset(df1, Donor !=4)}
    df1.5 <- subset(df1, Donor == 5); df1.5 <- subset(df1.5, Abundance > 0) 
    if (dim(df1.5)[1] == 0) {df1 <- subset(df1, Donor !=5)}
    df1.6 <- subset(df1, Donor == 6); df1.6 <- subset(df1.6, Abundance > 0) 
    if (dim(df1.6)[1] == 0) {df1 <- subset(df1, Donor !=6)}
    }
  for(x in c("P", "D")){
  df<- subset(df1, PD_Colon == x)
  t <- cor(df$Abundance, df$TT, method = "spearman")
  
  if(is.na(t)){corr[["p.value"]] <- NA}else{
   corr <- cor.test(df$Abundance, df$TT, method = 'spearman')  
  }
  
  
  if(is.na(corr[["p.value"]])){
    sig <- ""
    }else{
      if(corr[["p.value"]]<0.0001){
        sig <- "****"
        }else{
          if(corr[["p.value"]]<0.001){
            sig <- "***"
            }else{
              if(corr[["p.value"]]<0.01){
                sig <- "**"
                }else{
                  if(corr[["p.value"]]<0.05){
                    sig<-"*"
                  }else{
                      sig<-""}}}}}
  
  if(is.na(t)){dfcor.otu <- rbind(dfcor.otu, data.frame(genus = i, cor = NA, 
                 PD_Colon = x, pvalue = corr[["p.value"]], sig = sig, comm = "Quantitative microbial community"))
  }else{
    dfcor.otu <- rbind(dfcor.otu, data.frame(genus = i, cor = corr[["estimate"]][["rho"]], 
                 PD_Colon = x, pvalue = corr[["p.value"]], sig = sig, comm = "Quantitative microbial community"))}
  
  if(x == "P"){corP <- paste0("Ï = ",round(t, digits = 2)," ", sig, sep = ""); corrP <- corr[["p.value"]]
  }else{corD <- paste0("Ï = ",round(t, digits = 2)," ", sig, sep = ""); corrD <- corr[["p.value"]]}
  }
  
  plot <- papertheme(ggplot(df1, aes(y = Abundance, x = TT1))) + geom_boxplot(aes(fill = Transit)) + geom_jitter(aes(color = Donor), size =4) + 
      facet_grid(rows = vars(PD_Colon), labeller = labeller(PD_Colon = PD_Colon.labs), scales = "free_x") 
  
  if(is.na(corrP)){corrP <- 1} 
  if(is.na(corrD)){corrD <- 1} 
  if (corrP > 0.05 & corrD > 0.05) {
    plot <- plot
    dat_text <- data.frame(label = c("", ""), PD_Colon = c("P",
        "D"), value = c(max(df1$Abundance), max(df1$Abundance)),
        xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
        hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
        TT1 = c(x_lab, x_lab))
} else {
    if (corrP > 0.05 & corrD <= 0.05) {
        plot <- plot + geom_smooth(data = subset(df1, PD_Colon =="D"), 
                                   aes(y = Abundance, x = TT1), method = lm,
                                   color = "black")
        dat_text <- data.frame(label = c("", corD), PD_Colon = c("P",
            "D"), value = c(max(df1$Abundance), max(df1$Abundance)),
            xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
            hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
            TT1 = c(x_lab, x_lab))
    } else {
        if (corrP <= 0.05 & corrD > 0.05) {
            plot <- plot + geom_smooth(data = subset(df1,
                PD_Colon == "P"), aes(y = Abundance, x = TT1),
                method = lm, color = "black")
            dat_text <- data.frame(label = c(corP, ""),
                PD_Colon = c("P", "D"), value = c(max(df1$Abundance),
                  max(df1$Abundance)),
                xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),TT1 = c(x_lab, x_lab))
        } else {
            plot <- plot + geom_smooth(aes(y = Abundance,
                x = TT1), method = lm, color = "black")
            dat_text <- data.frame(label = c(corP, corD),
                PD_Colon = c("P", "D"), value = c(max(df1$Abundance),
                  max(df1$Abundance)), 
                xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
                TT1 = c(x_lab, x_lab))
        }
    }
}
  dat_text$PD_Colon <- as.ordered(factor(dat_text$PD_Colon, levels = c("P", "D")))

plot <- plot + scale_color_manual(values = sixdonors) + 
  scale_fill_manual("SHIME transit", values = transitcolors, labels = c("Short", "Medium", "Long")) +
  # scale_x_discrete(labels = c("Short", "Medium", "Long"))+
  scale_x_continuous(breaks=c(8,16,24), labels=c("Short", "Medium", "Long")) +
  ylab(expression(paste("Absolute abundance (cells ",mL^{"-1"},")")))+ xlab("Transit time") + 
  scale_y_continuous(labels = scales::scientific) 
if(i == "Otu00029"){plot <- plot + 
   guides(fill=guide_legend(ncol=1, title.position = "top", hjust = 0.5),
          color=guide_legend(nrow=3, title.position = "top", hjust = 0.5)) +
   theme(legend.title.align = 0.5, legend.title = element_text(size=20), 
         legend.text = element_text(size=20), legend.position = c(0.5,0.85), legend.box = "horizontal")}else{
           if(i == "Otu00042"){plot <- plot + 
   guides(fill=guide_legend(ncol=1, title.position = "top", hjust = 0.5),
          color=guide_legend(nrow=3, title.position = "top", hjust = 0.5)) +
   theme(legend.title.align = 0.5, legend.title = element_text(size=20), 
         legend.text = element_text(size=20), legend.position = c(0.5,0.30), legend.box = "horizontal")}else{
   plot <- plot + theme(legend.position = "none")}}
         plot <- plot + theme(strip.background =element_rect(fill="#D6E0F4"),
         # legend.box = "horizontal",
         strip.text.x = element_text(size = 20),
         strip.text.y = element_blank(),
         # strip.text.y = element_text(size = 20, angle = 270),
         legend.title = element_text(size=20), legend.background = element_blank())
         
    


  if(i == "X"){plot <- plot + geom_text(correlations)}else{
  # plot <- plot + geom_text(data= dat_text,  mapping = aes(x = as.numeric(TT1), y = as.numeric(value), label = label, family = 'sans'), size = 7)
  plot <- plot + geom_text(data= dat_text,  mapping = aes(x = xpos, y = ypos, label = label, 
                                        hjust=hjustvar,vjust=vjustvar, family = 'sans'), size = 7)
}
 OTUQMPlist[[i]] <- plot; y = y+1
  
  }


# Proportional
Top15.prop.otu <- psmelt(subset_samples(phylobj.prop.otu, Ninetransit == 3)); unique(Top15.prop.otu$OTU)
## Subset according to top most abundant
df.test <- data.frame()
for(i in Top.OTU){
  df.test2 <- subset(Top15.prop.otu, OTU == i)
  df.test <- rbind(df.test, df.test2)
}
Top15.prop.otu <- df.test
unique(Top15.prop.otu$OTU)
## Add extra column days for graphs
Top.prop.otu <- Top15.prop.otu
Top.prop.otu$Transit <- as.ordered(factor(Top.prop.otu$Transit, levels = c("ST", "MT", "LT")))
Top.prop.otu$PD_Colon <- as.ordered(factor(Top.prop.otu$PD_Colon, levels = c("P", "D", "FS")))
Top.prop.otu$Donor <- as.ordered(factor(Top.prop.otu$Donor, levels = c("1", "2", "3", "4", "5", "6")))

## Add numbers
df.1 <- subset(Top.prop.otu, PD_Colon == "P" & Transit == "ST"); df.1$TT <- 8; df.1$TT1 <- 8
df.2 <- subset(Top.prop.otu, PD_Colon == "P" & Transit == "MT"); df.2$TT <- 16; df.2$TT1 <- 16
df.3 <- subset(Top.prop.otu, PD_Colon == "P" & Transit == "LT"); df.3$TT <- 24; df.3$TT1 <- 24
df.4 <- subset(Top.prop.otu, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 13; df.4$TT1 <- 8
df.5 <- subset(Top.prop.otu, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 26; df.5$TT1 <- 16
df.6 <- subset(Top.prop.otu, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 39; df.6$TT1 <- 24
Top.prop.otu <- rbind(df.1, df.2, df.3, df.4, df.5, df.6)
rm(df.1, df.2, df.3, df.4, df.5, df.6)


## Perform tests
dfcor.otu <- data.frame(); OTUPMPlist <- list()
for(i in na.omit(unique(Top.prop.otu$OTU))){
  df1 <- subset(Top.prop.otu, OTU == i)
  if(abovezero == T){
    df1.1 <- subset(df1, Donor == 1); df1.1 <- subset(df1.1, Abundance > 0)
    if (dim(df1.1)[1] == 0) {df1 <- subset(df1, Donor !=1)}
    df1.2 <- subset(df1, Donor == 2); df1.2 <- subset(df1.2, Abundance > 0) 
    if (dim(df1.2)[1] == 0) {df1 <- subset(df1, Donor !=2)}
    df1.3 <- subset(df1, Donor == 3); df1.3 <- subset(df1.3, Abundance > 0)
    if (dim(df1.3)[1] == 0) {df1 <- subset(df1, Donor !=3)}
    df1.4 <- subset(df1, Donor == 4); df1.4 <- subset(df1.4, Abundance > 0) 
    if (dim(df1.4)[1] == 0) {df1 <- subset(df1, Donor !=4)}
    df1.5 <- subset(df1, Donor == 5); df1.5 <- subset(df1.5, Abundance > 0) 
    if (dim(df1.5)[1] == 0) {df1 <- subset(df1, Donor !=5)}
    df1.6 <- subset(df1, Donor == 6); df1.6 <- subset(df1.6, Abundance > 0) 
    if (dim(df1.6)[1] == 0) {df1 <- subset(df1, Donor !=6)}
    }
  for(x in c("P", "D")){
  df<- subset(df1, PD_Colon == x)
  t <- cor(df$Abundance, df$TT, method = "spearman")
  
  if(is.na(t)){corr[["p.value"]] <- NA}else{
   corr <- cor.test(df$Abundance, df$TT, method = 'spearman')  
  }
  
  
  if(is.na(corr[["p.value"]])){
    sig <- ""
    }else{
      if(corr[["p.value"]]<0.0001){
        sig <- "****"
        }else{
          if(corr[["p.value"]]<0.001){
            sig <- "***"
            }else{
              if(corr[["p.value"]]<0.01){
                sig <- "**"
                }else{
                  if(corr[["p.value"]]<0.05){
                    sig<-"*"
                  }else{
                      sig<-""}}}}}
  
  if(is.na(t)){dfcor.otu <- rbind(dfcor.otu, data.frame(genus = i, cor = NA, 
                 PD_Colon = x, pvalue = corr[["p.value"]], sig = sig, comm = "Quantitative microbial community"))
  }else{
    dfcor.otu <- rbind(dfcor.otu, data.frame(genus = i, cor = corr[["estimate"]][["rho"]], 
                 PD_Colon = x, pvalue = corr[["p.value"]], sig = sig, comm = "Quantitative microbial community"))}
  
  if(x == "P"){corP <- paste0("Ï = ",round(t, digits = 2)," ", sig, sep = ""); corrP <- corr[["p.value"]]
  }else{corD <- paste0("Ï = ",round(t, digits = 2)," ", sig, sep = ""); corrD <- corr[["p.value"]]}
  }
  
  plot <- papertheme(ggplot(df1, aes(y = Abundance, x = TT1))) + geom_boxplot(aes(fill = Transit)) + geom_jitter(aes(color = Donor), size =4) + 
      facet_grid(rows = vars(PD_Colon), labeller = labeller(PD_Colon = PD_Colon.labs), scales = "free_x") 

  
  
  if(is.na(corrP)){corrP <- 1} 
  if(is.na(corrD)){corrD <- 1} 
  if (corrP > 0.05 & corrD > 0.05) {
    plot <- plot
    dat_text <- data.frame(label = c("", ""), PD_Colon = c("P",
        "D"), value = c(max(df1$Abundance), max(df1$Abundance)),
        xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
        TT1 = c(x_lab, x_lab))
} else {
    if (corrP > 0.05 & corrD <= 0.05) {
        plot <- plot + geom_smooth(data = subset(df1, PD_Colon =="D"), 
                                   aes(y = Abundance, x = TT1), method = lm,
                                   color = "black")
        dat_text <- data.frame(label = c("", corD), PD_Colon = c("P",
            "D"), value = c(max(df1$Abundance), max(df1$Abundance)),
            xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
            TT1 = c(x_lab, x_lab))
    } else {
        if (corrP <= 0.05 & corrD > 0.05) {
            plot <- plot + geom_smooth(data = subset(df1,
                PD_Colon == "P"), aes(y = Abundance, x = TT1),
                method = lm, color = "black")
            dat_text <- data.frame(label = c(corP, ""),
                PD_Colon = c("P", "D"), value = c(max(df1$Abundance),
                  max(df1$Abundance)), 
                xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),TT1 = c(x_lab, x_lab))
        } else {
            plot <- plot + geom_smooth(aes(y = Abundance,
                x = TT1), method = lm, color = "black")
            dat_text <- data.frame(label = c(corP, corD),
                PD_Colon = c("P", "D"), value = c(max(df1$Abundance),
                  max(df1$Abundance)), 
                xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),TT1 = c(x_lab, x_lab))
        }
    }
}
  dat_text$PD_Colon <- as.ordered(factor(dat_text$PD_Colon, levels = c("P", "D")))

plot <- plot + scale_color_manual(values = sixdonors) + 
  scale_fill_manual("SHIME transit", values = transitcolors, labels = c("Short", "Medium", "Long")) + 
  guides(fill=guide_legend(ncol=1, title.position = "top", hjust = 0.5), 
         color=guide_legend(nrow=3, title.position = "top", hjust = 0.5)) +
  # scale_x_discrete(labels = c("Short", "Medium", "Long"))+ 
  scale_x_continuous(breaks=c(8,16,24), labels=c("Short", "Medium", "Long")) +
  theme(legend.title.align = 0.5, legend.title = element_text(size=20), legend.text = element_text(size=20))+
  ylab("Proportional abundance (%)")+ xlab("Transit time") 
  
plt_legend <- g_legend(plot)

if(i == "Otu00144"){plot <- plot + 
   guides(fill=guide_legend(ncol=1, title.position = "top", hjust = 0.5),
          color=guide_legend(nrow=3, title.position = "top", hjust = 0.5)) +
   theme(legend.title.align = 0.5, legend.title = element_text(size=20), 
         legend.text = element_text(size=20), legend.position = c(0.5,0.30), legend.box = "horizontal")}else{
   plot <- plot + theme(legend.position = "none")}
         plot <- plot + theme(strip.background =element_rect(fill="#D6E0F4"),
         # legend.box = "horizontal",
         strip.text.x = element_text(size = 20),
         strip.text.y = element_blank(),
         # strip.text.y = element_text(size = 20, angle = 270),
         legend.title = element_text(size=20), legend.background = element_blank())
         
  if(i == "Otu00015" | i == "Otu00013" | i == "Otu00144"| i == "Otu00014"| i == "Otu00042" |  i == "Otu00025"| i == "Otu00008"| i == "Otu00038"|
      i == "Otu00189"| i == "Otu00073"| i == "Otu00004"){
    plot <- plot + theme(strip.text.y = element_text(size = 20, angle = 270))
  }




  if(i == "X"){plot <- plot + geom_text(correlations)}else{
  # plot <- plot + geom_text(data= dat_text,  mapping = aes(x = as.numeric(TT1), y = as.numeric(value), label = label, family = 'sans'), size = 7)
  plot <- plot + geom_text(data= dat_text,  mapping = aes(x = xpos, y = ypos, label = label, 
                                        hjust=hjustvar,vjust=vjustvar, family = 'sans'), size = 7)
}
 OTUPMPlist[[i]] <- plot; y = y+1
  
  }







# Combine OTUplots
otuplots <- list(); x = 1
genera <- na.omit(unique(Top.otu$OTU))
genera <- sort(genera, decreasing = F)
for(i in genera){ 
  plot1 <- OTUQMPlist[[i]]; plot2 <- OTUPMPlist[[i]]
  plot <- ggarrange(plot1, plot2)
  # print(plot)
  
  m =i
  if(stringr::str_detect(m, "Otu0000")){
      y <- str_replace(m, "Otu0000", "OTU")
    }else{if(stringr::str_detect(m, "Otu000")){
      y <- str_replace(m, "Otu000", "OTU")
    }else{if(stringr::str_detect(m, "Otu00")){
      y <- str_replace(m, "Otu00", "OTU")
    }else{if(stringr::str_detect(m, "Otu0")){
      y <- str_replace(m, "Otu0", "OTU")
    }}}}
  
  plot <- annotate_figure(plot, text_grob(y, color = "black", face = "bold", size = 25))
  # print(plot)
  otuplots[[i]] <- plot
  x = x +1
}

# ggarrange(plotlist = otuplots)



# Combine genera and OTU's
allplotslist <- list()
for(i in sort(Top.genera, decreasing = F)){
  plot1 <- plots[[i]]
dfje <- subset(df.combined, genus == i)
dfje <- subset(dfje, rowname!="Otu00102")
dfje <- subset(dfje, rowname!="Otu00216")
dfje <- subset(dfje, rowname!="Otu00261")
dfje <- subset(dfje, rowname!="Otu00078")
dfje <- subset(dfje, rowname!="Otu00133")
dfje <- subset(dfje, rowname!="Otu00214")
dfje <- subset(dfje, rowname!="Otu00189")
dfje <- subset(dfje, mean > 5)
plotjes <- list()
plotjes[[1]] <- plot1; y = 2
for(x in unique(dfje$rowname)){
  plot2 <- otuplots[[x]]
    plotjes[[y]] <- plot2;  y = y+1
}

if(i == "Ruminococcus"){xtralist <- plotjes
  plotjes[[y]] <- ggarrange(NULL, plt_legend ,nrow = 2)}
if(i == "Megamonas"){xtralist2 <- plotjes}
if(i == "Acidaminococcus"){plotA <- ggarrange(plotlist = plotjes, nrow = 1, ncol = 2)}
if(i == "Bilophila"){plotB <- ggarrange(plotlist = plotjes, nrow = 1, ncol = 2)}
if(i == "Bilophila"){plotA <- ggarrange(plotlist = plotjes, nrow = 1, ncol = 2)}
if(i == "Cloacibacillus"){plotB <- ggarrange(plotlist = plotjes, nrow = 1, ncol = 2)}
if(i == "Megamonas"){plotC <- ggarrange(plotlist = plotjes, nrow = 1, ncol = 2)}
if(i == "Pseudomonas"){plotD <- ggarrange(plotlist = plotjes, nrow = 1, ncol = 2)}

allplotslist[[i]] <- ggarrange(plotlist = plotjes, 
                               nrow = ceiling((nrow(dfje)+1)/4), 
                               ncol = 4)
}

# ggarrange(plotlist =  allplotslist, ncol = 1)

# Rearrange them maybe?
plotAB <- ggarrange(plotA+ ggpubr::border(color = "grey", size = 1), 
                    plotB+ ggpubr::border(color = "grey", size = 1), ncol = 2, widths = c(2,2))
plotCD <- ggarrange(plotC+ ggpubr::border(color = "grey", size = 1), 
                    plotD+ ggpubr::border(color = "grey", size = 1), ncol = 2, widths = c(2,2))

# plotE <- ggarrange(xtralist2[[1]], xtralist2[[2]], xtralist2[[3]],  plt_legend, ncol = 4, nrow = 1)
ggarrange(plotAB , allplotslist[["Bacteroides"]]+ ggpubr::border(color = "grey", size = 1), 
          allplotslist[["Dialister"]]+ ggpubr::border(color = "grey", size = 1), 
          plotCD, allplotslist[["Prevotella"]]+ ggpubr::border(color = "grey", size = 1),
          # plotE+ ggpubr::border(color = "grey", size = 1), 
          ncol = 1, heights = c(1,2,1,1,1))
```

## Weird generas
```{r Paperplot correlation colons split AND combined, fig.height=26, fig.width=35}

# fig.height = 6.5 per row and width = 8.75 per col

# order <- c("Veillonella", 	"Sutterella", 	"Ruminococcus", 	"Ruminococcaceae", 	"Roseburia", 	"Pseudomonas", 	"Prevotella", 	"Phascolarctobacterium", 	"Parabacteroides", 	"Mitsuokella", 	"Megasphaera", 	"Megamonas", 	"Lachnospiraceae", 	"Fusobacterium", 	"Fusicatenibacter", 	"Faecalibacterium", 	"Erysipelotrichaceae", 	"Enterobacteriaceae", 	"Eisenbergiella", 	"Dorea", 	"Dialister", 	"Desulfovibrio", 	"Coprococcus", 	"Clostridium_XlVa", 	"Clostridiales", 	"Cloacibacillus", 	"Blautia", 	"Bilophila", 	"Bifidobacterium", 	"Bacteroides", 	"Bacteria", 	"Anaerostipes", 	"Anaeroglobus", 	"Alistipes", 	"Akkermansia", 	"Acidaminococcus")



# df.OTU <-subset(df.combined, sum > 10)
df.OTU <-subset(df.combined, mean > 5); x_lab = 16; 
hjustv = -0.1; vjustv = 1.5; xInf <- -Inf; yInf <- Inf # Top Left (h0,v1, -inf, +inf)  Top Right (h1,v1, +inf, +inf)
topnum <- 10; abovezero = T
# Retrieve top most abundant genera
## Absolute
phylobj.QMP_stable <- subset_samples(phylobj.QMP, Ninetransit == 3)
Top15phylobj <- aggregate_taxa(phylobj.QMP_stable, level = "Genus")
Top15phylobj <- aggregate_top_taxa(phylobj.QMP_stable, top = topnum, "Genus")
## Proportional
phylobj.prop_2 = transform_sample_counts(phylobj.QMP, function(x) 100 * x/sum(x))
phylobj.prop_stable <- subset_samples(phylobj.prop_2, Ninetransit == 3)
phylobj.prop_stable <- aggregate_top_taxa(phylobj.prop_stable, top = topnum, "Genus")
## Take consensus top genera
Top.genera <- rbind(subset(as.data.frame(phylobj.prop_stable@tax_table), Genus != "Other"), 
                    subset(as.data.frame(Top15phylobj@tax_table), Genus != "Other"))
## Remove insignificant genera
df <- data.frame()
for(i in significantlist){
  df2 <- subset(Top.genera, Genus == i)
  df <- rbind(df, df2)
}
Top.genera <- unique(df$Genus)
# Remove those that are not in the significant changes list (significantlist)
Top.genera <- sort(Top.genera, decreasing = T)
Top.genera <- c("Lachnospiraceae", 	"Enterobacteriaceae", 	"Clostridium_XlVa",  	"Akkermansia")

# GENUS LEVEL
# Absolute
phylobj.QMP_stable <- subset_samples(phylobj.QMP, Ninetransit == 3)
Top15phylobj <- aggregate_taxa(phylobj.QMP_stable, level = "Genus")
Top15 <- psmelt(Top15phylobj); unique(Top15$OTU); unique(Top15$OTU) %in% Top15levels
## Subset according to top most abundant genera
df.test <- data.frame();
for(i in Top.genera){
  df.test2 <- subset(Top15, Genus == i)
  df.test <- rbind(df.test, df.test2)
}
Top15 <- df.test
x <- data.frame(x = unique(Top15$OTU) %in% Top15levels, genus = unique(Top15$OTU))  #check if labelled
Top15$OTU <- as.ordered(factor(Top15$OTU, levels = Top15levels))
## Add extra column days for graphs
Top <- Top15
Top$Transit <- as.ordered(factor(Top$Transit, levels = c("ST", "MT", "LT")))
Top$PD_Colon <- as.ordered(factor(Top$PD_Colon, levels = c("P", "D", "FS")))
Top$Donor <- as.ordered(factor(Top$Donor, levels = c("1", "2", "3", "4", "5", "6")))
Top$Genus <- as.ordered(factor(Top$Genus, levels = Top15levels))
## Add numbers
df.1 <- subset(Top, PD_Colon == "P" & Transit == "ST"); df.1$TT <- 8; df.1$TT1 <- 8
df.2 <- subset(Top, PD_Colon == "P" & Transit == "MT"); df.2$TT <- 16; df.2$TT1 <- 16
df.3 <- subset(Top, PD_Colon == "P" & Transit == "LT"); df.3$TT <- 24; df.3$TT1 <- 24
df.4 <- subset(Top, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 13; df.4$TT1 <- 8
df.5 <- subset(Top, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 26; df.5$TT1 <- 16
df.6 <- subset(Top, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 39; df.6$TT1 <- 24
# df.4 <- subset(Top, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 8
# df.5 <- subset(Top, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 16
# df.6 <- subset(Top, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 24

Top <- rbind(df.1, df.2, df.3, df.4, df.5, df.6)
rm(df.1, df.2, df.3, df.4, df.5, df.6)
## Perform test
dfcor <- data.frame(); df.otutest <- data.frame(); QMPlist <- list(); y =1
for(i in na.omit(unique(Top$Genus))){
  df1 <- subset(Top, Genus == i);  
  if(abovezero == T){
    df1.1 <- subset(df1, Donor == 1); df1.1 <- subset(df1.1, Abundance > 0)
    if (dim(df1.1)[1] == 0) {df1 <- subset(df1, Donor !=1)}
    df1.2 <- subset(df1, Donor == 2); df1.2 <- subset(df1.2, Abundance > 0) 
    if (dim(df1.2)[1] == 0) {df1 <- subset(df1, Donor !=2)}
    df1.3 <- subset(df1, Donor == 3); df1.3 <- subset(df1.3, Abundance > 0)
    if (dim(df1.3)[1] == 0) {df1 <- subset(df1, Donor !=3)}
    df1.4 <- subset(df1, Donor == 4); df1.4 <- subset(df1.4, Abundance > 0) 
    if (dim(df1.4)[1] == 0) {df1 <- subset(df1, Donor !=4)}
    df1.5 <- subset(df1, Donor == 5); df1.5 <- subset(df1.5, Abundance > 0) 
    if (dim(df1.5)[1] == 0) {df1 <- subset(df1, Donor !=5)}
    df1.6 <- subset(df1, Donor == 6); df1.6 <- subset(df1.6, Abundance > 0) 
    if (dim(df1.6)[1] == 0) {df1 <- subset(df1, Donor !=6)}
    }
  for(x in c("P", "D")){
  df<- subset(df1, PD_Colon == x)
  t <- cor(df$Abundance, df$TT, method = "spearman")
  corr <- cor.test(df$Abundance, df$TT, method = 'spearman'); 
  
  if(is.na(corr[["p.value"]])){
    sig <- "-"
    }else{
      if(corr[["p.value"]]<0.0001){
        sig <- "****"
        }else{
          if(corr[["p.value"]]<0.001){
            sig <- "***"
            }else{
              if(corr[["p.value"]]<0.01){
                sig <- "**"
                }else{
                  if(corr[["p.value"]]<0.05){
                    sig<-"*"
                  }else{
                      sig<-""}}}}}
  
  dfcor <- rbind(dfcor, data.frame(genus = i, cor = corr[["estimate"]][["rho"]], 
             PD_Colon = x, pvalue = corr[["p.value"]], sig = sig, comm = "Quantitative microbial community"))
  
  if(x == "P"){corP <- paste0("Ï = ",round(t, digits = 2)," ", sig, sep = ""); corrP <- corr[["p.value"]]
  }else{corD <- paste0("Ï = ",round(t, digits = 2)," ", sig, sep = ""); corrD <- corr[["p.value"]]}
  }
  
  # dat_text <- data.frame(label = c(corP, corD),
  # PD_Colon   = c("P","D"),
  # value   = c(max(df1$Abundance), max(df1$Abundance)), 
  # TT = c(8,8))

  
plot <- papertheme(ggplot(df1, aes(y = Abundance, x = TT1))) + geom_boxplot(aes(fill = Transit)) + geom_jitter(aes(color = Donor), size = 4) + 
    facet_grid(rows = vars(PD_Colon), labeller = labeller(PD_Colon = PD_Colon.labs), scales = "free_x") 

  if(is.na(corrP)){corrP <- 1} 
  if(is.na(corrD)){corrD <- 1} 
  if (corrP > 0.05 & corrD > 0.05) {
    plot <- plot
    dat_text <- data.frame(label = c("", ""), 
                           PD_Colon = c("P", "D"), 
                           value = c(max(df1$Abundance), max(df1$Abundance)),
                           xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                           hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
                           TT1 = c(x_lab, x_lab))
} else {
    if (corrP > 0.05 & corrD <= 0.05 ) {
        plot <- plot + geom_smooth(data = subset(df1, PD_Colon =="D"), aes(y = Abundance, x = TT1), method = lm,
                                   color = "black")
        dat_text <- data.frame(label = c("", corD), PD_Colon = c("P","D"), 
                               value = c(max(df1$Abundance), max(df1$Abundance)),
                               xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                               hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
                               TT1 = c(x_lab, x_lab))
    } else {
        if (corrP <= 0.05 & corrD > 0.05) {
            plot <- plot + geom_smooth(data = subset(df1,  PD_Colon == "P"), aes(y = Abundance, x = TT1),
                method = lm, color = "black")
            dat_text <- data.frame(label = c(corP, ""),
                                   PD_Colon = c("P", "D"), 
                                   value = c(max(df1$Abundance),max(df1$Abundance)),
                                   xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                                   hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
                                   TT1 = c(x_lab, x_lab))
        } else {
            plot <- plot + geom_smooth(aes(y = Abundance, x = TT1), method = lm, color = "black")
            dat_text <- data.frame(label = c(corP, corD),
                                   PD_Colon = c("P", "D"), 
                                   value = c(max(df1$Abundance),max(df1$Abundance)),
                                   xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                                   hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
                                   TT1 = c(x_lab, x_lab))
        }
    }
}
  dat_text$PD_Colon <- as.ordered(factor(dat_text$PD_Colon, levels = c("P", "D")))
  
 plot <- plot + scale_color_manual(values = sixdonors) + scale_fill_manual("SHIME transit", values = transitcolors, labels= c("Short", "Medium", "Long")) +
  scale_x_continuous(breaks=c(8,16,24), labels=c("Short", "Medium", "Long")) +
   # scale_x_discrete(labels = c("Short", "Medium", "Long"))+
   ylab(expression(paste("Absolute abundance (cells ",mL^{"-1"},")")))+ xlab("Transit time") + 
   scale_y_continuous(labels = scales::scientific) 
 
 if(i == "Acidaminococcus"){plot <- plot + 
   guides(fill=guide_legend(ncol=1, title.position = "top", hjust = 0.5),
          color=guide_legend(nrow=3, title.position = "top", hjust = 0.5)) +
   theme(legend.title.align = 0.5, legend.title = element_text(size=20), 
         legend.text = element_text(size=20), legend.position = c(0.5,0.35), legend.box = "horizontal")}else{
   plot <- plot + theme(legend.position = "none")}
         plot <- plot + theme(strip.background =element_rect(fill="#D6E0F4"),
         # legend.box = "horizontal",
         strip.text.x = element_text(size = 20),
         strip.text.y = element_blank(),
         # strip.text.y = element_text(size = 20, angle = 270),
         legend.title = element_text(size=20), legend.background = element_blank()) 
  
 
 
if(i == "X"){plot <- plot + geom_text(correlations)}else{
  # plot <- plot + geom_text(data= dat_text,  mapping = aes(x = as.numeric(TT1), y = as.numeric(value), label = label, family = 'sans'), size = 7)
  plot <- plot + geom_text(data= dat_text,  mapping = aes(x = xpos, y = ypos, label = label, 
                                        hjust=hjustvar,vjust=vjustvar, family = 'sans'), size = 7)
  }
QMPlist[[i]] <- plot; y = y+1
  }



# Proportional
phylobj.prop_2 = transform_sample_counts(phylobj.QMP, function(x) 100 * x/sum(x))
phylobj.prop_stable <- subset_samples(phylobj.prop_2, Ninetransit == 3)
Top15phylobj.prop <- aggregate_taxa(phylobj.prop_stable, level = "Genus")
Top15.prop <- psmelt(Top15phylobj.prop); unique(Top15.prop$OTU); unique(Top15.prop$OTU) %in% Top15levels
## Subset according to top most abundant genera
df.test <- data.frame();
for(i in Top.genera){
  df.test2 <- subset(Top15.prop, Genus == i)
  df.test <- rbind(df.test, df.test2)
}
Top15.prop <- df.test
x <- data.frame(x = unique(Top15$OTU) %in% Top15levels, genus = unique(Top15$OTU))  #check if labelled
Top15$OTU <- as.ordered(factor(Top15$OTU, levels = Top15levels))
x <- data.frame(x = unique(Top15.prop$OTU) %in% Top15levels, genus = unique(Top15.prop$OTU))  #check if labelled
Top15.prop$OTU <- as.ordered(factor(Top15.prop$OTU, levels = Top15levels))
## Add extra column days for graphs
Top.prop <- Top15.prop
Top.prop$Transit <- as.ordered(factor(Top.prop$Transit, levels = c("ST", "MT", "LT")))
Top.prop$PD_Colon <- as.ordered(factor(Top.prop$PD_Colon, levels = c("P", "D", "FS")))
Top.prop$Donor <- as.ordered(factor(Top.prop$Donor, levels = c("1", "2", "3", "4", "5", "6")))
Top.prop$Genus <- as.ordered(factor(Top.prop$Genus, levels = Top15levels))
## Add numbers
df.1 <- subset(Top.prop, PD_Colon == "P" & Transit == "ST"); df.1$TT <- 8; df.1$TT1 <- 8
df.2 <- subset(Top.prop, PD_Colon == "P" & Transit == "MT"); df.2$TT <- 16; df.2$TT1 <- 16
df.3 <- subset(Top.prop, PD_Colon == "P" & Transit == "LT"); df.3$TT <- 24; df.3$TT1 <- 24
df.4 <- subset(Top.prop, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 13; df.4$TT1 <- 8
df.5 <- subset(Top.prop, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 26; df.5$TT1 <- 16
df.6 <- subset(Top.prop, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 39; df.6$TT1 <- 24
# df.4 <- subset(Top.prop, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 8
# df.5 <- subset(Top.prop, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 16
# df.6 <- subset(Top.prop, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 24
Top.prop <- rbind(df.1, df.2, df.3, df.4, df.5, df.6)
rm(df.1, df.2, df.3, df.4, df.5, df.6)
dfcor2 <- data.frame(); PMPlist <- list(); y = 1

for(i in na.omit(unique(Top.prop$Genus))){
  df1 <- subset(Top.prop, Genus == i)
  if(abovezero == T){
    df1.1 <- subset(df1, Donor == 1); df1.1 <- subset(df1.1, Abundance > 0)
    if (dim(df1.1)[1] == 0) {df1 <- subset(df1, Donor !=1)}
    df1.2 <- subset(df1, Donor == 2); df1.2 <- subset(df1.2, Abundance > 0) 
    if (dim(df1.2)[1] == 0) {df1 <- subset(df1, Donor !=2)}
    df1.3 <- subset(df1, Donor == 3); df1.3 <- subset(df1.3, Abundance > 0)
    if (dim(df1.3)[1] == 0) {df1 <- subset(df1, Donor !=3)}
    df1.4 <- subset(df1, Donor == 4); df1.4 <- subset(df1.4, Abundance > 0) 
    if (dim(df1.4)[1] == 0) {df1 <- subset(df1, Donor !=4)}
    df1.5 <- subset(df1, Donor == 5); df1.5 <- subset(df1.5, Abundance > 0) 
    if (dim(df1.5)[1] == 0) {df1 <- subset(df1, Donor !=5)}
    df1.6 <- subset(df1, Donor == 6); df1.6 <- subset(df1.6, Abundance > 0) 
    if (dim(df1.6)[1] == 0) {df1 <- subset(df1, Donor !=6)}
    }
  for(x in c("P", "D")){
  df<- subset(df1, PD_Colon == x)
  t <- cor(df$Abundance, df$TT, method = "spearman")
  corr <- cor.test(df$Abundance, df$TT, method = 'spearman'); 
  
  if(is.na(corr[["p.value"]])){
    sig <- "-"
    }else{
      if(corr[["p.value"]]<0.0001){
        sig <- "****"
        }else{
          if(corr[["p.value"]]<0.001){
            sig <- "***"
            }else{
              if(corr[["p.value"]]<0.01){
                sig <- "**"
                }else{
                  if(corr[["p.value"]]<0.05){
                    sig<-"*"
                  }else{
                      sig<-""}}}}}
  
  dfcor2 <- rbind(dfcor2, data.frame(genus = i, cor = corr[["estimate"]][["rho"]], 
                 PD_Colon = x, pvalue = corr[["p.value"]], sig = sig, comm = "Proportional microbial community"))
  
  if(x == "P"){corP <- paste0("Ï = ",round(t, digits = 2)," ", sig, sep = ""); corrP <- corr[["p.value"]]}
  if(x == "D"){corD <- paste0("Ï = ",round(t, digits = 2)," ", sig, sep = ""); corrD <- corr[["p.value"]]}
  }
  
  #   dat_text <- data.frame(label = c(corP, corD),
  # PD_Colon   = c("P","D"),
  # value   = c(max(df1$Abundance), max(df1$Abundance)), 
  # TT = c(8,8))
  # dat_text$PD_Colon <- as.ordered(factor(dat_text$PD_Colon, levels = c("P", "D")))
  
plot <- papertheme(ggplot(df1, aes(y = Abundance, x = TT1))) + geom_boxplot(aes(fill = Transit)) + geom_jitter(aes(color = Donor), size =4) + 
      facet_grid(rows = vars(PD_Colon), labeller = labeller(PD_Colon = PD_Colon.labs), scales = "free_x") 

  if(is.na(corrP)){corrP <- 1} 
  if(is.na(corrD)){corrD <- 1} 
  if (corrP > 0.05 & corrD > 0.05) {
    plot <- plot
    dat_text <- data.frame(label = c("", ""), PD_Colon = c("P",
        "D"), value = c(max(df1$Abundance), max(df1$Abundance)),
        xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
        hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
        TT1 = c(x_lab, x_lab))
} else {
    if (corrP > 0.05 & corrD <= 0.05) {
        plot <- plot + geom_smooth(data = subset(df1, PD_Colon =="D"), 
                                   aes(y = Abundance, x = TT1), method = lm,
                                   color = "black")
        dat_text <- data.frame(label = c("", corD), PD_Colon = c("P",
            "D"), value = c(max(df1$Abundance), max(df1$Abundance)),
            xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
            hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
            TT1 = c(x_lab, x_lab))
    } else {
        if (corrP <= 0.05 & corrD > 0.05) {
            plot <- plot + geom_smooth(data = subset(df1,
                PD_Colon == "P"), aes(y = Abundance, x = TT1),
                method = lm, color = "black")
            dat_text <- data.frame(label = c(corP, ""),
                PD_Colon = c("P", "D"), value = c(max(df1$Abundance),
                  max(df1$Abundance)), 
                xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),TT1 = c(x_lab, x_lab))
        } else {
            plot <- plot + geom_smooth(aes(y = Abundance,
                x = TT1), method = lm, color = "black")
            dat_text <- data.frame(label = c(corP, corD),
                PD_Colon = c("P", "D"), value = c(max(df1$Abundance),
                  max(df1$Abundance)), 
                xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
                TT1 = c(x_lab, x_lab))
        }
    }
}
  dat_text$PD_Colon <- as.ordered(factor(dat_text$PD_Colon, levels = c("P", "D")))

plot <- plot + scale_color_manual(values = sixdonors) + scale_fill_manual(values = transitcolors) +
  scale_x_continuous(breaks=c(8,16,24), labels=c("Short", "Medium", "Long")) +
  ylab("Proportional abundance (%)")+ xlab("Transit time")  
  # scale_x_discrete(labels = c("Short", "Medium", "Long"))+
  
  if(i == "Akkermansia"){plot <- plot +  guides(fill=guide_legend(ncol=1, title.position = "top", hjust = 0.5),
          color=guide_legend(nrow=3, title.position = "top", hjust = 0.5)) +
   theme(legend.title.align = 0.5, legend.title = element_text(size=20), 
         legend.text = element_text(size=20), legend.position = c(0.5,0.8), legend.box = "horizontal")
  }else{
   plot <- plot + theme(legend.position = "none")}

 plot <- plot +  theme(strip.background =element_rect(fill="#D6E0F4"),
          strip.text.x = element_text(size = 20),
          strip.text.y = element_blank(), 
          # strip.text.y = element_text(size = 20, angle = 270),
          legend.title = element_text(size=20), legend.background = element_blank()) 
if(i == "Ruminococcus"){
    plot <- plot + theme(strip.text.y = element_text(size = 20, angle = 270))
  }
  
if(i == "X"){plot <- plot + geom_text(correlations)}else{
  # plot <- plot + geom_text(data= dat_text,  mapping = aes(x = as.numeric(TT1), y = as.numeric(value),
  #                                                         label = label, family = 'sans'), size = 7)
  plot <- plot + geom_text(data= dat_text,  mapping = aes(x = xpos, y = ypos, label = label, 
                                        hjust=hjustvar,vjust=vjustvar, family = 'sans'), size = 7)
}
PMPlist[[i]] <- plot; y = y+1
    }

#Combine abs and prop abundances in one
dfcor <- rbind(dfcor, dfcor2)
dfcor <- na.omit(dfcor); dfcor$comm <- as.ordered(factor(dfcor$comm, 
                              levels = c("Quantitative microbial community", "Proportional microbial community")))
dfcor$PD_Colon <- as.ordered(factor(dfcor$PD_Colon, levels = c("P", "D")))
dfcor$genus <- as.ordered(factor(dfcor$genus, levels = Top.genera))
plot <- papertheme(ggplot(dfcor, aes(x = PD_Colon, y = genus, fill =cor))) + geom_tile() +
  scale_fill_gradient2(na.value = "grey50",low="blue", high="red") + 
  geom_text(aes(label = sig)) +
  theme(axis.text.y = element_text(face="italic")) + ylab("Genus") + xlab("Colon vessel") + 
  scale_x_discrete(breaks=c("P","D"), labels=c("Proximal colon", "Distal colon")) +
  labs(fill = expression(rho)) +
  facet_grid(cols = vars(comm)) +theme(strip.background = element_rect(fill = "#D6E0F4"))
plot
dfcorall <- dfcor


# Combine plots
plots <- list(); x = 1
genera <- na.omit(unique(Top.prop$Genus))
genera <- sort(genera, decreasing = F)
for(i in genera){ 
  plot1 <- QMPlist[[i]]; plot2 <- PMPlist[[i]]
  plot <- ggarrange(plot1, plot2)
  plot <- annotate_figure(plot, text_grob(i, color = "black", face = "bold.italic", size = 25))
  # print(plot)
  plots[[i]] <- plot
  x = x +1
}

plots1 <- plots[1:8]; plots2 <- plots[9:16]
# ggarrange(plotlist = plots1, nrow = 2, ncol = 4)
# ggarrange(plotlist = plots2, nrow = 2, ncol = 4)
# ggarrange(plotlist = plots, nrow = 4, ncol = 4)








# OTU LEVEL
# Create OTU list with the genera
Top.OTU <- data.frame(); df.OTU <-df.combined
for(i in Top.genera){
  df.test <- subset(df.OTU, genus == i)
  Top.OTU <- rbind(Top.OTU, df.test)
}
Top.OTU <- unique(Top.OTU$rowname)

# ABSOLUTE ABUNDANCE
Top15.otu <- psmelt(subset_samples(phylobj.QMP.otu, Ninetransit == 3)); unique(Top15.otu$OTU)
## Subset according to top most abundant
df.test <- data.frame()
for(i in Top.OTU){
  df.test2 <- subset(Top15.otu, OTU == i)
  df.test <- rbind(df.test, df.test2)
}
Top15.otu <- df.test
unique(Top15.otu$OTU)
## Add extra column days for graphs
Top.otu <- Top15.otu
Top.otu$Transit <- as.ordered(factor(Top.otu$Transit, levels = c("ST", "MT", "LT")))
Top.otu$PD_Colon <- as.ordered(factor(Top.otu$PD_Colon, levels = c("P", "D", "FS")))
Top.otu$Donor <- as.ordered(factor(Top.otu$Donor, levels = c("1", "2", "3", "4", "5", "6")))

## Add numbers
df.1 <- subset(Top.otu, PD_Colon == "P" & Transit == "ST"); df.1$TT <- 8; df.1$TT1 <- 8
df.2 <- subset(Top.otu, PD_Colon == "P" & Transit == "MT"); df.2$TT <- 16; df.2$TT1 <- 16
df.3 <- subset(Top.otu, PD_Colon == "P" & Transit == "LT"); df.3$TT <- 24; df.3$TT1 <- 24
df.4 <- subset(Top.otu, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 13; df.4$TT1 <- 8
df.5 <- subset(Top.otu, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 26; df.5$TT1 <- 16
df.6 <- subset(Top.otu, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 39; df.6$TT1 <- 24
Top.otu <- rbind(df.1, df.2, df.3, df.4, df.5, df.6)
rm(df.1, df.2, df.3, df.4, df.5, df.6)


## Perform tests
dfcor.otu <- data.frame(); OTUQMPlist <- list()
for(i in na.omit(unique(Top.otu$OTU))){
  df1 <- subset(Top.otu, OTU == i)
  if(abovezero == T){
    df1.1 <- subset(df1, Donor == 1); df1.1 <- subset(df1.1, Abundance > 0)
    if (dim(df1.1)[1] == 0) {df1 <- subset(df1, Donor !=1)}
    df1.2 <- subset(df1, Donor == 2); df1.2 <- subset(df1.2, Abundance > 0) 
    if (dim(df1.2)[1] == 0) {df1 <- subset(df1, Donor !=2)}
    df1.3 <- subset(df1, Donor == 3); df1.3 <- subset(df1.3, Abundance > 0)
    if (dim(df1.3)[1] == 0) {df1 <- subset(df1, Donor !=3)}
    df1.4 <- subset(df1, Donor == 4); df1.4 <- subset(df1.4, Abundance > 0) 
    if (dim(df1.4)[1] == 0) {df1 <- subset(df1, Donor !=4)}
    df1.5 <- subset(df1, Donor == 5); df1.5 <- subset(df1.5, Abundance > 0) 
    if (dim(df1.5)[1] == 0) {df1 <- subset(df1, Donor !=5)}
    df1.6 <- subset(df1, Donor == 6); df1.6 <- subset(df1.6, Abundance > 0) 
    if (dim(df1.6)[1] == 0) {df1 <- subset(df1, Donor !=6)}
    }
  for(x in c("P", "D")){
  df<- subset(df1, PD_Colon == x)
  t <- cor(df$Abundance, df$TT, method = "spearman")
  
  if(is.na(t)){corr[["p.value"]] <- NA}else{
   corr <- cor.test(df$Abundance, df$TT, method = 'spearman')  
  }
  
  
  if(is.na(corr[["p.value"]])){
    sig <- ""
    }else{
      if(corr[["p.value"]]<0.0001){
        sig <- "****"
        }else{
          if(corr[["p.value"]]<0.001){
            sig <- "***"
            }else{
              if(corr[["p.value"]]<0.01){
                sig <- "**"
                }else{
                  if(corr[["p.value"]]<0.05){
                    sig<-"*"
                  }else{
                      sig<-""}}}}}
  
  if(is.na(t)){dfcor.otu <- rbind(dfcor.otu, data.frame(genus = i, cor = NA, 
                 PD_Colon = x, pvalue = corr[["p.value"]], sig = sig, comm = "Quantitative microbial community"))
  }else{
    dfcor.otu <- rbind(dfcor.otu, data.frame(genus = i, cor = corr[["estimate"]][["rho"]], 
                 PD_Colon = x, pvalue = corr[["p.value"]], sig = sig, comm = "Quantitative microbial community"))}
  
  if(x == "P"){corP <- paste0("Ï = ",round(t, digits = 2)," ", sig, sep = ""); corrP <- corr[["p.value"]]
  }else{corD <- paste0("Ï = ",round(t, digits = 2)," ", sig, sep = ""); corrD <- corr[["p.value"]]}
  }
  
  plot <- papertheme(ggplot(df1, aes(y = Abundance, x = TT1))) + geom_boxplot(aes(fill = Transit)) + geom_jitter(aes(color = Donor), size =4) + 
      facet_grid(rows = vars(PD_Colon), labeller = labeller(PD_Colon = PD_Colon.labs), scales = "free_x") 
  
  if(is.na(corrP)){corrP <- 1} 
  if(is.na(corrD)){corrD <- 1} 
  if (corrP > 0.05 & corrD > 0.05) {
    plot <- plot
    dat_text <- data.frame(label = c("", ""), PD_Colon = c("P",
        "D"), value = c(max(df1$Abundance), max(df1$Abundance)),
        xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
        hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
        TT1 = c(x_lab, x_lab))
} else {
    if (corrP > 0.05 & corrD <= 0.05) {
        plot <- plot + geom_smooth(data = subset(df1, PD_Colon =="D"), 
                                   aes(y = Abundance, x = TT1), method = lm,
                                   color = "black")
        dat_text <- data.frame(label = c("", corD), PD_Colon = c("P",
            "D"), value = c(max(df1$Abundance), max(df1$Abundance)),
            xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
            hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
            TT1 = c(x_lab, x_lab))
    } else {
        if (corrP <= 0.05 & corrD > 0.05) {
            plot <- plot + geom_smooth(data = subset(df1,
                PD_Colon == "P"), aes(y = Abundance, x = TT1),
                method = lm, color = "black")
            dat_text <- data.frame(label = c(corP, ""),
                PD_Colon = c("P", "D"), value = c(max(df1$Abundance),
                  max(df1$Abundance)),
                xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),TT1 = c(x_lab, x_lab))
        } else {
            plot <- plot + geom_smooth(aes(y = Abundance,
                x = TT1), method = lm, color = "black")
            dat_text <- data.frame(label = c(corP, corD),
                PD_Colon = c("P", "D"), value = c(max(df1$Abundance),
                  max(df1$Abundance)), 
                xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
                TT1 = c(x_lab, x_lab))
        }
    }
}
  dat_text$PD_Colon <- as.ordered(factor(dat_text$PD_Colon, levels = c("P", "D")))

plot <- plot + scale_color_manual(values = sixdonors) + 
  scale_fill_manual("SHIME transit", values = transitcolors, labels = c("Short", "Medium", "Long")) +
  # scale_x_discrete(labels = c("Short", "Medium", "Long"))+
  scale_x_continuous(breaks=c(8,16,24), labels=c("Short", "Medium", "Long")) +
  ylab(expression(paste("Absolute abundance (cells ",mL^{"-1"},")")))+ xlab("Transit time") + 
  scale_y_continuous(labels = scales::scientific) 
if(i == "Otu00029"){plot <- plot + 
   guides(fill=guide_legend(ncol=1, title.position = "top", hjust = 0.5),
          color=guide_legend(nrow=3, title.position = "top", hjust = 0.5)) +
   theme(legend.title.align = 0.5, legend.title = element_text(size=20), 
         legend.text = element_text(size=20), legend.position = c(0.5,0.85), legend.box = "horizontal")}else{
           if(i == "Otu00042"){plot <- plot + 
   guides(fill=guide_legend(ncol=1, title.position = "top", hjust = 0.5),
          color=guide_legend(nrow=3, title.position = "top", hjust = 0.5)) +
   theme(legend.title.align = 0.5, legend.title = element_text(size=20), 
         legend.text = element_text(size=20), legend.position = c(0.5,0.30), legend.box = "horizontal")}else{
   plot <- plot + theme(legend.position = "none")}}
         plot <- plot + theme(strip.background =element_rect(fill="#D6E0F4"),
         # legend.box = "horizontal",
         strip.text.x = element_text(size = 20),
         strip.text.y = element_blank(),
         # strip.text.y = element_text(size = 20, angle = 270),
         legend.title = element_text(size=20), legend.background = element_blank())
         
    


  if(i == "X"){plot <- plot + geom_text(correlations)}else{
  # plot <- plot + geom_text(data= dat_text,  mapping = aes(x = as.numeric(TT1), y = as.numeric(value), label = label, family = 'sans'), size = 7)
  plot <- plot + geom_text(data= dat_text,  mapping = aes(x = xpos, y = ypos, label = label, 
                                        hjust=hjustvar,vjust=vjustvar, family = 'sans'), size = 7)
}
 OTUQMPlist[[i]] <- plot; y = y+1
  
  }


# Proportional
Top15.prop.otu <- psmelt(subset_samples(phylobj.prop.otu, Ninetransit == 3)); unique(Top15.prop.otu$OTU)
## Subset according to top most abundant
df.test <- data.frame()
for(i in Top.OTU){
  df.test2 <- subset(Top15.prop.otu, OTU == i)
  df.test <- rbind(df.test, df.test2)
}
Top15.prop.otu <- df.test
unique(Top15.prop.otu$OTU)
## Add extra column days for graphs
Top.prop.otu <- Top15.prop.otu
Top.prop.otu$Transit <- as.ordered(factor(Top.prop.otu$Transit, levels = c("ST", "MT", "LT")))
Top.prop.otu$PD_Colon <- as.ordered(factor(Top.prop.otu$PD_Colon, levels = c("P", "D", "FS")))
Top.prop.otu$Donor <- as.ordered(factor(Top.prop.otu$Donor, levels = c("1", "2", "3", "4", "5", "6")))

## Add numbers
df.1 <- subset(Top.prop.otu, PD_Colon == "P" & Transit == "ST"); df.1$TT <- 8; df.1$TT1 <- 8
df.2 <- subset(Top.prop.otu, PD_Colon == "P" & Transit == "MT"); df.2$TT <- 16; df.2$TT1 <- 16
df.3 <- subset(Top.prop.otu, PD_Colon == "P" & Transit == "LT"); df.3$TT <- 24; df.3$TT1 <- 24
df.4 <- subset(Top.prop.otu, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 13; df.4$TT1 <- 8
df.5 <- subset(Top.prop.otu, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 26; df.5$TT1 <- 16
df.6 <- subset(Top.prop.otu, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 39; df.6$TT1 <- 24
Top.prop.otu <- rbind(df.1, df.2, df.3, df.4, df.5, df.6)
rm(df.1, df.2, df.3, df.4, df.5, df.6)


## Perform tests
dfcor.otu <- data.frame(); OTUPMPlist <- list()
for(i in na.omit(unique(Top.prop.otu$OTU))){
  df1 <- subset(Top.prop.otu, OTU == i)
  if(abovezero == T){
    df1.1 <- subset(df1, Donor == 1); df1.1 <- subset(df1.1, Abundance > 0)
    if (dim(df1.1)[1] == 0) {df1 <- subset(df1, Donor !=1)}
    df1.2 <- subset(df1, Donor == 2); df1.2 <- subset(df1.2, Abundance > 0) 
    if (dim(df1.2)[1] == 0) {df1 <- subset(df1, Donor !=2)}
    df1.3 <- subset(df1, Donor == 3); df1.3 <- subset(df1.3, Abundance > 0)
    if (dim(df1.3)[1] == 0) {df1 <- subset(df1, Donor !=3)}
    df1.4 <- subset(df1, Donor == 4); df1.4 <- subset(df1.4, Abundance > 0) 
    if (dim(df1.4)[1] == 0) {df1 <- subset(df1, Donor !=4)}
    df1.5 <- subset(df1, Donor == 5); df1.5 <- subset(df1.5, Abundance > 0) 
    if (dim(df1.5)[1] == 0) {df1 <- subset(df1, Donor !=5)}
    df1.6 <- subset(df1, Donor == 6); df1.6 <- subset(df1.6, Abundance > 0) 
    if (dim(df1.6)[1] == 0) {df1 <- subset(df1, Donor !=6)}
    }
  for(x in c("P", "D")){
  df<- subset(df1, PD_Colon == x)
  t <- cor(df$Abundance, df$TT, method = "spearman")
  
  if(is.na(t)){corr[["p.value"]] <- NA}else{
   corr <- cor.test(df$Abundance, df$TT, method = 'spearman')  
  }
  
  
  if(is.na(corr[["p.value"]])){
    sig <- ""
    }else{
      if(corr[["p.value"]]<0.0001){
        sig <- "****"
        }else{
          if(corr[["p.value"]]<0.001){
            sig <- "***"
            }else{
              if(corr[["p.value"]]<0.01){
                sig <- "**"
                }else{
                  if(corr[["p.value"]]<0.05){
                    sig<-"*"
                  }else{
                      sig<-""}}}}}
  
  if(is.na(t)){dfcor.otu <- rbind(dfcor.otu, data.frame(genus = i, cor = NA, 
                 PD_Colon = x, pvalue = corr[["p.value"]], sig = sig, comm = "Quantitative microbial community"))
  }else{
    dfcor.otu <- rbind(dfcor.otu, data.frame(genus = i, cor = corr[["estimate"]][["rho"]], 
                 PD_Colon = x, pvalue = corr[["p.value"]], sig = sig, comm = "Quantitative microbial community"))}
  
  if(x == "P"){corP <- paste0("Ï = ",round(t, digits = 2)," ", sig, sep = ""); corrP <- corr[["p.value"]]
  }else{corD <- paste0("Ï = ",round(t, digits = 2)," ", sig, sep = ""); corrD <- corr[["p.value"]]}
  }
  
  plot <- papertheme(ggplot(df1, aes(y = Abundance, x = TT1))) + geom_boxplot(aes(fill = Transit)) + geom_jitter(aes(color = Donor), size =4) + 
      facet_grid(rows = vars(PD_Colon), labeller = labeller(PD_Colon = PD_Colon.labs), scales = "free_x") 

  
  
  if(is.na(corrP)){corrP <- 1} 
  if(is.na(corrD)){corrD <- 1} 
  if (corrP > 0.05 & corrD > 0.05) {
    plot <- plot
    dat_text <- data.frame(label = c("", ""), PD_Colon = c("P",
        "D"), value = c(max(df1$Abundance), max(df1$Abundance)),
        xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
        TT1 = c(x_lab, x_lab))
} else {
    if (corrP > 0.05 & corrD <= 0.05) {
        plot <- plot + geom_smooth(data = subset(df1, PD_Colon =="D"), 
                                   aes(y = Abundance, x = TT1), method = lm,
                                   color = "black")
        dat_text <- data.frame(label = c("", corD), PD_Colon = c("P",
            "D"), value = c(max(df1$Abundance), max(df1$Abundance)),
            xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
            TT1 = c(x_lab, x_lab))
    } else {
        if (corrP <= 0.05 & corrD > 0.05) {
            plot <- plot + geom_smooth(data = subset(df1,
                PD_Colon == "P"), aes(y = Abundance, x = TT1),
                method = lm, color = "black")
            dat_text <- data.frame(label = c(corP, ""),
                PD_Colon = c("P", "D"), value = c(max(df1$Abundance),
                  max(df1$Abundance)), 
                xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),TT1 = c(x_lab, x_lab))
        } else {
            plot <- plot + geom_smooth(aes(y = Abundance,
                x = TT1), method = lm, color = "black")
            dat_text <- data.frame(label = c(corP, corD),
                PD_Colon = c("P", "D"), value = c(max(df1$Abundance),
                  max(df1$Abundance)), 
                xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),TT1 = c(x_lab, x_lab))
        }
    }
}
  dat_text$PD_Colon <- as.ordered(factor(dat_text$PD_Colon, levels = c("P", "D")))

plot <- plot + scale_color_manual(values = sixdonors) + 
  scale_fill_manual("SHIME transit", values = transitcolors, labels = c("Short", "Medium", "Long")) + 
  guides(fill=guide_legend(ncol=1, title.position = "top", hjust = 0.5), 
         color=guide_legend(nrow=3, title.position = "top", hjust = 0.5)) +
  # scale_x_discrete(labels = c("Short", "Medium", "Long"))+ 
  scale_x_continuous(breaks=c(8,16,24), labels=c("Short", "Medium", "Long")) +
  theme(legend.title.align = 0.5, legend.title = element_text(size=20), legend.text = element_text(size=20))+
  ylab("Proportional abundance (%)")+ xlab("Transit time") 
  
plt_legend <- g_legend(plot)

if(i == "Otu00144"){plot <- plot + 
   guides(fill=guide_legend(ncol=1, title.position = "top", hjust = 0.5),
          color=guide_legend(nrow=3, title.position = "top", hjust = 0.5)) +
   theme(legend.title.align = 0.5, legend.title = element_text(size=20), 
         legend.text = element_text(size=20), legend.position = c(0.5,0.30), legend.box = "horizontal")}else{
   plot <- plot + theme(legend.position = "none")}
         plot <- plot + theme(strip.background =element_rect(fill="#D6E0F4"),
         # legend.box = "horizontal",
         strip.text.x = element_text(size = 20),
         strip.text.y = element_blank(),
         # strip.text.y = element_text(size = 20, angle = 270),
         legend.title = element_text(size=20), legend.background = element_blank())
         
  if(i == "Otu00070" | i == "Otu00116" | i == "Otu00046"| i == "Otu00019"| i == "Otu00042" |  i == "Otu00025"| i == "Otu00008"| i == "Otu00038"|
      i == "Otu00189"| i == "Otu00073"){
    plot <- plot + theme(strip.text.y = element_text(size = 20, angle = 270))
  }




  if(i == "X"){plot <- plot + geom_text(correlations)}else{
  # plot <- plot + geom_text(data= dat_text,  mapping = aes(x = as.numeric(TT1), y = as.numeric(value), label = label, family = 'sans'), size = 7)
  plot <- plot + geom_text(data= dat_text,  mapping = aes(x = xpos, y = ypos, label = label, 
                                        hjust=hjustvar,vjust=vjustvar, family = 'sans'), size = 7)
}
 OTUPMPlist[[i]] <- plot; y = y+1
  
  }







# Combine OTUplots
otuplots <- list(); x = 1
genera <- na.omit(unique(Top.otu$OTU))
genera <- sort(genera, decreasing = F)
for(i in genera){ 
  plot1 <- OTUQMPlist[[i]]; plot2 <- OTUPMPlist[[i]]
  plot <- ggarrange(plot1, plot2)
  # print(plot)
  
  m =i
  if(stringr::str_detect(m, "Otu0000")){
      y <- str_replace(m, "Otu0000", "OTU")
    }else{if(stringr::str_detect(m, "Otu000")){
      y <- str_replace(m, "Otu000", "OTU")
    }else{if(stringr::str_detect(m, "Otu00")){
      y <- str_replace(m, "Otu00", "OTU")
    }else{if(stringr::str_detect(m, "Otu0")){
      y <- str_replace(m, "Otu0", "OTU")
    }}}}
  
  plot <- annotate_figure(plot, text_grob(y, color = "black", face = "bold", size = 25))
  # print(plot)
  otuplots[[i]] <- plot
  x = x +1
}

# ggarrange(plotlist = otuplots)



# Combine genera and OTU's
allplotslist <- list()
for(i in sort(Top.genera, decreasing = F)){
  plot1 <- plots[[i]]
dfje <- subset(df.combined, genus == i)
dfje <- subset(dfje, rowname!="Otu00102")
dfje <- subset(dfje, rowname!="Otu00216")
dfje <- subset(dfje, rowname!="Otu00261")
dfje <- subset(dfje, rowname!="Otu00078")
dfje <- subset(dfje, rowname!="Otu00133")
dfje <- subset(dfje, rowname!="Otu00214")
dfje <- subset(dfje, mean > 5)
plotjes <- list()
plotjes[[1]] <- plot1; y = 2
for(x in unique(dfje$rowname)){
  plot2 <- otuplots[[x]]
    plotjes[[y]] <- plot2;  y = y+1
}

if(i == "Ruminococcus"){xtralist <- plotjes
  plotjes[[y]] <- ggarrange(NULL, plt_legend ,nrow = 2)}
if(i == "Megamonas"){xtralist2 <- plotjes}
if(i == "Lachnospiraceae"){xtralist <- plotjes}
if(i == "Akkermansia"){xtralist2 <- plotjes}
if(i == "Akkermansia"){plotA <- ggarrange(plotlist = plotjes, nrow = 1, ncol = 2)}
if(i == "Clostridium_XlVa"){plotB <- ggarrange(plotlist = plotjes, nrow = 1, ncol = 3)}
if(i == "Clostridium_XlVa"){
  plotjes[[y]] <- ggarrange(NULL, plt_legend ,nrow = 2)}
if(i == "Cloacibacillus"){plotC <- ggarrange(plotlist = plotjes, nrow = 1, ncol = 2)}
if(i == "Pseudomonas"){plotD <- ggarrange(plotlist = plotjes, nrow = 1, ncol = 2)}

allplotslist[[i]] <- ggarrange(plotlist = plotjes, 
                               nrow = ceiling((nrow(dfje)+1)/4), 
                               ncol = 4)
}

ggarrange(plotlist =  allplotslist, ncol = 1)


plotE <- ggarrange(xtralist2[[1]], xtralist2[[2]],  xtralist[[1]],
                   xtralist[[2]], xtralist[[3]], xtralist[[4]], xtralist[[5]], xtralist[[6]], ncol = 4, nrow = 2)

# Rearrange them maybe?
plotAB <- ggarrange(plotA+ ggpubr::border(color = "grey", size = 1), 
                    plotB+ ggpubr::border(color = "grey", size = 1), ncol = 2, widths = c(2,3))
plotCD <- ggarrange(plotC+ ggpubr::border(color = "grey", size = 1), 
                    plotD+ ggpubr::border(color = "grey", size = 1), ncol = 2, widths = c(2,2))

ggarrange(plotE , allplotslist[[2]]+ ggpubr::border(color = "grey", size = 1), 
          allplotslist[[3]]+ ggpubr::border(color = "grey", size = 1),ncol = 1, heights = c(2,1,1))
```



# (Figure 4)
## All with lines
```{r Paperplot correlation colons split AND combined, fig.height=28, fig.width=30}
# fig.height = 6.5 per row and width = 8.75 per col

# order <- c("Veillonella", 	"Sutterella", 	"Ruminococcus", 	"Ruminococcaceae", 	"Roseburia", 	"Pseudomonas", 	"Prevotella", 	"Phascolarctobacterium", 	"Parabacteroides", 	"Mitsuokella", 	"Megasphaera", 	"Megamonas", 	"Lachnospiraceae", 	"Fusobacterium", 	"Fusicatenibacter", 	"Faecalibacterium", 	"Erysipelotrichaceae", 	"Enterobacteriaceae", 	"Eisenbergiella", 	"Dorea", 	"Dialister", 	"Desulfovibrio", 	"Coprococcus", 	"Clostridium_XlVa", 	"Clostridiales", 	"Cloacibacillus", 	"Blautia", 	"Bilophila", 	"Bifidobacterium", 	"Bacteroides", 	"Bacteria", 	"Anaerostipes", 	"Anaeroglobus", 	"Alistipes", 	"Akkermansia", 	"Acidaminococcus")



# df.OTU <-subset(df.combined, sum > 10)
df.OTU <-subset(df.combined, mean > 5); x_lab = 16; 
hjustv = -0.1; vjustv = 1.5; xInf <- -Inf; yInf <- Inf # Top Left (h0,v1, -inf, +inf)  Top Right (h1,v1, +inf, +inf)
topnum <- 16; abovezero = T
# Retrieve top most abundant genera
## Absolute
phylobj.QMP_stable <- subset_samples(phylobj.QMP, Ninetransit == 3)
Top15phylobj <- aggregate_taxa(phylobj.QMP_stable, level = "Genus")
Top15phylobj <- aggregate_top_taxa(phylobj.QMP_stable, top = topnum, "Genus")
## Proportional
phylobj.prop_2 = transform_sample_counts(phylobj.QMP, function(x) 100 * x/sum(x))
phylobj.prop_stable <- subset_samples(phylobj.prop_2, Ninetransit == 3)
phylobj.prop_stable <- aggregate_top_taxa(phylobj.prop_stable, top = topnum, "Genus")
## Take consensus top genera
Top.genera <- rbind(subset(as.data.frame(phylobj.prop_stable@tax_table), Genus != "Other")
                    # , subset(as.data.frame(Top15phylobj@tax_table), Genus != "Other")
                    )
## Remove insignificant genera
df <- data.frame()
for(i in significantlist){
  df2 <- subset(Top.genera, Genus == i)
  df <- rbind(df, df2)
}
Top.genera <- unique(df$Genus)
# Remove those that are not in the significant changes list (significantlist)
Top.genera <- sort(Top.genera, decreasing = T)

 # Top.genera <- c("Veillonella", "Ruminococcus", "Pseudomonas", "Prevotella", "Parabacteroides", "Mitsuokella", "Megamonas", 
 #                 # "Lachnospiraceae",
 #                 "Faecalibacterium",
 #                 "Enterobacteriaceae", "Dialister", "Clostridium_XlVa", "Bilophila", "Bifidobacterium", "Bacteroides", 
 #                 "Akkermansia", "Acidaminococcus")

# GENUS LEVEL
# Absolute
phylobj.QMP_stable <- subset_samples(phylobj.QMP, Ninetransit == 3)
Top15phylobj <- aggregate_taxa(phylobj.QMP_stable, level = "Genus")
Top15 <- psmelt(Top15phylobj); unique(Top15$OTU); unique(Top15$OTU) %in% Top15levels
## Subset according to top most abundant genera
df.test <- data.frame();
for(i in Top.genera){
  df.test2 <- subset(Top15, Genus == i)
  df.test <- rbind(df.test, df.test2)
}
Top15 <- df.test
x <- data.frame(x = unique(Top15$OTU) %in% Top15levels, genus = unique(Top15$OTU))  #check if labelled
# Top15$OTU <- as.ordered(factor(Top15$OTU, levels = Top15levels))
## Add extra column days for graphs
Top <- Top15
Top$Transit <- as.ordered(factor(Top$Transit, levels = c("ST", "MT", "LT")))
Top$PD_Colon <- as.ordered(factor(Top$PD_Colon, levels = c("P", "D", "FS")))
Top$Donor <- as.ordered(factor(Top$Donor, levels = c("1", "2", "3", "4", "5", "6")))
# Top$Genus <- as.ordered(factor(Top$Genus, levels = Top15levels))
## Add numbers
df.1 <- subset(Top, PD_Colon == "P" & Transit == "ST"); df.1$TT <- 8; df.1$TT1 <- 8
df.2 <- subset(Top, PD_Colon == "P" & Transit == "MT"); df.2$TT <- 16; df.2$TT1 <- 16
df.3 <- subset(Top, PD_Colon == "P" & Transit == "LT"); df.3$TT <- 24; df.3$TT1 <- 24
df.4 <- subset(Top, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 13; df.4$TT1 <- 8
df.5 <- subset(Top, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 26; df.5$TT1 <- 16
df.6 <- subset(Top, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 39; df.6$TT1 <- 24
# df.4 <- subset(Top, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 8
# df.5 <- subset(Top, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 16
# df.6 <- subset(Top, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 24

Top <- rbind(df.1, df.2, df.3, df.4, df.5, df.6)
rm(df.1, df.2, df.3, df.4, df.5, df.6)
## Perform test
dfcor <- data.frame(); df.otutest <- data.frame(); QMPlist <- list(); y =1
for(i in na.omit(unique(Top$Genus))){
    df1 <- subset(Top, Genus == i);  print(paste(i, max(df1$Abundance)))
  if(abovezero == T){
    df1.1 <- subset(df1, Donor == 1); df1.1 <- subset(df1.1, Abundance > 0)
    if (dim(df1.1)[1] == 0) {df1 <- subset(df1, Donor !=1)}
    df1.2 <- subset(df1, Donor == 2); df1.2 <- subset(df1.2, Abundance > 0) 
    if (dim(df1.2)[1] == 0) {df1 <- subset(df1, Donor !=2)}
    df1.3 <- subset(df1, Donor == 3); df1.3 <- subset(df1.3, Abundance > 0)
    if (dim(df1.3)[1] == 0) {df1 <- subset(df1, Donor !=3)}
    df1.4 <- subset(df1, Donor == 4); df1.4 <- subset(df1.4, Abundance > 0) 
    if (dim(df1.4)[1] == 0) {df1 <- subset(df1, Donor !=4)}
    df1.5 <- subset(df1, Donor == 5); df1.5 <- subset(df1.5, Abundance > 0) 
    if (dim(df1.5)[1] == 0) {df1 <- subset(df1, Donor !=5)}
    df1.6 <- subset(df1, Donor == 6); df1.6 <- subset(df1.6, Abundance > 0) 
    if (dim(df1.6)[1] == 0) {df1 <- subset(df1, Donor !=6)}
    }
  for(x in c("P", "D")){
  df<- subset(df1, PD_Colon == x)
  t <- cor(df$Abundance, df$TT, method = "spearman")
  corr <- cor.test(df$Abundance, df$TT, method = 'spearman'); 
  
  if(is.na(corr[["p.value"]])){
    sig <- "-"
    }else{
      if(corr[["p.value"]]<0.0001){
        sig <- "****"
        }else{
          if(corr[["p.value"]]<0.001){
            sig <- "***"
            }else{
              if(corr[["p.value"]]<0.01){
                sig <- "**"
                }else{
                  if(corr[["p.value"]]<0.05){
                    sig<-"*"
                  }else{
                      sig<-""}}}}}
  
  dfcor <- rbind(dfcor, data.frame(genus = i, cor = corr[["estimate"]][["rho"]], 
             PD_Colon = x, pvalue = corr[["p.value"]], sig = sig, comm = "Quantitative microbial community"))
  
  if(x == "P"){corP <- paste0("Ï = ",round(t, digits = 2)," ", sig, sep = ""); corrP <- corr[["p.value"]]
  }else{corD <- paste0("Ï = ",round(t, digits = 2)," ", sig, sep = ""); corrD <- corr[["p.value"]]}
  }
  
  # dat_text <- data.frame(label = c(corP, corD),
  # PD_Colon   = c("P","D"),
  # value   = c(max(df1$Abundance), max(df1$Abundance)), 
  # TT = c(8,8))

  
plot <- papertheme(ggplot(df1, aes(y = Abundance, x = TT1))) + geom_boxplot(aes(fill = Transit)) + geom_jitter(aes(color = Donor), size = 4) + 
    facet_grid(rows = vars(PD_Colon), labeller = labeller(PD_Colon = PD_Colon.labs), scales = "free_x") 

  if(is.na(corrP)){corrP <- 1} 
  if(is.na(corrD)){corrD <- 1} 
if(i == "Bifidobacterium" | i == "Bilophila" | i == "Dialister"|i == " Prevotella" | i == "Veillonella"){
      plot <- plot + geom_smooth(aes(y = Abundance, x = TT1), method = lm, color = "black")
      if(i == "Bifidobacterium"|i == "Veillonella"){dat_text <- data.frame(label = c(corP, corD),
                                                        PD_Colon = c("P", "D"),                                                                value = c(max(df1$Abundance),max(df1$Abundance)),
                             xpos = c(Inf, Inf), ypos =  c(Inf, Inf),
                             hjustvar = c(1.1, 1.1), vjustvar = c(1.5, 1.5),
                             TT1 = c(x_lab, x_lab))}else{
      dat_text <- data.frame(label = c(corP, corD),
                             PD_Colon = c("P", "D"), 
                             value = c(max(df1$Abundance),max(df1$Abundance)),
                             xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                             hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
                             TT1 = c(x_lab, x_lab))}
      }else{
        if(i == "Pseudomonas"){
          plot <- plot + geom_smooth(data = subset(df1,  PD_Colon == "P"), 
                                     aes(y = Abundance, x = TT1),
                                     method = lm, color = "black")
          dat_text <- data.frame(label = c(corP, ""),
                                 PD_Colon = c("P", "D"), 
                                 value = c(max(df1$Abundance),max(df1$Abundance)),
                                 xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                                 hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
                                 TT1 = c(x_lab, x_lab))
          }else{
            if(i == "Clostridium_XlVa"){
              plot <- plot + geom_smooth(data = subset(df1, PD_Colon =="D"), 
                                         aes(y = Abundance, x = TT1), method = lm,
                                         color = "black")
              dat_text <- data.frame(label = c("", corD), PD_Colon = c("P","D"),
                                     value = c(max(df1$Abundance), max(df1$Abundance)),
                                     xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                                     hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
                                     TT1 = c(x_lab, x_lab))
              }else{
                plot <- plot
                dat_text <- data.frame(label = c("", ""), 
                                       PD_Colon = c("P", "D"), 
                                       value = c(max(df1$Abundance), max(df1$Abundance)),
                                       xpos = c(xInf, xInf), ypos =  c(yInf, yInf),
                                       hjustvar = c(hjustv, hjustv), vjustvar = c(vjustv, vjustv),
                                       TT1 = c(x_lab, x_lab))
              }
          }
      }
  
    

  dat_text$PD_Colon <- as.ordered(factor(dat_text$PD_Colon, levels = c("P", "D")))
  
 plot <- plot + scale_color_manual(values = sixdonors) + scale_fill_manual("SHIME transit", values = transitcolors, labels= c("Short", "Medium", "Long")) +
  scale_x_continuous(breaks=c(8,16,24), labels=c("Short", "Medium", "Long")) +
   # scale_x_discrete(labels = c("Short", "Medium", "Long"))+
   ylab(expression(paste("Absolute abundance (cells ",mL^{"-1"},")")))+ xlab("Transit time") + 
   scale_y_continuous(labels = scales::scientific) 
 
 if(i == "Acidaminococcus"){plot <- plot + 
   guides(fill=guide_legend(ncol=1, title.position = "top", hjust = 0.5),
          color=guide_legend(nrow=3, title.position = "top", hjust = 0.5)) +
   theme(legend.title.align = 0.5, legend.title = element_text(size=20), 
         legend.text = element_text(size=20), legend.position = c(0.5,0.35), 
         strip.text.y = element_text(size = 20, angle = 270),legend.box = "horizontal")}else{
   plot <- plot + theme(legend.position = "none")}
         plot <- plot + theme(strip.background =element_rect(fill="#D6E0F4"),
         strip.text.x = element_text(size = 20),
         # strip.text.y = element_blank(),
         strip.text.y = element_text(size = 20, angle = 270),
         legend.title = element_text(size=20), legend.background = element_blank()) 
  
 
 
if(i == "X"){plot <- plot + geom_text(correlations)}else{
  # plot <- plot + geom_text(data= dat_text,  mapping = aes(x = as.numeric(TT1), y = as.numeric(value), label = label, family = 'sans'), size = 7)
  plot <- plot + geom_text(data= dat_text,  mapping = aes(x = xpos, y = ypos, label = label, 
                                        hjust=hjustvar,vjust=vjustvar, family = 'sans'), size = 7)
}
         
if(i == "Veillonella"){
  xtra_text <- data.frame(label = c("a", "b", "c", "a", "b", "c"),
  PD_Colon   = c("P","P","P","D","D","D"),
  TT1 = c(8,16,24,8,16,24),
  Transit  = c("ST", "MT", "LT", "ST", "MT", "LT"),
  Abundance     = c(max(df1$Abundance)+2e+08, 5e+08, 3e+08, 
                max(df1$Abundance),  5e+08, 3e+08))
  xtra_text$PD_Colon <- factor(xtra_text$PD_Colon, levels = c("P", "D"))
  xtra_text$Transit <- factor(xtra_text$Transit, levels = c("ST", "MT", "LT"))
  
  plot <- plot + geom_text(data= xtra_text,  mapping = aes(x = TT1, y = as.numeric(Abundance),
                                                          label = label, family = 'sans'), size = 7)}
         
if(i == "Faecalibacterium"){
  xtra_text <- data.frame(label = c("a", "b*", "b*", "a", "b*", "b*"),
  PD_Colon   = c("P","P","P","D","D","D"),
  TT1 = c(8,16,24,8,16,24),
  Transit  = c("ST", "MT", "LT", "ST", "MT", "LT"),
  Abundance     = c(4e+08, 1e+08, 1e+08, 
                2e+08,  1e+08, 1e+08))
  xtra_text$PD_Colon <- factor(xtra_text$PD_Colon, levels = c("P", "D"))
  xtra_text$Transit <- factor(xtra_text$Transit, levels = c("ST", "MT", "LT"))
  
  plot <- plot + geom_text(data= xtra_text,  mapping = aes(x = TT1, y = as.numeric(Abundance),
                                                          label = label, family = 'sans'), size = 7)}
         
if(i == "Ruminococcus"){
  xtra_text <- data.frame(label = c("a*", "a*", "b*", "a*", "a*", "a*"),
  PD_Colon   = c("P","P","P","D","D","D"),
  TT1 = c(8,16,24,8,16,24),
  Transit  = c("ST", "MT", "LT", "ST", "MT", "LT"),
  value     = c(6e+08, 6e+08, 6e+08, 
                max(df1$Abundance)+0.2e+09, max(df1$Abundance)+0.2e+09, max(df1$Abundance)+0.2e+09))
  xtra_text$PD_Colon <- factor(xtra_text$PD_Colon, levels = c("P", "D"))
  xtra_text$Transit <- factor(xtra_text$Transit, levels = c("ST", "MT", "LT"))
  
  plot <- plot + geom_text(data= xtra_text,  mapping = aes(x = TT1, y = as.numeric(value),
                                                          label = label, family = 'sans'), size = 7)}

         
if(i == "Pseudomonas"){
  xtra_text <- data.frame(label = c("a", "b*", "c", "a", "b*", "b"),
  PD_Colon   = c("P","P","P","D","D","D"),
  TT1 = c(8,16,24,8,16,24),
  Transit  = c("ST", "MT", "LT", "ST", "MT", "LT"),
  value     = c(2.2e+08, 2.5e+08, 2.8e+08, 
                max(df1$Abundance), max(df1$Abundance)+1.0e+08, max(df1$Abundance)+1.0e+08))
  xtra_text$PD_Colon <- factor(xtra_text$PD_Colon, levels = c("P", "D"))
  xtra_text$Transit <- factor(xtra_text$Transit, levels = c("ST", "MT", "LT"))
  
  plot <- plot + geom_text(data= xtra_text,  mapping = aes(x = TT1, y = as.numeric(value),
                                                          label = label, family = 'sans'), size = 7)}
         
         
if(i == "Prevotella"){
  xtra_text <- data.frame(label = c("a", "b", "c*", "a", "b", "c*"),
  PD_Colon   = c("P","P","P","D","D","D"),
  TT1 = c(8,16,24,8,16,24),
  Transit  = c("ST", "MT", "LT", "ST", "MT", "LT"),
  value     = c(2.5e+08, 2.5e+08, max(df1$Abundance)+2.2e+08, 
                2.5e+08, 2.5e+08, 5e+08))
  xtra_text$PD_Colon <- factor(xtra_text$PD_Colon, levels = c("P", "D"))
  xtra_text$Transit <- factor(xtra_text$Transit, levels = c("ST", "MT", "LT"))
  
  plot <- plot + geom_text(data= xtra_text,  mapping = aes(x = TT1, y = as.numeric(value),
                                                          label = label, family = 'sans'), size = 7)}
 
         
if(i == "Parabacteroides"){
  xtra_text <- data.frame(label = c("a*", "b*", "c", "a*", "b*", "b"),
  PD_Colon   = c("P","P","P","D","D","D"),
  TT1 = c(8,16,24,8,16,24),
  Transit  = c("ST", "MT", "LT", "ST", "MT", "LT"),
  value     = c(1e+08, max(df1$Abundance), max(df1$Abundance)+0.3e+08, 
                1e+08, max(df1$Abundance), max(df1$Abundance)))
  xtra_text$PD_Colon <- factor(xtra_text$PD_Colon, levels = c("P", "D"))
  xtra_text$Transit <- factor(xtra_text$Transit, levels = c("ST", "MT", "LT"))
  
  plot <- plot + geom_text(data= xtra_text,  mapping = aes(x = TT1, y = as.numeric(value),
                                                          label = label, family = 'sans'), size = 7)} 
         
         
if(i == "Mitsuokella"){
  xtra_text <- data.frame(label = c("a", "a", "a", "a", "a", "a"),
  PD_Colon   = c("P","P","P","D","D","D"),
  TT1 = c(8,16,24,8,16,24),
  Transit  = c("ST", "MT", "LT", "ST", "MT", "LT"),
  value     = c(max(df1$Abundance)+0.5e+08, max(df1$Abundance)+0.5e+08, max(df1$Abundance)+0.5e+08, 
                max(df1$Abundance), max(df1$Abundance), max(df1$Abundance)))
  xtra_text$PD_Colon <- factor(xtra_text$PD_Colon, levels = c("P", "D"))
  xtra_text$Transit <- factor(xtra_text$Transit, levels = c("ST", "MT", "LT"))
  
  plot <- plot + geom_text(data= xtra_text,  mapping = aes(x = TT1, y = as.numeric(value),
                                                          label = label, family = 'sans'), size = 7)}         

         
         
if(i == "Megamonas"){
  xtra_text <- data.frame(label = c("a", "a", "b", "a", "a", "a"),
  PD_Colon   = c("P","P","P","D","D","D"),
  TT1 = c(8,16,24,8,16,24),
  Transit  = c("ST", "MT", "LT", "ST", "MT", "LT"),
  value     = c(max(df1$Abundance), max(df1$Abundance), max(df1$Abundance)+1e+08, 
                max(df1$Abundance)-1e+08, max(df1$Abundance)-1e+08, max(df1$Abundance)-1e+08))
  xtra_text$PD_Colon <- factor(xtra_text$PD_Colon, levels = c("P", "D"))
  xtra_text$Transit <- factor(xtra_text$Transit, levels = c("ST", "MT", "LT"))
  
  plot <- plot + geom_text(data= xtra_text,  mapping = aes(x = TT1, y = as.numeric(value),
                                                          label = label, family = 'sans'), size = 7)}         
 
if(i == "Lachnospiraceae"){
  xtra_text <- data.frame(label = c("a*", "b*", "b*", "a*", "b*", "b*"),
  PD_Colon   = c("P","P","P","D","D","D"),
  TT1 = c(8,16,24,8,16,24),
  Transit  = c("ST", "MT", "LT", "ST", "MT", "LT"),
  value     = c(2.5e+08, max(df1$Abundance)+0.5e+08, max(df1$Abundance)+0.5e+08, 
                2.5e+08, max(df1$Abundance)+0.5e+08, max(df1$Abundance)+0.5e+08))
  xtra_text$PD_Colon <- factor(xtra_text$PD_Colon, levels = c("P", "D"))
  xtra_text$Transit <- factor(xtra_text$Transit, levels = c("ST", "MT", "LT"))
  
  plot <- plot + geom_text(data= xtra_text,  mapping = aes(x = TT1, y = as.numeric(value),
                                                          label = label, family = 'sans'), size = 7)}   

if(i == "Enterobacteriaceae"){
  xtra_text <- data.frame(label = c("a", "b*", "b*", "ab", "a*", "b*"),
  PD_Colon   = c("P","P","P","D","D","D"),
  TT1 = c(8,16,24,8,16,24),
  Transit  = c("ST", "MT", "LT", "ST", "MT", "LT"),
  value     = c(1e+09, max(df1$Abundance)+0.15e+09, max(df1$Abundance)+0.15e+09, 
                1e+09,1e+09,1e+09))
  xtra_text$PD_Colon <- factor(xtra_text$PD_Colon, levels = c("P", "D"))
  xtra_text$Transit <- factor(xtra_text$Transit, levels = c("ST", "MT", "LT"))
  
  plot <- plot + geom_text(data= xtra_text,  mapping = aes(x = TT1, y = as.numeric(value),
                                                          label = label, family = 'sans'), size = 7)}    
if(i == "Dialister"){
  xtra_text <- data.frame(label = c("a", "b", "b*", "a", "b", "b*"),
  PD_Colon   = c("P","P","P","D","D","D"),
  TT1 = c(8,16,24,8,16,24),
  Transit  = c("ST", "MT", "LT", "ST", "MT", "LT"),
  value     = c(1.4e+08, 1.7e+08, 2.3e+08, 
                1.4e+08, 1.7e+08, 2.3e+08))
  xtra_text$PD_Colon <- factor(xtra_text$PD_Colon, levels = c("P", "D"))
  xtra_text$Transit <- factor(xtra_text$Transit, levels = c("ST", "MT", "LT"))
  
  plot <- plot + geom_text(data= xtra_text,  mapping = aes(x = TT1, y = as.numeric(value),
                                                          label = label, family = 'sans'), size = 7)}
         
if(i == "Clostridium_XlVa"){
  xtra_text <- data.frame(label = c("a*", "b", "c*", "a*", "b", "b*"),
  PD_Colon   = c("P","P","P","D","D","D"),
  TT1 = c(8,16,24,8,16,24),
  Transit  = c("ST", "MT", "LT", "ST", "MT", "LT"),
  value     = c(6e+08, 7e+08, max(df1$Abundance)+0.2e+09, 
                6e+08, 7e+08, 7e+08))
  xtra_text$PD_Colon <- factor(xtra_text$PD_Colon, levels = c("P", "D"))
  xtra_text$Transit <- factor(xtra_text$Transit, levels = c("ST", "MT", "LT"))
  
  plot <- plot + geom_text(data= xtra_text,  mapping = aes(x = TT1, y = as.numeric(value),
                                                          label = label, family = 'sans'), size = 7)} 
         
         
if(i == "Bilophila"){
  xtra_text <- data.frame(label = c("a*", "b*", "b*", "a*", "b*", "b*"),
  PD_Colon   = c("P","P","P","D","D","D"),
  TT1 = c(8,16,24,8,16,24),
  Transit  = c("ST", "MT", "LT", "ST", "MT", "LT"),
  value     = c(5e+08, 8e+08, max(df1$Abundance)+0.1e+09, 
                7e+08, 9e+08, 7e+08))
  xtra_text$PD_Colon <- factor(xtra_text$PD_Colon, levels = c("P", "D"))
  xtra_text$Transit <- factor(xtra_text$Transit, levels = c("ST", "MT", "LT"))
  
  plot <- plot + geom_text(data= xtra_text,  mapping = aes(x = TT1, y = as.numeric(value),
                                                          label = label, family = 'sans'), size = 7)}          
if(i == "Bifidobacterium"){
  xtra_text <- data.frame(label = c("a", "b", "c*", "a", "b", "b*"),
  PD_Colon   = c("P","P","P","D","D","D"),
  TT1 = c(8,16,24,8,16,24),
  Transit  = c("ST", "MT", "LT", "ST", "MT", "LT"),
  value     = c(max(df1$Abundance)+0.5e+09, 8e+08, 4e+08, 
                max(df1$Abundance)+0.5e+09, 8e+08, 8e+08))
  xtra_text$PD_Colon <- factor(xtra_text$PD_Colon, levels = c("P", "D"))
  xtra_text$Transit <- factor(xtra_text$Transit, levels = c("ST", "MT", "LT"))
  
  plot <- plot + geom_text(data= xtra_text,  mapping = aes(x = TT1, y = as.numeric(value),
                                                          label = label, family = 'sans'), size = 7)}         

if(i == "Bacteroides"){
  xtra_text <- data.frame(label = c("a*", "b", "b", "a*", "b", "b"),
  PD_Colon   = c("P","P","P","D","D","D"),
  TT1 = c(8,16,24,8,16,24),
  Transit  = c("ST", "MT", "LT", "ST", "MT", "LT"),
  value     = c(2e+09, max(df1$Abundance)+0.25e+09, max(df1$Abundance)+0.25e+09, 
                2e+09, max(df1$Abundance), max(df1$Abundance)))
  xtra_text$PD_Colon <- factor(xtra_text$PD_Colon, levels = c("P", "D"))
  xtra_text$Transit <- factor(xtra_text$Transit, levels = c("ST", "MT", "LT"))
  
  plot <- plot + geom_text(data= xtra_text,  mapping = aes(x = TT1, y = as.numeric(value),
                                                          label = label, family = 'sans'), size = 7)}   
         
         
if(i == "Akkermansia"){
  xtra_text <- data.frame(label = c("a*", "a*", "a*", "a*", "a*", "a*"),
  PD_Colon   = c("P","P","P","D","D","D"),
  TT1 = c(8,16,24,8,16,24),
  Transit  = c("ST", "MT", "LT", "ST", "MT", "LT"),
  value     = c(1e+09, 1e+09, 1e+09, 
                1e+09, 1.5e+09, max(df1$Abundance)+0.5e+09))
  xtra_text$PD_Colon <- factor(xtra_text$PD_Colon, levels = c("P", "D"))
  xtra_text$Transit <- factor(xtra_text$Transit, levels = c("ST", "MT", "LT"))
  
  plot <- plot + geom_text(data= xtra_text,  mapping = aes(x = TT1, y = as.numeric(value),
                                                          label = label, family = 'sans'), size = 7)} 
         
if(i == "Acidaminococcus"){
  xtra_text <- data.frame(label = c("a", "a", "a", "a", "a", "a"),
  PD_Colon   = c("P","P","P","D","D","D"),
  TT1 = c(8,16,24,8,16,24),
  Transit  = c("ST", "MT", "LT", "ST", "MT", "LT"),
  value     = c(max(df1$Abundance)-0.5e+08, max(df1$Abundance), max(df1$Abundance)+0.6e+08, 
                2e+08, 2.35e+08, 2e+08))
  xtra_text$PD_Colon <- factor(xtra_text$PD_Colon, levels = c("P", "D"))
  xtra_text$Transit <- factor(xtra_text$Transit, levels = c("ST", "MT", "LT"))
  
  plot <- plot + geom_text(data= xtra_text,  mapping = aes(x = TT1, y = as.numeric(value),
                                                          label = label, family = 'sans'), size = 7)} 
          

         
         
QMPlist[[i]] <- plot; y = y+1
  }

ggarrange(plotlist = QMPlist)










# OTU LEVEL
# Create OTU list with the genera
Top.OTU <- data.frame(); df.OTU <-df.combined
for(i in Top.genera){
  df.test <- subset(df.OTU, genus == i)
  Top.OTU <- rbind(Top.OTU, df.test)
}
Top.OTU <- unique(Top.OTU$rowname)



# ABSOLUTE ABUNDANCE
Top15.otu <- psmelt(subset_samples(phylobj.QMP.otu, Ninetransit == 3))
## Subset according to top most abundant
df.test <- data.frame()
for(i in Top.OTU){
  df.test2 <- subset(Top15.otu, OTU == i)
  df.test <- rbind(df.test, df.test2)
}
Top15.otu <- df.test
unique(Top15.otu$OTU)

## Add extra column days for graphs
Top.otu <- Top15.otu
Top.otu$Transit <- as.ordered(factor(Top.otu$Transit, levels = c("ST", "MT", "LT")))
Top.otu$PD_Colon <- as.ordered(factor(Top.otu$PD_Colon, levels = c("P", "D", "FS")))
Top.otu$Donor <- as.ordered(factor(Top.otu$Donor, levels = c("1", "2", "3", "4", "5", "6")))

## Add numbers
df.1 <- subset(Top.otu, PD_Colon == "P" & Transit == "ST"); df.1$TT <- 8; df.1$TT1 <- 8
df.2 <- subset(Top.otu, PD_Colon == "P" & Transit == "MT"); df.2$TT <- 16; df.2$TT1 <- 16
df.3 <- subset(Top.otu, PD_Colon == "P" & Transit == "LT"); df.3$TT <- 24; df.3$TT1 <- 24
df.4 <- subset(Top.otu, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 13; df.4$TT1 <- 8
df.5 <- subset(Top.otu, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 26; df.5$TT1 <- 16
df.6 <- subset(Top.otu, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 39; df.6$TT1 <- 24
Top.otu <- rbind(df.1, df.2, df.3, df.4, df.5, df.6)
rm(df.1, df.2, df.3, df.4, df.5, df.6)


## Perform tests
dfcor.otu <- data.frame(); OTUQMPlistP <- list();  OTUQMPlistD <- list()
for(i in na.omit(unique(Top.otu$OTU))){
  
  df1 <- subset(Top.otu, OTU == i)
  if(abovezero == T){
    df1.1 <- subset(df1, Donor == 1); df1.1 <- subset(df1.1, Abundance > 0)
    if (dim(df1.1)[1] == 0) {df1 <- subset(df1, Donor !=1)}
    df1.2 <- subset(df1, Donor == 2); df1.2 <- subset(df1.2, Abundance > 0) 
    if (dim(df1.2)[1] == 0) {df1 <- subset(df1, Donor !=2)}
    df1.3 <- subset(df1, Donor == 3); df1.3 <- subset(df1.3, Abundance > 0)
    if (dim(df1.3)[1] == 0) {df1 <- subset(df1, Donor !=3)}
    df1.4 <- subset(df1, Donor == 4); df1.4 <- subset(df1.4, Abundance > 0) 
    if (dim(df1.4)[1] == 0) {df1 <- subset(df1, Donor !=4)}
    df1.5 <- subset(df1, Donor == 5); df1.5 <- subset(df1.5, Abundance > 0) 
    if (dim(df1.5)[1] == 0) {df1 <- subset(df1, Donor !=5)}
    df1.6 <- subset(df1, Donor == 6); df1.6 <- subset(df1.6, Abundance > 0) 
    if (dim(df1.6)[1] == 0) {df1 <- subset(df1, Donor !=6)}
  }
  
  
if(nrow(df1) >0){
  m = unique(df1$OTU)
  if(stringr::str_detect(m, "Otu0000")){
      y <- str_replace(m, "Otu0000", "OTU")
    }else{if(stringr::str_detect(m, "Otu000")){
      y <- str_replace(m, "Otu000", "OTU")
    }else{if(stringr::str_detect(m, "Otu00")){
      y <- str_replace(m, "Otu00", "OTU")
    }else{if(stringr::str_detect(m, "Otu0")){
      y <- str_replace(m, "Otu0", "OTU")
    }}}}
  
  # if(stringr::str_detect(m, "Otu0000")){
  #     y <- str_replace(m, "Otu0000", "")
  #   }else{if(stringr::str_detect(m, "Otu000")){
  #     y <- str_replace(m, "Otu000", "")
  #   }else{if(stringr::str_detect(m, "Otu00")){
  #     y <- str_replace(m, "Otu00", "")
  #   }else{if(stringr::str_detect(m, "Otu0")){
  #     y <- str_replace(m, "Otu0", "")
  #   }}}}

  df1$OTU <- y
 
for(x in c("P", "D")){
  df2 <- subset(df1, PD_Colon == x)
  df2$comb <- paste0(df2$PD_Colon, df2$Transit)
  df.avg <-  df2 %>% group_by(comb) %>% summarise(Abundance = mean(Abundance), OTU = OTU, TT = TT1, Transit = Transit, PD_Colon = PD_Colon)
  df.avg<- df.avg[!duplicated(df.avg),]
  
  df.ST <- as.numeric(subset(df.avg, Transit == "ST")$Abundance)
  df.MT <- as.numeric(subset(df.avg, Transit == "MT")$Abundance)
  df.LT <- as.numeric(subset(df.avg, Transit == "LT")$Abundance)  
  
  plot <- ggplot(df.avg, aes(x = TT, y = Abundance))
  
  if(df.MT > df.ST && df.MT > df.LT){plot <- plot + geom_line(color = "#8d6516", size = 2) #red
  }else{
    if(df.ST > df.MT && df.MT > df.LT){plot <- plot + geom_line(color = "#168d65", size =2)
    }else{
    if(df.MT > df.ST && df.LT > df.MT){plot <- plot + geom_line(color = "#65168d", size = 2)
    }else{
      plot <- plot + geom_line(color = "black", size = 2)
    }
  }}
  
  plot <- plot + theme_void() 
  plot <- plot + ylab(y)+  theme(strip.text.y = element_blank(), axis.title.y = element_text(angle = 0, size = 20))
  

  #   if(df.MT > df.ST && df.MT > df.LT){
  #   plot <- plot + theme(panel.background = element_rect(fill = 'lightblue', color = 'purple'))
  #   }else{plot <- plot}
  
  if(x == "P"){OTUQMPlistP[[i]] <- plot}else{OTUQMPlistD[[i]] <- plot}
  
  }}
  
} 
ggarrange(plotlist = OTUQMPlistP)


# Combine genera and OTU's
allplotslist <- list()
for(i in sort(Top.genera, decreasing = F)){
  plot1 <- QMPlist[[i]]
dfje <- subset(df.combined, genus == i)
dfje <- subset(dfje, rowname!="Otu00102")
dfje <- subset(dfje, rowname!="Otu00216")
dfje <- subset(dfje, rowname!="Otu00261")
dfje <- subset(dfje, rowname!="Otu00078")
dfje <- subset(dfje, rowname!="Otu00133")
dfje <- subset(dfje, rowname!="Otu00214")
dfje <- subset(dfje, rowname!="Otu00189")
dfje <- subset(dfje, mean > 5| rowname=="Otu00021"| rowname=="Otu00026")
plotjes <- list()
plotjes[[1]] <- plot1
plotsP <- list(); plotsD <- list()
for(x in unique(dfje$rowname)){
  plotsP[[x]] <- OTUQMPlistP[[x]] 
  plotsD[[x]] <- OTUQMPlistD[[x]]
}
plotsPP <- ggarrange(plotlist = plotsP, ncol = 1)
plotsPP <- ggarrange(NULL,plotsPP, NULL,NULL, nrow = 4, heights = c(0.5,length(unique(dfje$rowname)), 9 - length(unique(dfje$rowname)), 1))
plotsDD <- ggarrange(plotlist = plotsD, ncol = 1)
plotsDD <- ggarrange(plotsDD, NULL,NULL, nrow = 3, heights = c(length(unique(dfje$rowname)), 9 - length(unique(dfje$rowname)), 1))
plotjes[[2]] <- ggarrange(plotsPP, plotsDD, nrow = 2)
  

# if(i == "Ruminococcus"){xtralist <- plotjes
#   plotjes[[y]] <- ggarrange(NULL, plt_legend ,nrow = 2)}
# if(i == "Megamonas"){xtralist2 <- plotjes}
# if(i == "Acidaminococcus"){plotA <- ggarrange(plotlist = plotjes, nrow = 1, ncol = 2)}
# if(i == "Bilophila"){plotB <- ggarrange(plotlist = plotjes, nrow = 1, ncol = 2)}
# if(i == "Bilophila"){plotA <- ggarrange(plotlist = plotjes, nrow = 1, ncol = 2)}
# if(i == "Cloacibacillus"){plotB <- ggarrange(plotlist = plotjes, nrow = 1, ncol = 2)}
# if(i == "Megamonas"){plotC <- ggarrange(plotlist = plotjes, nrow = 1, ncol = 2)}
# if(i == "Pseudomonas"){plotD <- ggarrange(plotlist = plotjes, nrow = 1, ncol = 2)}

allplotslist[[i]] <- ggarrange(plotlist = plotjes,  
                               ncol = 2, widths = c(1,0.4))

allplotslist[[i]] <- annotate_figure(allplotslist[[i]], top = text_grob(i, face = "italic", size = 22))

}




ggarrange(plotlist = allplotslist, ncol = 4, nrow = 4)
```

#### Statistics
```{r}

phylobj.QMP_stable <- subset_samples(phylobj.QMP, Ninetransit == 3)
Top15phylobj <- aggregate_taxa(phylobj.QMP_stable, level = "Genus")
Top15 <- psmelt(Top15phylobj); unique(Top15$OTU); unique(Top15$OTU) %in% Top15levels
## Subset according to top most abundant genera
df.test <- data.frame();
for(i in c(Top.genera, "Faecalibacterium")){
  df.test2 <- subset(Top15, Genus == i)
  df.test <- rbind(df.test, df.test2)
}
Top15 <- df.test
x <- data.frame(x = unique(Top15$OTU) %in% Top15levels, genus = unique(Top15$OTU))  #check if labelled
# Top15$OTU <- as.ordered(factor(Top15$OTU, levels = Top15levels))
## Add extra column days for graphs
Top <- Top15
Top$Transit <- as.ordered(factor(Top$Transit, levels = c("ST", "MT", "LT")))
Top$PD_Colon <- as.ordered(factor(Top$PD_Colon, levels = c("P", "D", "FS")))
Top$Donor <- as.ordered(factor(Top$Donor, levels = c("1", "2", "3", "4", "5", "6")))
# Top$Genus <- as.ordered(factor(Top$Genus, levels = Top15levels))




# Non parametric to 1-way ANOVA, alternative - Kruskal-Wallis
correction <- "holm"

# Between transits
gglist <- list(); x = 1; df.pwc <- data.frame()
message(paste("Kruskal-Wallis tests with ",correction," correction between transit times"))
for(gener in na.omit(unique(Top$Genus))){
  my_dat <- subset(Top, Genus == gener)
  for (i in c("P", "D")) {
  my_data <- subset(my_dat, PD_Colon == i); my_data <- subset(my_data, Ninetransit == 3)
   if(abovezero == T){
    df1.1 <- subset(my_data, Donor == 1); df1.1 <- subset(df1.1, Abundance > 0)
    if (dim(df1.1)[1] == 0) {my_data <- subset(my_data, Donor !=1)}
    df1.2 <- subset(my_data, Donor == 2); df1.2 <- subset(df1.2, Abundance > 0) 
    if (dim(df1.2)[1] == 0) {my_data <- subset(my_data, Donor !=2)}
    df1.3 <- subset(my_data, Donor == 3); df1.3 <- subset(df1.3, Abundance > 0)
    if (dim(df1.3)[1] == 0) {my_data <- subset(my_data, Donor !=3)}
    df1.4 <- subset(my_data, Donor == 4); df1.4 <- subset(df1.4, Abundance > 0) 
    if (dim(df1.4)[1] == 0) {my_data <- subset(my_data, Donor !=4)}
    df1.5 <- subset(my_data, Donor == 5); df1.5 <- subset(df1.5, Abundance > 0) 
    if (dim(df1.5)[1] == 0) {my_data <- subset(my_data, Donor !=5)}
    df1.6 <- subset(my_data, Donor == 6); df1.6 <- subset(df1.6, Abundance > 0) 
    if (dim(df1.6)[1] == 0) {my_data <- subset(my_data, Donor !=6)}
    }
    my_data2 <- my_data
    res.kruskal <- my_data2 %>% rstatix::kruskal_test(Abundance ~ Transit)
    
    # Effect size The interpretation values commonly in published literature are: 
    # 0.01- < 0.06 (small effect), 0.06 - < 0.14 (moderate    effect) and >= 0.14 (large effect).
    effsize <- my_data2 %>% kruskal_effsize(Abundance ~ Transit)
    if(effsize$effsize>0.13){
      effsize <- paste0(", effsize= ", signif(effsize$effsize, digits = 5)," (large effect size is detected)")
    }else{if(effsize$effsize>0.06){
      effsize <- paste0(", effsize= ", signif(effsize$effsize, digits = 5)," (moderate effect size is detected)")
    }else{effsize <- paste0(", effsize= ", signif(effsize$effsize, digits = 5)," (WARNING: low effect size is detected)")}}
    
    # Pairwise comparisons using Wilcoxonâs test:
    pwc <- my_data2 %>%  wilcox_test(Abundance ~ Transit, p.adjust.method = correction)
    pwc <- subset(pwc, p.adj.signif != "ns")
    pwc$Vessel <- i; pwc$Transit <- gener; df.pwc <- rbind(df.pwc, pwc)
    # print(pwc) # print pairwise comparisons
    
    if(res.kruskal$p < 0.05){
      message(crayon::green(paste0("Kruskal-Wallis: ",gener,", in ",i, ", p = ",res.kruskal$p, effsize)))
      }else{
      message(crayon::red(paste0("Kruskal-Wallis: ",gener,", in ",i, ", p = ",res.kruskal$p, effsize)))}
    
    # Visualization: box plots with p-values
    pwc <- pwc %>% add_xy_position(x = "Transit")
    gglist[[x]]<- ggboxplot(my_data2, x = "Transit", y = "Abundance") +
    stat_pvalue_manual(pwc, hide.ns = TRUE) +
    labs(title = paste0(gener,", Vessel ",i),
    subtitle = get_test_label(res.kruskal, detailed = TRUE),
    caption = get_pwc_label(pwc))
    x = x +1
    
}
}






# Between vessels
gglist <- list(); x = 1; df.pwc1 <- data.frame()
message(paste("Kruskal-Wallis tests with ",correction," correction between vessels"))
for(gener in na.omit(unique(Top$Genus))){
my_dat <- subset(Top, Genus == gener)
for (j in c("ST", "MT", "LT")) {
    my_data <- subset(my_dat, Transit == j); my_data <- subset(my_data, Ninetransit == 3)
   if(abovezero == T){
    df1.1 <- subset(my_data, Donor == 1); df1.1 <- subset(df1.1, Abundance > 0)
    if (dim(df1.1)[1] == 0) {my_data <- subset(my_data, Donor !=1)}
    df1.2 <- subset(my_data, Donor == 2); df1.2 <- subset(df1.2, Abundance > 0) 
    if (dim(df1.2)[1] == 0) {my_data <- subset(my_data, Donor !=2)}
    df1.3 <- subset(my_data, Donor == 3); df1.3 <- subset(df1.3, Abundance > 0)
    if (dim(df1.3)[1] == 0) {my_data <- subset(my_data, Donor !=3)}
    df1.4 <- subset(my_data, Donor == 4); df1.4 <- subset(df1.4, Abundance > 0) 
    if (dim(df1.4)[1] == 0) {my_data <- subset(my_data, Donor !=4)}
    df1.5 <- subset(my_data, Donor == 5); df1.5 <- subset(df1.5, Abundance > 0) 
    if (dim(df1.5)[1] == 0) {my_data <- subset(my_data, Donor !=5)}
    df1.6 <- subset(my_data, Donor == 6); df1.6 <- subset(df1.6, Abundance > 0) 
    if (dim(df1.6)[1] == 0) {my_data <- subset(my_data, Donor !=6)}
    }
    my_data2 <- my_data  
  
    res.kruskal <- my_data2 %>% rstatix::kruskal_test(Abundance ~ PD_Colon)
    test1<- subset(my_data2, PD_Colon == "P"); test<- subset(my_data2, PD_Colon == "D")
    # Effect size The interpretation values commonly in published literature are: 
    # 0.01- < 0.06 (small effect), 0.06 - < 0.14 (moderate    effect) and >= 0.14 (large effect).
    effsize <- my_data2 %>% kruskal_effsize(Abundance ~ PD_Colon)
    if(effsize$effsize>0.13){
      effsize <- paste0(", effsize= ", signif(effsize$effsize, digits = 5)," (large effect size is detected)")
    }else{if(effsize$effsize>0.06){ 
      effsize <- paste0(", effsize= ", signif(effsize$effsize, digits = 5)," (moderate effect size is detected)")
    }else{effsize <- paste0(", effsize= ", signif(effsize$effsize, digits = 5)," (WARNING: low effect size is detected)")}}
    
    # Pairwise comparisons using Wilcoxonâs test:
    pwc <- my_data2 %>%  wilcox_test(Abundance ~PD_Colon, p.adjust.method = "bonferroni", paired = F)
    pwc <- subset(pwc, p != "ns")
    pwc$Genus <- gener; pwc$Transit <- j; df.pwc1 <- rbind(df.pwc1, pwc)
    # print(pwc) # print pairwise comparisons
    
       
    if(res.kruskal$p < 0.05){
      message(crayon::green(paste0("Kruskal-Wallis: Transit ",j,", p = ",res.kruskal$p, effsize)))
      }else{
      message(crayon::red(paste0("Kruskal-Wallis: Transit ",j,", p = ",res.kruskal$p, effsize)))}
    
    # Visualization: box plots with p-values
    pwc <- pwc %>% add_xy_position(x = "PD_Colon")
    gglist[[x]]<- ggboxplot(my_data2, x = "PD_Colon", y = "Abundance") +
    stat_pvalue_manual(pwc, hide.ns = TRUE) +
    labs(title = paste0(gener,", ",j),
    subtitle = get_test_label(res.kruskal, detailed = TRUE),
    caption = get_pwc_label(pwc))
    x = x +1}}; 
ggarrange(plotlist = gglist, nrow = 2, ncol = 3)
df.pwc.total <- df.pwc; df.pwc1.total <- df.pwc1
View(df.pwc.total); View(df.pwc1.total)
```



## Paperplot genera correlations and OTU of Top most abundant genera
```{r Paperplot correlation colons split AND combined, fig.height=23, fig.width=20}
# order <- c("Veillonella", 	"Sutterella", 	"Ruminococcus", 	"Ruminococcaceae", 	"Roseburia", 	"Pseudomonas", 	"Prevotella", 	"Phascolarctobacterium", 	"Parabacteroides", 	"Mitsuokella", 	"Megasphaera", 	"Megamonas", 	"Lachnospiraceae", 	"Fusobacterium", 	"Fusicatenibacter", 	"Faecalibacterium", 	"Erysipelotrichaceae", 	"Enterobacteriaceae", 	"Eisenbergiella", 	"Dorea", 	"Dialister", 	"Desulfovibrio", 	"Coprococcus", 	"Clostridium_XlVa", 	"Clostridiales", 	"Cloacibacillus", 	"Blautia", 	"Bilophila", 	"Bifidobacterium", 	"Bacteroides", 	"Bacteria", 	"Anaerostipes", 	"Anaeroglobus", 	"Alistipes", 	"Akkermansia", 	"Acidaminococcus")
# df.OTU <-subset(df.combined, sum > 10)
df.OTU <-subset(df.combined, mean > 5);
topnum <- 15

# Retrieve top most abundant genera
## Absolute
phylobj.QMP_stable <- subset_samples(phylobj.QMP, Ninetransit == 3)
Top15phylobj <- aggregate_taxa(phylobj.QMP_stable, level = "Genus")
Top15phylobj <- aggregate_top_taxa(phylobj.QMP_stable, top = topnum, "Genus")
## Proportional
phylobj.prop_2 = transform_sample_counts(phylobj.QMP, function(x) 100 * x/sum(x))
phylobj.prop_stable <- subset_samples(phylobj.prop_2, Ninetransit == 3)
phylobj.prop_stable <- aggregate_top_taxa(phylobj.prop_stable, top = topnum, "Genus")
## Take consensus top genera
Top.genera <- rbind(subset(as.data.frame(phylobj.prop_stable@tax_table), Genus != "Other"), 
                    subset(as.data.frame(Top15phylobj@tax_table), Genus != "Other"))
## Remove unsignificant genera
df <- data.frame()
for(i in significantlist){
  df2 <- subset(Top.genera, Genus == i)
  df <- rbind(df, df2)
}
Top.genera <- unique(df$Genus)
# Remove those that are not in the significant changes list (significantlist)
Top.genera <- sort(Top.genera, decreasing = T)

# GENUS LEVEL
# Absolute
phylobj.QMP_stable <- subset_samples(phylobj.QMP, Ninetransit == 3)
Top15phylobj <- aggregate_taxa(phylobj.QMP_stable, level = "Genus")
Top15 <- psmelt(Top15phylobj); unique(Top15$OTU); unique(Top15$OTU) %in% Top15levels

## Subset according to top most abundant genera
df.test <- data.frame();
for(i in Top.genera){
  df.test2 <- subset(Top15, Genus == i)
  df.test <- rbind(df.test, df.test2)
}
Top15 <- df.test
x <- data.frame(x = unique(Top15$OTU) %in% Top15levels, genus = unique(Top15$OTU))  #check if labelled
Top15$OTU <- as.ordered(factor(Top15$OTU, levels = Top15levels))

## Add extra column days for graphs
Top <- Top15
Top$Transit <- as.ordered(factor(Top$Transit, levels = c("ST", "MT", "LT")))
Top$PD_Colon <- as.ordered(factor(Top$PD_Colon, levels = c("P", "D", "FS")))
Top$Donor <- as.ordered(factor(Top$Donor, levels = c("1", "2", "3", "4", "5", "6")))
Top$Genus <- as.ordered(factor(Top$Genus, levels = Top15levels))

## Add numbers
df.1 <- subset(Top, PD_Colon == "P" & Transit == "ST"); df.1$TT <- 8
df.2 <- subset(Top, PD_Colon == "P" & Transit == "MT"); df.2$TT <- 16
df.3 <- subset(Top, PD_Colon == "P" & Transit == "LT"); df.3$TT <- 24
df.4 <- subset(Top, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 13
df.5 <- subset(Top, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 26
df.6 <- subset(Top, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 39
Top <- rbind(df.1, df.2, df.3, df.4, df.5, df.6)
rm(df.1, df.2, df.3, df.4, df.5, df.6)

## Perform test
dfcor <- data.frame(); df.otutest <- data.frame()
for(i in na.omit(unique(Top$Genus))){
  df1 <- subset(Top, Genus == i);  df1 <- subset(df1, Abundance > 0)
  for(x in c("P", "D")){
  df<- subset(df1, PD_Colon == x)
  t <- cor(df$Abundance, df$TT, method = "spearman")
  corr <- cor.test(df$Abundance, df$TT, method = 'spearman'); 
  
  if(is.na(corr[["p.value"]])){
    sig <- "-"
    }else{
      if(corr[["p.value"]]<0.0001){
        sig <- "****"
        }else{
          if(corr[["p.value"]]<0.001){
            sig <- "***"
            }else{
              if(corr[["p.value"]]<0.01){
                sig <- "**"
                }else{
                  if(corr[["p.value"]]<0.05){
                    sig<-"*"
                  }else{
                      sig<-""}}}}}
  
  dfcor <- rbind(dfcor, data.frame(genus = i, cor = corr[["estimate"]][["rho"]], 
                 PD_Colon = x, pvalue = corr[["p.value"]], sig = sig, comm = "Quantitative microbial community"))
  }}

# Proportional
phylobj.prop_2 = transform_sample_counts(phylobj.QMP, function(x) 100 * x/sum(x))
phylobj.prop_stable <- subset_samples(phylobj.prop_2, Ninetransit == 3)
Top15phylobj.prop <- aggregate_taxa(phylobj.prop_stable, level = "Genus")
Top15.prop <- psmelt(Top15phylobj.prop); unique(Top15.prop$OTU); unique(Top15.prop$OTU) %in% Top15levels
## Subset according to top most abundant genera
df.test <- data.frame();
for(i in Top.genera){
  df.test2 <- subset(Top15.prop, Genus == i)
  df.test <- rbind(df.test, df.test2)
}
Top15.prop <- df.test
x <- data.frame(x = unique(Top15$OTU) %in% Top15levels, genus = unique(Top15$OTU))  #check if labelled
Top15$OTU <- as.ordered(factor(Top15$OTU, levels = Top15levels))
x <- data.frame(x = unique(Top15.prop$OTU) %in% Top15levels, genus = unique(Top15.prop$OTU))  #check if labelled
Top15.prop$OTU <- as.ordered(factor(Top15.prop$OTU, levels = Top15levels))
## Add extra column days for graphs
Top.prop <- Top15.prop
Top.prop$Transit <- as.ordered(factor(Top.prop$Transit, levels = c("ST", "MT", "LT")))
Top.prop$PD_Colon <- as.ordered(factor(Top.prop$PD_Colon, levels = c("P", "D", "FS")))
Top.prop$Donor <- as.ordered(factor(Top.prop$Donor, levels = c("1", "2", "3", "4", "5", "6")))
Top.prop$Genus <- as.ordered(factor(Top.prop$Genus, levels = Top15levels))
## Add numbers
df.1 <- subset(Top.prop, PD_Colon == "P" & Transit == "ST"); df.1$TT <- 8
df.2 <- subset(Top.prop, PD_Colon == "P" & Transit == "MT"); df.2$TT <- 16
df.3 <- subset(Top.prop, PD_Colon == "P" & Transit == "LT"); df.3$TT <- 24
df.4 <- subset(Top.prop, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 13
df.5 <- subset(Top.prop, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 26
df.6 <- subset(Top.prop, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 39
Top.prop <- rbind(df.1, df.2, df.3, df.4, df.5, df.6)
rm(df.1, df.2, df.3, df.4, df.5, df.6)
dfcor2 <- data.frame()
for(i in na.omit(unique(Top.prop$Genus))){
  df1 <- subset(Top.prop, Genus == i);  df1 <- subset(df1, Abundance > 0)
  for(x in c("P", "D")){
  df<- subset(df1, PD_Colon == x)
  t <- cor(df$Abundance, df$TT, method = "spearman")
  corr <- cor.test(df$Abundance, df$TT, method = 'spearman'); 
  
  if(is.na(corr[["p.value"]])){
    sig <- "-"
    }else{
      if(corr[["p.value"]]<0.0001){
        sig <- "****"
        }else{
          if(corr[["p.value"]]<0.001){
            sig <- "***"
            }else{
              if(corr[["p.value"]]<0.01){
                sig <- "**"
                }else{
                  if(corr[["p.value"]]<0.05){
                    sig<-"*"
                  }else{
                      sig<-""}}}}}
  
  dfcor2 <- rbind(dfcor2, data.frame(genus = i, cor = corr[["estimate"]][["rho"]], 
                 PD_Colon = x, pvalue = corr[["p.value"]], sig = sig, comm = "Proportional microbial community"))
  }}

#Combine abs and prop abundances in one
dfcor <- rbind(dfcor, dfcor2)
dfcor <- na.omit(dfcor); dfcor$comm <- as.ordered(factor(dfcor$comm, 
                              levels = c("Quantitative microbial community", "Proportional microbial community")))
dfcor$PD_Colon <- as.ordered(factor(dfcor$PD_Colon, levels = c("P", "D")))
dfcor$genus <- as.ordered(factor(dfcor$genus, levels = Top.genera))
plot <- papertheme(ggplot(dfcor, aes(x = PD_Colon, y = genus, fill =cor))) + geom_tile() +
  scale_fill_gradient2(na.value = "grey50",low="blue", high="red") + 
  geom_text(aes(label = sig)) +
  theme(axis.text.y = element_text(face="italic")) + ylab("Genus") + xlab("Colon vessel") + 
  scale_x_discrete(breaks=c("P","D"), labels=c("Proximal colon", "Distal colon")) +
  labs(fill = expression(rho)) +
  facet_grid(cols = vars(comm)) +theme(strip.background = element_rect(fill = "#D6E0F4"))
plot
dfcorall <- dfcor



# OTU LEVEL
# Create OTU list with the genera
Top.OTU <- data.frame(); df.OTU <-df.combined
for(i in Top.genera){
  df.test <- subset(df.OTU, genus == i)
  Top.OTU <- rbind(Top.OTU, df.test)
}
Top.OTU <- unique(Top.OTU$rowname)
# Absolute
Top15.otu <- psmelt(subset_samples(phylobj.QMP.otu, Ninetransit == 3)); unique(Top15.otu$OTU)
## Subset according to top most abundant
df.test <- data.frame()
for(i in Top.OTU){
  df.test2 <- subset(Top15.otu, OTU == i)
  df.test <- rbind(df.test, df.test2)
}
Top15.otu <- df.test
unique(Top15.otu$OTU)
## Add extra column days for graphs
Top.otu <- Top15.otu
Top.otu$Transit <- as.ordered(factor(Top.otu$Transit, levels = c("ST", "MT", "LT")))
Top.otu$PD_Colon <- as.ordered(factor(Top.otu$PD_Colon, levels = c("P", "D", "FS")))
Top.otu$Donor <- as.ordered(factor(Top.otu$Donor, levels = c("1", "2", "3", "4", "5", "6")))
## Add numbers
df.1 <- subset(Top.otu, PD_Colon == "P" & Transit == "ST"); df.1$TT <- 8
df.2 <- subset(Top.otu, PD_Colon == "P" & Transit == "MT"); df.2$TT <- 16
df.3 <- subset(Top.otu, PD_Colon == "P" & Transit == "LT"); df.3$TT <- 24
df.4 <- subset(Top.otu, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 13
df.5 <- subset(Top.otu, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 26
df.6 <- subset(Top.otu, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 39
Top.otu <- rbind(df.1, df.2, df.3, df.4, df.5, df.6)
rm(df.1, df.2, df.3, df.4, df.5, df.6)
## Perform tests
dfcor.otu <- data.frame()
for(i in na.omit(unique(Top.otu$OTU))){
  df1 <- subset(Top.otu, OTU == i);  df1 <- subset(df1, Abundance > 0)
  for(x in c("P", "D")){
  df<- subset(df1, PD_Colon == x)
  t <- cor(df$Abundance, df$TT, method = "spearman")
  if(is.na(t)){corr[["p.value"]] <- NA}else{
   corr <- cor.test(df$Abundance, df$TT, method = 'spearman')  
  }
  
  
  if(is.na(corr[["p.value"]])){
    sig <- ""
    }else{
      if(corr[["p.value"]]<0.0001){
        sig <- "****"
        }else{
          if(corr[["p.value"]]<0.001){
            sig <- "***"
            }else{
              if(corr[["p.value"]]<0.01){
                sig <- "**"
                }else{
                  if(corr[["p.value"]]<0.05){
                    sig<-"*"
                  }else{
                      sig<-""}}}}}
  
  if(is.na(t)){dfcor.otu <- rbind(dfcor.otu, data.frame(genus = i, cor = NA, 
                 PD_Colon = x, pvalue = corr[["p.value"]], sig = sig, comm = "Quantitative microbial community"))
  }else{
    dfcor.otu <- rbind(dfcor.otu, data.frame(genus = i, cor = corr[["estimate"]][["rho"]], 
                 PD_Colon = x, pvalue = corr[["p.value"]], sig = sig, comm = "Quantitative microbial community"))}
  }}


# Proportional
Top15.prop.otu <- psmelt(subset_samples(phylobj.prop.otu, Ninetransit == 3)); unique(Top15.prop.otu$OTU)
## Subset according to top most abundant
df.test <- data.frame()
for(i in Top.OTU){
  df.test2 <- subset(Top15.prop.otu, OTU == i)
  df.test <- rbind(df.test, df.test2)
}
Top15.prop.otu <- df.test
unique(Top15.prop.otu$OTU)
## Add extra column days for graphs
Top.prop.otu <- Top15.prop.otu
Top.prop.otu$Transit <- as.ordered(factor(Top.prop.otu$Transit, levels = c("ST", "MT", "LT")))
Top.prop.otu$PD_Colon <- as.ordered(factor(Top.prop.otu$PD_Colon, levels = c("P", "D", "FS")))
Top.prop.otu$Donor <- as.ordered(factor(Top.prop.otu$Donor, levels = c("1", "2", "3", "4", "5", "6")))
## Add numbers
df.1 <- subset(Top.prop.otu, PD_Colon == "P" & Transit == "ST"); df.1$TT <- 8
df.2 <- subset(Top.prop.otu, PD_Colon == "P" & Transit == "MT"); df.2$TT <- 16
df.3 <- subset(Top.prop.otu, PD_Colon == "P" & Transit == "LT"); df.3$TT <- 24
df.4 <- subset(Top.prop.otu, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 13
df.5 <- subset(Top.prop.otu, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 26
df.6 <- subset(Top.prop.otu, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 39
Top.prop.otu <- rbind(df.1, df.2, df.3, df.4, df.5, df.6)
rm(df.1, df.2, df.3, df.4, df.5, df.6)
dfcor2.otu <- data.frame()
for(i in na.omit(unique(Top.prop.otu$OTU))){
  df1 <- subset(Top.prop.otu, OTU == i);  df1 <- subset(df1, Abundance > 0)
  for(x in c("P", "D")){
  df<- subset(df1, PD_Colon == x)
  t <- cor(df$Abundance, df$TT, method = "spearman")
  
  if(is.na(t)){corr[["p.value"]] <- NA}else{
   corr <- cor.test(df$Abundance, df$TT, method = 'spearman')  
  }
  
  if(is.na(corr[["p.value"]])){
    sig <- ""
    }else{
      if(corr[["p.value"]]<0.0001){
        sig <- "****"
        }else{
          if(corr[["p.value"]]<0.001){
            sig <- "***"
            }else{
              if(corr[["p.value"]]<0.01){
                sig <- "**"
                }else{
                  if(corr[["p.value"]]<0.05){
                    sig<-"*"
                  }else{
                      sig<-""}}}}}
  
  
  if(is.na(t)){dfcor2.otu <- rbind(dfcor2.otu, data.frame(genus = i, cor = NA, 
                 PD_Colon = x, pvalue = corr[["p.value"]], sig = sig, comm = "Proportional microbial community"))
  }else{
    dfcor2.otu <- rbind(dfcor2.otu, data.frame(genus = i, cor = corr[["estimate"]][["rho"]], 
                 PD_Colon = x, pvalue = corr[["p.value"]], sig = sig, comm = "Proportional microbial community"))}
  }}
#Combine abs and prop abundances in one
dfcor.otu <- rbind(dfcor.otu, dfcor2.otu)
# dfcor.otu <- na.omit(dfcor.otu) # Will remove 4 OTUs altogether!
dfcor.otu$comm <- as.ordered(factor(dfcor.otu$comm, 
                              levels = c("Quantitative microbial community", "Proportional microbial community")))
dfcor.otu$PD_Colon <- as.ordered(factor(dfcor.otu$PD_Colon, levels = c("P", "D")))
plot <- papertheme(ggplot(dfcor.otu, aes(x = PD_Colon, y = genus, fill =cor))) + geom_tile() +
  scale_fill_gradient2(na.value = "grey50",low="blue", high="red") + 
  geom_text(aes(label = sig)) +
  theme(axis.text.y = element_text(face="italic")) + ylab("Genus") + xlab("Colon vessel") + 
  scale_x_discrete(breaks=c("P","D"), labels=c("Proximal colon", "Distal colon")) +
  labs(fill = expression(rho)) +
  facet_grid(cols = vars(comm)) +theme(strip.background = element_rect(fill = "#D6E0F4"))
plot
# Combine Genus and OTU levels 
dfcorall.otu <- dfcor.otu
dfcorall.gen.otu <- data.frame(); orderlist <- data.frame()
for (i in Top.genera){
  testotu <- subset(df.OTU, genus ==i); orderlist2 <- data.frame(genus = i)
  df <- subset(dfcorall, genus == i)
  for(x in sort(testotu$rowname, decreasing = F)){
    df2 <- subset(dfcorall.otu, genus == x)
    
    # Remove 00 from OTUs
    if(stringr::str_detect(x, "Otu0000")){
      x <- str_replace(x, "Otu0000", "OTU")
    }else{if(stringr::str_detect(x, "Otu000")){
      x <- str_replace(x, "Otu000", "OTU")
    }else{if(stringr::str_detect(x, "Otu00")){
      x <- str_replace(x, "Otu00", "OTU")
    }else{if(stringr::str_detect(x, "Otu0")){
      x <- str_replace(x, "Otu0", "OTU")
    }}}}
    df2$genus <- x
    
    
    df <- rbind(df2, df) # Do it the other way round for aesthetics 
    ord <- data.frame(genus = x); orderlist2 <- rbind(ord, orderlist2) # Do it the other way round for aesthetics 
  }
  orderlist <- rbind(orderlist, orderlist2)
  dfcorall.gen.otu <- rbind(dfcorall.gen.otu, df)
}
orderlist <- orderlist$genus
dfcorall.gen.otu$genus <- as.ordered(factor(dfcorall.gen.otu$genus, levels = orderlist))
plot <- papertheme(ggplot(dfcorall.gen.otu, aes(x = PD_Colon, y = genus, fill =cor))) + geom_tile() +
  scale_fill_gradient2(na.value = "grey50",low="#5e135c", high="#125c13", breaks = c(-0.8,-0.4,0,0.4,0.8)) +
  geom_text(aes(label = sig, fontface = "bold")) +
  # theme(axis.text.y = element_text(face="italic")) +
  ylab("Genus") + xlab("Colon vessel") + 
  scale_x_discrete(breaks=c("P","D"), labels=c("Proximal colon", "Distal colon")) +
  labs(fill = expression(rho)) +
  facet_grid(cols = vars(comm)) +theme(strip.background = element_rect(fill = "#D6E0F4"))+
  theme(legend.key.width= unit(2, 'cm'))
plot
# Create data frame with combinations of colon compartments
# GENUS LEVEL
dfcor <- data.frame()
for(i in na.omit(unique(Top$Genus))){
  df1 <- subset(Top, Genus == i);  df1 <- subset(df1, Abundance > 0)
  
  df<- df1
  t <- cor(df$Abundance, df$TT, method = "spearman")
  corr <- cor.test(df$Abundance, df$TT, method = 'spearman'); 
  
  if(is.na(corr[["p.value"]])){
    sig <- "-"
    }else{
      if(corr[["p.value"]]<0.0001){
        sig <- "****"
        }else{
          if(corr[["p.value"]]<0.001){
            sig <- "***"
            }else{
              if(corr[["p.value"]]<0.01){
                sig <- "**"
                }else{
                  if(corr[["p.value"]]<0.05){
                    sig<-"*"
                  }else{
                      sig<-""}}}}}
  
  dfcor <- rbind(dfcor, data.frame(genus = i, cor = corr[["estimate"]][["rho"]], 
                 PD_Colon = "Combined", pvalue = corr[["p.value"]], sig = sig, comm = "Quantitative microbial community"))
  }
dfcor2 <- data.frame()
for(i in na.omit(unique(Top.prop$Genus))){
  df1 <- subset(Top.prop, Genus == i);  df1 <- subset(df1, Abundance > 0)
  
  df<- df1
  t <- cor(df$Abundance, df$TT, method = "spearman")
  corr <- cor.test(df$Abundance, df$TT, method = 'spearman'); 
  
  if(is.na(corr[["p.value"]])){
    sig <- "-"
    }else{
      if(corr[["p.value"]]<0.0001){
        sig <- "****"
        }else{
          if(corr[["p.value"]]<0.001){
            sig <- "***"
            }else{
              if(corr[["p.value"]]<0.01){
                sig <- "**"
                }else{
                  if(corr[["p.value"]]<0.05){
                    sig<-"*"
                  }else{
                      sig<-""}}}}}
  
  dfcor2 <- rbind(dfcor2, data.frame(genus = i, cor = corr[["estimate"]][["rho"]], 
                 PD_Colon = "Combined", pvalue = corr[["p.value"]], sig = sig, comm = "Proportional microbial community"))
  }
#Combine abs and prop abundances in one
dfcor <- rbind(dfcor, dfcor2); dfcor.for.genera <- dfcor
dfcor <- na.omit(dfcor); dfcor$comm <- as.ordered(factor(dfcor$comm, 
                              levels = c("Quantitative microbial community", "Proportional microbial community")))
dfcorcombined <- dfcor
dfcor$genus <- as.ordered(factor(dfcor$genus, levels = Top.genera))
plot <- papertheme(ggplot(dfcor, aes(x = "1", y = genus, fill =cor))) + geom_tile() +
  scale_fill_gradient2(na.value = "grey50",low="blue", high="red", 
                       breaks = c(-0.8,-0.4,0,0.4,0.8), labels = c(-0.8,-0.4,0,0.4,0.8)) + 
  geom_text(aes(label = sig)) +
  theme(axis.text.y = element_text(face="italic")) + ylab("Genus") + xlab("") + 
  scale_x_discrete(breaks=c("P","D"), labels=c("Proximal colon", "Distal colon")) +
  labs(fill = expression(rho)) +
  facet_grid(cols = vars(comm)) +theme(strip.background = element_rect(fill = "#D6E0F4")) + 
  theme(legend.key.width= unit(2, 'cm'))
plot
dfcorall <- rbind(dfcorall, dfcor)
dfcorall$PD_Colon <- as.ordered(factor(dfcorall$PD_Colon, levels = c("P", "D", "Combined")))
dfcorall$genus <- as.ordered(factor(dfcorall$genus, levels = Top.genera))
plot <- papertheme(ggplot(dfcorall, aes(x = PD_Colon, y = genus, fill =cor))) + geom_tile() +
  scale_fill_gradient2(na.value = "grey50",low="#5e135c", high="#125c13", breaks = c(-0.8,-0.4,0,0.4,0.8)) +
  geom_text(aes(label = sig, fontface = "bold")) +
  theme(axis.text.y = element_text(face="italic")) + ylab("Genus") + xlab("Colon vessel") + 
  scale_x_discrete(breaks=c("P","D", "Combined"), labels=c("Proximal colon", "Distal colon", "Combined")) +
  labs(fill = expression(rho)) +
  facet_grid(cols = vars(comm)) +theme(strip.background = element_rect(fill = "#D6E0F4")) + 
  theme(legend.key.width= unit(2, 'cm'))
plot
# OTU LEVEL
## Perform tests
dfcor.otu <- data.frame()
for(i in na.omit(unique(Top.otu$OTU))){
  df1 <- subset(Top.otu, OTU == i);  df <- subset(df1, Abundance > 0)
  t <- cor(df$Abundance, df$TT, method = "spearman")
  if(is.na(t)){corr[["p.value"]] <- NA}else{
   corr <- cor.test(df$Abundance, df$TT, method = 'spearman')  
  }
  
  if(is.na(corr[["p.value"]])){
    sig <- ""
    }else{
      if(corr[["p.value"]]<0.0001){
        sig <- "****"
        }else{
          if(corr[["p.value"]]<0.001){
            sig <- "***"
            }else{
              if(corr[["p.value"]]<0.01){
                sig <- "**"
                }else{
                  if(corr[["p.value"]]<0.05){
                    sig<-"*"
                  }else{
                      sig<-""}}}}}
  
  if(is.na(t)){dfcor.otu <- rbind(dfcor.otu, data.frame(genus = i, cor = NA, 
                 PD_Colon = "Combined", pvalue = corr[["p.value"]], sig = sig, comm = "Quantitative microbial community"))
  }else{
    dfcor.otu <- rbind(dfcor.otu, data.frame(genus = i, cor = corr[["estimate"]][["rho"]], 
                 PD_Colon = "Combined", pvalue = corr[["p.value"]], sig = sig, comm = "Quantitative microbial community"))}
  }
dfcor2.otu <- data.frame()
for(i in na.omit(unique(Top.prop.otu$OTU))){
  df1 <- subset(Top.prop.otu, OTU == i);  df <- subset(df1, Abundance > 0)
  t <- cor(df$Abundance, df$TT, method = "spearman")
  
  if(is.na(t)){corr[["p.value"]] <- NA}else{
   corr <- cor.test(df$Abundance, df$TT, method = 'spearman')  
  }
  
  if(is.na(corr[["p.value"]])){
    sig <- ""
    }else{
      if(corr[["p.value"]]<0.0001){
        sig <- "****"
        }else{
          if(corr[["p.value"]]<0.001){
            sig <- "***"
            }else{
              if(corr[["p.value"]]<0.01){
                sig <- "**"
                }else{
                  if(corr[["p.value"]]<0.05){
                    sig<-"*"
                  }else{
                      sig<-""}}}}}
  
  
  if(is.na(t)){dfcor2.otu <- rbind(dfcor2.otu, data.frame(genus = i, cor = NA, 
                 PD_Colon = "Combined", pvalue = corr[["p.value"]], sig = sig, comm = "Proportional microbial community"))
  }else{
    dfcor2.otu <- rbind(dfcor2.otu, data.frame(genus = i, cor = corr[["estimate"]][["rho"]], 
                 PD_Colon = "Combined", pvalue = corr[["p.value"]], sig = sig, comm = "Proportional microbial community"))}
  }
#Combine abs and prop abundances in one
dfcor.otu <- rbind(dfcor.otu, dfcor2.otu)
# dfcor <- na.omit(dfcor) #enabling this will mess up stuff!
dfcor.otu$comm <- as.ordered(factor(dfcor.otu$comm, 
                              levels = c("Quantitative microbial community", "Proportional microbial community")))
# Combine Genus and OTU levels 
dfcorall5 <- subset(dfcorall, PD_Colon == "Combined")
# dfcorall.gen.otu <- data.frame(); orderlist <- data.frame()
for (i in Top.genera){
  testotu <- subset(df.OTU, genus ==i)
  df <- subset(dfcorall5, genus == i)
  for(x in sort(testotu$rowname, decreasing = F)){
    df2 <- subset(dfcor.otu, genus == x)
    
    # Remove 00 from OTUs
    if(stringr::str_detect(x, "Otu0000")){
      x <- str_replace(x, "Otu0000", "OTU")
    }else{if(stringr::str_detect(x, "Otu000")){
      x <- str_replace(x, "Otu000", "OTU")
    }else{if(stringr::str_detect(x, "Otu00")){
      x <- str_replace(x, "Otu00", "OTU")
    }else{if(stringr::str_detect(x, "Otu0")){
      x <- str_replace(x, "Otu0", "OTU")
    }}}}
    df2$genus <- x
    df <- rbind(df2, df) # Do it the other way round for aesthetics 
    }
  dfcorall.gen.otu <- rbind(dfcorall.gen.otu, df)
  }
dfcorall.gen.otu$genus <- as.ordered(factor(dfcorall.gen.otu$genus, levels = orderlist))
dfcorall.gen.otu$PD_Colon <- as.ordered(factor(dfcorall.gen.otu$PD_Colon, levels = c("P", "D", "Combined")))
plot <- papertheme(ggplot(dfcorall.gen.otu, aes(x = PD_Colon, y = genus, fill =cor))) + geom_tile() +
  scale_fill_gradient2(na.value = "grey50",low="#5e135c", high="#125c13", breaks = c(-0.8,-0.4,0,0.4,0.8)) +
  geom_text(aes(label = sig, fontface = "bold")) +  ylab("Genus") + xlab("Colon vessel") 
plot <- italicgenera(plot)  
plot <- plot +  scale_x_discrete(breaks=c("P","D", "Combined"), labels=c("Proximal colon", "Distal colon", "Combined")) +
  labs(fill = expression(rho)) +
  facet_grid(cols = vars(comm)) +theme(strip.background = element_rect(fill = "#D6E0F4"))+
  theme(legend.key.width= unit(2, 'cm'))
plot
# saveplots()
# ggsave(plot = plot, width = 15, height = 25, dpi = 300, filename = "Paperplot_correlation_genus_otu_top15.png")
# Only show OTUs that have significant correlations
dfcorall.gen.otu.sig <- data.frame()
for(i in unique(dfcorall.gen.otu$genus)){
  df <- subset(dfcorall.gen.otu, genus == i)
  
  n1 <- df[1,4]; if(is.na(n1)){n1 <- 1}; n2 <- df[2,4]; if(is.na(n2)){n2 <- 1}
  n3 <- df[3,4]; if(is.na(n3)){n3 <- 1}; n4 <- df[4,4]; if(is.na(n4)){n4 <- 1}
  n5 <- df[5,4]; if(is.na(n5)){n5 <- 1}; n6 <- df[6,4]; if(is.na(n6)){n6 <- 1}
  if(n1<0.05 | n2<0.05 | n3<0.05 | n4<0.05 | n5<0.05 | n6<0.05){
    # message("nice")
    dfcorall.gen.otu.sig <- rbind(dfcorall.gen.otu.sig, df)}
  if(i == "Akkermansia" | i == "OTU18"){dfcorall.gen.otu.sig <- rbind(dfcorall.gen.otu.sig, df)}
}
plot <- papertheme(ggplot(dfcorall.gen.otu.sig, aes(x = PD_Colon, y = genus, fill =cor))) + geom_tile() +
  scale_fill_gradient2(na.value = "grey50",low="#5e135c", high="#125c13", breaks = c(-0.8,-0.4,0,0.4,0.8)) +
  geom_text(aes(label = sig, fontface = "bold")) + ylab("") + xlab("Colon vessel")
# plot <- italicgenera(plot)
plot <- bolditalicgenera(plot)
plot <- plot +  scale_x_discrete(breaks=c("P","D", "Combined"), labels=c("Proximal colon", "Distal colon", "Combined")) +
  labs(fill = expression(rho)) +
  facet_grid(cols = vars(comm)) +theme(strip.background = element_rect(fill = "#D6E0F4"))+
  theme(legend.key.width= unit(2, 'cm'))
plot
```


### Supplementary paperplot genera correlations and OTU of Top other abundant genera
```{r Paperplot correlation colons split AND combined, fig.height=20, fig.width=20}
# order <- c("Veillonella", 	"Sutterella", 	"Ruminococcus", 	"Ruminococcaceae", 	"Roseburia", 	"Pseudomonas", 	"Prevotella", 	"Phascolarctobacterium", 	"Parabacteroides", 	"Mitsuokella", 	"Megasphaera", 	"Megamonas", 	"Lachnospiraceae", 	"Fusobacterium", 	"Fusicatenibacter", 	"Faecalibacterium", 	"Erysipelotrichaceae", 	"Enterobacteriaceae", 	"Eisenbergiella", 	"Dorea", 	"Dialister", 	"Desulfovibrio", 	"Coprococcus", 	"Clostridium_XlVa", 	"Clostridiales", 	"Cloacibacillus", 	"Blautia", 	"Bilophila", 	"Bifidobacterium", 	"Bacteroides", 	"Bacteria", 	"Anaerostipes", 	"Anaeroglobus", 	"Alistipes", 	"Akkermansia", 	"Acidaminococcus")
# df.OTU <-subset(df.combined, sum > 10)
df.OTU <-subset(df.combined, mean > 5);
topnum <- 15
# Retrieve top most abundant genera
## Absolute
phylobj.QMP_stable <- subset_samples(phylobj.QMP, Ninetransit == 3)
Top15phylobj <- aggregate_taxa(phylobj.QMP_stable, level = "Genus")
Top15phylobj <- aggregate_top_taxa(phylobj.QMP_stable, top = topnum, "Genus")
## Proportional
phylobj.prop_2 = transform_sample_counts(phylobj.QMP, function(x) 100 * x/sum(x))
phylobj.prop_stable <- subset_samples(phylobj.prop_2, Ninetransit == 3)
phylobj.prop_stable <- aggregate_top_taxa(phylobj.prop_stable, top = topnum, "Genus")
## Take consensus top genera
Top.genera <- rbind(subset(as.data.frame(phylobj.prop_stable@tax_table), Genus != "Other"), 
                    subset(as.data.frame(Top15phylobj@tax_table), Genus != "Other"))
## Remove unsignificant genera
df <- data.frame()
for(i in significantlist){
  df2 <- subset(Top.genera, Genus == i)
  df <- rbind(df, df2)
}
Top.genera <- unique(df$Genus)
# Remove top genera from the significant list
df <- data.frame(Genus = significantlist); 
for(i in Top.genera){
  df <- subset(df, Genus != i)
  # print(i)
  }
Top.genera <- sort(unique(df$Genus))
# Remove those that are not in the significant changes list (significantlist)
Top.genera <- sort(Top.genera, decreasing = T)
Top.genera2 <- Top.genera
# Start, but first split it up in 3 sets of genera
for (numberino in c(1:2)) {
  if(numberino == 1){Top.genera <- Top.genera2[1:25]}
  if(numberino == 2){Top.genera <- Top.genera2[26:45]}
  # if(numberino == 3){Top.genera <- Top.genera2[31:45]}
# GENUS LEVEL
# Absolute
phylobj.QMP_stable <- subset_samples(phylobj.QMP, Ninetransit == 3)
Top15phylobj <- aggregate_taxa(phylobj.QMP_stable, level = "Genus")
Top15 <- psmelt(Top15phylobj); unique(Top15$OTU); unique(Top15$OTU) %in% Top15levels
test <- as.data.frame(unique(Top15$OTU))
## Subset according to top most abundant genera
df.test <- data.frame();
for(i in Top.genera){
  df.test2 <- subset(Top15, Genus == i)
  df.test <- rbind(df.test, df.test2)
}
Top15 <- df.test
x <- data.frame(x = unique(Top15$OTU) %in% Top15levels, genus = unique(Top15$OTU))  #check if labelled
Top15$OTU <- as.ordered(factor(Top15$OTU, levels = Top15levels))
## Add extra column days for graphs
Top <- Top15
Top$Transit <- as.ordered(factor(Top$Transit, levels = c("ST", "MT", "LT")))
Top$PD_Colon <- as.ordered(factor(Top$PD_Colon, levels = c("P", "D", "FS")))
Top$Donor <- as.ordered(factor(Top$Donor, levels = c("1", "2", "3", "4", "5", "6")))
Top$Genus <- as.ordered(factor(Top$Genus, levels = Top15levels))
## Add numbers
df.1 <- subset(Top, PD_Colon == "P" & Transit == "ST"); df.1$TT <- 8
df.2 <- subset(Top, PD_Colon == "P" & Transit == "MT"); df.2$TT <- 16
df.3 <- subset(Top, PD_Colon == "P" & Transit == "LT"); df.3$TT <- 24
df.4 <- subset(Top, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 13
df.5 <- subset(Top, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 26
df.6 <- subset(Top, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 39
Top <- rbind(df.1, df.2, df.3, df.4, df.5, df.6)
rm(df.1, df.2, df.3, df.4, df.5, df.6)
## Perform test
dfcor <- data.frame(); df.otutest <- data.frame()
for(i in na.omit(unique(Top$Genus))){
  df1 <- subset(Top, Genus == i);  df1 <- subset(df1, Abundance > 0)
  for(x in c("P", "D")){
  df<- subset(df1, PD_Colon == x)
  t <- cor(df$Abundance, df$TT, method = "spearman")
  
  if(is.na(t)){corr[["p.value"]] <- NA}else{
   corr <- cor.test(df$Abundance, df$TT, method = 'spearman')  
  }
  
  if(is.na(corr[["p.value"]])){
    sig <- "-"
    }else{
      if(corr[["p.value"]]<0.0001){
        sig <- "****"
        }else{
          if(corr[["p.value"]]<0.001){
            sig <- "***"
            }else{
              if(corr[["p.value"]]<0.01){
                sig <- "**"
                }else{
                  if(corr[["p.value"]]<0.05){
                    sig<-"*"
                  }else{
                      sig<-""}}}}}
  
  dfcor <- rbind(dfcor, data.frame(genus = i, cor = corr[["estimate"]][["rho"]], 
                 PD_Colon = x, pvalue = corr[["p.value"]], sig = sig, comm = "Quantitative microbial community"))
  }}
# Proportional
phylobj.prop_2 = transform_sample_counts(phylobj.QMP, function(x) 100 * x/sum(x))
phylobj.prop_stable <- subset_samples(phylobj.prop_2, Ninetransit == 3)
Top15phylobj.prop <- aggregate_taxa(phylobj.prop_stable, level = "Genus")
Top15.prop <- psmelt(Top15phylobj.prop); unique(Top15.prop$OTU); unique(Top15.prop$OTU) %in% Top15levels
## Subset according to top most abundant genera
df.test <- data.frame();
for(i in Top.genera){
  df.test2 <- subset(Top15.prop, Genus == i)
  df.test <- rbind(df.test, df.test2)
}
Top15.prop <- df.test
x <- data.frame(x = unique(Top15$OTU) %in% Top15levels, genus = unique(Top15$OTU))  #check if labelled
Top15$OTU <- as.ordered(factor(Top15$OTU, levels = Top15levels))
x <- data.frame(x = unique(Top15.prop$OTU) %in% Top15levels, genus = unique(Top15.prop$OTU))  #check if labelled
Top15.prop$OTU <- as.ordered(factor(Top15.prop$OTU, levels = Top15levels))
## Add extra column days for graphs
Top.prop <- Top15.prop
Top.prop$Transit <- as.ordered(factor(Top.prop$Transit, levels = c("ST", "MT", "LT")))
Top.prop$PD_Colon <- as.ordered(factor(Top.prop$PD_Colon, levels = c("P", "D", "FS")))
Top.prop$Donor <- as.ordered(factor(Top.prop$Donor, levels = c("1", "2", "3", "4", "5", "6")))
Top.prop$Genus <- as.ordered(factor(Top.prop$Genus, levels = Top15levels))
## Add numbers
df.1 <- subset(Top.prop, PD_Colon == "P" & Transit == "ST"); df.1$TT <- 8
df.2 <- subset(Top.prop, PD_Colon == "P" & Transit == "MT"); df.2$TT <- 16
df.3 <- subset(Top.prop, PD_Colon == "P" & Transit == "LT"); df.3$TT <- 24
df.4 <- subset(Top.prop, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 13
df.5 <- subset(Top.prop, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 26
df.6 <- subset(Top.prop, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 39
Top.prop <- rbind(df.1, df.2, df.3, df.4, df.5, df.6)
rm(df.1, df.2, df.3, df.4, df.5, df.6)
dfcor2 <- data.frame()
for(i in na.omit(unique(Top.prop$Genus))){
  df1 <- subset(Top.prop, Genus == i);  df1 <- subset(df1, Abundance > 0)
  for(x in c("P", "D")){
  df<- subset(df1, PD_Colon == x)
  t <- cor(df$Abundance, df$TT, method = "spearman")
  
  if(is.na(t)){corr[["p.value"]] <- NA}else{
   corr <- cor.test(df$Abundance, df$TT, method = 'spearman')  
  }
  
  if(is.na(corr[["p.value"]])){
    sig <- "-"
    }else{
      if(corr[["p.value"]]<0.0001){
        sig <- "****"
        }else{
          if(corr[["p.value"]]<0.001){
            sig <- "***"
            }else{
              if(corr[["p.value"]]<0.01){
                sig <- "**"
                }else{
                  if(corr[["p.value"]]<0.05){
                    sig<-"*"
                  }else{
                      sig<-""}}}}}
  
  dfcor2 <- rbind(dfcor2, data.frame(genus = i, cor = corr[["estimate"]][["rho"]], 
                 PD_Colon = x, pvalue = corr[["p.value"]], sig = sig, comm = "Proportional microbial community"))
  }}
#Combine abs and prop abundances in one
dfcor <- rbind(dfcor, dfcor2)
dfcor <- na.omit(dfcor); dfcor$comm <- as.ordered(factor(dfcor$comm, 
                              levels = c("Quantitative microbial community", "Proportional microbial community")))
dfcor$PD_Colon <- as.ordered(factor(dfcor$PD_Colon, levels = c("P", "D")))
dfcor$genus <- as.ordered(factor(dfcor$genus, levels = Top.genera))
plot <- papertheme(ggplot(dfcor, aes(x = PD_Colon, y = genus, fill =cor))) + geom_tile() +
  scale_fill_gradient2(na.value = "grey50",low="blue", high="red") + 
  geom_text(aes(label = sig)) +
  theme(axis.text.y = element_text(face="italic")) + ylab("Genus") + xlab("Colon vessel") + 
  scale_x_discrete(breaks=c("P","D"), labels=c("Proximal colon", "Distal colon")) +
  labs(fill = expression(rho)) +
  facet_grid(cols = vars(comm)) +theme(strip.background = element_rect(fill = "#D6E0F4"))
plot
dfcorall <- dfcor
# OTU LEVEL
# Create OTU list with the genera
Top.OTU <- data.frame(); df.OTU <-df.combined
for(i in Top.genera){
  df.test <- subset(df.OTU, genus == i)
  Top.OTU <- rbind(Top.OTU, df.test)
}
Top.OTU <- unique(Top.OTU$rowname)
# Absolute
Top15.otu <- psmelt(subset_samples(phylobj.QMP.otu, Ninetransit == 3)); unique(Top15.otu$OTU)
## Subset according to top most abundant
df.test <- data.frame()
for(i in Top.OTU){
  df.test2 <- subset(Top15.otu, OTU == i)
  df.test <- rbind(df.test, df.test2)
}
Top15.otu <- df.test
unique(Top15.otu$OTU)
## Add extra column days for graphs
Top.otu <- Top15.otu
Top.otu$Transit <- as.ordered(factor(Top.otu$Transit, levels = c("ST", "MT", "LT")))
Top.otu$PD_Colon <- as.ordered(factor(Top.otu$PD_Colon, levels = c("P", "D", "FS")))
Top.otu$Donor <- as.ordered(factor(Top.otu$Donor, levels = c("1", "2", "3", "4", "5", "6")))
## Add numbers
df.1 <- subset(Top.otu, PD_Colon == "P" & Transit == "ST"); df.1$TT <- 8
df.2 <- subset(Top.otu, PD_Colon == "P" & Transit == "MT"); df.2$TT <- 16
df.3 <- subset(Top.otu, PD_Colon == "P" & Transit == "LT"); df.3$TT <- 24
df.4 <- subset(Top.otu, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 13
df.5 <- subset(Top.otu, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 26
df.6 <- subset(Top.otu, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 39
Top.otu <- rbind(df.1, df.2, df.3, df.4, df.5, df.6)
rm(df.1, df.2, df.3, df.4, df.5, df.6)

## Perform tests
dfcor.otu <- data.frame()
for(i in na.omit(unique(Top.otu$OTU))){
  df1 <- subset(Top.otu, OTU == i);  df1 <- subset(df1, Abundance > 0)
  for(x in c("P", "D")){
  df<- subset(df1, PD_Colon == x)
  t <- cor(df$Abundance, df$TT, method = "spearman")
  
  if(is.na(t)){corr[["p.value"]] <- NA}else{
   corr <- cor.test(df$Abundance, df$TT, method = 'spearman')  
  }
  
  
  if(is.na(corr[["p.value"]])){
    sig <- ""
    }else{
      if(corr[["p.value"]]<0.0001){
        sig <- "****"
        }else{
          if(corr[["p.value"]]<0.001){
            sig <- "***"
            }else{
              if(corr[["p.value"]]<0.01){
                sig <- "**"
                }else{
                  if(corr[["p.value"]]<0.05){
                    sig<-"*"
                  }else{
                      sig<-""}}}}}
  
  if(is.na(t)){dfcor.otu <- rbind(dfcor.otu, data.frame(genus = i, cor = NA, 
                 PD_Colon = x, pvalue = corr[["p.value"]], sig = sig, comm = "Quantitative microbial community"))
  }else{
    dfcor.otu <- rbind(dfcor.otu, data.frame(genus = i, cor = corr[["estimate"]][["rho"]], 
                 PD_Colon = x, pvalue = corr[["p.value"]], sig = sig, comm = "Quantitative microbial community"))}
  }}
# Proportional
Top15.prop.otu <- psmelt(subset_samples(phylobj.prop.otu, Ninetransit == 3)); unique(Top15.prop.otu$OTU)
## Subset according to top most abundant
df.test <- data.frame()
for(i in Top.OTU){
  df.test2 <- subset(Top15.prop.otu, OTU == i)
  df.test <- rbind(df.test, df.test2)
}
Top15.prop.otu <- df.test
unique(Top15.prop.otu$OTU)
## Add extra column days for graphs
Top.prop.otu <- Top15.prop.otu
Top.prop.otu$Transit <- as.ordered(factor(Top.prop.otu$Transit, levels = c("ST", "MT", "LT")))
Top.prop.otu$PD_Colon <- as.ordered(factor(Top.prop.otu$PD_Colon, levels = c("P", "D", "FS")))
Top.prop.otu$Donor <- as.ordered(factor(Top.prop.otu$Donor, levels = c("1", "2", "3", "4", "5", "6")))
## Add numbers
df.1 <- subset(Top.prop.otu, PD_Colon == "P" & Transit == "ST"); df.1$TT <- 8
df.2 <- subset(Top.prop.otu, PD_Colon == "P" & Transit == "MT"); df.2$TT <- 16
df.3 <- subset(Top.prop.otu, PD_Colon == "P" & Transit == "LT"); df.3$TT <- 24
df.4 <- subset(Top.prop.otu, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 13
df.5 <- subset(Top.prop.otu, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 26
df.6 <- subset(Top.prop.otu, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 39
Top.prop.otu <- rbind(df.1, df.2, df.3, df.4, df.5, df.6)
rm(df.1, df.2, df.3, df.4, df.5, df.6)
dfcor2.otu <- data.frame()
for(i in na.omit(unique(Top.prop.otu$OTU))){
  df1 <- subset(Top.prop.otu, OTU == i);  df1 <- subset(df1, Abundance > 0)
  for(x in c("P", "D")){
  df<- subset(df1, PD_Colon == x)
  t <- cor(df$Abundance, df$TT, method = "spearman")
  
  if(is.na(t)){corr[["p.value"]] <- NA}else{
   corr <- cor.test(df$Abundance, df$TT, method = 'spearman')  
  }
  
  if(is.na(corr[["p.value"]])){
    sig <- ""
    }else{
      if(corr[["p.value"]]<0.0001){
        sig <- "****"
        }else{
          if(corr[["p.value"]]<0.001){
            sig <- "***"
            }else{
              if(corr[["p.value"]]<0.01){
                sig <- "**"
                }else{
                  if(corr[["p.value"]]<0.05){
                    sig<-"*"
                  }else{
                      sig<-""}}}}}
  
  
  if(is.na(t)){dfcor2.otu <- rbind(dfcor2.otu, data.frame(genus = i, cor = NA, 
                 PD_Colon = x, pvalue = corr[["p.value"]], sig = sig, comm = "Proportional microbial community"))
  }else{
    dfcor2.otu <- rbind(dfcor2.otu, data.frame(genus = i, cor = corr[["estimate"]][["rho"]], 
                 PD_Colon = x, pvalue = corr[["p.value"]], sig = sig, comm = "Proportional microbial community"))}
  }}
#Combine abs and prop abundances in one
dfcor.otu <- rbind(dfcor.otu, dfcor2.otu)
# dfcor.otu <- na.omit(dfcor.otu) # Will remove 4 OTUs altogether!
dfcor.otu$comm <- as.ordered(factor(dfcor.otu$comm, 
                              levels = c("Quantitative microbial community", "Proportional microbial community")))
dfcor.otu$PD_Colon <- as.ordered(factor(dfcor.otu$PD_Colon, levels = c("P", "D")))
plot <- papertheme(ggplot(dfcor.otu, aes(x = PD_Colon, y = genus, fill =cor))) + geom_tile() +
  scale_fill_gradient2(na.value = "grey50",low="blue", high="red") + 
  geom_text(aes(label = sig)) +
  theme(axis.text.y = element_text(face="italic")) + ylab("Genus") + xlab("Colon vessel") + 
  scale_x_discrete(breaks=c("P","D"), labels=c("Proximal colon", "Distal colon")) +
  labs(fill = expression(rho)) +
  facet_grid(cols = vars(comm)) +theme(strip.background = element_rect(fill = "#D6E0F4"))
plot
# Combine Genus and OTU levels 
dfcorall.otu <- dfcor.otu
dfcorall.gen.otu <- data.frame(); orderlist <- data.frame()
for (i in Top.genera){
  testotu <- subset(df.OTU, genus ==i); orderlist2 <- data.frame(genus = i)
  df <- subset(dfcorall, genus == i)
  for(x in sort(testotu$rowname, decreasing = F)){
    df2 <- subset(dfcorall.otu, genus == x)
    
    # Remove 00 from OTUs
    if(stringr::str_detect(x, "Otu0000")){
      x <- str_replace(x, "Otu0000", "OTU")
    }else{if(stringr::str_detect(x, "Otu000")){
      x <- str_replace(x, "Otu000", "OTU")
    }else{if(stringr::str_detect(x, "Otu00")){
      x <- str_replace(x, "Otu00", "OTU")
    }else{if(stringr::str_detect(x, "Otu0")){
      x <- str_replace(x, "Otu0", "OTU")
    }}}}
    df2$genus <- x
    
    
    df <- rbind(df2, df) # Do it the other way round for aesthetics 
    ord <- data.frame(genus = x); orderlist2 <- rbind(ord, orderlist2) # Do it the other way round for aesthetics 
  }
  orderlist <- rbind(orderlist, orderlist2)
  dfcorall.gen.otu <- rbind(dfcorall.gen.otu, df)
}
orderlist <- orderlist$genus
dfcorall.gen.otu$genus <- as.ordered(factor(dfcorall.gen.otu$genus, levels = orderlist))
plot <- papertheme(ggplot(dfcorall.gen.otu, aes(x = PD_Colon, y = genus, fill =cor))) + geom_tile() +
  scale_fill_gradient2(na.value = "grey50",low="#5e135c", high="#125c13", breaks = c(-0.8,-0.4,0,0.4,0.8)) +
  geom_text(aes(label = sig, fontface = "bold")) +
  # theme(axis.text.y = element_text(face="italic")) +
  ylab("Genus") + xlab("Colon vessel") + 
  scale_x_discrete(breaks=c("P","D"), labels=c("Proximal colon", "Distal colon")) +
  labs(fill = expression(rho)) +
  facet_grid(cols = vars(comm)) +theme(strip.background = element_rect(fill = "#D6E0F4"))+
  theme(legend.key.width= unit(2, 'cm'))
plot




# Create data frame with combinations of colon compartments
# GENUS LEVEL
dfcor <- data.frame()
for(i in na.omit(unique(Top$Genus))){
  df1 <- subset(Top, Genus == i);  df1 <- subset(df1, Abundance > 0)
  
  df<- df1
  t <- cor(df$Abundance, df$TT, method = "spearman")
  
  if(is.na(t)){corr[["p.value"]] <- NA}else{
   corr <- cor.test(df$Abundance, df$TT, method = 'spearman')  
  }
  
  if(is.na(corr[["p.value"]])){
    sig <- "-"
    }else{
      if(corr[["p.value"]]<0.0001){
        sig <- "****"
        }else{
          if(corr[["p.value"]]<0.001){
            sig <- "***"
            }else{
              if(corr[["p.value"]]<0.01){
                sig <- "**"
                }else{
                  if(corr[["p.value"]]<0.05){
                    sig<-"*"
                  }else{
                      sig<-""}}}}}
  
  dfcor <- rbind(dfcor, data.frame(genus = i, cor = corr[["estimate"]][["rho"]], 
                 PD_Colon = "Combined", pvalue = corr[["p.value"]], sig = sig, comm = "Quantitative microbial community"))
  }
dfcor2 <- data.frame()
for(i in na.omit(unique(Top.prop$Genus))){
  df1 <- subset(Top.prop, Genus == i);  df1 <- subset(df1, Abundance > 0)
  
  df<- df1
  t <- cor(df$Abundance, df$TT, method = "spearman")
  
  if(is.na(t)){corr[["p.value"]] <- NA}else{
   corr <- cor.test(df$Abundance, df$TT, method = 'spearman')  
  }
  
  if(is.na(corr[["p.value"]])){
    sig <- "-"
    }else{
      if(corr[["p.value"]]<0.0001){
        sig <- "****"
        }else{
          if(corr[["p.value"]]<0.001){
            sig <- "***"
            }else{
              if(corr[["p.value"]]<0.01){
                sig <- "**"
                }else{
                  if(corr[["p.value"]]<0.05){
                    sig<-"*"
                  }else{
                      sig<-""}}}}}
  
  dfcor2 <- rbind(dfcor2, data.frame(genus = i, cor = corr[["estimate"]][["rho"]], 
                 PD_Colon = "Combined", pvalue = corr[["p.value"]], sig = sig, comm = "Proportional microbial community"))
  }
#Combine abs and prop abundances in one
dfcor <- rbind(dfcor, dfcor2); dfcor.for.genera <- dfcor
dfcor <- na.omit(dfcor); dfcor$comm <- as.ordered(factor(dfcor$comm, 
                              levels = c("Quantitative microbial community", "Proportional microbial community")))
dfcorcombined <- dfcor
dfcor$genus <- as.ordered(factor(dfcor$genus, levels = Top.genera))
plot <- papertheme(ggplot(dfcor, aes(x = "1", y = genus, fill =cor))) + geom_tile() +
  scale_fill_gradient2(na.value = "grey50",low="blue", high="red", 
                       breaks = c(-0.8,-0.4,0,0.4,0.8), labels = c(-0.8,-0.4,0,0.4,0.8)) + 
  geom_text(aes(label = sig)) +
  theme(axis.text.y = element_text(face="italic")) + ylab("Genus") + xlab("") + 
  scale_x_discrete(breaks=c("P","D"), labels=c("Proximal colon", "Distal colon")) +
  labs(fill = expression(rho)) +
  facet_grid(cols = vars(comm)) +theme(strip.background = element_rect(fill = "#D6E0F4")) + 
  theme(legend.key.width= unit(2, 'cm'))
plot
dfcorall <- rbind(dfcorall, dfcor)
dfcorall$PD_Colon <- as.ordered(factor(dfcorall$PD_Colon, levels = c("P", "D", "Combined")))
dfcorall$genus <- as.ordered(factor(dfcorall$genus, levels = Top.genera))
plot <- papertheme(ggplot(dfcorall, aes(x = PD_Colon, y = genus, fill =cor))) + geom_tile() +
  scale_fill_gradient2(na.value = "grey50",low="#5e135c", high="#125c13", breaks = c(-0.8,-0.4,0,0.4,0.8)) +
  geom_text(aes(label = sig, fontface = "bold")) +
  theme(axis.text.y = element_text(face="italic")) + ylab("Genus") + xlab("Colon vessel") + 
  scale_x_discrete(breaks=c("P","D", "Combined"), labels=c("Proximal colon", "Distal colon", "Combined")) +
  labs(fill = expression(rho)) +
  facet_grid(cols = vars(comm)) +theme(strip.background = element_rect(fill = "#D6E0F4")) + 
  theme(legend.key.width= unit(2, 'cm'))
plot


# OTU LEVEL
## Perform tests
dfcor.otu <- data.frame()
for(i in na.omit(unique(Top.otu$OTU))){
  df1 <- subset(Top.otu, OTU == i);  df <- subset(df1, Abundance > 0)
  t <- cor(df$Abundance, df$TT, method = "spearman")
  
  if(is.na(t)){corr[["p.value"]] <- NA}else{
   corr <- cor.test(df$Abundance, df$TT, method = 'spearman')  
  }
  
  if(is.na(corr[["p.value"]])){
    sig <- ""
    }else{
      if(corr[["p.value"]]<0.0001){
        sig <- "****"
        }else{
          if(corr[["p.value"]]<0.001){
            sig <- "***"
            }else{
              if(corr[["p.value"]]<0.01){
                sig <- "**"
                }else{
                  if(corr[["p.value"]]<0.05){
                    sig<-"*"
                  }else{
                      sig<-""}}}}}
  
  if(is.na(t)){dfcor.otu <- rbind(dfcor.otu, data.frame(genus = i, cor = NA, 
                 PD_Colon = "Combined", pvalue = corr[["p.value"]], sig = sig, comm = "Quantitative microbial community"))
  }else{
    dfcor.otu <- rbind(dfcor.otu, data.frame(genus = i, cor = corr[["estimate"]][["rho"]], 
                 PD_Colon = "Combined", pvalue = corr[["p.value"]], sig = sig, comm = "Quantitative microbial community"))}
  }
dfcor2.otu <- data.frame()
for(i in na.omit(unique(Top.prop.otu$OTU))){
  df1 <- subset(Top.prop.otu, OTU == i);  df <- subset(df1, Abundance > 0)
  t <- cor(df$Abundance, df$TT, method = "spearman")
  
  if(is.na(t)){corr[["p.value"]] <- NA}else{
   corr <- cor.test(df$Abundance, df$TT, method = 'spearman')  
  }
  
  if(is.na(corr[["p.value"]])){
    sig <- ""
    }else{
      if(corr[["p.value"]]<0.0001){
        sig <- "****"
        }else{
          if(corr[["p.value"]]<0.001){
            sig <- "***"
            }else{
              if(corr[["p.value"]]<0.01){
                sig <- "**"
                }else{
                  if(corr[["p.value"]]<0.05){
                    sig<-"*"
                  }else{
                      sig<-""}}}}}
  
  
  if(is.na(t)){dfcor2.otu <- rbind(dfcor2.otu, data.frame(genus = i, cor = NA, 
                 PD_Colon = "Combined", pvalue = corr[["p.value"]], sig = sig, comm = "Proportional microbial community"))
  }else{
    dfcor2.otu <- rbind(dfcor2.otu, data.frame(genus = i, cor = corr[["estimate"]][["rho"]], 
                 PD_Colon = "Combined", pvalue = corr[["p.value"]], sig = sig, comm = "Proportional microbial community"))}
  }
#Combine abs and prop abundances in one
dfcor.otu <- rbind(dfcor.otu, dfcor2.otu)
# dfcor <- na.omit(dfcor) #enabling this will mess up stuff!
dfcor.otu$comm <- as.ordered(factor(dfcor.otu$comm, 
                              levels = c("Quantitative microbial community", "Proportional microbial community")))
# Combine Genus and OTU levels 
dfcorall5 <- subset(dfcorall, PD_Colon == "Combined")
# dfcorall.gen.otu <- data.frame(); orderlist <- data.frame()
for (i in Top.genera){
  testotu <- subset(df.OTU, genus ==i)
  df <- subset(dfcorall5, genus == i)
  for(x in sort(testotu$rowname, decreasing = F)){
    df2 <- subset(dfcor.otu, genus == x)
    
    # Remove 00 from OTUs
    if(stringr::str_detect(x, "Otu0000")){
      x <- str_replace(x, "Otu0000", "OTU")
    }else{if(stringr::str_detect(x, "Otu000")){
      x <- str_replace(x, "Otu000", "OTU")
    }else{if(stringr::str_detect(x, "Otu00")){
      x <- str_replace(x, "Otu00", "OTU")
    }else{if(stringr::str_detect(x, "Otu0")){
      x <- str_replace(x, "Otu0", "OTU")
    }}}}
    df2$genus <- x
    df <- rbind(df2, df) # Do it the other way round for aesthetics 
    }
  dfcorall.gen.otu <- rbind(dfcorall.gen.otu, df)
  }
dfcorall.gen.otu$genus <- as.ordered(factor(dfcorall.gen.otu$genus, levels = orderlist))
dfcorall.gen.otu$PD_Colon <- as.ordered(factor(dfcorall.gen.otu$PD_Colon, levels = c("P", "D", "Combined")))
plot <- papertheme(ggplot(dfcorall.gen.otu, aes(x = PD_Colon, y = genus, fill =cor))) + geom_tile() +
  scale_fill_gradient2(na.value = "grey50",low="#5e135c", high="#125c13", breaks = c(-0.8,-0.4,0,0.4,0.8)) +
  geom_text(aes(label = sig, fontface = "bold")) +  ylab("Genus") + xlab("Colon vessel") 
plot <- italicgenera(plot)  
plot <- plot +  scale_x_discrete(breaks=c("P","D", "Combined"), labels=c("Proximal colon", "Distal colon", "Combined")) +
  labs(fill = expression(rho)) +
  facet_grid(cols = vars(comm)) +theme(strip.background = element_rect(fill = "#D6E0F4"))+
  theme(legend.key.width= unit(2, 'cm'))
print(plot)
# saveplots()
# ggsave(plot = plot, width = 15, height = 25, dpi = 300, filename = "Paperplot_correlation_genus_otu_top15.png")
# Only show OTUs that have significant correlations
dfcorall.gen.otu.sig <- data.frame()
for(i in unique(dfcorall.gen.otu$genus)){
  df <- subset(dfcorall.gen.otu, genus == i)
  
  n1 <- df[1,4]; if(is.na(n1)){n1 <- 1}; n2 <- df[2,4]; if(is.na(n2)){n2 <- 1}
  n3 <- df[3,4]; if(is.na(n3)){n3 <- 1}; n4 <- df[4,4]; if(is.na(n4)){n4 <- 1}
  n5 <- df[5,4]; if(is.na(n5)){n5 <- 1}; n6 <- df[6,4]; if(is.na(n6)){n6 <- 1}
  if(n1<0.05 | n2<0.05 | n3<0.05 | n4<0.05 | n5<0.05 | n6<0.05){
    # message("nice")
    dfcorall.gen.otu.sig <- rbind(dfcorall.gen.otu.sig, df)}
  if(i == "Akkermansia" | i == "OTU18"| i == "OTU359"| i == "OTU234"|i == "OTU229"){dfcorall.gen.otu.sig <- rbind(dfcorall.gen.otu.sig, df)}
}
plot <- papertheme(ggplot(dfcorall.gen.otu.sig, aes(x = PD_Colon, y = genus, fill =cor))) + geom_tile() +
  scale_fill_gradient2(na.value = "grey50",low="#5e135c", high="#125c13", breaks = c(-0.8,-0.4,0,0.4,0.8)) +
  geom_text(aes(label = sig, fontface = "bold")) + ylab("") + xlab("Colon vessel")
plot <- bolditalicgenera(plot)
# plot <- italicgenera(plot)
plot <- plot +  scale_x_discrete(breaks=c("P","D", "Combined"), labels=c("Proximal colon", "Distal colon", "Combined")) +
  labs(fill = expression(rho)) +
  facet_grid(cols = vars(comm)) +theme(strip.background = element_rect(fill = "#D6E0F4"))+
  theme(legend.key.width= unit(2, 'cm'))
print(plot)
} 
```




```{r fig.height=20, fig.width=17}
plot + theme(legend.position = "right", legend.key.width= unit(1, 'cm'), legend.key.height =  unit(1.1, 'cm'))+ ylab(NULL)
plot + theme(legend.position = "bottom", legend.key.width= unit(2, 'cm'), legend.key.height =  unit(0.5, 'cm')) + ylab(NULL)
```


## Ordination with shapes
```{r Plot_PCoAs, fig.height=10, fig.width=16, include=TRUE}
# Save location of the plots 
saveplots()
fourcolours = c("#d18fa9","#d1b78f","#8fd1b7","#8fa9d1")
phylobj.QMP2 <- subset_samples(phylobj.QMP, Ninetransit == 3)

samp <- sample_data(phylobj.QMP2)
samp1 <- subset(samp, Donor ==1 | Donor ==2); samp1$intTransit <- "Short"
samp2 <- subset(samp, Donor ==3 | Donor ==4); samp2$intTransit <- "Medium"
samp3 <- subset(samp, Donor ==5 | Donor ==6); samp3$intTransit <- "Long"
samp <- rbind(samp1, samp2, samp3) 
samp$intTransit <- as.ordered(factor(samp$intTransit, levels=c("Short", "Medium", "Long"))) 
phylobj.QMP2 <- merge_phyloseq(sample_data(samp), phylobj.QMP2)

# ON GENUS LEVEL
# Ordination graphs  (PCoA or DCA)
ord <- ordinate(phylobj.QMP2, "PCoA", "bray") 
extract_eigenvalue <- function(ordination) UseMethod("extract_eigenvalue", ordination)
extract_eigenvalue = function(ordination) ordination$values$Relative_eig # for PCoA
eigvec = extract_eigenvalue(ord)
fracvar = eigvec[1:2]/sum(eigvec)
percvar = round(100*fracvar, 0)
plot <- plot_ordination(phylobj.QMP2, ord, color = "Transit" ,
                        shape = "intTransit"
                        )
plot <- papertheme(plot, size = 20)
plot <- plot + geom_point(size = 6, alpha = 0.8) +
  scale_color_manual (name = "SHIME transit", values = transitcolors, labels = c("Short", "Medium", "Long")) +
  scale_shape(name = expression(paste(italic("In vivo"), " transit")))+
  scale_fill_manual(values = transitcolors, labels = c("Short", "Medium", "Long")) +
  
  
  
  # guides(alpha = "none", fill=guide_legend(ncol=1, title.position = "top"), 
            # color=guide_legend(ncol = 3, title.position = "top", title.hjust = 0.5, override.aes = list(size=5))) +

  theme(strip.background =element_rect(fill="#D6E0F4"),
        legend.box = "horizontal",
        strip.text.x = element_text(size = 20),
        strip.text.y = element_text(size = 20, angle = 270),
        legend.title = element_text(size=20), legend.background = element_blank()) +
  guides (size = "none") 
strivar <- as(c("PCoA1", "PCoA2"), "character"); strivar <- paste0(strivar, " ", percvar, "%")
plot <- plot + xlab(strivar[1]) + ylab(strivar[2])
# Reorder PD_Colon
newtab = data.table(plot$data)
newtab$PD_Colon <- factor(newtab$PD_Colon, levels = c("P", "D"))
newtab$Donor <- factor(newtab$Donor, levels = c(1,2,3,4,5,6))
newtab$abund <- rep("Quantitative microbial community composition",nrow(newtab)) # make new column
plot$data <- newtab
plot <- plot + 
  # scale_size_manual(values=c(6,6,6,4,4,4)) + scale_shape_manual(values=c(0,15,1,16,2,17)) +
  facet_grid(cols = vars(abund),rows = vars(PD_Colon), labeller = labeller(Donor = donor.labs, PD_Colon = PD_Colon.labsPC)) +
  # theme(legend.position = c(0.15,0.4))
  theme(legend.position = c(0.29,0.41))
plot
# ggsave(plot = plot, width = 14, height = 10, dpi = 400, filename = "PCOAQMP.png")





## Proportional abundances with QMP
phylobj.QMP.prop = subset_samples(phylobj.prop, Transit != "FS")
phylobj.QMP.prop <- subset_samples(phylobj.QMP.prop, Ninetransit ==3)

samp <- sample_data(phylobj.QMP.prop)
samp1 <- subset(samp, Donor ==1 | Donor ==2); samp1$intTransit <- "Short"
samp2 <- subset(samp, Donor ==3 | Donor ==4); samp2$intTransit <- "Medium"
samp3 <- subset(samp, Donor ==5 | Donor ==6); samp3$intTransit <- "Long"
samp <- rbind(samp1, samp2, samp3); samp$intTransit <- as.ordered(factor(samp$intTransit, 
                                                      levels=c("Short", "Medium", "Long"))) 
phylobj.QMP2 <- merge_phyloseq(sample_data(samp), phylobj.QMP.prop)



ord2 <- ordinate(phylobj.QMP2, "PCoA", "bray")
extract_eigenvalue <- function(ordination) UseMethod("extract_eigenvalue", ordination)
extract_eigenvalue = function(ordination) ordination$values$Relative_eig # for PCoA
eigvec2 = extract_eigenvalue(ord2)
fracvar2 = eigvec2[1:2]/sum(eigvec2)
percvar2 = round(100*fracvar2, 0)
plot2 <- plot_ordination(phylobj.QMP2, ord2, color = "Transit", shape = "intTransit") 
plot2 <- papertheme(plot2, size = 20)
plot2 <- plot2 + geom_point(size = 6, alpha = 0.8) +
  scale_color_manual (values = transitcolors, name = "SHIME transit") +
  scale_fill_manual(values = transitcolors, name = "Load") +
  theme(strip.background =element_rect(fill="#D6E0F4"),
        strip.text.x = element_text(size = 20),
        strip.text.y = element_text(size = 20, angle = 270),
        legend.title = element_text(size=20)) +
  guides (size = "none") 
strivar2 <- as(c("PCoA1", "PCoA2"), "character"); strivar2 <- paste0(strivar2, " ", percvar2, "%")
plot2 <- plot2 + xlab(strivar2[1]) + ylab(strivar2[2])
# Reorder PD_Colon
newtab2 = data.table(plot2$data)
newtab2$PD_Colon <- factor(newtab2$PD_Colon, levels = c("P", "D"))
newtab2$abund <- rep("Proportional microbial community composition",nrow(newtab2)) # make new column
plot2$data <- newtab2
plot2 <- plot2 + 
  facet_grid(cols = vars(abund), rows = vars(PD_Colon), labeller = labeller(Donor = donor.labs, PD_Colon = PD_Colon.labsPC)) +
  theme(legend.position = "none")
plot2
comboplot <- ggarrange(plot, plot2, ncol = 2, labels = "AUTO", font.label = list(size = 20))
comboplot
# ggsave(plot = comboplot, width = 16, height = 10, dpi = 400, filename = "Figure2.png")
# ggsave(plot = comboplot, width = 16, height = 10, dpi = 400, filename = "Figure2_high_res.tiff")
```

# Figure 2
## Paperplot ordination with shapes split per colon
```{r Plot_PCoAs, fig.height=9.5, fig.width=12.5, include=TRUE}
# Save location of the plots 
saveplots()
fourcolours = c("#d18fa9","#d1b78f","#8fd1b7","#8fa9d1")
phylobj.QMP2 <- subset_samples(phylobj.QMP, Ninetransit == 3)

samp <- sample_data(phylobj.QMP2)
samp1 <- subset(samp, Donor ==1 | Donor ==2); samp1$intTransit <- "Short"
samp2 <- subset(samp, Donor ==3 | Donor ==4); samp2$intTransit <- "Medium"
samp3 <- subset(samp, Donor ==5 | Donor ==6); samp3$intTransit <- "Long"
samp <- rbind(samp1, samp2, samp3) 
samp$intTransit <- as.ordered(factor(samp$intTransit, levels=c("Short", "Medium", "Long"))) 
phylobj.QMP2 <- merge_phyloseq(sample_data(samp), phylobj.QMP2)

# ON GENUS LEVEL
# Ordination graphs  (PCoA or DCA)
phylobj.QMP3 <- subset_samples(phylobj.QMP2, PD_Colon == "P")
ord <- ordinate(phylobj.QMP3, "PCoA", "bray") 
extract_eigenvalue <- function(ordination) UseMethod("extract_eigenvalue", ordination)
extract_eigenvalue = function(ordination) ordination$values$Relative_eig # for PCoA
eigvec = extract_eigenvalue(ord)
fracvar = eigvec[1:2]/sum(eigvec)
percvar = round(100*fracvar, 0)
plot <- plot_ordination(phylobj.QMP2, ord, color = "Transit" ,
                        shape = "intTransit"
                        )
plot <- papertheme(plot, size = 20)
plot <- plot + geom_point(size = 6, alpha = 0.8) +
  scale_color_manual (name = "SHIME transit", values = transitcolors, labels = c("Short", "Medium", "Long")) +
  scale_shape(name = expression(paste(italic("In vivo"), " transit")))+
  scale_fill_manual(values = transitcolors, labels = c("Short", "Medium", "Long")) +


  theme(strip.background =element_rect(fill="#D6E0F4"),
        legend.box = "horizontal",
        strip.text.x = element_text(size = 20),
        strip.text.y = element_text(size = 20, angle = 270),
        legend.title = element_text(size=20), legend.background = element_blank()) +
  guides (size = "none") 
strivar <- as(c("PCoA1", "PCoA2"), "character"); strivar <- paste0(strivar, " ", percvar, "%")
plot <- plot + xlab(strivar[1]) + ylab(strivar[2])
# Reorder PD_Colon
newtab = data.table(plot$data)
newtab$PD_Colon <- factor(newtab$PD_Colon, levels = c("P", "D"))
newtab$Donor <- factor(newtab$Donor, levels = c(1,2,3,4,5,6))
# newtab$abund <- rep("Quantitative microbial community composition",nrow(newtab)) # make new column
newtab$abund <- rep("Quantitative community composition",nrow(newtab)) # make new column
plot$data <- newtab; newtabbis1 = newtab
plot1 <- plot + 
  facet_grid(cols = vars(abund),rows = vars(PD_Colon), 
             labeller = labeller(Donor = donor.labs, PD_Colon = PD_Colon.labsPC)) +
  theme(legend.position = "none")
# plot1


phylobj.QMP3 <- subset_samples(phylobj.QMP2, PD_Colon == "D")
ord <- ordinate(phylobj.QMP3, "PCoA", "bray") 
extract_eigenvalue <- function(ordination) UseMethod("extract_eigenvalue", ordination)
extract_eigenvalue = function(ordination) ordination$values$Relative_eig # for PCoA
eigvec = extract_eigenvalue(ord)
fracvar = eigvec[1:2]/sum(eigvec)
percvar = round(100*fracvar, 0)
plot <- plot_ordination(phylobj.QMP2, ord, color = "Transit" ,
                        shape = "intTransit"
                        )
plot <- papertheme(plot, size = 20)
plot <- plot + geom_point(size = 6, alpha = 0.8) +
  scale_color_manual (name = "SHIME transit", values = transitcolors, labels = c("Short", "Medium", "Long")) +
  scale_shape(name = expression(paste(italic("In vivo"), " transit")))+
  scale_fill_manual(values = transitcolors, labels = c("Short", "Medium", "Long")) +


  theme(strip.background =element_rect(fill="#D6E0F4"),
        legend.box = "horizontal",
        strip.text.x = element_text(size = 20),
        strip.text.y = element_text(size = 20, angle = 270),
        legend.title = element_text(size=20), legend.background = element_blank()) #+
  # guides (size = "none", color = "none") 
strivar <- as(c("PCoA1", "PCoA2"), "character"); strivar <- paste0(strivar, " ", percvar, "%")
plot <- plot + xlab(strivar[1]) + ylab(strivar[2])

# Reorder PD_Colon
newtab = data.table(plot$data)
newtab$PD_Colon <- factor(newtab$PD_Colon, levels = c("P", "D"))
newtab$Donor <- factor(newtab$Donor, levels = c(1,2,3,4,5,6))
# newtab$abund <- rep("Quantitative microbial community composition",nrow(newtab)) # make new column
newtab$abund <- rep("Quantitative community composition",nrow(newtab)) # make new column
plot$data <- newtab; newtabbis2 = newtab
plot2 <- plot + 
  facet_grid(#cols = vars(abund),
             rows = vars(PD_Colon), 
             labeller = labeller(Donor = donor.labs, PD_Colon = PD_Colon.labsPC)) +
  # theme(legend.position = c(0.85,0.21)) +
  theme(legend.position = "bottom", legend.justification = "right") +
  guides(shape = guide_legend(ncol = 3, direction = "horizontal",
                                                    title.position = "left"))+
  xlim(-0.65,0.3) + ylim(-0.6,0.3)
# plot2



plot <- ggarrange(plot1 + xlim(min(newtabbis1$Axis.2, newtabbis2$Axis.2)-0.1,
                                max(newtabbis1$Axis.1, newtabbis2$Axis.1)+0.15) + 
                    ylim(min(newtabbis1$Axis.2, newtabbis2$Axis.2),
                         max(newtabbis1$Axis.2, newtabbis2$Axis.2)), 
                  plot2 + xlim(min(newtabbis1$Axis.2, newtabbis2$Axis.2)-0.1,
                               max(newtabbis1$Axis.1, newtabbis2$Axis.1)+0.15) + 
                    ylim(min(newtabbis1$Axis.2, newtabbis2$Axis.2),
                         max(newtabbis1$Axis.2, newtabbis2$Axis.2)) +
                    guides(size = "none", color = "none"), 
                  nrow = 2, heights = c(1,1.1))


plot11 <- plot1 + xlim(min(newtabbis1$Axis.2, newtabbis2$Axis.2)-0.1,
                                max(newtabbis1$Axis.1, newtabbis2$Axis.1)+0.15) + 
                    ylim(min(newtabbis1$Axis.2, newtabbis2$Axis.2),
                         max(newtabbis1$Axis.2, newtabbis2$Axis.2))
plot12 <- plot2 + xlim(min(newtabbis1$Axis.2, newtabbis2$Axis.2)-0.1,
                               max(newtabbis1$Axis.1, newtabbis2$Axis.1)+0.15) + 
                    ylim(min(newtabbis1$Axis.2, newtabbis2$Axis.2),
                         max(newtabbis1$Axis.2, newtabbis2$Axis.2))



## Proportional abundances with QMP
phylobj.QMP.prop = subset_samples(phylobj.prop, Transit != "FS")
phylobj.QMP.prop <- subset_samples(phylobj.QMP.prop, Ninetransit ==3)

samp <- sample_data(phylobj.QMP.prop)
samp1 <- subset(samp, Donor ==1 | Donor ==2); samp1$intTransit <- "Short"
samp2 <- subset(samp, Donor ==3 | Donor ==4); samp2$intTransit <- "Medium"
samp3 <- subset(samp, Donor ==5 | Donor ==6); samp3$intTransit <- "Long"
samp <- rbind(samp1, samp2, samp3); samp$intTransit <- as.ordered(factor(samp$intTransit, 
                                                      levels=c("Short", "Medium", "Long"))) 
phylobj.QMP2 <- merge_phyloseq(sample_data(samp), phylobj.QMP.prop)
phylobj.QMP3 <- subset_samples(phylobj.QMP2, PD_Colon == "P")


ord2 <- ordinate(phylobj.QMP3, "PCoA", "bray")
extract_eigenvalue <- function(ordination) UseMethod("extract_eigenvalue", ordination)
extract_eigenvalue = function(ordination) ordination$values$Relative_eig # for PCoA
eigvec2 = extract_eigenvalue(ord2)
fracvar2 = eigvec2[1:2]/sum(eigvec2)
percvar2 = round(100*fracvar2, 0)
plot2 <- plot_ordination(phylobj.QMP2, ord2, color = "Transit", shape = "intTransit") 
plot2 <- papertheme(plot2, size = 20)
plot2 <- plot2 + geom_point(size = 6, alpha = 0.8) +
  scale_color_manual (name = "SHIME transit", values = transitcolors, labels = c("Short", "Medium", "Long")) +
  scale_shape(name = expression(paste(italic("In vivo"), " transit")))+
  scale_fill_manual(values = transitcolors, labels = c("Short", "Medium", "Long")) +
  theme(strip.background =element_rect(fill="#D6E0F4"),
        strip.text.x = element_text(size = 20),
        strip.text.y = element_text(size = 20, angle = 270),
        legend.title = element_text(size=20)) +
  guides (size = "none") 
strivar2 <- as(c("PCoA1", "PCoA2"), "character"); strivar2 <- paste0(strivar2, " ", percvar2, "%")
plot2 <- plot2 + xlab(strivar2[1]) + ylab(strivar2[2])
# Reorder PD_Colon
newtab2 = data.table(plot2$data)
newtab2$PD_Colon <- factor(newtab2$PD_Colon, levels = c("P", "D"))
# newtab2$abund <- rep("Proportional microbial community composition",nrow(newtab2)) # make new column
newtab2$abund <- rep("Proportional community composition",nrow(newtab2)) # make new column
plot2$data <- newtab2; newtabbis2 = newtab2
plot1 <- plot2 + 
  facet_grid(cols = vars(abund), rows = vars(PD_Colon), labeller = labeller(Donor = donor.labs, PD_Colon = PD_Colon.labsPC)) +
  theme(legend.position = "none")
# plot1

phylobj.QMP3 <- subset_samples(phylobj.QMP2, PD_Colon == "D")

ord2 <- ordinate(phylobj.QMP3, "PCoA", "bray")
extract_eigenvalue <- function(ordination) UseMethod("extract_eigenvalue", ordination)
extract_eigenvalue = function(ordination) ordination$values$Relative_eig # for PCoA
eigvec2 = extract_eigenvalue(ord2)
fracvar2 = eigvec2[1:2]/sum(eigvec2)
percvar2 = round(100*fracvar2, 0)

plot2 <- plot_ordination(phylobj.QMP2, ord2, color = "Transit", shape = "intTransit") 
plot2 <- papertheme(plot2, size = 20)
plot2 <- plot2 + geom_point(size = 6, alpha = 0.8) +
  scale_color_manual (name = "SHIME transit", values = transitcolors, labels = c("Short", "Medium", "Long")) +
  scale_shape(name = expression(paste(italic("In vivo"), " transit")))+
  scale_fill_manual(values = transitcolors, labels = c("Short", "Medium", "Long")) +
  theme(strip.background =element_rect(fill="#D6E0F4"),
        legend.box = "horizontal",
        strip.text.x = element_text(size = 20),
        strip.text.y = element_text(size = 20, angle = 270),
        legend.title = element_text(size=20), legend.background = element_blank()) #+
  # guides (size = "none", shape = "none")

strivar2 <- as(c("PCoA1", "PCoA2"), "character"); strivar2 <- paste0(strivar2, " ", percvar2, "%")
plot2 <- plot2 + xlab(strivar2[1]) + ylab(strivar2[2])

# Reorder PD_Colon
newtab2 = data.table(plot2$data)
newtab2$PD_Colon <- factor(newtab2$PD_Colon, levels = c("P", "D"))
# newtab2$abund <- rep("Proportional microbial community composition",nrow(newtab2)) # make new column
newtab2$abund <- rep("Proportional community composition",nrow(newtab2)) # make new column
plot2$data <- newtab2; newtabbis1 = newtab2
plot2 <- plot2 + 
  facet_grid(#cols = vars(abund), 
             rows = vars(PD_Colon), labeller = labeller(Donor = donor.labs, PD_Colon = PD_Colon.labsPC)) +
  theme(legend.position = "bottom", legend.justification = "left", legend.margin = margin(t = 2, r = 0, b = 2, l = 2, unit = "mm")) +
  # theme(legend.position = c(0.2,-0.5)) +
  guides(color = guide_legend(ncol = 3, direction = "horizontal",
                                                    title.position = "left"))
  # theme(legend.position = c(0.16,0.21)) 
  # theme(legend.position = "none")
# plot2


plot13 <- plot1 + xlim(min(newtabbis1$Axis.2, newtabbis2$Axis.2)-0.1,
                                max(newtabbis1$Axis.1, newtabbis2$Axis.1)) + 
                    ylim(min(newtabbis1$Axis.2, newtabbis2$Axis.2),
                         max(newtabbis1$Axis.2, newtabbis2$Axis.2))
plot14 <- plot2 + xlim(min(newtabbis1$Axis.2, newtabbis2$Axis.2),
                               max(newtabbis1$Axis.1, newtabbis2$Axis.1)) + 
                    ylim(min(newtabbis1$Axis.2, newtabbis2$Axis.2),
                         max(newtabbis1$Axis.2, newtabbis2$Axis.2))

plot2 <- ggarrange(plot1 + xlim(min(newtabbis1$Axis.2, newtabbis2$Axis.2)-0.1,
                                max(newtabbis1$Axis.1, newtabbis2$Axis.1)) + 
                    ylim(min(newtabbis1$Axis.2, newtabbis2$Axis.2),
                         max(newtabbis1$Axis.2, newtabbis2$Axis.2)), 
                  plot2 + xlim(min(newtabbis1$Axis.2, newtabbis2$Axis.2),
                               max(newtabbis1$Axis.1, newtabbis2$Axis.1)) + 
                    ylim(min(newtabbis1$Axis.2, newtabbis2$Axis.2),
                         max(newtabbis1$Axis.2, newtabbis2$Axis.2)) +
                    guides (size = "none", shape = "none"), 
                  nrow = 2, heights = c(1,1.1))

plot2

# plot2 <- ggarrange(plot1, plot2, nrow = 2, heights = c(1,0.95))

comboplot <- ggarrange(plot, plot2, ncol = 2, labels = "AUTO", font.label = list(size = 20))
comboplot


comboplot2 <- ggarrange(plot11 + theme(legend.position = "bottom"), 
                        plot13 + theme(legend.position = "bottom"), 
                        plot12 + theme(legend.position = "bottom"),
                        plot14 + theme(legend.position = "bottom"), labels = c("A", "B", "", ""), 
                        font.label = list(size = 20), nrow = 2, ncol=2, 
                        common.legend = T, legend = "bottom"); comboplot2
# ggsave(plot = comboplot, width = 16, height = 10, dpi = 400, filename = "Figure2.png")
# ggsave(plot = comboplot, width = 16, height = 10, dpi = 400, filename = "Figure2_high_res.tiff")
```


## Ordination without shapes
```{r Plot_PCoAs, fig.height=10, fig.width=16, include=TRUE}
# Save location of the plots 
saveplots()
fourcolours = c("#d18fa9","#d1b78f","#8fd1b7","#8fa9d1")
phylobj.QMP2 <- subset_samples(phylobj.QMP, Ninetransit == 3)
# ON GENUS LEVEL
# Ordination graphs  (PCoA or DCA)
ord <- ordinate(phylobj.QMP2, "PCoA", "bray") 
extract_eigenvalue <- function(ordination) UseMethod("extract_eigenvalue", ordination)
extract_eigenvalue = function(ordination) ordination$values$Relative_eig # for PCoA
eigvec = extract_eigenvalue(ord)
fracvar = eigvec[1:2]/sum(eigvec)
percvar = round(100*fracvar, 0)
plot <- plot_ordination(phylobj.QMP2, ord, color = "Transit" 
                        # shape = "Donor"
                        )
plot <- papertheme(plot, size = 20)
plot <- plot + geom_point(size = 6, alpha = 0.7) +
  scale_color_manual (name = "SHIME transit", values = transitcolors, labels = c("Short", "Medium", "Long")) +
  scale_fill_manual(values = transitcolors, labels = c("Short", "Medium", "Long")) +
  theme(strip.background =element_rect(fill="#D6E0F4"),
        strip.text.x = element_text(size = 20),
        strip.text.y = element_text(size = 20, angle = 270),
        legend.title = element_text(size=20)) +
  guides (size = "none") 
strivar <- as(c("PCoA1", "PCoA2"), "character"); strivar <- paste0(strivar, " ", percvar, "%")
plot <- plot + xlab(strivar[1]) + ylab(strivar[2])
# Reorder PD_Colon
newtab = data.table(plot$data)
newtab$PD_Colon <- factor(newtab$PD_Colon, levels = c("P", "D"))
newtab$Donor <- factor(newtab$Donor, levels = c(1,2,3,4,5,6))
newtab$abund <- rep("Quantitative microbial community ",nrow(newtab)) # make new column
plot$data <- newtab
plot <- plot + 
  # scale_size_manual(values=c(6,6,6,4,4,4)) + scale_shape_manual(values=c(0,15,1,16,2,17)) +
  facet_grid(cols = vars(abund),rows = vars(PD_Colon), labeller = labeller(Donor = donor.labs, PD_Colon = PD_Colon.labsPC)) +
  theme(legend.position = c(0.15,0.4))
plot
# ggsave(plot = plot, width = 14, height = 10, dpi = 400, filename = "PCOAQMP.png")
## Proportional abundances with QMP
phylobj.QMP.prop = subset_samples(phylobj.prop, Transit != "FS")
phylobj.QMP.prop <- subset_samples(phylobj.QMP.prop, Ninetransit ==3)
ord2 <- ordinate(phylobj.QMP.prop, "PCoA", "bray")
extract_eigenvalue <- function(ordination) UseMethod("extract_eigenvalue", ordination)
extract_eigenvalue = function(ordination) ordination$values$Relative_eig # for PCoA
eigvec2 = extract_eigenvalue(ord2)
fracvar2 = eigvec2[1:2]/sum(eigvec2)
percvar2 = round(100*fracvar2, 0)
plot2 <- plot_ordination(phylobj.QMP.prop, ord2, color = "Transit") 
plot2 <- papertheme(plot2, size = 20)
plot2 <- plot2 + geom_point(size = 6, alpha = 0.7) +
  scale_color_manual (values = transitcolors, name = "SHIME transit") +
  scale_fill_manual(values = transitcolors, name = "Load") +
  theme(strip.background =element_rect(fill="#D6E0F4"),
        strip.text.x = element_text(size = 20),
        strip.text.y = element_text(size = 20, angle = 270),
        legend.title = element_text(size=20)) +
  guides (size = "none") 
strivar2 <- as(c("PCoA1", "PCoA2"), "character"); strivar2 <- paste0(strivar2, " ", percvar2, "%")
plot2 <- plot2 + xlab(strivar2[1]) + ylab(strivar2[2])
# Reorder PD_Colon
newtab2 = data.table(plot2$data)
newtab2$PD_Colon <- factor(newtab2$PD_Colon, levels = c("P", "D"))
newtab2$abund <- rep("Proportional microbial community",nrow(newtab2)) # make new column
plot2$data <- newtab2
plot2 <- plot2 + 
  facet_grid(cols = vars(abund), rows = vars(PD_Colon), labeller = labeller(Donor = donor.labs, PD_Colon = PD_Colon.labsPC)) +
  theme(legend.position = "none")
plot2
comboplot <- ggarrange(plot, plot2, ncol = 2, labels = "AUTO", font.label = list(size = 20))
comboplot
# ggsave(plot = comboplot, width = 16, height = 10, dpi = 400, filename = "Figure2.png")
# ggsave(plot = comboplot, width = 16, height = 10, dpi = 400, filename = "Figure2_high_res.tiff")
```



## statistics
```{r Ordination statistics QMP}
df.ord.stat <- data.frame()
set.seed(6314)
# Test Of Equality Of Variances
VarianceP <- subset_samples(phylobj.QMP, PD_Colon == "P")
Bray_45P <- phyloseq::distance(VarianceP, method="bray")
df_45P <- as(sample_data(VarianceP), "data.frame")
groupsP <- df_45P[["Nutrient_Load"]]
groupsP
modP <- betadisper(Bray_45P, groupsP)
anova(modP)
VarianceD <- subset_samples(phylobj.QMP, PD_Colon == "D")
Bray_45D <- phyloseq::distance(VarianceD, method="bray")
df_45D <- as(sample_data(VarianceD), "data.frame")
groupsD <- df_45D[["Nutrient_Load"]]
groupsD
modD <- betadisper(Bray_45D, groupsD)
anova(modD)
# Adonis
phylobj.FS.PD_Colonp <- subset_samples(phylobj.QMP, PD_Colon == "P")
metadata1 <- as(sample_data(phylobj.FS.PD_Colonp), "data.frame")
df.ord.stat <- adonis(phyloseq::distance(phylobj.FS.PD_Colonp, method="bray") ~ Nutrient_Load,
                      data = metadata1)
df.ord.stat <- df.ord.stat$aov.tab
df.ord.stat <- df.ord.stat[-c(2,3), ]
df.ord.stat$Nutrient_Load <- "All"
df.ord.stat$Nutrient_Load2 <- "All"
df.ord.stat$PD_Colon <- "PC"
rownames(df.ord.stat) <- c(paste("All, PC"))
for (j in c("100%","33%","20%")) {
  if (j =="100%") {Treat <- c("33%","20%","10%")}
  if (j =="33%") {Treat <- c("20%","10%")}
  if (j =="20%") {Treat <- c("10%")}
  for(x in Treat){
    phylobj.FS.PD_Colonpd <- subset_samples(phylobj.FS.PD_Colonp, Nutrient_Load == x|Nutrient_Load==j)
    metadata2 <- as(sample_data(phylobj.FS.PD_Colonpd), "data.frame")
    df.ord.stat2 <- adonis(phyloseq::distance(phylobj.FS.PD_Colonpd, method="bray") ~ Nutrient_Load,
                 data = metadata2)
    df.ord.stat2 <- df.ord.stat2$aov.tab
    df.ord.stat2 <- df.ord.stat2[-c(2,3), ]
    df.ord.stat2$Nutrient_Load <- j
    df.ord.stat2$Nutrient_Load2 <- x
    df.ord.stat2$PD_Colon <- "PC"
    rownames(df.ord.stat2) <- c(paste("PC",j,",",x))
    df.ord.stat<- rbind(df.ord.stat,df.ord.stat2)
  }
}
phylobj.FS.PD_Colonp <- subset_samples(phylobj.QMP, PD_Colon == "D")
metadata1 <- as(sample_data(phylobj.FS.PD_Colonp), "data.frame")
df.ord.stat2 <- adonis(phyloseq::distance(phylobj.FS.PD_Colonp, method="bray") ~ Nutrient_Load,
       data = metadata1)
df.ord.stat2 <- df.ord.stat2$aov.tab
df.ord.stat2 <- df.ord.stat2[-c(2,3), ]
df.ord.stat2$Nutrient_Load <- j
    df.ord.stat2$Nutrient_Load2 <- x
    df.ord.stat2$PD_Colon <- "DC"
rownames(df.ord.stat2) <- c(paste("All, DC"))
df.ord.stat<- rbind(df.ord.stat,df.ord.stat2)
for (j in c("100%","33%","20%")) {
  if (j =="100%") {Treat <- c("33%","20%","10%")}
  if (j =="33%") {Treat <- c("20%","10%")}
  if (j =="20%") {Treat <- c("10%")}
  for(x in Treat){
    phylobj.FS.PD_Colonpd <- subset_samples(phylobj.FS.PD_Colonp, Nutrient_Load == x|Nutrient_Load==j)
    metadata2 <- as(sample_data(phylobj.FS.PD_Colonpd), "data.frame")
    df.ord.stat2 <- adonis(phyloseq::distance(phylobj.FS.PD_Colonpd, method="bray") ~ Nutrient_Load,
                           data = metadata2)
    df.ord.stat2 <- df.ord.stat2$aov.tab
    df.ord.stat2 <- df.ord.stat2[-c(2,3), ]
    df.ord.stat2$Nutrient_Load <- j
    df.ord.stat2$Nutrient_Load2 <- x
    df.ord.stat2$PD_Colon <- "DC"
    rownames(df.ord.stat2) <- c(paste("DC",j,",",x))
    df.ord.stat<- rbind(df.ord.stat,df.ord.stat2)
  }}
rm(Treat, df.ord.stat2)
saveplots()
write.csv2(df.ord.stat, "ordination.statistics.QMP.csv")
```

# RDA - 3 parameters
## Genus
```{r RDA genus from dataframe, fig.width=13.5, fig.height=15, echo=TRUE, eval = TRUE}
neon <- c("#2f89fc") ; labelcolour <- c("#fc0202"); linecolour <- c("#fc0202"); showR = TRUE
PDcolours <- c("#434e52", "#5b8c85" ) ; ndigits <- 2
legendsize <- 15 ; labelsize <- 6; ygrob <- 0.97; neonsize <- 7; topn <- 15
if(showR == TRUE){xgrob <- 0.85}else{xgrob <- 0.9}; abb_digits = T; dgts = 1; abbnum = 3; alphnumb = 0.62
lefttitle <- "Absolute community abundance" ; righttitle <- "Proportional community abundance"
legendsize <- 18; pointsize = 9; grobsize <- 15; abbrev <- F; plotobjects <- TRUE; plotobjects2 <- F
pos <- position_jitter(width = 0.0, seed = 2)
saveplots()


# ABSOLUTE DATA
## Make data frame of all genera
phylobj.QMP1 <- aggregate_taxa(phylobj.QMP, level = "Genus"); phylobj.QMP1 <- subset_samples(phylobj.QMP1, Ninetransit == 3)
a <- as.data.frame(vegan_otu(phylobj.QMP1))
b <- as.data.frame(sample_data(phylobj.QMP1))
df.phylobj.all <- cbind(Samplenumber = b$Samplenumber,a,b$Day, 
       Transit=b$Transit, Donor = b$Donor, PD_Colon =b$PD_Colon)
cols2.df <- ncol(df.phylobj.all)-4
df.phylobj.all.genus <- df.phylobj.all[2:cols2.df]
## Make data frame of top 15 genera
Top15phylobj <- aggregate_top_taxa(phylobj.QMP, topn, "Genus"); Top15phylobj <- subset_samples(Top15phylobj, Ninetransit == 3)
a <- as.data.frame(vegan_otu(Top15phylobj))
b <- as.data.frame(sample_data(Top15phylobj))
df.Top15phylobj <- cbind(Samplenumber = b$Samplenumber,a,b$Day, 
        Transit=b$Transit, Donor = b$Donor, PD_Colon =b$PD_Colon)
cols.df <- ncol(df.Top15phylobj)-4
df.Top15phylobj.otu <- df.Top15phylobj[2:cols.df]


# Nutrient Load
## All genera
rdafunct <- rda(df.phylobj.all.genus~Transit + Condition(Donor) + Condition(PD_Colon), df.phylobj.all)
coef(rdafunct)

## permutation test of the results
anova.cca(rdafunct,step=1000)
anova.cca(rdafunct,by="term",step=1000)
df.2A <- data.frame("Transit","time","QMP", "Absolute")
rownames(df.2A) <- df.2A$X.Nutrient.
num = anova.cca(rdafunct,step=1000)
num$Df <- as.factor(num$Df); num$Variance <- as.factor(num$Variance); num$F <- as.factor(num$F); num$`Pr(>F)` <- as.factor(num$`Pr(>F)`)
colnames(df.2A) <- colnames(num)
df.2A <- rbind(df.2A, num)

## explanatory power of the variables included == constrained waarde (variance corrected)
Runadj <- RsquareAdj(rdafunct)$r.squared
Radj <- RsquareAdj(rdafunct)$adj.r.squared
## As label on plot
Radjall2 <- signif(Radj, digits = ndigits)
rdafunct <- rda(df.Top15phylobj.otu~Transit + Condition(Donor) + Condition(PD_Colon), df.Top15phylobj)
coef(rdafunct)
## permutation test of the results
anova.cca(rdafunct,step=1000)
anova.cca(rdafunct,by="term",step=1000)
df.A <- data.frame("Transit","time","QMP", "Absolute")
rownames(df.A) <- df.A$X.Nutrient.
num = anova.cca(rdafunct,step=1000)
num$Df <- as.factor(num$Df); num$Variance <- as.factor(num$Variance); num$F <- as.factor(num$F); num$`Pr(>F)` <- as.factor(num$`Pr(>F)`)
colnames(df.A) <- colnames(num)
df.A <- rbind(df.A, num)
## explanatory power of the variables included == constrained waarde (variance corrected) 
Runadj <- RsquareAdj(rdafunct)$r.squared
Radj <- RsquareAdj(rdafunct)$adj.r.squared
Radj


## As label on plot
Radj2 <- signif(Radj, digits = ndigits)
if(showR == TRUE){grob <- grobTree(textGrob(paste("R\u00B2 =",Radjall2, "\nR\u00B2 top", topn, "=", Radj2), x=xgrob,  y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))
}else{
  grob <- grobTree(textGrob(paste("R\u00B2= ",Radjall2), x=xgrob,  y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))
}
## labelling axes
if(abb_digits==T){
 Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,dgts)
 Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,dgts) 
 }else{
   if(round(summary(rdafunct)$cont$importance[2,1]*100,0)==0){
     Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,1)
     }else{
       Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,0)
       }
   if(round(summary(rdafunct)$cont$importance[2,2]*100,0)==0){
     Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,1)
     }else{
       Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,0)
     }
 }
## plot SCALING2
if(plotobjects2 == TRUE){
  plot(rdafunct,scaling=2,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","wa","cn"))
  plot(rdafunct,scaling=1,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","lc","cn"))
}



## ggplot
rdascores <- scores(rdafunct, choices = 1:2, scaling = 2, display = "sites")
firstaxispoints <- rdascores[, 1]; secondaxispoints <- rdascores[, 2]
points <- data.frame(fa = firstaxispoints, sa = secondaxispoints, Donor = df.Top15phylobj$Donor, 
    P_D = df.Top15phylobj$PD_Colon, Transit = df.Top15phylobj$Transit)
points$P_D <- mapvalues(points$P_D, from = c("P", "D"), to = c("Proximal", "Distal"))
rdascorevfa <- data.frame(scores(rdafunct, choices = 1:2, scaling = 2, display = "species"))
rdafactor <- data.frame(summary(rdafunct)$centroids)
rownames(rdafactor) <- mapvalues(rownames(rdafactor), from = c("TransitST", "TransitMT", "TransitLT"), 
                                 to = c("Short", "Medium", "Long"))
plotobj <- ggplot(as.data.frame(points)) + geom_point(aes(x = fa, y = sa, size = pointsize, color = Transit), alpha = 0.8) +
  scale_colour_manual(values = transitcolors, name = "Transit") + xlab(paste("RDA1", Dim1, "%", sep = " ")) + 
    ylab(paste("RDA2", Dim2, "%", sep = " ")) + annotation_custom(grob) + 
    geom_segment(data = rdascorevfa, aes(x = 0, xend = RDA1, y = 0, yend = RDA2), 
        color = linecolour) + geom_text(data = rdafactor, aes(x = RDA1, y = RDA2), 
                                         # fill = "grey", alpha = 0.5,
    color = neon, label = rownames(rdafactor), size = neonsize)
if(abbrev == TRUE){plotobj <- plotobj + geom_label_repel(data = rdascorevfa,  aes(x = RDA1, y = RDA2, 
                                          label = abbreviate(colnames(a),abbnum),
                                          segment.color = "grey50"
                                          # vjust = ifelse(RDA2 >= 0, -0.1, 0.7), 
                                          # hjust = ifelse(RDA1 >= 0, 0, 0.5)
                                          ), 
                                          fill = "grey", alpha = alphnumb, label.size = NA,
                                          color = labelcolour, size = labelsize,
                                          fontface = "italic",
                                          position = pos)
} else { 
      plotobj <- plotobj + geom_label_repel(data = rdascorevfa,  aes(x = RDA1, y = RDA2, 
                                          label = colnames(a),
                                          segment.color = "grey50"
                                          # vjust = ifelse(RDA2 >= 0, -0.1, 0.7), 
                                          # hjust = ifelse(RDA1 >= 0, 0, 0.5)
                                          ), 
                                          fill = "grey", alpha = alphnumb, label.size = NA,
                                          color = labelcolour, size = labelsize,
                                          fontface = "italic",
                                          position = pos)
      }
plotobj <- papertheme(plotobj)
plotobj <- plotobj + guides(color = guide_legend(ncol = 1), size = FALSE) + theme(legend.position = "right")
mylegendPDfunctnet <- g_legend(plotobj)
plotobj <- plotobj + theme(legend.position =  c(0.88,0.05), 
        legend.text = element_text(size=legendsize,family='sans',angle=0),
        legend.title = element_text(size=legendsize,family='sans',angle=0),
        legend.direction = "horizontal") 
plotobjrdatop15PDA <- plotobj + xlim(min(rdascorevfa) - 7500, max(points$fa + 1)) + 
  # ggtitle(lefttitle) + 
  theme(plot.title = element_text(hjust = 0.5, face="bold"))
if(plotobjects == TRUE){plotobjrdatop15PDA}
if(abbrev == TRUE){ggsave(plot = plotobjrdatop15PDA, width = 13.5, height = 15, dpi = 200, filename = "Supplementary_figure9_low_res_ab.png")} else {
  ggsave(plot = plotobjrdatop15PDA, width = 13.5, height = 15, dpi = 200, filename = "Supplementary_figure9_low_res.png")
  ggsave(plot = plotobjrdatop15PDA, width = 13.5, height = 15, dpi = 600, filename = "Supplementary_figure9_high_res.tiff")
}


# Donor
## All genera
rdafunct <- rda(df.phylobj.all.genus~Donor + Condition(Transit) + Condition(PD_Colon), df.phylobj.all)
coef(rdafunct)
## permutation test of the results
anova.cca(rdafunct,step=1000)
anova.cca(rdafunct,by="term",step=1000)
df.2C <- data.frame("Donor","","QMP", "Absolute")
rownames(df.2C) <- df.2C$X.Nutrient.
num = anova.cca(rdafunct,step=1000)
num$Df <- as.factor(num$Df); num$Variance <- as.factor(num$Variance); num$F <- as.factor(num$F); num$`Pr(>F)` <- as.factor(num$`Pr(>F)`)
colnames(df.2C) <- colnames(num)
df.2C <- rbind(df.2C, num)
## explanatory power of the variables included == constrained waarde (variance corrected)
Runadj <- RsquareAdj(rdafunct)$r.squared
Radj <- RsquareAdj(rdafunct)$adj.r.squared
## As label on plot
Radjall2 <- signif(Radj, digits = ndigits)
rdafunct <- rda(df.Top15phylobj.otu~Donor + Condition(Transit) + Condition(PD_Colon), df.Top15phylobj)
#summary(rdafunct)
coef(rdafunct)
## permutation test of the results
anova.cca(rdafunct,step=1000)
anova.cca(rdafunct,by="term",step=1000)
df.C <- data.frame("Donor","","QMP", "Absolute")
rownames(df.C) <- df.C$X.Nutrient. ; num = anova.cca(rdafunct,step=1000)
num$Df <- as.factor(num$Df); num$Variance <- as.factor(num$Variance); num$F <- as.factor(num$F) 
num$`Pr(>F)` <- as.factor(num$`Pr(>F)`); colnames(df.C) <- colnames(num)
df.C <- rbind(df.C, num)
## explanatory power of the variables included == constrained waarde (variance corrected) 
Runadj <- RsquareAdj(rdafunct)$r.squared
Radj <- RsquareAdj(rdafunct)$adj.r.squared
Radj
Radj2 <- signif(Radj, digits = ndigits)
if(showR == TRUE){grob <- grobTree(textGrob(paste("R\u00B2 =",Radjall2, "\nR\u00B2 top", topn, "=", Radj2), x=xgrob,  y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))
}else{
  grob <- grobTree(textGrob(paste("R\u00B2= ",Radjall2), x=xgrob,  y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))
}
## labelling axes
if(abb_digits==T){
 Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,dgts)
 Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,dgts) 
 }else{
   if(round(summary(rdafunct)$cont$importance[2,1]*100,0)==0){
     Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,1)
     }else{
       Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,0)
       }
   if(round(summary(rdafunct)$cont$importance[2,2]*100,0)==0){
     Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,1)
     }else{
       Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,0)
     }
   }
## plot SCALING2
if(plotobjects2 == TRUE){
  plot(rdafunct,scaling=2,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","wa","cn"))
  plot(rdafunct,scaling=1,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","lc","cn"))
}
## ggplot
rdascores <- scores(rdafunct, choices = 1:2, scaling = 2, display = "sites")
firstaxispoints <- rdascores[, 1]; secondaxispoints <- rdascores[, 2]
points <- data.frame(fa = firstaxispoints, sa = secondaxispoints, Donor = df.Top15phylobj$Donor, 
                     P_D = df.Top15phylobj$PD_Colon, Transit = df.Top15phylobj$Transit)
points$P_D <- mapvalues(points$P_D, from = c("P", "D"), to = c("Proximal", "Distal"))
rdascorevfa <- data.frame(scores(rdafunct, choices = 1:2, scaling = 2, display = "species"))
rdafactor <- data.frame(summary(rdafunct)$centroids)
rownames(rdafactor) <- mapvalues(rownames(rdafactor), from = c("Donor1","Donor2","Donor3","Donor4","Donor5","Donor6"),
                                 to = c("1","2", "3", "4", "5", "6"))
plotobj <- ggplot(as.data.frame(points)) + geom_point(aes(x = fa, y = sa, size = pointsize, color = Donor), alpha = 0.8) +
  scale_colour_manual(values = sixdonors, name = "Donor") + 
  xlab(paste("RDA1", Dim1, "%", sep = " ")) + ylab(paste("RDA2", Dim2, "%", sep = " ")) +
  annotation_custom(grob) +
  geom_segment(data = rdascorevfa, aes(x = 0, xend = RDA1, y = 0, yend = RDA2), color = linecolour) +
  geom_text(data = rdafactor, aes(x = RDA1, y = RDA2), color = neon, label = rownames(rdafactor), size = neonsize)
  # geom_text_repel(data = rdafactor, aes(x = RDA1, y = RDA2), color = neon, label = rownames(rdafactor), size = neonsize)
if(abbrev == TRUE){plotobj <- plotobj + geom_label_repel(data = rdascorevfa,  aes(x = RDA1, y = RDA2, 
                                          label = abbreviate(colnames(a),abbnum),
                                          segment.color = "grey50"
                                          # vjust = ifelse(RDA2 >= 0, -0.1, 0.7), 
                                          # hjust = ifelse(RDA1 >= 0, 0, 0.5)
                                          ), 
                                          fill = "grey", alpha = alphnumb, label.size = NA,
                                          color = labelcolour, size = labelsize,
                                          fontface = "italic",
                                          position = pos)
} else {
      plotobj <- plotobj + geom_label_repel(data = rdascorevfa,  aes(x = RDA1, y = RDA2, 
                                          label = colnames(a),
                                          segment.color = "grey50"
                                          # vjust = ifelse(RDA2 >= 0, -0.1, 0.7), 
                                          # hjust = ifelse(RDA1 >= 0, 0, 0.5)
                                          ), 
                                          fill = "grey", alpha = alphnumb, label.size = NA,
                                          color = labelcolour, size = labelsize,
                                          fontface = "italic",
                                          position = pos)}
plotobj <- papertheme(plotobj)
plotobj <- plotobj + guides(color = guide_legend(ncol = 1), size = FALSE) + theme(legend.position = "right")
mylegendPDfunctnet <- g_legend(plotobj)
plotobj <- plotobj + theme(legend.position =  c(0.95,0.09), 
        legend.text = element_text(size=legendsize,family='sans',angle=0),
        legend.title = element_text(size=legendsize,family='sans',angle=0),
        legend.direction = "vertical") 
plotobjrdatop15PDC <- plotobj + xlim(min(rdascorevfa) - 8000, max(points$fa + 1))
if(plotobjects == TRUE){plotobjrdatop15PDC}
if(abbrev == TRUE){ggsave(plot = plotobjrdatop15PDC, width = 13.5, height = 15, dpi = 200, filename = "Supplementary_figure8_low_res_ab.png")} else {ggsave(plot = plotobjrdatop15PDC, width = 13.5, height = 15, dpi = 200, filename = "Supplementary_figure8_low_res.png")
  ggsave(plot = plotobjrdatop15PDC, width = 13.5, height = 15, dpi = 600, filename = "Supplementary_figure8_high_res.tiff")}


# PD_Colon
## All genera
rdafunct <- rda(df.phylobj.all.genus~PD_Colon + Condition(Transit) + Condition(Donor),
                df.phylobj.all)
coef(rdafunct)
## permutation test of the results
anova.cca(rdafunct,step=1000)
anova.cca(rdafunct,by="term",step=1000)
df.2E <- data.frame("PD","COLON","QMP", "Absolute")
rownames(df.2E) <- df.2E$X.Nutrient.
num = anova.cca(rdafunct,step=1000)
num$Df <- as.factor(num$Df); num$Variance <- as.factor(num$Variance); num$F <- as.factor(num$F); num$`Pr(>F)` <- as.factor(num$`Pr(>F)`)
colnames(df.2E) <- colnames(num)
df.2E <- rbind(df.2E, num)
## explanatory power of the variables included == constrained waarde (variance corrected)
Runadj <- RsquareAdj(rdafunct)$r.squared
Radj <- RsquareAdj(rdafunct)$adj.r.squared
## As label on plot
Radjall2 <- signif(Radj, digits = ndigits)
rdafunct <- rda(df.Top15phylobj.otu~PD_Colon + Condition(Transit) + Condition(Donor),df.Top15phylobj)
#summary(rdafunct)
coef(rdafunct)
## permutation test of the results
anova.cca(rdafunct,step=1000)
anova.cca(rdafunct,by="term",step=1000)
df.E <- data.frame("PD","COLON","QMP", "Absolute")
rownames(df.E) <- df.E$X.Nutrient. ; num = anova.cca(rdafunct,step=1000)
num$Df <- as.factor(num$Df); num$Variance <- as.factor(num$Variance); num$F <- as.factor(num$F); 
num$`Pr(>F)` <- as.factor(num$`Pr(>F)`) ; colnames(df.E) <- colnames(num)
df.E <- rbind(df.E, num)
## explanatory power of the variables included == constrained waarde (variance corrected) 
Runadj <- RsquareAdj(rdafunct)$r.squared
Radj <- RsquareAdj(rdafunct)$adj.r.squared
Radj
Radj2 <- signif(Radj, digits = ndigits)
if(showR == TRUE){grob <- grobTree(textGrob(paste("R\u00B2 =",Radjall2, "\nR\u00B2 top", topn, "=", Radj2), x=xgrob,  y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))
}else{
  grob <- grobTree(textGrob(paste("R\u00B2= ",Radjall2), x=xgrob,  y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))
}
## labelling axes
if(abb_digits==T){
 Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,dgts)
 Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,dgts) 
 }else{
   if(round(summary(rdafunct)$cont$importance[2,1]*100,0)==0){
     Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,1)
     }else{
       Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,0)
       }
   if(round(summary(rdafunct)$cont$importance[2,2]*100,0)==0){
     Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,1)
     }else{
       Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,0)
     }
   }
## plot SCALING2
if(plotobjects2 == TRUE){
  plot(rdafunct,scaling=2,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","wa","cn"))
  plot(rdafunct,scaling=1,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","lc","cn"))
}
## ggplot
rdascores <- scores(rdafunct, choices = 1:2, scaling = 2, display = "sites")
firstaxispoints <- rdascores[, 1]; secondaxispoints <- rdascores[, 2]
points <- data.frame(fa = firstaxispoints, sa = secondaxispoints, Donor = df.Top15phylobj$Donor, 
                     P_D = df.Top15phylobj$PD_Colon, Transit = df.Top15phylobj$Transit)
points$P_D <- mapvalues(points$P_D, from = c("P", "D"), to = c("Proximal", "Distal"))
rdascorevfa <- data.frame(scores(rdafunct, choices = 1:2, scaling = 2, display = "species"))
rdafactor <- data.frame(summary(rdafunct)$centroids)
rownames(rdafactor) <- mapvalues(rownames(rdafactor), from = c("PD_ColonP","PD_ColonD"),
                                 to = c("Proximal", "Distal"))
plotobj <- ggplot(as.data.frame(points)) + geom_point(aes(x = fa, y = sa, size = pointsize, color = points$P_D), alpha = 0.8) +
  scale_colour_manual(values = PDcolours, name = "PD_Colon") + 
  xlab(paste("RDA1", Dim1, "%", sep = " ")) + ylab(paste("PC1", Dim2, "%", sep = " ")) +
  annotation_custom(grob) +
  geom_segment(data = rdascorevfa, aes(x = 0, xend = RDA1, y = 0, yend = PC1), color = linecolour) +
  geom_text(data = rdafactor, aes(x = RDA1, y = PC1), color = neon, label = rownames(rdafactor), size = neonsize)
plotobj <- plotobj + geom_text(data = rdascorevfa, aes(x = RDA1, y = PC1, label = abbreviate(colnames(a),3),
  vjust = ifelse(PC1 >= 0, -0.1, 0.7), hjust = ifelse(RDA1 >= 0, 0, 0.5)), color = labelcolour, size = labelsize, 
  #parse = TRUE,    
  fontface = "italic") 
plotobj <- papertheme(plotobj)
plotobj <- plotobj + guides(color = guide_legend(ncol = 1), size = FALSE) + theme(legend.position = "right")
mylegendPDfunctnet <- g_legend(plotobj)
plotobj <- plotobj + theme(legend.position = "none")
plotobjrdatop15PDE <- plotobj + xlim(min(rdascorevfa) - 1, max(points$fa + 1))
if(plotobjects == TRUE){plotobjrdatop15PDE}
# PROPORTIONAL DATA
saveplots()
## Make data frame of all genera
phylobj.QMP.prop = transform_sample_counts(phylobj.QMP, function(x) 100 * x/sum(x))
phylobj.QMP.prop <- subset_samples(phylobj.QMP.prop, Ninetransit ==3)
phylobj.QMP.prop <- aggregate_taxa(phylobj.QMP.prop, "Genus")
a <- as.data.frame(vegan_otu(phylobj.QMP.prop))
b <- as.data.frame(sample_data(phylobj.QMP.prop))
df.phylobj.QMP <- cbind(Samplenumber = b$Samplenumber,a,b$Day, Transit=b$Transit, Donor = b$Donor, PD_Colon =b$PD_Colon)
df.phylobj.QMP.otu <- df.phylobj.QMP[2:cols2.df]
## Top 15 genera
Top15phylobj.prop = transform_sample_counts(Top15phylobj, function(x) 100 * x/sum(x))
Top15phylobj.prop <- subset_samples(Top15phylobj.prop, Ninetransit ==3)
a <- as.data.frame(vegan_otu(Top15phylobj.prop))
b <- as.data.frame(sample_data(Top15phylobj.prop))
df.Top15phylobj <- cbind(Samplenumber = b$Samplenumber,a,b$Day, Transit=b$Transit, Donor = b$Donor, PD_Colon =b$PD_Colon)
df.Top15phylobj.otu <- df.Top15phylobj[2:cols.df]
# Transit time
## all genera 
rdafunct <- rda(df.phylobj.QMP.otu~Transit + Condition(Donor) + Condition(PD_Colon), df.phylobj.QMP)
#summary(rdafunct)
coef(rdafunct)
## permutation test of the results
anova.cca(rdafunct,step=1000)
anova.cca(rdafunct,by="term",step=1000)
df.2B <- data.frame("Transit","time","QMP", "Proportional")
rownames(df.2B) <- df.2B$X.Nutrient. ;num = anova.cca(rdafunct,step=1000)
num$Df <- as.factor(num$Df); num$Variance <- as.factor(num$Variance); num$F <- as.factor(num$F); 
num$`Pr(>F)` <- as.factor(num$`Pr(>F)`); colnames(df.2B) <- colnames(num)
df.2B <- rbind(df.2B, num)
## explanatory power of the variables included == constrained waarde (variance corrected) 
Runadj <- RsquareAdj(rdafunct)$r.squared
Radj <- RsquareAdj(rdafunct)$adj.r.squared
Radj
## As label on plot
Radjall2 <- signif(Radj, digits = ndigits)
rdafunct <- rda(df.Top15phylobj.otu~Transit + Condition(Donor) + Condition(PD_Colon), df.Top15phylobj)
#summary(rdafunct)
coef(rdafunct)
## permutation test of the results
anova.cca(rdafunct,step=1000)
anova.cca(rdafunct,by="term",step=1000)
df.B <- data.frame("Transit","time","QMP", "Proportional")
rownames(df.B) <- df.B$X.Nutrient. ;num = anova.cca(rdafunct,step=1000)
num$Df <- as.factor(num$Df); num$Variance <- as.factor(num$Variance); num$F <- as.factor(num$F); 
num$`Pr(>F)` <- as.factor(num$`Pr(>F)`); colnames(df.B) <- colnames(num)
df.B <- rbind(df.B, num)
## explanatory power of the variables included == constrained waarde (variance corrected) 
Runadj <- RsquareAdj(rdafunct)$r.squared
Radj <- RsquareAdj(rdafunct)$adj.r.squared
Radj
## As label on plot
Radj2 <- signif(Radj, digits = ndigits)
if(showR == TRUE){grob <- grobTree(textGrob(paste("R\u00B2 =",Radjall2, "\nR\u00B2 top", topn, "=", Radj2), x=xgrob,  y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))
}else{
  grob <- grobTree(textGrob(paste("R\u00B2= ",Radjall2), x=xgrob,  y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))
}
## labelling axes
if(abb_digits==T){
 Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,dgts)
 Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,dgts) 
 }else{
   if(round(summary(rdafunct)$cont$importance[2,1]*100,0)==0){
     Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,1)
     }else{
       Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,0)
       }
   if(round(summary(rdafunct)$cont$importance[2,2]*100,0)==0){
     Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,1)
     }else{
       Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,0)
     }
   }
## plot SCALING2
if(plotobjects2 == TRUE){
  plot(rdafunct,scaling=2,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","wa","cn"))
  plot(rdafunct,scaling=1,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","lc","cn"))
}
## ggplot
rdascores <- scores(rdafunct, choices = 1:2, scaling = 2, display = "sites")
firstaxispoints <- rdascores[, 1]; secondaxispoints <- rdascores[, 2]
points <- data.frame(fa = firstaxispoints, sa = secondaxispoints, Donor = df.Top15phylobj$Donor, 
    P_D = df.Top15phylobj$PD_Colon, Transit = df.Top15phylobj$Transit)
points$P_D <- mapvalues(points$P_D, from = c("P", "D"), to = c("Proximal", "Distal"))
rdascorevfa <- data.frame(scores(rdafunct, choices = 1:2, scaling = 2, display = "species"))
rdafactor <- data.frame(summary(rdafunct)$centroids)
rownames(rdafactor) <- mapvalues(rownames(rdafactor), from = c("TransitST", "TransitMT", "TransitLT"), 
                                 to = c("Short", "Medium", "Long"))
plotobj <- ggplot(as.data.frame(points)) + geom_point(aes(x = fa, y = sa, size = pointsize, color = Transit), alpha = 0.8) +
  scale_colour_manual(values = transitcolors, name = "Transit") + xlab(paste("RDA1", Dim1, "%", sep = " ")) + 
    ylab(paste("RDA2", Dim2, "%", sep = " ")) + annotation_custom(grob) + 
    geom_segment(data = rdascorevfa, aes(x = 0, xend = RDA1, y = 0, yend = RDA2), 
        color = linecolour) + geom_text(data = rdafactor, aes(x = RDA1, y = RDA2), 
    color = neon, label = rownames(rdafactor), size = neonsize)
if(abbrev == TRUE){plotobj <- plotobj + geom_label_repel(data = rdascorevfa,  aes(x = RDA1, y = RDA2, 
                                          label = abbreviate(colnames(a),abbnum),
                                          segment.color = "grey50"
                                          # vjust = ifelse(RDA2 >= 0, -0.1, 0.7), 
                                          # hjust = ifelse(RDA1 >= 0, 0, 0.5)
                                          ), 
                                          fill = "grey", alpha = alphnumb, label.size = NA,
                                          color = labelcolour, size = labelsize,
                                          fontface = "italic",
                                          position = pos)
} else {options(ggrepel.max.overlaps = Inf)
      plotobj <- plotobj + geom_label_repel(data = rdascorevfa,  aes(x = RDA1, y = RDA2, 
                                          label = colnames(a),
                                          segment.color = "grey50"
                                          # vjust = ifelse(RDA2 >= 0, -0.1, 0.7), 
                                          # hjust = ifelse(RDA1 >= 0, 0, 0.5)
                                          ), 
                                          fill = "grey", alpha = alphnumb, label.size = NA,
                                          color = labelcolour, size = labelsize,
                                          fontface = "italic",
                                          position = pos)
      }
plotobj <- papertheme(plotobj)
plotobj <- plotobj + guides(color = guide_legend(ncol = 1), size = FALSE) + 
  theme(legend.position = "right", legend.text = element_text(size = legendsize), legend.title = element_text(size = legendsize))
mylegendNL <- g_legend(plotobj)
plotobj <- plotobj + theme(legend.position =  c(0.88,0.05), 
        legend.text = element_text(size=legendsize,family='sans',angle=0),
        legend.title = element_text(size=legendsize,family='sans',angle=0),
        legend.direction = "horizontal") 
plotobjrdatop15PDB <- plotobj + xlim(min(rdascorevfa) - 1, max(points$fa + 1)) + 
  # ggtitle(righttitle) +
  theme(plot.title = element_text(hjust = 0.5, face="bold"))
if(plotobjects == TRUE){plotobjrdatop15PDB}
if(abbrev == TRUE) {ggsave(plot = plotobjrdatop15PDB, width = 13.5, height = 15, dpi = 200, filename = "Supplementary_figure11_low_res_ab.png")} else {
  ggsave(plot = plotobjrdatop15PDB, width = 13.5, height = 15, dpi = 200, filename = "Supplementary_figure11_low_res.png")
  ggsave(plot = plotobjrdatop15PDB, width = 13.5, height = 15, dpi = 600, filename = "Supplementary_figure11_high_res.tiff")
}
# Donor
## all genera 
rdafunct <- rda(df.phylobj.QMP.otu~Donor+ Condition(Transit) + Condition(PD_Colon), df.phylobj.QMP)
#summary(rdafunct)
coef(rdafunct)
## permutation test of the results
anova.cca(rdafunct,step=1000)
anova.cca(rdafunct,by="term",step=1000)
df.2D <- data.frame("Donor","","QMP", "Proportional")
rownames(df.2D) <- df.2D$X.Nutrient. ;num = anova.cca(rdafunct,step=1000)
num$Df <- as.factor(num$Df); num$Variance <- as.factor(num$Variance); num$F <- as.factor(num$F); 
num$`Pr(>F)` <- as.factor(num$`Pr(>F)`); colnames(df.2D) <- colnames(num)
df.2D <- rbind(df.2D, num)
## explanatory power of the variables included == constrained waarde (variance corrected) 
Runadj <- RsquareAdj(rdafunct)$r.squared
Radj <- RsquareAdj(rdafunct)$adj.r.squared
Radj
## As label on plot
Radjall2 <- signif(Radj, digits = ndigits)
rdafunct <- rda(df.Top15phylobj.otu~Donor+ Condition(Transit) + Condition(PD_Colon), df.Top15phylobj)
#summary(rdafunct)
coef(rdafunct)
## permutation test of the results
anova.cca(rdafunct,step=1000)
anova.cca(rdafunct,by="term",step=1000)
df.D <- data.frame("Donor","","QMP", "Proportional")
rownames(df.D) <- df.D$X.Nutrient. ; num = anova.cca(rdafunct,step=1000)
num$Df <- as.factor(num$Df); num$Variance <- as.factor(num$Variance); num$F <- as.factor(num$F); 
num$`Pr(>F)` <- as.factor(num$`Pr(>F)`); colnames(df.D) <- colnames(num)
df.D <- rbind(df.D, num)
## explanatory power of the variables included == constrained waarde (variance corrected) 
Runadj <- RsquareAdj(rdafunct)$r.squared
Radj <- RsquareAdj(rdafunct)$adj.r.squared
Radj
Radj2 <- signif(Radj, digits = ndigits)
if(showR == TRUE){grob <- grobTree(textGrob(paste("R\u00B2 =",Radjall2, "\nR\u00B2 top", topn, "=", Radj2), x=xgrob,  y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))
}else{
  grob <- grobTree(textGrob(paste("R\u00B2= ",Radjall2), x=xgrob,  y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))
}
## labelling axes
if(abb_digits==T){
 Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,dgts)
 Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,dgts) 
 }else{
   if(round(summary(rdafunct)$cont$importance[2,1]*100,0)==0){
     Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,1)
     }else{
       Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,0)
       }
   if(round(summary(rdafunct)$cont$importance[2,2]*100,0)==0){
     Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,1)
     }else{
       Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,0)
     }
   }
   
## plot SCALING2
if(plotobjects2 == TRUE){
  plot(rdafunct,scaling=2,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","wa","cn"))
  plot(rdafunct,scaling=1,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","lc","cn"))
}
## ggplot
rdascores <- scores(rdafunct, choices = 1:2, scaling = 2, display = "sites")
firstaxispoints <- rdascores[, 1]; secondaxispoints <- rdascores[, 2]
points <- data.frame(fa = firstaxispoints, sa = secondaxispoints, Donor = df.Top15phylobj$Donor, 
                     P_D = df.Top15phylobj$PD_Colon, Transit = df.Top15phylobj$Transit)
points$P_D <- mapvalues(points$P_D, from = c("P", "D"), to = c("Proximal", "Distal"))
rdascorevfa <- data.frame(scores(rdafunct, choices = 1:2, scaling = 2, display = "species"))
rdafactor <- data.frame(summary(rdafunct)$centroids)
rownames(rdafactor) <- mapvalues(rownames(rdafactor), from = c("Donor1","Donor2","Donor3","Donor4","Donor5","Donor6"),
                                 to = c("1","2", "3", "4", "5", "6"))
plotobj <- ggplot(as.data.frame(points)) + geom_point(aes(x = fa, y = sa, size = pointsize, color = Donor), alpha = 0.8) +
  scale_colour_manual(values = sixdonors, name = "Donor") + 
  xlab(paste("RDA1", Dim1, "%", sep = " ")) + ylab(paste("RDA2", Dim2, "%", sep = " ")) +
  annotation_custom(grob) +
  geom_segment(data = rdascorevfa, aes(x = 0, xend = RDA1, y = 0, yend = RDA2), color = linecolour) +
  geom_text(data = rdafactor, aes(x = RDA1, y = RDA2), color = neon, label = rownames(rdafactor), size = neonsize)
if(abbrev == TRUE){plotobj <- plotobj + geom_label_repel(data = rdascorevfa,  aes(x = RDA1, y = RDA2, 
                                          label = abbreviate(colnames(a),abbnum),
                                          segment.color = "grey50"
                                          # vjust = ifelse(RDA2 >= 0, -0.1, 0.7), 
                                          # hjust = ifelse(RDA1 >= 0, 0, 0.5)
                                          ), 
                                          fill = "grey", alpha = alphnumb, label.size = NA,
                                          color = labelcolour, size = labelsize,
                                          fontface = "italic",
                                          position = pos)
} else {
      plotobj <- plotobj + geom_label_repel(data = rdascorevfa,  aes(x = RDA1, y = RDA2, 
                                          label = colnames(a),
                                          segment.color = "grey50"
                                          # vjust = ifelse(RDA2 >= 0, -0.1, 0.7), 
                                          # hjust = ifelse(RDA1 >= 0, 0, 0.5)
                                          ), 
                                          fill = "grey", alpha = alphnumb, label.size = NA,
                                          color = labelcolour, size = labelsize,
                                          fontface = "italic",
                                          position = pos)}
plotobj <- papertheme(plotobj)
plotobj <- plotobj + guides(color = guide_legend(ncol = 1), size = FALSE) + 
  theme(legend.position = "right", legend.text = element_text(size = legendsize), legend.title = element_text(size = legendsize))
mylegendDo <- g_legend(plotobj)
plotobj <- plotobj + theme(legend.position =  c(0.95,0.09), 
        legend.text = element_text(size=legendsize,family='sans',angle=0),
        legend.title = element_text(size=legendsize,family='sans',angle=0),
        legend.direction = "vertical") 
plotobjrdatop15PDD <- plotobj + xlim(min(rdascorevfa) - 1, max(points$fa + 3)) 
if(plotobjects == TRUE){plotobjrdatop15PDD}
if (abbrev == TRUE){
  ggsave(plot = plotobjrdatop15PDD, width = 13.5, height = 15, dpi = 200, filename = "Supplementary_figure10_low_res_ab.png")
  } else
    {
  ggsave(plot = plotobjrdatop15PDD, width = 13.5, height = 15, dpi = 200, filename = "Supplementary_figure10_low_res.png")
  ggsave(plot = plotobjrdatop15PDD, width = 13.5, height = 15, dpi = 600, filename = "Supplementary_figure10_high_res.tiff")
    }
# PD_Colon
## all genera 
rdafunct <- rda(df.phylobj.QMP.otu~PD_Colon+Condition(Transit) + Condition(Donor), df.phylobj.QMP)
#summary(rdafunct)
coef(rdafunct)
## permutation test of the results
anova.cca(rdafunct,step=1000)
anova.cca(rdafunct,by="term",step=1000)
df.2F <- data.frame("PD","Colon","QMP", "Proportional")
rownames(df.2F) <- df.2F$X.Nutrient. ;num = anova.cca(rdafunct,step=1000)
num$Df <- as.factor(num$Df); num$Variance <- as.factor(num$Variance); num$F <- as.factor(num$F); 
num$`Pr(>F)` <- as.factor(num$`Pr(>F)`); colnames(df.2F) <- colnames(num)
df.2F <- rbind(df.2F, num)
## explanatory power of the variables included == constrained waarde (variance corrected) 
Runadj <- RsquareAdj(rdafunct)$r.squared
Radj <- RsquareAdj(rdafunct)$adj.r.squared
Radj
## As label on plot
Radjall2 <- signif(Radj, digits = ndigits)
rdafunct <- rda(df.Top15phylobj.otu~PD_Colon+Condition(Transit) + Condition(Donor), df.Top15phylobj)
#summary(rdafunct)
coef(rdafunct)
## permutation test of the results
anova.cca(rdafunct,step=1000)
anova.cca(rdafunct,by="term",step=1000)
df.F <- data.frame("PD","Colon","QMP", "Proportional")
rownames(df.F) <- df.C$X.Nutrient. ; num = anova.cca(rdafunct,step=1000)
num$Df <- as.factor(num$Df); num$Variance <- as.factor(num$Variance); num$F <- as.factor(num$F); 
num$`Pr(>F)` <- as.factor(num$`Pr(>F)`); colnames(df.F) <- colnames(num)
df.F <- rbind(df.F, num)
## explanatory power of the variables included == constrained waarde (variance corrected) 
Runadj <- RsquareAdj(rdafunct)$r.squared
Radj <- RsquareAdj(rdafunct)$adj.r.squared
Radj
Radj2 <- signif(Radj, digits = ndigits)
if(showR == TRUE){grob <- grobTree(textGrob(paste("R\u00B2 =",Radjall2, "\nR\u00B2 top", topn, "=", Radj2), x=xgrob,  y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))
}else{
  grob <- grobTree(textGrob(paste("R\u00B2= ",Radjall2), x=xgrob,  y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))
}
## labelling axes
if(abb_digits==T){
 Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,dgts)
 Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,dgts) 
 }else{
   if(round(summary(rdafunct)$cont$importance[2,1]*100,0)==0){
     Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,1)
     }else{
       Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,0)
       }
   if(round(summary(rdafunct)$cont$importance[2,2]*100,0)==0){
     Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,1)
     }else{
       Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,0)
     }
   }
## plot SCALING2
if(plotobjects2 == TRUE){
  plot(rdafunct,scaling=2,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","wa","cn"))
  plot(rdafunct,scaling=1,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","lc","cn"))
}
## ggplot
rdascores <- scores(rdafunct, choices = 1:2, scaling = 2, display = "sites")
firstaxispoints <- rdascores[, 1]; secondaxispoints <- rdascores[, 2]
points <- data.frame(fa = firstaxispoints, sa = secondaxispoints, Donor = df.Top15phylobj$Donor, 
                     P_D = df.Top15phylobj$PD_Colon, Transit = df.Top15phylobj$Transit)
points$P_D <- mapvalues(points$P_D, from = c("1", "2"), to = c("Proximal", "Distal"))
rdascorevfa <- data.frame(scores(rdafunct, choices = 1:2, scaling = 2, display = "species"))
rdafactor <- data.frame(summary(rdafunct)$centroids)
rownames(rdafactor) <- mapvalues(rownames(rdafactor), from = c("PD_ColonP","PD_ColonD"),
                                 to = c("Proximal", "Distal"))
plotobj <- ggplot(as.data.frame(points)) + geom_point(aes(x = fa, y = sa, size = pointsize, color = points$P_D), alpha = 0.8) +
  scale_colour_manual(values = vesselcolors, name = "Colon") + 
  xlab(paste("RDA1", Dim1, "%", sep = " ")) + ylab(paste("PC1", Dim2, "%", sep = " ")) +
  annotation_custom(grob) +
  geom_segment(data = rdascorevfa, aes(x = 0, xend = RDA1, y = 0, yend = PC1), color = linecolour) +
  geom_text(data = rdafactor, aes(x = RDA1, y = PC1), color = neon, label = rownames(rdafactor), size = neonsize)
plotobj <- plotobj + geom_text(data = rdascorevfa, aes(x = RDA1, y = PC1, label = abbreviate(colnames(a),3),
  vjust = ifelse(PC1 >= 0, -0.1, 0.7), hjust = ifelse(RDA1 >= 0, 0, 0.5)), color = labelcolour, size = labelsize, 
  #parse = TRUE,  
  fontface = "italic") 
plotobj <- papertheme(plotobj)
plotobj <- plotobj + guides(color = guide_legend(ncol = 1), size = FALSE) + 
  theme(legend.position = "right", legend.text = element_text(size = legendsize), legend.title = element_text(size = legendsize))
mylegendPD <- g_legend(plotobj)
plotobj <- plotobj + theme(legend.position = "none")
plotobjrdatop15PDF <- plotobj + xlim(min(rdascorevfa) - 1, max(points$fa + 1))
if(plotobjects == TRUE){plotobjrdatop15PDF}
# Combine the plots into 1
comboplot <- ggarrange(ggarrange(plotobjrdatop15PDA, plotobjrdatop15PDC, plotobjrdatop15PDE, 
                                 nrow = 3, labels = c("A", "C","E"), heights = c(1,0.95,0.95), font.label = list(size = 20)),
                       ggarrange(plotobjrdatop15PDB, plotobjrdatop15PDD, plotobjrdatop15PDF,
                                 nrow = 3, labels = c("B", "D","F"), heights = c(1,0.95,0.95), font.label = list(size = 20)),
                       ggarrange(mylegendNL, mylegendDo, mylegendPD,
                                 nrow = 3, labels = c("", "", "")),
                       ncol = 3, labels = c("","",""), widths = c(1.5,1.5,0.3))
comboplot
saveplots()
# ggsave(plot = comboplot, width = 13.5, height = 15, dpi = 300, filename = "Paperplot.RDA.QMP.png")
df.anova.cca.QMP <- rbind(df.A, df.B, df.C, df.D, df.E, df.F)
tabler <- tableGrob(df.anova.cca.QMP)
grid.newpage()
grid.draw(tabler)
df.anova.cca.QMP2 <- rbind(df.2A, df.2B, df.2C, df.2D, df.2E, df.2F)
tabler <- tableGrob(df.anova.cca.QMP2)
grid.newpage()
grid.draw(tabler)
# df.anova.cca <- cbind(df.anova.cca.Met, df.anova.cca.QMP)
Abbreviations <- paste(abbreviate(colnames(a), 3),"=",colnames(a), collapse = ", ")
Abbreviations
```

## OTU
```{r RDA OTU from dataframe, fig.width=13.5, fig.height=15, echo=TRUE, eval = TRUE}
neon <- c("#2f89fc") ; labelcolour <- c("#fc0202"); linecolour <- c("#fc0202")
PDcolours <- c("#434e52", "#5b8c85" ) ; ndigits <- 2
sixdonors <- c("#974908", "#089648", "#480896", "#085696", "#960856", "#569608")
sixdonors <- c("#5c68c1", "#c15c68", "#488740", "#c1b55c", "#8cd3cb", "#D99DC3")
legendsize <- 15 ; labelsize <- 6; ygrob <- 0.97; neonsize <- 7; topn <- 30
showR = TRUE
if(showR == TRUE){xgrob <- 0.85}else{xgrob <- 0.9}
lefttitle <- "Absolute community abundance" ; righttitle <- "Proportional community abundance"
legendsize <- 18; pointsize = 10; grobsize <- 15; abbrev <- FALSE; plotobjects <- TRUE
saveplots()
# ABSOLUTE DATA
## Make data frame of all genera
a <- as.data.frame(vegan_otu(phylobj.QMP.otu))
b <- as.data.frame(sample_data(phylobj.QMP.otu))
df.phylobj.all <- cbind(Samplenumber = b$Samplenumber,a,b$Day, 
       Nutrient_Load=b$Nutrient_Load, Donor = b$Donor, PD_Colon =b$PD_Colon)
cols2.df <- ncol(df.phylobj.all)-4
df.phylobj.all.genus <- df.phylobj.all[2:cols2.df]
## Make data frame of top 15 genera
Top15phylobj <- aggregate_top_otu(phylobj.QMP.otu, topn)
# Top15phylobj <- phylobj.QMP.otu
a <- as.data.frame(vegan_otu(Top15phylobj))
b <- as.data.frame(sample_data(Top15phylobj))
df.Top15phylobj <- cbind(Samplenumber = b$Samplenumber,a,b$Nutrient_Load.Day, 
        Nutrient_Load=b$Nutrient_Load, Donor = b$Donor, PD_Colon =b$PD_Colon)
cols.df <- ncol(df.Top15phylobj)-4
df.Top15phylobj.otu <- df.Top15phylobj[2:cols.df]
# Nutrient Load
## All genera
rdafunct <- rda(df.phylobj.all.genus~Nutrient_Load + Condition(Donor) + Condition(PD_Colon), df.phylobj.all)
coef(rdafunct)
## permutation test of the results
anova.cca(rdafunct,step=1000)
anova.cca(rdafunct,by="term",step=1000)
df.2A <- data.frame("Nutrient","Load","QMP", "Absolute")
rownames(df.2A) <- df.2A$X.Nutrient.
num = anova.cca(rdafunct,step=1000)
num$Df <- as.factor(num$Df); num$Variance <- as.factor(num$Variance); num$F <- as.factor(num$F); num$`Pr(>F)` <- as.factor(num$`Pr(>F)`)
colnames(df.2A) <- colnames(num)
df.2A <- rbind(df.2A, num)
## explanatory power of the variables included == constrained waarde (variance corrected)
Runadj <- RsquareAdj(rdafunct)$r.squared
Radj <- RsquareAdj(rdafunct)$adj.r.squared
## As label on plot
Radjall2 <- signif(Radj, digits = ndigits)
rdafunct <- rda(df.Top15phylobj.otu~Nutrient_Load + Condition(Donor) + Condition(PD_Colon), df.Top15phylobj)
coef(rdafunct)
## permutation test of the results
anova.cca(rdafunct,step=1000)
anova.cca(rdafunct,by="term",step=1000)
df.A <- data.frame("Nutrient","Load","QMP", "Absolute")
rownames(df.A) <- df.A$X.Nutrient.
num = anova.cca(rdafunct,step=1000)
num$Df <- as.factor(num$Df); num$Variance <- as.factor(num$Variance); num$F <- as.factor(num$F); num$`Pr(>F)` <- as.factor(num$`Pr(>F)`)
colnames(df.A) <- colnames(num)
df.A <- rbind(df.A, num)
## explanatory power of the variables included == constrained waarde (variance corrected) 
Runadj <- RsquareAdj(rdafunct)$r.squared
Radj <- RsquareAdj(rdafunct)$adj.r.squared
Radj
## As label on plot
Radj2 <- signif(Radj, digits = ndigits)
if(showR == TRUE){grob <- grobTree(textGrob(paste("R\u00B2 =",Radjall2, "\nR\u00B2 top", topn, "=", Radj2), x=xgrob,  y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))
}else{
  grob <- grobTree(textGrob(paste("R\u00B2= ",Radjall2), x=xgrob,  y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))
}
## labelling axes
if(abb_digits==T){
 Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,dgts)
 Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,dgts) 
 }else{
   if(round(summary(rdafunct)$cont$importance[2,1]*100,0)==0){
     Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,1)
     }else{
       Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,0)
       }
   if(round(summary(rdafunct)$cont$importance[2,2]*100,0)==0){
     Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,1)
     }else{
       Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,0)
     }
   }
## plot SCALING2
if(plotobjects == TRUE){
  plot(rdafunct,scaling=2,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","wa","cn"))
  plot(rdafunct,scaling=1,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","lc","cn"))
}
## ggplot
rdascores <- scores(rdafunct, choices = 1:2, scaling = 2, display = "sites")
firstaxispoints <- rdascores[, 1]; secondaxispoints <- rdascores[, 2]
points <- data.frame(fa = firstaxispoints, sa = secondaxispoints, Donor = df.Top15phylobj$Donor, 
    P_D = df.Top15phylobj$PD_Colon, Nutrient_Load = df.Top15phylobj$Nutrient_Load)
points$P_D <- mapvalues(points$P_D, from = c("P", "D"), to = c("Proximal", "Distal"))
rdascorevfa <- data.frame(scores(rdafunct, choices = 1:2, scaling = 2, display = "species"))
rdafactor <- data.frame(summary(rdafunct)$centroids)
rownames(rdafactor) <- mapvalues(rownames(rdafactor), from = c("Nutrient_Load100%", "Nutrient_Load33%", "Nutrient_Load20%",
   "Nutrient_Load10%"), to = c("100", "33", "20", "10"))
plotobj <- ggplot(as.data.frame(points)) + geom_point(aes(x = fa, y = sa, size = pointsize, color = Nutrient_Load)) +
  scale_colour_manual(values = fourcolours, name = "Nutrient load") + xlab(paste("RDA1", Dim1, "%", sep = " ")) + 
    ylab(paste("RDA2", Dim2, "%", sep = " ")) + annotation_custom(grob) + 
    geom_segment(data = rdascorevfa, aes(x = 0, xend = RDA1, y = 0, yend = RDA2), 
        color = linecolour) + geom_text(data = rdafactor, aes(x = RDA1, y = RDA2), 
    color = neon, label = rownames(rdafactor), size = neonsize)
if(abbrev == TRUE){plotobj <- plotobj + geom_text(data = rdascorevfa, aes(x = RDA1, y = RDA2, label = abbreviate(colnames(a),3),
    vjust = ifelse(RDA2 >= 0, -0.1, 0.7), hjust = ifelse(RDA1 >= 0, 0, 0.5)), color = labelcolour, size = labelsize,
    #parse = TRUE,  
    fontface = "italic")
} else {
  # x <- colnames(a);  x <- sub("Otu000", "OTU", x); x <- sub("Otu00", "OTU", x);x <- sub("Otu0", "OTU", x)
  x <- colnames(a);  x <- sub("Otu000", "", x); x <- sub("Otu00", "", x);x <- sub("Otu0", "", x)
  plotobj <- plotobj + geom_label_repel(data = rdascorevfa,  aes(x = RDA1, y = RDA2, label = x,
                                                                 segment.color = "grey50"),
                                        fill = "grey", alpha = alphnumb, label.size = NA,
                                          color = labelcolour, size = labelsize,
                                          fontface = "italic",  position = pos)
  # plotobj <- plotobj + geom_text_repel(data = rdascorevfa,  aes(x = RDA1, y = RDA2, label = x,
  #                                                               segment.color = "grey50"
  #                                                               # vjust = ifelse(RDA2 >= 0, -0.1, 0.7),
  #                                                               # hjust = ifelse(RDA1 >= 0, 0, 0.5)
  #                                                               ), 
  #                                      color = labelcolour, size = labelsize, fontface = "italic")
      }
plotobj <- papertheme(plotobj)
plotobj <- plotobj + guides(color = guide_legend(ncol = 1), size = FALSE) + theme(legend.position = "right")
mylegendPDfunctnet <- g_legend(plotobj)
plotobj <- plotobj + theme(legend.position =  c(0.88,0.05), 
        legend.text = element_text(size=legendsize,family='sans',angle=0),
        legend.title = element_text(size=legendsize,family='sans',angle=0),
        legend.direction = "horizontal") 
plotobjrdatop15PDA <- plotobj + xlim(min(rdascorevfa) - 1000, max(points$fa + 1)) + 
  # ggtitle(lefttitle) + 
  theme(plot.title = element_text(hjust = 0.5, face="bold"))
if(plotobjects == TRUE){plotobjrdatop15PDA}
if(abbrev == TRUE){ggsave(plot = plotobjrdatop15PDA, width = 13.5, height = 15, dpi = 200, filename = "Paperplot.RDA.otu.ABS_NL_ab.png")} else {
  ggsave(plot = plotobjrdatop15PDA, width = 13.5, height = 15, dpi = 200, filename = "Supplementary_figure13_low_res.png")
  ggsave(plot = plotobjrdatop15PDA, width = 13.5, height = 15, dpi = 600, filename = "Supplementary_figure13_high_res.tiff")
  
}
Abbreviations <- paste(abbreviate(colnames(a), 3),"=",colnames(a), collapse = ", ")
Abbreviations
# Donor
## All genera
rdafunct <- rda(df.phylobj.all.genus~Donor + Condition(Nutrient_Load) + Condition(PD_Colon), df.phylobj.all)
coef(rdafunct)
## permutation test of the results
anova.cca(rdafunct,step=1000)
anova.cca(rdafunct,by="term",step=1000)
df.2C <- data.frame("Donor","","QMP", "Absolute")
rownames(df.2C) <- df.2C$X.Nutrient.
num = anova.cca(rdafunct,step=1000)
num$Df <- as.factor(num$Df); num$Variance <- as.factor(num$Variance); num$F <- as.factor(num$F); num$`Pr(>F)` <- as.factor(num$`Pr(>F)`)
colnames(df.2C) <- colnames(num)
df.2C <- rbind(df.2C, num)
## explanatory power of the variables included == constrained waarde (variance corrected)
Runadj <- RsquareAdj(rdafunct)$r.squared
Radj <- RsquareAdj(rdafunct)$adj.r.squared
## As label on plot
Radjall2 <- signif(Radj, digits = ndigits)
rdafunct <- rda(df.Top15phylobj.otu~Donor + Condition(Nutrient_Load) + Condition(PD_Colon), df.Top15phylobj)
#summary(rdafunct)
coef(rdafunct)
## permutation test of the results
anova.cca(rdafunct,step=1000)
anova.cca(rdafunct,by="term",step=1000)
df.C <- data.frame("Donor","","QMP", "Absolute")
rownames(df.C) <- df.C$X.Nutrient. ; num = anova.cca(rdafunct,step=1000)
num$Df <- as.factor(num$Df); num$Variance <- as.factor(num$Variance); num$F <- as.factor(num$F) 
num$`Pr(>F)` <- as.factor(num$`Pr(>F)`); colnames(df.C) <- colnames(num)
df.C <- rbind(df.C, num)
## explanatory power of the variables included == constrained waarde (variance corrected) 
Runadj <- RsquareAdj(rdafunct)$r.squared
Radj <- RsquareAdj(rdafunct)$adj.r.squared
Radj
Radj2 <- signif(Radj, digits = ndigits)
if(showR == TRUE){grob <- grobTree(textGrob(paste("R\u00B2 =",Radjall2, "\nR\u00B2 top", topn, "=", Radj2), x=xgrob,  y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))
}else{
  grob <- grobTree(textGrob(paste("R\u00B2= ",Radjall2), x=xgrob,  y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))
}
## labelling axes
if(abb_digits==T){
 Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,dgts)
 Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,dgts) 
 }else{
   if(round(summary(rdafunct)$cont$importance[2,1]*100,0)==0){
     Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,1)
     }else{
       Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,0)
       }
   if(round(summary(rdafunct)$cont$importance[2,2]*100,0)==0){
     Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,1)
     }else{
       Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,0)
     }
   }
## plot SCALING2
if(plotobjects == TRUE){
  plot(rdafunct,scaling=2,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","wa","cn"))
  plot(rdafunct,scaling=1,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","lc","cn"))
}
## ggplot
rdascores <- scores(rdafunct, choices = 1:2, scaling = 2, display = "sites")
firstaxispoints <- rdascores[, 1]; secondaxispoints <- rdascores[, 2]
points <- data.frame(fa = firstaxispoints, sa = secondaxispoints, Donor = df.Top15phylobj$Donor, 
                     P_D = df.Top15phylobj$PD_Colon, Nutrient_Load = df.Top15phylobj$Nutrient_Load)
points$P_D <- mapvalues(points$P_D, from = c("P", "D"), to = c("Proximal", "Distal"))
rdascorevfa <- data.frame(scores(rdafunct, choices = 1:2, scaling = 2, display = "species"))
rdafactor <- data.frame(summary(rdafunct)$centroids)
rownames(rdafactor) <- mapvalues(rownames(rdafactor), from = c("Donor1","Donor2","Donor3","Donor4","Donor5","Donor6"),
                                 to = c("1","2", "3", "4", "5", "6"))
plotobj <- ggplot(as.data.frame(points)) + geom_point(aes(x = fa, y = sa, size = pointsize, color = Donor)) +
  scale_colour_manual(values = sixdonors, name = "Donor") + 
  xlab(paste("RDA1", Dim1, "%", sep = " ")) + ylab(paste("RDA2", Dim2, "%", sep = " ")) +
  annotation_custom(grob) +
  geom_segment(data = rdascorevfa, aes(x = 0, xend = RDA1, y = 0, yend = RDA2), color = linecolour) +
  geom_text(data = rdafactor, aes(x = RDA1, y = RDA2), color = neon, label = rownames(rdafactor), size = neonsize)
if(abbrev == TRUE){plotobj <- plotobj + geom_text(data = rdascorevfa, aes(x = RDA1, y = RDA2, label = abbreviate(colnames(a),3),
    vjust = ifelse(RDA2 >= 0, -0.1, 0.7), hjust = ifelse(RDA1 >= 0, 0, 0.5)), color = labelcolour, size = labelsize, 
    #parse = TRUE,   
    fontface = "italic")
} else {
  # x <- colnames(a);  x <- sub("Otu000", "OTU", x); x <- sub("Otu00", "OTU", x);x <- sub("Otu0", "OTU", x)
  x <- colnames(a);  x <- sub("Otu000", "", x); x <- sub("Otu00", "", x);x <- sub("Otu0", "", x)
  plotobj <- plotobj + geom_label_repel(data = rdascorevfa,  aes(x = RDA1, y = RDA2, label = x,
                                                                 segment.color = "grey50"),
                                        fill = "grey", alpha = alphnumb, label.size = NA,
                                          color = labelcolour, size = labelsize,
                                          fontface = "italic",  position = pos)
  # plotobj <- plotobj + geom_text_repel(data = rdascorevfa,  aes(x = RDA1, y = RDA2, label = x,
  #                                                               segment.color = "grey50"
  #                                                               # vjust = ifelse(RDA2 >= 0, -0.1, 0.7),
  #                                                               # hjust = ifelse(RDA1 >= 0, 0, 0.5)
  #                                                               ), 
  #                                      color = labelcolour, size = labelsize, fontface = "italic")
      }
plotobj <- papertheme(plotobj)
plotobj <- plotobj + guides(color = guide_legend(ncol = 1), size = FALSE) + theme(legend.position = "right")
mylegendPDfunctnet <- g_legend(plotobj)
plotobj <- plotobj + theme(legend.position =  c(0.95,0.09), 
        legend.text = element_text(size=legendsize,family='sans',angle=0),
        legend.title = element_text(size=legendsize,family='sans',angle=0),
        legend.direction = "vertical") 
plotobjrdatop15PDC <- plotobj + xlim(min(rdascorevfa) - 1000, max(points$fa + 1))
if(plotobjects == TRUE){plotobjrdatop15PDC}
if(abbrev == TRUE){ggsave(plot = plotobjrdatop15PDC, width = 13.5, height = 15, dpi = 200, filename = "Paperplot.RDA.otu.ABS_Donor_ab.png")} else {
  ggsave(plot = plotobjrdatop15PDC, width = 13.5, height = 15, dpi = 200, filename = "Supplementary_figure12_low_res.png")
  ggsave(plot = plotobjrdatop15PDC, width = 13.5, height = 15, dpi = 600, filename = "Supplementary_figure12_high_res.tiff")
  }
Abbreviations <- paste(abbreviate(colnames(a), 3),"=",colnames(a), collapse = ", ")
Abbreviations
# PD_Colon
## All genera
rdafunct <- rda(df.phylobj.all.genus~PD_Colon + Condition(Nutrient_Load) + Condition(Donor),
                df.phylobj.all)
coef(rdafunct)
## permutation test of the results
anova.cca(rdafunct,step=1000)
anova.cca(rdafunct,by="term",step=1000)
df.2E <- data.frame("PD","COLON","QMP", "Absolute")
rownames(df.2E) <- df.2E$X.Nutrient.
num = anova.cca(rdafunct,step=1000)
num$Df <- as.factor(num$Df); num$Variance <- as.factor(num$Variance); num$F <- as.factor(num$F); num$`Pr(>F)` <- as.factor(num$`Pr(>F)`)
colnames(df.2E) <- colnames(num)
df.2E <- rbind(df.2E, num)
## explanatory power of the variables included == constrained waarde (variance corrected)
Runadj <- RsquareAdj(rdafunct)$r.squared
Radj <- RsquareAdj(rdafunct)$adj.r.squared
## As label on plot
Radjall2 <- signif(Radj, digits = ndigits)
rdafunct <- rda(df.Top15phylobj.otu~PD_Colon + Condition(Nutrient_Load) + Condition(Donor),df.Top15phylobj)
#summary(rdafunct)
coef(rdafunct)
## permutation test of the results
anova.cca(rdafunct,step=1000)
anova.cca(rdafunct,by="term",step=1000)
df.E <- data.frame("PD","COLON","QMP", "Absolute")
rownames(df.E) <- df.E$X.Nutrient. ; num = anova.cca(rdafunct,step=1000)
num$Df <- as.factor(num$Df); num$Variance <- as.factor(num$Variance); num$F <- as.factor(num$F); 
num$`Pr(>F)` <- as.factor(num$`Pr(>F)`) ; colnames(df.E) <- colnames(num)
df.E <- rbind(df.E, num)
## explanatory power of the variables included == constrained waarde (variance corrected) 
Runadj <- RsquareAdj(rdafunct)$r.squared
Radj <- RsquareAdj(rdafunct)$adj.r.squared
Radj
Radj2 <- signif(Radj, digits = ndigits)
if(showR == TRUE){grob <- grobTree(textGrob(paste("R\u00B2 =",Radjall2, "\nR\u00B2 top", topn, "=", Radj2), x=xgrob,  y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))
}else{
  grob <- grobTree(textGrob(paste("R\u00B2= ",Radjall2), x=xgrob,  y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))
}
## labelling axes
if(abb_digits==T){
 Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,dgts)
 Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,dgts) 
 }else{
   if(round(summary(rdafunct)$cont$importance[2,1]*100,0)==0){
     Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,1)
     }else{
       Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,0)
       }
   if(round(summary(rdafunct)$cont$importance[2,2]*100,0)==0){
     Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,1)
     }else{
       Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,0)
     }
   }
## plot SCALING2
if(plotobjects == TRUE){
  plot(rdafunct,scaling=2,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","wa","cn"))
  plot(rdafunct,scaling=1,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","lc","cn"))
}
## ggplot
rdascores <- scores(rdafunct, choices = 1:2, scaling = 2, display = "sites")
firstaxispoints <- rdascores[, 1]; secondaxispoints <- rdascores[, 2]
points <- data.frame(fa = firstaxispoints, sa = secondaxispoints, Donor = df.Top15phylobj$Donor, 
                     P_D = df.Top15phylobj$PD_Colon, Nutrient_Load = df.Top15phylobj$Nutrient_Load)
points$P_D <- mapvalues(points$P_D, from = c("P", "D"), to = c("Proximal", "Distal"))
rdascorevfa <- data.frame(scores(rdafunct, choices = 1:2, scaling = 2, display = "species"))
rdafactor <- data.frame(summary(rdafunct)$centroids)
rownames(rdafactor) <- mapvalues(rownames(rdafactor), from = c("PD_ColonP","PD_ColonD"),
                                 to = c("Proximal", "Distal"))
plotobj <- ggplot(as.data.frame(points)) + geom_point(aes(x = fa, y = sa, size = pointsize, color = points$P_D)) +
  scale_colour_manual(values = vesselcolors, name = "PD_Colon") + 
  xlab(paste("RDA1", Dim1, "%", sep = " ")) + ylab(paste("PC1", Dim2, "%", sep = " ")) +
  annotation_custom(grob) +
  geom_segment(data = rdascorevfa, aes(x = 0, xend = RDA1, y = 0, yend = PC1), color = linecolour) +
  geom_text(data = rdafactor, aes(x = RDA1, y = PC1), color = neon, label = rownames(rdafactor), size = neonsize)
plotobj <- plotobj + geom_text(data = rdascorevfa, aes(x = RDA1, y = PC1, label = abbreviate(colnames(a),3),
  vjust = ifelse(PC1 >= 0, -0.1, 0.7), hjust = ifelse(RDA1 >= 0, 0, 0.5)), color = labelcolour, size = labelsize, 
  #parse = TRUE,    
  fontface = "italic") 
plotobj <- papertheme(plotobj)
plotobj <- plotobj + guides(color = guide_legend(ncol = 1), size = FALSE) + theme(legend.position = "right")
mylegendPDfunctnet <- g_legend(plotobj)
plotobj <- plotobj + theme(legend.position = "none")
plotobjrdatop15PDE <- plotobj + xlim(min(rdascorevfa) - 1, max(points$fa + 1))
if(plotobjects == TRUE){plotobjrdatop15PDE}
Abbreviations <- paste(abbreviate(colnames(a), 3),"=",colnames(a), collapse = ", ")
Abbreviations
# PROPORTIONAL DATA
## Make data frame of all genera
phylobj.QMP.prop = transform_sample_counts(phylobj.QMP.otu, function(x) 100 * x/sum(x))
a <- as.data.frame(vegan_otu(phylobj.QMP.prop))
b <- as.data.frame(sample_data(phylobj.QMP.prop))
df.phylobj.QMP <- cbind(Samplenumber = b$Samplenumber,a,b$Nutrient_Load.Day, Nutrient_Load=b$Nutrient_Load, Donor = b$Donor, PD_Colon =b$PD_Colon)
df.phylobj.QMP.otu <- df.phylobj.QMP[2:cols2.df]
## Top 15 genera
Top15phylobj.prop = transform_sample_counts(Top15phylobj, function(x) 100 * x/sum(x))
a <- as.data.frame(vegan_otu(Top15phylobj.prop))
b <- as.data.frame(sample_data(Top15phylobj.prop))
df.Top15phylobj <- cbind(Samplenumber = b$Samplenumber,a,b$Nutrient_Load.Day, Nutrient_Load=b$Nutrient_Load, Donor = b$Donor, PD_Colon =b$PD_Colon)
df.Top15phylobj.otu <- df.Top15phylobj[2:cols.df]
# Nutrient Load
## all genera 
rdafunct <- rda(df.phylobj.QMP.otu~Nutrient_Load + Condition(Donor) + Condition(PD_Colon), df.phylobj.QMP)
#summary(rdafunct)
coef(rdafunct)
## permutation test of the results
anova.cca(rdafunct,step=1000)
anova.cca(rdafunct,by="term",step=1000)
df.2B <- data.frame("Nutrient","Load","QMP", "Proportional")
rownames(df.2B) <- df.2B$X.Nutrient. ;num = anova.cca(rdafunct,step=1000)
num$Df <- as.factor(num$Df); num$Variance <- as.factor(num$Variance); num$F <- as.factor(num$F); 
num$`Pr(>F)` <- as.factor(num$`Pr(>F)`); colnames(df.2B) <- colnames(num)
df.2B <- rbind(df.2B, num)
## explanatory power of the variables included == constrained waarde (variance corrected) 
Runadj <- RsquareAdj(rdafunct)$r.squared
Radj <- RsquareAdj(rdafunct)$adj.r.squared
Radj
## As label on plot
Radjall2 <- signif(Radj, digits = ndigits)
rdafunct <- rda(df.Top15phylobj.otu~Nutrient_Load + Condition(Donor) + Condition(PD_Colon), df.Top15phylobj)
#summary(rdafunct)
coef(rdafunct)
## permutation test of the results
anova.cca(rdafunct,step=1000)
anova.cca(rdafunct,by="term",step=1000)
df.B <- data.frame("Nutrient","Load","QMP", "Proportional")
rownames(df.B) <- df.B$X.Nutrient. ;num = anova.cca(rdafunct,step=1000)
num$Df <- as.factor(num$Df); num$Variance <- as.factor(num$Variance); num$F <- as.factor(num$F); 
num$`Pr(>F)` <- as.factor(num$`Pr(>F)`); colnames(df.B) <- colnames(num)
df.B <- rbind(df.B, num)
## explanatory power of the variables included == constrained waarde (variance corrected) 
Runadj <- RsquareAdj(rdafunct)$r.squared
Radj <- RsquareAdj(rdafunct)$adj.r.squared
Radj
## As label on plot
Radj2 <- signif(Radj, digits = ndigits)
if(showR == TRUE){grob <- grobTree(textGrob(paste("R\u00B2 =",Radjall2, "\nR\u00B2 top", topn, "=", Radj2), x=xgrob,  y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))
}else{
  grob <- grobTree(textGrob(paste("R\u00B2= ",Radjall2), x=xgrob,  y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))
}
## labelling axes
if(abb_digits==T){
 Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,dgts)
 Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,dgts) 
 }else{
   if(round(summary(rdafunct)$cont$importance[2,1]*100,0)==0){
     Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,1)
     }else{
       Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,0)
       }
   if(round(summary(rdafunct)$cont$importance[2,2]*100,0)==0){
     Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,1)
     }else{
       Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,0)
     }
   }
## plot SCALING2
if(plotobjects == TRUE){
  plot(rdafunct,scaling=2,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","wa","cn"))
  plot(rdafunct,scaling=1,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","lc","cn"))
}
## ggplot
rdascores <- scores(rdafunct, choices = 1:2, scaling = 2, display = "sites")
firstaxispoints <- rdascores[, 1]; secondaxispoints <- rdascores[, 2]
points <- data.frame(fa = firstaxispoints, sa = secondaxispoints, Donor = df.Top15phylobj$Donor, 
    P_D = df.Top15phylobj$PD_Colon, Nutrient_Load = df.Top15phylobj$Nutrient_Load)
points$P_D <- mapvalues(points$P_D, from = c("P", "D"), to = c("Proximal", "Distal"))
rdascorevfa <- data.frame(scores(rdafunct, choices = 1:2, scaling = 2, display = "species"))
rdafactor <- data.frame(summary(rdafunct)$centroids)
rownames(rdafactor) <- mapvalues(rownames(rdafactor), from = c("Nutrient_Load100%", "Nutrient_Load33%", "Nutrient_Load20%",
   "Nutrient_Load10%"), to = c("100", "33", "20", "10"))
plotobj <- ggplot(as.data.frame(points)) + geom_point(aes(x = fa, y = sa, size = pointsize, color = Nutrient_Load)) +
  scale_colour_manual(values = fourcolours, name = "Nutrient load") + xlab(paste("RDA1", Dim1, "%", sep = " ")) + 
    ylab(paste("RDA2", Dim2, "%", sep = " ")) + annotation_custom(grob) + 
    geom_segment(data = rdascorevfa, aes(x = 0, xend = RDA1, y = 0, yend = RDA2), 
        color = linecolour) + geom_text(data = rdafactor, aes(x = RDA1, y = RDA2), 
    color = neon, label = rownames(rdafactor), size = neonsize)
if(abbrev == TRUE){plotobj <- plotobj + geom_text(data = rdascorevfa, aes(x = RDA1, y = RDA2, label = abbreviate(colnames(a),3),
    vjust = ifelse(RDA2 >= 0, -0.1, 0.7), hjust = ifelse(RDA1 >= 0, 0, 0.5)), color = labelcolour, size = labelsize,
    #parse = TRUE, 
    fontface = "italic")
} else {
  # x <- colnames(a);  x <- sub("Otu000", "OTU", x); x <- sub("Otu00", "OTU", x);x <- sub("Otu0", "OTU", x)
  x <- colnames(a);  x <- sub("Otu000", "", x); x <- sub("Otu00", "", x);x <- sub("Otu0", "", x)
  plotobj <- plotobj + geom_label_repel(data = rdascorevfa,  aes(x = RDA1, y = RDA2, label = x,
                                                                 segment.color = "grey50"),
                                        fill = "grey", alpha = alphnumb, label.size = NA,
                                          color = labelcolour, size = labelsize,
                                          fontface = "italic",  position = pos)
  # plotobj <- plotobj + geom_text_repel(data = rdascorevfa,  aes(x = RDA1, y = RDA2, label = x,
  #                                                               segment.color = "grey50"
  #                                                               # vjust = ifelse(RDA2 >= 0, -0.1, 0.7),
  #                                                               # hjust = ifelse(RDA1 >= 0, 0, 0.5)
  #                                                               ), 
  #                                      color = labelcolour, size = labelsize, fontface = "italic")
      }
plotobj <- papertheme(plotobj)
plotobj <- plotobj + guides(color = guide_legend(ncol = 1), size = FALSE) + 
  theme(legend.position = "right", legend.text = element_text(size = legendsize), legend.title = element_text(size = legendsize))
mylegendNL <- g_legend(plotobj)
plotobj <- plotobj + theme(legend.position =  c(0.88,0.05), 
        legend.text = element_text(size=legendsize,family='sans',angle=0),
        legend.title = element_text(size=legendsize,family='sans',angle=0),
        legend.direction = "horizontal") 
plotobjrdatop15PDB <- plotobj + xlim(min(rdascorevfa) - 1, max(points$fa + 1)) + 
  # ggtitle(righttitle) +
  theme(plot.title = element_text(hjust = 0.5, face="bold"))
if(plotobjects == TRUE){plotobjrdatop15PDB}
if(abbrev == TRUE) {ggsave(plot = plotobjrdatop15PDB, width = 13.5, height = 15, dpi = 200, filename = "Paperplot.RDA.otu.PROP_NL_ab.png")} else {
  ggsave(plot = plotobjrdatop15PDB, width = 13.5, height = 15, dpi = 200, filename = "Supplementary_figure15_low_res.png")
  ggsave(plot = plotobjrdatop15PDB, width = 13.5, height = 15, dpi = 600, filename = "Supplementary_figure15_high_res.tiff")
}
Abbreviations <- paste(abbreviate(colnames(a), 3),"=",colnames(a), collapse = ", ")
Abbreviations
# Donor
## all genera 
rdafunct <- rda(df.phylobj.QMP.otu~Donor+ Condition(Nutrient_Load) + Condition(PD_Colon), df.phylobj.QMP)
#summary(rdafunct)
coef(rdafunct)
## permutation test of the results
anova.cca(rdafunct,step=1000)
anova.cca(rdafunct,by="term",step=1000)
df.2D <- data.frame("Donor","","QMP", "Proportional")
rownames(df.2D) <- df.2D$X.Nutrient. ;num = anova.cca(rdafunct,step=1000)
num$Df <- as.factor(num$Df); num$Variance <- as.factor(num$Variance); num$F <- as.factor(num$F); 
num$`Pr(>F)` <- as.factor(num$`Pr(>F)`); colnames(df.2D) <- colnames(num)
df.2D <- rbind(df.2D, num)
## explanatory power of the variables included == constrained waarde (variance corrected) 
Runadj <- RsquareAdj(rdafunct)$r.squared
Radj <- RsquareAdj(rdafunct)$adj.r.squared
Radj
## As label on plot
Radjall2 <- signif(Radj, digits = ndigits)
rdafunct <- rda(df.Top15phylobj.otu~Donor+ Condition(Nutrient_Load) + Condition(PD_Colon), df.Top15phylobj)
#summary(rdafunct)
coef(rdafunct)
## permutation test of the results
anova.cca(rdafunct,step=1000)
anova.cca(rdafunct,by="term",step=1000)
df.D <- data.frame("Donor","","QMP", "Proportional")
rownames(df.D) <- df.D$X.Nutrient. ; num = anova.cca(rdafunct,step=1000)
num$Df <- as.factor(num$Df); num$Variance <- as.factor(num$Variance); num$F <- as.factor(num$F); 
num$`Pr(>F)` <- as.factor(num$`Pr(>F)`); colnames(df.D) <- colnames(num)
df.D <- rbind(df.D, num)
## explanatory power of the variables included == constrained waarde (variance corrected) 
Runadj <- RsquareAdj(rdafunct)$r.squared
Radj <- RsquareAdj(rdafunct)$adj.r.squared
Radj
Radj2 <- signif(Radj, digits = ndigits)
if(showR == TRUE){grob <- grobTree(textGrob(paste("R\u00B2 =",Radjall2, "\nR\u00B2 top", topn, "=", Radj2), x=xgrob,  y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))
}else{
  grob <- grobTree(textGrob(paste("R\u00B2= ",Radjall2), x=xgrob,  y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))
}
## labelling axes
if(abb_digits==T){
 Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,dgts)
 Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,dgts) 
 }else{
   if(round(summary(rdafunct)$cont$importance[2,1]*100,0)==0){
     Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,1)
     }else{
       Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,0)
       }
   if(round(summary(rdafunct)$cont$importance[2,2]*100,0)==0){
     Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,1)
     }else{
       Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,0)
     }
   }
## plot SCALING2
if(plotobjects == TRUE){
  plot(rdafunct,scaling=2,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","wa","cn"))
  plot(rdafunct,scaling=1,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","lc","cn"))
}
## ggplot
rdascores <- scores(rdafunct, choices = 1:2, scaling = 2, display = "sites")
firstaxispoints <- rdascores[, 1]; secondaxispoints <- rdascores[, 2]
points <- data.frame(fa = firstaxispoints, sa = secondaxispoints, Donor = df.Top15phylobj$Donor, 
                     P_D = df.Top15phylobj$PD_Colon, Nutrient_Load = df.Top15phylobj$Nutrient_Load)
points$P_D <- mapvalues(points$P_D, from = c("P", "D"), to = c("Proximal", "Distal"))
rdascorevfa <- data.frame(scores(rdafunct, choices = 1:2, scaling = 2, display = "species"))
rdafactor <- data.frame(summary(rdafunct)$centroids)
rownames(rdafactor) <- mapvalues(rownames(rdafactor), from = c("Donor1","Donor2","Donor3","Donor4","Donor5","Donor6"),
                                 to = c("1","2", "3", "4", "5", "6"))
plotobj <- ggplot(as.data.frame(points)) + geom_point(aes(x = fa, y = sa, size = pointsize, color = Donor)) +
  scale_colour_manual(values = sixdonors, name = "Donor") + 
  xlab(paste("RDA1", Dim1, "%", sep = " ")) + ylab(paste("RDA2", Dim2, "%", sep = " ")) +
  annotation_custom(grob) +
  geom_segment(data = rdascorevfa, aes(x = 0, xend = RDA1, y = 0, yend = RDA2), color = linecolour) +
  geom_text(data = rdafactor, aes(x = RDA1, y = RDA2), color = neon, label = rownames(rdafactor), size = neonsize)
if(abbrev == TRUE){plotobj <- plotobj + geom_text(data = rdascorevfa, aes(x = RDA1, y = RDA2, label = abbreviate(colnames(a),3),
    vjust = ifelse(RDA2 >= 0, -0.1, 0.7), hjust = ifelse(RDA1 >= 0, 0, 0.5)), color = labelcolour, size = labelsize,
    #parse = TRUE, 
    fontface = "italic")
} else {
  # x <- colnames(a);  x <- sub("Otu000", "OTU", x); x <- sub("Otu00", "OTU", x);x <- sub("Otu0", "OTU", x)
  x <- colnames(a);  x <- sub("Otu000", "", x); x <- sub("Otu00", "", x);x <- sub("Otu0", "", x)
  plotobj <- plotobj + geom_label_repel(data = rdascorevfa,  aes(x = RDA1, y = RDA2, label = x,
                                                                 segment.color = "grey50"),
                                        fill = "grey", alpha = alphnumb, label.size = NA,
                                          color = labelcolour, size = labelsize,
                                          fontface = "italic",  position = pos)
  # plotobj <- plotobj + geom_text_repel(data = rdascorevfa,  aes(x = RDA1, y = RDA2, label = x,
  #                                                               segment.color = "grey50"
  #                                                               # vjust = ifelse(RDA2 >= 0, -0.1, 0.7),
  #                                                               # hjust = ifelse(RDA1 >= 0, 0, 0.5)
  #                                                               ), 
  #                                      color = labelcolour, size = labelsize, fontface = "italic")
      }
plotobj <- papertheme(plotobj)
plotobj <- plotobj + guides(color = guide_legend(ncol = 1), size = FALSE) + 
  theme(legend.position = "right", legend.text = element_text(size = legendsize), legend.title = element_text(size = legendsize))
mylegendDo <- g_legend(plotobj)
plotobj <- plotobj + theme(legend.position =  c(0.95,0.09), 
        legend.text = element_text(size=legendsize,family='sans',angle=0),
        legend.title = element_text(size=legendsize,family='sans',angle=0),
        legend.direction = "vertical") 
plotobjrdatop15PDD <- plotobj + xlim(min(rdascorevfa) - 1, max(points$fa + 3)) 
if(plotobjects == TRUE){plotobjrdatop15PDD}
if (abbrev == TRUE){ggsave(plot = plotobjrdatop15PDD, width = 13.5, height = 15, dpi = 200, filename = "Paperplot.RDA.otu.PROP_Donor_ab.png")} else{
  ggsave(plot = plotobjrdatop15PDD, width = 13.5, height = 15, dpi = 200, filename = "Supplementary_figure14_low_res.png")
  ggsave(plot = plotobjrdatop15PDD, width = 13.5, height = 15, dpi = 600, filename = "Supplementary_figure14_high_res.tiff")
}
Abbreviations <- paste(abbreviate(colnames(a), 3),"=",colnames(a), collapse = ", ")
Abbreviations
# PD_Colon
## all genera 
rdafunct <- rda(df.phylobj.QMP.otu~PD_Colon+Condition(Nutrient_Load) + Condition(Donor), df.phylobj.QMP)
#summary(rdafunct)
coef(rdafunct)
## permutation test of the results
anova.cca(rdafunct,step=1000)
anova.cca(rdafunct,by="term",step=1000)
df.2F <- data.frame("PD","Colon","QMP", "Proportional")
rownames(df.2F) <- df.2F$X.Nutrient. ;num = anova.cca(rdafunct,step=1000)
num$Df <- as.factor(num$Df); num$Variance <- as.factor(num$Variance); num$F <- as.factor(num$F); 
num$`Pr(>F)` <- as.factor(num$`Pr(>F)`); colnames(df.2F) <- colnames(num)
df.2F <- rbind(df.2F, num)
## explanatory power of the variables included == constrained waarde (variance corrected) 
Runadj <- RsquareAdj(rdafunct)$r.squared
Radj <- RsquareAdj(rdafunct)$adj.r.squared
Radj
## As label on plot
Radjall2 <- signif(Radj, digits = ndigits)
rdafunct <- rda(df.Top15phylobj.otu~PD_Colon+Condition(Nutrient_Load) + Condition(Donor), df.Top15phylobj)
#summary(rdafunct)
coef(rdafunct)
## permutation test of the results
anova.cca(rdafunct,step=1000)
anova.cca(rdafunct,by="term",step=1000)
df.F <- data.frame("PD","Colon","QMP", "Proportional")
rownames(df.F) <- df.C$X.Nutrient. ; num = anova.cca(rdafunct,step=1000)
num$Df <- as.factor(num$Df); num$Variance <- as.factor(num$Variance); num$F <- as.factor(num$F); 
num$`Pr(>F)` <- as.factor(num$`Pr(>F)`); colnames(df.F) <- colnames(num)
df.F <- rbind(df.F, num)
## explanatory power of the variables included == constrained waarde (variance corrected) 
Runadj <- RsquareAdj(rdafunct)$r.squared
Radj <- RsquareAdj(rdafunct)$adj.r.squared
Radj
Radj2 <- signif(Radj, digits = ndigits)
if(showR == TRUE){grob <- grobTree(textGrob(paste("R\u00B2 =",Radjall2, "\nR\u00B2 top", topn, "=", Radj2), x=xgrob,  y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))
}else{
  grob <- grobTree(textGrob(paste("R\u00B2= ",Radjall2), x=xgrob,  y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))
}
## labelling axes
if(abb_digits==T){
 Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,dgts)
 Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,dgts) 
 }else{
   if(round(summary(rdafunct)$cont$importance[2,1]*100,0)==0){
     Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,1)
     }else{
       Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,0)
       }
   if(round(summary(rdafunct)$cont$importance[2,2]*100,0)==0){
     Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,1)
     }else{
       Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,0)
     }
   }
## plot SCALING2
if(plotobjects == TRUE){
  plot(rdafunct,scaling=2,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","wa","cn"))
  plot(rdafunct,scaling=1,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","lc","cn"))
}
## ggplot
rdascores <- scores(rdafunct, choices = 1:2, scaling = 2, display = "sites")
firstaxispoints <- rdascores[, 1]; secondaxispoints <- rdascores[, 2]
points <- data.frame(fa = firstaxispoints, sa = secondaxispoints, Donor = df.Top15phylobj$Donor, 
                     P_D = df.Top15phylobj$PD_Colon, Nutrient_Load = df.Top15phylobj$Nutrient_Load)
points$P_D <- mapvalues(points$P_D, from = c("1", "2"), to = c("Proximal", "Distal"))
rdascorevfa <- data.frame(scores(rdafunct, choices = 1:2, scaling = 2, display = "species"))
rdafactor <- data.frame(summary(rdafunct)$centroids)
rownames(rdafactor) <- mapvalues(rownames(rdafactor), from = c("PD_ColonP","PD_ColonD"),
                                 to = c("Proximal", "Distal"))
plotobj <- ggplot(as.data.frame(points)) + geom_point(aes(x = fa, y = sa, size = pointsize, color = points$P_D)) +
  scale_colour_manual(values = PDcolours, name = "Colon") + 
  xlab(paste("RDA1", Dim1, "%", sep = " ")) + ylab(paste("PC1", Dim2, "%", sep = " ")) +
  annotation_custom(grob) +
  geom_segment(data = rdascorevfa, aes(x = 0, xend = RDA1, y = 0, yend = PC1), color = linecolour) +
  geom_text(data = rdafactor, aes(x = RDA1, y = PC1), color = neon, label = rownames(rdafactor), size = neonsize)
plotobj <- plotobj + geom_text(data = rdascorevfa, aes(x = RDA1, y = PC1, label = abbreviate(colnames(a),3),
  vjust = ifelse(PC1 >= 0, -0.1, 0.7), hjust = ifelse(RDA1 >= 0, 0, 0.5)), color = labelcolour, size = labelsize, 
  #parse = TRUE,  
  fontface = "italic") 
plotobj <- papertheme(plotobj)
plotobj <- plotobj + guides(color = guide_legend(ncol = 1), size = FALSE) + 
  theme(legend.position = "right", legend.text = element_text(size = legendsize), legend.title = element_text(size = legendsize))
mylegendPD <- g_legend(plotobj)
plotobj <- plotobj + theme(legend.position = "none")
plotobjrdatop15PDF <- plotobj + xlim(min(rdascorevfa) - 1, max(points$fa + 1))
if(plotobjects == TRUE){plotobjrdatop15PDF}
# Combine the plots into 1
comboplot <- ggarrange(ggarrange(plotobjrdatop15PDA, plotobjrdatop15PDC, plotobjrdatop15PDE, 
                                 nrow = 3, labels = c("A", "C","E"), heights = c(1,0.95,0.95), font.label = list(size = 20)),
                       ggarrange(plotobjrdatop15PDB, plotobjrdatop15PDD, plotobjrdatop15PDF,
                                 nrow = 3, labels = c("B", "D","F"), heights = c(1,0.95,0.95), font.label = list(size = 20)),
                       ggarrange(mylegendNL, mylegendDo, mylegendPD,
                                 nrow = 3, labels = c("", "", "")),
                       ncol = 3, labels = c("","",""), widths = c(1.5,1.5,0.3))
# comboplot
saveplots()
# ggsave(plot = comboplot, width = 13.5, height = 15, dpi = 300, filename = "Paperplot.RDA.otu.QMP.png")
df.anova.cca.QMP <- rbind(df.A, df.B, df.C, df.D, df.E, df.F)
tabler <- tableGrob(df.anova.cca.QMP)
grid.newpage()
grid.draw(tabler)
df.anova.cca.QMP2 <- rbind(df.2A, df.2B, df.2C, df.2D, df.2E, df.2F)
tabler <- tableGrob(df.anova.cca.QMP2)
grid.newpage()
grid.draw(tabler)
# df.anova.cca <- cbind(df.anova.cca.Met, df.anova.cca.QMP)
Abbreviations <- paste(abbreviate(colnames(a), 3),"=",colnames(a), collapse = ", ")
Abbreviations
```

# db-RDA - QMP
## Buildup model
```{r RDA_microbial load and percentage intact, eval=, echo=TRUE, fig.height=15, fig.width=13.5}
# General parameters
## Create in vivo transit variable
test <- as.data.frame(sample_data(phylobj.QMP))
# test <- subset(test, select = -c(InTransit))
df.1 <- subset(test, Donor ==1 | Donor == 2); df.1$ivTransit <- "STD"
df.2 <- subset(test, Donor ==3 | Donor == 4); df.2$ivTransit <- "MTD"
df.3 <- subset(test, Donor ==5 | Donor == 6); df.3$ivTransit <- "LTD"
test <- rbind(df.1, df.2, df.3)
test$ivTransit <- factor(test$ivTransit) 
phylobj.QMP <- merge_phyloseq(phylobj.QMP, sample_data(test))

set.seed(0604); labelsint <- c("Int"); labelstot <- c("ML") 
neon <- c("#2f89fc"); neonsize <- 7 ;labelcolour <- c("#fc0202"); linecolour <- c("#fc0202")
legendsize <- 15 ; labelsize <- 6; grobsize <- 14
xgrob <- 0.7;  ygrob <- 0.93; xgrobb <- 0.83
leg.A <- c(0.13,0.77)
leg.C <- c(0.07,0.7)
leg.E <- c(0.13,0.84)
lefttitle <- "Microbial load" ; righttitle <- "Percentage intact microbial population"

neon <- c("#2f89fc") ; labelcolour <- c("#fc0202"); linecolour <- c("#fc0202"); showR = TRUE
PDcolours <- c("#434e52", "#5b8c85" ) ; ndigits <- 2
legendsize <- 15 ; labelsize <- 6; ygrob <- 0.97; neonsize <- 7; topn <- 100
if(showR == TRUE){xgrob <- 0.85}else{xgrob <- 0.9}; abb_digits = T; dgts = 1; abbnum = 3; alphnumb = 0.62
lefttitle <- "Absolute community abundance" ; righttitle <- "Proportional community abundance"
legendsize <- 18; pointsize = 9; grobsize <- 15; abbrev <- F; plotobjects <- TRUE; plotobjects2 <- F
pos <- position_jitter(width = 0.0, seed = 2)

saveplots()
# ABSOLUTE DATA
## Make data frame of all genera
phylobj.QMP1 <- aggregate_taxa(phylobj.QMP, level = "Genus"); phylobj.QMP1 <- subset_samples(phylobj.QMP1, Ninetransit == 3)
a <- as.data.frame(vegan_otu(phylobj.QMP1))
b <- as.data.frame(sample_data(phylobj.QMP1))
df.phylobj.all <- cbind(Samplenumber = b$Samplenumber,a,Day = b$Day, 
       Transit=b$Transit, Donor = b$Donor, PD_Colon =b$PD_Colon, ivTransit = b$ivTransit)
cols2.df <- ncol(df.phylobj.all)-5
df.phylobj.all.genus <- df.phylobj.all[2:cols2.df]

## Make data frame of top 15 genera
Top15phylobj <- aggregate_top_taxa(phylobj.QMP, topn, "Genus"); Top15phylobj <- subset_samples(Top15phylobj, Ninetransit == 3)
a <- as.data.frame(vegan_otu(Top15phylobj))
b <- as.data.frame(sample_data(Top15phylobj))
df.Top15phylobj <- cbind(Samplenumber = b$Samplenumber,a,Day = b$Day, 
        Transit=b$Transit, Donor = b$Donor, PD_Colon =b$PD_Colon, ivTransit = b$ivTransit)
cols.df <- ncol(df.Top15phylobj)-5
df.Top15phylobj.otu <- df.Top15phylobj[2:cols.df]

# Define data frames
genus_table <- df.phylobj.all.genus %>%
   mutate_all(as.numeric)
sampleinfo <- subset(df.phylobj.all, select = c(Day, Donor, Transit, ivTransit, PD_Colon))


# Develop correct model
## Compute dissimilarity matrix
 rdaplot <- genus_table; df.sub <- genus_table
 # rdajaccard <- vegdist(rdaplot,method="jaccard",binary="FALSE")
 rdabray <- vegdist(rdaplot,method="bray",binary="FALSE")
## Perform PCoA
 # pcoail <- cmdscale(rdajaccard,k=(nrow(rdaplot)-1),eig=TRUE)
 pcoail <- cmdscale(rdabray,k=(nrow(rdaplot)-1),eig=TRUE)
 pcoascores <- pcoail$points
## Automatic stepwise model building for constrained ordination methods
 mod0 <- rda(pcoascores ~ 1, sampleinfo)  # Model with intercept only
 mod1 <- rda(pcoascores ~ ., sampleinfo)  # Model with all explanatory variables
 ordi <- ordiR2step(mod0, scope = formula(mod1), perm.max = 200)
 # ordistep(mod0, scope = formula(mod1), perm.max = 200)
 ordi
 ordi$anova
 
test <- ordi[["anova"]]
test <- rownames_to_column(test) %>%
  mutate_all(funs(gsub("[[:punct:]]", "", .)))
test <- test %>%
  mutate_all(funs(gsub(" ", "", .)))
test <- test[1:nrow(test) - 1,]
variables <- test$rowname; message(crayon::green(paste("Hey, your variables are:"))); variables
```

## 1) donors - Run the models
```{r db-RDA, eval=, echo=TRUE, fig.height=15, fig.width=13.5}
neon <- c("#2f89fc") ; labelcolour <- c("#fc0202");
labelcolour <- "#fc0202"; plotlabels = F
linecolour <- c("#fc0202"); showR = FALSE
PDcolours <- c("#434e52", "#5b8c85" ) ; ndigits <- 2
legendsize <- 15 ; labelsize <- 6; ygrob <- 0.97; neonsize <- 7; topn <- 15
if(showR == TRUE){xgrob <- 0.85}else{xgrob <- 0.9}; abb_digits = T; dgts = 1; abbnum = 3; alphnumb = 0.62
lefttitle <- "Absolute community abundance" ; righttitle <- "Proportional community abundance"
legendsize <- 18; pointsize = 9; grobsize <- 15; abbrev <- F; plotobjects <- TRUE; plotobjects2 <- F
pos <- position_jitter(width = 0.0, seed = 2)
saveplots()




# Go through all the selected variables in the model
 Radjs <- c(); CheckRadjs <- c(); Panova <- c()
 
# SHIME transit time
## to plot original species on graph: capscale function!!!
 dbrdail <- dbrda(df.sub~ Transit + Condition(PD_Colon) + Condition(Donor), sampleinfo, dist="bray")
 summary(dbrdail)
 coef(dbrdail)
 plot(dbrdail)
 
## permutation test of the results
 anova <- anova.cca(dbrdail,step=1000)
 Panova <- c(Panova, "Transit"= anova[1,4])
 
 # cores <- makeCluster(4) # Define amount of clusters
 # anova.cca(dbrdail,by="axis",step=1000, parallel = cores)
 # stopCluster(cores)
## explanatory power of the variables included
 Runadj <- RsquareAdj(dbrdail)$r.squared
 Radj <- RsquareAdj(dbrdail)$adj.r.squared
 Radjs <- c(Radjs, "Transit" = Radj)
 
## Perform RDA with pcoascores to check if its the same
 rdafunct <- rda(pcoascores~Transit + Condition(PD_Colon) + Condition(Donor), sampleinfo)
 # summary(rdafunct); coef(rdafunct)
 
## explanatory power of the variables included
 Runadj <- RsquareAdj(rdafunct)$r.squared
 Radj <- RsquareAdj(rdafunct)$adj.r.squared
 CheckRadjs <- c(CheckRadjs, "Transit" = Radj)
 
 
 ## As label on plot
Radj2 <- signif(Radj, digits = ndigits)
if(showR == TRUE){grob <- grobTree(textGrob(paste("R\u00B2 =",Radj2, "\nR\u00B2 top", topn, "=", Radj2), x=xgrob,  y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))
}else{
  grob <- grobTree(textGrob(paste("R\u00B2= ",Radj2), x=xgrob,  y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))
}
## labelling axes
if(abb_digits==T){
 Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,dgts)
 Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,dgts) 
 }else{
   if(round(summary(rdafunct)$cont$importance[2,1]*100,0)==0){
     Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,1)
     }else{
       Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,0)
       }
   if(round(summary(rdafunct)$cont$importance[2,2]*100,0)==0){
     Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,1)
     }else{
       Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,0)
     }
 }
## plot SCALING2
if(plotobjects2 == TRUE){
  plot(rdafunct,scaling=2,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","wa","cn"))
  plot(rdafunct,scaling=1,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","lc","cn"))
}
 
 
## ggplot
rdascores <- scores(rdafunct, choices = 1:2, scaling = 2, display = "sites")
firstaxispoints <- rdascores[, 1]; secondaxispoints <- rdascores[, 2]
points <- data.frame(fa = firstaxispoints, sa = secondaxispoints, Donor = sampleinfo$Donor, 
    P_D = sampleinfo$PD_Colon, Transit = sampleinfo$Transit, ivTransit = sampleinfo$ivTransit)
points$P_D <- mapvalues(points$P_D, from = c("P", "D"), to = c("Proximal", "Distal"))

rdascorevfa <- data.frame(scores(rdafunct, choices = 1:2, scaling = 2, display = "species"))
rdafactor <- data.frame(summary(rdafunct)$centroids)
rownames(rdafactor) <- mapvalues(rownames(rdafactor), from = c("TransitST", "TransitMT", "TransitLT"), 
                                 to = c("Short", "Medium", "Long"))
plotobj <- ggplot(as.data.frame(points)) + geom_point(aes(x = fa, y = sa, size = pointsize, color = Transit), alpha = 0.8) +
  scale_colour_manual(values = transitcolors, name = "Transit") + xlab(paste("RDA1", Dim1, "%", sep = " ")) + 
    ylab(paste("RDA2", Dim2, "%", sep = " ")) + annotation_custom(grob) + 
    geom_segment(data = rdascorevfa, aes(x = 0, xend = RDA1, y = 0, yend = RDA2), 
        color = linecolour) + geom_text(data = rdafactor, aes(x = RDA1, y = RDA2), 
                                         # fill = "grey", alpha = 0.5,
    color = neon, label = rownames(rdafactor), size = neonsize)
if(plotlabels == T){
if(abbrev == TRUE){
  plotobj <- plotobj + geom_label_repel(data = rdascorevfa,  aes(x = RDA1, y = RDA2, 
                                          label = abbreviate(colnames(a),abbnum),
                                          segment.color = "grey50"
                                          # vjust = ifelse(RDA2 >= 0, -0.1, 0.7), 
                                          # hjust = ifelse(RDA1 >= 0, 0, 0.5)
                                          ), 
                                          fill = "grey", alpha = alphnumb, label.size = NA,
                                          color = labelcolour, size = labelsize,
                                          fontface = "italic",
                                          position = pos)
} else { 
      plotobj <- plotobj + geom_label_repel(data = rdascorevfa,  aes(x = RDA1, y = RDA2, 
                                          label = colnames(a),
                                          segment.color = "grey50"
                                          # vjust = ifelse(RDA2 >= 0, -0.1, 0.7), 
                                          # hjust = ifelse(RDA1 >= 0, 0, 0.5)
                                          ), 
                                          fill = "grey", 
                                          alpha = alphnumb, 
                                          label.size = NA,
                                          # color = labelcolour, 
                                          size = labelsize,
                                          fontface = "italic",
                                          position = pos)
      }}
plotobj <- papertheme(plotobj)
plotobj <- plotobj + guides(color = guide_legend(ncol = 1), size = "none") + theme(legend.position = "right")
mylegendPDfunctnet <- g_legend(plotobj)
plotobj <- plotobj + theme(legend.position =  c(0.88,0.05), 
        legend.text = element_text(size=legendsize,family='sans',angle=0),
        legend.title = element_text(size=legendsize,family='sans',angle=0),
        legend.direction = "horizontal") 
plotobjrdatop15PDA <- plotobj + xlim(min(rdascorevfa)-0.5, max(points$fa + 0.5)) + 
  # ggtitle(lefttitle) + 
  theme(plot.title = element_text(hjust = 0.5, face="bold"))
if(plotobjects == TRUE){plotobjrdatop15PDA}


 
 
 
 
 
 
 
 
# PD_Colon
## to plot original species on graph: capscale function!!!
 dbrdail <- dbrda(df.sub~ PD_Colon + Condition(Transit) + Condition(Donor), sampleinfo, dist="bray")
 summary(dbrdail)
 coef(dbrdail)
 plot(dbrdail)
 
## permutation test of the results
 anova.cca(dbrdail,step=1000)
 anova <- anova.cca(dbrdail,step=1000)
 Panova <- c(Panova, "PD_Colon"= anova[1,4])
 
 # cores <- makeCluster(4) # Define amount of clusters
 # anova.cca(dbrdail,by="axis",step=1000, parallel = cores)
 # stopCluster(cores)
 
## explanatory power of the variables included
 Runadj <- RsquareAdj(dbrdail)$r.squared
 Radj <- RsquareAdj(dbrdail)$adj.r.squared
 Radjs <- c(Radjs, "PD_Colon" = Radj)
 
## Perform RDA with pcoascores to check if its the same
 rdafunct <- rda(pcoascores~PD_Colon + Condition(Transit) + Condition(Donor),sampleinfo)
 # summary(rdafunct); coef(rdafunct)
 
## explanatory power of the variables included
 Runadj <- RsquareAdj(rdafunct)$r.squared
 Radj <- RsquareAdj(rdafunct)$adj.r.squared
 CheckRadjs <- c(CheckRadjs, "PD_Colon" = Radj)
 
 
# Donor
## to plot original species on graph: capscale function!!!
 dbrdail <- dbrda(df.sub~ Donor + Condition(Transit) + Condition(PD_Colon), sampleinfo, dist="bray")
 summary(dbrdail)
 coef(dbrdail)
 plot(dbrdail)
 
## permutation test of the results
 anova.cca(dbrdail,step=1000)
 anova <- anova.cca(dbrdail,step=1000)
 Panova <- c(Panova, "Donor" = anova[1,4])
 # cores <- makeCluster(4) # Define amount of clusters
 # anova.cca(dbrdail,by="axis",step=1000, parallel = cores)
 # stopCluster(cores)
 
## explanatory power of the variables included
 Runadj <- RsquareAdj(dbrdail)$r.squared
 Radj <- RsquareAdj(dbrdail)$adj.r.squared
 Radjs <- c(Radjs, "Donor" = Radj)
 
## Perform RDA with pcoascores to check if its the same
 rdafunct <- rda(pcoascores~Donor + Condition(Transit) + Condition(PD_Colon),sampleinfo)
 # summary(rdafunct); coef(rdafunct)
 
## explanatory power of the variables included
 Runadj <- RsquareAdj(rdafunct)$r.squared
 Radj <- RsquareAdj(rdafunct)$adj.r.squared
 CheckRadjs <- c(CheckRadjs, "Donor" = Radj) 
 

 
 
 # ordiR2step(mod0, scope = formula(mod1), perm.max = 200)
 # ordistep(mod0, scope = formula(mod1), perm.max = 200)
 # Radjs
 # CheckRadjs
 Radjs_ABS <- Radjs; CheckRadjs_ABS <- CheckRadjs
 
 
 
# Adjust p-values because of multiple testing
 # Correction for multiple testing (BenjaminiâHochberg procedure, FDR) 
 # Panova
 # p.adjust(Panova)
 Panova_ABS <- Panova
 padj_abs <- p.adjust(Panova, method = "holm")
 padj_abs <- p.adjust(Panova)
 
  
 
 Radjs_ABS
 CheckRadjs_ABS
 CheckRadjs_ABS_donor <- CheckRadjs_ABS
 Panova_ABS
 padj_abs
```

## 2) in vivo transit - Run the models
```{r db-RDA, eval=, echo=TRUE, fig.height=15, fig.width=13.5}
# Go through all the selected variables in the model
 Radjs <- c(); CheckRadjs <- c(); Panova <- c()
 
# SHIME transit time
## to plot original species on graph: capscale function!!!
 dbrdail <- dbrda(df.sub~ Transit + Condition(PD_Colon) + Condition(ivTransit), sampleinfo, dist="bray")
 summary(dbrdail)
 coef(dbrdail)
 plot(dbrdail)
 
## permutation test of the results
 anova <- anova.cca(dbrdail,step=1000)
 Panova <- c(Panova, "Transit"= anova[1,4])
 
 # cores <- makeCluster(4) # Define amount of clusters
 # anova.cca(dbrdail,by="axis",step=1000, parallel = cores)
 # stopCluster(cores)
## explanatory power of the variables included
 Runadj <- RsquareAdj(dbrdail)$r.squared
 Radj <- RsquareAdj(dbrdail)$adj.r.squared
 Radjs <- c(Radjs, "Transit" = Radj)
 
## Perform RDA with pcoascores to check if its the same
 rdafunct <- rda(pcoascores~Transit + Condition(PD_Colon) + Condition(ivTransit), sampleinfo)
 # summary(rdafunct); coef(rdafunct)
 
## explanatory power of the variables included
 Runadj <- RsquareAdj(rdafunct)$r.squared
 Radj <- RsquareAdj(rdafunct)$adj.r.squared
 CheckRadjs <- c(CheckRadjs, "Transit" = Radj)
 
 
# PD_Colon
## to plot original species on graph: capscale function!!!
 dbrdail <- dbrda(df.sub~ PD_Colon + Condition(Transit) + Condition(ivTransit), sampleinfo, dist="bray")
 summary(dbrdail)
 coef(dbrdail)
 plot(dbrdail)
 
## permutation test of the results
 anova.cca(dbrdail,step=1000)
 anova <- anova.cca(dbrdail,step=1000)
 Panova <- c(Panova, "PD_Colon"= anova[1,4])
 
 # cores <- makeCluster(4) # Define amount of clusters
 # anova.cca(dbrdail,by="axis",step=1000, parallel = cores)
 # stopCluster(cores)
 
## explanatory power of the variables included
 Runadj <- RsquareAdj(dbrdail)$r.squared
 Radj <- RsquareAdj(dbrdail)$adj.r.squared
 Radjs <- c(Radjs, "PD_Colon" = Radj)
 
## Perform RDA with pcoascores to check if its the same
 rdafunct <- rda(pcoascores~PD_Colon + Condition(Transit) + Condition(ivTransit),sampleinfo)
 # summary(rdafunct); coef(rdafunct)
 
## explanatory power of the variables included
 Runadj <- RsquareAdj(rdafunct)$r.squared
 Radj <- RsquareAdj(rdafunct)$adj.r.squared
 CheckRadjs <- c(CheckRadjs, "PD_Colon" = Radj)
 
 
# ivTransit
## to plot original species on graph: capscale function!!!
 dbrdail <- dbrda(df.sub~ ivTransit + Condition(Transit) + Condition(PD_Colon), sampleinfo, dist="bray")
 summary(dbrdail)
 coef(dbrdail)
 plot(dbrdail)
 
## permutation test of the results
 anova.cca(dbrdail,step=1000)
 anova <- anova.cca(dbrdail,step=1000)
 Panova <- c(Panova, "ivTransit" = anova[1,4])
 # cores <- makeCluster(4) # Define amount of clusters
 # anova.cca(dbrdail,by="axis",step=1000, parallel = cores)
 # stopCluster(cores)
 
## explanatory power of the variables included
 Runadj <- RsquareAdj(dbrdail)$r.squared
 Radj <- RsquareAdj(dbrdail)$adj.r.squared
 Radjs <- c(Radjs, "ivTransit" = Radj)
 
## Perform RDA with pcoascores to check if its the same
 rdafunct <- rda(pcoascores~ivTransit + Condition(Transit) + Condition(PD_Colon),sampleinfo)
 # summary(rdafunct); coef(rdafunct)
 
## explanatory power of the variables included
 Runadj <- RsquareAdj(rdafunct)$r.squared
 Radj <- RsquareAdj(rdafunct)$adj.r.squared
 CheckRadjs <- c(CheckRadjs, "ivTransit" = Radj) 
 
 

 # ordiR2step(mod0, scope = formula(mod1), perm.max = 200)
 # ordistep(mod0, scope = formula(mod1), perm.max = 200)
 # Radjs
 # CheckRadjs
 Radjs_ABS <- Radjs; CheckRadjs_ABS <- CheckRadjs
 
 
 
# Adjust p-values because of multiple testing
 # Correction for multiple testing (BenjaminiâHochberg procedure, FDR) 
 # Panova
 # p.adjust(Panova)
 Panova_ABS <- Panova
 padj_abs <- p.adjust(Panova, method = "holm")
 padj_abs <- p.adjust(Panova)
 
  
 
 Radjs_ABS
 CheckRadjs_ABS
 CheckRadjs_ABS_donor
 Panova_ABS
 padj_abs
```

### Plot them
```{r fig.height=15, fig.width=13.5}
neon <- c("#2f89fc") ; labelcolour <- c("#fc0202");
labelcolour <- "#fc0202"; plotlabels = F
linecolour <- c("#fc0202"); showR = FALSE
PDcolours <- c("#434e52", "#5b8c85" ) ; ndigits <- 2
legendsize <- 15 ; labelsize <- 6; ygrob <- 0.9; neonsize <- 7; topn <- 15
if(showR == TRUE){xgrob <- 0.85}else{xgrob <- 0.8}; abb_digits = T; dgts = 1; abbnum = 3; alphnumb = 0.62+0.1
lefttitle <- "Absolute community abundance" ; righttitle <- "Proportional community abundance"
legendsize <- 18; pointsize = 9; grobsize <- 15; abbrev <- F; plotobjects <- TRUE; plotobjects2 <- F
pos <- position_jitter(width = 0.0, seed = 2)
saveplots()

xgrobb = 0.65


# Go through all the selected variables in the model
 Radjs <- c(); CheckRadjs <- c(); Panova <- c()
 
 
# SHIME transit time
## to plot original species on graph: capscale function!!!
 dbrdail <- dbrda(df.sub~ Transit + Condition(PD_Colon) + Condition(ivTransit), sampleinfo, dist="bray")
 summary(dbrdail)
 coef(dbrdail)
 plot(dbrdail)
 
## permutation test of the results
 anova <- anova.cca(dbrdail,step=1000)
 Panova <- c(Panova, "Transit"= anova[1,4])
 
 # cores <- makeCluster(4) # Define amount of clusters
 # anova.cca(dbrdail,by="axis",step=1000, parallel = cores)
 # stopCluster(cores)
## explanatory power of the variables included
 Runadj <- RsquareAdj(dbrdail)$r.squared
 Radj <- RsquareAdj(dbrdail)$adj.r.squared
 Radjs <- c(Radjs, "Transit" = Radj)
 
## Perform RDA with pcoascores to check if its the same
 rdafunctR <- rda(pcoascores~Transit + Condition(PD_Colon) + Condition(ivTransit), sampleinfo)
 rdafunct <- capscale(df.sub~ Transit + Condition(PD_Colon) + Condition(ivTransit), sampleinfo, dist="bray")
 # summary(rdafunct); coef(rdafunct)
 
## explanatory power of the variables included
 Runadj <- RsquareAdj(rdafunctR)$r.squared
 Radj <- RsquareAdj(rdafunctR)$adj.r.squared
 CheckRadjs <- c(CheckRadjs, "Transit" = Radj)
 
 
 ## As label on plot
Radj2 <- signif(Radj, digits = ndigits)
  grob <- grobTree(textGrob(paste("R\u00B2= ",Radj2), x=xgrob,  y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))

## labelling axes
if(abb_digits==T){
 Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,dgts)
 Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,dgts) 
 }else{
   if(round(summary(rdafunct)$cont$importance[2,1]*100,0)==0){
     Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,1)
     }else{
       Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,0)
       }
   if(round(summary(rdafunct)$cont$importance[2,2]*100,0)==0){
     Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,1)
     }else{
       Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,0)
     }
 }
## plot SCALING2
if(plotobjects2 == TRUE){
  plot(rdafunct,scaling=2,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","wa","cn"))
  plot(rdafunct,scaling=1,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","lc","cn"))
}
 
 
## ggplot
rdascores <- scores(rdafunct, choices = 1:2, scaling = 2, display = "sites")
firstaxispoints <- rdascores[, 1]; secondaxispoints <- rdascores[, 2]
points <- data.frame(fa = firstaxispoints, sa = secondaxispoints, Donor = sampleinfo$Donor, 
    P_D = sampleinfo$PD_Colon, Transit = sampleinfo$Transit, ivTransit = sampleinfo$ivTransit)
points$P_D <- mapvalues(points$P_D, from = c("P", "D"), to = c("Proximal", "Distal"))

rdascorevfa <- data.frame(scores(rdafunct, choices = 1:2, scaling = 2, display = "species"))
rdafactor <- data.frame(summary(rdafunct)$centroids)
rownames(rdafactor) <- mapvalues(rownames(rdafactor), from = c("TransitST", "TransitMT", "TransitLT"), 
                                 to = c("Short", "Medium", "Long"))
plotobj <- ggplot(as.data.frame(points)) + geom_point(aes(x = fa, y = sa, size = pointsize, color = Transit), alpha = 0.8) +
  scale_colour_manual(values = transitcolors, name = "Transit") + xlab(paste("RDA1", Dim1, "%", sep = " ")) + 
    ylab(paste("RDA2", Dim2, "%", sep = " ")) + annotation_custom(grob) + 
    geom_segment(data = rdascorevfa, aes(x = 0, xend = CAP1, y = 0, yend = CAP2), 
        color = linecolour) + geom_text(data = rdafactor, aes(x = CAP1, y = CAP2), 
                                         # fill = "grey", alpha = 0.5,
    color = neon, label = rownames(rdafactor), size = neonsize)
if(plotlabels == T){
if(abbrev == TRUE){
  plotobj <- plotobj + geom_label_repel(data = rdascorevfa,  aes(x = CAP1, y = CAP2, 
                                          label = abbreviate(rownames(rdascorevfa),abbnum),
                                          segment.color = "grey50"
                                          # vjust = ifelse(RDA2 >= 0, -0.1, 0.7), 
                                          # hjust = ifelse(RDA1 >= 0, 0, 0.5)
                                          ), 
                                          fill = "grey", alpha = alphnumb, label.size = NA,
                                          color = labelcolour, size = labelsize,
                                          fontface = "italic",
                                          position = pos)
} else { 
      plotobj <- plotobj + geom_label_repel(data = rdascorevfa,  aes(x = CAP1, y = CAP2, 
                                          label = colnames(a),
                                          segment.color = "grey50"
                                          # vjust = ifelse(RDA2 >= 0, -0.1, 0.7), 
                                          # hjust = ifelse(RDA1 >= 0, 0, 0.5)
                                          ), 
                                          fill = "grey", 
                                          alpha = alphnumb, 
                                          label.size = NA,
                                          # color = labelcolour, 
                                          size = labelsize,
                                          fontface = "italic",
                                          position = pos)
      }}
plotobj <- papertheme(plotobj)
plotobj <- plotobj + guides(color = guide_legend(ncol = 1), size = "none") + theme(legend.position = "right")
mylegendPDfunctnet <- g_legend(plotobj)
plotobj <- plotobj + theme(legend.position = "none"           , 
        legend.text = element_text(size=legendsize,family='sans',angle=0),
        legend.title = element_text(size=legendsize,family='sans',angle=0),
        legend.direction = "horizontal") 
plotobjrdatop15PDA <- plotobj + xlim(min(rdascorevfa)-0.5, max(points$fa + 0.5)) + 
  # ggtitle(lefttitle) + 
  theme(plot.title = element_text(hjust = 0.5, face="bold"))
if(plotobjects == TRUE){plotobjrdatop15PDA}


 
 
 
 
 
 
 
 
# PD_Colon
## to plot original species on graph: capscale function!!!
 dbrdail <- dbrda(df.sub~ PD_Colon + Condition(Transit) + Condition(ivTransit), sampleinfo, dist="bray")
 summary(dbrdail)
 coef(dbrdail)
 plot(dbrdail)
 
## permutation test of the results
 anova.cca(dbrdail,step=1000)
 anova <- anova.cca(dbrdail,step=1000)
 Panova <- c(Panova, "PD_Colon"= anova[1,4])
 
 # cores <- makeCluster(4) # Define amount of clusters
 # anova.cca(dbrdail,by="axis",step=1000, parallel = cores)
 # stopCluster(cores)
 
## explanatory power of the variables included
 Runadj <- RsquareAdj(dbrdail)$r.squared
 Radj <- RsquareAdj(dbrdail)$adj.r.squared
 Radjs <- c(Radjs, "PD_Colon" = Radj)
 
## Perform RDA with pcoascores to check if its the same
 rdafunctR <- rda(pcoascores~PD_Colon + Condition(Transit) + Condition(ivTransit),sampleinfo)
rdafunct <- capscale(df.sub~ PD_Colon + Condition(Transit) + Condition(ivTransit), sampleinfo, dist="bray")
 # summary(rdafunct); coef(rdafunct)
 
## explanatory power of the variables included
 Runadj <- RsquareAdj(rdafunctR)$r.squared
 Radj <- RsquareAdj(rdafunctR)$adj.r.squared
 CheckRadjs <- c(CheckRadjs, "PD_Colon" = Radj)
 
 
  ## As label on plot
Radj2 <- signif(Radj, digits = ndigits)
if(showR == TRUE){grob <- grobTree(textGrob(paste("R\u00B2 =",Radj2, "\nR\u00B2 top", topn, "=", Radj2), x=xgrobb,  y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))
}else{
  grob <- grobTree(textGrob(paste("R\u00B2= ",Radj2), x=xgrob,  y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))
}
## labelling axes
if(abb_digits==T){
 Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,dgts)
 Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,dgts) 
 }else{
   if(round(summary(rdafunct)$cont$importance[2,1]*100,0)==0){
     Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,1)
     }else{
       Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,0)
       }
   if(round(summary(rdafunct)$cont$importance[2,2]*100,0)==0){
     Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,1)
     }else{
       Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,0)
     }
 }
## plot SCALING2
if(plotobjects2 == TRUE){
  plot(rdafunct,scaling=2,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","wa","cn"))
  plot(rdafunct,scaling=1,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","lc","cn"))
}
 
 
## ggplot
rdascores <- scores(rdafunct, choices = 1:2, scaling = 2, display = "sites")
firstaxispoints <- rdascores[, 1]; secondaxispoints <- rdascores[, 2]
points <- data.frame(fa = firstaxispoints, sa = secondaxispoints, Donor = sampleinfo$Donor, 
    P_D = sampleinfo$PD_Colon, Transit = sampleinfo$Transit, ivTransit = sampleinfo$ivTransit)
points$P_D <- mapvalues(points$P_D, from = c("P", "D"), to = c("Proximal", "Distal"))

rdascorevfa <- data.frame(scores(rdafunct, choices = 1:2, scaling = 2, display = "species"))
rdafactor <- data.frame(summary(rdafunct)$centroids)
rownames(rdafactor) <- mapvalues(rownames(rdafactor), from = c("PD_ColonD", "PD_ColonP"), 
                                 to = c("Distal", "Proximal"))
plotobj <- ggplot(as.data.frame(points)) + geom_point(aes(x = fa, y = sa, size = pointsize, color = P_D), alpha = 0.8) +
  scale_colour_manual(values = c("Proximal" = "#cdb174", "Distal" = "#a48d5c"), name = "Vessel") +
 xlab(paste("RDA1", Dim1, "%", sep = " ")) + 
    ylab(paste("PC1", Dim2, "%", sep = " ")) + annotation_custom(grob) + 
    geom_segment(data = rdascorevfa, aes(x = 0, xend = RDA1, y = 0, yend =PC1), 
        color = linecolour) + geom_text(data = rdafactor, aes(x = RDA1, y = PC1), 
                                         # fill = "grey", alpha = 0.5,
    color = neon, label = rownames(rdafactor), size = neonsize)
plotobj <- papertheme(plotobj)
plotobj <- plotobj + guides(color = guide_legend(ncol = 1), size = "none") + theme(legend.position = "right")
mylegendPDfunctnet <- g_legend(plotobj)
plotobj <- plotobj + theme(legend.position =  "none", 
        legend.text = element_text(size=legendsize,family='sans',angle=0),
        legend.title = element_text(size=legendsize,family='sans',angle=0),
        legend.direction = "horizontal") 
plotobjrdatop15PDB <- plotobj + xlim(min(rdascorevfa)-0.5, max(points$fa + 0.5)) + 
  # ggtitle(lefttitle) + 
  theme(plot.title = element_text(hjust = 0.5, face="bold"))
if(plotobjects == TRUE){plotobjrdatop15PDB}
 
 
# ivTransit
## to plot original species on graph: capscale function!!!
 dbrdail <- dbrda(df.sub~ ivTransit + Condition(Transit) + Condition(PD_Colon), sampleinfo, dist="bray")
 summary(dbrdail)
 coef(dbrdail)
 plot(dbrdail)
 
## permutation test of the results
 anova.cca(dbrdail,step=1000)
 anova <- anova.cca(dbrdail,step=1000)
 Panova <- c(Panova, "Donor" = anova[1,4])
 # cores <- makeCluster(4) # Define amount of clusters
 # anova.cca(dbrdail,by="axis",step=1000, parallel = cores)
 # stopCluster(cores)
 
## explanatory power of the variables included
 Runadj <- RsquareAdj(dbrdail)$r.squared
 Radj <- RsquareAdj(dbrdail)$adj.r.squared
 Radjs <- c(Radjs, "Donor" = Radj)
 
## Perform RDA with pcoascores to check if its the same
 rdafunctR <- rda(pcoascores~ivTransit + Condition(Transit) + Condition(PD_Colon),sampleinfo)
rdafunct <- capscale(df.sub~ ivTransit + Condition(PD_Colon) + Condition(Transit), sampleinfo, dist="bray")
 # summary(rdafunct); coef(rdafunct)
 
## explanatory power of the variables included
 Runadj <- RsquareAdj(rdafunctR)$r.squared
 Radj <- RsquareAdj(rdafunctR)$adj.r.squared
 CheckRadjs <- c(CheckRadjs, "ivTransit" = Radj) 
 
xgrobb = 0.65
 
   ## As label on plot
Radj2 <- signif(Radj, digits = ndigits)
if(showR == TRUE){grob <- grobTree(textGrob(paste("R\u00B2 =",Radj2, "\nR\u00B2 top", topn, "=", Radj2), x=xgrob,  y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))
}else{
  grob <- grobTree(textGrob(paste("Donor R\u00B2 =", signif(CheckRadjs_ABS_donor[[3]], digits = ndigits), 
                                  "\nIn vivo transit R\u00B2= ",Radj2), x=xgrobb, 
                            y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))
}
xgrobb
## labelling axes
if(abb_digits==T){
 Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,dgts)
 Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,dgts) 
 }else{
   if(round(summary(rdafunct)$cont$importance[2,1]*100,0)==0){
     Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,1)
     }else{
       Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,0)
       }
   if(round(summary(rdafunct)$cont$importance[2,2]*100,0)==0){
     Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,1)
     }else{
       Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,0)
     }
 }
## plot SCALING2
if(plotobjects2 == TRUE){
  plot(rdafunct,scaling=2,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","wa","cn"))
  plot(rdafunct,scaling=1,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","lc","cn"))
}
 
 
## ggplot
rdascores <- scores(rdafunct, choices = 1:2, scaling = 2, display = "sites")
firstaxispoints <- rdascores[, 1]; secondaxispoints <- rdascores[, 2]
points <- data.frame(fa = firstaxispoints, sa = secondaxispoints, Donor = sampleinfo$Donor, 
    P_D = sampleinfo$PD_Colon, Transit = sampleinfo$Transit, ivTransit = sampleinfo$ivTransit)
points$P_D <- mapvalues(points$P_D, from = c("P", "D"), to = c("Proximal", "Distal"))

rdascorevfa <- data.frame(scores(rdafunct, choices = 1:2, scaling = 2, display = "species"))
rdafactor <- data.frame(summary(rdafunct)$centroids)
rownames(rdafactor) <- mapvalues(rownames(rdafactor), from = c("ivTransitLTD", "ivTransitMTD", "ivTransitSTD"), 
                                 to = c("Long", "Medium", "Short"))
plotobj <- ggplot(as.data.frame(points)) + geom_point(aes(x = fa, y = sa, size = pointsize, color = Donor, shape = ivTransit), alpha = 0.8) +
  scale_colour_manual(values = sixdonors, name = "Donor")+ 
  scale_shape_manual(name = expression(paste(italic("In vivo"), " transit")), 
              values = c("STD" = 16, "MTD" = 17, "LTD" = 15), labels = c("Short", "Medium", "Long"))+
  xlab(paste("RDA1", Dim1, "%", sep = " ")) + 
    ylab(paste("RDA2", Dim2, "%", sep = " ")) + annotation_custom(grob) + 
    geom_segment(data = rdascorevfa, aes(x = 0, xend = RDA1, y = 0, yend =RDA2), 
        color = linecolour) + geom_text(data = rdafactor, aes(x = RDA1, y = RDA2), 
                                         # fill = "grey", alpha = 0.5,
    color = neon, label = rownames(rdafactor), size = neonsize)
plotobj <- papertheme(plotobj)
plotobj <- plotobj + guides(color = guide_legend(ncol = 1), size = "none") + theme(legend.position = "right")
mylegendPDfunctnet <- g_legend(plotobj)
plotobj <- plotobj + theme(legend.position =  "none", 
        legend.text = element_text(size=legendsize,family='sans',angle=0),
        legend.title = element_text(size=legendsize,family='sans',angle=0),
        legend.direction = "horizontal") 

plotobjrdatop15PDC <- plotobj + xlim(min(rdascorevfa)-0.5, max(points$fa + 0.5)) + 
  # ggtitle(lefttitle) + 
  theme(plot.title = element_text(hjust = 0.5, face="bold"))
if(plotobjects == TRUE){plotobjrdatop15PDC}
 

ggarrange(plotobjrdatop15PDA, plotobjrdatop15PDB, plotobjrdatop15PDC, nrow = 3, ncol = 1)
 
 
 
 
# Adjust p-values because of multiple testing
 # Correction for multiple testing (BenjaminiâHochberg procedure, FDR) 
 # Panova
 # p.adjust(Panova)
 Panova_ABS <- Panova
 padj_abs <- p.adjust(Panova, method = "holm")
 padj_abs <- p.adjust(Panova)
 

```


### Paperplot plot them
```{r fig.height=17, fig.width=15}
neon <- c("#2f89fc") ; labelcolour <- c("#fc0202");
labelcolour <- "#fc0202"; plotlabels = T
linecolour <- c("#fc0202"); showR = FALSE
PDcolours <- c("#434e52", "#5b8c85" ) ; ndigits <- 2
legendsize <- 15 ; labelsize <- 6; ygrob <- 0.91; neonsize <- 7; topn <- 15
if(showR == TRUE){xgrob <- 0.85}else{xgrob <- 0.8}; abb_digits = T; dgts = 1; abbnum = 3; alphnumb = 0.62+0.1
lefttitle <- "Absolute community abundance" ; righttitle <- "Proportional community abundance"
legendsize <- 18; pointsize = 9; grobsize <- 15; abbrev <- F; plotobjects <- TRUE; plotobjects2 <- F
pos <- position_jitter(width = 0.0, seed = 2)
saveplots()

xgrobb = 0.65


# Go through all the selected variables in the model
 Radjs <- c(); CheckRadjs <- c(); Panova <- c()
 
 
 # Define data frames for top10
 ## Make data frame of top 10 genera
Top10phylobj <- aggregate_top_taxa(phylobj.QMP, 10, "Genus"); Top10phylobj <- subset_samples(Top10phylobj, Ninetransit == 3)
a <- as.data.frame(vegan_otu(Top10phylobj))
b <- as.data.frame(sample_data(Top10phylobj))
df.Top10phylobj <- cbind(Samplenumber = b$Samplenumber,a, Day = b$Day, 
        Transit=b$Transit, Donor = b$Donor, PD_Colon =b$PD_Colon, ivTransit = b$ivTransit)
cols.df <- ncol(df.Top10phylobj)-5
df.Top10phylobj.genus <- df.Top10phylobj[2:cols.df]
genus_table <- df.Top10phylobj.genus %>%
   mutate_all(as.numeric)

sampleinfo10 <- subset(df.Top10phylobj, select = c(Day, Donor, Transit, ivTransit, PD_Colon))


# Develop correct model
## Compute dissimilarity matrix
 rdaplot10 <- genus_table; df.sub10 <- subset(genus_table, select = c(-Other))
 
 
# SHIME transit time
## to plot original species on graph: capscale function!!!
 dbrdail <- dbrda(df.sub~ Transit + Condition(PD_Colon) + Condition(ivTransit), sampleinfo, dist="bray")
 summary(dbrdail)
 coef(dbrdail)
 plot(dbrdail)
 
## permutation test of the results
 anova <- anova.cca(dbrdail,step=1000)
 Panova <- c(Panova, "Transit"= anova[1,4])
 
 # cores <- makeCluster(4) # Define amount of clusters
 # anova.cca(dbrdail,by="axis",step=1000, parallel = cores)
 # stopCluster(cores)
## explanatory power of the variables included
 Runadj <- RsquareAdj(dbrdail)$r.squared
 Radj <- RsquareAdj(dbrdail)$adj.r.squared
 Radjs <- c(Radjs, "Transit" = Radj)
 
## Perform RDA with pcoascores to check if its the same
 rdafunctR <- rda(pcoascores~Transit + Condition(PD_Colon) + Condition(ivTransit), sampleinfo)
 rdafunct <- capscale(df.sub~ Transit + Condition(PD_Colon) + Condition(ivTransit), sampleinfo, dist="bray")
 # summary(rdafunct); coef(rdafunct)
 
## explanatory power of the variables included
 Runadj <- RsquareAdj(rdafunctR)$r.squared
 Radj <- RsquareAdj(rdafunctR)$adj.r.squared
 CheckRadjs <- c(CheckRadjs, "Transit" = Radj)
 
 
 ## As label on plot
Radj2 <- signif(Radj, digits = ndigits)
  grob <- grobTree(textGrob(paste("R\u00B2= ",Radj2), x=xgrob,  y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))

## labelling axes
if(abb_digits==T){
 Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,dgts)
 Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,dgts) 
 }else{
   if(round(summary(rdafunct)$cont$importance[2,1]*100,0)==0){
     Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,1)
     }else{
       Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,0)
       }
   if(round(summary(rdafunct)$cont$importance[2,2]*100,0)==0){
     Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,1)
     }else{
       Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,0)
     }
 }
## plot SCALING2
if(plotobjects2 == TRUE){
  plot(rdafunct,scaling=2,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","wa","cn"))
  plot(rdafunct,scaling=1,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","lc","cn"))
}
 rdafunct10 <- capscale(df.sub10~ Transit + Condition(PD_Colon) + Condition(ivTransit), sampleinfo10, dist="bray")
 
## ggplot
rdascores <- scores(rdafunct10, choices = 1:2, scaling = 2, display = "sites")
firstaxispoints <- rdascores[, 1]; secondaxispoints <- rdascores[, 2]
points <- data.frame(fa = firstaxispoints, sa = secondaxispoints, Donor = sampleinfo10$Donor, 
    P_D = sampleinfo10$PD_Colon, Transit = sampleinfo10$Transit, ivTransit = sampleinfo10$ivTransit)
points$P_D <- mapvalues(points$P_D, from = c("P", "D"), to = c("Proximal", "Distal"))

rdascorevfa <- data.frame(scores(rdafunct10, choices = 1:2, scaling = 2, display = "species"))
rdafactor <- data.frame(summary(rdafunct10)$centroids)
rownames(rdafactor) <- mapvalues(rownames(rdafactor), from = c("TransitST", "TransitMT", "TransitLT"), 
                                 to = c("Short", "Medium", "Long"))
plotobj <- ggplot(as.data.frame(points)) + geom_point(aes(x = fa, y = sa, size = pointsize, color = Transit), alpha = 0.8) +
  scale_colour_manual(values = transitcolors, name = "SHIME Transit", labels = c("Short", "Medium", "Long")) + 
  xlab(paste("RDA1", Dim1, "%", sep = " ")) + 
    ylab(paste("RDA2", Dim2, "%", sep = " ")) + annotation_custom(grob) + 
    geom_segment(data = rdascorevfa, aes(x = 0, xend = CAP1, y = 0, yend = CAP2), 
        color = linecolour) + 
      ggrepel::geom_label_repel(data = rdafactor, aes(x = CAP1, y = CAP2),
                              segment.color = "grey50", alpha = alphnumb+.15, 
                                         # fill = "grey", alpha = 0.5,
    color = neon, label = rownames(rdafactor), size = neonsize)
if(plotlabels == T){
if(abbrev == TRUE){
  plotobj <- plotobj + geom_label_repel(data = rdascorevfa,  aes(x = CAP1, y = CAP2, 
                                          label = abbreviate(colnames(subset(a, select = c(-Other))),abbnum),
                                          segment.color = "grey50"
                                          # vjust = ifelse(RDA2 >= 0, -0.1, 0.7), 
                                          # hjust = ifelse(RDA1 >= 0, 0, 0.5)
                                          ), 
                                          color = linecolour, alpha = alphnumb, label.size = NA,
                                          size = labelsize, fontface = "italic",
                                          position = pos)
} else { 
      plotobj <- plotobj + geom_label_repel(data = rdascorevfa,  aes(x = CAP1, y = CAP2, 
                                          label = colnames(subset(a, select = c(-Other))),
                                          segment.color = "grey50"
                                          # vjust = ifelse(RDA2 >= 0, -0.1, 0.7), 
                                          # hjust = ifelse(RDA1 >= 0, 0, 0.5)
                                          ), 
                                          color = linecolour, alpha = alphnumb, label.size = NA,
                                          size = labelsize, fontface = "italic",
                                          position = pos)
      }}
plotobj <- papertheme(plotobj)
plotobj <- plotobj + guides(color = guide_legend(ncol = 1, direction = "horizontal",
      title.position = "top", override.aes = list(size = 5)), size = "none") + theme(legend.position = "right")
mylegendPDfunctnet <- g_legend(plotobj)
plotobj <- plotobj + theme(legend.position = "none"           , 
        legend.text = element_text(size=legendsize,family='sans',angle=0),
        legend.title = element_text(size=legendsize,family='sans',angle=0),
        legend.direction = "horizontal") 
plotobjrdatop15PDA <- plotobj + xlim(min(rdascorevfa)-0.5, max(points$fa + 0.5)) + 
  # ggtitle(lefttitle) + 
  theme(plot.title = element_text(hjust = 0.5, face="bold"))
if(plotobjects == TRUE){plotobjrdatop15PDA}


 
 
 


# Donor
## to plot original species on graph: capscale function!!!
 dbrdail <- dbrda(df.sub~ Donor+ Condition(PD_Colon) + Condition(Transit), sampleinfo, dist="bray")
 summary(dbrdail)
 coef(dbrdail)
 plot(dbrdail)
 
## permutation test of the results
 anova <- anova.cca(dbrdail,step=1000)
 Panova <- c(Panova, "Donor"= anova[1,4])
 
 # cores <- makeCluster(4) # Define amount of clusters
 # anova.cca(dbrdail,by="axis",step=1000, parallel = cores)
 # stopCluster(cores)
## explanatory power of the variables included
 Runadj <- RsquareAdj(dbrdail)$r.squared
 Radj <- RsquareAdj(dbrdail)$adj.r.squared
 Radjs <- c(Radjs, "Donor" = Radj)
 
## Perform RDA with pcoascores to check if its the same
 rdafunctR <- rda(pcoascores~Donor + Condition(PD_Colon) + Condition(Transit), sampleinfo)
 rdafunct <- capscale(df.sub~ Donor + Condition(PD_Colon) + Condition(Transit), sampleinfo, dist="bray")
 # summary(rdafunct); coef(rdafunct)
 
## explanatory power of the variables included
 Runadj <- RsquareAdj(rdafunctR)$r.squared
 Radj <- RsquareAdj(rdafunctR)$adj.r.squared
 CheckRadjs <- c(CheckRadjs, "Donor" = Radj)
 
 
 ## As label on plot
Radj2 <- signif(Radj, digits = ndigits)
  grob <- grobTree(textGrob(paste("R\u00B2= ",Radj2), x=xgrob,  y=ygrob+0.05, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))

## labelling axes
if(abb_digits==T){
 Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,dgts)
 Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,dgts) 
 }else{
   if(round(summary(rdafunct)$cont$importance[2,1]*100,0)==0){
     Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,1)
     }else{
       Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,0)
       }
   if(round(summary(rdafunct)$cont$importance[2,2]*100,0)==0){
     Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,1)
     }else{
       Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,0)
     }
 }
## plot SCALING2
if(plotobjects2 == TRUE){
  plot(rdafunct,scaling=2,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","wa","cn"))
  plot(rdafunct,scaling=1,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","lc","cn"))
}
 rdafunct10 <- capscale(df.sub10~ Donor + Condition(PD_Colon) + Condition(Transit), sampleinfo10, dist="bray")
 
## ggplot
rdascores <- scores(rdafunct10, choices = 1:2, scaling = 2, display = "sites")
firstaxispoints <- rdascores[, 1]; secondaxispoints <- rdascores[, 2]
points <- data.frame(fa = firstaxispoints, sa = secondaxispoints, Donor = sampleinfo10$Donor, 
    P_D = sampleinfo10$PD_Colon, Transit = sampleinfo10$Transit, ivTransit = sampleinfo10$ivTransit)
points$P_D <- mapvalues(points$P_D, from = c("P", "D"), to = c("Proximal", "Distal"))

rdascorevfa <- data.frame(scores(rdafunct10, choices = 1:2, scaling = 2, display = "species"))
rdafactor <- data.frame(summary(rdafunct10)$centroids)
rownames(rdafactor) <- mapvalues(rownames(rdafactor), from = c("Donor1","Donor2","Donor3",
                                                               "Donor4","Donor5","Donor6"), 
                                 to = rep(1:6))
plotobj <- ggplot(as.data.frame(points)) + geom_point(aes(x = fa, y = sa, size = pointsize, color = Donor), alpha = 0.8) +
  scale_colour_manual(values = sixdonors, name = "Donor") + xlab(paste("RDA1", Dim1, "%", sep = " ")) + 
    ylab(paste("RDA2", Dim2, "%", sep = " ")) + annotation_custom(grob) + 
    geom_segment(data = rdascorevfa, aes(x = 0, xend = CAP1, y = 0, yend = CAP2), 
        color = linecolour) + 
      ggrepel::geom_label_repel(data = rdafactor, aes(x = CAP1, y = CAP2),
                              segment.color = "grey50", alpha = alphnumb+.15, 
                                         # fill = "grey", alpha = 0.5,
    color = neon, label = rownames(rdafactor), size = neonsize)
if(plotlabels == T){
if(abbrev == TRUE){
  plotobj <- plotobj + geom_label_repel(data = rdascorevfa,  aes(x = CAP1, y = CAP2, 
                                          label = abbreviate(colnames(subset(a, select = c(-Other))),abbnum),
                                          segment.color = "grey50"
                                          # vjust = ifelse(RDA2 >= 0, -0.1, 0.7), 
                                          # hjust = ifelse(RDA1 >= 0, 0, 0.5)
                                          ), 
                                          color = linecolour, alpha = alphnumb, label.size = NA,
                                          size = labelsize, fontface = "italic",
                                          position = pos)
} else { 
      plotobj <- plotobj + geom_label_repel(data = rdascorevfa,  aes(x = CAP1, y = CAP2, 
                                          label = colnames(subset(a, select = c(-Other))),
                                          segment.color = "grey50"
                                          # vjust = ifelse(RDA2 >= 0, -0.1, 0.7), 
                                          # hjust = ifelse(RDA1 >= 0, 0, 0.5)
                                          ), 
                                          color = linecolour, alpha = alphnumb, label.size = NA,
                                          size = labelsize, fontface = "italic",
                                          position = pos)
      }}
plotobj <- papertheme(plotobj)
plotobj <- plotobj + guides(color = guide_legend(ncol = 2, direction = "horizontal",
      title.position = "top", override.aes = list(size = 5)), size = "none") + theme(legend.position = "right")
mylegendPDfunctnet <- g_legend(plotobj)
plotobj <- plotobj + theme(legend.position = "none"           , 
        legend.text = element_text(size=legendsize,family='sans',angle=0),
        legend.title = element_text(size=legendsize,family='sans',angle=0),
        legend.direction = "horizontal") 
plotobjrdatop15PDB <- plotobj + xlim(min(rdascorevfa)-0.5, max(points$fa + 0.5)) + 
  # ggtitle(lefttitle) + 
  theme(plot.title = element_text(hjust = 0.5, face="bold"))
if(plotobjects == TRUE){plotobjrdatop15PDB}




# iv transit time
## to plot original species on graph: capscale function!!!
 dbrdail <- dbrda(df.sub~ ivTransit + Condition(PD_Colon) + Condition(Transit), sampleinfo, dist="bray")
 summary(dbrdail)
 coef(dbrdail)
 plot(dbrdail)
 
## permutation test of the results
 anova <- anova.cca(dbrdail,step=1000)
 Panova <- c(Panova, "ivTransit"= anova[1,4])
 
 # cores <- makeCluster(4) # Define amount of clusters
 # anova.cca(dbrdail,by="axis",step=1000, parallel = cores)
 # stopCluster(cores)
## explanatory power of the variables included
 Runadj <- RsquareAdj(dbrdail)$r.squared
 Radj <- RsquareAdj(dbrdail)$adj.r.squared
 Radjs <- c(Radjs, "ivTransit" = Radj)
 
## Perform RDA with pcoascores to check if its the same
 rdafunctR <- rda(pcoascores~ivTransit + Condition(PD_Colon) + Condition(Transit), sampleinfo)
 rdafunct <- capscale(df.sub~ ivTransit + Condition(PD_Colon) + Condition(Transit), sampleinfo, dist="bray")
 # summary(rdafunct); coef(rdafunct)
 
## explanatory power of the variables included
 Runadj <- RsquareAdj(rdafunctR)$r.squared
 Radj <- RsquareAdj(rdafunctR)$adj.r.squared
 CheckRadjs <- c(CheckRadjs, "ivTransit" = Radj)
 
 
 ## As label on plot
Radj2 <- signif(Radj, digits = ndigits)
  grob <- grobTree(textGrob(paste("R\u00B2= ",Radj2), x=xgrob,  y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))

## labelling axes
if(abb_digits==T){
 Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,dgts)
 Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,dgts) 
 }else{
   if(round(summary(rdafunct)$cont$importance[2,1]*100,0)==0){
     Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,1)
     }else{
       Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,0)
       }
   if(round(summary(rdafunct)$cont$importance[2,2]*100,0)==0){
     Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,1)
     }else{
       Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,0)
     }
 }
## plot SCALING2
if(plotobjects2 == TRUE){
  plot(rdafunct,scaling=2,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","wa","cn"))
  plot(rdafunct,scaling=1,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","lc","cn"))
}
 rdafunct10 <- capscale(df.sub10~ ivTransit + Condition(PD_Colon) + Condition(Transit), sampleinfo10, dist="bray")
 lblInvivo <- "\U0001D43C\U0001D62F \U0001D637\U0001d62a\U0001D637\U0001D630"
## ggplot
rdascores <- scores(rdafunct10, choices = 1:2, scaling = 2, display = "sites")
firstaxispoints <- rdascores[, 1]; secondaxispoints <- rdascores[, 2]
points <- data.frame(fa = firstaxispoints, sa = secondaxispoints, Donor = sampleinfo10$Donor, 
    P_D = sampleinfo10$PD_Colon, Transit = sampleinfo10$Transit, ivTransit = sampleinfo10$ivTransit)
points$P_D <- mapvalues(points$P_D, from = c("P", "D"), to = c("Proximal", "Distal"))

rdascorevfa <- data.frame(scores(rdafunct10, choices = 1:2, scaling = 2, display = "species"))
rdafactor <- data.frame(summary(rdafunct10)$centroids)
rownames(rdafactor) <- mapvalues(rownames(rdafactor), from = c("ivTransitSTD", "ivTransitMTD", "ivTransitLTD"), 
                                 to = c("Short", "Medium", "Long"))
plotobj <- ggplot(as.data.frame(points)) + geom_point(aes(x = fa, y = sa, size = pointsize, color = ivTransit), alpha = 0.8) +
  scale_colour_manual(values = transitcolors2, name = substitute(paste(italic("In vivo"), " transit")), labels = c("Short", "Medium", "Long")) +
  xlab(paste("RDA1", Dim1, "%", sep = " ")) + 
    ylab(paste("RDA2", Dim2, "%", sep = " ")) + annotation_custom(grob) + 
    geom_segment(data = rdascorevfa, aes(x = 0, xend = CAP1, y = 0, yend = CAP2), 
        color = linecolour) + 
      ggrepel::geom_label_repel(data = rdafactor, aes(x = CAP1, y = CAP2),
                              segment.color = "grey50", alpha = alphnumb+.15, 
                                         # fill = "grey", alpha = 0.5,
    color = neon, label = rownames(rdafactor), size = neonsize)
if(plotlabels == T){
if(abbrev == TRUE){
  plotobj <- plotobj + geom_label_repel(data = rdascorevfa,  aes(x = CAP1, y = CAP2, 
                                          label = abbreviate(colnames(subset(a, select = c(-Other))),abbnum),
                                          segment.color = "grey50"
                                          # vjust = ifelse(RDA2 >= 0, -0.1, 0.7), 
                                          # hjust = ifelse(RDA1 >= 0, 0, 0.5)
                                          ), 
                                          color = linecolour, alpha = alphnumb, label.size = NA,
                                          size = labelsize, fontface = "italic",
                                          position = pos)
} else { 
      plotobj <- plotobj + geom_label_repel(data = rdascorevfa,  aes(x = CAP1, y = CAP2, 
                                          label = colnames(subset(a, select = c(-Other))),
                                          segment.color = "grey50"
                                          # vjust = ifelse(RDA2 >= 0, -0.1, 0.7), 
                                          # hjust = ifelse(RDA1 >= 0, 0, 0.5)
                                          ), 
                                          color = linecolour, alpha = alphnumb, label.size = NA,
                                          size = labelsize, fontface = "italic",
                                          position = pos)
      }}
plotobj <- papertheme(plotobj)
plotobj <- plotobj + guides(color = guide_legend(ncol = 1, direction = "horizontal",
      title.position = "top", override.aes = list(size = 5)), size = "none") + theme(legend.position = "right")
mylegendPDfunctnet <- g_legend(plotobj)
plotobj <- plotobj + theme(legend.position = "none"           , 
        legend.text = element_text(size=legendsize,family='sans',angle=0),
        legend.title = element_text(size=legendsize,family='sans',angle=0),
        legend.direction = "horizontal") 
plotobjrdatop15PDC <- plotobj + xlim(min(rdascorevfa)-0.5, max(points$fa + 0.5)) + 
  # ggtitle(lefttitle) + 
  theme(plot.title = element_text(hjust = 0.5, face="bold"))
if(plotobjects == TRUE){plotobjrdatop15PDC}



# Vessel
## to plot original species on graph: capscale function!!!
 dbrdail <- dbrda(df.sub~ PD_Colon + Condition(Transit) + Condition(ivTransit), sampleinfo, dist="bray")
 summary(dbrdail)
 coef(dbrdail)
 plot(dbrdail)
 
## permutation test of the results
 anova <- anova.cca(dbrdail,step=1000)
 Panova <- c(Panova, "PD"= anova[1,4])
 
 # cores <- makeCluster(4) # Define amount of clusters
 # anova.cca(dbrdail,by="axis",step=1000, parallel = cores)
 # stopCluster(cores)
## explanatory power of the variables included
 Runadj <- RsquareAdj(dbrdail)$r.squared
 Radj <- RsquareAdj(dbrdail)$adj.r.squared
 Radjs <- c(Radjs, "PD" = Radj)
 
## Perform RDA with pcoascores to check if its the same
 rdafunctR <- rda(pcoascores~PD_Colon + Condition(Transit) + Condition(ivTransit), sampleinfo)
 rdafunct <- capscale(df.sub~ PD_Colon + Condition(Transit) + Condition(ivTransit), sampleinfo, dist="bray")
 # summary(rdafunct); coef(rdafunct)
 
## explanatory power of the variables included
 Runadj <- RsquareAdj(rdafunctR)$r.squared
 Radj <- RsquareAdj(rdafunctR)$adj.r.squared
 CheckRadjs <- c(CheckRadjs, "PD" = Radj)
 
 
 ## As label on plot
Radj2 <- signif(Radj, digits = ndigits)
  grob <- grobTree(textGrob(paste("R\u00B2= ",Radj2), x=xgrob,  y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))

## labelling axes
if(abb_digits==T){
 Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,dgts)
 Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,dgts) 
 }else{
   if(round(summary(rdafunct)$cont$importance[2,1]*100,0)==0){
     Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,1)
     }else{
       Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,0)
       }
   if(round(summary(rdafunct)$cont$importance[2,2]*100,0)==0){
     Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,1)
     }else{
       Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,0)
     }
 }
## plot SCALING2
if(plotobjects2 == TRUE){
  plot(rdafunct,scaling=2,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","wa","cn"))
  plot(rdafunct,scaling=1,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","lc","cn"))
}
 rdafunct10 <- capscale(df.sub10~ PD_Colon + Condition(Transit) + Condition(ivTransit), sampleinfo10, dist="bray")
 
## ggplot
rdascores <- scores(rdafunct10, choices = 1:2, scaling = 2, display = "sites")
firstaxispoints <- rdascores[, 1]; secondaxispoints <- rdascores[, 2]
points <- data.frame(fa = firstaxispoints, sa = secondaxispoints, Donor = sampleinfo10$Donor, 
    P_D = sampleinfo10$PD_Colon, Transit = sampleinfo10$Transit, ivTransit = sampleinfo10$ivTransit)
points$P_D <- mapvalues(points$P_D, from = c("P", "D"), to = c("Proximal", "Distal"))

rdascorevfa <- data.frame(scores(rdafunct10, choices = 1:2, scaling = 2, display = "species"))
rdafactor <- data.frame(summary(rdafunct10)$centroids)
rownames(rdafactor) <- mapvalues(rownames(rdafactor), from = c("PD_ColonD", "PD_ColonP"), 
                                 to = c("Distal", "Proximal"))
plotobj <- ggplot(as.data.frame(points)) + geom_point(aes(x = fa, y = sa, size = pointsize, color = P_D), alpha = 0.8) +
  scale_colour_manual(values = c("Proximal" = "#cdb174", "Distal" = "#a48d5c"), name = "Vessel") + 
  xlab(paste("RDA1", Dim1, "%", sep = " ")) + 
    ylab(paste("RDA2", Dim2, "%", sep = " ")) + annotation_custom(grob) + 
    geom_segment(data = rdascorevfa, aes(x = 0, xend = CAP1, y = 0, yend = MDS1), 
        color = linecolour) + 
      ggrepel::geom_label_repel(data = rdafactor, aes(x = CAP1, y = MDS1),
                              segment.color = "grey50", alpha = alphnumb+.15, 
                                         # fill = "grey", alpha = 0.5,
    color = neon, label = rownames(rdafactor), size = neonsize)
if(plotlabels == T){
if(abbrev == TRUE){
  plotobj <- plotobj + geom_label_repel(data = rdascorevfa,  aes(x = CAP1, y = MDS1, 
                                          label = abbreviate(colnames(subset(a, select = c(-Other))),abbnum),
                                          segment.color = "grey50"
                                          # vjust = ifelse(RDA2 >= 0, -0.1, 0.7), 
                                          # hjust = ifelse(RDA1 >= 0, 0, 0.5)
                                          ), 
                                          color = linecolour, alpha = alphnumb, label.size = NA,
                                          size = labelsize, fontface = "italic",
                                          position = pos)
} else { 
      plotobj <- plotobj + geom_label_repel(data = rdascorevfa,  aes(x = CAP1, y = MDS1, 
                                          label = colnames(subset(a, select = c(-Other))),
                                          segment.color = "grey50"
                                          # vjust = ifelse(RDA2 >= 0, -0.1, 0.7), 
                                          # hjust = ifelse(RDA1 >= 0, 0, 0.5)
                                          ), 
                                          color = linecolour, alpha = alphnumb, label.size = NA,
                                          size = labelsize, fontface = "italic",
                                          position = pos)
      }}
plotobj <- papertheme(plotobj)
plotobj <- plotobj + guides(color = guide_legend(ncol = 1, direction = "horizontal",
      title.position = "top", override.aes = list(size = 5)), size = "none") + theme(legend.position = "right")
mylegendPDfunctnet <- g_legend(plotobj)
plotobj <- plotobj + theme(legend.position = "none"           , 
        legend.text = element_text(size=legendsize,family='sans',angle=0),
        legend.title = element_text(size=legendsize,family='sans',angle=0),
        legend.direction = "horizontal") 
plotobjrdatop15PDD <- plotobj + xlim(min(rdascorevfa)-0.5, max(points$fa + 0.5)) + 
  # ggtitle(lefttitle) + 
  theme(plot.title = element_text(hjust = 0.5, face="bold"))
if(plotobjects == TRUE){plotobjrdatop15PDD}




 
 

ggarrange(plotobjrdatop15PDA + theme(legend.position = c(0.81,0.1)), 
          plotobjrdatop15PDB + theme(legend.position = c(0.15,0.1)),
          plotobjrdatop15PDC + theme(legend.position = c(0.15,0.1)), 
          plotobjrdatop15PDD + theme(legend.position = c(0.15,0.1)),nrow = 2, ncol = 2) 
 
 
 
 
# Adjust p-values because of multiple testing
 # Correction for multiple testing (BenjaminiâHochberg procedure, FDR) 
 # Panova
 # p.adjust(Panova)
 Panova_ABS <- Panova
 padj_abs <- p.adjust(Panova, method = "holm")
 padj_abs <- p.adjust(Panova)
 

```



# db-RDA - PMP
## Buildup model
```{r db-RDA, eval=, echo=TRUE, fig.height=15, fig.width=13.5}
# Proportional
 ## Make data frame of all genera
phylobj.QMP.prop = transform_sample_counts(phylobj.QMP, function(x) 100 * x/sum(x))
phylobj.QMP.prop <- subset_samples(phylobj.QMP.prop, Ninetransit ==3)
phylobj.QMP.prop <- aggregate_taxa(phylobj.QMP.prop, "Genus")
a <- as.data.frame(vegan_otu(phylobj.QMP.prop))
b <- as.data.frame(sample_data(phylobj.QMP.prop))
df.phylobj.QMP <- cbind(Samplenumber = b$Samplenumber,a,b$Day, Transit=b$Transit, ivTransit=b$ivTransit,Donor = b$Donor, PD_Colon =b$PD_Colon)
df.phylobj.QMP.otu <- df.phylobj.QMP[2:cols2.df]


 
# Define data frames
genus_table <- df.phylobj.QMP.otu %>%
   mutate_all(as.numeric)
sampleinfo <- subset(df.phylobj.all, select = c(Day, Donor, Transit, ivTransit, PD_Colon))

# Develop correct model
## Compute dissimilarity matrix
 rdaplot <- genus_table; df.sub <- genus_table
 rdajaccard <- vegdist(rdaplot,method="jaccard",binary="FALSE")
 rdabray <- vegdist(rdaplot,method="bray",binary="FALSE")
## Perform PCoA
 pcoail <- cmdscale(rdajaccard,k=(nrow(rdaplot)-1),eig=TRUE)
 pcoail <- cmdscale(rdabray,k=(nrow(rdaplot)-1),eig=TRUE)
 pcoascores <- pcoail$points
## Automatic stepwise model building for constrained ordination methods
 mod0 <- rda(pcoascores ~ 1, sampleinfo)  # Model with intercept only
 mod1 <- rda(pcoascores ~ ., sampleinfo)  # Model with all explanatory variables
 ordi <- ordiR2step(mod0, scope = formula(mod1), perm.max = 200)
 # ordistep(mod0, scope = formula(mod1), perm.max = 200)
 ordi
 ordi$anova
 
test <- ordi[["anova"]]
test <- rownames_to_column(test) %>%
  mutate_all(funs(gsub("[[:punct:]]", "", .)))
test <- test %>%
  mutate_all(funs(gsub(" ", "", .)))
test <- test[1:nrow(test) - 1,]
variables <- test$rowname; message(crayon::green(paste("Hey, your variables are:"))); variables
# Go through all the selected variables in the model
 Radjs <- c(); CheckRadjs <- c(); Panova <- c()
```


## 1) donors - Run the models
```{r db-RDA, eval=, echo=TRUE, fig.height=15, fig.width=13.5}
neon <- c("#2f89fc") ; labelcolour <- c("#fc0202");
labelcolour <- "#fc0202"; plotlabels = F
linecolour <- c("#fc0202"); showR = FALSE
PDcolours <- c("#434e52", "#5b8c85" ) ; ndigits <- 2
legendsize <- 15 ; labelsize <- 6; ygrob <- 0.97; neonsize <- 7; topn <- 15
if(showR == TRUE){xgrob <- 0.85}else{xgrob <- 0.9}; abb_digits = T; dgts = 1; abbnum = 3; alphnumb = 0.62
lefttitle <- "Absolute community abundance" ; righttitle <- "Proportional community abundance"
legendsize <- 18; pointsize = 9; grobsize <- 15; abbrev <- F; plotobjects <- TRUE; plotobjects2 <- F
pos <- position_jitter(width = 0.0, seed = 2)
saveplots()




# Go through all the selected variables in the model
 Radjs <- c(); CheckRadjs <- c(); Panova <- c()
 
# SHIME transit time
## to plot original species on graph: capscale function!!!
 dbrdail <- dbrda(df.sub~ Transit + Condition(PD_Colon) + Condition(Donor), sampleinfo, dist="bray")
 summary(dbrdail)
 coef(dbrdail)
 plot(dbrdail)
 
## permutation test of the results
 anova <- anova.cca(dbrdail,step=1000)
 Panova <- c(Panova, "Transit"= anova[1,4])
 
 # cores <- makeCluster(4) # Define amount of clusters
 # anova.cca(dbrdail,by="axis",step=1000, parallel = cores)
 # stopCluster(cores)
## explanatory power of the variables included
 Runadj <- RsquareAdj(dbrdail)$r.squared
 Radj <- RsquareAdj(dbrdail)$adj.r.squared
 Radjs <- c(Radjs, "Transit" = Radj)
 
## Perform RDA with pcoascores to check if its the same
 rdafunct <- rda(pcoascores~Transit + Condition(PD_Colon) + Condition(Donor), sampleinfo)
 # summary(rdafunct); coef(rdafunct)
 
## explanatory power of the variables included
 Runadj <- RsquareAdj(rdafunct)$r.squared
 Radj <- RsquareAdj(rdafunct)$adj.r.squared
 CheckRadjs <- c(CheckRadjs, "Transit" = Radj)
 
 
 ## As label on plot
Radj2 <- signif(Radj, digits = ndigits)
if(showR == TRUE){grob <- grobTree(textGrob(paste("R\u00B2 =",Radj2, "\nR\u00B2 top", topn, "=", Radj2), x=xgrob,  y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))
}else{
  grob <- grobTree(textGrob(paste("R\u00B2= ",Radj2), x=xgrob,  y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))
}
## labelling axes
if(abb_digits==T){
 Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,dgts)
 Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,dgts) 
 }else{
   if(round(summary(rdafunct)$cont$importance[2,1]*100,0)==0){
     Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,1)
     }else{
       Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,0)
       }
   if(round(summary(rdafunct)$cont$importance[2,2]*100,0)==0){
     Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,1)
     }else{
       Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,0)
     }
 }
## plot SCALING2
if(plotobjects2 == TRUE){
  plot(rdafunct,scaling=2,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","wa","cn"))
  plot(rdafunct,scaling=1,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","lc","cn"))
}
 
 
## ggplot
rdascores <- scores(rdafunct, choices = 1:2, scaling = 2, display = "sites")
firstaxispoints <- rdascores[, 1]; secondaxispoints <- rdascores[, 2]
points <- data.frame(fa = firstaxispoints, sa = secondaxispoints, Donor = sampleinfo$Donor, 
    P_D = sampleinfo$PD_Colon, Transit = sampleinfo$Transit, ivTransit = sampleinfo$ivTransit)
points$P_D <- mapvalues(points$P_D, from = c("P", "D"), to = c("Proximal", "Distal"))

rdascorevfa <- data.frame(scores(rdafunct, choices = 1:2, scaling = 2, display = "species"))
rdafactor <- data.frame(summary(rdafunct)$centroids)
rownames(rdafactor) <- mapvalues(rownames(rdafactor), from = c("TransitST", "TransitMT", "TransitLT"), 
                                 to = c("Short", "Medium", "Long"))
plotobj <- ggplot(as.data.frame(points)) + geom_point(aes(x = fa, y = sa, size = pointsize, color = Transit), alpha = 0.8) +
  scale_colour_manual(values = transitcolors, name = "Transit") + xlab(paste("RDA1", Dim1, "%", sep = " ")) + 
    ylab(paste("RDA2", Dim2, "%", sep = " ")) + annotation_custom(grob) + 
    geom_segment(data = rdascorevfa, aes(x = 0, xend = RDA1, y = 0, yend = RDA2), 
        color = linecolour) + geom_text(data = rdafactor, aes(x = RDA1, y = RDA2), 
                                         # fill = "grey", alpha = 0.5,
    color = neon, label = rownames(rdafactor), size = neonsize)
if(plotlabels == T){
if(abbrev == TRUE){
  plotobj <- plotobj + geom_label_repel(data = rdascorevfa,  aes(x = RDA1, y = RDA2, 
                                          label = abbreviate(colnames(a),abbnum),
                                          segment.color = "grey50"
                                          # vjust = ifelse(RDA2 >= 0, -0.1, 0.7), 
                                          # hjust = ifelse(RDA1 >= 0, 0, 0.5)
                                          ), 
                                          fill = "grey", alpha = alphnumb, label.size = NA,
                                          color = labelcolour, size = labelsize,
                                          fontface = "italic",
                                          position = pos)
} else { 
      plotobj <- plotobj + geom_label_repel(data = rdascorevfa,  aes(x = RDA1, y = RDA2, 
                                          label = colnames(a),
                                          segment.color = "grey50"
                                          # vjust = ifelse(RDA2 >= 0, -0.1, 0.7), 
                                          # hjust = ifelse(RDA1 >= 0, 0, 0.5)
                                          ), 
                                          fill = "grey", 
                                          alpha = alphnumb, 
                                          label.size = NA,
                                          # color = labelcolour, 
                                          size = labelsize,
                                          fontface = "italic",
                                          position = pos)
      }}
plotobj <- papertheme(plotobj)
plotobj <- plotobj + guides(color = guide_legend(ncol = 1), size = "none") + theme(legend.position = "right")
mylegendPDfunctnet <- g_legend(plotobj)
plotobj <- plotobj + theme(legend.position =  c(0.88,0.05), 
        legend.text = element_text(size=legendsize,family='sans',angle=0),
        legend.title = element_text(size=legendsize,family='sans',angle=0),
        legend.direction = "horizontal") 
plotobjrdatop15PDA <- plotobj + xlim(min(rdascorevfa)-0.5, max(points$fa + 0.5)) + 
  # ggtitle(lefttitle) + 
  theme(plot.title = element_text(hjust = 0.5, face="bold"))
if(plotobjects == TRUE){plotobjrdatop15PDA}


 
 
 
 
 
 
 
 
# PD_Colon
## to plot original species on graph: capscale function!!!
 dbrdail <- dbrda(df.sub~ PD_Colon + Condition(Transit) + Condition(Donor), sampleinfo, dist="bray")
 summary(dbrdail)
 coef(dbrdail)
 plot(dbrdail)
 
## permutation test of the results
 anova.cca(dbrdail,step=1000)
 anova <- anova.cca(dbrdail,step=1000)
 Panova <- c(Panova, "PD_Colon"= anova[1,4])
 
 # cores <- makeCluster(4) # Define amount of clusters
 # anova.cca(dbrdail,by="axis",step=1000, parallel = cores)
 # stopCluster(cores)
 
## explanatory power of the variables included
 Runadj <- RsquareAdj(dbrdail)$r.squared
 Radj <- RsquareAdj(dbrdail)$adj.r.squared
 Radjs <- c(Radjs, "PD_Colon" = Radj)
 
## Perform RDA with pcoascores to check if its the same
 rdafunct <- rda(pcoascores~PD_Colon + Condition(Transit) + Condition(Donor),sampleinfo)
 # summary(rdafunct); coef(rdafunct)
 
## explanatory power of the variables included
 Runadj <- RsquareAdj(rdafunct)$r.squared
 Radj <- RsquareAdj(rdafunct)$adj.r.squared
 CheckRadjs <- c(CheckRadjs, "PD_Colon" = Radj)
 
 
# Donor
## to plot original species on graph: capscale function!!!
 dbrdail <- dbrda(df.sub~ Donor + Condition(Transit) + Condition(PD_Colon), sampleinfo, dist="bray")
 summary(dbrdail)
 coef(dbrdail)
 plot(dbrdail)
 
## permutation test of the results
 anova.cca(dbrdail,step=1000)
 anova <- anova.cca(dbrdail,step=1000)
 Panova <- c(Panova, "Donor" = anova[1,4])
 # cores <- makeCluster(4) # Define amount of clusters
 # anova.cca(dbrdail,by="axis",step=1000, parallel = cores)
 # stopCluster(cores)
 
## explanatory power of the variables included
 Runadj <- RsquareAdj(dbrdail)$r.squared
 Radj <- RsquareAdj(dbrdail)$adj.r.squared
 Radjs <- c(Radjs, "Donor" = Radj)
 
## Perform RDA with pcoascores to check if its the same
 rdafunct <- rda(pcoascores~Donor + Condition(Transit) + Condition(PD_Colon),sampleinfo)
 # summary(rdafunct); coef(rdafunct)
 
## explanatory power of the variables included
 Runadj <- RsquareAdj(rdafunct)$r.squared
 Radj <- RsquareAdj(rdafunct)$adj.r.squared
 CheckRadjs <- c(CheckRadjs, "Donor" = Radj) 
 

 
 
 # ordiR2step(mod0, scope = formula(mod1), perm.max = 200)
 # ordistep(mod0, scope = formula(mod1), perm.max = 200)
 # Radjs
 # CheckRadjs
 Radjs_ABS <- Radjs; CheckRadjs_ABS <- CheckRadjs
 
 
 
# Adjust p-values because of multiple testing
 # Correction for multiple testing (BenjaminiâHochberg procedure, FDR) 
 # Panova
 # p.adjust(Panova)
 Panova_ABS <- Panova
 padj_abs <- p.adjust(Panova, method = "holm")
 padj_abs <- p.adjust(Panova)
 
  
 
 Radjs_ABS
 CheckRadjs_ABS
 CheckRadjs_ABS_donor <- CheckRadjs_ABS
 Panova_ABS
 padj_abs
```

## 2) in vivo transit - Run the models
```{r db-RDA, eval=, echo=TRUE, fig.height=15, fig.width=13.5}
# Go through all the selected variables in the model
 Radjs <- c(); CheckRadjs <- c(); Panova <- c()
 
# SHIME transit time
## to plot original species on graph: capscale function!!!
 dbrdail <- dbrda(df.sub~ Transit + Condition(PD_Colon) + Condition(ivTransit), sampleinfo, dist="bray")
 summary(dbrdail)
 coef(dbrdail)
 plot(dbrdail)
 
## permutation test of the results
 anova <- anova.cca(dbrdail,step=1000)
 Panova <- c(Panova, "Transit"= anova[1,4])
 
 # cores <- makeCluster(4) # Define amount of clusters
 # anova.cca(dbrdail,by="axis",step=1000, parallel = cores)
 # stopCluster(cores)
## explanatory power of the variables included
 Runadj <- RsquareAdj(dbrdail)$r.squared
 Radj <- RsquareAdj(dbrdail)$adj.r.squared
 Radjs <- c(Radjs, "Transit" = Radj)
 
## Perform RDA with pcoascores to check if its the same
 rdafunct <- rda(pcoascores~Transit + Condition(PD_Colon) + Condition(ivTransit), sampleinfo)
 # summary(rdafunct); coef(rdafunct)
 
## explanatory power of the variables included
 Runadj <- RsquareAdj(rdafunct)$r.squared
 Radj <- RsquareAdj(rdafunct)$adj.r.squared
 CheckRadjs <- c(CheckRadjs, "Transit" = Radj)
 
 
# PD_Colon
## to plot original species on graph: capscale function!!!
 dbrdail <- dbrda(df.sub~ PD_Colon + Condition(Transit) + Condition(ivTransit), sampleinfo, dist="bray")
 summary(dbrdail)
 coef(dbrdail)
 plot(dbrdail)
 
## permutation test of the results
 anova.cca(dbrdail,step=1000)
 anova <- anova.cca(dbrdail,step=1000)
 Panova <- c(Panova, "PD_Colon"= anova[1,4])
 
 # cores <- makeCluster(4) # Define amount of clusters
 # anova.cca(dbrdail,by="axis",step=1000, parallel = cores)
 # stopCluster(cores)
 
## explanatory power of the variables included
 Runadj <- RsquareAdj(dbrdail)$r.squared
 Radj <- RsquareAdj(dbrdail)$adj.r.squared
 Radjs <- c(Radjs, "PD_Colon" = Radj)
 
## Perform RDA with pcoascores to check if its the same
 rdafunct <- rda(pcoascores~PD_Colon + Condition(Transit) + Condition(ivTransit),sampleinfo)
 # summary(rdafunct); coef(rdafunct)
 
## explanatory power of the variables included
 Runadj <- RsquareAdj(rdafunct)$r.squared
 Radj <- RsquareAdj(rdafunct)$adj.r.squared
 CheckRadjs <- c(CheckRadjs, "PD_Colon" = Radj)
 
 
# ivTransit
## to plot original species on graph: capscale function!!!
 dbrdail <- dbrda(df.sub~ ivTransit + Condition(Transit) + Condition(PD_Colon), sampleinfo, dist="bray")
 summary(dbrdail)
 coef(dbrdail)
 plot(dbrdail)
 
## permutation test of the results
 anova.cca(dbrdail,step=1000)
 anova <- anova.cca(dbrdail,step=1000)
 Panova <- c(Panova, "ivTransit" = anova[1,4])
 # cores <- makeCluster(4) # Define amount of clusters
 # anova.cca(dbrdail,by="axis",step=1000, parallel = cores)
 # stopCluster(cores)
 
## explanatory power of the variables included
 Runadj <- RsquareAdj(dbrdail)$r.squared
 Radj <- RsquareAdj(dbrdail)$adj.r.squared
 Radjs <- c(Radjs, "ivTransit" = Radj)
 
## Perform RDA with pcoascores to check if its the same
 rdafunct <- rda(pcoascores~ivTransit + Condition(Transit) + Condition(PD_Colon),sampleinfo)
 # summary(rdafunct); coef(rdafunct)
 
## explanatory power of the variables included
 Runadj <- RsquareAdj(rdafunct)$r.squared
 Radj <- RsquareAdj(rdafunct)$adj.r.squared
 CheckRadjs <- c(CheckRadjs, "ivTransit" = Radj) 
 
 

 # ordiR2step(mod0, scope = formula(mod1), perm.max = 200)
 # ordistep(mod0, scope = formula(mod1), perm.max = 200)
 # Radjs
 # CheckRadjs
 Radjs_ABS <- Radjs; CheckRadjs_ABS <- CheckRadjs
 
 
 
# Adjust p-values because of multiple testing
 # Correction for multiple testing (BenjaminiâHochberg procedure, FDR) 
 # Panova
 # p.adjust(Panova)
 Panova_ABS <- Panova
 padj_abs <- p.adjust(Panova, method = "holm")
 padj_abs <- p.adjust(Panova)
 
  
 
 Radjs_ABS
 CheckRadjs_ABS
 CheckRadjs_ABS_donor
 Panova_ABS
 padj_abs
```

### Plot them
```{r fig.height=17, fig.width=15}
neon <- c("#2f89fc") ; labelcolour <- c("#fc0202");
labelcolour <- "#fc0202"; plotlabels = F
linecolour <- c("#fc0202"); showR = FALSE
PDcolours <- c("#434e52", "#5b8c85" ) ; ndigits <- 2
legendsize <- 15 ; labelsize <- 6; ygrob <- 0.9; neonsize <- 7; topn <- 15
if(showR == TRUE){xgrob <- 0.85}else{xgrob <- 0.8}; abb_digits = T; dgts = 1; abbnum = 3; alphnumb = 0.62
lefttitle <- "Absolute community abundance" ; righttitle <- "Proportional community abundance"
legendsize <- 18; pointsize = 9; grobsize <- 15; abbrev <- F; plotobjects <- TRUE; plotobjects2 <- F
pos <- position_jitter(width = 0.0, seed = 2)
saveplots()




# Go through all the selected variables in the model
 Radjs <- c(); CheckRadjs <- c(); Panova <- c()
 
# SHIME transit time
## to plot original species on graph: capscale function!!!
 dbrdail <- dbrda(df.sub~ Transit + Condition(PD_Colon) + Condition(ivTransit), sampleinfo, dist="bray")
 summary(dbrdail)
 coef(dbrdail)
 plot(dbrdail)
 
## permutation test of the results
 anova <- anova.cca(dbrdail,step=1000)
 Panova <- c(Panova, "Transit"= anova[1,4])
 
 # cores <- makeCluster(4) # Define amount of clusters
 # anova.cca(dbrdail,by="axis",step=1000, parallel = cores)
 # stopCluster(cores)
## explanatory power of the variables included
 Runadj <- RsquareAdj(dbrdail)$r.squared
 Radj <- RsquareAdj(dbrdail)$adj.r.squared
 Radjs <- c(Radjs, "Transit" = Radj)
 
## Perform RDA with pcoascores to check if its the same
 rdafunct <- rda(pcoascores~Transit + Condition(PD_Colon) + Condition(ivTransit), sampleinfo)
 # summary(rdafunct); coef(rdafunct)
 
## explanatory power of the variables included
 Runadj <- RsquareAdj(rdafunct)$r.squared
 Radj <- RsquareAdj(rdafunct)$adj.r.squared
 CheckRadjs <- c(CheckRadjs, "Transit" = Radj)
 
 
 ## As label on plot
Radj2 <- signif(Radj, digits = ndigits)
if(showR == TRUE){grob <- grobTree(textGrob(paste("R\u00B2 =",Radj2, "\nR\u00B2 top", topn, "=", Radj2), x=xgrob,  y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))
}else{
  grob <- grobTree(textGrob(paste("R\u00B2= ",Radj2), x=xgrob,  y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))
}
## labelling axes
if(abb_digits==T){
 Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,dgts)
 Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,dgts) 
 }else{
   if(round(summary(rdafunct)$cont$importance[2,1]*100,0)==0){
     Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,1)
     }else{
       Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,0)
       }
   if(round(summary(rdafunct)$cont$importance[2,2]*100,0)==0){
     Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,1)
     }else{
       Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,0)
     }
 }
## plot SCALING2
if(plotobjects2 == TRUE){
  plot(rdafunct,scaling=2,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","wa","cn"))
  plot(rdafunct,scaling=1,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","lc","cn"))
}
 
 
## ggplot
rdascores <- scores(rdafunct, choices = 1:2, scaling = 2, display = "sites")
firstaxispoints <- rdascores[, 1]; secondaxispoints <- rdascores[, 2]
points <- data.frame(fa = firstaxispoints, sa = secondaxispoints, Donor = sampleinfo$Donor, 
    P_D = sampleinfo$PD_Colon, Transit = sampleinfo$Transit, ivTransit = sampleinfo$ivTransit)
points$P_D <- mapvalues(points$P_D, from = c("P", "D"), to = c("Proximal", "Distal"))

rdascorevfa <- data.frame(scores(rdafunct, choices = 1:2, scaling = 2, display = "species"))
rdafactor <- data.frame(summary(rdafunct)$centroids)
rownames(rdafactor) <- mapvalues(rownames(rdafactor), from = c("TransitST", "TransitMT", "TransitLT"), 
                                 to = c("Short", "Medium", "Long"))
plotobj <- ggplot(as.data.frame(points)) + geom_point(aes(x = fa, y = sa, 
                                                          size = pointsize, color = Transit), alpha = 0.8) + scale_color_manual("SHIME transit", values = transitcolors, labels = c("Short", "Medium", "Long"))+
   xlab(paste("RDA1", Dim1, "%", sep = " ")) + 
    ylab(paste("RDA2", Dim2, "%", sep = " ")) + annotation_custom(grob) + 
    geom_segment(data = rdascorevfa, aes(x = 0, xend = RDA1, y = 0, yend = RDA2), 
        color = linecolour) + geom_text(data = rdafactor, aes(x = RDA1, y = RDA2), 
                                         # fill = "grey", alpha = 0.5,
    color = neon, label = rownames(rdafactor), size = neonsize)
if(plotlabels == T){
if(abbrev == TRUE){
  plotobj <- plotobj + geom_label_repel(data = rdascorevfa,  aes(x = RDA1, y = RDA2, 
                                          label = abbreviate(colnames(a),abbnum),
                                          segment.color = "grey50"
                                          # vjust = ifelse(RDA2 >= 0, -0.1, 0.7), 
                                          # hjust = ifelse(RDA1 >= 0, 0, 0.5)
                                          ), 
                                          fill = "grey", alpha = alphnumb, label.size = NA,
                                          color = labelcolour, size = labelsize,
                                          fontface = "italic",
                                          position = pos)
} else { 
      plotobj <- plotobj + geom_label_repel(data = rdascorevfa,  aes(x = RDA1, y = RDA2, 
                                          label = colnames(a),
                                          segment.color = "grey50"
                                          # vjust = ifelse(RDA2 >= 0, -0.1, 0.7), 
                                          # hjust = ifelse(RDA1 >= 0, 0, 0.5)
                                          ), 
                                          fill = "grey", 
                                          alpha = alphnumb, 
                                          label.size = NA,
                                          # color = labelcolour, 
                                          size = labelsize,
                                          fontface = "italic",
                                          position = pos)
      }}
plotobj <- papertheme(plotobj)
plotobj <- plotobj + guides(color = guide_legend(ncol = 1, direction = "horizontal",
      title.position = "top", override.aes = list(size = 5)), shape = guide_legend(ncol = 1,direction = "horizontal", override.aes = list(size = 5), title.position = "top"), size = "none") +
  theme(legend.position = "right") 
mylegendPDfunctnet <- g_legend(plotobj)
plotobj <- plotobj + theme(legend.position =  c(0.88,0.05), 
        legend.text = element_text(size=legendsize,family='sans',angle=0),
        legend.title = element_text(size=legendsize,family='sans',angle=0),
        legend.direction = "horizontal") 
plotobjrdatop15PDD <- plotobj + xlim(min(rdascorevfa)-0.5, max(points$fa + 0.5)) + 
  # ggtitle(lefttitle) + 
  theme(plot.title = element_text(hjust = 0.5, face="bold"))
if(plotobjects == TRUE){plotobjrdatop15PDA}


 
 
 
 
 
 
 
 
# PD_Colon
## to plot original species on graph: capscale function!!!
 dbrdail <- dbrda(df.sub~ PD_Colon + Condition(Transit) + Condition(ivTransit), sampleinfo, dist="bray")
 summary(dbrdail)
 coef(dbrdail)
 plot(dbrdail)
 
## permutation test of the results
 anova.cca(dbrdail,step=1000)
 anova <- anova.cca(dbrdail,step=1000)
 Panova <- c(Panova, "PD_Colon"= anova[1,4])
 
 # cores <- makeCluster(4) # Define amount of clusters
 # anova.cca(dbrdail,by="axis",step=1000, parallel = cores)
 # stopCluster(cores)
 
## explanatory power of the variables included
 Runadj <- RsquareAdj(dbrdail)$r.squared
 Radj <- RsquareAdj(dbrdail)$adj.r.squared
 Radjs <- c(Radjs, "PD_Colon" = Radj)
 
## Perform RDA with pcoascores to check if its the same
 rdafunct <- rda(pcoascores~PD_Colon + Condition(Transit) + Condition(ivTransit),sampleinfo)
 # summary(rdafunct); coef(rdafunct)
 
## explanatory power of the variables included
 Runadj <- RsquareAdj(rdafunct)$r.squared
 Radj <- RsquareAdj(rdafunct)$adj.r.squared
 CheckRadjs <- c(CheckRadjs, "PD_Colon" = Radj)
 
 
  ## As label on plot
Radj2 <- signif(Radj, digits = ndigits)
if(showR == TRUE){grob <- grobTree(textGrob(paste("R\u00B2 =",Radj2, "\nR\u00B2 top", topn, "=", Radj2), x=xgrob,  y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))
}else{
  grob <- grobTree(textGrob(paste("R\u00B2= ",Radj2), x=xgrob,  y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))
}
## labelling axes
if(abb_digits==T){
 Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,dgts)
 Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,dgts) 
 }else{
   if(round(summary(rdafunct)$cont$importance[2,1]*100,0)==0){
     Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,1)
     }else{
       Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,0)
       }
   if(round(summary(rdafunct)$cont$importance[2,2]*100,0)==0){
     Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,1)
     }else{
       Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,0)
     }
 }
## plot SCALING2
if(plotobjects2 == TRUE){
  plot(rdafunct,scaling=2,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","wa","cn"))
  plot(rdafunct,scaling=1,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","lc","cn"))
}
 
 
## ggplot
rdascores <- scores(rdafunct, choices = 1:2, scaling = 2, display = "sites")
firstaxispoints <- rdascores[, 1]; secondaxispoints <- rdascores[, 2]
points <- data.frame(fa = firstaxispoints, sa = secondaxispoints, Donor = sampleinfo$Donor, 
    P_D = sampleinfo$PD_Colon, Transit = sampleinfo$Transit, ivTransit = sampleinfo$ivTransit)
points$P_D <- mapvalues(points$P_D, from = c("P", "D"), to = c("Proximal", "Distal"))

rdascorevfa <- data.frame(scores(rdafunct, choices = 1:2, scaling = 2, display = "species"))
rdafactor <- data.frame(summary(rdafunct)$centroids)
rownames(rdafactor) <- mapvalues(rownames(rdafactor), from = c("PD_ColonD", "PD_ColonP"), 
                                 to = c("Distal", "Proximal"))
plotobj <- ggplot(as.data.frame(points)) + geom_point(aes(x = fa, y = sa, size = pointsize, color = P_D), alpha = 0.8) +
  scale_colour_manual(values = c("Proximal" = "#cdb174", "Distal" = "#a48d5c"), name = "Vessel") +
  xlab(paste("RDA1", Dim1, "%", sep = " ")) + 
    ylab(paste("PC1", Dim2, "%", sep = " ")) + annotation_custom(grob) + 
    geom_segment(data = rdascorevfa, aes(x = 0, xend = RDA1, y = 0, yend =PC1), 
        color = linecolour) + geom_text(data = rdafactor, aes(x = RDA1, y = PC1), 
                                         # fill = "grey", alpha = 0.5,
    color = neon, label = rownames(rdafactor), size = neonsize)
plotobj <- papertheme(plotobj)
plotobj <- plotobj + guides(color = guide_legend(ncol =1, direction = "horizontal",
      title.position = "top", override.aes = list(size = 5)), shape = guide_legend(ncol = 1,direction = "horizontal", override.aes = list(size = 5), title.position = "top"), size = "none") +
  theme(legend.position = "right") 
mylegendPDfunctnet <- g_legend(plotobj)
plotobj <- plotobj + theme(legend.position =  c(0.88,0.05), 
        legend.text = element_text(size=legendsize,family='sans',angle=0),
        legend.title = element_text(size=legendsize,family='sans',angle=0),
        legend.direction = "horizontal") 
plotobjrdatop15PDE <- plotobj + xlim(min(rdascorevfa)-0.5, max(points$fa + 0.5)) + 
  # ggtitle(lefttitle) + 
  theme(plot.title = element_text(hjust = 0.5, face="bold"))
if(plotobjects == TRUE){plotobjrdatop15PDE}
 
 
# Donor
## to plot original species on graph: capscale function!!!
 dbrdail <- dbrda(df.sub~ ivTransit + Condition(Transit) + Condition(PD_Colon), sampleinfo, dist="bray")
 summary(dbrdail)
 coef(dbrdail)
 plot(dbrdail)
 
## permutation test of the results
 anova.cca(dbrdail,step=1000)
 anova <- anova.cca(dbrdail,step=1000)
 Panova <- c(Panova, "Donor" = anova[1,4])
 # cores <- makeCluster(4) # Define amount of clusters
 # anova.cca(dbrdail,by="axis",step=1000, parallel = cores)
 # stopCluster(cores)
 
## explanatory power of the variables included
 Runadj <- RsquareAdj(dbrdail)$r.squared
 Radj <- RsquareAdj(dbrdail)$adj.r.squared
 Radjs <- c(Radjs, "Donor" = Radj)
 
## Perform RDA with pcoascores to check if its the same
 rdafunct <- rda(pcoascores~ivTransit + Condition(Transit) + Condition(PD_Colon),sampleinfo)
 # summary(rdafunct); coef(rdafunct)
 
## explanatory power of the variables included
 Runadj <- RsquareAdj(rdafunct)$r.squared
 Radj <- RsquareAdj(rdafunct)$adj.r.squared
 CheckRadjs <- c(CheckRadjs, "Donor" = Radj) 
 

 xgrobb = 0.65
   ## As label on plot
Radj2 <- signif(Radj, digits = ndigits)
if(showR == TRUE){grob <- grobTree(textGrob(paste("R\u00B2 =",Radj2, "\nR\u00B2 top", topn, "=", Radj2), x=xgrob,  y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))
}else{
  grob <- grobTree(textGrob(paste("Donor R\u00B2 =", signif(CheckRadjs_ABS_donor[[3]], digits = ndigits), 
                                  "\nIn vivo transit R\u00B2= ",Radj2), x=xgrobb, 
                            y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))
}
## labelling axes
if(abb_digits==T){
 Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,dgts)
 Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,dgts) 
 }else{
   if(round(summary(rdafunct)$cont$importance[2,1]*100,0)==0){
     Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,1)
     }else{
       Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,0)
       }
   if(round(summary(rdafunct)$cont$importance[2,2]*100,0)==0){
     Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,1)
     }else{
       Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,0)
     }
 }
## plot SCALING2
if(plotobjects2 == TRUE){
  plot(rdafunct,scaling=2,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","wa","cn"))
  plot(rdafunct,scaling=1,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","lc","cn"))
}
 
 
## ggplot
rdascores <- scores(rdafunct, choices = 1:2, scaling = 2, display = "sites")
firstaxispoints <- rdascores[, 1]; secondaxispoints <- rdascores[, 2]
points <- data.frame(fa = firstaxispoints, sa = secondaxispoints, Donor = sampleinfo$Donor, 
    P_D = sampleinfo$PD_Colon, Transit = sampleinfo$Transit, ivTransit = sampleinfo$ivTransit)
points$P_D <- mapvalues(points$P_D, from = c("P", "D"), to = c("Proximal", "Distal"))

rdascorevfa <- data.frame(scores(rdafunct, choices = 1:2, scaling = 2, display = "species"))
rdafactor <- data.frame(summary(rdafunct)$centroids)
rownames(rdafactor) <- mapvalues(rownames(rdafactor), from = c("ivTransitLTD", "ivTransitMTD", "ivTransitSTD"), 
                                 to = c("Long", "Medium", "Short"))
plotobj <- ggplot(as.data.frame(points)) + geom_point(aes(x = fa, y = sa, size = pointsize, color = Donor, shape = ivTransit), alpha = 0.8) +
  scale_colour_manual(values = sixdonors, name = "Donor")+ 
  scale_shape_manual(name = expression(paste(italic("In vivo"), " transit")), 
              values = c("STD" = 16, "MTD" = 17, "LTD" = 15), labels = c("Short", "Medium", "Long"))+
  xlab(paste("RDA1", Dim1, "%", sep = " ")) + 
    ylab(paste("RDA2", Dim2, "%", sep = " ")) + annotation_custom(grob) + 
    geom_segment(data = rdascorevfa, aes(x = 0, xend = RDA1, y = 0, yend =RDA2), 
        color = linecolour) + geom_text(data = rdafactor, aes(x = RDA1, y = RDA2), 
                                         # fill = "grey", alpha = 0.5,
    color = neon, label = rownames(rdafactor), size = neonsize)
plotobj <- papertheme(plotobj)
plotobj <- plotobj + guides(color = guide_legend(ncol = 2, direction = "horizontal",
      title.position = "top", override.aes = list(size = 5)), shape = guide_legend(ncol = 1,direction = "horizontal", override.aes = list(size = 5), title.position = "top"), size = "none") +
  theme(legend.position = "right") 
mylegendPDfunctnet <- g_legend(plotobj)
plotobj <- plotobj + theme(legend.position =  c(0.88,0.5), 
        legend.text = element_text(size=legendsize,family='sans',angle=0),
        legend.title = element_text(size=legendsize,family='sans',angle=0),
        legend.direction = "horizontal") 
plotobjrdatop15PDF <- plotobj + xlim(min(rdascorevfa)-0.5, max(points$fa + 0.8)) + 
  # ggtitle(lefttitle) + 
  theme(plot.title = element_text(hjust = 0.5, face="bold"))
if(plotobjects == TRUE){plotobjrdatop15PDF}
 

ggarrange(plotobjrdatop15PDA + ggtitle("Quantitative microbiome composition") + theme(legend.position = "none"), 
          plotobjrdatop15PDD + ggtitle("Proportional microbiome composition") + theme(legend.position = c(0.85,0.3)), 
          plotobjrdatop15PDC, plotobjrdatop15PDF + theme(legend.position = c(0.85,0.5)), 
          plotobjrdatop15PDB, plotobjrdatop15PDE + theme(legend.position = c(0.85,0.3)),
          nrow = 3, ncol = 2)
 
 
 
 
# Adjust p-values because of multiple testing
 # Correction for multiple testing (BenjaminiâHochberg procedure, FDR) 
 # Panova
 # p.adjust(Panova)
 Panova_ABS <- Panova
 padj_abs <- p.adjust(Panova, method = "holm")
 padj_abs <- p.adjust(Panova)
 

```



### Paperplot plot them
```{r fig.height=17, fig.width=15}
neon <- c("#2f89fc") ; labelcolour <- c("#fc0202");
labelcolour <- "#fc0202"; plotlabels = T
linecolour <- c("#fc0202"); showR = FALSE
PDcolours <- c("#434e52", "#5b8c85" ) ; ndigits <- 2
legendsize <- 15 ; labelsize <- 6; ygrob <- 0.91; neonsize <- 7; topn <- 15
if(showR == TRUE){xgrob <- 0.85}else{xgrob <- 0.8}; abb_digits = T; dgts = 1; abbnum = 3; alphnumb = 0.62+0.1
lefttitle <- "Absolute community abundance" ; righttitle <- "Proportional community abundance"
legendsize <- 18; pointsize = 9; grobsize <- 15; abbrev <- F; plotobjects <- TRUE; plotobjects2 <- F
pos <- position_jitter(width = 0.0, seed = 2)
saveplots()

xgrobb = 0.65


# Go through all the selected variables in the model
 Radjs <- c(); CheckRadjs <- c(); Panova <- c()
 
 
 # Define data frames for top10
 ## Make data frame of top 10 genera
Top10phylobj <- aggregate_top_taxa(phylobj.QMP.prop, 10, "Genus"); Top10phylobj <- subset_samples(Top10phylobj, Ninetransit == 3)
a <- as.data.frame(vegan_otu(Top10phylobj))
b <- as.data.frame(sample_data(Top10phylobj))
df.Top10phylobj <- cbind(Samplenumber = b$Samplenumber,a, Day = b$Day, 
        Transit=b$Transit, Donor = b$Donor, PD_Colon =b$PD_Colon, ivTransit = b$ivTransit)
cols.df <- ncol(df.Top10phylobj)-5
df.Top10phylobj.genus <- df.Top10phylobj[2:cols.df]
genus_table <- df.Top10phylobj.genus %>%
   mutate_all(as.numeric)

sampleinfo10 <- subset(df.Top10phylobj, select = c(Day, Donor, Transit, ivTransit, PD_Colon))


# Develop correct model
## Compute dissimilarity matrix
 rdaplot10 <- genus_table; df.sub10 <- subset(genus_table, select = c(-Other))
 
 
# SHIME transit time
## to plot original species on graph: capscale function!!!
 dbrdail <- dbrda(df.sub~ Transit + Condition(PD_Colon) + Condition(ivTransit), sampleinfo, dist="bray")
 summary(dbrdail)
 coef(dbrdail)
 plot(dbrdail)
 
## permutation test of the results
 anova <- anova.cca(dbrdail,step=1000)
 Panova <- c(Panova, "Transit"= anova[1,4])
 
 # cores <- makeCluster(4) # Define amount of clusters
 # anova.cca(dbrdail,by="axis",step=1000, parallel = cores)
 # stopCluster(cores)
## explanatory power of the variables included
 Runadj <- RsquareAdj(dbrdail)$r.squared
 Radj <- RsquareAdj(dbrdail)$adj.r.squared
 Radjs <- c(Radjs, "Transit" = Radj)
 
## Perform RDA with pcoascores to check if its the same
 rdafunctR <- rda(pcoascores~Transit + Condition(PD_Colon) + Condition(ivTransit), sampleinfo)
 rdafunct <- capscale(df.sub~ Transit + Condition(PD_Colon) + Condition(ivTransit), sampleinfo, dist="bray")
 # summary(rdafunct); coef(rdafunct)
 
## explanatory power of the variables included
 Runadj <- RsquareAdj(rdafunctR)$r.squared
 Radj <- RsquareAdj(rdafunctR)$adj.r.squared
 CheckRadjs <- c(CheckRadjs, "Transit" = Radj)
 
 
 ## As label on plot
Radj2 <- signif(Radj, digits = ndigits)
  grob <- grobTree(textGrob(paste("R\u00B2= ",Radj2), x=xgrob,  y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))

## labelling axes
if(abb_digits==T){
 Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,dgts)
 Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,dgts) 
 }else{
   if(round(summary(rdafunct)$cont$importance[2,1]*100,0)==0){
     Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,1)
     }else{
       Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,0)
       }
   if(round(summary(rdafunct)$cont$importance[2,2]*100,0)==0){
     Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,1)
     }else{
       Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,0)
     }
 }
## plot SCALING2
if(plotobjects2 == TRUE){
  plot(rdafunct,scaling=2,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","wa","cn"))
  plot(rdafunct,scaling=1,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","lc","cn"))
}
 rdafunct10 <- capscale(df.sub10~ Transit + Condition(PD_Colon) + Condition(ivTransit), sampleinfo10, dist="bray")
 
## ggplot
rdascores <- scores(rdafunct10, choices = 1:2, scaling = 2, display = "sites")
firstaxispoints <- rdascores[, 1]; secondaxispoints <- rdascores[, 2]
points <- data.frame(fa = firstaxispoints, sa = secondaxispoints, Donor = sampleinfo10$Donor, 
    P_D = sampleinfo10$PD_Colon, Transit = sampleinfo10$Transit, ivTransit = sampleinfo10$ivTransit)
points$P_D <- mapvalues(points$P_D, from = c("P", "D"), to = c("Proximal", "Distal"))

rdascorevfa <- data.frame(scores(rdafunct10, choices = 1:2, scaling = 2, display = "species"))
rdafactor <- data.frame(summary(rdafunct10)$centroids)
rownames(rdafactor) <- mapvalues(rownames(rdafactor), from = c("TransitST", "TransitMT", "TransitLT"), 
                                 to = c("Short", "Medium", "Long"))
plotobj <- ggplot(as.data.frame(points)) + geom_point(aes(x = fa, y = sa, size = pointsize, color = Transit), alpha = 0.8) +
  scale_colour_manual(values = transitcolors, name = "SHIME Transit", labels = c("Short", "Medium", "Long")) + 
  xlab(paste("RDA1", Dim1, "%", sep = " ")) + 
    ylab(paste("RDA2", Dim2, "%", sep = " ")) + annotation_custom(grob) + 
    geom_segment(data = rdascorevfa, aes(x = 0, xend = CAP1, y = 0, yend = CAP2), 
        color = linecolour) + 
      ggrepel::geom_label_repel(data = rdafactor, aes(x = CAP1, y = CAP2),
                              segment.color = "grey50", alpha = alphnumb+.15, 
                                         # fill = "grey", alpha = 0.5,
    color = neon, label = rownames(rdafactor), size = neonsize)
if(plotlabels == T){
if(abbrev == TRUE){
  plotobj <- plotobj + geom_label_repel(data = rdascorevfa,  aes(x = CAP1, y = CAP2, 
                                          label = abbreviate(colnames(subset(a, select = c(-Other))),abbnum),
                                          segment.color = "grey50"
                                          # vjust = ifelse(RDA2 >= 0, -0.1, 0.7), 
                                          # hjust = ifelse(RDA1 >= 0, 0, 0.5)
                                          ), 
                                          color = linecolour, alpha = alphnumb, label.size = NA,
                                          size = labelsize, fontface = "italic",
                                          position = pos)
} else { 
      plotobj <- plotobj + geom_label_repel(data = rdascorevfa,  aes(x = CAP1, y = CAP2, 
                                          label = colnames(subset(a, select = c(-Other))),
                                          segment.color = "grey50"
                                          # vjust = ifelse(RDA2 >= 0, -0.1, 0.7), 
                                          # hjust = ifelse(RDA1 >= 0, 0, 0.5)
                                          ), 
                                          color = linecolour, alpha = alphnumb, label.size = NA,
                                          size = labelsize, fontface = "italic",
                                          position = pos)
      }}
plotobj <- papertheme(plotobj)
plotobj <- plotobj + guides(color = guide_legend(ncol = 1, direction = "horizontal",
      title.position = "top", override.aes = list(size = 5)), size = "none") + theme(legend.position = "right")
mylegendPDfunctnet <- g_legend(plotobj)
plotobj <- plotobj + theme(legend.position = "none"           , 
        legend.text = element_text(size=legendsize,family='sans',angle=0),
        legend.title = element_text(size=legendsize,family='sans',angle=0),
        legend.direction = "horizontal") 
plotobjrdatop15PDE <- plotobj + xlim(min(rdascorevfa)-0.5, max(points$fa + 0.5)) + 
  # ggtitle(lefttitle) + 
  theme(plot.title = element_text(hjust = 0.5, face="bold"))
if(plotobjects == TRUE){plotobjrdatop15PDE}


 
 
 


# Donor
## to plot original species on graph: capscale function!!!
 dbrdail <- dbrda(df.sub~ Donor+ Condition(PD_Colon) + Condition(Transit), sampleinfo, dist="bray")
 summary(dbrdail)
 coef(dbrdail)
 plot(dbrdail)
 
## permutation test of the results
 anova <- anova.cca(dbrdail,step=1000)
 Panova <- c(Panova, "Donor"= anova[1,4])
 
 # cores <- makeCluster(4) # Define amount of clusters
 # anova.cca(dbrdail,by="axis",step=1000, parallel = cores)
 # stopCluster(cores)
## explanatory power of the variables included
 Runadj <- RsquareAdj(dbrdail)$r.squared
 Radj <- RsquareAdj(dbrdail)$adj.r.squared
 Radjs <- c(Radjs, "Donor" = Radj)
 
## Perform RDA with pcoascores to check if its the same
 rdafunctR <- rda(pcoascores~Donor + Condition(PD_Colon) + Condition(Transit), sampleinfo)
 rdafunct <- capscale(df.sub~ Donor + Condition(PD_Colon) + Condition(Transit), sampleinfo, dist="bray")
 # summary(rdafunct); coef(rdafunct)
 
## explanatory power of the variables included
 Runadj <- RsquareAdj(rdafunctR)$r.squared
 Radj <- RsquareAdj(rdafunctR)$adj.r.squared
 CheckRadjs <- c(CheckRadjs, "Donor" = Radj)
 
 
 ## As label on plot
Radj2 <- signif(Radj, digits = ndigits)
  grob <- grobTree(textGrob(paste("R\u00B2= ",Radj2), x=xgrob,  y=ygrob+0.05, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))

## labelling axes
if(abb_digits==T){
 Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,dgts)
 Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,dgts) 
 }else{
   if(round(summary(rdafunct)$cont$importance[2,1]*100,0)==0){
     Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,1)
     }else{
       Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,0)
       }
   if(round(summary(rdafunct)$cont$importance[2,2]*100,0)==0){
     Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,1)
     }else{
       Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,0)
     }
 }
## plot SCALING2
if(plotobjects2 == TRUE){
  plot(rdafunct,scaling=2,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","wa","cn"))
  plot(rdafunct,scaling=1,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","lc","cn"))
}
 rdafunct10 <- capscale(df.sub10~ Donor + Condition(PD_Colon) + Condition(Transit), sampleinfo10, dist="bray")
 
## ggplot
rdascores <- scores(rdafunct10, choices = 1:2, scaling = 2, display = "sites")
firstaxispoints <- rdascores[, 1]; secondaxispoints <- rdascores[, 2]
points <- data.frame(fa = firstaxispoints, sa = secondaxispoints, Donor = sampleinfo10$Donor, 
    P_D = sampleinfo10$PD_Colon, Transit = sampleinfo10$Transit, ivTransit = sampleinfo10$ivTransit)
points$P_D <- mapvalues(points$P_D, from = c("P", "D"), to = c("Proximal", "Distal"))

rdascorevfa <- data.frame(scores(rdafunct10, choices = 1:2, scaling = 2, display = "species"))
rdafactor <- data.frame(summary(rdafunct10)$centroids)
rownames(rdafactor) <- mapvalues(rownames(rdafactor), from = c("Donor1","Donor2","Donor3",
                                                               "Donor4","Donor5","Donor6"), 
                                 to = rep(1:6))
plotobj <- ggplot(as.data.frame(points)) + geom_point(aes(x = fa, y = sa, size = pointsize, color = Donor), alpha = 0.8) +
  scale_colour_manual(values = sixdonors, name = "Donor") + xlab(paste("RDA1", Dim1, "%", sep = " ")) + 
    ylab(paste("RDA2", Dim2, "%", sep = " ")) + annotation_custom(grob) + 
    geom_segment(data = rdascorevfa, aes(x = 0, xend = CAP1, y = 0, yend = CAP2), 
        color = linecolour) + 
      ggrepel::geom_label_repel(data = rdafactor, aes(x = CAP1, y = CAP2),
                              segment.color = "grey50", alpha = alphnumb+.15, 
                                         # fill = "grey", alpha = 0.5,
    color = neon, label = rownames(rdafactor), size = neonsize)
if(plotlabels == T){
if(abbrev == TRUE){
  plotobj <- plotobj + geom_label_repel(data = rdascorevfa,  aes(x = CAP1, y = CAP2, 
                                          label = abbreviate(colnames(subset(a, select = c(-Other))),abbnum),
                                          segment.color = "grey50"
                                          # vjust = ifelse(RDA2 >= 0, -0.1, 0.7), 
                                          # hjust = ifelse(RDA1 >= 0, 0, 0.5)
                                          ), 
                                          color = linecolour, alpha = alphnumb, label.size = NA,
                                          size = labelsize, fontface = "italic",
                                          position = pos)
} else { 
      plotobj <- plotobj + geom_label_repel(data = rdascorevfa,  aes(x = CAP1, y = CAP2, 
                                          label = colnames(subset(a, select = c(-Other))),
                                          segment.color = "grey50"
                                          # vjust = ifelse(RDA2 >= 0, -0.1, 0.7), 
                                          # hjust = ifelse(RDA1 >= 0, 0, 0.5)
                                          ), 
                                          color = linecolour, alpha = alphnumb, label.size = NA,
                                          size = labelsize, fontface = "italic",
                                          position = pos)
      }}
plotobj <- papertheme(plotobj)
plotobj <- plotobj + guides(color = guide_legend(ncol = 2, direction = "horizontal",
      title.position = "top", override.aes = list(size = 5)), size = "none") + theme(legend.position = "right")
mylegendPDfunctnet <- g_legend(plotobj)
plotobj <- plotobj + theme(legend.position = "none"           , 
        legend.text = element_text(size=legendsize,family='sans',angle=0),
        legend.title = element_text(size=legendsize,family='sans',angle=0),
        legend.direction = "horizontal") 
plotobjrdatop15PDF <- plotobj + xlim(min(rdascorevfa)-0.5, max(points$fa + 0.5)) + 
  # ggtitle(lefttitle) + 
  theme(plot.title = element_text(hjust = 0.5, face="bold"))
if(plotobjects == TRUE){plotobjrdatop15PDF}




# iv transit time
## to plot original species on graph: capscale function!!!
 dbrdail <- dbrda(df.sub~ ivTransit + Condition(PD_Colon) + Condition(Transit), sampleinfo, dist="bray")
 summary(dbrdail)
 coef(dbrdail)
 plot(dbrdail)
 
## permutation test of the results
 anova <- anova.cca(dbrdail,step=1000)
 Panova <- c(Panova, "ivTransit"= anova[1,4])
 
 # cores <- makeCluster(4) # Define amount of clusters
 # anova.cca(dbrdail,by="axis",step=1000, parallel = cores)
 # stopCluster(cores)
## explanatory power of the variables included
 Runadj <- RsquareAdj(dbrdail)$r.squared
 Radj <- RsquareAdj(dbrdail)$adj.r.squared
 Radjs <- c(Radjs, "ivTransit" = Radj)
 
## Perform RDA with pcoascores to check if its the same
 rdafunctR <- rda(pcoascores~ivTransit + Condition(PD_Colon) + Condition(Transit), sampleinfo)
 rdafunct <- capscale(df.sub~ ivTransit + Condition(PD_Colon) + Condition(Transit), sampleinfo, dist="bray")
 # summary(rdafunct); coef(rdafunct)
 
## explanatory power of the variables included
 Runadj <- RsquareAdj(rdafunctR)$r.squared
 Radj <- RsquareAdj(rdafunctR)$adj.r.squared
 CheckRadjs <- c(CheckRadjs, "ivTransit" = Radj)
 
 
 ## As label on plot
Radj2 <- signif(Radj, digits = ndigits)
  grob <- grobTree(textGrob(paste("R\u00B2= ",Radj2), x=xgrob,  y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))

## labelling axes
if(abb_digits==T){
 Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,dgts)
 Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,dgts) 
 }else{
   if(round(summary(rdafunct)$cont$importance[2,1]*100,0)==0){
     Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,1)
     }else{
       Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,0)
       }
   if(round(summary(rdafunct)$cont$importance[2,2]*100,0)==0){
     Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,1)
     }else{
       Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,0)
     }
 }
## plot SCALING2
if(plotobjects2 == TRUE){
  plot(rdafunct,scaling=2,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","wa","cn"))
  plot(rdafunct,scaling=1,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","lc","cn"))
}
 rdafunct10 <- capscale(df.sub10~ ivTransit + Condition(PD_Colon) + Condition(Transit), sampleinfo10, dist="bray")
 lblInvivo <- "\U0001D43C\U0001D62F \U0001D637\U0001d62a\U0001D637\U0001D630"
## ggplot
rdascores <- scores(rdafunct10, choices = 1:2, scaling = 2, display = "sites")
firstaxispoints <- rdascores[, 1]; secondaxispoints <- rdascores[, 2]
points <- data.frame(fa = firstaxispoints, sa = secondaxispoints, Donor = sampleinfo10$Donor, 
    P_D = sampleinfo10$PD_Colon, Transit = sampleinfo10$Transit, ivTransit = sampleinfo10$ivTransit)
points$P_D <- mapvalues(points$P_D, from = c("P", "D"), to = c("Proximal", "Distal"))

rdascorevfa <- data.frame(scores(rdafunct10, choices = 1:2, scaling = 2, display = "species"))
rdafactor <- data.frame(summary(rdafunct10)$centroids)
rownames(rdafactor) <- mapvalues(rownames(rdafactor), from = c("ivTransitSTD", "ivTransitMTD", "ivTransitLTD"), 
                                 to = c("Short", "Medium", "Long"))
plotobj <- ggplot(as.data.frame(points)) + geom_point(aes(x = fa, y = sa, size = pointsize, color = ivTransit), alpha = 0.8) +
  scale_colour_manual(values = transitcolors2, name = substitute(paste(italic("In vivo"), " transit")), labels = c("Short", "Medium", "Long")) +
  xlab(paste("RDA1", Dim1, "%", sep = " ")) + 
    ylab(paste("RDA2", Dim2, "%", sep = " ")) + annotation_custom(grob) + 
    geom_segment(data = rdascorevfa, aes(x = 0, xend = CAP1, y = 0, yend = CAP2), 
        color = linecolour) + 
      ggrepel::geom_label_repel(data = rdafactor, aes(x = CAP1, y = CAP2),
                              segment.color = "grey50", alpha = alphnumb+.15, 
                                         # fill = "grey", alpha = 0.5,
    color = neon, label = rownames(rdafactor), size = neonsize)
if(plotlabels == T){
if(abbrev == TRUE){
  plotobj <- plotobj + geom_label_repel(data = rdascorevfa,  aes(x = CAP1, y = CAP2, 
                                          label = abbreviate(colnames(subset(a, select = c(-Other))),abbnum),
                                          segment.color = "grey50"
                                          # vjust = ifelse(RDA2 >= 0, -0.1, 0.7), 
                                          # hjust = ifelse(RDA1 >= 0, 0, 0.5)
                                          ), 
                                          color = linecolour, alpha = alphnumb, label.size = NA,
                                          size = labelsize, fontface = "italic",
                                          position = pos)
} else { 
      plotobj <- plotobj + geom_label_repel(data = rdascorevfa,  aes(x = CAP1, y = CAP2, 
                                          label = colnames(subset(a, select = c(-Other))),
                                          segment.color = "grey50"
                                          # vjust = ifelse(RDA2 >= 0, -0.1, 0.7), 
                                          # hjust = ifelse(RDA1 >= 0, 0, 0.5)
                                          ), 
                                          color = linecolour, alpha = alphnumb, label.size = NA,
                                          size = labelsize, fontface = "italic",
                                          position = pos)
      }}
plotobj <- papertheme(plotobj)
plotobj <- plotobj + guides(color = guide_legend(ncol = 1, direction = "horizontal",
      title.position = "top", override.aes = list(size = 5, fill=NA)), size = "none") + theme(legend.position = "right")
mylegendPDfunctnet <- g_legend(plotobj)
plotobj <- plotobj + theme(legend.position = "none"           , 
        legend.text = element_text(size=legendsize,family='sans',angle=0),
        legend.title = element_text(size=legendsize,family='sans',angle=0),
        legend.direction = "horizontal") 
plotobjrdatop15PDG <- plotobj + xlim(min(rdascorevfa)-0.5, max(points$fa + 0.5)) + 
  # ggtitle(lefttitle) + 
  theme(plot.title = element_text(hjust = 0.5, face="bold"))
if(plotobjects == TRUE){plotobjrdatop15PDG}



# Vessel
## to plot original species on graph: capscale function!!!
 dbrdail <- dbrda(df.sub~ PD_Colon + Condition(Transit) + Condition(ivTransit), sampleinfo, dist="bray")
 summary(dbrdail)
 coef(dbrdail)
 plot(dbrdail)
 
## permutation test of the results
 anova <- anova.cca(dbrdail,step=1000)
 Panova <- c(Panova, "PD"= anova[1,4])
 
 # cores <- makeCluster(4) # Define amount of clusters
 # anova.cca(dbrdail,by="axis",step=1000, parallel = cores)
 # stopCluster(cores)
## explanatory power of the variables included
 Runadj <- RsquareAdj(dbrdail)$r.squared
 Radj <- RsquareAdj(dbrdail)$adj.r.squared
 Radjs <- c(Radjs, "PD" = Radj)
 
## Perform RDA with pcoascores to check if its the same
 rdafunctR <- rda(pcoascores~PD_Colon + Condition(Transit) + Condition(ivTransit), sampleinfo)
 rdafunct <- capscale(df.sub~ PD_Colon + Condition(Transit) + Condition(ivTransit), sampleinfo, dist="bray")
 # summary(rdafunct); coef(rdafunct)
 
## explanatory power of the variables included
 Runadj <- RsquareAdj(rdafunctR)$r.squared
 Radj <- RsquareAdj(rdafunctR)$adj.r.squared
 CheckRadjs <- c(CheckRadjs, "PD" = Radj)
 
 
 ## As label on plot
Radj2 <- signif(Radj, digits = ndigits)
  grob <- grobTree(textGrob(paste("R\u00B2= ",Radj2), x=xgrob,  y=ygrob, hjust=0,  gp=gpar(col="black", fontsize=grobsize, fontface="italic")))

## labelling axes
if(abb_digits==T){
 Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,dgts)
 Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,dgts) 
 }else{
   if(round(summary(rdafunct)$cont$importance[2,1]*100,0)==0){
     Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,1)
     }else{
       Dim1 <- round(summary(rdafunct)$cont$importance[2,1]*100,0)
       }
   if(round(summary(rdafunct)$cont$importance[2,2]*100,0)==0){
     Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,1)
     }else{
       Dim2 <- round(summary(rdafunct)$cont$importance[2,2]*100,0)
     }
 }
## plot SCALING2
if(plotobjects2 == TRUE){
  plot(rdafunct,scaling=2,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","wa","cn"))
  plot(rdafunct,scaling=1,main="Triplot RDA datanormalised_subset ~ functional -scaling 1 ",type = "text",display=c("sp","lc","cn"))
}
 rdafunct10 <- capscale(df.sub10~ PD_Colon + Condition(Transit) + Condition(ivTransit), sampleinfo10, dist="bray")
 
## ggplot
rdascores <- scores(rdafunct10, choices = 1:2, scaling = 2, display = "sites")
firstaxispoints <- rdascores[, 1]; secondaxispoints <- rdascores[, 2]
points <- data.frame(fa = firstaxispoints, sa = secondaxispoints, Donor = sampleinfo10$Donor, 
    P_D = sampleinfo10$PD_Colon, Transit = sampleinfo10$Transit, ivTransit = sampleinfo10$ivTransit)
points$P_D <- mapvalues(points$P_D, from = c("P", "D"), to = c("Proximal", "Distal"))

rdascorevfa <- data.frame(scores(rdafunct10, choices = 1:2, scaling = 2, display = "species"))
rdafactor <- data.frame(summary(rdafunct10)$centroids)
rownames(rdafactor) <- mapvalues(rownames(rdafactor), from = c("PD_ColonD", "PD_ColonP"), 
                                 to = c("Distal", "Proximal"))
plotobj <- ggplot(as.data.frame(points)) + geom_point(aes(x = fa, y = sa, size = pointsize, color = P_D), alpha = 0.8) +
  scale_colour_manual(values = c("Proximal" = "#cdb174", "Distal" = "#a48d5c"), name = "Vessel") + 
  xlab(paste("RDA1", Dim1, "%", sep = " ")) + 
    ylab(paste("RDA2", Dim2, "%", sep = " ")) + annotation_custom(grob) + 
    geom_segment(data = rdascorevfa, aes(x = 0, xend = CAP1, y = 0, yend = MDS1), 
        color = linecolour) + 
      ggrepel::geom_label_repel(data = rdafactor, aes(x = CAP1, y = MDS1),
                              segment.color = "grey50", alpha = alphnumb+.15, 
                                         # fill = "grey", alpha = 0.5,
    color = neon, label = rownames(rdafactor), size = neonsize)
if(plotlabels == T){
if(abbrev == TRUE){
  plotobj <- plotobj + geom_label_repel(data = rdascorevfa,  aes(x = CAP1, y = MDS1, 
                                          label = abbreviate(colnames(subset(a, select = c(-Other))),abbnum),
                                          segment.color = "grey50"
                                          # vjust = ifelse(RDA2 >= 0, -0.1, 0.7), 
                                          # hjust = ifelse(RDA1 >= 0, 0, 0.5)
                                          ), 
                                          color = linecolour, alpha = alphnumb, label.size = NA,
                                          size = labelsize, fontface = "italic",
                                          position = pos)
} else { 
      plotobj <- plotobj + geom_label_repel(data = rdascorevfa,  aes(x = CAP1, y = MDS1, 
                                          label = colnames(subset(a, select = c(-Other))),
                                          segment.color = "grey50"
                                          # vjust = ifelse(RDA2 >= 0, -0.1, 0.7), 
                                          # hjust = ifelse(RDA1 >= 0, 0, 0.5)
                                          ), 
                                          color = linecolour, alpha = alphnumb, label.size = NA,
                                          size = labelsize, fontface = "italic",
                                          position = pos)
      }}
plotobj <- papertheme(plotobj)
plotobj <- plotobj + guides(color = guide_legend(ncol = 1, direction = "horizontal",
      title.position = "top", override.aes = list(size = 5)), size = "none") + theme(legend.position = "right")
mylegendPDfunctnet <- g_legend(plotobj)
plotobj <- plotobj + theme(legend.position = "none"           , 
        legend.text = element_text(size=legendsize,family='sans',angle=0),
        legend.title = element_text(size=legendsize,family='sans',angle=0),
        legend.direction = "horizontal") 
plotobjrdatop15PDH <- plotobj + xlim(min(rdascorevfa)-0.5, max(points$fa + 0.5)) + 
  # ggtitle(lefttitle) + 
  theme(plot.title = element_text(hjust = 0.5, face="bold"))
if(plotobjects == TRUE){plotobjrdatop15PDH}




 
 

ggarrange(plotobjrdatop15PDE + theme(legend.position = c(0.81,0.1)), 
          plotobjrdatop15PDF + theme(legend.position = c(0.15,0.1)),
          plotobjrdatop15PDG + theme(legend.position = c(0.15,0.1)), 
          plotobjrdatop15PDH + theme(legend.position = c(0.15,0.1)),nrow = 2, ncol = 2)
 
 
 
 
# Adjust p-values because of multiple testing
 # Correction for multiple testing (BenjaminiâHochberg procedure, FDR) 
 # Panova
 # p.adjust(Panova)
 Panova_ABS <- Panova
 padj_abs <- p.adjust(Panova, method = "holm")
 padj_abs <- p.adjust(Panova)
 

```

# Figure S9 
### Combine paperplots
```{r fig.height=21.5, fig.width=15}
ggarrange(plotobjrdatop15PDA + ggtitle("Quantitative microbiome composition") , 
          plotobjrdatop15PDE + theme(legend.position = c(0.15,0.1))+ ggtitle("Proportional microbiome composition"),
          plotobjrdatop15PDB , 
          plotobjrdatop15PDF + theme(legend.position = c(0.15,0.9)),nrow = 2, ncol = 2, 
          labels = c("A", "B", "C", "D"),font.label = list(size = 20))


ggarrange(plotobjrdatop15PDC+ggtitle("Quantitative microbiome composition"), 
          plotobjrdatop15PDG + theme(legend.position = c(0.15,0.1))+ ggtitle("Proportional microbiome composition"),
          plotobjrdatop15PDD, 
          plotobjrdatop15PDH + theme(legend.position = c(0.15,0.9)),nrow = 2, ncol = 2, 
          labels = c("E", "F", "G", "H"), font.label = list(size = 20))




ggarrange(plotobjrdatop15PDA + ggtitle("Quantitative microbiome composition") , 
          plotobjrdatop15PDE + theme(legend.position = c(0.14,0.82))+ ggtitle("Proportional microbiome composition"),
          plotobjrdatop15PDB , 
          plotobjrdatop15PDF + theme(legend.position = c(0.13,0.84)),
          plotobjrdatop15PDC, 
          plotobjrdatop15PDG + theme(legend.position = c(0.13,0.84)),
          plotobjrdatop15PDD, 
          plotobjrdatop15PDH + theme(legend.position = c(0.14,0.84)),nrow = 4, ncol = 2, 
          labels = "AUTO",font.label = list(size = 20))



```



# Splines tryout
```{r}
library(splines)
library(Ecdat)

data(Clothing)
head(Clothing)


model <- lm(tsales~bs(inv2, knots = c(12,20)), data = Clothing)
summary(model)

inv2lims <- range(Clothing$inv2)
inv2.grid <- seq(from = inv2lims[1], to = inv2lims[2])
pred <- predict(model, newdata = list(inv2 = inv2.grid), se = T)

# GENUS LEVEL
# Absolute
phylobj.QMP_stable <- subset_samples(phylobj.QMP, Ninetransit == 3)
Top15phylobj <- aggregate_taxa(phylobj.QMP_stable, level = "Genus")
Top15 <- psmelt(Top15phylobj); unique(Top15$OTU); unique(Top15$OTU) %in% Top15levels
## Subset according to top most abundant genera
df.test <- data.frame();
for(i in Top.genera){
  df.test2 <- subset(Top15, Genus == i)
  df.test <- rbind(df.test, df.test2)
}
Top15 <- df.test
x <- data.frame(x = unique(Top15$OTU) %in% Top15levels, genus = unique(Top15$OTU))  #check if labelled
Top15$OTU <- as.ordered(factor(Top15$OTU, levels = Top15levels))
## Add extra column days for graphs
Top <- Top15
Top$Transit <- as.ordered(factor(Top$Transit, levels = c("ST", "MT", "LT")))
Top$PD_Colon <- as.ordered(factor(Top$PD_Colon, levels = c("P", "D", "FS")))
Top$Donor <- as.ordered(factor(Top$Donor, levels = c("1", "2", "3", "4", "5", "6")))
Top$Genus <- as.ordered(factor(Top$Genus, levels = Top15levels))
## Add numbers
df.1 <- subset(Top, PD_Colon == "P" & Transit == "ST"); df.1$TT <- 8; df.1$TT1 <- 8
df.2 <- subset(Top, PD_Colon == "P" & Transit == "MT"); df.2$TT <- 16; df.2$TT1 <- 16
df.3 <- subset(Top, PD_Colon == "P" & Transit == "LT"); df.3$TT <- 24; df.3$TT1 <- 24
df.4 <- subset(Top, PD_Colon == "D" & Transit == "ST"); df.4$TT <- 13; df.4$TT1 <- 8
df.5 <- subset(Top, PD_Colon == "D" & Transit == "MT"); df.5$TT <- 26; df.5$TT1 <- 16
df.6 <- subset(Top, PD_Colon == "D" & Transit == "LT"); df.6$TT <- 39; df.6$TT1 <- 24
Top <- rbind(df.1, df.2, df.3, df.4, df.5, df.6)
rm(df.1, df.2, df.3, df.4, df.5, df.6)
## Perform test
dfcor <- data.frame(); df.otutest <- data.frame(); QMPlist <- list(); y =1

  df1 <- subset(Top, Genus == "Bacteroides");  
  if(abovezero == T){
    df1.1 <- subset(df1, Donor == 1); df1.1 <- subset(df1.1, Abundance > 0)
    if (dim(df1.1)[1] == 0) {df1 <- subset(df1, Donor !=1)}
    df1.2 <- subset(df1, Donor == 2); df1.2 <- subset(df1.2, Abundance > 0) 
    if (dim(df1.2)[1] == 0) {df1 <- subset(df1, Donor !=2)}
    df1.3 <- subset(df1, Donor == 3); df1.3 <- subset(df1.3, Abundance > 0)
    if (dim(df1.3)[1] == 0) {df1 <- subset(df1, Donor !=3)}
    df1.4 <- subset(df1, Donor == 4); df1.4 <- subset(df1.4, Abundance > 0) 
    if (dim(df1.4)[1] == 0) {df1 <- subset(df1, Donor !=4)}
    df1.5 <- subset(df1, Donor == 5); df1.5 <- subset(df1.5, Abundance > 0) 
    if (dim(df1.5)[1] == 0) {df1 <- subset(df1, Donor !=5)}
    df1.6 <- subset(df1, Donor == 6); df1.6 <- subset(df1.6, Abundance > 0) 
    if (dim(df1.6)[1] == 0) {df1 <- subset(df1, Donor !=6)}
    }

  df<- subset(df1, PD_Colon == "P")
df<- df1

  BP02gam1 = mgcv::gam(TT~Abundance, data = df)
summary(BP02gam1)
  
  
model <- lm(Abundance~bs(as.numeric(TT), knots = c(12,20)), data = df)
summary(model)

model <- lm(Abundance~TT, data = df)
plot(model, which = 1)
lm_y <- lm(y ~ x, data = Sample_data)


# GAM
# To run a GAM, we use:
df <- df1
gam_y <- gam(df$Abundance ~ s(df$TT, k = 2), method = "REML")

# To extract the fitted values, we can use predict just like normal:
x_new <- seq(min(df$TT), max(df$TT), length.out = 3); x_new <- c(8,13,16,24,26,39)
y_pred <- predict(gam_y, data.frame(x = x_new))

ggplot(df, aes(TT, Abundance)) + geom_point() + geom_smooth(method = "gam", formula = y ~s(x, k = 5))


par(mfrow = c(2,2))
gam.check(gam_y)
summary(gam_y)
summary.gam(gam_y)
plot(gam_y)

```


# Quality control
```{r Quality control objects, fig.height=18, fig.width=18}
sharedfile = "QC.opti_mcc.shared"
taxfile = "QC.opti_mcc.0.03.cons.taxonomy"
display.brewer.pal(n = 25, name = 'Dark2')
sevencolors = brewer.pal(n = 7, name = "Set2")
getpallette = colorRampPalette(brewer.pal(n = 40, name = "Set3"))
fortycolors = getpallette(40)

# Import mothur data
QC_mothur_data <- import_mothur(mothur_shared_file = sharedfile,
                             mothur_constaxonomy_file = taxfile)
# To see if the samples are correct 
# sample_names(QC_mothur_data) <- c("BL1" = "BL_1", "Bl_01_pl20" = "Bl_2", "Bl_02_pl19" = "Bl_3",
#         "Bl_02_pl20" = "Bl_4", "Bl_03_pl19" = "Bl_5", "Bl_03_pl20" = "Bl_6", "Bl_04_pl19" = "Bl_7", "Bl_04_pl20" = "Bl_8", 
#         "NC_pl19"="NC_1", "NC_pl20"="NC_2", "PC_pl19"="PC_1", "PC_pl20"="PC_2", "RS_pl19"="RS_1", "RS_pl20"="RS_2")
sample_names(QC_mothur_data) 
# Rename the comumns to reflect tax
colnames(tax_table(QC_mothur_data)) <- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus")
  ## Rename certain genera in taxonomy of phyloseq 
  tax.tab1 <- data.frame(tax_table(QC_mothur_data))
  tax.tab1$Genus <- as.character(tax.tab1$Genus)
  tax.tab1$Genus[tax.tab1$Genus == "Lachnospiraceae_ge"] <- "Ruminococcus"
  tax.tab1$Genus[tax.tab1$Genus == "Ruminococcus_1"] <- "Ruminococcus"
  tax.tab1$Genus[tax.tab1$Genus == "Ruminococcus_2"] <- "Ruminococcus"
  tax.tab1$Genus[tax.tab1$Genus == "Coprococcus_1"] <- "Coprococcus"
  tax.tab1$Genus[tax.tab1$Genus == "Coprococcus_2"] <- "Coprococcus"
  tax.tab1$Genus[tax.tab1$Genus == "Lachnospira"] <- "Lachnospiraceae"
  tax.tab1$Genus[tax.tab1$Genus == "Lachnospiraceae_NK4A136_group"] <- "Lachnospiraceae"
  tax.tab1$Genus[tax.tab1$Genus == "Lachnospiraceae_unclassified"] <- "Lachnospiraceae"
  tax.tab1$Genus[tax.tab1$Genus == "Ruminococcaceae_ge"] <- "Ruminococcaceae"
  tax.tab1$Genus[tax.tab1$Genus == "Ruminococcaceae_UCG-002"] <- "Ruminococcaceae"
  tax.tab1$Genus[tax.tab1$Genus == "Ruminococcaceae_unclassified"] <- "Ruminococcaceae"
  tax.tab1$Genus[tax.tab1$Genus == "Clostridia_unclassified"] <- "Clostridia"
  tax.tab1$Genus[tax.tab1$Genus == "Clostridiales_unclassified"] <- "Clostridiales"
  tax.tab1$Genus[tax.tab1$Genus == "Erysipelotrichaceae_UCG-003"] <- "Erysipelotrichaceae"
  tax.tab1$Genus[tax.tab1$Genus == "Christensenellaceae_R-7_group"] <- "Christensenellaceae"
  tax.tab1$Genus <- as.factor(tax.tab1$Genus)
  tax.tab1 <- tax_table(as.matrix(tax.tab1))
  tax_table(QC_mothur_data) <- tax.tab1
  
  QCcolors = c("Akkermansia" = "#1a1259", "Alistipes" = "#CD9BCD", "Bacteroides"="#3194cd", "Cupriavidus" ="#233938",
             "Bilophila"="#0d654d", "Blautia"="#149700", "Enterobacteriaceae" = "#85cb4f", "Lachnoclostridium"="#63415a", "Lachnospiraceae"="#937777", "Clostridiales"="#503365", "Mitsuokella"="#874444", "Other"="#bfbfbf", "Parabacteroides"= "#874444", "Phascolarctobacterium"="#d86d36", "Pseudomonas" ="#883488", "Ruminococcus"="#fbaa22", "Agathobacter" = "#8DD3C7", "Anaerostipes" = "#CFECBB", "Barnesiella" = "#F4F4B9", "Butyricicoccus" = "#CFCCCF", "Christensenellaceae" = "#D1A7B9",
"Clostridia" = "#F4867C", "Coprococcus" = "#C0979F", "Dorea" = "#86B1CD" , "Erysipelotrichaceae" = "#CEB28B", "Faecalibacterium" =  "#EDBC63", "Fusicatenibacter" = "#C2D567", "Roseburia" =  "#CDD796" , "Ruminococcaceae" =   "#d45d79", "Runella" =   "#E9D3DE", "Subdoligranulum" = "#445c3c")
  
  QCcolors = c("Akkermansia" = "#1a1259", "Alistipes" = "#CD9BCD", "Bacteroides"="#3194cd", "Cupriavidus" ="#233938",
             "Blautia"="#149700", "Lachnoclostridium"="#63415a", "Lachnospiraceae"="#937777", "Clostridiales"="#503365", "Other"="#bfbfbf", "Phascolarctobacterium"="#d86d36",  "Ruminococcus"="#fbaa22", "Agathobacter" = "#8DD3C7", "Anaerostipes" = "#CFECBB",  "Butyricicoccus" = "#CFCCCF", "Christensenellaceae" = "#D1A7B9",
"Clostridia" = "#F4867C", "Coprococcus" = "#C0979F", "Dorea" = "#86B1CD" , "Erysipelotrichaceae" = "#CEB28B", "Faecalibacterium" =  "#EDBC63", "Fusicatenibacter" = "#C2D567", "Ruminococcaceae" =   "#d45d79", "Runella" =   "#E9D3DE", "Subdoligranulum" = "#445c3c")
  
  QCcolors2 <- sort(QCcolors)
  

  
  
# Proportional composition at genus level
QC.prop <- aggregate_top_taxa(QC_mothur_data, top= 25, "Genus")
QC.prop <- transform_sample_counts(QC.prop, function(x) 100 * x/sum(x))
QC.df <- psmelt(QC.prop)
QC.df <- subset(QC.df, Sample != "BL4")
# Order parameters
# QC.df$Sample <- as.ordered(factor(QC.df$Sample, levels=c("BL1", "BL3", "BL4", "BL5", "BL6", "BL7","BL8", "BL9", "BL10", "BL11", "BL12",
#                                                          "BL13", "NC1", "PC1", "PC2", "PC3", "PC4", "RS1", "RS2", "RS3"))) 
QC.df$Sample <- as.ordered(factor(QC.df$Sample, levels=c("BL1", "BL3", "BL4", "BL5", "BL6", "BL7","BL8", "BL9", "BL10", "BL11", "BL12",
                                                         "BL13", "NC1", "PC1", "PC2", "PC3", "PC4", "RS1", "RS2", "RS3"))) 

 
levels <- c("Agathobacter",	"Akkermansia",	"Alistipes",	"Anaerostipes",	"Bacteroides",	"Barnesiella",	"Blautia",	"Butyricicoccus",	"Christensenellaceae",	"Clostridia", "Clostridiales",	"Coprococcus", "Cupriavidus", 	"Dorea",	"Erysipelotrichaceae",	"Faecalibacterium",	"Fusicatenibacter",	"Lachnoclostridium",	"Lachnospiraceae",	"Phascolarctobacterium",	"Pseudomonas",	"Roseburia",	"Ruminococcaceae",	"Ruminococcus",	"Runella",	"Subdoligranulum",	"Other")
unique(QC.df$Genus)%in%levels; x <- data.frame(x = unique(QC.df$Genus)%in%levels, y = unique(QC.df$Genus))
QC.df$OTU <- as.ordered(factor(QC.df$OTU,levels=levels))
QC.df$Genus <- as.ordered(factor(QC.df$Genus,levels=levels))

plot <- ggplot(data = QC.df, aes(x = Sample, y = as.numeric(Abundance), fill = factor(Genus))) +
  geom_bar(position="stack", stat="identity") +  coord_flip() + xlab("Sample") + ylab("Relative abundance (%)") +
  theme(legend.position = "bottom") +
  scale_fill_manual("Genus", values = QCcolors)
plot <- papertheme(plot, 20)
plot1 <- plot + theme(legend.text = element_text(face = 'italic', size = 16), legend.position = "bottom") 
plot1
saveplots()
ggsave(plot = plot, width = 10, height = 25, dpi = 100, filename = "Paperplot.QC.prop.png")



plot <- ggplot(data = QC.df, aes(y = Sample, x = as.numeric(Abundance), fill = factor(Genus))) +
  geom_bar(position="stack", stat="identity") +  coord_flip() + ylab("Sample") + xlab("Proportional abundance (%)") +
  theme(legend.position = "bottom") +
  scale_fill_manual("Genus", values = QCcolors)
plot <- papertheme(plot, 20)
plot1b <- plot + theme(legend.text = element_text(face = 'italic', size = 16), legend.position = "bottom") 
plot1b

# With reads
# Proportional composition at genus level
QC.prop <- aggregate_top_taxa(QC_mothur_data, top= 25, "Genus")
# QC.prop <- transform_sample_counts(QC.prop, function(x) 100 * x/sum(x))
QC.df <- psmelt(QC.prop)
QC.df <- subset(QC.df, Sample != "BL4")
# Order parameters
# QC.df$Sample <- as.ordered(factor(QC.df$Sample, levels=c("BL1", "BL3", "BL4", "BL5", "BL6", "BL7","BL8", "BL9", "BL10", "BL11", "BL12",
#                                                          "BL13", "NC1", "PC1", "PC2", "PC3", "PC4", "RS1", "RS2", "RS3"))) 
QC.df$Sample <- as.ordered(factor(QC.df$Sample, levels=rev(c("BL1", "BL3", "BL4", "BL5", "BL6", "BL7","BL8", "BL9", "BL10", "BL11", "BL12",
                                                         "BL13", "NC1", "PC1", "PC2", "PC3", "PC4", "RS1", "RS2", "RS3")))) 
 
levels <- c("Agathobacter",	"Akkermansia",	"Alistipes",	"Anaerostipes",	"Bacteroides",	"Barnesiella",	"Blautia",	"Butyricicoccus",	"Christensenellaceae",	"Clostridia", "Clostridiales",	"Coprococcus", "Cupriavidus", 	"Dorea",	"Erysipelotrichaceae",	"Faecalibacterium",	"Fusicatenibacter",	"Lachnoclostridium",	"Lachnospiraceae",	"Phascolarctobacterium",	"Pseudomonas",	"Roseburia",	"Ruminococcaceae",	"Ruminococcus",	"Runella",	"Subdoligranulum",	"Other")
unique(QC.df$Genus)%in%levels; x <- data.frame(x = unique(QC.df$Genus)%in%levels, y = unique(QC.df$Genus))
QC.df$OTU <- as.ordered(factor(QC.df$OTU,levels=levels))
QC.df$Genus <- as.ordered(factor(QC.df$Genus,levels=levels))
plot <- ggplot(data = QC.df, aes(x = Sample, y = as.numeric(Abundance), fill = factor(Genus))) +
  geom_bar(position="stack", stat="identity") +  coord_flip() + xlab("Sample") + ylab("Reads abundance (reads)") +
  theme(legend.position = "bottom") + theme(axis.title.y = element_blank(), axis.ticks.y = element_blank()) +
  scale_fill_manual("Genus", values = QCcolors)
plot <- papertheme(plot, 20)
plot2 <- plot + theme(legend.text = element_text(face = 'italic', size = 16), legend.position = "bottom") 
plot2



# Rarefaction
QC_df.OTU <- as.data.frame(otu_table(QC_mothur_data))
QC_df.OTU2 <- QC_df.OTU
QC_df.OTU <- transpose(QC_df.OTU)
QC_df.OTU[] <- lapply(QC_df.OTU, as.numeric)
colnames(QC_df.OTU) <- rownames(QC_df.OTU2)
rownames(QC_df.OTU) <- colnames(QC_df.OTU2)
set.seed=711
datarare <-t(QC_df.OTU)
# datarare <- datarare[,1:12]
datarare <- subset(datarare, select = -c(BL4))
datarare <- datarare[,1:11]

 ## Find minimal total number of reads of all samples = mimimal number of 'species'
 raremax <- min(colSums(datarare))
 
 ## Draw rarefaction curves
plot3 <- rarecurve(round(t(datarare)), step = 20, sample = raremax, col = c("blue"), cex = 0.6)
print("average sample size of blanks: 502.18 +- 575.07, with NC: 471.583 +- 558.456")


# Rarefaction
QC_df.OTU <- as.data.frame(otu_table(QC_mothur_data))
QC_df.OTU2 <- QC_df.OTU
QC_df.OTU <- transpose(QC_df.OTU)
QC_df.OTU[] <- lapply(QC_df.OTU, as.numeric)
colnames(QC_df.OTU) <- rownames(QC_df.OTU2)
rownames(QC_df.OTU) <- colnames(QC_df.OTU2)
set.seed=711
datarare <-t(QC_df.OTU)
# datarare <- datarare[,1:12]
datarare <- subset(datarare, select = -c(BL4))
# datarare <- datarare[,1:11]

 ## Find minimal total number of reads of all samples = mimimal number of 'species'
 raremax <- min(colSums(datarare))
 
 ## Draw rarefaction curves
plot4 <- rarecurve(round(t(datarare)), step = 20, sample = raremax, col = c("blue"), cex = 0.6)
print("average sample size of blanks: 502.18 +- 575.07, with NC: 471.583 +- 558.456")

ggarrange(plot1, plot2, common.legend = T)
# rm(QC.prop, QC_df.OTU2, QCcolors, levels, QC_mothur_data)




df.OTU_QC <- transpose(as.data.frame(otu_table(QC_mothur_data)))
colnames(df.OTU_QC) <- rownames(as.data.frame(otu_table(QC_mothur_data)))
rownames(df.OTU_QC) <- colnames(as.data.frame(otu_table(QC_mothur_data)))
df.Tax_QC <- as.data.frame(tax_table(QC_mothur_data))
# Quality Control samples 
  # Taxa distribution
    readnr <- rowSums(df.OTU_QC)
    OTUnr <- rowSums(df.OTU_QC!=0) 
    genusnr <- colSums(tax_summarize(df.OTU_QC,df.Tax_QC,"Genus")!=0)
    familynr <- colSums(tax_summarize(df.OTU_QC,df.Tax_QC,"Family")!=0)
    ordernr <- colSums(tax_summarize(df.OTU_QC,df.Tax_QC,"Order")!=0)
    classnr <- colSums(tax_summarize(df.OTU_QC,df.Tax_QC,"Class")!=0)
    phylumnr <- colSums(tax_summarize(df.OTU_QC,df.Tax_QC,"Phylum")!=0)
    taxoverview_QC <- data.frame(Sample=rownames(df.OTU_QC),OTU=OTUnr,Genus=genusnr,Family=familynr,Order=ordernr,Class=classnr,Phylum=phylumnr,Reads=readnr)
    taxoverview <- reshape2::melt(taxoverview_QC,id.vars=c("Sample"))
    taxoverview <- subset(taxoverview, Sample != "BL4")
    
    QClevels = c("BL1",  "BL3" ,"BL5", "BL6" , "BL7",  "BL8" , "BL9" , "BL10" ,"BL11" ,"BL12", "BL13"  , "NC1" , "PC1",  "PC2",  "PC3" , "PC4",  "RS1" , "RS2",  "RS3" )
    QClevels = rev(QClevels)
    taxoverview$Sample <- as.ordered(factor(taxoverview$Sample,
                                          levels=QClevels))
    
    
    
    plottaxoverview <- ggplot(taxoverview,aes(x=Sample,y=as.numeric(value))) +
      geom_bar(aes(fill=factor(variable)),stat='identity',position="identity") +
      facet_grid(.~variable, 
                 scales="free",
                 drop=TRUE) + 
      coord_flip()
    plottaxoverview <- papertheme(plottaxoverview)
    plottaxoverview <- plottaxoverview + scale_fill_manual(values=sevencolors)+
      theme(strip.background =element_rect(fill="#D6E0F4")) 
    plottaxoverview <- plottaxoverview + ylab("Number of unique taxa") + xlab("Sample") + theme(legend.title=element_blank(),legend.position="none") + geom_label(data = subset(taxoverview, variable == "Reads"), aes(label = value))
    plottaxoverview
    
    
    plottaxoverview <- ggplot(subset(taxoverview, variable == "OTU") ,aes(x=Sample,y=as.numeric(value))) +
      geom_bar(aes(fill=factor(variable)),stat='identity',position="identity") +
      facet_grid(.~variable, 
                 scales="free",
                 drop=TRUE) + 
      coord_flip()
    plottaxoverview <- papertheme(plottaxoverview)
    plottaxoverview <- plottaxoverview + scale_fill_manual(values=sevencolors[[1]])+
      theme(strip.background =element_rect(fill="#D6E0F4")) 
    plottaxoverview2 <- plottaxoverview + ylab("") + xlab("Sample") +
      scale_y_continuous(limits = c(0,2300), breaks = c(0, 1000,2000)) +
      theme(legend.title=element_blank(),legend.position="none")  
      # geom_label(data = subset(taxoverview, variable == "Reads"), aes(label = value))
    plottaxoverview2
   
      plottaxoverview <- ggplot(subset(taxoverview, variable == "Reads") ,aes(x=Sample,y=as.numeric(value))) +
      geom_bar(aes(fill=factor(variable)),stat='identity',position="identity") +
      facet_grid(.~variable, 
                 scales="free",
                 drop=TRUE) + 
      coord_flip()
    plottaxoverview <- papertheme(plottaxoverview)
    plottaxoverview <- plottaxoverview + scale_fill_manual(values=sevencolors[[7]])+
      theme(strip.background =element_rect(fill="#D6E0F4")) 
    plottaxoverview3 <- plottaxoverview + ylab("") + xlab("") + theme(legend.title=element_blank(),legend.position="none", axis.title.y = element_blank(), axis.ticks.y = element_blank(), axis.text.y = element_blank())  +
      scale_y_continuous(limits = c(-1000, 56000), breaks = c(0, 25000, 50000)) +
      geom_label(data = subset(taxoverview, variable == "Reads"), aes(label = value), size = 4)
    plottaxoverview3
    
  
    
        plottaxoverview <- ggplot(subset(taxoverview, variable != "Reads" &variable != "OTU" ) ,aes(x=Sample,y=as.numeric(value))) +
      geom_bar(aes(fill=factor(variable)),stat='identity',position="identity") +
      facet_grid(.~variable, 
                 scales="free",
                 drop=TRUE) + 
      coord_flip()
    plottaxoverview <- papertheme(plottaxoverview)
    plottaxoverview <- plottaxoverview + scale_fill_manual(values=c("#FC8D62", "#8DA0CB", "#E78AC3", "#A6D854", "#FFD92F"))+
      theme(strip.background =element_rect(fill="#D6E0F4")) 
    plottaxoverview4 <- plottaxoverview + ylab("Number of unique taxa") + xlab("Sample") +
      
      theme(legend.title=element_blank(),legend.position="none", axis.title.y = element_blank(), axis.ticks.y = element_blank(), axis.text.y = element_blank())  
      # geom_label(data = subset(taxoverview, variable == "Reads"), aes(label = value))
    plottaxoverview4
    
 combplot<-  ggarrange(plottaxoverview2, plottaxoverview4, plottaxoverview3, ncol = 3, widths = c(1.35,5,1))
    
   
    ggarrange(combplot, plot1b, labels = "AUTO" , ncol = 1, font.label = list(size = 20))

```

## All data
```{r QC pure data, fig.height=40, fig.width=15}
df.OTU_QC <- transpose(as.data.frame(otu_table(QC_mothur_data)))
colnames(df.OTU_QC) <- rownames(as.data.frame(otu_table(QC_mothur_data)))
rownames(df.OTU_QC) <- colnames(as.data.frame(otu_table(QC_mothur_data)))
df.Tax_QC <- as.data.frame(tax_table(QC_mothur_data))
# Quality Control samples 
  # Taxa distribution
    readnr <- rowSums(df.OTU_QC)
    OTUnr <- rowSums(df.OTU_QC!=0) 
    genusnr <- colSums(tax_summarize(df.OTU_QC,df.Tax_QC,"Genus")!=0)
    familynr <- colSums(tax_summarize(df.OTU_QC,df.Tax_QC,"Family")!=0)
    ordernr <- colSums(tax_summarize(df.OTU_QC,df.Tax_QC,"Order")!=0)
    classnr <- colSums(tax_summarize(df.OTU_QC,df.Tax_QC,"Class")!=0)
    phylumnr <- colSums(tax_summarize(df.OTU_QC,df.Tax_QC,"Phylum")!=0)
    taxoverview_QC <- data.frame(Sample=rownames(df.OTU_QC),OTU=OTUnr,Genus=genusnr,Family=familynr,Order=ordernr,Class=classnr,Phylum=phylumnr,Reads=readnr)
    taxoverview <- reshape2::melt(taxoverview_QC,id.vars=c("Sample"))
    
    plottaxoverview <- ggplot(taxoverview,aes(x=Sample,y=as.numeric(value))) +
      geom_bar(aes(fill=factor(variable)),stat='identity',position="identity") +
      facet_grid(.~variable, 
                 # scales="free", 
                 drop=TRUE) + coord_flip()
    plottaxoverview <- papertheme(plottaxoverview)
    plottaxoverview <- plottaxoverview + scale_fill_manual(values=sevencolors)
    plottaxoverview <- plottaxoverview + ylab("Number of unique taxa") + xlab("Sample") + theme(legend.title=element_blank(),legend.position="none")
    plottaxoverview
    
  # Composition at genus level
    genustable_QC_bar <- decostand(tax_summarize(df.OTU_QC,df.Tax_QC,"Genus"),MARGIN=2,"total")*100
    selecttop20par <- rowSums(genustable_QC_bar)
    genustable_QC_bar <- rownames_to_column(genustable_QC_bar)
    genustable_QC_bar <- as.data.frame(genustable_QC_bar %>% top_n(29,selecttop20par))
    other <- 100-colSums(genustable_QC_bar[,-1])
    rownames(genustable_QC_bar) <- genustable_QC_bar$rowname
    genustable_QC_bar <- genustable_QC_bar[,-1]
    genustable_QC_bar <- rbind(genustable_QC_bar,other)
    rownames(genustable_QC_bar)[nrow(genustable_QC_bar)] <- "Other"
    
    genustable_QC_bar <- as.data.frame(t(genustable_QC_bar))
    genustable_QC_bar <- cbind(genustable_QC_bar,rownames=rownames(genustable_QC_bar))
    genustable_QC_bar <- melt(genustable_QC_bar,id.vars=c('rownames'))
    p <- ggplot(data=genustable_QC_bar,aes(x=rownames,y=as.numeric(as.character(value))))
    p <- p + geom_bar(stat="identity",aes(fill=variable)) 
    p <- p + theme(axis.text.x = element_text(angle = 0, hjust = 1))  
      scale_fill_manual(values=fortycolors,name="genus")
    p <- p + coord_flip() + ylab("Relative abundance (%)") + xlab(NULL) + theme(legend.text=element_text(face='italic')) + guides(fill=guide_legend(ncol=2)) + theme(legend.justification = "center")
    p
    
    
# Samples 
df.OTU <- transpose(as.data.frame(otu_table(moth_merge)))
colnames(df.OTU) <- rownames(as.data.frame(otu_table(moth_merge)))
rownames(df.OTU) <- colnames(as.data.frame(otu_table(moth_merge)))
df.Tax_QC <- as.data.frame(tax_table(moth_merge))
  # Taxa distribution
    readnr <- rowSums(df.OTU)
    OTUnr <- rowSums(df.OTU!=0)
    genusnr <- colSums(tax_summarize(df.OTU,df.Tax_QC,"Genus")!=0)
    familynr <- colSums(tax_summarize(df.OTU,df.Tax_QC,"Family")!=0)
    ordernr <- colSums(tax_summarize(df.OTU,df.Tax_QC,"Order")!=0)
    classnr <- colSums(tax_summarize(df.OTU,df.Tax_QC,"Class")!=0)
    phylumnr <- colSums(tax_summarize(df.OTU,df.Tax_QC,"Phylum")!=0)
    
    taxoverview_QC <- data.frame(Sample=rownames(df.OTU),OTU=OTUnr,Genus=genusnr,Family=familynr,Order=ordernr,Class=classnr,Phylum=phylumnr,Reads=readnr)
    taxoverview <- melt(taxoverview_QC,id.vars=c("Sample"))

    
    plottaxoverview <- ggplot(taxoverview,aes(x=Sample,y=as.numeric(value))) 
    plottaxoverview <- papertheme(plottaxoverview)+ geom_bar(aes(fill=factor(variable)),stat='identity',position="identity") + facet_grid(.~variable,scales="free",drop=TRUE) + coord_flip()
    plottaxoverview <- plottaxoverview + scale_fill_manual(values=sevencolors)
    plottaxoverview <- plottaxoverview + ylab("Number of unique taxa") + xlab("Sample") + theme(legend.title=element_blank(),legend.position="none", strip.background =element_rect(fill="#D6E0F4"), 
    strip.text.x = element_text(size = 20), strip.text.y = element_text(size = 20))
    plottaxoverview
    
saveplots()
ggsave(plot = plottaxoverview, width = 15, height = 40, dpi = 150, filename = "QC_taxadistribution.png")
# Remove sample 175 (no reads)
# df.OTU <- transpose(as.data.frame(otu_table(subset_samples(moth_merge, sample !="X175"))))
# colnames(df.OTU) <- rownames(as.data.frame(otu_table(subset_samples(moth_merge, sample !="X175"))))
# rownames(df.OTU) <- colnames(as.data.frame(otu_table(subset_samples(moth_merge, sample !="X175"))))
# df.Tax_QC <- as.data.frame(tax_table(subset_samples(moth_merge, sample !="X175")))
 
 # Rarefaction
    set.seed=711
    
    datarare <-t(df.OTU)
    # Find minimal total number of reads of all samples = mimimal number of 'species'
    raremax <- min(colSums(datarare))
    # Draw rarefaction curves
    rarecurve(round(t(datarare)), step = 20, sample = raremax, col = c("blue"), cex = 0.6)
  
        # datarare <-t(df.OTU[50:60,])
```

```{r QC removed singleton data, fig.height=25, fig.width=10}
# QC_mothur_data_NS = filter_taxa(QC_mothur_data, function(x) sum(x > 1) > (0.05*length(x)), TRUE)
QC_mothur_data_NS <- prune_taxa(taxa_sums(QC_mothur_data) > 1, QC_mothur_data)
df.OTU_QC_NS <- transpose(as.data.frame(otu_table(QC_mothur_data_NS)))
colnames(df.OTU_QC_NS) <- rownames(as.data.frame(otu_table(QC_mothur_data_NS)))
rownames(df.OTU_QC_NS) <- colnames(as.data.frame(otu_table(QC_mothur_data_NS)))
df.Tax_QC <- as.data.frame(tax_table(QC_mothur_data_NS))
# df.OTU_QC_NS <- transpose(df.OTU_QC_NS); df.Tax_QC <- transpose(df.Tax_QC)
# Quality Control samples 
  # Taxa distribution
    readnr <- rowSums(df.OTU_QC_NS)
    OTUnr <- rowSums(df.OTU_QC_NS!=0)
    genusnr <- colSums(tax_summarize(df.OTU_QC_NS,df.Tax_QC,"Genus")!=0)
    familynr <- colSums(tax_summarize(df.OTU_QC_NS,df.Tax_QC,"Family")!=0)
    ordernr <- colSums(tax_summarize(df.OTU_QC_NS,df.Tax_QC,"Order")!=0)
    classnr <- colSums(tax_summarize(df.OTU_QC_NS,df.Tax_QC,"Class")!=0)
    phylumnr <- colSums(tax_summarize(df.OTU_QC_NS,df.Tax_QC,"Phylum")!=0)
    
    taxoverview_QC <- data.frame(Sample=rownames(df.OTU_QC_NS),OTU=OTUnr,Genus=genusnr,
                                 Family=familynr,Order=ordernr,Class=classnr,Phylum=phylumnr,Reads=readnr)
    taxoverview <- reshape2::melt(taxoverview_QC,id.vars=c("Sample"))
    
    plottaxoverview <- ggplot(taxoverview,aes(x=Sample,y=as.numeric(value))) +
      geom_bar(aes(fill=factor(variable)),stat='identity',position="identity") + 
      facet_grid(.~variable,scales="free",drop=TRUE) + coord_flip()
    plottaxoverview <- plottaxoverview + scale_fill_manual(values=sevencolors)
    plottaxoverview <- plottaxoverview + ylab("Number of unique taxa") + xlab("Sample") +
      theme(legend.title=element_blank(),legend.position="none")
    plottaxoverview
    
  # Composition at genus level
    genustable_QC_bar <- decostand(tax_summarize(df.OTU_QC_NS,df.Tax_QC,"Genus"),MARGIN=2,"total")*100
    selecttop20par <- rowSums(genustable_QC_bar)
    genustable_QC_bar <- rownames_to_column(genustable_QC_bar)
    genustable_QC_bar <- as.data.frame(genustable_QC_bar %>% top_n(29,selecttop20par))
    other <- 100-colSums(genustable_QC_bar[,-1])
    rownames(genustable_QC_bar) <- genustable_QC_bar$rowname
    genustable_QC_bar <- genustable_QC_bar[,-1]
    genustable_QC_bar <- rbind(genustable_QC_bar,other)
    rownames(genustable_QC_bar)[nrow(genustable_QC_bar)] <- "Other"
    
    genustable_QC_bar <- as.data.frame(t(genustable_QC_bar))
    genustable_QC_bar <- cbind(genustable_QC_bar,rownames=rownames(genustable_QC_bar))
    genustable_QC_bar <- melt(genustable_QC_bar,id.vars=c('rownames'))
    p <- ggplot(data=genustable_QC_bar,aes(x=rownames,y=as.numeric(as.character(value))))
    p <- p + geom_bar(stat="identity",aes(fill=variable)) 
    p <- p + theme(axis.text.x = element_text(angle = 0, hjust = 1)) + scale_fill_manual(values=fortycolors,name="genus")
    p <- p + coord_flip() + ylab("Relative abundance (%)") + xlab(NULL) + theme(legend.text=element_text(face='italic')) + guides(fill=guide_legend(ncol=2)) + theme(legend.justification = "center")
    p
# Samples 
summarize_phyloseq(moth_merge)
moth_merge_NS <- prune_taxa(taxa_sums(moth_merge) > 1, moth_merge) 
summarize_phyloseq(moth_merge_NS)
df.OTU_NS <- transpose(as.data.frame(otu_table(moth_merge_NS)))
colnames(df.OTU_NS) <- rownames(as.data.frame(otu_table(moth_merge_NS)))
rownames(df.OTU_NS) <- colnames(as.data.frame(otu_table(moth_merge_NS)))
df.Tax_NS <- as.data.frame(tax_table(moth_merge_NS))
df.OTU_NS <- df.OTU_NS[ order(as.numeric(row.names(df.OTU_NS))), ]
  # Taxa distribution
    readnr <- rowSums(df.OTU_NS)
    OTUnr <- rowSums(df.OTU_NS!=0)
    genusnr <- colSums(tax_summarize(df.OTU_NS,df.Tax_NS,"Genus")!=0)
    familynr <- colSums(tax_summarize(df.OTU_NS,df.Tax_NS,"Family")!=0)
    ordernr <- colSums(tax_summarize(df.OTU_NS,df.Tax_NS,"Order")!=0)
    classnr <- colSums(tax_summarize(df.OTU_NS,df.Tax_NS,"Class")!=0)
    phylumnr <- colSums(tax_summarize(df.OTU_NS,df.Tax_NS,"Phylum")!=0)
    
    taxoverview_QC <- data.frame(Sample=rownames(df.OTU_NS),
                                 OTU=OTUnr, Genus=genusnr, Family=familynr, 
                                 Order=ordernr, Class=classnr, Phylum=phylumnr, Reads=readnr)
    taxoverview <- melt(taxoverview_QC,id.vars=c("Sample"))
    
    plottaxoverview <- ggplot(taxoverview,aes(x=Sample,y=as.numeric(value))) +
      geom_bar(aes(fill=factor(variable)),stat='identity',position="identity") +
    facet_grid(.~variable,scales="free",drop=TRUE) + coord_flip()
    plottaxoverview <- plottaxoverview + scale_fill_manual(values=sevencolors)
    plottaxoverview <- plottaxoverview + ylab("Number of unique taxa") + xlab("Sample") +
    theme(legend.title=element_blank(),legend.position="none")
    plottaxoverview <- papertheme(plottaxoverview, size = 15)
plottaxoverview <- plottaxoverview + theme(strip.background = element_rect(fill="#D6E0F4"),
                                           strip.text.x = element_text(size = 15), 
                                           legend.position = "none")
    plottaxoverview
    saveplots()
ggsave(plot = plottaxoverview, width = 10, height = 25, dpi = 150, filename = "QC_taxadistribution_dataset.png")
# Remove sample 175 (1 read)
# df.OTU_NS <- transpose(as.data.frame(otu_table(subset_samples(moth_merge_NS, sample !="X175"))))
# colnames(df.OTU_NS) <- rownames(as.data.frame(otu_table(subset_samples(moth_merge_NS, sample !="X175"))))
# rownames(df.OTU_NS) <- colnames(as.data.frame(otu_table(subset_samples(moth_merge_NS, sample !="X175"))))
# df.Tax_NS <- as.data.frame(tax_table(subset_samples(moth_merge_NS, sample !="X175")))
  # Rarefaction
    set.seed=711
    
    datarare <-t(df.OTU_NS)
    
    # Find minimal total number of reads of all samples = mimimal number of 'species'
    raremax <- min(colSums(datarare))
    # Draw rarefaction curves
    rarecurve(round(t(datarare)), step = 20, sample = raremax, col = c("blue"), cex = 0.6)
    
        # datarare <-t(df.OTU[50:60,])
```

```{r QC Waste Not Want Not, fig.height=5, fig.width=12}
# Samples (sample 175 remove, no reads)
# df.OTU_NS <- as.data.frame(otu_table(subset_samples(moth_merge_NS, sample !="X175")))
# df.Tax_NS <- as.data.frame(tax_table(subset_samples(moth_merge_NS, sample !="X175")))
df.OTU_NS <- as.data.frame(otu_table(moth_merge_NS))
df.Tax_NS <- as.data.frame(tax_table(moth_merge_NS))

# Apply 'arbitrary cut-offs' from the waste not want not paper
minobs = 1
prevalence <- apply(as.matrix(df.OTU_NS), 1 , 
                    function (x, minobs){sum(x>=minobs)}, minobs)/ncol(df.OTU_NS)
prevalencefilter <- prevalence > 0.05
df.OTUwnwn <- df.OTU_NS[prevalencefilter,]
df.OTU_NSwnwn <- df.OTUwnwn[rowSums(df.OTUwnwn)>0.5*ncol(df.OTUwnwn),]

# Transpose dataframe
df.OTU_NS_wnwn <- transpose(df.OTU_NSwnwn)
df.OTU_NS_wnwn[] <- lapply(df.OTU_NS_wnwn, as.numeric)
colnames(df.OTU_NS_wnwn) <- rownames(df.OTU_NSwnwn)
rownames(df.OTU_NS_wnwn) <- colnames(df.OTU_NSwnwn)
moth_merge2 <- moth_merge_NS
otu_table(moth_merge2) <- otu_table(as.matrix(df.OTU_NSwnwn), taxa_are_rows = TRUE)
rm(df.OTU_NSwnwn)

# Taxa distribution
readnr <- rowSums(df.OTU_NS_wnwn)
OTUnr <- rowSums(df.OTU_NS_wnwn!=0)
genusnr <- colSums(tax_summarize(df.OTU_NS_wnwn,df.Tax_NS,"Genus")!=0)
familynr <- colSums(tax_summarize(df.OTU_NS_wnwn,df.Tax_NS,"Family")!=0)
ordernr <- colSums(tax_summarize(df.OTU_NS_wnwn,df.Tax_NS,"Order")!=0)
classnr <- colSums(tax_summarize(df.OTU_NS_wnwn,df.Tax_NS,"Class")!=0)
phylumnr <- colSums(tax_summarize(df.OTU_NS_wnwn,df.Tax_NS,"Phylum")!=0)
    
taxoverview_QC <- data.frame(Sample=rownames(df.OTU_NS_wnwn),
                                 OTU=OTUnr, Genus=genusnr, Family=familynr, 
                                 Order=ordernr, Class=classnr, Phylum=phylumnr, Reads=readnr)
taxoverview <- melt(taxoverview_QC,id.vars=c("Sample"))
    
plottaxoverview <- ggplot(taxoverview,aes(x=Sample,y=as.numeric(value))) + geom_bar(aes(fill=factor(variable)),stat='identity',position="identity") + facet_grid(.~variable,scales="free",drop=TRUE) + coord_flip()
# plottaxoverview <- plottaxoverview + scale_fill_manual(values=sevencolors)
plottaxoverview <- plottaxoverview + ylab("Number of unique taxa") + xlab("Sample") + theme(legend.title=element_blank(),legend.position="none")
plottaxoverview <- papertheme(plottaxoverview, size = 15)
plottaxoverview <- plottaxoverview + theme(legend.position = "none", 
                                           strip.background = element_rect(fill="#D6E0F4"),
                                           strip.text.x = element_text(size = 15))
plottaxoverview
# saveplots()
# ggsave(plot = plottaxoverview, width = 10, height = 25, dpi = 150, filename = "QC_taxadistribution_datasetwnwn.png")
  # Rarefaction
    set.seed=711
    datarare <-t(df.OTU_NS_wnwn)
    # datarare <- df.OTU_NS_wnwn
    # Find minimal total number of reads of all samples = mimimal number of 'species'
    raremax <- min(colSums(datarare))
    # Draw rarefaction curves
  x <- round(t(datarare))
rarecurve(round(t(datarare)), step = 20, 
          # sample = raremax, 
          col = c("blue"), cex = 0.6)
```
